include: "rules/common.smk"
configfile: "config/config.yaml"

include: "rules/common.py"
include: "rules/DownloadAndPreprocess.smk"
include: "rules/Alignment.smk"
include: "rules/MakeBigwigs.smk"
# include: "rules/CutAndTagChecks.smk"
include: "rules/QC.smk"
include: "rules/ExpressionAnalysis.smk"
include: "rules/featureCounts.smk"
include: "rules/PeakCalling.smk"
include: "rules/SplicingAnalysis.smk"
include: "rules/ChromatinPehnotypesAnalysis.smk"
include: "rules/ProSeqAnalysis.smk"
include: "rules/QTLTools.smk"
include: "rules/Coloc.smk"
include: "rules/report.smk"
include: "rules/BensJupyterNotebookRules.smk"

## Inserted by Carlos
include: "rules/chRNASeqSlopes.smk"
#include: "rules/slopeQTLs.smk"

wildcard_constraints:
    chrom="|".join(autosomes),
    IndID="|".join(Fastq_samples['IndID']),
    Rep="\d+",
    QTLsGenotypeSet="|".join(['', 'HarmonizedSNPs'])

localrules: all, CopyFastqFromLocal, DownloadFastqFromLink, Download1KG_GRCh38, STAR_make_index, DownloadHg38Ref, DownloadFastqFromLink_SE
ruleorder: CopyFastqFromLocal > DownloadFastqFromLink


rule all:
    input:
        # expand("logs/PlotGruberQTLs/PlotFromShellScript.{pheno}.log", pheno=["H3K4ME3", "H3K27AC", "H3K4ME1"]),
        # "QTLs/sQTLs/permutation_pass_chunks/1.1000.txt.gz",
        # expand("FastqFastp/{Phenotype}/{IndID}/{Rep}.R2.fastq.gz", zip, Phenotype=Fastq_samples['Phenotype'], IndID=Fastq_samples['IndID'], Rep=Fastq_samples['RepNumber']),
        # expand("Alignments/Hisat2_Align/{Phenotype}/{IndID}.{Rep}.wasp_filterd.markdup.sorted.bam", zip, Phenotype= ChromatinProfilingSamples_df['Phenotype'], IndID=ChromatinProfilingSamples_df['IndID'], Rep=ChromatinProfilingSamples_df['RepNumber']),
        # expand("Alignments/STAR_Align/{Phenotype}/{IndID}/{Rep}/Filtered.bam", zip, Phenotype=RNASeqSamples_df['Phenotype'], IndID=RNASeqSamples_df['IndID'], Rep=RNASeqSamples_df['RepNumber']),
        # expand("bigwigs/{Phenotype}/{IndID}.{Rep}.bw", zip, Phenotype=Fastq_samples['Phenotype'], IndID=Fastq_samples['IndID'], Rep=Fastq_samples['RepNumber']),
        expand("QC/mbv/plots/{Phenotype}/{IndID}.{Rep}.pdf", zip, Phenotype=chRNASeqSamples_df['Phenotype'], IndID=chRNASeqSamples_df['IndID'], Rep=chRNASeqSamples_df['RepNumber']),
        #Make bigwigs from test cut and tag
        # "../output/QC/AutosomeCountsPerSamples.tsv",
        # expand("bigwigs_FromNonWASPFilteredReads/{Phenotype}/{IndID}.{Rep}.bw", Phenotype=["POL2S5", "POL2S2", "H3K9ME3", "H3K36ME3", "H3K79ME2"], IndID="NA19210", Rep="1"),
        multiext("ReferenceGenome/Annotations/gencode.v34.primary_assembly.annotation.gtf", "_all_exons.txt.gz", "_all_introns.bed.gz", "_fiveprime.bed.gz", "_threeprime.bed.gz"),
        expand("SplicingAnalysis/IR/{Phenotype}/Counts.txt", Phenotype=RNASeqPhenotypes),
        # expand("PeakCalling/{Phenotype}.peaks.saf", Phenotype=ChromatinProfilingPhenotypes)
        # expand("SplicingAnalysis/leafcutter/clustering/chr{chrom}/leafcutter_perind.counts.gz", chrom=autosomes),
        "SplicingAnalysis/leafcutter/clustering/autosomes/leafcutter_perind.counts.gz",
        #expand("QTLs/QTLTools/{Phenotype}/OnlyFirstReps.sorted.qqnorm.bed.pca", Phenotype=["chRNA.IR", "Expression.Splicing", "H3K4ME1", "H3K27AC", "CTCF", "H3K4ME3"]),
        #"QTLs/QTLTools/Expression.Splicing/Genotypes/WholeGenome.vcf.gz",
        # "Multiqc/multiqc_report.html",
        #expand("QTLs/QTLTools/{Phenotype}/PermutationPass.FDR_Added.txt.gz", Phenotype=MyPhenotypes),
        expand("featureCounts/{Phenotype}/NormFactors.tsv", Phenotype=["chRNA.Expression.Splicing", "Expression.Splicing", "H3K4ME1", "H3K4ME3", "H3K27AC", "CTCF"]),
        "hyprcoloc/GenewiseSummaryStatsInput",
        "hyprcoloc/GenewiseSummaryStatsInputHarmonizedSNPs",
        "ReferenceGenome/Annotations/GTFTools_BasicAnnotations/gencode.v34.chromasomal.tss.bed",
        expand("featureCounts/AtTSS/{Phenotype}/Counts.txt", Phenotype=["H3K27AC", "H3K4ME3"]),
        expand("featureCounts/{Phenotype}/Counts.txt", Phenotype=["MetabolicLabelled.30min", "MetabolicLabelled.60min"]),
        #expand("QTLs/QTLTools/{Phenotype}/PermutationPass{QTLsGenotypeSet}ForColoc.txt.gz", Phenotype=PhenotypesToColoc, QTLsGenotypeSet=['', 'HarmonizedSNPs']),
        #### Inserted by Carlos
        expand(
            "IntronWindowCounts/{sample}.{windowStyle}.bed.gz",
            sample=NaRNASeq.index,
            windowStyle=["IntronWindows_equalLength", "IntronWindows"],
        ),
        expand(
            "slopes/{sample}.tab.gz",
            sample=NaRNASeq.index,
        ),

rule Gather_chRNASeq_Fastq:
    """
    Useful to force rexecution of all chRNAseq related analysis after
    identifying sample swaps and correcting them in samples.tsv config
    """
    input:
        expand("Fastq/{Phenotype}/{IndID}/{Rep}.R1.fastq.gz", zip, Phenotype=chRNASeqSamples_df['Phenotype'], IndID=chRNASeqSamples_df['IndID'], Rep=chRNASeqSamples_df['RepNumber']),

