<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>20200110_PickSamplesForGrowth</title>

<script src="site_libs/header-attrs-2.7/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">20200110_PickSamplesForGrowth</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-12-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>ChromatinSplicingQTLs/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstree89517ffeaecf50fe2639320bb002b1a5ce1b1858targetblank89517ffa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/89517ffeaecf50fe2639320bb002b1a5ce1b1858" target="_blank">89517ff</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstree89517ffeaecf50fe2639320bb002b1a5ce1b1858targetblank89517ffa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/89517ffeaecf50fe2639320bb002b1a5ce1b1858" target="_blank">89517ff</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    ._.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store
    Ignored:    code/.Rhistory
    Ignored:    code/._.DS_Store
    Ignored:    code/._README.md
    Ignored:    code/._dag.pdf
    Ignored:    code/._report.html
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    code/.snakemake/
    Ignored:    code/Alignments/
    Ignored:    code/ENCODE/
    Ignored:    code/ExpressionAnalysis/
    Ignored:    code/Fastq/
    Ignored:    code/FastqFastp/
    Ignored:    code/FastqFastpSE/
    Ignored:    code/FastqSE/
    Ignored:    code/Gather_gwas_summary_stats/
    Ignored:    code/Genotypes/
    Ignored:    code/Multiqc/
    Ignored:    code/Multiqc_chRNA/
    Ignored:    code/PeakCalling/
    Ignored:    code/Phenotypes/
    Ignored:    code/PlotGruberQTLs/
    Ignored:    code/ProCapAnalysis/
    Ignored:    code/QC/
    Ignored:    code/QTLs/
    Ignored:    code/ReferenceGenome/
    Ignored:    code/Session.vim
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TODO
    Ignored:    code/bigwigs/
    Ignored:    code/bigwigs_FromNonWASPFilteredReads/
    Ignored:    code/config/.DS_Store
    Ignored:    code/config/._.DS_Store
    Ignored:    code/config/ExternalFastqDataAccessions/
    Ignored:    code/config/OldSamplesConfig/
    Ignored:    code/dag.pdf
    Ignored:    code/featureCounts/
    Ignored:    code/gwas_summary_stats/
    Ignored:    code/hyprcoloc/
    Ignored:    code/logs/
    Ignored:    code/notebooks/.ipynb_checkpoints/
    Ignored:    code/out.hap.ld
    Ignored:    code/out.log
    Ignored:    code/report.html
    Ignored:    code/rules/OldRules/
    Ignored:    code/scratch/
    Ignored:    code/scripts/GTFtools_0.8.0/
    Ignored:    code/scripts/__pycache__/
    Ignored:    code/scripts/liftOverBedpe/liftOverBedpe.py
    Ignored:    code/snakemake.log
    Ignored:    code/snakemake.sbatch.log
    Ignored:    data/._PRJEB1350_RunTable.Ding_etal_CTCF.txt
    Ignored:    data/._igsr_samples.tsv
    Ignored:    data/._list_gwas_summary_statistics_PMID27863252.csv
    Ignored:    data/GrowthNotes/._20210830_GrowthNotes_chRNA.ConcentrationsToReplate.txt

Untracked files:
    Untracked:  code/snakemake_profiles/slurm/__pycache__/

Unstaged changes:
    Modified:   code/scripts/GenometracksByGenotype
    Modified:   output/hyprcoloc_results/ForColoc/hyprcoloc.results.txt.gz

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/20200110_PickSamplesForGrowth.Rmd</code>) and HTML (<code>docs/20200110_PickSamplesForGrowth.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/304e9b71d38372d012263365ae4c32724ecccddf/analysis/20200110_PickSamplesForGrowth.Rmd" target="_blank">304e9b7</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2021-02-11
</td>
<td>
added data about picking samples
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(readxl)
library(knitr)

LCLs.1000Genomes.vec &lt;- read.table(&quot;../data/GiladStrains/YRI.list.from1000Genomes.txt&quot;, sep=&#39;\t&#39;) %&gt;% pull(V1) %&gt;%
  gsub(&quot;^NA&quot;, &quot;&quot;, ., perl=T)

LCLs.GiladFreezer &lt;- read.table(&quot;../data/GiladStrains/Human_LCLs.txt&quot;, sep = &#39;\t&#39;, header = T, stringsAsFactors = F)

LCLs.GiladFreezer.LCL.YRI.vec &lt;- read.table(&quot;../data/GiladStrains/LCL_YRI.GiladCryoSheetTab.list.txt&quot;, stringsAsFactors = F) %&gt;% pull(V1) %&gt;% as.character()

GrowthNotes &lt;- read_delim(&quot;../data/GrowthNotes/GrowthNotes.tsv&quot;, delim = &#39;\t&#39;)

SamplesToRestock.vec &lt;- read.table(&quot;../data/GiladStrains/Human_LCLs.ToRestock.txt&quot;, stringsAsFactors = F) %&gt;% pull(V1) %&gt;% as.character()</code></pre>
<p>Ok now figure out which lines to pull</p>
<pre class="r"><code>AlreadyDone.vec &lt;- GrowthNotes %&gt;%
  filter(!is.na(`Raw Count1 (x10^6)`)) %&gt;%
  pull(`Cell Line`) %&gt;% as.character()

#Lines that I can replenish from Briana&#39;s stocks
LinesToPutBack.vec &lt;- intersect(SamplesToRestock.vec, AlreadyDone.vec)

LinesStillNeedToRestock.vec &lt;- setdiff(SamplesToRestock.vec, AlreadyDone.vec)

#Are there any YRI lines in Gilad lab freezer that aren&#39;t in YRI box
LCLs.GiladFreezer.MainTab.YRI.vec &lt;-  LCLs.GiladFreezer %&gt;%
  filter(Human.LCL.ID %in% LCLs.1000Genomes.vec) %&gt;%
  pull(Human.LCL.ID) %&gt;% as.character()
setdiff(LCLs.GiladFreezer.MainTab.YRI.vec, LCLs.GiladFreezer.LCL.YRI.vec)</code></pre>
<pre><code> [1] &quot;18489&quot; &quot;18498&quot; &quot;18501&quot; &quot;18503&quot; &quot;18504&quot; &quot;18509&quot; &quot;18511&quot; &quot;18516&quot; &quot;18518&quot;
[10] &quot;18519&quot; &quot;18521&quot; &quot;18522&quot; &quot;18852&quot; &quot;18855&quot; &quot;18856&quot; &quot;18857&quot; &quot;18858&quot; &quot;18859&quot;
[19] &quot;18860&quot; &quot;18861&quot; &quot;18862&quot; &quot;18871&quot; &quot;18872&quot; &quot;18907&quot; &quot;18909&quot; &quot;18912&quot; &quot;18913&quot;
[28] &quot;18914&quot; &quot;18916&quot; &quot;19092&quot; &quot;19093&quot; &quot;19099&quot; &quot;19100&quot; &quot;19102&quot; &quot;19103&quot; &quot;19114&quot;
[37] &quot;19116&quot; &quot;19119&quot; &quot;19120&quot; &quot;19127&quot; &quot;19129&quot; &quot;19130&quot; &quot;19131&quot; &quot;19132&quot; &quot;19137&quot;
[46] &quot;19138&quot; &quot;19139&quot; &quot;19140&quot; &quot;19142&quot; &quot;19143&quot; &quot;19144&quot; &quot;19145&quot; &quot;19152&quot; &quot;19153&quot;
[55] &quot;19159&quot; &quot;19161&quot; &quot;19171&quot; &quot;19173&quot; &quot;19190&quot; &quot;19201&quot; &quot;19202&quot; &quot;19203&quot; &quot;19204&quot;
[64] &quot;19205&quot; &quot;19206&quot; &quot;19210&quot; &quot;19211&quot; &quot;19222&quot; &quot;18854&quot; &quot;19208&quot;</code></pre>
<pre class="r"><code>#And the reverse?
setdiff(LCLs.GiladFreezer.LCL.YRI.vec, LCLs.GiladFreezer.MainTab.YRI.vec)</code></pre>
<pre><code>[1] &quot;19193&quot; &quot;19128&quot; &quot;18520&quot;</code></pre>
<pre class="r"><code>BlacklistFromYang &lt;- c(&quot;18502&quot;, &quot;18516&quot;, &quot;18916&quot;, &quot;19012&quot;)
BlacklistFromBriana &lt;- c(&quot;18517&quot;, &quot;19128&quot;)</code></pre>
<p>Let’s include those reverse matches in this experiment. Let’s come up with a list of “StillToDo” strains based on the following points</p>
<ol style="list-style-type: decimal">
<li>Not already done</li>
<li>Not in Yang or Briana’s blacklist</li>
<li>Pull from the LinesStillNeedToRestock.vec first, and then from other lines in LCLs.GiladFreezer and not in LCLs.GiladFreezer.LCL.YRI.vec</li>
<li>Should be in 1000Genome’s project (some samples are not, though they are clearly YRI from Coriell)</li>
</ol>
<pre class="r"><code>#Start with set of LCLs in GiladFreezer Main tab
StillToDo.vec &lt;- LCLs.GiladFreezer.MainTab.YRI.vec %&gt;%
  #Add list of LCLs in GiladFreezer YRI tab
  union(LCLs.GiladFreezer.MainTab.YRI.vec) %&gt;%
  #intersect with 1000Genome&#39;s YRI list
  intersect(LCLs.1000Genomes.vec) %&gt;%
  #Subtract out the blacklist strains and the ones we&#39;ve already done
  setdiff(BlacklistFromYang) %&gt;%
  setdiff(BlacklistFromBriana) %&gt;%
  setdiff(AlreadyDone.vec) %&gt;% sort()
length(StillToDo.vec)</code></pre>
<pre><code>[1] 65</code></pre>
<p>From that list, let’s prioritize a list of 30 strains for this experiment by</p>
<ol style="list-style-type: decimal">
<li>Including strains I still have to restock because I took a vial and didn’t replenish it from 2019</li>
<li>The two lines that we need to redo because of swapped pellet</li>
<li>Prioritize strains have more than 2 vials in the freezer</li>
<li>Things that more published data already</li>
</ol>
<pre class="r"><code>NumVials &lt;- read_delim(&quot;../data/GiladStrains/Human_LCLs.NumberOfVials.counts.txt&quot;, delim = &#39;\t&#39;, col_names = c(&quot;Strain&quot;, &quot;Count&quot;))


LinesStillNeedToRestock.vec</code></pre>
<pre><code>[1] &quot;19209&quot; &quot;18870&quot; &quot;19222&quot; &quot;19143&quot; &quot;18510&quot; &quot;18499&quot; &quot;18498&quot;</code></pre>
<pre class="r"><code>LinesToRedoFromBriannasStocks.vec &lt;- c(&quot;18861&quot;, &quot;19140&quot;)

Points1And2 &lt;- union(LinesStillNeedToRestock.vec,LinesToRedoFromBriannasStocks.vec)

#Points 3 and 4
LinesWithMoreThan2Vials.vec &lt;- NumVials %&gt;%
  filter(Strain %in% StillToDo.vec) %&gt;%
  filter(Count &gt; 2) %&gt;%
  pull(Strain) %&gt;%
  setdiff(Points1And2)
LinesWithMoreThan2Vials.vec</code></pre>
<pre><code> [1] &quot;18486&quot; &quot;18497&quot; &quot;18500&quot; &quot;18501&quot; &quot;18503&quot; &quot;18504&quot; &quot;18505&quot; &quot;18508&quot; &quot;18509&quot;
[10] &quot;18515&quot; &quot;18518&quot; &quot;18521&quot; &quot;18522&quot; &quot;18852&quot; &quot;18853&quot; &quot;18854&quot; &quot;18855&quot; &quot;18857&quot;
[19] &quot;18859&quot; &quot;18860&quot; &quot;18862&quot; &quot;18871&quot; &quot;18872&quot; &quot;18914&quot; &quot;19100&quot; &quot;19102&quot; &quot;19103&quot;
[28] &quot;19114&quot; &quot;19116&quot; &quot;19120&quot; &quot;19127&quot; &quot;19129&quot; &quot;19132&quot; &quot;19139&quot; &quot;19141&quot; &quot;19142&quot;
[37] &quot;19144&quot; &quot;19145&quot; &quot;19147&quot; &quot;19159&quot; &quot;19161&quot; &quot;19173&quot; &quot;19190&quot; &quot;19201&quot; &quot;19202&quot;
[46] &quot;19203&quot; &quot;19204&quot; &quot;19205&quot; &quot;19206&quot; &quot;19207&quot; &quot;19208&quot; &quot;19211&quot; &quot;19223&quot; &quot;19238&quot;
[55] &quot;19239&quot; &quot;19257&quot;</code></pre>
<pre class="r"><code>LCLs.PreviousMolecularDatasets &lt;- read_excel(&quot;../data/GiladStrains/individuals.data.types.xlsx&quot;, col_names =c(&quot;Line&quot;, &quot;NumberAssays&quot;, &quot;Assay1&quot;, &quot;Assay2&quot;, &quot;Assay3&quot;, &quot;Assay4&quot;, &quot;Assay5&quot;, &quot;Assay6&quot;, &quot;Assay7&quot;, &quot;Assay8&quot;))
head(LCLs.PreviousMolecularDatasets)</code></pre>
<pre><code># A tibble: 6 x 10
   Line NumberAssays Assay1 Assay2 Assay3   Assay4  Assay5  Assay6 Assay7 Assay8
  &lt;dbl&gt;        &lt;dbl&gt; &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt;    &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt; 
1 18499            8 RNAseq Decay  prot     H3K27ac methyl… DNase  ribo   4su   
2 18498            8 RNAseq Decay  prot     H3K27ac methyl… DNase  ribo   4su   
3 19159            7 RNAseq Decay  methyla… H3K27ac DNase   ribo   4su    &lt;NA&gt;  
4 19150            1 RNAseq &lt;NA&gt;   &lt;NA&gt;     &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;   &lt;NA&gt;   &lt;NA&gt;  
5 19153            7 RNAseq Decay  prot     H3K27ac DNase   ribo   4su    &lt;NA&gt;  
6 19152            7 RNAseq Decay  prot     H3K27ac DNase   ribo   4su    &lt;NA&gt;  </code></pre>
<pre class="r"><code>Points3And4 &lt;- LCLs.PreviousMolecularDatasets %&gt;%
  filter(Line %in% LinesWithMoreThan2Vials.vec) %&gt;%
  arrange(desc(NumberAssays)) %&gt;%
  slice_head(n=32-length(Points1And2)) %&gt;%
  pull(Line) %&gt;% as.character()

LinesToThaw &lt;- union(Points3And4, Points1And2)
length(LinesToThaw)</code></pre>
<pre><code>[1] 32</code></pre>
<p>Lastly, let’s obtain the exact location of these strains</p>
<pre class="r"><code>#Read in location of strains
GiladLabLinesLocation &lt;- read_delim(&quot;../data/GiladStrains/Large_Cryo_2019_Human_LCLs.tsv&quot;, delim = &#39;\t&#39;, comment = &#39;#&#39;)

GiladLabLinesLocation %&gt;%
  filter(Line %in% LinesToThaw) %&gt;%
  # filter(is.na(Notes1)) %&gt;%
  arrange(Box) %&gt;%
  distinct(Line, .keep_all = T) %&gt;%
  dplyr::select(Box, Position, Line, Notes1) %&gt;%
  write_tsv(&quot;../data/GiladStrains/20200111_CellsToWake.tsv&quot;)


GiladLabLinesLocation %&gt;%
  filter(Line %in% c(&quot;18870&quot;, &quot;18510&quot;))</code></pre>
<pre><code># A tibble: 12 x 11
   LineRowNumber LineColumnNumber LineColumnLetter LineReferenceString
           &lt;dbl&gt;            &lt;dbl&gt; &lt;chr&gt;            &lt;chr&gt;              
 1           149                2 B                Human LCLs&#39;!B149   
 2           149                3 C                Human LCLs&#39;!C149   
 3           149                4 D                Human LCLs&#39;!D149   
 4           149                5 E                Human LCLs&#39;!E149   
 5           149                6 F                Human LCLs&#39;!F149   
 6           149                7 G                Human LCLs&#39;!G149   
 7           149                8 H                Human LCLs&#39;!H149   
 8           149                9 I                Human LCLs&#39;!I149   
 9           149               10 J                Human LCLs&#39;!J149   
10           152                2 B                Human LCLs&#39;!B152   
11           152                3 C                Human LCLs&#39;!C152   
12           261                4 D                Human LCLs&#39;!D261   
# … with 7 more variables: Notes1ReferenceString &lt;chr&gt;,
#   Notes2ReferenceString &lt;chr&gt;, Box &lt;dbl&gt;, Position &lt;dbl&gt;, Line &lt;chr&gt;,
#   Notes1 &lt;chr&gt;, Notes2 &lt;lgl&gt;</code></pre>
<div id="update" class="section level2">
<h2>Update:</h2>
<p>As of 1/19, only 14/32 strains that we woke up are growing well. therefore, let’s thaw 13 more strains, bringing the total to 27, which, assuming no more than three drop out due to poor growth, will leave us with a batch of 24 for fractionation.</p>
<p>Let’s pick which ones to do by picking from the list of cell lines left to do, subtracting the 14 lines that are growing well, and then intersecting amongst the gilad lab YRI cryo rack strains which are probably stocked in better condition and I think will grow better…</p>
<pre class="r"><code>GiladLabYRI.Rack &lt;- read_delim(&quot;../data/GiladStrains/LCL_YRI.Gilad.Positions2.txt&quot;, delim = &#39;\t&#39;) %&gt;%
  mutate(Line=as.character(Line))

FourteenLinesAlreadyGrowing &lt;- GrowthNotes %&gt;%
  filter(Thawed == &quot;1/12/21&quot;) %&gt;%
  filter(`CellFractionedAttempted?` == &quot;Y&quot;) %&gt;%
  pull(`Cell Line`) %&gt;% as.character()

CellLinesToThaw_Jan19 &lt;- LCLs.1000Genomes.vec %&gt;%
  setdiff(AlreadyDone.vec) %&gt;%
  setdiff(FourteenLinesAlreadyGrowing) %&gt;%
  base::intersect(GiladLabYRI.Rack %&gt;% pull(Line))
  
CellLinesToThaw_Jan19</code></pre>
<pre><code> [1] &quot;19225&quot; &quot;18500&quot; &quot;18505&quot; &quot;18517&quot; &quot;18497&quot; &quot;18853&quot; &quot;19128&quot; &quot;19147&quot; &quot;18508&quot;
[10] &quot;18510&quot; &quot;18515&quot; &quot;18502&quot; &quot;19239&quot; &quot;19238&quot; &quot;19207&quot; &quot;19257&quot; &quot;19209&quot; &quot;19223&quot;
[19] &quot;18520&quot;</code></pre>
<pre class="r"><code>GiladLabYRI.Rack %&gt;%
  filter(Line %in% CellLinesToThaw_Jan19) %&gt;%
  slice_head(n=13) %&gt;%
  write_tsv(&quot;../data/GiladStrains/20200119_CellsToWake.tsv&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-conda_cos6-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /project2/yangili1/bjf79/ChromatinSplicingQTLs/code/.snakemake/conda/4480a43d58942cece9fb73087fc984b8/lib/R/lib/libRblas.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] knitr_1.31      readxl_1.3.1    forcats_0.5.1   stringr_1.4.0  
 [5] dplyr_1.0.5     purrr_0.3.4     readr_1.4.0     tidyr_1.1.3    
 [9] tibble_3.1.0    ggplot2_3.3.3   tidyverse_1.3.0

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.0  xfun_0.20         bslib_0.2.4       haven_2.3.1      
 [5] colorspace_2.0-0  vctrs_0.3.7       generics_0.1.0    htmltools_0.5.1.1
 [9] yaml_2.2.1        utf8_1.2.1        rlang_0.4.10      jquerylib_0.1.3  
[13] later_1.1.0.1     pillar_1.5.1      withr_2.4.1       glue_1.4.2       
[17] DBI_1.1.1         dbplyr_2.1.1      modelr_0.1.8      lifecycle_1.0.0  
[21] cellranger_1.1.0  munsell_0.5.0     gtable_0.3.0      workflowr_1.6.2  
[25] rvest_1.0.0       evaluate_0.14     ps_1.6.0          httpuv_1.5.5     
[29] fansi_0.4.2       broom_0.7.6       Rcpp_1.0.6        promises_1.2.0.1 
[33] backports_1.2.1   scales_1.1.1      jsonlite_1.7.2    fs_1.5.0         
[37] hms_1.0.0         digest_0.6.27     stringi_1.4.3     rprojroot_2.0.2  
[41] grid_3.6.1        cli_2.4.0         tools_3.6.1       magrittr_2.0.1   
[45] sass_0.3.1        crayon_1.4.1      whisker_0.4       pkgconfig_2.0.3  
[49] ellipsis_0.3.1    xml2_1.3.2        reprex_1.0.0      lubridate_1.7.10 
[53] rstudioapi_0.13   assertthat_0.2.1  rmarkdown_2.7     httr_1.4.2       
[57] R6_2.5.0          git2r_0.28.0      compiler_3.6.1   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
