<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Benjamin Fair" />

<meta name="date" content="2022-01-24" />

<title>Untitled</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Untitled</h1>
<h4 class="author">Benjamin Fair</h4>
<h4 class="date">January 24, 2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-01-24
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>ChromatinSplicingQTLs/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstreecc1c333631f88b8b65947c07e7a1032b2a016ae2targetblankcc1c333a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/cc1c333631f88b8b65947c07e7a1032b2a016ae2" target="_blank">cc1c333</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstreecc1c333631f88b8b65947c07e7a1032b2a016ae2targetblankcc1c333a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/cc1c333631f88b8b65947c07e7a1032b2a016ae2" target="_blank">cc1c333</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    ._.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store
    Ignored:    code/.Rhistory
    Ignored:    code/._.DS_Store
    Ignored:    code/._README.md
    Ignored:    code/._dag.pdf
    Ignored:    code/._report.html
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    code/.snakemake/
    Ignored:    code/Alignments/
    Ignored:    code/ENCODE/
    Ignored:    code/ExpressionAnalysis/
    Ignored:    code/Fastq/
    Ignored:    code/FastqFastp/
    Ignored:    code/FastqFastpSE/
    Ignored:    code/FastqSE/
    Ignored:    code/Gather_gwas_summary_stats/
    Ignored:    code/Genotypes/
    Ignored:    code/Multiqc/
    Ignored:    code/Multiqc_chRNA/
    Ignored:    code/PeakCalling/
    Ignored:    code/Phenotypes/
    Ignored:    code/PlotGruberQTLs/
    Ignored:    code/ProCapAnalysis/
    Ignored:    code/QC/
    Ignored:    code/QTLs/
    Ignored:    code/ReferenceGenome/
    Ignored:    code/Session.vim
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TODO
    Ignored:    code/bigwigs/
    Ignored:    code/bigwigs_FromNonWASPFilteredReads/
    Ignored:    code/config/.DS_Store
    Ignored:    code/config/._.DS_Store
    Ignored:    code/config/ExternalFastqDataAccessions/
    Ignored:    code/config/OldSamplesConfig/
    Ignored:    code/dag.pdf
    Ignored:    code/featureCounts/
    Ignored:    code/gwas_summary_stats/
    Ignored:    code/hyprcoloc/
    Ignored:    code/logs/
    Ignored:    code/notebooks/.ipynb_checkpoints/
    Ignored:    code/out.hap.ld
    Ignored:    code/out.log
    Ignored:    code/report.html
    Ignored:    code/rules/OldRules/
    Ignored:    code/scratch/
    Ignored:    code/scripts/GTFtools_0.8.0/
    Ignored:    code/scripts/__pycache__/
    Ignored:    code/scripts/liftOverBedpe/liftOverBedpe.py
    Ignored:    code/snakemake.log
    Ignored:    code/snakemake.sbatch.log
    Ignored:    data/._PRJEB1350_RunTable.Ding_etal_CTCF.txt
    Ignored:    data/._igsr_samples.tsv
    Ignored:    data/._list_gwas_summary_statistics_PMID27863252.csv
    Ignored:    data/GrowthNotes/._20210830_GrowthNotes_chRNA.ConcentrationsToReplate.txt

Untracked files:
    Untracked:  code/snakemake_profiles/slurm/__pycache__/

Unstaged changes:
    Modified:   analysis/20211217_GenewiseColocFirstLook.Rmd
    Modified:   analysis/20220114_ColocalizationEffectSizeCorrelations.Rmd
    Modified:   code/rules/report.smk
    Modified:   code/scripts/BamToBigwig.sh
    Modified:   code/scripts/GenometracksByGenotype

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/20211217_GenewiseColocFirstLook.Rmd</code>) and HTML (<code>docs/20211217_GenewiseColocFirstLook.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/a6be25f31d620baf3752385a3c95b6fc9c8a66eb/analysis/20211217_GenewiseColocFirstLook.Rmd" target="_blank">a6be25f</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-01-04
</td>
<td>
remove pro-cap from leafcutter cluster rule
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/237b54974507ae639dd505ca0019ae3c6c854af2/docs/20211217_GenewiseColocFirstLook.html" target="_blank">237b549</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2021-12-28
</td>
<td>
rebuilt some site analyses
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/89517ffeaecf50fe2639320bb002b1a5ce1b1858/analysis/20211217_GenewiseColocFirstLook.Rmd" target="_blank">89517ff</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2021-12-22
</td>
<td>
added genewise coloc rmd
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>I have colocalized a bunch of molecular phenotypes in the window of each gene (Locus). Chromatin phenotypes and other genes must be within +/- 100kb of the gene. Splicing events must be within the gene. Nominal permutation Pvalues must be under 0.01. The real results are subject to change, since there are many more molecular phenotypes I haven’t included in this analysis, due to computational impatience, and that I am still collecting data. So I will evenetually include more phenotype which will change the results. More details of the methods are embedded in the snakemake pipeline. Here I will get some intuitions about the results.</p>
</div>
<div id="analysis" class="section level2">
<h2>Analysis</h2>
<p>First, load necessary libraries</p>
<pre class="r"><code>library(tidyverse)
library(viridis)
library(gplots)
library(data.table)

dat &lt;- read_tsv(&quot;../output/hyprcoloc_results/ForColoc/hyprcoloc.results.txt.gz&quot;, col_names = c(&quot;Locus&quot;, &quot;iteration&quot;, &#39;ColocalizedTraits&#39;, &#39;ColocPr&#39;, &#39;RegionalPr&#39;, &quot;topSNP&quot;, &quot;TopSNPFinemapPr&quot;, &quot;DroppedTrait&quot;), skip=1)</code></pre>
<p>First question:</p>
<p>how many traits attempted to colocalized per gene:</p>
<pre class="r"><code>Median &lt;- dat %&gt;%
  mutate(NumTraits = str_count(ColocalizedTraits, &quot;,&quot;) + 1) %&gt;%
  group_by(Locus) %&gt;%
  summarise(TotalTraitsAttemptedColocalization = sum(NumTraits)+1) %&gt;%
  pull(TotalTraitsAttemptedColocalization) %&gt;% median()

dat %&gt;%
  mutate(NumTraits = str_count(ColocalizedTraits, &quot;,&quot;) + 1) %&gt;%
  group_by(Locus) %&gt;%
  summarise(TotalTraitsAttemptedColocalization = sum(NumTraits)+1) %&gt;%
  ggplot(aes(x=TotalTraitsAttemptedColocalization)) +
  geom_bar() +
  ylab(&quot;cumulative fraction&quot;) +
  xlab(&quot;Number molecular traits attempted to colocalize for each gene locus&quot;) +
  xlim(c(-1,20)) +
  geom_vline(xintercept=Median) +
  theme_light()</code></pre>
<p><img src="figure/20211217_GenewiseColocFirstLook.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/237b54974507ae639dd505ca0019ae3c6c854af2/docs/figure/20211217_GenewiseColocFirstLook.Rmd/unnamed-chunk-2-1.png" target="_blank">237b549</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2021-12-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Median of 4 traits attempted to colocalize</p>
<p>Now how many traits actually colocalized per gene, and how many clusters</p>
<pre class="r"><code>#How many traits colocalizes for each gene
Median &lt;- dat %&gt;%
  mutate(NumTraits = str_count(ColocalizedTraits, &quot;,&quot;) ) %&gt;%
  mutate(NumColocalizedTraits  = case_when(
    NumTraits &gt; 0 ~ NumTraits +1,
    TRUE ~ 0
  )) %&gt;%
  group_by(Locus) %&gt;%
  summarise(TotalTraitsColocalized = sum(NumColocalizedTraits)) %&gt;%
  pull(TotalTraitsColocalized) %&gt;% median()

#How many traits colocalizes for each gene
dat %&gt;%
  mutate(NumTraits = str_count(ColocalizedTraits, &quot;,&quot;) ) %&gt;%
  mutate(NumColocalizedTraits  = case_when(
    NumTraits &gt; 0 ~ NumTraits +1,
    TRUE ~ 0
  )) %&gt;%
  group_by(Locus) %&gt;%
  summarise(TotalTraitsColocalized = sum(NumColocalizedTraits)) %&gt;%
  ggplot(aes(x=TotalTraitsColocalized)) +
  # stat_ecdf() +
  # geom_histogram(binwidth = 1) +
  geom_bar() +
  ylab(&quot;Count&quot;) +
  xlim(c(-1,20)) +
  xlab(&quot;Number molecular traits colocalize for each gene locus&quot;) +
  geom_vline(xintercept=Median) +
  theme_light()</code></pre>
<p><img src="figure/20211217_GenewiseColocFirstLook.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/237b54974507ae639dd505ca0019ae3c6c854af2/docs/figure/20211217_GenewiseColocFirstLook.Rmd/unnamed-chunk-3-1.png" target="_blank">237b549</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2021-12-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#How many clusters
dat %&gt;%
  mutate(Cluster = !ColocalizedTraits == &quot;None&quot;) %&gt;%
  group_by(Locus) %&gt;%
  summarize(NumColocClusters = sum(Cluster)) %&gt;%
  ggplot(aes(x=as.factor(NumColocClusters))) +
  geom_bar() +
  ylab(&quot;Count&quot;) +
  # xlim(c(-1,10)) +
  xlab(&quot;Number colocalized clusters per gene&quot;) +
  theme_light()</code></pre>
<p><img src="figure/20211217_GenewiseColocFirstLook.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Now how many colocalizations involve each type of molecular trait. I’m envisioning a matrix of how many colocalizations were attempted for each trait family pair, and how many times they were actually colocalized.</p>
<pre class="r"><code>set.seed(0)
ToSplit &lt;- dat %&gt;%
  select(Locus, ColocalizedTraits, iteration, DroppedTrait) %&gt;%
  # filter(Locus %in% (dat$Locus %&gt;% unique() %&gt;% head(200))) %&gt;%
  mutate(ColocalizedTraits = str_replace(ColocalizedTraits, paste0(Locus, &quot;, &quot;), &quot;&quot;)) %&gt;%
  separate_rows(ColocalizedTraits, sep = &#39; &#39;) %&gt;%
  # recode for better labelling
  mutate(Trait = case_when(
    ColocalizedTraits == &quot;None&quot; ~ DroppedTrait,
    TRUE ~ ColocalizedTraits
  )) %&gt;%
  select(Locus, Trait, iteration) %&gt;%
  group_by(Locus)

# Phenoclasses &lt;- dat %&gt;%
#   pull(pheno_class) %&gt;% unique()
# 
# files &lt;- paste0(&quot;../code/QTLs/QTLTools/&quot;, Phenoclasses, &quot;/OnlyFirstReps.sorted.qqnorm.bed.gz&quot;)
# 
# bed.list &lt;- lapply(files, fread, sep=&#39;\t&#39;, select=2:6) %&gt;%
#   setNames(Phenoclasses)
# positions &lt;- bind_rows(bed.list, .id=&quot;source&quot;) %&gt;%
#   unite(phenotype_full, source, pid, sep = &quot;;&quot;) %&gt;%
#   select(-gid)


ToMat &lt;- function(df){
  xp &lt;- df %&gt;%
  select(Trait, iteration) %&gt;%
  deframe()

  mat &lt;- apply(as.matrix(xp), 1, 
                   function(x) as.numeric(x==as.matrix(xp)))
  rownames(mat) &lt;- colnames(mat)
  mat &lt;- as.matrix(mat)
  diag(x=mat) &lt;- NA
return(mat)
}

SumColocsByPhenotypeClass &lt;- function(mat){
  mat %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;Trait1&quot;) %&gt;%
  gather(key=&quot;Trait2&quot;, value=&quot;IsColoc&quot;, -Trait1) %&gt;%
  mutate(
    Trait1 = str_replace(Trait1, &quot;(.+?);.+&quot;, &quot;\\1&quot;),
    Trait2 = str_replace(Trait2, &quot;(.+?);.+&quot;, &quot;\\1&quot;)
    ) %&gt;%
  drop_na() %&gt;%
  group_by(Trait1, Trait2) %&gt;%
  summarise(NumColocs = sum(IsColoc)) %&gt;%
  return()
}

SumColocAttemptsByPhenotypeClass &lt;- function(mat){
  mat %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;Trait1&quot;) %&gt;%
  gather(key=&quot;Trait2&quot;, value=&quot;IsColoc&quot;, -Trait1) %&gt;%
  mutate(
    Trait1 = str_replace(Trait1, &quot;(.+?);.+&quot;, &quot;\\1&quot;),
    Trait2 = str_replace(Trait2, &quot;(.+?);.+&quot;, &quot;\\1&quot;)
    ) %&gt;%
  drop_na() %&gt;%
  mutate(IsColoc=1) %&gt;%
  group_by(Trait1, Trait2) %&gt;%
  summarise(NumColocs = sum(IsColoc)) %&gt;%
  return()
}



Split.list &lt;- setNames(group_split(ToSplit), deframe(group_keys(ToSplit)))

Mat.list &lt;- lapply(Split.list, ToMat)

## Some sanity checks that this code works as expected
Mat.list[[&quot;ENSG00000196735.13&quot;]] %&gt;%
  write.table(row.names = F, col.names = F)</code></pre>
<pre><code>NA 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
1 NA 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
1 1 NA 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
1 1 1 NA 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 NA 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 NA 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 1 NA 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 1 1 NA 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 1 1 1 NA 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 NA 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 NA 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 1 1 NA 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 NA 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 1 NA 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 1 1 NA 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 NA 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 NA 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 NA 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 NA 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 NA 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 NA 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 NA 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 NA 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 NA 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 NA 1 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 NA 0 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 NA 0 0 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 NA 1 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 NA 0 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 NA 0 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 NA 0 0 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 NA 1 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 NA 0 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 NA 0 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 NA 0 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 NA 0 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 NA 0 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 NA 0
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 NA</code></pre>
<pre class="r"><code>Mat.list[[100]] %&gt;%
  write.table(row.names = T, col.names = F)</code></pre>
<pre><code>&quot;Expression.Splicing;H3K4ME3;H3K4ME3_peak_17313,&quot; NA 1 0 0 0
&quot;polyA.Splicing.Subset_YRI;16:579180:580858:clu_32206_+&quot; 1 NA 0 0 0
&quot;H3K27AC;H3K27AC_peak_34486,&quot; 0 0 NA 1 0
&quot;H3K4ME3;H3K4ME3_peak_17317&quot; 0 0 1 NA 0
&quot;H3K4ME3;H3K4ME3_peak_17316&quot; 0 0 0 0 NA</code></pre>
<pre class="r"><code>SumColocs.list &lt;- lapply(Mat.list, SumColocsByPhenotypeClass)
SumAttemptedColocs.list &lt;- lapply(Mat.list, SumColocAttemptsByPhenotypeClass)


Coloc.ByClass.Summary &lt;- bind_rows(SumColocs.list, .id=&quot;Locus&quot;) %&gt;%
  inner_join(
    bind_rows(SumAttemptedColocs.list, .id=&quot;Locus&quot;),
    by=c(&quot;Trait1&quot;, &quot;Trait2&quot;, &quot;Locus&quot;),
    suffix = c(&quot;.Success&quot;, &quot;.Attemped&quot;)
  ) %&gt;%
  # filter(Locus == &quot;ENSG00000114378.17&quot;)
  group_by(Trait1, Trait2) %&gt;%
  summarise(
    Colocs = sum(NumColocs.Success),
    ColocAttempts = sum(NumColocs.Attemped))

Coloc.ByClass.Summary %&gt;%
  arrange(Trait1, Trait2) %&gt;%
  gather(key=&quot;AttemptOrSuccess&quot;, value=&quot;NumColocs&quot;, Colocs, ColocAttempts) %&gt;%
  mutate(NumColocs = NumColocs/2) %&gt;%
  ggplot(aes(x=Trait1, y=Trait2, fill=NumColocs)) +
  geom_raster() +
  scale_fill_viridis(trans=&quot;log10&quot;, option=&quot;B&quot;) +
  facet_wrap(~AttemptOrSuccess) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))</code></pre>
<p><img src="figure/20211217_GenewiseColocFirstLook.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/237b54974507ae639dd505ca0019ae3c6c854af2/docs/figure/20211217_GenewiseColocFirstLook.Rmd/unnamed-chunk-4-1.png" target="_blank">237b549</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2021-12-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#Plot fill color by rate
Coloc.ByClass.Summary %&gt;%
  mutate(ColocRate = Colocs/ColocAttempts * 100) %&gt;%
  arrange(Trait1, Trait2) %&gt;%
  ggplot(aes(x=Trait1, y=Trait2, fill=ColocRate)) +
  geom_raster() +
  scale_fill_viridis(option=&quot;B&quot;) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))</code></pre>
<p><img src="figure/20211217_GenewiseColocFirstLook.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-2">
Past versions of unnamed-chunk-4-2.png
</button>
</p>
<div id="fig-unnamed-chunk-4-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/237b54974507ae639dd505ca0019ae3c6c854af2/docs/figure/20211217_GenewiseColocFirstLook.Rmd/unnamed-chunk-4-2.png" target="_blank">237b549</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2021-12-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Coloc.ByClass.Summary.ForHeatmap &lt;- Coloc.ByClass.Summary %&gt;%
  mutate(ColocRate = Colocs/ColocAttempts * 100) %&gt;%
  select(Trait1, Trait2, ColocRate)

Coloc.ByClass.Summary.ForHeatmap %&gt;%
  rename(Trait1=Trait2, Trait2=Trait1) %&gt;%
  bind_rows(Coloc.ByClass.Summary.ForHeatmap) %&gt;%
  distinct(.keep_all=T) %&gt;%
  pivot_wider(names_from = &quot;Trait1&quot;, values_from = &quot;ColocRate&quot;, values_fill=NA, names_sort=T) %&gt;%
  column_to_rownames(&quot;Trait2&quot;) %&gt;%
  as.matrix() %&gt;%
  # heatmap.2(trace=&quot;none&quot;, col=brewer.pal(51,&quot;Spectral&quot;))
  # heatmap.2(trace=&quot;none&quot;, col=colorpanel(75, &quot;blue&quot;, &quot;black&quot;, &quot;yellow&quot;), cexRow = 1, cexCol = 1, offsetRow=0, offsetCol = 0, margins=c(11,11), dendrogram = &quot;column&quot;)
  heatmap.2(trace=&quot;none&quot;, col=viridis(15, option=&quot;B&quot;), cexRow = 1, cexCol = 1, offsetRow=0, offsetCol = 0, margins=c(11,11), dendrogram = &quot;column&quot;)</code></pre>
<p><img src="figure/20211217_GenewiseColocFirstLook.Rmd/unnamed-chunk-4-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Now let’s look at the distance between colocalized trait pairs, for different phenotype classes</p>
<pre class="r"><code>DatToPlotDist &lt;- dat %&gt;%
  select(Locus, ColocalizedTraits, iteration, DroppedTrait) %&gt;%
  separate_rows(ColocalizedTraits, sep = &#39;, &#39;) %&gt;%
  gather(key = &quot;ColocalizationStatus&quot;, value=&quot;phenotype_full&quot;, ColocalizedTraits, DroppedTrait) %&gt;%
  filter(!is.na(phenotype_full)) %&gt;%
  filter(!phenotype_full == &quot;None&quot;) %&gt;%
  mutate(pheno_class = str_replace(phenotype_full, &quot;(.+?);.+$&quot;, &quot;\\1&quot;))


Phenoclasses &lt;- DatToPlotDist %&gt;%
  pull(pheno_class) %&gt;% unique()

files &lt;- paste0(&quot;../code/QTLs/QTLTools/&quot;, Phenoclasses, &quot;/OnlyFirstReps.sorted.qqnorm.bed.gz&quot;)

bed.list &lt;- lapply(files, fread, sep=&#39;\t&#39;, select=2:6) %&gt;%
  setNames(Phenoclasses)
positions &lt;- bind_rows(bed.list, .id=&quot;source&quot;) %&gt;%
  unite(phenotype_full, source, pid, sep = &quot;;&quot;) %&gt;%
  select(-gid)

head(positions)</code></pre>
<pre><code>                                    phenotype_full start   end strand
1: polyA.Splicing.Subset_YRI;1:15947:16607:clu_1_- 15947 16607      -
2: polyA.Splicing.Subset_YRI;1:16310:16607:clu_1_- 16310 16607      -
3: polyA.Splicing.Subset_YRI;1:17055:17233:clu_2_- 17055 17233      -
4: polyA.Splicing.Subset_YRI;1:17055:17915:clu_2_- 17055 17915      -
5: polyA.Splicing.Subset_YRI;1:17368:17606:clu_2_- 17368 17606      -
6: polyA.Splicing.Subset_YRI;1:17368:17915:clu_2_- 17368 17915      -</code></pre>
<pre class="r"><code>DatToPlotDist &lt;- DatToPlotDist %&gt;%
  left_join(positions, by=&quot;phenotype_full&quot;)

GeneWithPromoterPhenotypes &lt;- c(&quot;Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;, &quot;chRNA.Expression.Splicing&quot;, &quot;MetabolicLabelled.60min&quot;, &quot;MetabolicLabelled.30min&quot;)

PairwiseColocalizedPhenotypes &lt;- DatToPlotDist %&gt;%
  # head(1000) %&gt;%
  mutate(start = case_when(
    (pheno_class %in% GeneWithPromoterPhenotypes) &amp; (strand==&quot;-&quot;) ~ as.integer(end-1),
    TRUE ~ start
  )) %&gt;%
  mutate(end = case_when(
    (pheno_class %in% GeneWithPromoterPhenotypes) &amp; (strand==&quot;+&quot;) ~ as.integer(start+1),
    TRUE ~ end
  )) %&gt;%
  mutate(pheno_class = recode(pheno_class,
                                   Expression.Splicing = &quot;Expression (Geuvadis)&quot;,
                                   Expression.Splicing.Subset_YRI = &quot;Expression (YRI.Subset)&quot;,
                                   polyA.Splicing.Subset_YRI = &quot;Splicing (leafcutter.YRI.Subset)&quot;,
                                   chRNA.Expression.Splicing = &quot;Expression (chRNA)&quot;)) %&gt;%
  # filter(Locus_snp == &quot;ENSG00000002822.15_7:1998126:G:A&quot;) %&gt;%
  left_join(., ., by = c(&quot;Locus&quot;, &quot;iteration&quot;)) %&gt;% 
  filter(phenotype_full.x != phenotype_full.y) %&gt;% 
  rowwise() %&gt;%
  mutate(name = toString(sort(c(phenotype_full.x,phenotype_full.y)))) %&gt;% 
  distinct(Locus_snp, name, .keep_all = T) %&gt;%
  mutate(pheno_class_name = toString(sort(c(pheno_class.x,pheno_class.y)))) %&gt;%
  mutate(Distance_between = case_when(
    # If features overlap
    ((start.x &gt;= start.y) &amp; (start.x &lt;= end.y)) |
    ((end.x &gt;= start.y) &amp; (end.x &lt;= end.y)) |
    ((start.x &lt;= start.y) &amp; (end.x &gt;= end.y)) ~ as.integer(0),
    TRUE ~ min(abs(end.x - start.y), abs(start.x - end.y))
  )) %&gt;%
  mutate(PairwiseColocClass=&quot;Colocalized&quot;)


PairwiseColocalizedPhenotypes %&gt;%
  mutate(ColocStatus = paste(ColocalizationStatus.x, ColocalizationStatus.y)) %&gt;%
  pull(ColocStatus) %&gt;% table()</code></pre>
<pre><code>.
ColocalizedTraits ColocalizedTraits 
                              25421 </code></pre>
<pre class="r"><code>PairwiseAllPhenotypes &lt;- DatToPlotDist %&gt;%
  # head(1000) %&gt;%
  mutate(start = case_when(
    (pheno_class %in% GeneWithPromoterPhenotypes) &amp; (strand==&quot;-&quot;) ~ as.integer(end-1),
    TRUE ~ start
  )) %&gt;%
  mutate(end = case_when(
    (pheno_class %in% GeneWithPromoterPhenotypes) &amp; (strand==&quot;+&quot;) ~ as.integer(start+1),
    TRUE ~ end
  )) %&gt;%
  mutate(pheno_class = recode(pheno_class,
                                   Expression.Splicing = &quot;Expression (Geuvadis)&quot;,
                                   Expression.Splicing.Subset_YRI = &quot;Expression (YRI.Subset)&quot;,
                                   polyA.Splicing.Subset_YRI = &quot;Splicing (leafcutter.YRI.Subset)&quot;,
                                   chRNA.Expression.Splicing = &quot;Expression (chRNA)&quot;)) %&gt;%
  # filter(Locus_snp == &quot;ENSG00000002822.15_7:1998126:G:A&quot;) %&gt;%
  left_join(., ., by = c(&quot;Locus&quot;)) %&gt;% 
  filter(phenotype_full.x != phenotype_full.y) %&gt;% 
  rowwise() %&gt;%
  mutate(name = toString(sort(c(phenotype_full.x,phenotype_full.y)))) %&gt;% 
  distinct(Locus_snp, name, .keep_all = T) %&gt;%
  mutate(pheno_class_name = toString(sort(c(pheno_class.x,pheno_class.y)))) %&gt;%
  mutate(Distance_between = case_when(
    # If features overlap
    ((start.x &gt;= start.y) &amp; (start.x &lt;= end.y)) |
    ((end.x &gt;= start.y) &amp; (end.x &lt;= end.y)) |
    ((start.x &lt;= start.y) &amp; (end.x &gt;= end.y)) ~ as.integer(0),
    TRUE ~ min(abs(end.x - start.y), abs(start.x - end.y))
  )) %&gt;%
  mutate(PairwiseColocClass=&quot;AllAttemped&quot;)

PairwiseAllPhenotypes %&gt;%
  mutate(ColocStatus = paste(ColocalizationStatus.x, ColocalizationStatus.y)) %&gt;%
  pull(ColocStatus) %&gt;% table()</code></pre>
<pre><code>.
ColocalizedTraits ColocalizedTraits      ColocalizedTraits DroppedTrait 
                              40632                               25184 
          DroppedTrait DroppedTrait 
                              12858 </code></pre>
<pre class="r"><code>bind_rows(PairwiseAllPhenotypes, PairwiseColocalizedPhenotypes) %&gt;%
  mutate(Distance_between = Distance_between+1) %&gt;%
  ggplot(aes(x=Distance_between, color=PairwiseColocClass)) +
  geom_boxplot(outlier.shape =NA) +
  # geom_density() +
  # scale_x_continuous(trans=&quot;log10&quot;) +
  coord_cartesian(xlim=c(1,100000)) +
  facet_wrap(~pheno_class_name) +
  theme_classic()</code></pre>
<p><img src="figure/20211217_GenewiseColocFirstLook.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok as expected, trait pairs that colocalize, for most all types of phenotypes, are closer than those that do not colocalize (compared to all traits that I attempted colocalization)… These plots might be kind of hard to interpret. Maybe a simpler summary is to present the median distance for those that do colocalize, versus those that I attempted colocalization for.</p>
<pre class="r"><code># 
bind_rows(PairwiseAllPhenotypes, PairwiseColocalizedPhenotypes) %&gt;%
  group_by(pheno_class_name, PairwiseColocClass) %&gt;%
  summarize(MedianDistanceBetweenTraitPairs = median(Distance_between)) %&gt;%
  ggplot(aes(x=PairwiseColocClass, y=MedianDistanceBetweenTraitPairs, fill=PairwiseColocClass)) +
  geom_col() +
  # facet_wrap(~pheno_class_name) +
  theme_bw()</code></pre>
<p><img src="figure/20211217_GenewiseColocFirstLook.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DistancesToPlot &lt;-bind_rows(PairwiseAllPhenotypes, PairwiseColocalizedPhenotypes) %&gt;%
  mutate(Distance_between = Distance_between+1) %&gt;%
  group_by(pheno_class.x, pheno_class.y, PairwiseColocClass) %&gt;%
  summarize(MedianDistanceBetweenTraitPairs = median(Distance_between))

DistancesToPlot %&gt;%
  rename(pheno_class.x=pheno_class.y, pheno_class.y=pheno_class.x) %&gt;%
  bind_rows(DistancesToPlot) %&gt;%
  ggplot(aes(x=pheno_class.x, y=pheno_class.y, fill=MedianDistanceBetweenTraitPairs)) +
  geom_raster() +
  # scale_fill_continuous(trans=&quot;log10&quot;) +
  scale_fill_viridis(trans=&quot;log10&quot;, option=&quot;B&quot;, direction = -1) +
  facet_wrap(~PairwiseColocClass) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))</code></pre>
<p><img src="figure/20211217_GenewiseColocFirstLook.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>AsFacetsPlot &lt;- DistancesToPlot %&gt;%
  rename(pheno_class.x=pheno_class.y, pheno_class.y=pheno_class.x) %&gt;%
  bind_rows(DistancesToPlot) %&gt;%
  mutate(MedianDistanceBetweenTraitPairs= MedianDistanceBetweenTraitPairs/1000) %&gt;%
  arrange(desc(pheno_class.x), pheno_class.y) %&gt;%
  ggplot(aes(x=PairwiseColocClass, y=MedianDistanceBetweenTraitPairs, fill=PairwiseColocClass)) +
  geom_col() +
  facet_grid(rows = vars(pheno_class.x), cols = vars(pheno_class.y), switch=&quot;y&quot;,labeller = label_wrap_gen(width=10)) +
  ylab(&quot;Median distance between trait pairs (kb)&quot;) +
  theme_bw() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  theme(strip.text = element_text(size = 6))</code></pre>
<pre class="r"><code>ggsave(&quot;../code/scratch/PairwiseDistances.pdf&quot;, AsFacetsPlot, height=5, width=7)</code></pre>
</div>
<div id="conclusions" class="section level2">
<h2>Conclusions</h2>
<p>Some thing make sense. polyA.splicing traits often colocalize with other polyA.Splicing traits, which is unsurprisingly biologically and also technically since intra-cluster introns are not independent quantifications. H3K27ME3 and H3K4ME3 also tend to colocalize.</p>
<p>The colocalization rate might be more informative, in the sense that the results match by expectations for which traits might colocalize.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] data.table_1.13.2 gplots_3.0.1.1    viridis_0.5.1    
 [4] viridisLite_0.3.0 forcats_0.4.0     stringr_1.4.0    
 [7] dplyr_0.8.3       purrr_0.3.4       readr_1.3.1      
[10] tidyr_1.1.0       tibble_2.1.3      ggplot2_3.3.3    
[13] tidyverse_1.3.0  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5         lubridate_1.7.4    lattice_0.20-38   
 [4] gtools_3.8.1       assertthat_0.2.1   rprojroot_2.0.2   
 [7] digest_0.6.20      R6_2.4.0           cellranger_1.1.0  
[10] backports_1.1.4    reprex_0.3.0       evaluate_0.14     
[13] httr_1.4.1         pillar_1.4.2       rlang_0.4.10      
[16] readxl_1.3.1       rstudioapi_0.10    gdata_2.18.0      
[19] whisker_0.3-2      R.oo_1.22.0        R.utils_2.9.0     
[22] rmarkdown_1.13     labeling_0.3       munsell_0.5.0     
[25] broom_0.5.2        compiler_3.6.1     httpuv_1.5.1      
[28] modelr_0.1.8       xfun_0.8           pkgconfig_2.0.2   
[31] htmltools_0.3.6    tidyselect_1.1.0   gridExtra_2.3     
[34] workflowr_1.6.2    fansi_0.4.0        crayon_1.3.4      
[37] dbplyr_1.4.2       withr_2.4.1        later_0.8.0       
[40] R.methodsS3_1.7.1  bitops_1.0-6       grid_3.6.1        
[43] nlme_3.1-140       jsonlite_1.6       gtable_0.3.0      
[46] lifecycle_0.1.0    DBI_1.1.0          git2r_0.26.1      
[49] magrittr_1.5       scales_1.1.0       KernSmooth_2.23-15
[52] cli_2.2.0          stringi_1.4.3      farver_2.1.0      
[55] fs_1.3.1           promises_1.0.1     xml2_1.3.2        
[58] ellipsis_0.2.0.1   generics_0.0.2     vctrs_0.3.1       
[61] tools_3.6.1        glue_1.3.1         hms_0.5.3         
[64] yaml_2.2.0         colorspace_1.4-1   caTools_1.17.1.2  
[67] rvest_0.3.5        knitr_1.23         haven_2.3.1       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
