<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>20210604_Check_chRNASeq_Genotypes</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<div class="fluid-row" id="header">



<h1 class="title toc-ignore">20210604_Check_chRNASeq_Genotypes</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-06-04
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>ChromatinSplicingQTLs/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstreed908cecd8af9abe62d8777837b416f71e9358639targetblankd908ceca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/d908cecd8af9abe62d8777837b416f71e9358639" target="_blank">d908cec</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstreed908cecd8af9abe62d8777837b416f71e9358639targetblankd908ceca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    ._.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store
    Ignored:    code/.Rhistory
    Ignored:    code/._.DS_Store
    Ignored:    code/.snakemake/
    Ignored:    code/Alignments/
    Ignored:    code/ENCODE/
    Ignored:    code/ExpressionAnalysis/
    Ignored:    code/Fastq/
    Ignored:    code/FastqFastp/
    Ignored:    code/Genotypes/
    Ignored:    code/Log.out
    Ignored:    code/Log.progress.out
    Ignored:    code/Multiqc/
    Ignored:    code/MyOutput.sam
    Ignored:    code/PeakCalling/
    Ignored:    code/Phenotypes/
    Ignored:    code/PlotGruberQTLs/
    Ignored:    code/ProCapAnalysis/
    Ignored:    code/QC/
    Ignored:    code/QTLs/
    Ignored:    code/ReferenceGenome/
    Ignored:    code/Session.vim
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TODO
    Ignored:    code/_STARtmp/
    Ignored:    code/bigwigs/
    Ignored:    code/bigwigs_FromNonWASPFilteredReads/
    Ignored:    code/config/.DS_Store
    Ignored:    code/config/._.DS_Store
    Ignored:    code/config/ExternalFastqDataAccessions/
    Ignored:    code/featureCounts/
    Ignored:    code/logs/
    Ignored:    code/out.log
    Ignored:    code/output/
    Ignored:    code/rules/OldRules/
    Ignored:    code/scratch/
    Ignored:    code/scripts/GTFtools_0.8.0/
    Ignored:    code/scripts/liftOverBedpe/liftOverBedpe.py
    Ignored:    code/snakemake.log
    Ignored:    code/snakemake.sbatch.log
    Ignored:    data/._PRJEB1350_RunTable.Ding_etal_CTCF.txt

Untracked files:
    Untracked:  analysis/20210604_CheckSampleGenotypes.Rmd
    Untracked:  code/envs/deeptools.yml
    Untracked:  code/rules/ProSeqAnalysis.smk
    Untracked:  code/scripts/PlotPCA_FromPhenotypeTable.R
    Untracked:  code/scripts/Plot_mbv.R
    Untracked:  code/scripts/PreparePhenotypeTableFromFeatureCounts_SubsetGeneList.R
    Untracked:  code/scripts/PreparePhenotypeTable_ProCap.R
    Untracked:  code/snakemake_profiles/slurm/__pycache__/
    Untracked:  data/20210604_chRNA_SampleIDs_FromBamToFix.txt
    Untracked:  output/QC/

Unstaged changes:
    Modified:   code/Snakefile
    Modified:   code/config/samples.tsv
    Modified:   code/rules/ExpressionAnalysis.smk
    Modified:   code/rules/MakeBigwigs.smk
    Modified:   code/rules/QC.smk
    Modified:   code/rules/QTLTools.smk
    Modified:   code/rules/common.py
    Modified:   code/scripts/CalculateNormFactorsForBigwig.R
    Modified:   code/scripts/PreparePhenotypeTablesFromFeatureCounts_ChromatinProfilingPeaks.R

Staged changes:
    Modified:   .gitmodules
    New:        code/scripts/qtltools

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="motivation" class="section level2">
<h2>Motivation</h2>
<p>From our first batch of chRNA-seq samples (~40 YRI lines), I found no eQTLs which was a bit concerning. I want to verify that there were no sample swaps. Therefore, I ran <a href="https://qtltools.github.io/qtltools/">QTLtools</a> <code>mbv</code> command on each bam to see which 1000 genome’s sample matches best, and saved the results to a file included in this repo. Here I will analyze those results, and figure out if there were any sample swaps.</p>
</div>
<div id="analysis" class="section level2">
<h2>analysis</h2>
<p>first load libraries and data</p>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching packages ──────────────────────────────── tidyverse 1.2.1 ──</code></pre>
<pre><code>✓ ggplot2 3.2.1     ✓ purrr   0.3.3
✓ tibble  3.0.4     ✓ dplyr   1.0.2
✓ tidyr   1.1.2     ✓ stringr 1.4.0
✓ readr   1.4.0     ✓ forcats 0.5.0</code></pre>
<pre><code>── Conflicts ─────────────────────────────────── tidyverse_conflicts() ──
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(knitr)

dat &lt;- read_tsv(&quot;../output/QC/20210604_mbv.summary.txt.gz&quot;)</code></pre>
<pre><code>
── Column specification ─────────────────────────────────────────────────
cols(
  SampleID = col_character(),
  n_het_covered = col_double(),
  perc_het_consistent = col_double(),
  perc_hom_consistent = col_double(),
  fn = col_character()
)</code></pre>
<pre class="r"><code>head(dat) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">SampleID</th>
<th align="right">n_het_covered</th>
<th align="right">perc_het_consistent</th>
<th align="right">perc_hom_consistent</th>
<th align="left">fn</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">HG00096</td>
<td align="right">0</td>
<td align="right">NaN</td>
<td align="right">1</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA18486.1.txt</td>
</tr>
<tr class="even">
<td align="left">HG00097</td>
<td align="right">0</td>
<td align="right">NaN</td>
<td align="right">1</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA18486.1.txt</td>
</tr>
<tr class="odd">
<td align="left">HG00099</td>
<td align="right">0</td>
<td align="right">NaN</td>
<td align="right">1</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA18486.1.txt</td>
</tr>
<tr class="even">
<td align="left">HG00100</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">NaN</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA18486.1.txt</td>
</tr>
<tr class="odd">
<td align="left">HG00101</td>
<td align="right">0</td>
<td align="right">NaN</td>
<td align="right">1</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA18486.1.txt</td>
</tr>
<tr class="even">
<td align="left">HG00102</td>
<td align="right">0</td>
<td align="right">NaN</td>
<td align="right">1</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA18486.1.txt</td>
</tr>
</tbody>
</table>
<p>Now tidy the data a bit plot the results for one example</p>
<pre class="r"><code># Extract sample name from filename (fn)
dat &lt;- dat %&gt;%
  mutate(ExpectedSampleID = str_replace(fn, &quot;.+Splicing\\/(.+?)\\..+$&quot;, &quot;\\1&quot;))

dat %&gt;%
  filter(ExpectedSampleID==&quot;NA19130&quot;) %&gt;%
  mutate(IsExpectedSample= (ExpectedSampleID == SampleID)) %&gt;%
  arrange(IsExpectedSample) %&gt;%
  ggplot(aes(x=perc_het_consistent, y=perc_hom_consistent, color=IsExpectedSample, label=SampleID)) +
  geom_text(size=2) +
  theme_classic() +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/20210604_CheckSampleGenotypes.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, at least for that one sample, the data looks great and as expected. The expected sample (line NA19130) is a clear outlier with a higher fraction of concordant reads matching the expected genotypes for line NA19130 at both homozygous and heterozygous sites. Let’s make this plot for all sequenced samples…</p>
<pre class="r"><code>dat %&gt;%
  mutate(IsExpectedSample= (ExpectedSampleID == SampleID)) %&gt;%
  arrange(IsExpectedSample) %&gt;%
  ggplot(aes(x=perc_het_consistent, y=perc_hom_consistent, color=IsExpectedSample, label=SampleID)) +
  geom_text(size=2) +
  facet_wrap(~ExpectedSampleID, scale=&quot;free&quot;) +
  theme_classic() +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/20210604_CheckSampleGenotypes.Rmd/unnamed-chunk-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Ok, clearly there a lot of sample swaps where the best match is clear and is not the expected sample. Though there a lot of ‘good’ samples that look like the first example I plotted, and there are also some samples that just don’t have enough data to make a reliable plot. I am noting that all of the samples where there is a clear match that is not the expected sample, match to a different sample in this batch of samples we made libraries for. This is most consistent with sample swapping during our cell prep or library prep, rather than the cell lines that we thawed being some other cell line.</p>
<p>Let’s write some quick rules to output the best match for each line, so we can systematically correct the sample swapping… For example, something along the lines of this algorithm:</p>
<ol style="list-style-type: decimal">
<li>If there less than x sites considered for calculating percent concordant sites, then the plot is unreliable and let’s say the best match is ‘undetermined’. Will have to determine what x is in a bit…</li>
<li>If the line that has the highest percent_het_consistent is also the line with the highest percent_hom_consistent, then that is the best match. Else, the best match is ‘undetermined’.</li>
</ol>
<p>First, let’s look at the plot above, and compare it to the sum of <code>n_het_covered</code> field for each sample as a proxy for good the data is and whether we should exclude certain samples (Point #1)</p>
<pre class="r"><code>dat.x &lt;- dat %&gt;%
  group_by(ExpectedSampleID) %&gt;%
  summarise(n_het_covered_sum = sum(n_het_covered, na.rm=T)) %&gt;%
  arrange(n_het_covered_sum)</code></pre>
<pre><code>`summarise()` ungrouping output (override with `.groups` argument)</code></pre>
<pre class="r"><code>ggplot(dat.x, aes(x=reorder(ExpectedSampleID, n_het_covered_sum), y=n_het_covered_sum)) +
  geom_col() +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90))</code></pre>
<p><img src="figure/20210604_CheckSampleGenotypes.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>head(dat.x, 10)</code></pre>
<pre><code># A tibble: 10 x 2
   ExpectedSampleID n_het_covered_sum
   &lt;chr&gt;                        &lt;dbl&gt;
 1 NA18520                          0
 2 NA19114                          0
 3 NA19238                          0
 4 NA18486                        983
 5 NA19160                       3159
 6 NA19152                      45651
 7 NA18907                      67066
 8 NA19119                      80334
 9 NA19223                     216688
10 NA19257                     345384</code></pre>
<p>Ok, let’s consider all those samples with less n_het_covered_sum than sample 19119 as samples to drop and automatically call unknown.</p>
<pre class="r"><code>SamplesToDrop &lt;- dat.x %&gt;%
  filter(n_het_covered_sum &lt; 80334) %&gt;%
  pull(ExpectedSampleID)

SamplesToDrop</code></pre>
<pre><code>[1] &quot;NA18520&quot; &quot;NA19114&quot; &quot;NA19238&quot; &quot;NA18486&quot; &quot;NA19160&quot; &quot;NA19152&quot; &quot;NA18907&quot;</code></pre>
<p>Now let’s find the best match for the rest…</p>
<pre class="r"><code>BestMatches &lt;- dat %&gt;%
  filter(!ExpectedSampleID %in% SamplesToDrop) %&gt;%
  group_by(fn) %&gt;%
  mutate( BestHit_het = (perc_het_consistent == max(perc_het_consistent, na.rm = T)),
          BestHit_hom = (perc_hom_consistent == max(perc_hom_consistent, na.rm=T))) %&gt;%
  ungroup() %&gt;%
  filter(BestHit_hom &amp; BestHit_het) %&gt;%
  right_join(
    dat %&gt;% select(ExpectedSampleID) %&gt;% unique(),
    by = &quot;ExpectedSampleID&quot;
  ) %&gt;%
  select(ExpectedSampleID, BestMatch = SampleID, fn) %&gt;%
  mutate(IsSwapped = !ExpectedSampleID == BestMatch)

kable(BestMatches)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">ExpectedSampleID</th>
<th align="left">BestMatch</th>
<th align="left">fn</th>
<th align="left">IsSwapped</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">NA18497</td>
<td align="left">NA19130</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA18497.1.txt</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">NA18499</td>
<td align="left">NA19209</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA18499.1.txt</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">NA18505</td>
<td align="left">NA18486</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA18505.1.txt</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">NA18508</td>
<td align="left">NA18499</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA18508.1.txt</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">NA18511</td>
<td align="left">NA18511</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA18511.1.txt</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">NA18519</td>
<td align="left">NA18519</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA18519.1.txt</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">NA18522</td>
<td align="left">NA18852</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA18522.1.txt</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">NA18852</td>
<td align="left">NA18520</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA18852.1.txt</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">NA18858</td>
<td align="left">NA18858</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA18858.1.txt</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">NA18909</td>
<td align="left">NA18909</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA18909.1.txt</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">NA18912</td>
<td align="left">NA18912</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA18912.1.txt</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">NA18913</td>
<td align="left">NA18913</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA18913.1.txt</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">NA19093</td>
<td align="left">NA19093</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA19093.1.txt</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">NA19101</td>
<td align="left">NA19101</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA19101.1.txt</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">NA19102</td>
<td align="left">NA19099</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA19102.1.txt</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">NA19119</td>
<td align="left">NA19119</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA19119.1.txt</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">NA19128</td>
<td align="left">NA19140</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA19128.1.txt</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">NA19128</td>
<td align="left">NA18508</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA19128.2.txt</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">NA19130</td>
<td align="left">NA19130</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA19130.1.txt</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">NA19131</td>
<td align="left">NA19131</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA19131.1.txt</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">NA19137</td>
<td align="left">NA19152</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA19137.1.txt</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">NA19138</td>
<td align="left">NA19138</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA19138.1.txt</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">NA19140</td>
<td align="left">NA18522</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA19140.1.txt</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">NA19141</td>
<td align="left">NA19190</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA19141.1.txt</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">NA19153</td>
<td align="left">NA19171</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA19153.1.txt</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">NA19171</td>
<td align="left">NA19200</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA19171.1.txt</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">NA19190</td>
<td align="left">NA19114</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA19190.1.txt</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">NA19200</td>
<td align="left">NA19160</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA19200.1.txt</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">NA19201</td>
<td align="left">NA19225</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA19201.1.txt</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">NA19207</td>
<td align="left">NA19257</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA19207.1.txt</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">NA19209</td>
<td align="left">NA19238</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA19209.1.txt</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">NA19210</td>
<td align="left">NA19137</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA19210.1.txt</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">NA19257</td>
<td align="left">NA18505</td>
<td align="left">QC/mbv/data/chRNA.Expression.Splicing/NA19257.1.txt</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">NA18486</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA18520</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA18907</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA19114</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA19152</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA19160</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA19223</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA19225</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA19238</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
</tbody>
</table>
<p>Also, write out these results to a file… It might be handy if I want to fix the sample labels in my snakemake with a script. These exact results will be hard to replicate once I fix the sample labels in my snakemake, so I’ll save them to the data folder where I tend to not write to files to be overwritten.</p>
<pre class="r"><code>write_tsv(BestMatches, &quot;../data/20210604_chRNA_SampleIDs_FromBamToFix.txt&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.3 (2017-11-30)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] knitr_1.26      forcats_0.5.0   stringr_1.4.0   dplyr_1.0.2    
 [5] purrr_0.3.3     readr_1.4.0     tidyr_1.1.2     tibble_3.0.4   
 [9] ggplot2_3.2.1   tidyverse_1.2.1

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.0  xfun_0.20         haven_2.3.1       colorspace_2.0-0 
 [5] vctrs_0.3.6       generics_0.1.0    htmltools_0.4.0   yaml_2.2.0       
 [9] utf8_1.1.4        rlang_0.4.9       later_1.0.0       pillar_1.4.7     
[13] glue_1.4.2        withr_2.1.2       modelr_0.1.8      readxl_1.3.1     
[17] lifecycle_0.2.0   munsell_0.5.0     gtable_0.3.0      workflowr_1.5.0  
[21] cellranger_1.1.0  rvest_0.3.6       evaluate_0.14     labeling_0.3     
[25] httpuv_1.5.2      fansi_0.4.0       highr_0.8         broom_0.7.3      
[29] Rcpp_1.0.3        promises_1.1.0    scales_1.1.0      backports_1.1.5  
[33] jsonlite_1.6      farver_2.0.1      fs_1.3.1          hms_0.5.3        
[37] digest_0.6.27     stringi_1.4.3     grid_3.4.3        rprojroot_1.3-2  
[41] cli_2.0.0         tools_3.4.3       magrittr_1.5      lazyeval_0.2.2   
[45] crayon_1.3.4      pkgconfig_2.0.3   ellipsis_0.3.0    xml2_1.2.0       
[49] lubridate_1.7.9.2 assertthat_0.2.1  rmarkdown_2.6     httr_1.4.2       
[53] rstudioapi_0.10   R6_2.4.1          git2r_0.26.1      compiler_3.4.3   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
