<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Plot heatmap in many styles</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Plot heatmap in many styles</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-07-13
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>ChromatinSplicingQTLs/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstree15032a4aef6e03dc4e883c39e8d445b55044b9f5targetblank15032a4a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/15032a4aef6e03dc4e883c39e8d445b55044b9f5" target="_blank">15032a4</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstree15032a4aef6e03dc4e883c39e8d445b55044b9f5targetblank15032a4a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/15032a4aef6e03dc4e883c39e8d445b55044b9f5" target="_blank">15032a4</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store
    Ignored:    code/.RData
    Ignored:    code/._.DS_Store
    Ignored:    code/._README.md
    Ignored:    code/._report.html
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    code/.snakemake/
    Ignored:    code/Alignments/
    Ignored:    code/ENCODE/
    Ignored:    code/ExpressionAnalysis/
    Ignored:    code/FastqFastp/
    Ignored:    code/FastqFastpSE/
    Ignored:    code/Genotypes/
    Ignored:    code/IntronSlopes/
    Ignored:    code/Misc/
    Ignored:    code/MiscCountTables/
    Ignored:    code/Multiqc/
    Ignored:    code/Multiqc_chRNA/
    Ignored:    code/PeakCalling/
    Ignored:    code/Phenotypes/
    Ignored:    code/PlotGruberQTLs/
    Ignored:    code/PlotQTLs/
    Ignored:    code/ProCapAnalysis/
    Ignored:    code/QC/
    Ignored:    code/QTLs/
    Ignored:    code/ReferenceGenome/
    Ignored:    code/Rplots.pdf
    Ignored:    code/Session.vim
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TODO
    Ignored:    code/Tehranchi/
    Ignored:    code/bigwigs/
    Ignored:    code/bigwigs_FromNonWASPFilteredReads/
    Ignored:    code/config/.DS_Store
    Ignored:    code/config/._.DS_Store
    Ignored:    code/config/.ipynb_checkpoints/
    Ignored:    code/debug.ipynb
    Ignored:    code/debug_python.ipynb
    Ignored:    code/deepTools/
    Ignored:    code/featureCounts/
    Ignored:    code/gwas_summary_stats/
    Ignored:    code/hyprcoloc/
    Ignored:    code/igv_session.xml
    Ignored:    code/logs/
    Ignored:    code/notebooks/.ipynb_checkpoints/
    Ignored:    code/rules/.CalculatePi1.smk.swp
    Ignored:    code/rules/.ipynb_checkpoints/
    Ignored:    code/rules/OldRules/
    Ignored:    code/rules/notebooks/
    Ignored:    code/scratch/
    Ignored:    code/scripts/.ipynb_checkpoints/
    Ignored:    code/scripts/GTFtools_0.8.0/
    Ignored:    code/scripts/__pycache__/
    Ignored:    code/scripts/liftOverBedpe/liftOverBedpe.py
    Ignored:    code/snakemake.log
    Ignored:    code/snakemake.sbatch.log
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/._20220414203249_JASPAR2022_combined_matrices_25818_jaspar.txt
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-10.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-11.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-2.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-3.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-4.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-5.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-6.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-7.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-8.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022.csv

Untracked files:
    Untracked:  analysis/20220713_PlotHeatmapManyWays.Rmd
    Untracked:  code/snakemake_profiles/slurm/__pycache__/
    Untracked:  output/hyprcoloc_results/ForColoc/MolColocStandard/pairwisecor.txt.gz
    Untracked:  output/hyprcoloc_results/ForColoc/MolColocTest_001/pairwisecor.txt.gz
    Untracked:  output/hyprcoloc_results/ForColoc/MolColocTest_005/pairwisecor.txt.gz
    Untracked:  output/hyprcoloc_results/ForColoc/MolColocTest_01/pairwisecor.txt.gz
    Untracked:  output/hyprcoloc_results/ForColoc/MolColocTesteQTL_001/pairwisecor.txt.gz
    Untracked:  output/hyprcoloc_results/ForColoc/MolColocTesteQTL_005/pairwisecor.txt.gz
    Untracked:  output/hyprcoloc_results/ForColoc/MolColocTesteQTL_01/pairwisecor.txt.gz

Unstaged changes:
    Modified:   code/scripts/GenometracksByGenotype
    Modified:   data/Phenotypes_recode_for_Plotting.txt
    Modified:   output/hyprcoloc_results/ForColoc/MolColocStandard/hyprcoloc.results.OnlyColocalized.Stats.txt.gz
    Modified:   output/hyprcoloc_results/ForColoc/MolColocStandard/hyprcoloc.results.txt.gz
    Modified:   output/hyprcoloc_results/ForColoc/MolColocStandard/snpscores.txt.gz
    Modified:   output/hyprcoloc_results/ForColoc/MolColocTest_001/hyprcoloc.results.OnlyColocalized.Stats.txt.gz
    Modified:   output/hyprcoloc_results/ForColoc/MolColocTest_001/hyprcoloc.results.txt.gz
    Modified:   output/hyprcoloc_results/ForColoc/MolColocTest_001/snpscores.txt.gz
    Modified:   output/hyprcoloc_results/ForColoc/MolColocTest_005/hyprcoloc.results.OnlyColocalized.Stats.txt.gz
    Modified:   output/hyprcoloc_results/ForColoc/MolColocTest_005/hyprcoloc.results.txt.gz
    Modified:   output/hyprcoloc_results/ForColoc/MolColocTest_005/snpscores.txt.gz
    Modified:   output/hyprcoloc_results/ForColoc/MolColocTest_01/hyprcoloc.results.OnlyColocalized.Stats.txt.gz
    Modified:   output/hyprcoloc_results/ForColoc/MolColocTest_01/hyprcoloc.results.txt.gz
    Modified:   output/hyprcoloc_results/ForColoc/MolColocTest_01/snpscores.txt.gz
    Modified:   output/hyprcoloc_results/ForColoc/MolColocTesteQTL_001/hyprcoloc.results.OnlyColocalized.Stats.txt.gz
    Modified:   output/hyprcoloc_results/ForColoc/MolColocTesteQTL_001/hyprcoloc.results.txt.gz
    Modified:   output/hyprcoloc_results/ForColoc/MolColocTesteQTL_001/snpscores.txt.gz
    Modified:   output/hyprcoloc_results/ForColoc/MolColocTesteQTL_005/hyprcoloc.results.OnlyColocalized.Stats.txt.gz
    Modified:   output/hyprcoloc_results/ForColoc/MolColocTesteQTL_005/hyprcoloc.results.txt.gz
    Modified:   output/hyprcoloc_results/ForColoc/MolColocTesteQTL_005/snpscores.txt.gz
    Modified:   output/hyprcoloc_results/ForColoc/MolColocTesteQTL_01/hyprcoloc.results.OnlyColocalized.Stats.txt.gz
    Modified:   output/hyprcoloc_results/ForColoc/MolColocTesteQTL_01/hyprcoloc.results.txt.gz
    Modified:   output/hyprcoloc_results/ForColoc/MolColocTesteQTL_01/snpscores.txt.gz

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="intro" class="section level2">
<h2>Intro</h2>
<p>I’ve been making heatmaps to communicate the fraction of xQTLs (eg eQTLs/sQTLs/chromatinQTLs) that colocalize with yQTLs. There are many slightly different methods I’ve been using to make this heatmap, with slightly different interpretations. In this notebook, I will make lots of those heatmaps to compare the different methods.</p>
<p>Here is a list of the different methods I have been using, with the names I am calling these methods. Note that for all of these methods I can consider using different statistical thresholds for calling something a QTL, so in this notebook I’ll explore that too.</p>
<ul>
<li><em>Pi1</em>: “What is the pi1 statistic for yQTLs (ascertainment phenotype), in xQTLs (discovery phenotype)”. I can use different FDR thresholds for x (discovery phenotype)</li>
<li><em>ColocalizationRate</em>: “What fraction of xQTLs have at least one colocalizing yQTL”.</li>
<li><em>ColocalizationRateAmongAttempted</em>: “When there is both a xQTL and yQTL around the same gene, how often do they colocalize”</li>
<li><em>SpearmanCorrealtionThreshold</em>: What fraction of xQTLs have at least one yQTL with correlated association signal (spearman correlation coef &gt; Treshold)</li>
<li><em>SpearmanCorrealtionThresholdAmongAttempted</em>: When there is both a xQTL and yQTL around the same gene, how often do they have correlation association signals (spearman correlation coef &gt; Treshold)</li>
</ul>
<pre class="r"><code>library(tidyverse)
library(viridis)
library(gplots)
library(data.table)
library(qvalue)
library(purrr)
library(GGally)
library(pROC)

CalculatePi1 &lt;- function (dat.in) {
  return(tryCatch(1-qvalue(dat.in$Pvals.For.Pi1)$pi0, error=function(e) NULL))
}</code></pre>
<div id="pi1" class="section level3">
<h3>pi1</h3>
<p>First let’s read in data for Pi1 method</p>
<pre class="r"><code>FilesChunks &lt;- paste0(&quot;../code/scratch/PairwisePi1Traits.P.&quot;, 1:10, &quot;.txt.gz&quot;)


dat &lt;- lapply(FilesChunks, fread, sep=&#39;\t&#39;) %&gt;%
  bind_rows()

RecodeDat &lt;- read_tsv(&quot;../data/Phenotypes_recode_for_Plotting.txt&quot;)

RecodeVec &lt;- RecodeDat %&gt;%
  select(PC, ShorterAlias) %&gt;%
  deframe()

RecodeIncludePCs &lt;- RecodeDat %&gt;%
  filter(Include) %&gt;%
  pull(PC)

colnames(dat)</code></pre>
<pre><code> [1] &quot;PC1&quot;                      &quot;P1&quot;                      
 [3] &quot;GeneLocus&quot;                &quot;p_permutation.x&quot;         
 [5] &quot;singletrait_topvar.x&quot;     &quot;singletrait_topvar_chr.x&quot;
 [7] &quot;singletrait_topvar_pos.x&quot; &quot;FDR.x&quot;                   
 [9] &quot;PC2&quot;                      &quot;P2&quot;                      
[11] &quot;p_permutation.y&quot;          &quot;singletrait_topvar.y&quot;    
[13] &quot;singletrait_topvar_chr.y&quot; &quot;singletrait_topvar_pos.y&quot;
[15] &quot;FDR.y&quot;                    &quot;trait.x.p.in.y&quot;          </code></pre>
<p>I also need to simulate the null distribution to deal with the one to many problem…</p>
<pre class="r"><code>MaxSampleSizeToCreateANull &lt;- 100
NSamplesToEstimateDistribution &lt;- 10000
NullSimulatedTestStats &lt;- matrix(nrow=MaxSampleSizeToCreateANull, ncol=NSamplesToEstimateDistribution)
rownames(NullSimulatedTestStats) &lt;- paste0(&quot;runif_samplesize&quot;, 1:MaxSampleSizeToCreateANull)
colnames(NullSimulatedTestStats) &lt;- paste0(&quot;Sample_&quot;, 1:NSamplesToEstimateDistribution)


for (i in 1:MaxSampleSizeToCreateANull){
  SampleSizeFromUniform &lt;- i
  SampledDat &lt;- matrix(runif(SampleSizeFromUniform*NSamplesToEstimateDistribution), nrow=NSamplesToEstimateDistribution)
  SampleDatNullTestStatistics &lt;- -log10(apply(SampledDat, 1, min))
  NullSimulatedTestStats[i,] &lt;- SampleDatNullTestStatistics
}


NullSimulatedTestStats %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;runif_samplesize&quot;) %&gt;%
  slice(1:20) %&gt;%
  # mutate(runif_samplesize = as.numeric(str_remove(runif_samplesize, &quot;runif_samplesize&quot;))) %&gt;%
  gather(key=&quot;Sample&quot;, value=&quot;value&quot;, -runif_samplesize) %&gt;%
  ggplot(aes(x=value, color=runif_samplesize)) +
  geom_density() +
  theme_bw()</code></pre>
<p><img src="figure/20220713_PlotHeatmapManyWays.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ecdf.functions &lt;- apply(NullSimulatedTestStats, 1, ecdf)
ecdf.functions[[1]](1)</code></pre>
<pre><code>[1] 0.9036</code></pre>
<p>Now calculate pi1</p>
<pre class="r"><code>dat.split &lt;- dat %&gt;%
  filter(PC1 %in% RecodeIncludePCs) %&gt;%
  filter(PC2 %in% RecodeIncludePCs) %&gt;%
  mutate(PC1 = recode(PC1, !!!RecodeVec)) %&gt;%
  mutate(PC2 = recode(PC2, !!!RecodeVec)) %&gt;%
  group_by(PC1, P1, PC2) %&gt;%
  mutate(test.stat.obs = -log10(min(trait.x.p.in.y))) %&gt;%
  ungroup() %&gt;%
  add_count(PC1, P1, PC2) %&gt;%
  filter(n&lt;=100) %&gt;%
  group_by(PC1, PC2) %&gt;%
  rowwise() %&gt;%
  mutate(Pvals.For.Pi1 = 1-ecdf.functions[[n]](test.stat.obs)) %&gt;%
  ungroup() %&gt;%
  select(PC1, PC2, Pvals.For.Pi1) %&gt;%
  filter(!PC1==PC2) %&gt;%
  split(paste(.$PC1, .$PC2, sep = &quot;;&quot;))

dat.pi1 &lt;- lapply(dat.split, CalculatePi1) %&gt;%
  unlist() %&gt;%
  data.frame(pi1=.) %&gt;%
  rownames_to_column(&quot;PC1_PC2&quot;) %&gt;%
  separate(PC1_PC2, into=c(&quot;PC1&quot;, &quot;PC2&quot;), sep=&#39;;&#39;)

pi.heatmap &lt;- ggplot(dat.pi1, aes(x=PC1, y=PC2, fill=pi1)) +
  geom_raster() +
  geom_text(aes(label=signif(pi1*100, 2)), color=&quot;blue&quot;) +
  scale_fill_viridis(option=&quot;A&quot;, direction = -1, limits=c(0,1)) +
  coord_flip() +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(x=&quot;Discovery QTL phenotype&quot;, y=&quot;Phenotype assessed for overlap&quot;)
pi.heatmap  </code></pre>
<p><img src="figure/20220713_PlotHeatmapManyWays.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Now make the same heatmap but compare discovery FDR of 0.1 (like above), 0.05, 0.01</p>
<pre class="r"><code># Make a function to re-use

CalculatePi1Matrix &lt;- function(dat, DiscoveryFDR=0.1){
  dat.split &lt;- dat %&gt;%
  filter(PC1 %in% RecodeIncludePCs) %&gt;%
  filter(PC2 %in% RecodeIncludePCs) %&gt;%
  filter(FDR.x &lt;= DiscoveryFDR) %&gt;%
  mutate(PC1 = recode(PC1, !!!RecodeVec)) %&gt;%
  mutate(PC2 = recode(PC2, !!!RecodeVec)) %&gt;%
  group_by(PC1, P1, PC2) %&gt;%
  mutate(test.stat.obs = -log10(min(trait.x.p.in.y))) %&gt;%
  ungroup() %&gt;%
  add_count(PC1, P1, PC2) %&gt;%
  filter(n&lt;=100) %&gt;%
  group_by(PC1, PC2) %&gt;%
  rowwise() %&gt;%
  mutate(Pvals.For.Pi1 = 1-ecdf.functions[[n]](test.stat.obs)) %&gt;%
  ungroup() %&gt;%
  select(PC1, PC2, Pvals.For.Pi1) %&gt;%
  filter(!PC1==PC2) %&gt;%
  split(paste(.$PC1, .$PC2, sep = &quot;;&quot;))

dat.pi1 &lt;- lapply(dat.split, CalculatePi1) %&gt;%
  unlist() %&gt;%
  data.frame(pi1=.) %&gt;%
  rownames_to_column(&quot;PC1_PC2&quot;) %&gt;%
  separate(PC1_PC2, into=c(&quot;PC1&quot;, &quot;PC2&quot;), sep=&#39;;&#39;) %&gt;%
  mutate(DiscoveryFDR = DiscoveryFDR)
return(dat.pi1)
}

dat.pi.AtThresholds &lt;- bind_rows(
  CalculatePi1Matrix(dat, DiscoveryFDR=0.1),
  CalculatePi1Matrix(dat, DiscoveryFDR=0.05),
  CalculatePi1Matrix(dat, DiscoveryFDR=0.01),
)

pi.heatmap &lt;- ggplot(dat.pi.AtThresholds, aes(x=PC1, y=PC2, fill=pi1)) +
  geom_raster() +
  geom_text(aes(label=signif(pi1*100, 2)), color=&quot;blue&quot;, size=2) +
  scale_fill_viridis(option=&quot;A&quot;, direction = -1, limits=c(0,1)) +
  facet_wrap(~DiscoveryFDR) +
  coord_flip() +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(x=&quot;Discovery QTL phenotype&quot;, y=&quot;Phenotype assessed for overlap&quot;, &quot;Pi1 of xQTL among Discovery QTL classes&quot;)
pi.heatmap</code></pre>
<p><img src="figure/20220713_PlotHeatmapManyWays.Rmd/unnamed-chunk-5-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="colocalization-rate" class="section level3">
<h3>Colocalization Rate</h3>
<pre class="r"><code>dat.coloc &lt;- Sys.glob(&quot;../code/hyprcoloc/Results/ForColoc/MolColocTest*_*/results.txt.gz&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/hyprcoloc/Results/ForColoc/MolColocTest(.*?)_(.+?)/results.txt.gz&quot;, &quot;\\1_0.\\2&quot;)) %&gt;%
  lapply(read_tsv) %&gt;%
  bind_rows(.id=&quot;Threshold&quot;)

dat.coloc.tidy &lt;- dat.coloc %&gt;%
  unite(Locus, GeneLocus, Threshold, sep = &quot;:&quot;) %&gt;%
  left_join(., ., by = &quot;Locus&quot;) %&gt;% 
    filter(Trait.x != Trait.y) %&gt;% 
    separate(Trait.x, into = c(&quot;PC1&quot;,&quot;P1&quot;), sep = &quot;[, ;]&quot;, remove=F) %&gt;% 
    separate(Trait.y, into = c(&quot;PC2&quot;,&quot;P2&quot;), sep = &quot;[, ;]&quot;, remove=F) %&gt;% 
    rowwise() %&gt;%
    mutate(PC_ClassPair = paste(PC1, PC2)) %&gt;%
    ungroup() %&gt;%
    # pull(PC_ClassPair) %&gt;% unique()
    mutate(IsColocalizedPair = HyprcolocIteration.x == HyprcolocIteration.y) %&gt;%
    replace_na(list(IsColocalizedPair = FALSE))

NumQTLs &lt;- dat.coloc.tidy %&gt;%
  filter(PC1 %in% RecodeIncludePCs) %&gt;%
  mutate(PC1 = recode(PC1, !!!RecodeVec)) %&gt;%
  mutate(PC2 = recode(PC2, !!!RecodeVec)) %&gt;%  separate(Locus, into=c(&quot;Locus&quot;, &quot;Threshold&quot;), sep = &quot;:&quot;) %&gt;%
  distinct(Locus, Threshold, PC1, P1) %&gt;%
  count(Threshold, PC1)

coloc.rate.dat &lt;- dat.coloc.tidy %&gt;%
  filter(PC1 %in% RecodeIncludePCs) %&gt;%
  filter(PC2 %in% RecodeIncludePCs) %&gt;%
  mutate(PC1 = recode(PC1, !!!RecodeVec)) %&gt;%
  mutate(PC2 = recode(PC2, !!!RecodeVec)) %&gt;%
  filter(IsColocalizedPair) %&gt;%
  separate(Locus, into=c(&quot;Locus&quot;, &quot;Threshold&quot;), sep = &quot;:&quot;) %&gt;%
  mutate(PC2 = as.factor(PC2)) %&gt;%
  count(Locus, Threshold, PC1, P1, PC2, .drop=F) %&gt;%
  mutate(ContainsAtleastOneColoc = n&gt;0) %&gt;%
  group_by(PC1, PC2, Threshold) %&gt;%
  summarise(SumPC1sWithAtLeast1PC2Coloc = sum(ContainsAtleastOneColoc)) %&gt;%
  left_join(NumQTLs, by=c(&quot;Threshold&quot;, &quot;PC1&quot;)) %&gt;%
  mutate(FractionColocs = SumPC1sWithAtLeast1PC2Coloc/n)

coloc.rate.dat %&gt;%
  mutate(Threshold = str_remove(Threshold, &quot;^_&quot;)) %&gt;%
  filter(Threshold %in% c(&quot;0.001&quot;, &quot;0.005&quot;, &quot;0.01&quot;)) %&gt;%
ggplot(aes(x=PC1, y=PC2, fill=FractionColocs)) +
  geom_raster() +
  geom_text(aes(label=signif(FractionColocs*100, 2)), color=&quot;blue&quot;, size=2) +
  scale_fill_viridis(option=&quot;A&quot;, direction = -1, limits=c(0,1)) +
  facet_wrap(~Threshold) +
  coord_flip() +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  facet_wrap(~Threshold) +
  labs(x=&quot;Discovery QTL phenotype&quot;, y=&quot;Phenotype assessed for colocalization&quot;, &quot;Percent Discovery QTLs with at least one colocalizing xQTL&quot;)</code></pre>
<p><img src="figure/20220713_PlotHeatmapManyWays.Rmd/unnamed-chunk-6-1.png" width="960" style="display: block; margin: auto;" /> ### ColocalizationRateAmongAttempted</p>
<pre class="r"><code>coloc.rate.AmongAttempted.dat &lt;- dat.coloc.tidy %&gt;%
  filter(PC1 %in% RecodeIncludePCs) %&gt;%
  filter(PC2 %in% RecodeIncludePCs) %&gt;%
  mutate(PC1 = recode(PC1, !!!RecodeVec)) %&gt;%
  mutate(PC2 = recode(PC2, !!!RecodeVec)) %&gt;%
  separate(Locus, into=c(&quot;Locus&quot;, &quot;Threshold&quot;), sep = &quot;:&quot;) %&gt;%
  group_by(PC1, PC2, Threshold) %&gt;%
  summarise(FractionColocs = sum(IsColocalizedPair)/n())

coloc.rate.AmongAttempted.dat %&gt;%
  mutate(Threshold = str_remove(Threshold, &quot;^_&quot;)) %&gt;%
  filter(Threshold %in% c(&quot;0.001&quot;, &quot;0.005&quot;, &quot;0.01&quot;)) %&gt;%
ggplot(aes(x=PC1, y=PC2, fill=FractionColocs)) +
  geom_raster() +
  geom_text(aes(label=signif(FractionColocs*100, 2)), color=&quot;blue&quot;, size=2) +
  scale_fill_viridis(option=&quot;A&quot;, direction = -1, limits=c(0,1)) +
  facet_wrap(~Threshold) +
  coord_flip() +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  facet_wrap(~Threshold) +
  labs(x=&quot;Phenotype_x&quot;, y=&quot;Phenotype_y&quot;, title=&quot;Percent succesful colocalizations among attempted&quot;)</code></pre>
<p><img src="figure/20220713_PlotHeatmapManyWays.Rmd/unnamed-chunk-7-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="spearmancorrealtionthresholdamongattempted" class="section level2">
<h2>SpearmanCorrealtionThresholdAmongAttempted</h2>
<p>First read in the data, and evaluate how spearman correlation coefficient of association signal strength between two traits works as a predictor for colocalization (binary response). Use this to pick an ‘optimal’ threshold for calling this as “colocalized” using this spearman correlation method.</p>
<pre class="r"><code>pairwise.cor.results &lt;- fread(&quot;../code/hyprcoloc/Results/ForColoc/MolColocStandard/pairwisecor.txt.gz&quot;, col.names = c(&quot;Trait1&quot;,&quot;Trait2&quot;, &quot;cor.z.pearson&quot;, &quot;cor.z.spearman&quot;, &quot;cor.logp.pearson&quot;, &quot;GeneLocus&quot;))

spearman.dat &lt;- dat.coloc.tidy %&gt;%
  separate(Locus, into=c(&quot;GeneLocus&quot;, &quot;Threshold&quot;), sep = &quot;:&quot;) %&gt;%
  inner_join(pairwise.cor.results, by = c(&quot;GeneLocus&quot;, &quot;Trait.x&quot;=&quot;Trait1&quot;, &quot;Trait.y&quot;=&quot;Trait2&quot;))

ROC.results &lt;- spearman.dat %&gt;%
  filter(Threshold == &quot;_0.001&quot;) %&gt;%
  roc(formula=IsColocalizedPair~cor.z.spearman, data=.)

ROC.results$auc</code></pre>
<pre><code>Area under the curve: 0.9025</code></pre>
<pre class="r"><code>BestThreshold &lt;- coords(ROC.results, &quot;best&quot;, &quot;threshold&quot;)
BestThreshold</code></pre>
<pre><code>  threshold specificity sensitivity 
  0.3706721   0.7851396   0.8835937 </code></pre>
<pre class="r"><code>plot(ROC.results)
abline(v=BestThreshold[&quot;specificity&quot;])
abline(h=BestThreshold[&quot;sensitivity&quot;])</code></pre>
<p><img src="figure/20220713_PlotHeatmapManyWays.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>spearman.dat %&gt;%
  filter(str_detect(Threshold, &quot;^_&quot;)) %&gt;%
  select(Threshold, IsColocalizedPair, cor.z.spearman) %&gt;%
  ggplot(aes(x=cor.z.spearman, fill=IsColocalizedPair)) +
  geom_histogram() +
  geom_vline(xintercept = BestThreshold[&quot;threshold&quot;]) +
  facet_grid(rows=vars(IsColocalizedPair), cols=vars(Threshold), scales = &quot;free&quot;) +
  theme_bw() +
  labs(x=&quot;SpearmanCorrelationCoefficient&quot;, y=&quot;Number QTL pairs&quot;, title=&quot;Spearman correlation coefficient can\ndistinguish colocalized from uncolocalized trait pairs&quot;)</code></pre>
<p><img src="figure/20220713_PlotHeatmapManyWays.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>spearman.dat %&gt;%
  filter(str_detect(Threshold, &quot;^_&quot;)) %&gt;%
  mutate(IsTierTwoColocalized = cor.z.spearman &gt; BestThreshold[&quot;threshold&quot;]) %&gt;%
  count(IsTierTwoColocalized, IsColocalizedPair, Threshold) %&gt;%
  ggplot(aes(x=IsTierTwoColocalized, y=n, fill=IsColocalizedPair)) +
  geom_col() +
  facet_wrap(~Threshold) +
  labs(Title=&quot;Number colocalizations by methods&quot;, y=&quot;Number QTL trait pairs&quot;, x=&quot;Is Colocalized By Spearman Threshold&quot;) +
  theme_bw()</code></pre>
<p><img src="figure/20220713_PlotHeatmapManyWays.Rmd/unnamed-chunk-8-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok now that I’ve established some sort of reasonable threshold for calling something “colocalized” just by looking at spearman correlation of association signals, let’s make the heatmap. Note that TODO is to look at some of the trait pairs that colocalized by hyprcoloc and not spearman or vise-versa.</p>
<pre class="r"><code>spearman.dat %&gt;%
  filter(PC1 %in% RecodeIncludePCs) %&gt;%
  filter(PC2 %in% RecodeIncludePCs) %&gt;%
  mutate(PC1 = recode(PC1, !!!RecodeVec)) %&gt;%
  mutate(PC2 = recode(PC2, !!!RecodeVec)) %&gt;%
  mutate(Threshold = str_remove(Threshold, &quot;^_&quot;)) %&gt;%
  filter(Threshold %in% c(&quot;0.001&quot;, &quot;0.005&quot;, &quot;0.01&quot;)) %&gt;%
  group_by(PC1, PC2, Threshold) %&gt;%
  summarise(FractionColocs = sum(cor.z.spearman &gt; BestThreshold[&quot;threshold&quot;])/n()) %&gt;%
  ggplot(aes(x=PC1, y=PC2, fill=FractionColocs)) +
  geom_raster() +
  geom_text(aes(label=signif(FractionColocs*100, 2)), color=&quot;blue&quot;, size=2) +
  scale_fill_viridis(option=&quot;A&quot;, direction = -1, limits=c(0,1)) +
  facet_wrap(~Threshold) +
  coord_flip() +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  facet_wrap(~Threshold) +
  labs(x=&quot;Phenotype_x&quot;, y=&quot;Phenotype_y&quot;, title=&quot;Percent spearman method colocalizations of attempted&quot;)</code></pre>
<p><img src="figure/20220713_PlotHeatmapManyWays.Rmd/unnamed-chunk-9-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="spearmancorrealtionthreshold" class="section level2">
<h2>SpearmanCorrealtionThreshold</h2>
<p>Similar to plot above, but not symetrical because it is asking something a bit different, which I’ve explained above.</p>
<pre class="r"><code>NumQTLs &lt;- spearman.dat %&gt;%
  filter(PC1 %in% RecodeIncludePCs) %&gt;%
  mutate(PC1 = recode(PC1, !!!RecodeVec)) %&gt;%
  mutate(PC2 = recode(PC2, !!!RecodeVec)) %&gt;%  
  distinct(GeneLocus, Threshold, PC1, P1) %&gt;%
  count(Threshold, PC1)

spearman.coloc.rate.dat &lt;- spearman.dat %&gt;%
  filter(PC1 %in% RecodeIncludePCs) %&gt;%
  filter(PC2 %in% RecodeIncludePCs) %&gt;%
  mutate(PC1 = recode(PC1, !!!RecodeVec)) %&gt;%
  mutate(PC2 = recode(PC2, !!!RecodeVec)) %&gt;%
  filter(cor.z.spearman &gt; BestThreshold[&quot;threshold&quot;]) %&gt;%
  mutate(PC2 = as.factor(PC2)) %&gt;%
  count(GeneLocus, Threshold, PC1, P1, PC2, .drop=F) %&gt;%
  mutate(ContainsAtleastOneColoc = n&gt;0) %&gt;%
  group_by(PC1, PC2, Threshold) %&gt;%
  summarise(SumPC1sWithAtLeast1PC2Coloc = sum(ContainsAtleastOneColoc)) %&gt;%
  left_join(NumQTLs, by=c(&quot;Threshold&quot;, &quot;PC1&quot;)) %&gt;%
  mutate(FractionColocs = SumPC1sWithAtLeast1PC2Coloc/n)

spearman.coloc.rate.dat %&gt;%
  mutate(Threshold = str_remove(Threshold, &quot;^_&quot;)) %&gt;%
  filter(Threshold %in% c(&quot;0.001&quot;, &quot;0.005&quot;, &quot;0.01&quot;)) %&gt;%
ggplot(aes(x=PC1, y=PC2, fill=FractionColocs)) +
  geom_raster() +
  geom_text(aes(label=signif(FractionColocs*100, 2)), color=&quot;blue&quot;, size=2) +
  scale_fill_viridis(option=&quot;A&quot;, direction = -1, limits=c(0,1)) +
  facet_wrap(~Threshold) +
  coord_flip() +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  facet_wrap(~Threshold) +
  labs(x=&quot;Discovery QTL phenotype&quot;, y=&quot;Phenotype assessed for colocalization&quot;, &quot;Percent Discovery QTLs with at least one colocalizing xQTL&quot;, title=&quot;Colocalization determined by spearman correlation coef&quot;)</code></pre>
<p><img src="figure/20220713_PlotHeatmapManyWays.Rmd/unnamed-chunk-10-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="comparing-methods" class="section level2">
<h2>Comparing Methods</h2>
<p>As a quick way to visually compare all these different heatmaps, I’ll plot pariwise correlations of the heatmap values aross these many different methods.</p>
<ul>
<li>Still TODO…</li>
</ul>
</div>
<div id="more-exploration" class="section level2">
<h2>More exploration</h2>
<ul>
<li>One plot that I want to re-create and re-plot is the histogram or qq-plot of reciprocal pvalues (like for calculating pi1) but grouped/colored by whether the trait pairs are formally colocalized. The idea is to assess how often there is LD bleeding of signals that result in sharing by pi1, but not necessarily colocalization due to same causal variant.</li>
<li>Explore some biological ideas to highlight</li>
<li>how many chRNA specific sQTLs explain eQTLs. Do effect size directions make sense with model of splicing of cryptic intron –&gt; NMD –&gt; down-regulation. This bin of QTLs should be enriched among the QTLs that are not explained by chromatin</li>
<li>break chromatin up into promoter vs distal enhancer.</li>
</ul>
<p>TODO:</p>
<ul>
<li></li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] pROC_1.15.0       GGally_1.4.0      qvalue_2.16.0     data.table_1.14.2
 [5] gplots_3.0.1.1    viridis_0.5.1     viridisLite_0.3.0 forcats_0.4.0    
 [9] stringr_1.4.0     dplyr_1.0.9       purrr_0.3.4       readr_1.3.1      
[13] tidyr_1.2.0       tibble_3.1.7      ggplot2_3.3.6     tidyverse_1.3.0  

loaded via a namespace (and not attached):
 [1] httr_1.4.1         jsonlite_1.6       splines_3.6.1      R.utils_2.9.0     
 [5] modelr_0.1.8       gtools_3.9.2.2     assertthat_0.2.1   highr_0.9         
 [9] cellranger_1.1.0   yaml_2.2.0         pillar_1.7.0       backports_1.4.1   
[13] glue_1.6.2         digest_0.6.20      RColorBrewer_1.1-2 promises_1.0.1    
[17] rvest_0.3.5        colorspace_1.4-1   R.oo_1.22.0        htmltools_0.3.6   
[21] httpuv_1.5.1       plyr_1.8.4         pkgconfig_2.0.2    broom_1.0.0       
[25] haven_2.3.1        scales_1.1.0       gdata_2.18.0       later_0.8.0       
[29] git2r_0.26.1       farver_2.1.0       generics_0.1.3     ellipsis_0.3.2    
[33] withr_2.4.1        cli_3.3.0          magrittr_1.5       crayon_1.3.4      
[37] readxl_1.3.1       evaluate_0.15      R.methodsS3_1.7.1  fs_1.3.1          
[41] fansi_0.4.0        xml2_1.3.2         tools_3.6.1        hms_0.5.3         
[45] lifecycle_1.0.1    munsell_0.5.0      reprex_0.3.0       compiler_3.6.1    
[49] caTools_1.17.1.2   rlang_1.0.3        grid_3.6.1         rstudioapi_0.10   
[53] labeling_0.3       bitops_1.0-6       rmarkdown_1.13     gtable_0.3.0      
[57] DBI_1.1.0          reshape_0.8.8      reshape2_1.4.3     R6_2.4.0          
[61] gridExtra_2.3      lubridate_1.7.4    knitr_1.39         utf8_1.1.4        
[65] workflowr_1.6.2    rprojroot_2.0.2    KernSmooth_2.23-15 stringi_1.4.3     
[69] Rcpp_1.0.5         vctrs_0.4.1        dbplyr_1.4.2       tidyselect_1.1.2  
[73] xfun_0.31         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
