<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>20220920_Explore_chRNA_sQTLs</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">20220920_Explore_chRNA_sQTLs</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-09-22
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>ChromatinSplicingQTLs/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstreee9163e8a34fe090f28f17edee6c484ccd69c8d40targetblanke9163e8a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/e9163e8a34fe090f28f17edee6c484ccd69c8d40" target="_blank">e9163e8</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstreee9163e8a34fe090f28f17edee6c484ccd69c8d40targetblanke9163e8a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/e9163e8a34fe090f28f17edee6c484ccd69c8d40" target="_blank">e9163e8</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store
    Ignored:    code/.RData
    Ignored:    code/._.DS_Store
    Ignored:    code/._README.md
    Ignored:    code/._report.html
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    code/.snakemake/
    Ignored:    code/APA_Processing/
    Ignored:    code/Alignments/
    Ignored:    code/ChromHMM/
    Ignored:    code/ENCODE/
    Ignored:    code/ExpressionAnalysis/
    Ignored:    code/FastqFastp/
    Ignored:    code/FastqFastpSE/
    Ignored:    code/Genotypes/
    Ignored:    code/IntronSlopes/
    Ignored:    code/Misc/
    Ignored:    code/MiscCountTables/
    Ignored:    code/Multiqc/
    Ignored:    code/Multiqc_chRNA/
    Ignored:    code/NonCodingRNA_annotation/
    Ignored:    code/PeakCalling/
    Ignored:    code/Phenotypes/
    Ignored:    code/PlotGruberQTLs/
    Ignored:    code/PlotQTLs/
    Ignored:    code/ProCapAnalysis/
    Ignored:    code/QC/
    Ignored:    code/QTL_SNP_Enrichment/
    Ignored:    code/QTLs/
    Ignored:    code/ReferenceGenome/
    Ignored:    code/Rplots.pdf
    Ignored:    code/Session.vim
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TODO
    Ignored:    code/Tehranchi/
    Ignored:    code/bigwigs/
    Ignored:    code/bigwigs_FromNonWASPFilteredReads/
    Ignored:    code/config/.DS_Store
    Ignored:    code/config/._.DS_Store
    Ignored:    code/config/.ipynb_checkpoints/
    Ignored:    code/debug.ipynb
    Ignored:    code/debug_python.ipynb
    Ignored:    code/deepTools/
    Ignored:    code/featureCounts/
    Ignored:    code/gwas_summary_stats/
    Ignored:    code/hyprcoloc/
    Ignored:    code/igv_session.xml
    Ignored:    code/log
    Ignored:    code/logs/
    Ignored:    code/notebooks/.ipynb_checkpoints/
    Ignored:    code/rules/.ipynb_checkpoints/
    Ignored:    code/rules/OldRules/
    Ignored:    code/rules/notebooks/
    Ignored:    code/scratch/
    Ignored:    code/scripts/.ipynb_checkpoints/
    Ignored:    code/scripts/GTFtools_0.8.0/
    Ignored:    code/scripts/__pycache__/
    Ignored:    code/scripts/liftOverBedpe/liftOverBedpe.py
    Ignored:    code/snakemake.log
    Ignored:    code/snakemake.sbatch.log
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/._20220414203249_JASPAR2022_combined_matrices_25818_jaspar.txt
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-10.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-11.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-2.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-3.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-4.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-5.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-6.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-7.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-8.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022.csv

Untracked files:
    Untracked:  analysis/20220920_Explore_chRNA_sQTLs.Rmd
    Untracked:  code/snakemake_profiles/slurm/__pycache__/

Unstaged changes:
    Modified:   analysis/20220713_RPKM_v_TPM.Rmd
    Deleted:    code/envs/spliceq.yml
    Modified:   code/scripts/GenometracksByGenotype
    Modified:   code/scripts/Plot_QTLPermutationTestNumSig.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="intro" class="section level2">
<h2>Intro</h2>
<p>Carlos and I recently re-ran most of the QTL calling and colocalization pipelines, with some new phenotype tables, including his new de-novo called transcription units from our chRNA data. I expect this to improve some things. Let’s explore the data in many ways…</p>
</div>
<div id="analysis" class="section level2">
<h2>Analysis</h2>
<div id="count-number-qtls" class="section level3">
<h3>Count number QTLs</h3>
<p>…From permutation pass, with cis-eqtl test windows centered around the test phenotype.</p>
<pre class="r"><code>library(tidyverse)
library(data.table)

Files &lt;- Sys.glob(&quot;../code/QTLs/QTLTools/*/PermutationPass.FDR_Added.txt.gz&quot;)
Files</code></pre>
<pre><code> [1] &quot;../code/QTLs/QTLTools/CTCF/PermutationPass.FDR_Added.txt.gz&quot;                               
 [2] &quot;../code/QTLs/QTLTools/Expression.Splicing.Subset_YRI/PermutationPass.FDR_Added.txt.gz&quot;     
 [3] &quot;../code/QTLs/QTLTools/Expression.Splicing/PermutationPass.FDR_Added.txt.gz&quot;                
 [4] &quot;../code/QTLs/QTLTools/H3K27AC/PermutationPass.FDR_Added.txt.gz&quot;                            
 [5] &quot;../code/QTLs/QTLTools/H3K36ME3/PermutationPass.FDR_Added.txt.gz&quot;                           
 [6] &quot;../code/QTLs/QTLTools/H3K4ME1/PermutationPass.FDR_Added.txt.gz&quot;                            
 [7] &quot;../code/QTLs/QTLTools/H3K4ME3/PermutationPass.FDR_Added.txt.gz&quot;                            
 [8] &quot;../code/QTLs/QTLTools/MetabolicLabelled.30min.IER/PermutationPass.FDR_Added.txt.gz&quot;        
 [9] &quot;../code/QTLs/QTLTools/MetabolicLabelled.30min.IR/PermutationPass.FDR_Added.txt.gz&quot;         
[10] &quot;../code/QTLs/QTLTools/MetabolicLabelled.30min.IRjunctions/PermutationPass.FDR_Added.txt.gz&quot;
[11] &quot;../code/QTLs/QTLTools/MetabolicLabelled.30min.Splicing/PermutationPass.FDR_Added.txt.gz&quot;   
[12] &quot;../code/QTLs/QTLTools/MetabolicLabelled.30min/PermutationPass.FDR_Added.txt.gz&quot;            
[13] &quot;../code/QTLs/QTLTools/MetabolicLabelled.60min.IER/PermutationPass.FDR_Added.txt.gz&quot;        
[14] &quot;../code/QTLs/QTLTools/MetabolicLabelled.60min.IR/PermutationPass.FDR_Added.txt.gz&quot;         
[15] &quot;../code/QTLs/QTLTools/MetabolicLabelled.60min.IRjunctions/PermutationPass.FDR_Added.txt.gz&quot;
[16] &quot;../code/QTLs/QTLTools/MetabolicLabelled.60min.Splicing/PermutationPass.FDR_Added.txt.gz&quot;   
[17] &quot;../code/QTLs/QTLTools/MetabolicLabelled.60min/PermutationPass.FDR_Added.txt.gz&quot;            
[18] &quot;../code/QTLs/QTLTools/ProCap/PermutationPass.FDR_Added.txt.gz&quot;                             
[19] &quot;../code/QTLs/QTLTools/chRNA.Expression.Splicing/PermutationPass.FDR_Added.txt.gz&quot;          
[20] &quot;../code/QTLs/QTLTools/chRNA.Expression_cheRNA/PermutationPass.FDR_Added.txt.gz&quot;            
[21] &quot;../code/QTLs/QTLTools/chRNA.Expression_eRNA/PermutationPass.FDR_Added.txt.gz&quot;              
[22] &quot;../code/QTLs/QTLTools/chRNA.Expression_lncRNA/PermutationPass.FDR_Added.txt.gz&quot;            
[23] &quot;../code/QTLs/QTLTools/chRNA.Expression_ncRNA/PermutationPass.FDR_Added.txt.gz&quot;             
[24] &quot;../code/QTLs/QTLTools/chRNA.Expression_snoRNA/PermutationPass.FDR_Added.txt.gz&quot;            
[25] &quot;../code/QTLs/QTLTools/chRNA.IER/PermutationPass.FDR_Added.txt.gz&quot;                          
[26] &quot;../code/QTLs/QTLTools/chRNA.IR/PermutationPass.FDR_Added.txt.gz&quot;                           
[27] &quot;../code/QTLs/QTLTools/chRNA.IRjunctions/PermutationPass.FDR_Added.txt.gz&quot;                  
[28] &quot;../code/QTLs/QTLTools/chRNA.Slopes/PermutationPass.FDR_Added.txt.gz&quot;                       
[29] &quot;../code/QTLs/QTLTools/chRNA.Splicing.Order/PermutationPass.FDR_Added.txt.gz&quot;               
[30] &quot;../code/QTLs/QTLTools/chRNA.Splicing/PermutationPass.FDR_Added.txt.gz&quot;                     
[31] &quot;../code/QTLs/QTLTools/polyA.IER.Subset_YRI/PermutationPass.FDR_Added.txt.gz&quot;               
[32] &quot;../code/QTLs/QTLTools/polyA.IER/PermutationPass.FDR_Added.txt.gz&quot;                          
[33] &quot;../code/QTLs/QTLTools/polyA.IR.Subset_YRI/PermutationPass.FDR_Added.txt.gz&quot;                
[34] &quot;../code/QTLs/QTLTools/polyA.IR/PermutationPass.FDR_Added.txt.gz&quot;                           
[35] &quot;../code/QTLs/QTLTools/polyA.IRjunctions/PermutationPass.FDR_Added.txt.gz&quot;                  
[36] &quot;../code/QTLs/QTLTools/polyA.Splicing.Subset_YRI/PermutationPass.FDR_Added.txt.gz&quot;          
[37] &quot;../code/QTLs/QTLTools/polyA.Splicing/PermutationPass.FDR_Added.txt.gz&quot;                     </code></pre>
<p>Ok, actually, I think these files are out dated.</p>
<p>TODO: Make sure we have recently called these as snakemake target files, since Carlos updated these… I’m realizing there are lots and lots of things I just have to re-run from the snakemake before I can get up-to-date output files to do many QC types of analyses. So I’m going to to skip to only doing things with files that already exist as of today to get a sense of what results to expect once we finalize some analyses.</p>
<p>I think will mainly work with the colocalization output files here</p>
</div>
<div id="explore-colocalization-output" class="section level3">
<h3>Explore colocalization output</h3>
<pre class="r"><code>PhenotypeAliases &lt;- read_tsv(&quot;../data/Phenotypes_recode_for_Plotting.txt&quot;)

PC.ShortAliases &lt;- PhenotypeAliases %&gt;%
  dplyr::select(PC, ShorterAlias) %&gt;% deframe()

coloc.results &lt;- read_tsv(&quot;../code/hyprcoloc/Results/ForColoc/MolColocStandard/results.txt.gz&quot;)

coloc.results.tidycolocalized &lt;- read_tsv(&quot;../code/hyprcoloc/Results/ForColoc/MolColocStandard/tidy_results_OnlyColocalized.txt.gz&quot;) %&gt;%
  separate(phenotype_full, into=c(&quot;PC&quot;, &quot;P&quot;), sep=&quot;;&quot;)

# Get all the phenotype classes
coloc.results.tidycolocalized$PC %&gt;% unique()</code></pre>
<pre><code> [1] &quot;polyA.Splicing&quot;                   &quot;MetabolicLabelled.30min&quot;         
 [3] &quot;MetabolicLabelled.60min&quot;          &quot;H3K27AC&quot;                         
 [5] &quot;H3K4ME3&quot;                          &quot;H3K4ME1&quot;                         
 [7] &quot;chRNA.IER&quot;                        &quot;ProCap&quot;                          
 [9] &quot;polyA.Splicing.Subset_YRI&quot;        &quot;CTCF&quot;                            
[11] &quot;chRNA.Splicing.Order&quot;             &quot;chRNA.Expression_ncRNA&quot;          
[13] &quot;Expression.Splicing&quot;              &quot;Expression.Splicing.Subset_YRI&quot;  
[15] &quot;chRNA.Splicing&quot;                   &quot;MetabolicLabelled.30min.Splicing&quot;
[17] &quot;MetabolicLabelled.60min.Splicing&quot; &quot;chRNA.Expression.Splicing&quot;       
[19] &quot;H3K36ME3&quot;                         &quot;polyA.IER&quot;                       
[21] &quot;MetabolicLabelled.30min.IER&quot;      &quot;MetabolicLabelled.60min.IER&quot;     
[23] &quot;polyA.IER.Subset_YRI&quot;             &quot;chRNA.Slopes&quot;                    </code></pre>
<pre class="r"><code>#How many YRI poly A QTLs
coloc.results.tidycolocalized %&gt;%
  filter(PC==&quot;Expression.Splicing.Subset_YRI&quot;)</code></pre>
<pre><code># A tibble: 1,741 × 11
   snp      beta beta_se        p Locus PC    P     iteration ColocPr RegionalPr
   &lt;chr&gt;   &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;     &lt;dbl&gt;   &lt;dbl&gt;      &lt;dbl&gt;
 1 7:121…  0.561  0.119  9.18e- 6 ENSG… Expr… ENSG…         1   0.835      0.897
 2 7:199…  0.553  0.106  1.13e- 6 ENSG… Expr… ENSG…         1   0.308      0.531
 3 7:817…  1.69   0.143  8.35e-20 ENSG… Expr… ENSG…         1   0.896      0.923
 4 2:201…  0.388  0.0812 7.01e- 6 ENSG… Expr… ENSG…         1   0.940      0.963
 5 7:962… -0.335  0.0848 1.59e- 4 ENSG… Expr… ENSG…         3   0.634      0.697
 6 17:72…  0.324  0.0620 1.15e- 6 ENSG… Expr… ENSG…         1   0.959      0.986
 7 7:268…  0.554  0.110  2.71e- 6 ENSG… Expr… ENSG…         2   0.975      0.977
 8 6:822… -0.185  0.0337 4.12e- 7 ENSG… Expr… ENSG…         1   0.621      0.787
 9 11:33…  0.281  0.0475 6.45e- 8 ENSG… Expr… ENSG…         2   0.276      0.528
10 17:35… -0.526  0.108  4.95e- 6 ENSG… Expr… ENSG…         3   0.905      0.954
# … with 1,731 more rows, and 1 more variable: TopSNPFinemapPr &lt;dbl&gt;</code></pre>
<pre class="r"><code>#Of which, how many also contain chromatin
coloc.results.tidycolocalized %&gt;%
  filter(PC==&quot;Expression.Splicing.Subset_YRI&quot;)</code></pre>
<pre><code># A tibble: 1,741 × 11
   snp      beta beta_se        p Locus PC    P     iteration ColocPr RegionalPr
   &lt;chr&gt;   &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;     &lt;dbl&gt;   &lt;dbl&gt;      &lt;dbl&gt;
 1 7:121…  0.561  0.119  9.18e- 6 ENSG… Expr… ENSG…         1   0.835      0.897
 2 7:199…  0.553  0.106  1.13e- 6 ENSG… Expr… ENSG…         1   0.308      0.531
 3 7:817…  1.69   0.143  8.35e-20 ENSG… Expr… ENSG…         1   0.896      0.923
 4 2:201…  0.388  0.0812 7.01e- 6 ENSG… Expr… ENSG…         1   0.940      0.963
 5 7:962… -0.335  0.0848 1.59e- 4 ENSG… Expr… ENSG…         3   0.634      0.697
 6 17:72…  0.324  0.0620 1.15e- 6 ENSG… Expr… ENSG…         1   0.959      0.986
 7 7:268…  0.554  0.110  2.71e- 6 ENSG… Expr… ENSG…         2   0.975      0.977
 8 6:822… -0.185  0.0337 4.12e- 7 ENSG… Expr… ENSG…         1   0.621      0.787
 9 11:33…  0.281  0.0475 6.45e- 8 ENSG… Expr… ENSG…         2   0.276      0.528
10 17:35… -0.526  0.108  4.95e- 6 ENSG… Expr… ENSG…         3   0.905      0.954
# … with 1,731 more rows, and 1 more variable: TopSNPFinemapPr &lt;dbl&gt;</code></pre>
<pre class="r"><code>coloc.results.tidycolocalized %&gt;%
  group_by(Locus, snp) %&gt;%
  filter(any(PC==&quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
  summarise(
    ContainsChromatinEqtl = any(PC %in% c(&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;)),
    ContainsSqtl = any(PC %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;))
    ) %&gt;%
  ggplot(aes(x=1, fill=paste(ContainsChromatinEqtl, ContainsSqtl))) +
  geom_bar() +
  labs(title=&quot;More chromatin localization with eQTLs than splicing&quot;, y=&quot;Number of colocalizing eQTLs&quot;) +
  theme_classic() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())</code></pre>
<p><img src="figure/20220920_Explore_chRNA_sQTLs.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Is this plot sensitive to whether I call just consider H3K27AC for chromatin, as opposed to H3K4ME3</p>
<pre class="r"><code>bind_rows(
  coloc.results.tidycolocalized %&gt;%
    group_by(Locus, snp) %&gt;%
    filter(any(PC==&quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
    summarise(
      ContainsChromatinEqtl = any(PC %in% c(&quot;H3K27AC&quot;, &quot;H3K4ME3&quot;, &quot;H3K4ME1&quot;)),
      ContainsSqtl = any(PC %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;))
      ) %&gt;%
    mutate(x=&quot;chromatin = H3K4ME3|H3K4ME1|H3K27AC; splicing=chRNA|polyA&quot;),
  coloc.results.tidycolocalized %&gt;%
    group_by(Locus, snp) %&gt;%
    filter(any(PC==&quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
    summarise(
      ContainsChromatinEqtl = any(PC %in% c(&quot;H3K27AC&quot;)),
      ContainsSqtl = any(PC %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;))
      ) %&gt;%
    mutate(x=&quot;chromatin = H3K27AC; splicing=chRNA|polyA&quot;),
  coloc.results.tidycolocalized %&gt;%
    group_by(Locus, snp) %&gt;%
    filter(any(PC==&quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
    summarise(
      ContainsChromatinEqtl = any(PC %in% c(&quot;H3K4ME3&quot;)),
      ContainsSqtl = any(PC %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;))
      ) %&gt;%
    mutate(x=&quot;chromatin = H3K4ME3; splicing=chRNA|polyA no IR&quot;),
  coloc.results.tidycolocalized %&gt;%
    group_by(Locus, snp) %&gt;%
    filter(any(PC==&quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
    summarise(
      ContainsChromatinEqtl = any(PC %in% c(&quot;H3K27AC&quot;, &quot;H3K4ME3&quot;, &quot;H3K4ME1&quot;)),
      ContainsSqtl = any(PC %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;))
      ) %&gt;%
    mutate(x=&quot;chromatin = H3K27AC; splicing=chRNA|polyA no IR&quot;),
  coloc.results.tidycolocalized %&gt;%
    group_by(Locus, snp) %&gt;%
    filter(any(PC==&quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
    summarise(
      ContainsChromatinEqtl = any(PC %in% c(&quot;H3K27AC&quot;, &quot;H3K4ME3&quot;, &quot;H3K4ME1&quot;)),
      ContainsSqtl = any(PC %in% c(&quot;polyA.Splicing.Subset_YRI&quot;))
      ) %&gt;%
    mutate(x=&quot;chromatin = H3K27AC; splicing=polyA&quot;),
  coloc.results.tidycolocalized %&gt;%
    group_by(Locus, snp) %&gt;%
    filter(any(PC==&quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
    summarise(
      ContainsChromatinEqtl = any(PC %in% c(&quot;H3K27AC&quot;, &quot;H3K4ME3&quot;, &quot;H3K4ME1&quot;)),
      ContainsSqtl = any(PC %in% c(&quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;))
      ) %&gt;%
    mutate(x=&quot;chromatin = H3K27AC; splicing=chRNA&quot;),
  coloc.results.tidycolocalized %&gt;%
    group_by(Locus, snp) %&gt;%
    filter(any(PC==&quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
    summarise(
      ContainsChromatinEqtl = any(PC %in% c(&quot;H3K27AC&quot;, &quot;H3K4ME3&quot;, &quot;H3K4ME1&quot;)),
      ContainsSqtl = any(PC %in% c(&quot;chRNA.Splicing&quot;))
      ) %&gt;%
    mutate(x=&quot;chromatin = H3K27AC; splicing=chRNA no IR&quot;)
  ) %&gt;%
  ggplot(aes(x=x, fill=paste(ContainsChromatinEqtl, ContainsSqtl))) +
  geom_bar() +
  labs(title=&quot;More chromatin localization with eQTLs than splicing&quot;, y=&quot;Number of colocalizing eQTLs&quot;, x=&quot;&quot;) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))</code></pre>
<p><img src="figure/20220920_Explore_chRNA_sQTLs.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>I am curious about these QTLs that colocalize both chromatinQTL and sQTL effects. I will have to look at enrichment of SNP annotations under finemapping results (which I will look at later in this notebook I think) to get some sense of if they are mostly driven by splicing effects or by chromatin effects.</p>
<p>Next, let’s look at the direction of effects in each of these categories. For example, I expect (and have shown previously) that direction of chromatin QTL effects and eQTL effects are very concordant in colocalizing QTLs (eg Spearman correlation coef &gt;0.9). Are intron retention effects generally discordant in these genes with an intron retention QTL colocalizing with eQTL? Biologically I expect intron retention to generally down regulate. But then I suspect some intron retention QTL effects are somehow picking up on positive transcription effects. So I’m not exactly sure what to expect…</p>
<pre class="r"><code>coloc.results.tidycolocalized %&gt;%
  group_by(Locus, snp) %&gt;%
  filter(any(PC==&quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
  ungroup() %&gt;%
  filter(PC %in% c(&quot;Expression.Splicing.Subset_YRI&quot;,&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;)) %&gt;%
  left_join(., ., by=c(&quot;Locus&quot;, &quot;snp&quot;)) %&gt;%
  filter(!((P.x == P.y) &amp; (PC.x == PC.y))) %&gt;% 
  group_by(Locus, snp) %&gt;%
  mutate(
    ContainsChromatinEqtl = any(PC.x %in% c(&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;)),
    ContainsSqtl = any(PC.x %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;))
    ) %&gt;%
  ungroup() %&gt;%
  mutate(Contains.eQTL_Contains.sQTL = paste(ContainsChromatinEqtl, ContainsSqtl)) %&gt;%
  group_by(Contains.eQTL_Contains.sQTL, PC.x, PC.y) %&gt;%
  summarise(cor = cor(beta.x, beta.y, method=&quot;spearman&quot;)) %&gt;%
  mutate(PC.x = recode(PC.x, !!!PC.ShortAliases)) %&gt;%
  mutate(PC.y = recode(PC.y, !!!PC.ShortAliases)) %&gt;%
  ggplot(aes(x=PC.x, y=PC.y, fill=cor)) +
  geom_raster() +
  scale_fill_gradient2() +
  facet_wrap(~Contains.eQTL_Contains.sQTL) +
  scale_x_discrete(expand=c(0,0)) +
  scale_y_discrete(expand=c(0,0)) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  labs(x=&quot;TraitA Phenotype class&quot;, y=&quot;TraitB Phenotype class&quot;, fill=&quot;Spearman cor&quot;, title=&quot;Effect size correlation&quot;,
       caption = &quot;FacetTitle: IsChromatinQTL IsSplicingQTL&quot;)</code></pre>
<p><img src="figure/20220920_Explore_chRNA_sQTLs.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Hmm, it seems that intron retention might just be picking up eQTL signals, since it is positively correlated with eQTL effect size. And I don’t know how to explain that sQTLs are generally anti-correlated with eQTL effects in polyA (I would’ve expected no correlation).</p>
<p>And thankfully at least the effect directions of the chRNA sQTLs are positively correlated with the polyA sQTLs. These are all to be expected.</p>
<p>Next I want to plot a lot of these examples, as sashimi plots and coverage plots grouped by genotype of the top SNP, to get some intuitions on this. More specifically, is PSI larger in chRNA for the effects of opposite direction sQTL/eQTL effects (the presumed NMD targets)?</p>
<pre class="r"><code># AllFeats &lt;- read_tsv(&quot;../code/QC/AllQTLPhenotypes.PermutationTest.bed.gz&quot;, col_names=c(&quot;Chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;thickStart&quot;,&quot;thickStop&quot;, &quot;Color&quot;)) %&gt;%
#   separate(name, into=c(&quot;PC&quot;, &quot;P&quot;, &quot;topSNP.QTLTools&quot;, &quot;Pval&quot;, &quot;FDR&quot;), convert=T, sep=&quot;;&quot;)

AllFeats &lt;- Sys.glob(&quot;../code/QTLs/QTLTools/*/OnlyFirstReps.sorted.qqnorm.bed.gz&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/QTLs/QTLTools/(.+?)/OnlyFirstReps.sorted.qqnorm.bed.gz&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(fread, sep=&#39;\t&#39;, select=1:6) %&gt;%
  bind_rows(.id=&quot;PC&quot;)

sample_n_of &lt;- function(data, size, ...) {
  dots &lt;- quos(...)
  
  group_ids &lt;- data %&gt;% 
    group_by(!!! dots) %&gt;% 
    group_indices()
  
  sampled_groups &lt;- sample(unique(group_ids), size)
  
  data %&gt;% 
    filter(group_ids %in% sampled_groups)
}

ColocsToPlot.dat &lt;- coloc.results.tidycolocalized %&gt;%
  filter(PC %in% c(&quot;Expression.Splicing.Subset_YRI&quot;,&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;)) %&gt;%
  group_by(Locus, snp) %&gt;%
  filter(any(PC==&quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
  mutate(
    ContainsChromatinEqtl = if_else(any(PC %in% c(&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;)), &quot;hQTL&quot;, &quot;Not_hQTL&quot;),
    ContainsSqtl = if_else(any(PC %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;)), &quot;sQTL&quot;, &quot;Not_sQTL&quot;)
    ) %&gt;%
  ungroup() %&gt;%
  mutate(Group = paste(ContainsChromatinEqtl, ContainsSqtl, sep=&quot;.&quot;)) %&gt;%
  inner_join(AllFeats, by=c(&quot;PC&quot;, &quot;P&quot;=&quot;pid&quot;))

set.seed(0)
ColocsToPlot.Subset &lt;- ColocsToPlot.dat %&gt;%
  distinct(Locus, snp, Group, iteration) %&gt;%
  group_by(Group) %&gt;%
  sample_n(100) %&gt;%
  ungroup()</code></pre>
<p>Write bash script and other necessary stuff to execute my helper script for plotting</p>
<pre class="r"><code>#Create folder to house files and images
dir.create(&quot;../code/scratch/20220921_PlotColocs&quot;)

# Write bed files containing all colocalized features that will be plotted
ColocsToPlot.dat %&gt;%
  inner_join(ColocsToPlot.Subset) %&gt;%
  mutate(PC = recode(PC, !!!PC.ShortAliases)) %&gt;%
  mutate(name=paste(PC, P, &quot;beta=&quot;,signif(beta, 2)),
         dummyScore = 0) %&gt;%
  dplyr::select(`#Chr`, start, end, name, dummyScore, strand) %&gt;%
  write_tsv(&quot;../code/scratch/20220921_PlotColocs/.ColocFeats.bed&quot;)

# Write pygenometracks ini file for the above bed file
fileConn&lt;-file(&quot;../code/scratch/20220921_PlotColocs/.ColocFeats.ini&quot;)
writeLines(c(&quot;[feats]&quot;,
             &quot;type = bed&quot;,
             &quot;file = scratch/20220921_PlotColocs/.ColocFeats.bed&quot;,
             &quot;fontsize = 3&quot;,
             &quot;gene_rows = 10&quot;,
             &quot;height=5&quot;
             ), con=fileConn)
close(fileConn)

# Write a &quot;groups&quot; file for my custom script to determine which tracks to plot, and how to plot them
read_tsv(&quot;../code/scratch/bwList.Groups2.tsv&quot;) %&gt;%
  filter(Group_label %in% c(&quot;chRNA&quot;, &quot;polyA.RNA&quot;, &quot;MetabolicLabelled.30min&quot;, &quot;MetabolicLabelled.60min&quot;,&quot;H3K27AC&quot;, &quot;H3K4ME3&quot;, &quot;H3K4ME1&quot;, &quot;H3K36ME3&quot;)) %&gt;%
  write_tsv(&quot;../code/scratch/20220921_PlotColocs/.groups.tsv&quot;)

# Write a bash script that calls my custom script and pygenometracks to plot stuff
ColocsToPlot.dat %&gt;%
  inner_join(ColocsToPlot.Subset) %&gt;%
  rename(chrom=&quot;#Chr&quot;) %&gt;%
  group_by(Locus, snp, chrom, Group, iteration) %&gt;%
  summarise(
    min = min(start)-10,
    max= max(end)+10) %&gt;%
  ungroup() %&gt;%
  mutate(SnpPos = str_replace(snp, &quot;^(.+?):(.+?):.+?:.+?$&quot;, &quot;chr\\1:\\2&quot;)) %&gt;%
  mutate(chrom.num = str_replace(chrom, &quot;chr(.+?)$&quot;, &quot;\\1&quot;)) %&gt;%
  mutate(cmd = str_glue(&quot;python scripts/GenometracksByGenotype/AggregateBigwigsForPlotting.py --GroupSettingsFile scratch/20220921_PlotColocs/.groups.tsv --BigwigList scratch/bwList.tsv --SnpPos {SnpPos} --Normalization WholeGenome --Region {chrom}:{min}-{max} --BigwigListType KeyFile --OutputPrefix scratch/20220921_PlotColocs/. --VCF Genotypes/1KG_GRCh38/{chrom.num}.vcf.gz -vv --TracksTemplate scripts/GenometracksByGenotype/tracks_templates/GeneralPurposeColoredByGenotype.ini --Bed12GenesToIni scripts/GenometracksByGenotype/PremadeTracks/gencode.v26.FromGTEx.genes.bed12.gz  --FilterJuncsByBed scratch/20220921_PlotColocs/.ColocFeats.bed\npyGenomeTracks --tracks &lt;(cat scratch/20220921_PlotColocs/.tracks.ini scratch/20220921_PlotColocs/.ColocFeats.ini) --out scratch/20220921_PlotColocs/{Group}.{Locus}.{iteration}.pdf --region {chrom}:{min}-{max} --title {Group}_{Locus}_{snp} --trackLabelFraction 0.15\n\n&quot;)) %&gt;%
  dplyr::select(cmd) %&gt;%
  write.table(&quot;../code/scratch/20220921_PlotColocs/MakePlots.sh&quot;, quote=F, row.names = F, col.names = F)</code></pre>
<p>and execute the bash script to make a bunch of plots</p>
<pre class="bash"><code>cd /project2/yangili1/bjf79/ChromatinSplicingQTLs/code
bash scratch/20220921_PlotColocs/MakePlots.sh</code></pre>
<p>I’ve browsed perhaps 50 so far… some things i’ve noticed:</p>
<ul>
<li>a lot of plots aren’t interpretable… or the eQTL effects are barely noticeable in this raw coverage plot at the given scale</li>
<li>among the eQTL + sQTL + hQTL ones, a fair number of the interpretable ones are alt TSS</li>
<li>among the sQTL + eQTL ones, often we see intron retention things (not in always understandable ways yet) around the sQTLs</li>
<li>some NMD targets, though it’s not always more apparent in chRNA like I hypothesize</li>
<li>TODO:May want to get a better sense of the chRNA-specific sQTLs</li>
<li>TODO:Add annotations to plot about which introns are annotated or known NMD-related introns</li>
</ul>
<p>Next, let’s compare these groups of eQTLs (those that coloc with sQTL only, with hQTL only, or with both), to “unexplained eQTLs” (in this case, ones that colocalize with something, but not sQTL or hQTL), in terms of their finemap SNP enrichment over annotations (eg ChromHMM states, splice sites, etc)…</p>
<pre class="r"><code>dat &lt;- fread(&quot;../code/QTL_SNP_Enrichment/FinemapIntersections/MolColocStandard.bed.gz&quot;, col.names=c(&quot;SNP_chrom&quot;, &quot;SNP_start&quot;, &quot;SNP_end&quot;, &quot;SNP&quot;, &quot;PosteriorPr&quot;, &quot;Annotation_chrom&quot;, &quot;Annotation_start&quot;, &quot;Annotation_stop&quot;, &quot;Annotation&quot;, &quot;Overlap&quot;)) %&gt;%
  separate(SNP, into=c(&quot;SNP&quot;, &quot;HyprcolocCluster&quot;, &quot;GeneLocus&quot;), sep=&quot;_&quot;)


GeneLocus_Cluster_QTLSets &lt;- ColocsToPlot.dat %&gt;%
  distinct(Locus, iteration, Group) %&gt;%
  unite(Locus.iteration, Locus, iteration) %&gt;%
  rename(ClusterGroup=Group) %&gt;%
  split(.$ClusterGroup)
  

#Define the function
PlotDifferentialEnrichmentBetweenQTLSets &lt;- function(dat, GeneLocus_Cluster_QTLSet1, GeneLocus_Cluster_QTLSet2, seed=0, iterations=100){
  #GeneLocus_Cluster_QTLSet1 and GeneLocus_Cluster_QTLSet2 are dataframes with mandatory columns called Locus.iteration (with values of Locus_iteration), and a ClusterGroup column to name the two sets (ie value like &quot;SplicingClusters&quot; for GeneLocus_Cluster_QTLSet1 and &quot;HistoneClusters&quot; for GeneLocus_Cluster_QTLSet2)
  output &lt;- list()
  results &lt;- list()
  for (i in 1:iterations){
    results[[i]] &lt;-
    bind_rows(
    sample_frac(GeneLocus_Cluster_QTLSet1, replace=T),
    sample_frac(GeneLocus_Cluster_QTLSet2, replace=T),
    .id=&quot;QTL.Set&quot;) %&gt;%
    left_join(
      dat %&gt;%
      mutate(Locus.iteration = paste(GeneLocus, HyprcolocCluster, sep=&quot;_&quot;)),
      by = &quot;Locus.iteration&quot;
    ) %&gt;%
    mutate(Annotation = str_replace_all(Annotation, &quot;_0$&quot;, &quot;_UnannotatedButObserved&quot;)) %&gt;%
    mutate(Annotation = str_replace_all(Annotation, &quot;_1$&quot;, &quot;_Annotated&quot;)) %&gt;%
    mutate(Annotation = str_replace_all(Annotation, &quot;^\\d+?_&quot;, &quot;ChromHMM:&quot;)) %&gt;%
    filter(!is.na(Annotation)) %&gt;%
    mutate(Annotation = if_else(Annotation==&quot;.&quot;, &quot;No annotation&quot;, Annotation)) %&gt;%
    group_by(Annotation, ClusterGroup, QTL.Set) %&gt;%
    summarise(Sum = sum(PosteriorPr)) %&gt;%
    group_by(ClusterGroup, QTL.Set) %&gt;%
    mutate(FractionAnnotatedPosteriorInGroup = Sum/sum(Sum, na.rm=T)) %&gt;%
    ungroup() %&gt;%
    # select(-ClusterGroup) %&gt;%
    pivot_wider(names_from = &quot;QTL.Set&quot;, values_from = c(&quot;FractionAnnotatedPosteriorInGroup&quot;, &quot;Sum&quot;), id_cols=&quot;Annotation&quot;) %&gt;%
    mutate(Enrichment = FractionAnnotatedPosteriorInGroup_1/FractionAnnotatedPosteriorInGroup_2) %&gt;%
    mutate(i=i)
  }
  output[[&quot;data&quot;]] &lt;- bind_rows(results)
  output[[&quot;plot&quot;]] &lt;- output[[&quot;data&quot;]] %&gt;%
  group_by(Annotation) %&gt;%
  summarise(mean = exp(mean(log(Enrichment), na.rm=T)),
            Percentile10 = quantile(Enrichment, probs = 0.05, na.rm=T),
            Percentile90 = quantile(Enrichment, probs = 0.95, na.rm=T)) %&gt;%
  ggplot(aes(x=Annotation, y=log2(mean))) +
    geom_hline(yintercept = 0, color=&#39;red&#39;) +
    geom_col() +
    geom_errorbar(aes(ymin=log2(Percentile10), ymax=log2(Percentile90))) +
    theme_bw() +
    # scale_y_continuous(trans=&#39;log2&#39;) +
    theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
    coord_flip() +
    labs(x = &quot;Annotation&quot;, y=&quot;Finemapping posterior mass relative log2 fold enrichment\nQTL.Set2 &lt;----&gt; QTL.Set1&quot;,
         caption = &quot;Bootsrapped 90% confidence interval&quot;)
  return(output)
}</code></pre>
<p>Ok now that I have the plotting function set up, let’s compare the eQTLs with hQTL but not sQTL, to those without sQTL or hQTL…</p>
<pre class="r"><code>PlotDifferentialEnrichmentBetweenQTLSets(dat, GeneLocus_Cluster_QTLSets[[&quot;hQTL.Not_sQTL&quot;]], GeneLocus_Cluster_QTLSets[[&quot;Not_hQTL.Not_sQTL&quot;]])$plot +
  labs(caption=&quot;Set1=hQTL.Not_sQTL\nSet2=Not_hQTL.Not_sQTL&quot;)</code></pre>
<p><img src="figure/20220920_Explore_chRNA_sQTLs.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>there is enrichment in ChromHMM enhancer regions, and de-enrichment in splice sites, as expected.</p>
<p>…And now check those eQTLs with sQTLs but not hQTLs, to those without hQTLs or sQTLs</p>
<pre class="r"><code>PlotDifferentialEnrichmentBetweenQTLSets(dat, GeneLocus_Cluster_QTLSets[[&quot;Not_hQTL.sQTL&quot;]], GeneLocus_Cluster_QTLSets[[&quot;Not_hQTL.Not_sQTL&quot;]])$plot +
  labs(caption=&quot;Set1=Not_hQTL.sQTL\nSet2=Not_hQTL.Not_sQTL&quot;)</code></pre>
<p><img src="figure/20220920_Explore_chRNA_sQTLs.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>…Enrichment for splice sites as expected… Now let’s check those with both hQTLs or sQTLs to those without either…</p>
<pre class="r"><code>PlotDifferentialEnrichmentBetweenQTLSets(dat, GeneLocus_Cluster_QTLSets[[&quot;hQTL.sQTL&quot;]], GeneLocus_Cluster_QTLSets[[&quot;Not_hQTL.Not_sQTL&quot;]])$plot +
  labs(caption=&quot;Set1=hQTL.sQTL\nSet2=Not_hQTL.Not_sQTL&quot;)</code></pre>
<p><img src="figure/20220920_Explore_chRNA_sQTLs.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /> …so there is enrichment for PAS, that makes sense at it is a non splicing and non histone mechanism that can to regulate expression.</p>
</div>
<div id="ncrna-qtls" class="section level3">
<h3>ncRNA QTLs</h3>
<p>I think i want to also plot a lot of the new ncRNA QTLs… Let’s explore those</p>
<p>i would like to explore some of the following ideas:</p>
<ul>
<li>upstream antisense ncRNAs (from bidirectional promoter for a gene), which i expect to always have concordant effects with the gene.</li>
<li>check for intergenic ncRNAs. I expect them to largely be concordant too, as they may be products of active enhancers that also affect a gene. However i will be curious to check for cases where the signs are non concordant, to make sure I even believe them…</li>
</ul>
<pre class="r"><code>ncRNA.annotations &lt;- read_tsv(&quot;../code/NonCodingRNA_annotation/annotation/ncRNA.annotation.tab.gz&quot;)

head(ncRNA.annotations)</code></pre>
<pre><code># A tibble: 6 × 9
  gene_name rna_type uaRNA coRNA            rtRNA ctRNA lncRNA snoRNA pseudogene
  &lt;chr&gt;     &lt;chr&gt;    &lt;chr&gt; &lt;chr&gt;            &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt;     
1 ncRNA_1   incRNA   .     .                .     .     .      .      .         
2 ncRNA_2   incRNA   .     .                .     .     .      .      .         
3 ncRNA_3   srtRNA   .     .                ENSG… .     .      .      .         
4 ncRNA_4   srtRNA   .     .                ENSG… .     .      .      .         
5 ncRNA_5   coRNA    .     ENSG00000217801… .     .     .      .      .         
6 ncRNA_8   rtRNA    .     .                ncRN… .     .      .      .         </code></pre>
<pre class="r"><code>ncRNA.annotations %&gt;%
  count(rna_type) %&gt;%
  arrange(desc(n))</code></pre>
<pre><code># A tibble: 45 × 2
   rna_type         n
   &lt;chr&gt;        &lt;int&gt;
 1 srtRNA        5023
 2 incRNA        4924
 3 uaRNA         1609
 4 coRNA          362
 5 uaRNA,coRNA    303
 6 uaRNA,srtRNA   230
 7 rtRNA          164
 8 coRNA,srtRNA    90
 9 coRNA,pseudo    78
10 ctRNA           71
# … with 35 more rows</code></pre>
<pre class="r"><code>coloc.results.tidycolocalized %&gt;%
  filter(PC == &quot;chRNA.Expression_ncRNA&quot;)</code></pre>
<pre><code># A tibble: 1,760 × 11
   snp      beta beta_se        p Locus PC    P     iteration ColocPr RegionalPr
   &lt;chr&gt;   &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;     &lt;dbl&gt;   &lt;dbl&gt;      &lt;dbl&gt;
 1 11:64… -0.420  0.0572 1.26e-10 ENSG… chRN… ncRN…         2   0.992      0.993
 2 17:48… -0.310  0.0816 2.69e- 4 ENSG… chRN… ncRN…         1   0.675      0.790
 3 2:372… -0.611  0.122  2.90e- 6 ENSG… chRN… ncRN…         1   0.850      0.880
 4 1:331…  1.02   0.193  8.69e- 7 ENSG… chRN… ncRN…         2   0.818      0.832
 5 17:47… -0.460  0.0894 1.72e- 6 ENSG… chRN… ENSG…         1   0.972      0.996
 6 7:873… -0.961  0.119  4.28e-12 ENSG… chRN… ENSG…         2   0.862      0.884
 7 7:873… -0.961  0.119  4.28e-12 ENSG… chRN… ENSG…         2   0.852      0.880
 8 6:822… -0.385  0.102  3.15e- 4 ENSG… chRN… ENSG…         1   0.621      0.787
 9 17:45… -1.03   0.147  5.36e-10 ENSG… chRN… ENSG…         2   0.962      0.968
10 17:35…  0.490  0.0999 4.39e- 6 ENSG… chRN… ENSG…         1   0.846      0.973
# … with 1,750 more rows, and 1 more variable: TopSNPFinemapPr &lt;dbl&gt;</code></pre>
<p>Hmmm, this is kind of complicated but for simplicity for now, let’s just consider the ncRNAs with only one type of rna_type subannotation.</p>
<p>First, let’s confirm that effect sizes are generally concordant with eQTL and enhancer/promoter activity…</p>
<pre class="r"><code>coloc.results.tidycolocalized$PC %&gt;% unique()</code></pre>
<pre><code> [1] &quot;polyA.Splicing&quot;                   &quot;MetabolicLabelled.30min&quot;         
 [3] &quot;MetabolicLabelled.60min&quot;          &quot;H3K27AC&quot;                         
 [5] &quot;H3K4ME3&quot;                          &quot;H3K4ME1&quot;                         
 [7] &quot;chRNA.IER&quot;                        &quot;ProCap&quot;                          
 [9] &quot;polyA.Splicing.Subset_YRI&quot;        &quot;CTCF&quot;                            
[11] &quot;chRNA.Splicing.Order&quot;             &quot;chRNA.Expression_ncRNA&quot;          
[13] &quot;Expression.Splicing&quot;              &quot;Expression.Splicing.Subset_YRI&quot;  
[15] &quot;chRNA.Splicing&quot;                   &quot;MetabolicLabelled.30min.Splicing&quot;
[17] &quot;MetabolicLabelled.60min.Splicing&quot; &quot;chRNA.Expression.Splicing&quot;       
[19] &quot;H3K36ME3&quot;                         &quot;polyA.IER&quot;                       
[21] &quot;MetabolicLabelled.30min.IER&quot;      &quot;MetabolicLabelled.60min.IER&quot;     
[23] &quot;polyA.IER.Subset_YRI&quot;             &quot;chRNA.Slopes&quot;                    </code></pre>
<pre class="r"><code>coloc.results.tidycolocalized %&gt;%
  group_by(Locus, snp) %&gt;%
  filter(any(PC==&quot;Expression.Splicing.Subset_YRI&quot;) &amp; any(PC==&quot;chRNA.Expression_ncRNA&quot;)) %&gt;%
  ungroup() %&gt;%
  filter(PC %in% c(&quot;Expression.Splicing.Subset_YRI&quot;,&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;chRNA.Expression_ncRNA&quot;)) %&gt;%
  left_join(., ., by=c(&quot;Locus&quot;, &quot;snp&quot;)) %&gt;%
  filter(!((P.x == P.y) &amp; (PC.x == PC.y))) %&gt;%
  group_by(PC.x, PC.y) %&gt;%
  summarise(cor = cor(beta.x, beta.y, method=&quot;spearman&quot;)) %&gt;%
  mutate(PC.x = recode(PC.x, !!!PC.ShortAliases)) %&gt;%
  mutate(PC.y = recode(PC.y, !!!PC.ShortAliases)) %&gt;%
  ggplot(aes(x=PC.x, y=PC.y, fill=cor)) +
  geom_raster() +
  scale_fill_gradient2() +
  scale_x_discrete(expand=c(0,0)) +
  scale_y_discrete(expand=c(0,0)) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  labs(x=&quot;TraitA Phenotype class&quot;, y=&quot;TraitB Phenotype class&quot;, fill=&quot;Spearman cor&quot;, title=&quot;Effect size correlation&quot;)</code></pre>
<p><img src="figure/20220920_Explore_chRNA_sQTLs.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, now let’s break up the ncRNA phenotypes into the sub-groups…</p>
<p>This is a simplified way of doing this because the ncRNA annotation file is a bit complicated and i don’t fully know how to handle it yet.</p>
<pre class="r"><code>ncRNA.annotations %&gt;%
  count(rna_type) %&gt;%
  ggplot(aes(x=rna_type, y=n)) +
  coord_flip() +
  geom_col()</code></pre>
<p><img src="figure/20220920_Explore_chRNA_sQTLs.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#For simplicity, Only consider ncRNAs that only have one subtype
ncRNA.annotations %&gt;%
  filter(!str_detect(rna_type, &quot;,&quot;) )%&gt;%
  count(rna_type) %&gt;%
  ggplot(aes(x=rna_type, y=n)) +
  geom_col()</code></pre>
<p><img src="figure/20220920_Explore_chRNA_sQTLs.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok let’s try making that effect size heatmap…</p>
<pre class="r"><code>coloc.results.tidycolocalized %&gt;%
  left_join(
    ncRNA.annotations %&gt;%
      mutate(PC = &quot;chRNA.Expression_ncRNA&quot;) %&gt;%
      dplyr::select(1,2, PC) %&gt;%
      rename(P=gene_name) %&gt;%
      filter(!str_detect(rna_type, &quot;,&quot;))
  ) %&gt;%
  mutate(PC = case_when(
    !is.na(rna_type) ~ rna_type,
    TRUE ~ PC
  )) %&gt;%
  group_by(Locus, snp) %&gt;%
  filter(any(PC==&quot;Expression.Splicing.Subset_YRI&quot;) &amp; any(!is.na(rna_type))) %&gt;%
  ungroup() %&gt;%
  filter((PC %in% c(&quot;Expression.Splicing.Subset_YRI&quot;,&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;)) | (!is.na(rna_type))) %&gt;%
  left_join(., ., by=c(&quot;Locus&quot;, &quot;snp&quot;)) %&gt;%
  filter(!((P.x == P.y) &amp; (PC.x == PC.y))) %&gt;%
  group_by(PC.x, PC.y) %&gt;%
  summarise(cor = cor(beta.x, beta.y, method=&quot;spearman&quot;),
            n=n()) %&gt;%
  mutate(PC.x = recode(PC.x, !!!PC.ShortAliases)) %&gt;%
  mutate(PC.y = recode(PC.y, !!!PC.ShortAliases)) %&gt;%
  ggplot(aes(x=PC.x, y=PC.y, fill=cor)) +
  geom_raster() +
  scale_fill_gradient2() +
  geom_text(aes(label=n)) +
  scale_x_discrete(expand=c(0,0)) +
  scale_y_discrete(expand=c(0,0)) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  labs(x=&quot;TraitA Phenotype class&quot;, y=&quot;TraitB Phenotype class&quot;, fill=&quot;Spearman cor&quot;, title=&quot;Effect size correlation&quot;)</code></pre>
<p><img src="figure/20220920_Explore_chRNA_sQTLs.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] data.table_1.14.2 forcats_0.4.0     stringr_1.4.0     dplyr_1.0.9      
 [5] purrr_0.3.4       readr_1.3.1       tidyr_1.2.0       tibble_3.1.7     
 [9] ggplot2_3.3.6     tidyverse_1.3.0  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5        lubridate_1.7.4   assertthat_0.2.1  rprojroot_2.0.2  
 [5] digest_0.6.20     utf8_1.1.4        R6_2.4.0          cellranger_1.1.0 
 [9] backports_1.4.1   reprex_0.3.0      evaluate_0.15     httr_1.4.4       
[13] highr_0.9         pillar_1.7.0      rlang_1.0.5       readxl_1.3.1     
[17] rstudioapi_0.14   R.oo_1.22.0       R.utils_2.9.0     rmarkdown_1.13   
[21] labeling_0.3      munsell_0.5.0     broom_1.0.0       compiler_3.6.1   
[25] httpuv_1.5.1      modelr_0.1.8      xfun_0.31         pkgconfig_2.0.2  
[29] htmltools_0.5.3   tidyselect_1.1.2  workflowr_1.6.2   fansi_0.4.0      
[33] crayon_1.3.4      dbplyr_1.4.2      withr_2.5.0       later_0.8.0      
[37] R.methodsS3_1.7.1 grid_3.6.1        jsonlite_1.6      gtable_0.3.0     
[41] lifecycle_1.0.1   DBI_1.1.0         git2r_0.26.1      magrittr_1.5     
[45] scales_1.1.0      cli_3.3.0         stringi_1.4.3     farver_2.1.0     
[49] fs_1.5.2          promises_1.0.1    xml2_1.3.2        ellipsis_0.3.2   
[53] generics_0.1.3    vctrs_0.4.1       tools_3.6.1       glue_1.6.2       
[57] hms_0.5.3         fastmap_1.1.0     yaml_2.2.0        colorspace_1.4-1 
[61] rvest_0.3.5       knitr_1.39        haven_2.3.1      </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
