<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Does intron retention down regulate gene expression</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Does intron retention down regulate gene expression</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-10-26
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>ChromatinSplicingQTLs/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstree8df7e43cc493d35d443d132d58c6f327ee406d9ftargetblank8df7e43a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/8df7e43cc493d35d443d132d58c6f327ee406d9f" target="_blank">8df7e43</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstree8df7e43cc493d35d443d132d58c6f327ee406d9ftargetblank8df7e43a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/8df7e43cc493d35d443d132d58c6f327ee406d9f" target="_blank">8df7e43</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store
    Ignored:    code/.RData
    Ignored:    code/._.DS_Store
    Ignored:    code/._README.md
    Ignored:    code/._report.html
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    code/.snakemake/
    Ignored:    code/APA_Processing/
    Ignored:    code/Alignments/
    Ignored:    code/ChromHMM/
    Ignored:    code/ENCODE/
    Ignored:    code/ExpressionAnalysis/
    Ignored:    code/FastqFastp/
    Ignored:    code/FastqFastpSE/
    Ignored:    code/Genotypes/
    Ignored:    code/H3K36me3_CutAndTag.pdf
    Ignored:    code/IntronSlopes/
    Ignored:    code/Metaplots/
    Ignored:    code/Misc/
    Ignored:    code/MiscCountTables/
    Ignored:    code/Multiqc/
    Ignored:    code/Multiqc_chRNA/
    Ignored:    code/NonCodingRNA/
    Ignored:    code/NonCodingRNA_annotation/
    Ignored:    code/PeakCalling/
    Ignored:    code/Phenotypes/
    Ignored:    code/PlotGruberQTLs/
    Ignored:    code/PlotQTLs/
    Ignored:    code/ProCapAnalysis/
    Ignored:    code/QC/
    Ignored:    code/QTL_SNP_Enrichment/
    Ignored:    code/QTLs/
    Ignored:    code/RPKM_tables/
    Ignored:    code/ReferenceGenome/
    Ignored:    code/Rplots.pdf
    Ignored:    code/Session.vim
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TODO
    Ignored:    code/Tehranchi/
    Ignored:    code/bigwigs/
    Ignored:    code/bigwigs_FromNonWASPFilteredReads/
    Ignored:    code/config/.DS_Store
    Ignored:    code/config/._.DS_Store
    Ignored:    code/config/.ipynb_checkpoints/
    Ignored:    code/dag.pdf
    Ignored:    code/dag.png
    Ignored:    code/dag.svg
    Ignored:    code/debug.ipynb
    Ignored:    code/debug_python.ipynb
    Ignored:    code/deepTools/
    Ignored:    code/featureCounts/
    Ignored:    code/gwas_summary_stats/
    Ignored:    code/hyprcoloc/
    Ignored:    code/igv_session.xml
    Ignored:    code/log
    Ignored:    code/logs/
    Ignored:    code/notebooks/.ipynb_checkpoints/
    Ignored:    code/pi1/
    Ignored:    code/rules/.ipynb_checkpoints/
    Ignored:    code/rules/OldRules/
    Ignored:    code/rules/notebooks/
    Ignored:    code/scratch/
    Ignored:    code/scripts/.ipynb_checkpoints/
    Ignored:    code/scripts/GTFtools_0.8.0/
    Ignored:    code/scripts/__pycache__/
    Ignored:    code/scripts/liftOverBedpe/liftOverBedpe.py
    Ignored:    code/snakemake.dryrun.log
    Ignored:    code/snakemake.log
    Ignored:    code/snakemake.sbatch.log
    Ignored:    code/test.introns.bed
    Ignored:    code/test.introns2.bed
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/._20220414203249_JASPAR2022_combined_matrices_25818_jaspar.txt
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-10.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-11.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-2.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-3.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-4.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-5.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-6.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-7.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-8.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022.csv

Untracked files:
    Untracked:  code/snakemake_profiles/slurm/__pycache__/

Unstaged changes:
    Modified:   analysis/20221012_IntronRetentionAndExpressionConcordance.Rmd
    Modified:   code/scripts/GenometracksByGenotype

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/20221012_IntronRetentionAndExpressionConcordance.Rmd</code>) and HTML (<code>docs/20221012_IntronRetentionAndExpressionConcordance.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/8df7e43cc493d35d443d132d58c6f327ee406d9f/analysis/20221012_IntronRetentionAndExpressionConcordance.Rmd" target="_blank">8df7e43</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-26
</td>
<td>
update nb
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/8df7e43cc493d35d443d132d58c6f327ee406d9f/docs/20221012_IntronRetentionAndExpressionConcordance.html" target="_blank">8df7e43</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-26
</td>
<td>
update nb
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/61a245fa779318e64d6bb6154b21c4a792509781/analysis/20221012_IntronRetentionAndExpressionConcordance.Rmd" target="_blank">61a245f</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-24
</td>
<td>
update nb
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/c17b6b2fcd112a08cc7e7a2fb5bb45538dc3d339/analysis/20221012_IntronRetentionAndExpressionConcordance.Rmd" target="_blank">c17b6b2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-19
</td>
<td>
add un qqnorm phenotypes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/c17b6b2fcd112a08cc7e7a2fb5bb45538dc3d339/docs/20221012_IntronRetentionAndExpressionConcordance.html" target="_blank">c17b6b2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-19
</td>
<td>
add un qqnorm phenotypes
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ef85e5320603653a3919d55d51162a3ada4b35d8/analysis/20221012_IntronRetentionAndExpressionConcordance.Rmd" target="_blank">ef85e53</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-12
</td>
<td>
update index
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/3201a1943ba4ad08cdd1d521bbacbc97ad8e3689/analysis/20221012_IntronRetentionAndExpressionConcordance.Rmd" target="_blank">3201a19</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-12
</td>
<td>
add IR notebook
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/3201a1943ba4ad08cdd1d521bbacbc97ad8e3689/docs/20221012_IntronRetentionAndExpressionConcordance.html" target="_blank">3201a19</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-12
</td>
<td>
add IR notebook
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="intro" class="section level2">
<h2>Intro</h2>
<p>I previously showed that effect size directions for intron retention QTLs puzzingly positively correct with eQTLs (more intron retention = more expression). Somehow I think intron retention QTLs are often picking up on chromatin effects. Let’s break up the intron retention QTLs into groups by the location of the SNP (in a enhancer/promoter vs a splice site) and reassess the concordance of expression effects. I can also look at the same idea with normal leafcutter sQTLs, comparing introns by their annotation type. For example increase in splicing of annotated or “basic” tagged introns might generally increase expression, versus increase in unannoated or “NMD” tagged introns might decrease expression.</p>
<pre class="r"><code>library(tidyverse)
library(data.table)
library(qvalue)</code></pre>
<p>First let’s make some basic plots establishing that chRNA has more unannoated and NMD-specific splicing.</p>
<pre class="r"><code>NMD.transcript.introns &lt;- read_tsv(&quot;../code/SplicingAnalysis/Annotations/NMD/NMD_trancsript_introns.bed.gz&quot;, col_names=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;)) %&gt;%
  mutate(stop=stop+1) %&gt;%
  unite(intron, chrom:stop, strand)

Non.NMD.transcript.introns &lt;- read_tsv(&quot;../code/SplicingAnalysis/Annotations/NMD/NonNMD_trancsript_introns.bed.gz&quot;, col_names=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;)) %&gt;%
  mutate(stop=stop+1) %&gt;%
  unite(intron, chrom:stop, strand)


NMD.specific.introns &lt;- setdiff(NMD.transcript.introns$intron, Non.NMD.transcript.introns$intron)


SpliceJunctionCountTables &lt;- Sys.glob(&quot;../code/SplicingAnalysis/leafcutter/NormalizedPsiTables/PSI.JunctionCounts.*.bed.gz&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/SplicingAnalysis/leafcutter/NormalizedPsiTables/PSI.JunctionCounts.(.+?).bed.gz&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(read_tsv)

lapply(SpliceJunctionCountTables, dim)</code></pre>
<pre><code>$Expression.Splicing
[1] 198246    468

$MetabolicLabelled.30min
[1] 198246     72

$MetabolicLabelled.60min
[1] 198246     72

$chRNA.Expression.Splicing
[1] 198246     93</code></pre>
<pre class="r"><code># SpliceJunctionCountTables$Expression.Splicing %&gt;%
#   mutate(Intron=paste(`#Chrom`, start, end, strand, sep=&quot;_&quot;)) %&gt;%
#   mutate(Is.NMD.Intron = Intron %in% NMD.specific.introns) %&gt;%
#   group_by(Is.NMD.Intron) %&gt;%
#   summarise_if(is.numeric, sum, na.rm = TRUE) %&gt;%
#   dplyr::select(-start, -end) %&gt;%
#   column_to_rownames(&quot;Is.NMD.Intron&quot;) %&gt;%
#   t()

Sum.NMD.Intron.Counts &lt;- function(df){
  df %&gt;%
  mutate(Intron=paste(`#Chrom`, start, end, strand, sep=&quot;_&quot;)) %&gt;%
  mutate(Is.NMD.Intron = Intron %in% NMD.specific.introns) %&gt;%
  group_by(Is.NMD.Intron) %&gt;%
  summarise_if(is.numeric, sum, na.rm = TRUE) %&gt;%
  dplyr::select(-start, -end) %&gt;%
  column_to_rownames(&quot;Is.NMD.Intron&quot;) %&gt;%
  t() %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;IndID&quot;) %&gt;%
  return()
}

lapply(SpliceJunctionCountTables, Sum.NMD.Intron.Counts) %&gt;%
  bind_rows(.id=&quot;Source&quot;) %&gt;%
  dplyr::rename(c(&quot;NotNMD&quot;=&quot;FALSE&quot;, &quot;NMD&quot;=&quot;TRUE&quot;)) %&gt;%
  mutate(Fraction = NMD/(NMD+NotNMD)*100) %&gt;%
  ggplot(aes(x=Source, y=Fraction, color=Source)) +
  geom_boxplot(outlier.position=&quot;none&quot;) +
  geom_jitter() +
  theme_bw()</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/c17b6b2fcd112a08cc7e7a2fb5bb45538dc3d339/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-2-1.png" target="_blank">c17b6b2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>lapply(SpliceJunctionCountTables, Sum.NMD.Intron.Counts) %&gt;%
  bind_rows(.id=&quot;Source&quot;) %&gt;%
  dplyr::rename(c(&quot;NotNMD&quot;=&quot;FALSE&quot;, &quot;NMD&quot;=&quot;TRUE&quot;)) %&gt;%
  mutate(TotalJunctionCounts = NMD+NotNMD) %&gt;%
  ggplot(aes(x=Source, y=TotalJunctionCounts, color=Source)) +
  geom_boxplot(outlier.position=&quot;none&quot;) +
  geom_jitter() +
  theme_bw() +
  labs(title=&quot;Total juncs per dataset&quot;, y=&quot;Total num juncs&quot;)</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-2">
Past versions of unnamed-chunk-2-2.png
</button>
</p>
<div id="fig-unnamed-chunk-2-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/c17b6b2fcd112a08cc7e7a2fb5bb45538dc3d339/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-2-2.png" target="_blank">c17b6b2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="update" class="section level4">
<h4>Update:</h4>
<p>Yang is asking for the following plots to follow up on this one:</p>
<blockquote>
<ol style="list-style-type: lower-roman">
<li>same plot with unannotated junctions</li>
<li>instead of overall, compute % for each gene separately, and plot the CDF</li>
</ol>
</blockquote>
<pre class="r"><code>Intron.Annotations.basic &lt;- read_tsv(&quot;../code/SplicingAnalysis/regtools_annotate_combined/basic.bed.gz&quot;) %&gt;%
  filter(known_junction ==1) %&gt;%
  unite(intron, chrom, start, end, strand)
Introns.Annotations.comprehensive &lt;- read_tsv(&quot;../code/SplicingAnalysis/regtools_annotate_combined/comprehensive.bed.gz&quot;) %&gt;%
  filter(known_junction ==1) %&gt;%
  unite(intron, chrom, start, end, strand)
Introns.Annotations.all &lt;- read_tsv(&quot;../code/SplicingAnalysis/regtools_annotate_combined/comprehensive.bed.gz&quot;) %&gt;%
  unite(intron, chrom, start, end, strand)

AddIntronAnnotations &lt;- function(df){
  df %&gt;%
  mutate(intron = str_replace(junc, &quot;^(.+):(.+?):(.+?):clu.+?_([+-])$&quot;, &quot;\\1_\\2_\\3_\\4&quot;)) %&gt;%
  mutate(IntronAnnotation = case_when(
    intron %in% NMD.specific.introns ~ &quot;Annotated NMD&quot;,
    intron %in% Intron.Annotations.basic$intron ~ &quot;Annotated basic&quot;,
    intron %in% Introns.Annotations.comprehensive$intron ~ &quot;Annotated Not basic&quot;,
    TRUE ~ &quot;Unannotated&quot;
  )) %&gt;%
  dplyr::select(1:6, intron, IntronAnnotation, everything())
}

Long.table &lt;- lapply(SpliceJunctionCountTables, AddIntronAnnotations) %&gt;%
  lapply(pivot_longer,names_to=&quot;Sample&quot;, values_to=&quot;Count&quot;, -c(1:8)) %&gt;%
  bind_rows(.id=&quot;Dataset&quot;)</code></pre>
<p>And now make those plots</p>
<pre class="r"><code>P.i.dat &lt;- Long.table %&gt;%
  group_by(Sample, Dataset, IntronAnnotation) %&gt;%
  summarise(SumCounts = sum(Count)) %&gt;%
  ungroup() %&gt;%
  group_by(Sample, Dataset) %&gt;%
  mutate(Percent = SumCounts / sum(SumCounts) * 100) %&gt;%
  ungroup()
P.i &lt;- ggplot(P.i.dat, aes(x=Dataset, y=Percent, color=IntronAnnotation)) +
  geom_jitter(alpha=0.2, size=0.5) +
  geom_boxplot(outlier.shape=NA, color=&#39;black&#39;, fill=NA) +
  facet_wrap(~IntronAnnotation, scales=&quot;free_y&quot;) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  labs(y=&quot;Percent of splice junction reads&quot;)
P.i</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/8df7e43cc493d35d443d132d58c6f327ee406d9f/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-4-1.png" target="_blank">8df7e43</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/3201a1943ba4ad08cdd1d521bbacbc97ad8e3689/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-4-1.png" target="_blank">3201a19</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>genes &lt;- read_tsv(&quot;../code/ExpressionAnalysis/polyA/ExpressedGeneList.txt&quot;, col_names=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;name&quot;)) %&gt;%
  mutate(name = str_replace(name, &quot;^(E.+?)\\..+$&quot;, &quot;\\1&quot;))

p.ii.dat.joined &lt;- Introns.Annotations.all %&gt;%
  separate_rows(gene_id, sep = &quot;,&quot;) %&gt;%
  mutate(gene_id = str_replace(gene_id, &quot;^(E.+?)\\..+$&quot;, &quot;\\1&quot;)) %&gt;%
  filter(gene_id %in% genes$name) %&gt;%
  inner_join(Long.table)

p.ii.dat &lt;- p.ii.dat.joined %&gt;%
  group_by(gene_id, Dataset, IntronAnnotation) %&gt;%
  summarise(SumCounts = sum(Count, na.rm=F)) %&gt;%
  ungroup() %&gt;%
  group_by(Dataset, gene_id) %&gt;%
  mutate(Percent = SumCounts / sum(SumCounts, na.rm=F) * 100) %&gt;%
  ungroup()

p.ii &lt;- ggplot(p.ii.dat, aes(x=Percent, color=Dataset)) +
  stat_ecdf() +
  facet_wrap(~IntronAnnotation) +
  theme_bw() +
  labs(y=&quot;ecdf (14000 ProteinCodingGenes)&quot;, x=&quot;Percent splice junctions within gene&quot;)
p.ii</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-2">
Past versions of unnamed-chunk-4-2.png
</button>
</p>
<div id="fig-unnamed-chunk-4-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/8df7e43cc493d35d443d132d58c6f327ee406d9f/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-4-2.png" target="_blank">8df7e43</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/3201a1943ba4ad08cdd1d521bbacbc97ad8e3689/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-4-2.png" target="_blank">3201a19</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(&quot;figure/orig/SpliceJunctionsByTypeAndDataset.pdf&quot;, P.i, height=4, width=5.5)

ggsave(&quot;figure/orig/SpliceJunctionsByGeneTypeAndDataset.pdf&quot;, p.ii, height=4, width=5.5)</code></pre>
<p>let’s prcoeed to reading in other files</p>
<pre class="r"><code>PhenotypeAliases &lt;- read_tsv(&quot;../data/Phenotypes_recode_for_Plotting.txt&quot;)

PC.ShortAliases &lt;- PhenotypeAliases %&gt;%
  dplyr::select(PC, ShorterAlias) %&gt;% deframe()

coloc.results &lt;- read_tsv(&quot;../code/hyprcoloc/Results/ForColoc/MolColocStandard/results.txt.gz&quot;)

coloc.results.tidycolocalized &lt;- read_tsv(&quot;../code/hyprcoloc/Results/ForColoc/MolColocStandard/tidy_results_OnlyColocalized.txt.gz&quot;) %&gt;%
  separate(phenotype_full, into=c(&quot;PC&quot;, &quot;P&quot;), sep=&quot;;&quot;)

finemap.snps.annotated &lt;- read_tsv(&quot;../code/QTL_SNP_Enrichment/FinemapIntersections/MolColocStandard.bed.gz&quot;, col_names=c(&quot;SNPchrom&quot;, &quot;SNPstart&quot;, &quot;SNPstop&quot;, &quot;SNP_iteration_locus&quot;, &quot;FinemapPP&quot;, &quot;AnnotationChrom&quot;, &quot;AnnotationStart&quot;, &quot;AnnotatioStop&quot;, &quot;AnnotationClass&quot;, &quot;Overlap&quot;)) %&gt;%
  dplyr::select(-Overlap)</code></pre>
<p>Count chromatin-colocalizing and splicing-colocalizing eQTLs, and recreate previous observation about concordant effects</p>
<pre class="r"><code>coloc.results.tidycolocalized %&gt;%
  group_by(Locus, snp) %&gt;%
  filter(any(PC==&quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
  summarise(
    ContainsChromatinEqtl = any(PC %in% c(&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;)),
    ContainsSqtl = any(PC %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;))
    ) %&gt;%
  ggplot(aes(x=1, fill=paste(ContainsChromatinEqtl, ContainsSqtl))) +
  geom_bar() +
  labs(title=&quot;More chromatin localization with eQTLs than splicing&quot;, y=&quot;Number of colocalizing eQTLs&quot;) +
  theme_classic() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/8df7e43cc493d35d443d132d58c6f327ee406d9f/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-7-1.png" target="_blank">8df7e43</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/3201a1943ba4ad08cdd1d521bbacbc97ad8e3689/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-7-1.png" target="_blank">3201a19</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>coloc.results.tidycolocalized %&gt;%
  group_by(Locus, snp) %&gt;%
  filter(any(PC==&quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
  ungroup() %&gt;%
  filter(PC %in% c(&quot;Expression.Splicing.Subset_YRI&quot;,&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;)) %&gt;%
  left_join(., ., by=c(&quot;Locus&quot;, &quot;snp&quot;)) %&gt;%
  filter(!((P.x == P.y) &amp; (PC.x == PC.y))) %&gt;% 
  group_by(Locus, snp) %&gt;%
  mutate(
    ContainsChromatinEqtl = any(PC.x %in% c(&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;)),
    ContainsSqtl = any(PC.x %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;))
    ) %&gt;%
  ungroup() %&gt;%
  mutate(Contains.eQTL_Contains.sQTL = paste(ContainsChromatinEqtl, ContainsSqtl)) %&gt;%
  group_by(Contains.eQTL_Contains.sQTL, PC.x, PC.y) %&gt;%
  summarise(cor = cor(beta.x, beta.y, method=&quot;spearman&quot;)) %&gt;%
  mutate(PC.x = recode(PC.x, !!!PC.ShortAliases)) %&gt;%
  mutate(PC.y = recode(PC.y, !!!PC.ShortAliases)) %&gt;%
  ggplot(aes(x=PC.x, y=PC.y, fill=cor)) +
  geom_raster() +
  scale_fill_gradient2() +
  facet_wrap(~Contains.eQTL_Contains.sQTL) +
  scale_x_discrete(expand=c(0,0)) +
  scale_y_discrete(expand=c(0,0)) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  labs(x=&quot;TraitA Phenotype class&quot;, y=&quot;TraitB Phenotype class&quot;, fill=&quot;Spearman cor&quot;, title=&quot;Effect size correlation&quot;,
       caption = &quot;FacetTitle: IsChromatinQTL IsSplicingQTL&quot;)</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-2">
Past versions of unnamed-chunk-7-2.png
</button>
</p>
<div id="fig-unnamed-chunk-7-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/8df7e43cc493d35d443d132d58c6f327ee406d9f/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-7-2.png" target="_blank">8df7e43</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Now split QTLs by location of SNP as either in splice site, in enhancer/promoter, or neither</p>
<pre class="r"><code>#annotation types
finemap.snps.annotated$AnnotationClass %&gt;% unique()</code></pre>
<pre><code> [1] &quot;SpliceBranchpointRegion_0&quot; &quot;10_Txn_Elongation&quot;        
 [3] &quot;6_Weak_Enhancer&quot;           &quot;SpliceDonor_0&quot;            
 [5] &quot;2_Weak_Promoter&quot;           &quot;11_Weak_Txn&quot;              
 [7] &quot;1_Active_Promoter&quot;         &quot;12_Repressed&quot;             
 [9] &quot;4_Strong_Enhancer&quot;         &quot;13_Heterochrom/lo&quot;        
[11] &quot;SpliceAcceptor_0&quot;          &quot;SpliceBranchpointRegion_1&quot;
[13] &quot;ncRNA_coRNA&quot;               &quot;ncRNA_pseudo&quot;             
[15] &quot;9_Txn_Transition&quot;          &quot;SpliceAcceptor_1&quot;         
[17] &quot;ncRNA_srtRNA&quot;              &quot;8_Insulator&quot;              
[19] &quot;ncRNA_uaRNA&quot;               &quot;5_Strong_Enhancer&quot;        
[21] &quot;7_Weak_Enhancer&quot;           &quot;SpliceDonor_1&quot;            
[23] &quot;PAS_Region&quot;                &quot;ncRNA_incRNA&quot;             
[25] &quot;3_Poised_Promoter&quot;         &quot;.&quot;                        
[27] &quot;14_Repetitive/CNV&quot;         &quot;ncRNA_ctRNA&quot;              
[29] &quot;15_Repetitive/CNV&quot;         &quot;ncRNA_lncRNA&quot;             
[31] &quot;ncRNA_rtRNA&quot;               &quot;ncRNA_snoRNA&quot;             </code></pre>
<pre class="r"><code>coloc.results.tidycolocalized %&gt;%
  group_by(Locus, snp) %&gt;%
  filter(any(PC==&quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
  ungroup() %&gt;%
  filter(PC %in% c(&quot;Expression.Splicing.Subset_YRI&quot;,&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;)) %&gt;%
  left_join(., ., by=c(&quot;Locus&quot;, &quot;snp&quot;)) %&gt;%
  filter(!((P.x == P.y) &amp; (PC.x == PC.y))) %&gt;% 
  group_by(Locus, snp) %&gt;%
  mutate(
    ContainsChromatinEqtl = any(PC.x %in% c(&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;)),
    ContainsSqtl = any(PC.x %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;))
    ) %&gt;%
  ungroup() %&gt;%
  mutate(Contains.eQTL_Contains.sQTL = paste(ContainsChromatinEqtl, ContainsSqtl)) %&gt;%
  dplyr::select(-iteration.y) %&gt;%
  left_join(
    finemap.snps.annotated %&gt;%
      dplyr::select(SNP_iteration_locus, FinemapPP, AnnotationClass) %&gt;%
      separate(SNP_iteration_locus, into=c(&quot;snp&quot;, &quot;iteration.x&quot;, &quot;Locus&quot;), convert=T, sep=&quot;_&quot;)
  ) %&gt;%
  filter(FinemapPP &gt;0.25) %&gt;%
  mutate(AnnotationSuperclass = case_when(
    str_detect(AnnotationClass, &quot;Splice.+_1&quot;) ~ &quot;Annotated SS&quot;,
    str_detect(AnnotationClass, &quot;Splice.+_0$&quot;) ~ &quot;Unannotated SS&quot;,
    str_detect(AnnotationClass, &quot;Enhancer&quot;) ~ &quot;Enhancer&quot;,
    str_detect(AnnotationClass, &quot;Promoter&quot;) ~ &quot;Promoter&quot;,
    TRUE ~ &quot;Other&quot;
  )) %&gt;%
  mutate(PC.x = recode(PC.x, !!!PC.ShortAliases)) %&gt;%
  mutate(PC.y = recode(PC.y, !!!PC.ShortAliases)) %&gt;%
  mutate(Contains.eQTL_Contains.sQTL = recode(Contains.eQTL_Contains.sQTL,
                                              !!!c(&quot;TRUE TRUE&quot;=&quot;hQTL+sQTL&quot;,
                                                   &quot;TRUE FALSE&quot;=&quot;hQTL only&quot;,
                                                   &quot;FALSE TRUE&quot;=&quot;sQTL only&quot;))) %&gt;%
  group_by(Contains.eQTL_Contains.sQTL, PC.x, PC.y, AnnotationSuperclass) %&gt;%
  summarise(cor = cor(beta.x, beta.y, method=&quot;spearman&quot;), n=n()) %&gt;%
  ggplot(aes(x=PC.x, y=PC.y, fill=cor)) +
  geom_raster() +
  geom_text(aes(label=n), size=1.5) +
  scale_fill_gradient2() +
  facet_grid(rows=vars(Contains.eQTL_Contains.sQTL), cols=vars(AnnotationSuperclass)) +
  scale_x_discrete(expand=c(0,0)) +
  scale_y_discrete(expand=c(0,0)) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  labs(x=&quot;TraitA Phenotype class&quot;, y=&quot;TraitB Phenotype class&quot;, fill=&quot;Spearman cor&quot;, title=&quot;Effect size correlation for eQTLs that coloc with something&quot;,
       caption = str_wrap(&quot;FacetCols=Annotation class of SNP (Onlyfinemap PP &gt;0.25 included). FacetRows=Type of eQTL&quot;))</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/8df7e43cc493d35d443d132d58c6f327ee406d9f/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-9-1.png" target="_blank">8df7e43</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/c17b6b2fcd112a08cc7e7a2fb5bb45538dc3d339/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-9-1.png" target="_blank">c17b6b2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>I’m quite surprised just by the numbers of hQTL eQTLs with high finemap PP in unannoated splice site regions. This is clearly a really big annotation set that also includes lots of enhancer regions. This unannoated splice site region was defined as at least one spliced read mapping across all the data. Perhaps I shouldn’t be surprised, as any region is sufficeintly transcribed (including enhancers) will at some rate have some splice sites.</p>
<p>And even among the 7 intron retention QTLs that coloc with an eQTL but not hQTL, and are finemapped to annotated splice sites, the the direction of effects is if anything opposite what I was expecting</p>
<pre class="r"><code>coloc.results.tidycolocalized %&gt;%
  group_by(Locus, snp) %&gt;%
  filter(any(PC==&quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
  ungroup() %&gt;%
  filter(PC %in% c(&quot;Expression.Splicing.Subset_YRI&quot;,&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;)) %&gt;%
  left_join(., ., by=c(&quot;Locus&quot;, &quot;snp&quot;)) %&gt;%
  filter(!((P.x == P.y) &amp; (PC.x == PC.y))) %&gt;% 
  group_by(Locus, snp) %&gt;%
  mutate(
    ContainsChromatinEqtl = any(PC.x %in% c(&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;)),
    ContainsSqtl = any(PC.x %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;))
    ) %&gt;%
  ungroup() %&gt;%
  mutate(Contains.eQTL_Contains.sQTL = paste(ContainsChromatinEqtl, ContainsSqtl)) %&gt;%
  dplyr::select(-iteration.y) %&gt;%
  left_join(
    finemap.snps.annotated %&gt;%
      dplyr::select(SNP_iteration_locus, FinemapPP, AnnotationClass) %&gt;%
      separate(SNP_iteration_locus, into=c(&quot;snp&quot;, &quot;iteration.x&quot;, &quot;Locus&quot;), convert=T, sep=&quot;_&quot;)
  ) %&gt;%
  filter(FinemapPP &gt;0.25) %&gt;%
  mutate(AnnotationSuperclass = case_when(
    str_detect(AnnotationClass, &quot;Splice.+_1&quot;) ~ &quot;Annotated SS&quot;,
    str_detect(AnnotationClass, &quot;Splice.+_0$&quot;) ~ &quot;Unannotated SS&quot;,
    str_detect(AnnotationClass, &quot;Enhancer&quot;) ~ &quot;Enhancer&quot;,
    str_detect(AnnotationClass, &quot;Promoter&quot;) ~ &quot;Promoter&quot;,
    TRUE ~ &quot;Other&quot;
  )) %&gt;%
  mutate(PC.x = recode(PC.x, !!!PC.ShortAliases)) %&gt;%
  mutate(PC.y = recode(PC.y, !!!PC.ShortAliases)) %&gt;%
  mutate(Contains.eQTL_Contains.sQTL = recode(Contains.eQTL_Contains.sQTL,
                                              !!!c(&quot;TRUE TRUE&quot;=&quot;hQTL+sQTL&quot;,
                                                   &quot;TRUE FALSE&quot;=&quot;hQTL only&quot;,
                                                   &quot;FALSE TRUE&quot;=&quot;sQTL only&quot;))) %&gt;%
  filter(Contains.eQTL_Contains.sQTL==&quot;sQTL only&quot; &amp; AnnotationSuperclass==&quot;Annotated SS&quot;) %&gt;%
  ggplot(aes(x=beta.x, y=beta.y)) +
  geom_point() +
  geom_vline(xintercept=0) +
  geom_hline(yintercept=0) +
  facet_grid(rows=vars(PC.x), cols=vars(PC.y)) +
  theme_bw() +
  labs(x=&quot;TraitA Phenotype class&quot;, y=&quot;TraitB Phenotype class&quot;, title=&quot;Effect size correlation for eQTLs/sQTLs that are not hQTLs&quot;,
       caption = str_wrap(&quot;Only SNPs with finemap PP &gt; 0.25 in annotated splice site SNP included&quot;))</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/c17b6b2fcd112a08cc7e7a2fb5bb45538dc3d339/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-10-1.png" target="_blank">c17b6b2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Note there are only a few points of chRNA.IR intron retention with SNPs in splice sites that coloc with eQTL.</p>
<p>Let’s check how these plots look for NMD-specific introns…</p>
<pre class="r"><code>dat.to.plot &lt;- coloc.results.tidycolocalized %&gt;%
  group_by(Locus, snp) %&gt;%
  filter(any(PC==&quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
  ungroup() %&gt;%
  filter(PC %in% c(&quot;Expression.Splicing.Subset_YRI&quot;,&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;)) %&gt;%
  left_join(., ., by=c(&quot;Locus&quot;, &quot;snp&quot;)) %&gt;%
  filter(!((P.x == P.y) &amp; (PC.x == PC.y))) %&gt;% 
  group_by(Locus, snp) %&gt;%
  mutate(
    ContainsChromatinEqtl = any(PC.x %in% c(&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;)),
    ContainsSqtl = any(PC.x %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;))
    ) %&gt;%
  ungroup() %&gt;%
  mutate(Contains.eQTL_Contains.sQTL = paste(ContainsChromatinEqtl, ContainsSqtl)) %&gt;%
  dplyr::select(-iteration.y) %&gt;%
  left_join(
    finemap.snps.annotated %&gt;%
      dplyr::select(SNP_iteration_locus, FinemapPP, AnnotationClass) %&gt;%
      separate(SNP_iteration_locus, into=c(&quot;snp&quot;, &quot;iteration.x&quot;, &quot;Locus&quot;), convert=T, sep=&quot;_&quot;)
  ) %&gt;%
  filter(FinemapPP &gt;0.25) %&gt;%
  mutate(AnnotationSuperclass = case_when(
    str_detect(AnnotationClass, &quot;Splice.+_1&quot;) ~ &quot;Annotated SS&quot;,
    str_detect(AnnotationClass, &quot;Splice.+_0$&quot;) ~ &quot;Unannotated SS&quot;,
    str_detect(AnnotationClass, &quot;Enhancer&quot;) ~ &quot;Enhancer&quot;,
    str_detect(AnnotationClass, &quot;Promoter&quot;) ~ &quot;Promoter&quot;,
    TRUE ~ &quot;Other&quot;
  )) %&gt;%
  mutate(PC.x = recode(PC.x, !!!PC.ShortAliases)) %&gt;%
  mutate(PC.y = recode(PC.y, !!!PC.ShortAliases)) %&gt;%
  mutate(Contains.eQTL_Contains.sQTL = recode(Contains.eQTL_Contains.sQTL,
                                              !!!c(&quot;TRUE TRUE&quot;=&quot;hQTL+sQTL&quot;,
                                                   &quot;TRUE FALSE&quot;=&quot;hQTL only&quot;,
                                                   &quot;FALSE TRUE&quot;=&quot;sQTL only&quot;))) %&gt;%
  filter(Contains.eQTL_Contains.sQTL==&quot;sQTL only&quot;) %&gt;%
  mutate(New.x = str_replace(P.x, &quot;^(.+?):(.+?):(.+?):.+?_([-+])$&quot;, &quot;chr\\1_\\2_\\3_\\4&quot;)) %&gt;%
  mutate(New.y = str_replace(P.x, &quot;^(.+?):(.+?):(.+?):.+?_([-+])$&quot;, &quot;chr\\1_\\2_\\3_\\4&quot;)) %&gt;%
  mutate(ContainsNMDJunc = (New.x %in% NMD.specific.introns) | (New.y %in% NMD.specific.introns)) %&gt;%
  arrange(ContainsNMDJunc)
ggplot(dat.to.plot, aes(x=beta.x, y=beta.y, color=ContainsNMDJunc)) +
  geom_point(alpha=0.5) +
  geom_vline(xintercept=0) +
  geom_hline(yintercept=0) +
  geom_smooth(method=&#39;lm&#39;) +
  facet_grid(rows=vars(PC.x), cols=vars(PC.y)) +
  theme_bw() +
  labs(x=&quot;TraitA Phenotype class&quot;, y=&quot;TraitB Phenotype class&quot;, caption=&quot;Effect size correlation for eQTLs/sQTLs that are not hQTLs&quot;, title=&quot;sQTLs that increase annotated NMD junctions coloc with downward eQTLs&quot;)</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/8df7e43cc493d35d443d132d58c6f327ee406d9f/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-11-1.png" target="_blank">8df7e43</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/c17b6b2fcd112a08cc7e7a2fb5bb45538dc3d339/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-11-1.png" target="_blank">c17b6b2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>So there’s clearly a negative correlation between NMD-specific intron splicing and expression. But there’s only so many NMD introns annotated, and I expect a lot of splice changes due to genetic variation to be unannotated. Additionally, a lot of the red dots might the necessary down-regulated junctions when a NMD-junction gets up-regulated.</p>
<p>Let’s plot all the blue points with pygenometracks</p>
<pre class="r"><code>dir.create(&quot;../code/scratch/PlotNMDsQTLeQTLs/&quot;)

dat.to.plot.filtered &lt;- dat.to.plot %&gt;%
  filter(ContainsNMDJunc) %&gt;%
  filter(PC.y %in% &quot;Expression_polyA&quot;) %&gt;%
  distinct(Locus, .keep_all=T)

#Write out all juncs in clusters
dat.to.plot.filtered %&gt;%
  mutate(gid = str_replace(P.x, &quot;^(.+?):.+?:.+?:(.+?)&quot;, &quot;chr\\1_\\2&quot;)) %&gt;%
  inner_join(
    SpliceJunctionCountTables$Expression.Splicing %&gt;% dplyr::select(junc, gid)
  ) %&gt;%
  dplyr::select(junc) %&gt;%
  separate(junc, into=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;name&quot;), sep=&quot;:&quot;, convert=T) %&gt;%
  arrange(chrom, start, stop) %&gt;%
  write_tsv(&quot;../code/scratch/PlotNMDsQTLeQTLs/.juncs.bed&quot;)

# # Write out NMD introns
# dat.to.plot.filtered %&gt;%
#   dplyr::select(junc = P.x) %&gt;%
#   separate(junc, into=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;name&quot;), sep=&quot;:&quot;, convert=T) %&gt;%
#   mutate(chrom=paste0(&quot;chr&quot;, chrom)) %&gt;%
#   dplyr::select(chrom, start=newstart, stop) %&gt;%
#   arrange(chrom, start, stop) %&gt;%
#   write_tsv(&quot;../code/scratch/PlotNMDsQTLeQTLs/.NMD.introns.bed&quot;, col_names = F)
# 
# # Write out ini file for NMD introns track
# fileConn&lt;-file(&quot;../code/scratch/PlotNMDsQTLeQTLs/.NMD.introns.ini&quot;)
# writeLines(c(&quot;[NMD.introns]&quot;,&quot;file_type = bed&quot;, &quot;type = vlines&quot;, &quot;color=red&quot;,&quot;file = scratch/PlotNMDsQTLeQTLs/.NMD.introns.bed&quot;), con=fileConn)
# close(fileConn)

# Write out NMD introns
dat.to.plot.filtered %&gt;%
  dplyr::select(junc = P.x) %&gt;%
  separate(junc, into=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;name&quot;), sep=&quot;:&quot;, convert=T) %&gt;%
  mutate(chrom=paste0(&quot;chr&quot;, chrom)) %&gt;%
  arrange(chrom, start, stop) %&gt;%
  write_tsv(&quot;../code/scratch/PlotNMDsQTLeQTLs/.NMD.introns.bed&quot;, col_names = F)

# Write out ini file for NMD introns track
fileConn&lt;-file(&quot;../code/scratch/PlotNMDsQTLeQTLs/.NMD.introns.ini&quot;)
writeLines(c(&quot;[NMD.introns]&quot;,&quot;file_type = bed&quot;, &quot;color=red&quot;,&quot;file = scratch/PlotNMDsQTLeQTLs/.NMD.introns.bed&quot;), con=fileConn)
close(fileConn)

# Write out bash script
dat.to.plot.filtered %&gt;%
  separate(P.x, into=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;name&quot;), sep=&quot;:&quot;, convert=T) %&gt;%
  mutate(chrom=paste0(&quot;chr&quot;, chrom)) %&gt;%
  mutate(min=start-10000, max=stop+10000) %&gt;%
  mutate(SnpPos = str_replace(snp, &quot;^(.+?:.+?):.+?:.+?$&quot;, &quot;chr\\1&quot;)) %&gt;%
  mutate(rn = str_pad(row_number(), 2, pad=0)) %&gt;%
  mutate(cmd = str_glue(&quot;python scripts/GenometracksByGenotype/AggregateBigwigsForPlotting.py --VCF Genotypes/1KG_GRCh38/Autosomes.vcf.gz --SnpPos {SnpPos} --GroupSettingsFile Metaplots/bwGroups.tsv --BigwigList Metaplots/bwList.tsv --Normalization WholeGenome --Region {chrom}:{min}-{max} --BigwigListType KeyFile --OutputPrefix scratch/PlotNMDsQTLeQTLs/. -v --Bed12GenesToIni scripts/GenometracksByGenotype/PremadeTracks/gencode.v26.FromGTEx.genes.bed12.gz  --FilterJuncsByBed scratch/PlotNMDsQTLeQTLs/.juncs.bed\npyGenomeTracks --tracks &lt;(cat scratch/PlotNMDsQTLeQTLs/.tracks.ini scratch/PlotNMDsQTLeQTLs/.NMD.introns.ini) --out scratch/PlotNMDsQTLeQTLs/Plot_{rn}_{Locus}.pdf --region {chrom}:{min}-{max} --trackLabelFraction 0.15\n\n&quot;)) %&gt;%
  select(cmd) %&gt;%
  write.table(&quot;../code/scratch/PlotNMDsQTLeQTLs/MakePlots.sh&quot;, quote=F, row.names = F, col.names = F)</code></pre>
<p>..And plot them</p>
<pre class="bash"><code>conda activate GenometracksByGenotype
cd /project2/yangili1/bjf79/ChromatinSplicingQTLs/code
bash scratch/PlotNMDsQTLeQTLs/MakePlots.sh
</code></pre>
<p>And Let’s also expand these results by taking all NMD sQTLs and consider the beta on gene expression…</p>
<pre class="r"><code>sQTLs &lt;- c(&quot;../code/QTLs/QTLTools/polyA.Splicing.Subset_YRI/PermutationPassForColoc.txt.gz&quot;, &quot;../code/QTLs/QTLTools/chRNA.Splicing/PermutationPassForColoc.txt.gz&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/QTLs/QTLTools/(.+?)/PermutationPassForColoc.txt.gz&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(fread) %&gt;%
  bind_rows(.id=&quot;source&quot;) %&gt;%
  mutate(gene = str_replace(phe_id, &quot;.+:(EN.+?)$&quot;, &quot;\\1&quot;)) %&gt;%
  mutate(intron = str_replace(phe_id, &quot;^(.+):(.+?):(.+?):clu.+?_([+-]):EN.+$&quot;, &quot;chr\\1_\\2_\\3_\\4&quot;)) %&gt;%
  group_by(source) %&gt;%
  mutate(q = qvalue(adj_emp_pval)$qvalues) %&gt;%
  ungroup()


eQTLs &lt;- c(&quot;../code/QTLs/QTLTools/Expression.Splicing.Subset_YRI/PermutationPassForColoc.txt.gz&quot;, &quot;../code/QTLs/QTLTools/chRNA.Expression.Splicing/PermutationPassForColoc.txt.gz&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/QTLs/QTLTools/(.+?)/PermutationPassForColoc.txt.gz&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(fread) %&gt;%
  bind_rows(.id=&quot;source&quot;) %&gt;%
  mutate(gene = str_replace(phe_id, &quot;.+:(EN.+?)$&quot;, &quot;\\1&quot;)) %&gt;%
  group_by(source) %&gt;%
  mutate(q = qvalue(adj_emp_pval)$qvalues) %&gt;%
  ungroup()

H3K4me3.QTLs &lt;- c(&quot;../code/QTLs/QTLTools/H3K4ME3/PermutationPassForColoc.txt.gz&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/QTLs/QTLTools/(.+?)/PermutationPassForColoc.txt.gz&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(fread) %&gt;%
  bind_rows(.id=&quot;source&quot;) %&gt;%
  mutate(gene = str_replace(phe_id, &quot;.+:(EN.+?)$&quot;, &quot;\\1&quot;)) %&gt;%
  mutate(gene_peak = str_replace(phe_id, &quot;^(.+?):(.+?)$&quot;, &quot;\\2;\\1&quot;)) %&gt;%
  group_by(source) %&gt;%
  mutate(q = qvalue(adj_emp_pval)$qvalues) %&gt;%
  ungroup()

PeaksToTSS &lt;- Sys.glob(&quot;../code/Misc/PeaksClosestToTSS/*_assigned.tsv.gz&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/Misc/PeaksClosestToTSS/(.+?)_assigned.tsv.gz&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(read_tsv) %&gt;%
  bind_rows(.id=&quot;ChromatinMark&quot;) %&gt;%
  mutate(GenePeakPair = paste(gene, peak, sep = &quot;;&quot;)) %&gt;%
  distinct(ChromatinMark, peak, gene, .keep_all=T)

inner_join(eQTLs, sQTLs, by=&quot;gene&quot;, suffix=c(&quot;.eQTL&quot;, &quot;.sQTL&quot;)) %&gt;%
  mutate(Is.NMD.intron = intron %in% NMD.specific.introns) %&gt;%
  filter(q.eQTL &lt; 0.1 &amp; q.sQTL &lt; 0.1) %&gt;%
  mutate(source.eQTL = recode(source.eQTL, !!!c(&quot;polyA.Expression.YRI&quot;=&quot;Expression.Splicing.Subset_YRI&quot;, &quot;chRNA.Expression&quot;=&quot;chRNA.Expression.Splicing&quot;))) %&gt;%
  arrange(Is.NMD.intron) %&gt;%
  ggplot(aes(x=slope.eQTL, y=slope.sQTL, color=Is.NMD.intron)) +
  geom_point(alpha=0.5) +
  geom_vline(xintercept=0) +
  geom_hline(yintercept=0) +
  geom_smooth(method=&#39;lm&#39;) +
  facet_grid(rows = vars(source.eQTL), cols=vars(source.sQTL)) +
  theme_bw() +
  labs(title=&quot;genetic effects of NMD introns and host gene expression&quot;, x=&quot;beta, eQTL&quot;, y=&quot;beta, sQTL&quot;, caption=&quot;Only FDR&lt;10% sQTLs and eQTLs plotted&quot;)</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/8df7e43cc493d35d443d132d58c6f327ee406d9f/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-14-1.png" target="_blank">8df7e43</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/c17b6b2fcd112a08cc7e7a2fb5bb45538dc3d339/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-14-1.png" target="_blank">c17b6b2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>inner_join(eQTLs, H3K4me3.QTLs, by=&quot;gene&quot;, suffix=c(&quot;.eQTL&quot;, &quot;.sQTL&quot;)) %&gt;%
  filter(gene_peak %in% PeaksToTSS$GenePeakPair) %&gt;%
  filter(q.eQTL &lt; 0.1 &amp; q.sQTL &lt; 0.1) %&gt;%
  mutate(source.eQTL = recode(source.eQTL, !!!c(&quot;polyA.Expression.YRI&quot;=&quot;Expression.Splicing.Subset_YRI&quot;, &quot;chRNA.Expression&quot;=&quot;chRNA.Expression.Splicing&quot;))) %&gt;%
  ggplot(aes(x=slope.eQTL, y=slope.sQTL)) +
  geom_point(alpha=0.5) +
  geom_vline(xintercept=0) +
  geom_hline(yintercept=0) +
  geom_smooth(method=&#39;lm&#39;) +
  facet_grid(rows = vars(source.eQTL), cols=vars(source.sQTL)) +
  theme_bw() +
  labs(title=&quot;genetic effects of H3K4ME promoters and host gene expression&quot;, x=&quot;beta, eQTL&quot;, y=&quot;beta, H3K4ME3 QTL&quot;, caption=&quot;Only FDR&lt;10% H3K4ME3 QTLs and eQTLs plotted&quot;)</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-2">
Past versions of unnamed-chunk-14-2.png
</button>
</p>
<div id="fig-unnamed-chunk-14-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/8df7e43cc493d35d443d132d58c6f327ee406d9f/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-14-2.png" target="_blank">8df7e43</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/c17b6b2fcd112a08cc7e7a2fb5bb45538dc3d339/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-14-2.png" target="_blank">c17b6b2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Wait, upon further thought, I think I didn’t make the above plots correctly. I was just plotting the beta for the topSNP for each feature, not the same SNP between sQTL and eQTL, so the directions of SNP effect (which are always ALT/REF) might not be polarized the same way.</p>
<pre class="r"><code>TopSNPEffects.ByPairs &lt;- fread(&quot;../code/pi1/PairwisePi1Traits.P.all.txt.gz&quot;)

TopSNPEffects.ByPairs %&gt;%
  filter(
      (PC1 %in% c(&quot;chRNA.Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;)) &amp;
      (PC2 %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;)) &amp;
      (FDR.x &lt; 0.1) &amp;
      (FDR.y &lt; 0.1)) %&gt;%
  mutate(source.eQTL = recode(PC1, !!!c(&quot;Expression.Splicing.Subset_YRI&quot;=&quot;polyA.Expression.YRI&quot;, &quot;chRNA.Expression.Splicing&quot;=&quot;chRNA.Expression&quot;))) %&gt;%
  rename(source.sQTL=PC2) %&gt;%
  mutate(intron = str_replace(P2, &quot;^(.+):(.+?):(.+?):clu.+?_([+-])$&quot;, &quot;chr\\1_\\2_\\3_\\4&quot;)) %&gt;%
  mutate(Is.NMD.intron = intron %in% NMD.specific.introns) %&gt;%
  arrange(Is.NMD.intron) %&gt;%
  ggplot(aes(x=beta.x, y=trait.x.beta.in.y
, color=Is.NMD.intron)) +
  geom_point(alpha=0.5) +
  geom_vline(xintercept=0) +
  geom_hline(yintercept=0) +
  geom_smooth(method=&#39;lm&#39;) +
  facet_grid(rows = vars(source.eQTL), cols=vars(source.sQTL)) +
  theme_bw() +
  labs(title=&quot;genetic effects of NMD introns and host gene expression&quot;, x=&quot;beta, eQTL&quot;, y=&quot;beta, sQTL&quot;, caption=&quot;Only FDR&lt;10% sQTLs and eQTLs plotted. sQTL effects are relative to top eQTL SNP&quot;)</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/8df7e43cc493d35d443d132d58c6f327ee406d9f/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-15-1.png" target="_blank">8df7e43</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/c17b6b2fcd112a08cc7e7a2fb5bb45538dc3d339/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-15-1.png" target="_blank">c17b6b2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>TopSNPEffects.ByPairs %&gt;%
  filter(
      (PC2 %in% c(&quot;chRNA.Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;)) &amp;
      (PC1 %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;)) &amp;
      (FDR.x &lt; 0.1) &amp;
      (FDR.y &lt; 0.1)) %&gt;%
  mutate(source.eQTL = recode(PC2, !!!c(&quot;Expression.Splicing.Subset_YRI&quot;=&quot;polyA.Expression.YRI&quot;, &quot;chRNA.Expression.Splicing&quot;=&quot;chRNA.Expression&quot;))) %&gt;%
  rename(source.sQTL=PC1) %&gt;%
  mutate(intron = str_replace(P1, &quot;^(.+):(.+?):(.+?):clu.+?_([+-])$&quot;, &quot;chr\\1_\\2_\\3_\\4&quot;)) %&gt;%
  mutate(Is.NMD.intron = intron %in% NMD.specific.introns) %&gt;%
  arrange(Is.NMD.intron) %&gt;%
  ggplot(aes(y=beta.x, x=trait.x.beta.in.y
, color=Is.NMD.intron)) +
  geom_point(alpha=0.5) +
  geom_vline(xintercept=0) +
  geom_hline(yintercept=0) +
  geom_smooth(method=&#39;lm&#39;) +
  facet_grid(rows = vars(source.eQTL), cols=vars(source.sQTL)) +
  theme_bw() +
  labs(title=&quot;genetic effects of NMD introns and host gene expression&quot;, x=&quot;beta, eQTL&quot;, y=&quot;beta, sQTL&quot;, caption=&quot;Only FDR&lt;10% sQTLs and eQTLs plotted. eQTL effects are relative to top sQTL SNP&quot;)</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-2">
Past versions of unnamed-chunk-15-2.png
</button>
</p>
<div id="fig-unnamed-chunk-15-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/8df7e43cc493d35d443d132d58c6f327ee406d9f/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-15-2.png" target="_blank">8df7e43</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>I think I’ll want to further break down introns into - annotatic basic transcripts - annotated transripts (not NMD) - annotated NMD - unannotated</p>
<p>First read in the annotations…</p>
<pre class="r"><code>Intron.Annotations.basic &lt;- read_tsv(&quot;../code/SplicingAnalysis/regtools_annotate_combined/basic.bed.gz&quot;) %&gt;%
  filter(known_junction ==1) %&gt;%
  unite(intron, chrom, start, end, strand)
Introns.Annotations.comprehensive &lt;- read_tsv(&quot;../code/SplicingAnalysis/regtools_annotate_combined/comprehensive.bed.gz&quot;) %&gt;%
  filter(known_junction ==1) %&gt;%
  unite(intron, chrom, start, end, strand)</code></pre>
<p>Now remake the plots</p>
<pre class="r"><code>TopSNPEffects.ByPairs %&gt;%
  filter(
      (PC1 %in% c(&quot;chRNA.Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;)) &amp;
      (PC2 %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;)) &amp;
      (FDR.x &lt; 0.1) &amp;
      (FDR.y &lt; 0.1)) %&gt;%
  mutate(source.eQTL = recode(PC1, !!!c(&quot;Expression.Splicing.Subset_YRI&quot;=&quot;polyA.Expression.YRI&quot;, &quot;chRNA.Expression.Splicing&quot;=&quot;chRNA.Expression&quot;))) %&gt;%
  rename(source.sQTL=PC2) %&gt;%
  mutate(intron = str_replace(P2, &quot;^(.+):(.+?):(.+?):clu.+?_([+-])$&quot;, &quot;chr\\1_\\2_\\3_\\4&quot;)) %&gt;%
  mutate(IntronAnnotation = case_when(
    intron %in% NMD.specific.introns ~ &quot;Annotated NMD&quot;,
    intron %in% Intron.Annotations.basic$intron ~ &quot;Annotated basic&quot;,
    intron %in% Introns.Annotations.comprehensive$intron ~ &quot;Annotated Not basic&quot;,
    TRUE ~ &quot;Unannotated&quot;
  )) %&gt;%
  count(IntronAnnotation) %&gt;%
  ggplot(aes(x=IntronAnnotation, y=n, fill=IntronAnnotation)) +
  geom_col() +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  labs(title=&quot;Number sQTLs that are in an eGene&quot;, x=&quot;Intron Type&quot;, caption=&quot;FDR10% for sQTL and eQTL&quot;)</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/8df7e43cc493d35d443d132d58c6f327ee406d9f/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-17-1.png" target="_blank">8df7e43</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/c17b6b2fcd112a08cc7e7a2fb5bb45538dc3d339/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-17-1.png" target="_blank">c17b6b2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>TopSNPEffects.ByPairs %&gt;%
  filter(
      (PC1 %in% c(&quot;chRNA.Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;)) &amp;
      (PC2 %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;)) &amp;
      (FDR.x &lt; 0.1) &amp;
      (FDR.y &lt; 0.1)) %&gt;%
  mutate(source.eQTL = recode(PC1, !!!c(&quot;Expression.Splicing.Subset_YRI&quot;=&quot;polyA.Expression.YRI&quot;, &quot;chRNA.Expression.Splicing&quot;=&quot;chRNA.Expression&quot;))) %&gt;%
  rename(source.sQTL=PC2) %&gt;%
  mutate(intron = str_replace(P2, &quot;^(.+):(.+?):(.+?):clu.+?_([+-])$&quot;, &quot;chr\\1_\\2_\\3_\\4&quot;)) %&gt;%
  mutate(IntronAnnotation = case_when(
    intron %in% NMD.specific.introns ~ &quot;Annotated NMD&quot;,
    intron %in% Intron.Annotations.basic$intron ~ &quot;Annotated basic&quot;,
    intron %in% Introns.Annotations.comprehensive$intron ~ &quot;Annotated Not basic&quot;,
    TRUE ~ &quot;Unannotated&quot;
  )) %&gt;%
  ggplot(aes(y=beta.x, x=trait.x.beta.in.y
, color=IntronAnnotation)) +
  geom_point(alpha=0.5) +
  geom_vline(xintercept=0) +
  geom_hline(yintercept=0) +
  geom_smooth(method=&#39;lm&#39;) +
  facet_grid(rows = vars(source.eQTL), cols=vars(source.sQTL)) +
  theme_bw() +
  labs(title=&quot;genetic effects of NMD introns and host gene expression&quot;, x=&quot;beta, sQTL&quot;, y=&quot;beta, eQTL&quot;, caption=&quot;Only FDR&lt;10% sQTLs and eQTLs plotted. sQTL effects are relative to top eQTL SNP&quot;)</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-17-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-2">
Past versions of unnamed-chunk-17-2.png
</button>
</p>
<div id="fig-unnamed-chunk-17-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/8df7e43cc493d35d443d132d58c6f327ee406d9f/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-17-2.png" target="_blank">8df7e43</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>TopSNPEffects.ByPairs %&gt;%
  filter(
      (PC2 %in% c(&quot;chRNA.Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;)) &amp;
      (PC1 %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;)) &amp;
      (FDR.x &lt; 0.1) &amp;
      (FDR.y &lt; 0.1)) %&gt;%
  mutate(source.eQTL = recode(PC2, !!!c(&quot;Expression.Splicing.Subset_YRI&quot;=&quot;polyA.Expression.YRI&quot;, &quot;chRNA.Expression.Splicing&quot;=&quot;chRNA.Expression&quot;))) %&gt;%
  rename(source.sQTL=PC1) %&gt;%
  mutate(intron = str_replace(P1, &quot;^(.+):(.+?):(.+?):clu.+?_([+-])$&quot;, &quot;chr\\1_\\2_\\3_\\4&quot;)) %&gt;%
  mutate(IntronAnnotation = case_when(
    intron %in% NMD.specific.introns ~ &quot;Annotated NMD&quot;,
    intron %in% Intron.Annotations.basic$intron ~ &quot;Annotated basic&quot;,
    intron %in% Introns.Annotations.comprehensive$intron ~ &quot;Annotated Not basic&quot;,
    TRUE ~ &quot;Unannotated&quot;
  )) %&gt;%
  ggplot(aes(y=beta.x, x=trait.x.beta.in.y
, color=IntronAnnotation)) +
  geom_point(alpha=0.5) +
  geom_vline(xintercept=0) +
  geom_hline(yintercept=0) +
  geom_smooth(method=&#39;lm&#39;) +
  facet_grid(rows = vars(source.eQTL), cols=vars(source.sQTL)) +
  theme_bw() +
  labs(title=&quot;genetic effects of NMD introns and host gene expression&quot;, x=&quot;beta, sQTL&quot;, y=&quot;beta, eQTL&quot;, caption=&quot;Only FDR&lt;10% sQTLs and eQTLs plotted. eQTL effects are relative to top sQTL SNP&quot;)</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-17-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-3">
Past versions of unnamed-chunk-17-3.png
</button>
</p>
<div id="fig-unnamed-chunk-17-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/8df7e43cc493d35d443d132d58c6f327ee406d9f/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-17-3.png" target="_blank">8df7e43</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>There’s a few things I could do to improve this… one being that the effect sizes are standardized and therefore hard to interpret. Plotting eQTL effect sizes as logFC and splicing effect sizes as deltaPSI could be more interpretable. Also, Yang suggesting exploring these ideas by plotting things as QQ plots… let’s do that now…</p>
<p>For example, let’s plot eQTL P-values for different groups of SNPs, including sQTLs, hQTLs, etc…</p>
<pre class="r"><code>test.SNPs &lt;- paste0(&quot;../code/QTLs/QTLTools/&quot;, c(&quot;chRNA.Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;), &quot;/NominalPassForColoc.RandomSamplePvals.txt.gz&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/QTLs/QTLTools/(.+?)/NominalPassForColoc.RandomSamplePvals.txt.gz&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(read_tsv, col_names=c(&quot;trait.x.p.in.y&quot;)) %&gt;%
  bind_rows(.id=&quot;PC2&quot;) %&gt;%
  mutate(PC1 = &quot;TestSNPs&quot;) %&gt;%
  group_by(PC2) %&gt;%
  sample_n(5000) %&gt;%
  ungroup()

TopSNPEffects.ByPairs %&gt;%
  filter(
      (PC2 %in% c(&quot;chRNA.Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;)) &amp;
      (PC1 %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;H3K4ME3&quot;, &quot;H3K27AC&quot;)) &amp;
      (FDR.x &lt; 0.1)) %&gt;%
  bind_rows(test.SNPs) %&gt;%
  mutate(intron = case_when(
    PC1 %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;) ~ str_replace(P1, &quot;^(.+):(.+?):(.+?):clu.+?_([+-])$&quot;, &quot;chr\\1_\\2_\\3_\\4&quot;),
    TRUE ~ NA_character_)) %&gt;%
  mutate(IntronAnnotation = case_when(
    intron %in% NMD.specific.introns ~ &quot;Annotated NMD&quot;,
    intron %in% Intron.Annotations.basic$intron ~ &quot;Annotated basic&quot;,
    intron %in% Introns.Annotations.comprehensive$intron ~ &quot;Annotated Not basic&quot;,
    !is.na(intron) ~ &quot;Unannotated&quot;,
    TRUE ~ &quot;&quot;
  )) %&gt;%
  mutate(SnpSet = paste(PC1, IntronAnnotation)) %&gt;%
  group_by(SnpSet, PC2) %&gt;%
  mutate(ExpectedP = percent_rank(trait.x.p.in.y)) %&gt;%
  ungroup() %&gt;%
  ggplot(aes(x=-log10(ExpectedP), y=-log10(trait.x.p.in.y), color=SnpSet)) +
  geom_abline() +
  geom_point() +
  facet_wrap(~PC2) +
  theme_bw() +
  labs(title=&quot;QQ plot of eQTL P-values&quot;, color=&quot;xQTL SNP&quot;, y=&quot;-log10(P)&quot;)</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/8df7e43cc493d35d443d132d58c6f327ee406d9f/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-18-1.png" target="_blank">8df7e43</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok, hmm, is that even informative.. Well there is eQTL enrichment in all the types of QTLs tested. Let’s try breaking out the QQ plots by concordance of effect directions… I’ll plot the two mirror image QQ plots by concordance of effect directions</p>
<pre class="r"><code>QQ.plot.dat &lt;- TopSNPEffects.ByPairs %&gt;%
  filter(
      (PC2 %in% c(&quot;chRNA.Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;)) &amp;
      (PC1 %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;H3K4ME3&quot;, &quot;H3K27AC&quot;)) &amp;
      (FDR.x &lt; 0.1)) %&gt;%
  mutate(intron = case_when(
    PC1 %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;) ~ str_replace(P1, &quot;^(.+):(.+?):(.+?):clu.+?_([+-])$&quot;, &quot;chr\\1_\\2_\\3_\\4&quot;),
    TRUE ~ NA_character_)) %&gt;%
  mutate(IntronAnnotation = case_when(
    intron %in% NMD.specific.introns ~ &quot;Annotated NMD&quot;,
    intron %in% Intron.Annotations.basic$intron ~ &quot;Annotated basic&quot;,
    intron %in% Introns.Annotations.comprehensive$intron ~ &quot;Annotated Not basic&quot;,
    !is.na(intron) ~ &quot;Unannotated&quot;,
    TRUE ~ &quot;&quot;
  )) %&gt;%
  mutate(SnpSet = paste(PC1, IntronAnnotation)) %&gt;%
  mutate(Concordance = if_else(sign(beta.x)==sign(beta.y), &quot;Concordant&quot;, &quot;Discordant&quot;)) %&gt;%
  bind_rows(
    test.SNPs %&gt;%
      mutate(Concordance = &quot;Concordant&quot;, SnpSet=PC1),
    test.SNPs %&gt;%
      mutate(Concordance = &quot;Discordant&quot;, SnpSet=PC1)
    ) %&gt;%
  group_by(SnpSet, PC2, Concordance) %&gt;%
  mutate(ExpectedP = percent_rank(trait.x.p.in.y)) %&gt;%
  ungroup() %&gt;%
  mutate(Sign = recode(Concordance, !!!c(&quot;Discordant&quot;=-1, &quot;Concordant&quot;=1))) %&gt;%
  mutate(PC2 = recode(PC2, !!!c(&quot;chRNA.Expression.Splicing&quot;=&quot;chRNA&quot;, &quot;Expression.Splicing.Subset_YRI&quot;=&quot;polyA&quot;))) %&gt;%
  arrange(desc(trait.x.p.in.y))
ggplot(QQ.plot.dat, aes(x=-log10(ExpectedP)*Sign, y=-log10(trait.x.p.in.y)*Sign, color=SnpSet)) +
  geom_abline() +
  geom_point() +
  facet_wrap(~PC2) +
  theme_bw() +
  labs(title=&quot;QQ plot of eQTL P-values&quot;, color=&quot;xQTL SNP&quot;, y=&quot;Observed -log10(P)&quot;, x=&quot;Theoretical -log10(P)&quot;, caption=str_wrap(&quot;If eQTL and xQTL beta signs are same, right-side-up. If opposite, plot upside-down&quot;, 30))</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/8df7e43cc493d35d443d132d58c6f327ee406d9f/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-19-1.png" target="_blank">8df7e43</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>I like these plots but there is just too many colors… Let’s just plot sQTLs from polyA, to cut down on colors. Then seperately we can do just sQTLs from chRNA…</p>
<pre class="r"><code>QQ.plot.dat %&gt;%
  filter(!str_detect(SnpSet, &quot;chRNA&quot;)) %&gt;%
ggplot(aes(x=-log10(ExpectedP)*Sign, y=-log10(trait.x.p.in.y)*Sign, color=SnpSet)) +
  geom_abline() +
  geom_point() +
  facet_wrap(~PC2) +
  theme_bw() +
  labs(title=&quot;QQ plot of eQTL P-values&quot;, color=&quot;xQTL SNP&quot;, y=&quot;Observed -log10(P)&quot;, x=&quot;Theoretical -log10(P)&quot;, caption=str_wrap(&quot;If eQTL and xQTL beta signs are same, right-side-up. If opposite, plot upside-down&quot;, 30))</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/8df7e43cc493d35d443d132d58c6f327ee406d9f/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-20-1.png" target="_blank">8df7e43</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>QQ.plot.dat %&gt;%
  filter(!str_detect(SnpSet, &quot;polyA&quot;)) %&gt;%
ggplot(aes(x=-log10(ExpectedP)*Sign, y=-log10(trait.x.p.in.y)*Sign, color=SnpSet)) +
  geom_abline() +
  geom_point() +
  facet_wrap(~PC2) +
  theme_bw() +
  labs(title=&quot;QQ plot of eQTL P-values&quot;, color=&quot;xQTL SNP&quot;, y=&quot;Observed -log10(P)&quot;, x=&quot;Theoretical -log10(P)&quot;, caption=str_wrap(&quot;If eQTL and xQTL beta signs are same, right-side-up. If opposite, plot upside-down&quot;, 30))</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-20-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-2">
Past versions of unnamed-chunk-20-2.png
</button>
</p>
<div id="fig-unnamed-chunk-20-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/8df7e43cc493d35d443d132d58c6f327ee406d9f/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-20-2.png" target="_blank">8df7e43</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="enrichment-of-nmd-sqtls-in-eqtls" class="section level3">
<h3>enrichment of NMD sQTLs in eQTLs</h3>
<blockquote>
<p>Another way to show this, which would look more significant I think is to ask what % of non-eQTL sQTL affect NMD junctions of all non-eQTL sQTLs versus the same for sQTLs that are eQTLs</p>
</blockquote>
<pre class="r"><code>TopSNPEffects.ByPairs %&gt;%
  filter(
      (PC1 %in% c(&quot;chRNA.Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;))) %&gt;%
  # filter(
  #     (PC2 %in% c(&quot;chRNA.Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;)) &amp;
  #     (PC1 %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;)) ) %&gt;%
  filter(FDR.x &lt; 0.1) %&gt;%
  distinct(P1) %&gt;% nrow()</code></pre>
<pre><code>[1] 5358</code></pre>
<pre class="r"><code>TopSNPEffects.ByPairs %&gt;%
  filter(
      (PC1 %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;)) ) %&gt;%
  distinct(GeneLocus) %&gt;% nrow()</code></pre>
<pre><code>[1] 2621</code></pre>
<pre class="r"><code>datToPlot &lt;- TopSNPEffects.ByPairs %&gt;%
  filter(
      (PC2 %in% c(&quot;chRNA.Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;)) &amp;
      (PC1 %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;)) ) %&gt;%
  filter(FDR.x &lt; 0.1) %&gt;%
  mutate(intron = str_replace(P1, &quot;^(.+):(.+?):(.+?):clu.+?_([+-])$&quot;, &quot;chr\\1_\\2_\\3_\\4&quot;)) %&gt;%
  mutate(IntronAnnotation = case_when(
    intron %in% NMD.specific.introns ~ &quot;Annotated NMD&quot;,
    intron %in% Intron.Annotations.basic$intron ~ &quot;Annotated basic&quot;,
    intron %in% Introns.Annotations.comprehensive$intron ~ &quot;Annotated Not basic&quot;,
    TRUE ~ &quot;Unannotated&quot;
  )) %&gt;%
  mutate(Is_HostGene_eQTL = if_else(FDR.y &lt; 0.1, &quot;eGene&quot;, &quot;not eGene&quot;)) %&gt;%
  count(IntronAnnotation, Is_HostGene_eQTL) %&gt;%
  group_by(Is_HostGene_eQTL) %&gt;%
  mutate(Percent = n/sum(n)*100) %&gt;%
  mutate(rn = row_number()) %&gt;%
  ungroup()

ggplot(datToPlot, aes(x=Is_HostGene_eQTL, y=n, fill=IntronAnnotation)) +
  geom_col(position=&quot;stack&quot;) +
  geom_label(aes(label=signif(Percent, 3), vjust=rn), y=Inf) +
  theme_bw() +
  labs(y=&quot;Number sQTLs&quot;, caption=&quot;1406 eGenes identified in YRI with a test intron&quot;)</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/8df7e43cc493d35d443d132d58c6f327ee406d9f/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-21-1.png" target="_blank">8df7e43</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Let’s also break the original plot by ncRNAs for SNP regions</p>
<pre class="r"><code>coloc.results.tidycolocalized %&gt;%
  group_by(Locus, snp) %&gt;%
  filter(any(PC==&quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
  ungroup() %&gt;%
  filter(PC %in% c(&quot;Expression.Splicing.Subset_YRI&quot;,&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;)) %&gt;%
  left_join(., ., by=c(&quot;Locus&quot;, &quot;snp&quot;)) %&gt;%
  filter(!((P.x == P.y) &amp; (PC.x == PC.y))) %&gt;% 
  group_by(Locus, snp) %&gt;%
  mutate(
    ContainsChromatinEqtl = any(PC.x %in% c(&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;)),
    ContainsSqtl = any(PC.x %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;))
    ) %&gt;%
  ungroup() %&gt;%
  mutate(Contains.eQTL_Contains.sQTL = paste(ContainsChromatinEqtl, ContainsSqtl)) %&gt;%
  dplyr::select(-iteration.y) %&gt;%
  left_join(
    finemap.snps.annotated %&gt;%
      dplyr::select(SNP_iteration_locus, FinemapPP, AnnotationClass) %&gt;%
      separate(SNP_iteration_locus, into=c(&quot;snp&quot;, &quot;iteration.x&quot;, &quot;Locus&quot;), convert=T, sep=&quot;_&quot;)
  ) %&gt;%
  filter(FinemapPP &gt;0.25) %&gt;%
  mutate(AnnotationSuperclass = case_when(
    str_detect(AnnotationClass, &quot;Splice.+_1&quot;) ~ &quot;Annotated SS&quot;,
    str_detect(AnnotationClass, &quot;Splice.+_0$&quot;) ~ &quot;Unannotated SS&quot;,
    str_detect(AnnotationClass, &quot;Enhancer&quot;) ~ &quot;Enhancer&quot;,
    str_detect(AnnotationClass, &quot;Promoter&quot;) ~ &quot;Promoter&quot;,
    str_detect(AnnotationClass, &quot;ncRNA&quot;) ~ AnnotationClass,
    TRUE ~ &quot;Other&quot;
  )) %&gt;%
  group_by(Contains.eQTL_Contains.sQTL, PC.x, PC.y, AnnotationSuperclass) %&gt;%
  summarise(cor = cor(beta.x, beta.y, method=&quot;spearman&quot;), n=n()) %&gt;%
  mutate(PC.x = recode(PC.x, !!!PC.ShortAliases)) %&gt;%
  mutate(PC.y = recode(PC.y, !!!PC.ShortAliases)) %&gt;%
  mutate(Contains.eQTL_Contains.sQTL = recode(Contains.eQTL_Contains.sQTL,
                                              !!!c(&quot;TRUE TRUE&quot;=&quot;hQTL+sQTL&quot;,
                                                   &quot;TRUE FALSE&quot;=&quot;hQTL only&quot;,
                                                   &quot;FALSE TRUE&quot;=&quot;sQTL only&quot;))) %&gt;%
  ggplot(aes(x=PC.x, y=PC.y, fill=cor)) +
  geom_raster() +
  geom_text(aes(label=n), size=1.5) +
  scale_fill_gradient2() +
  facet_grid(rows=vars(Contains.eQTL_Contains.sQTL), cols=vars(AnnotationSuperclass)) +
  scale_x_discrete(expand=c(0,0)) +
  scale_y_discrete(expand=c(0,0)) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  labs(x=&quot;TraitA Phenotype class&quot;, y=&quot;TraitB Phenotype class&quot;, fill=&quot;Spearman cor&quot;, title=&quot;Effect size correlation for eQTLs that coloc with something&quot;,
       caption = str_wrap(&quot;FacetCols=Annotation class of SNP (Onlyfinemap PP &gt;0.25 included). FacetRows=Type of eQTL&quot;))</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-22-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/8df7e43cc493d35d443d132d58c6f327ee406d9f/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-22-1.png" target="_blank">8df7e43</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Perhaps we should plot the same ideas but breaking out the introns into classes by whether we have some good reason to think it would be an NMD target (ie unannoated, and or annotated with nonsense mediated decay transcript tag)</p>
<pre class="r"><code>coloc.results.tidycolocalized %&gt;%
  group_by(Locus, snp) %&gt;%
  filter(any(PC==&quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
  ungroup() %&gt;%
  filter(PC %in% c(&quot;Expression.Splicing.Subset_YRI&quot;,&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;)) %&gt;%
  left_join(., ., by=c(&quot;Locus&quot;, &quot;snp&quot;)) %&gt;%
  filter(!((P.x == P.y) &amp; (PC.x == PC.y))) %&gt;% 
  group_by(Locus, snp) %&gt;%
  mutate(
    ContainsChromatinEqtl = any(PC.x %in% c(&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;)),
    ContainsSqtl = any(PC.x %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;))
    ) %&gt;%
  ungroup() %&gt;%
  mutate(Contains.eQTL_Contains.sQTL = paste(ContainsChromatinEqtl, ContainsSqtl)) %&gt;%
  dplyr::select(-iteration.y) %&gt;%
  left_join(
    finemap.snps.annotated %&gt;%
      dplyr::select(SNP_iteration_locus, FinemapPP, AnnotationClass) %&gt;%
      separate(SNP_iteration_locus, into=c(&quot;snp&quot;, &quot;iteration.x&quot;, &quot;Locus&quot;), convert=T, sep=&quot;_&quot;)
  ) %&gt;%
  filter(FinemapPP &gt;0.25) %&gt;%
  mutate(AnnotationSuperclass = case_when(
    str_detect(AnnotationClass, &quot;Splice.+_1&quot;) ~ &quot;Annotated SS&quot;,
    str_detect(AnnotationClass, &quot;Splice.+_0$&quot;) ~ &quot;Unannotated SS&quot;,
    str_detect(AnnotationClass, &quot;Enhancer&quot;) ~ &quot;Enhancer&quot;,
    str_detect(AnnotationClass, &quot;Promoter&quot;) ~ &quot;Promoter&quot;,
    TRUE ~ &quot;Other&quot;
  )) %&gt;%
  mutate(PC.x = recode(PC.x, !!!PC.ShortAliases)) %&gt;%
  mutate(PC.y = recode(PC.y, !!!PC.ShortAliases)) %&gt;%
  mutate(Contains.eQTL_Contains.sQTL = recode(Contains.eQTL_Contains.sQTL,
                                              !!!c(&quot;TRUE TRUE&quot;=&quot;hQTL+sQTL&quot;,
                                                   &quot;TRUE FALSE&quot;=&quot;hQTL only&quot;,
                                                   &quot;FALSE TRUE&quot;=&quot;sQTL only&quot;))) %&gt;%
  filter(Contains.eQTL_Contains.sQTL==&quot;sQTL only&quot; &amp; AnnotationSuperclass==&quot;Annotated SS&quot;) %&gt;%
  ggplot(aes(x=beta.x, y=beta.y)) +
  geom_point() +
  geom_vline(xintercept=0) +
  geom_hline(yintercept=0) +
  facet_grid(rows=vars(PC.x), cols=vars(PC.y)) +
  theme_bw() +
  labs(x=&quot;TraitA Phenotype class&quot;, y=&quot;TraitB Phenotype class&quot;, title=&quot;Effect size correlation for eQTLs/sQTLs that are not hQTLs&quot;,
       caption = str_wrap(&quot;Only SNPs with finemap PP &gt; 0.25 in annotated splice site SNP included&quot;))</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/8df7e43cc493d35d443d132d58c6f327ee406d9f/docs/figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-23-1.png" target="_blank">8df7e43</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="write-out-bed-files-for-meta-qtl-plot" class="section level2">
<h2>Write out bed files for meta QTL plot</h2>
<p>As a side thing, I wrote a script to make metaQTL plots. That is, I lump the high/high allele genotypes for many QTLs and plot a metagene plot, along with the heterzygotes, and the low/low genotypes. This could be useful as an alternative to looking through a bunch of pygenometracks QTL plots. I think this list of eQTLs that coloc with hQTL only versus those that coloc with sQTL only would be a good test plot to look at. For example, I expect to see promoter effects at the hQTL eGenes but not so much at the sQTL eGenes, and definitely not at the sQTL eGenes with high finemapping probability as splice sites</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] qvalue_2.16.0     data.table_1.14.2 forcats_0.4.0     stringr_1.4.0    
 [5] dplyr_1.0.9       purrr_0.3.4       readr_1.3.1       tidyr_1.2.0      
 [9] tibble_3.1.7      ggplot2_3.3.6     tidyverse_1.3.0  

loaded via a namespace (and not attached):
 [1] httr_1.4.4        jsonlite_1.6      splines_3.6.1     R.utils_2.9.0    
 [5] modelr_0.1.8      assertthat_0.2.1  highr_0.9         cellranger_1.1.0 
 [9] yaml_2.2.0        pillar_1.7.0      backports_1.4.1   lattice_0.20-38  
[13] glue_1.6.2        digest_0.6.20     promises_1.0.1    rvest_0.3.5      
[17] colorspace_1.4-1  htmltools_0.5.3   httpuv_1.5.1      Matrix_1.2-18    
[21] R.oo_1.22.0       plyr_1.8.4        pkgconfig_2.0.2   broom_1.0.0      
[25] haven_2.3.1       scales_1.1.0      whisker_0.3-2     later_0.8.0      
[29] git2r_0.26.1      mgcv_1.8-40       generics_0.1.3    farver_2.1.0     
[33] ellipsis_0.3.2    withr_2.5.0       cli_3.3.0         magrittr_1.5     
[37] crayon_1.3.4      readxl_1.3.1      evaluate_0.15     R.methodsS3_1.7.1
[41] fs_1.5.2          fansi_0.4.0       nlme_3.1-140      xml2_1.3.2       
[45] tools_3.6.1       hms_0.5.3         lifecycle_1.0.1   munsell_0.5.0    
[49] reprex_0.3.0      compiler_3.6.1    rlang_1.0.5       grid_3.6.1       
[53] rstudioapi_0.14   labeling_0.3      rmarkdown_1.13    gtable_0.3.0     
[57] DBI_1.1.0         reshape2_1.4.3    R6_2.4.0          lubridate_1.7.4  
[61] knitr_1.39        fastmap_1.1.0     utf8_1.1.4        workflowr_1.6.2  
[65] rprojroot_2.0.2   stringi_1.4.3     Rcpp_1.0.5        vctrs_0.4.1      
[69] dbplyr_1.4.2      tidyselect_1.1.2  xfun_0.31        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
