<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Inspect unstandardized betas</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Inspect unstandardized betas</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-10-24
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>ChromatinSplicingQTLs/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstreef864f85adb4980fda6d516c6160926f726212539targetblankf864f85a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/f864f85adb4980fda6d516c6160926f726212539" target="_blank">f864f85</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstreef864f85adb4980fda6d516c6160926f726212539targetblankf864f85a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/f864f85adb4980fda6d516c6160926f726212539" target="_blank">f864f85</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/figure/
    Ignored:    code/.DS_Store
    Ignored:    code/.RData
    Ignored:    code/._.DS_Store
    Ignored:    code/._README.md
    Ignored:    code/._report.html
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    code/.snakemake/
    Ignored:    code/APA_Processing/
    Ignored:    code/Alignments/
    Ignored:    code/ChromHMM/
    Ignored:    code/ENCODE/
    Ignored:    code/ExpressionAnalysis/
    Ignored:    code/FastqFastp/
    Ignored:    code/FastqFastpSE/
    Ignored:    code/Genotypes/
    Ignored:    code/H3K36me3_CutAndTag.pdf
    Ignored:    code/IntronSlopes/
    Ignored:    code/Metaplots/
    Ignored:    code/Misc/
    Ignored:    code/MiscCountTables/
    Ignored:    code/Multiqc/
    Ignored:    code/Multiqc_chRNA/
    Ignored:    code/NonCodingRNA/
    Ignored:    code/NonCodingRNA_annotation/
    Ignored:    code/PeakCalling/
    Ignored:    code/Phenotypes/
    Ignored:    code/PlotGruberQTLs/
    Ignored:    code/PlotQTLs/
    Ignored:    code/ProCapAnalysis/
    Ignored:    code/QC/
    Ignored:    code/QTL_SNP_Enrichment/
    Ignored:    code/QTLs/
    Ignored:    code/RPKM_tables/
    Ignored:    code/ReferenceGenome/
    Ignored:    code/Rplots.pdf
    Ignored:    code/Session.vim
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TODO
    Ignored:    code/Tehranchi/
    Ignored:    code/bigwigs/
    Ignored:    code/bigwigs_FromNonWASPFilteredReads/
    Ignored:    code/config/.DS_Store
    Ignored:    code/config/._.DS_Store
    Ignored:    code/config/.ipynb_checkpoints/
    Ignored:    code/dag.pdf
    Ignored:    code/dag.png
    Ignored:    code/dag.svg
    Ignored:    code/debug.ipynb
    Ignored:    code/debug_python.ipynb
    Ignored:    code/deepTools/
    Ignored:    code/featureCounts/
    Ignored:    code/gwas_summary_stats/
    Ignored:    code/hyprcoloc/
    Ignored:    code/igv_session.xml
    Ignored:    code/log
    Ignored:    code/logs/
    Ignored:    code/notebooks/.ipynb_checkpoints/
    Ignored:    code/pi1/
    Ignored:    code/rules/.ipynb_checkpoints/
    Ignored:    code/rules/OldRules/
    Ignored:    code/rules/notebooks/
    Ignored:    code/scratch/
    Ignored:    code/scripts/.CalculatePi1_GetTraitPairs_AllTraits.R.swp
    Ignored:    code/scripts/.ipynb_checkpoints/
    Ignored:    code/scripts/GTFtools_0.8.0/
    Ignored:    code/scripts/__pycache__/
    Ignored:    code/scripts/liftOverBedpe/liftOverBedpe.py
    Ignored:    code/snakemake.dryrun.log
    Ignored:    code/snakemake.log
    Ignored:    code/snakemake.sbatch.log
    Ignored:    code/test.introns.bed
    Ignored:    code/test.introns2.bed
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/._20220414203249_JASPAR2022_combined_matrices_25818_jaspar.txt
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-10.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-11.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-2.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-3.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-4.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-5.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-6.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-7.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-8.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022.csv

Untracked files:
    Untracked:  analysis/20221024_InspectUnstandardizedBetas.Rmd
    Untracked:  code/snakemake_profiles/slurm/__pycache__/

Unstaged changes:
    Modified:   analysis/20220930_CheckSNP_QTL_Positions.Rmd
    Modified:   analysis/20221012_IntronRetentionAndExpressionConcordance.Rmd
    Modified:   code/rules/CreateUnstandardizedPhenotypeMatrices.smk
    Modified:   code/rules/QTLTools.smk
    Modified:   code/scripts/GenometracksByGenotype

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="intro" class="section level2">
<h2>Intro</h2>
<p>I mapped QTLs using phenotype table that was standardized, so the beta estimates are in standardized units which are less interpretable. I then used QTLtools again to get beta estimate where the input phenotype table was more interpretable units (ie, leafcutter intronic-psi for splicing, or log2CPM for expression), such that the betas will be more interpretable (ie, deltapsi for splicing, log2FoldChange for expression). Let’s explore that output a bit.</p>
<pre class="r"><code>library(tidyverse)
library(data.table)
library(GGally)

PhenotypeAliases &lt;- read_tsv(&quot;../data/Phenotypes_recode_for_Plotting.txt&quot;)

PermutationPass.dat &lt;- Sys.glob(&quot;../code/QTLs/QTLTools/*/PermutationPass.FDR_Added.txt.gz&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/QTLs/QTLTools/(.+?)/PermutationPass.FDR_Added.txt.gz&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(read_delim, delim=&#39; &#39;) %&gt;%
  bind_rows(.id=&quot;PhenotypeClass&quot;)</code></pre>
<p>First explore the data a bit… count QTLs for example…</p>
<pre class="r"><code>NumTestFeats &lt;- PermutationPass.dat %&gt;%
  count(PhenotypeClass)
NumQTLs &lt;- PermutationPass.dat %&gt;%
  filter(q&lt;0.1) %&gt;%
  count(PhenotypeClass)

ggplot(data = NumTestFeats,
         aes(x=PhenotypeClass, y=n)) +
  geom_col() +
  geom_text(aes(label=n), color=&quot;black&quot;, angle=70, hjust=-0.4, size=2) +
  geom_errorbar(
    data = NumQTLs,
    aes(y = n, ymin = n, ymax = n), color=&quot;black&quot;,lty=1, size=1.5) +
  geom_text(
    data = NumQTLs,
    aes(y=n, label=n), color=&quot;red&quot;, angle=90, hjust=-0.4, size=2) +
  scale_y_continuous(trans=&#39;log10&#39;, limits=c(1,1E6)) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  labs(y=&quot;Number features/QTLs&quot;, caption=&quot;Number of test features (black) and QTLs (red); FDR&lt;10%&quot;)</code></pre>
<p><img src="figure/20221024_InspectUnstandardizedBetas.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="inspect-in-unstandardized-qtltools-output" class="section level3">
<h3>Inspect in unstandardized QTLtools output</h3>
<pre class="r"><code>unstandardized.dat &lt;- Sys.glob(&quot;../code/QTLs/QTLTools/*/NominalPass_Unstandardized.txt.gz&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/QTLs/QTLTools/(.+?)/NominalPass_Unstandardized.txt.gz&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(read_delim, delim=&#39; &#39;, col_names=c(colnames(PermutationPass.dat)[c(2:12, 17:20)], &quot;best_hit&quot;)) %&gt;%
  bind_rows(.id=&quot;PhenotypeClass&quot;)



PermutationPass.dat.standardized &lt;- PermutationPass.dat %&gt;%
  inner_join(
    unstandardized.dat %&gt;%
      dplyr::select(PhenotypeClass, phe_id, var_id, slope),
    by=c(&quot;PhenotypeClass&quot;, &quot;phe_id&quot;, &quot;var_id&quot;),
    suffix=c(&quot;.standardized&quot;, &quot;.unstandardized&quot;)
  )


PermutationPass.dat.standardized %&gt;%
  filter(q&lt;0.1) %&gt;%
  ggplot(aes(x=slope.unstandardized)) +
  geom_histogram() +
  facet_wrap(~PhenotypeClass, scales=&quot;free&quot;) +
  theme_bw() +
  labs(title=&quot;Histogram of unstandardized effect sizes for QTLs&quot;)</code></pre>
<p><img src="figure/20221024_InspectUnstandardizedBetas.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>PermutationPass.dat.standardized %&gt;%
  filter(q&lt;0.1) %&gt;%
  ggplot(aes(x=slope.standardized, y=slope.unstandardized)) +
  geom_point(alpha=0.05) +
  # geom_hex(bins=50) +
  scale_fill_viridis_c() +
  facet_wrap(~PhenotypeClass, scales=&quot;free&quot;) +
  theme_bw() +
  labs(title=&quot;Unstandardized vs standardized effect sizes&quot;)</code></pre>
<p><img src="figure/20221024_InspectUnstandardizedBetas.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, these unstandardized effect sizes make sense. The sign is mostly always consistent with the sign of the standardized beta estimate, and the magnitude of things generally make sense… For example the beta for splicing (which is in units of delta psi for each dose of the alt allele) is generally capped at about 50%. The effect sizes for expression and chromatin QTLs are usually within abs(log2FC)&lt;1, so less than two fold change, though there are plenty of examples with larger effects. This matches my intuitions based on plotting a bunch of QTLs.</p>
</div>
</div>
<div id="recheck-splicingexpression-concordance" class="section level2">
<h2>Recheck splicing/expression concordance</h2>
<p>Remake plots from <a href="https://bfairkun.github.io/ChromatinSplicingQTLs/20221012_IntronRetentionAndExpressionConcordance.html">this notebook</a> with more interpretable effect sizes (eg delta-psi, and log2FC for expression) to explore the relationship between activation of NMD-inducing or crpytic spice sites and expression changes.</p>
<p>First, read in data…</p>
<pre class="r"><code>TopSNPEffects.ByPairs &lt;- fread(&quot;../code/pi1/PairwisePi1Traits.P.all.txt.gz&quot;)

NMD.transcript.introns &lt;- read_tsv(&quot;../code/SplicingAnalysis/Annotations/NMD/NMD_trancsript_introns.bed.gz&quot;, col_names=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;)) %&gt;%
  mutate(stop=stop+1) %&gt;%
  unite(intron, chrom:stop, strand)

Non.NMD.transcript.introns &lt;- read_tsv(&quot;../code/SplicingAnalysis/Annotations/NMD/NonNMD_trancsript_introns.bed.gz&quot;, col_names=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;)) %&gt;%
  mutate(stop=stop+1) %&gt;%
  unite(intron, chrom:stop, strand)

NMD.specific.introns &lt;- setdiff(NMD.transcript.introns$intron, Non.NMD.transcript.introns$intron)

Intron.Annotations.basic &lt;- read_tsv(&quot;../code/SplicingAnalysis/regtools_annotate_combined/basic.bed.gz&quot;) %&gt;%
  filter(known_junction ==1) %&gt;%
  unite(intron, chrom, start, end, strand)
Introns.Annotations.comprehensive &lt;- read_tsv(&quot;../code/SplicingAnalysis/regtools_annotate_combined/comprehensive.bed.gz&quot;) %&gt;%
  filter(known_junction ==1) %&gt;%
  unite(intron, chrom, start, end, strand)</code></pre>
<p>Now reproduce previous plot</p>
<pre class="r"><code>TopSNPEffects.ByPairs %&gt;%
  filter(
      (PC1 %in% c(&quot;chRNA.Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;)) &amp;
      (PC2 %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;)) &amp;
      (FDR.x &lt; 0.1) &amp;
      (FDR.y &lt; 0.1)) %&gt;%
  mutate(source.eQTL = recode(PC1, !!!c(&quot;Expression.Splicing.Subset_YRI&quot;=&quot;polyA.Expression.YRI&quot;, &quot;chRNA.Expression.Splicing&quot;=&quot;chRNA.Expression&quot;))) %&gt;%
  rename(source.sQTL=PC2) %&gt;%
  mutate(intron = str_replace(P2, &quot;^(.+):(.+?):(.+?):clu.+?_([+-])$&quot;, &quot;chr\\1_\\2_\\3_\\4&quot;)) %&gt;%
  mutate(IntronAnnotation = case_when(
    intron %in% NMD.specific.introns ~ &quot;Annotated NMD&quot;,
    intron %in% Intron.Annotations.basic$intron ~ &quot;Annotated basic&quot;,
    intron %in% Introns.Annotations.comprehensive$intron ~ &quot;Annotated Not basic&quot;,
    TRUE ~ &quot;Unannotated&quot;
  )) %&gt;%
  ggplot(aes(y=beta.x, x=trait.x.beta.in.y
, color=IntronAnnotation)) +
  geom_point(alpha=0.5) +
  geom_vline(xintercept=0) +
  geom_hline(yintercept=0) +
  geom_smooth(method=&#39;lm&#39;) +
  facet_grid(rows = vars(source.eQTL), cols=vars(source.sQTL)) +
  theme_bw() +
  labs(title=&quot;genetic effects of NMD introns and host gene expression&quot;, x=&quot;beta, sQTL&quot;, y=&quot;beta, eQTL&quot;, caption=&quot;Only FDR&lt;10% sQTLs and eQTLs plotted. sQTL effects are relative to top eQTL SNP&quot;)</code></pre>
<p><img src="figure/20221024_InspectUnstandardizedBetas.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, now plot similar plot, but make use unstandardized betas. Note that the previous plot was using slightly different cis-windows for QTL-mapping that the unstandardized beta estimates I have on hand, so some points might dissapear if the top QTL (in the gene-wise “ForColoc”&quot; style cis-QTL mapping) is outside the cis-window I used in the standard QTL mapping.</p>
<pre class="r"><code>unstandardized.dat %&gt;% distinct(PhenotypeClass)</code></pre>
<pre><code># A tibble: 15 × 1
   PhenotypeClass                  
   &lt;chr&gt;                           
 1 CTCF                            
 2 Expression.Splicing.Subset_YRI  
 3 Expression.Splicing             
 4 H3K27AC                         
 5 H3K36ME3                        
 6 H3K4ME1                         
 7 H3K4ME3                         
 8 MetabolicLabelled.30min.Splicing
 9 MetabolicLabelled.30min         
10 MetabolicLabelled.60min.Splicing
11 MetabolicLabelled.60min         
12 chRNA.Expression.Splicing       
13 chRNA.Splicing                  
14 polyA.Splicing.Subset_YRI       
15 polyA.Splicing                  </code></pre>
<pre class="r"><code>P &lt;- TopSNPEffects.ByPairs %&gt;%
  filter(
      (PC1 %in% c(&quot;chRNA.Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;)) &amp;
      (PC2 %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;)) &amp;
      (FDR.x &lt; 0.1) &amp;
      (FDR.y &lt; 0.1)) %&gt;%
  inner_join(
    unstandardized.dat %&gt;%
      dplyr::select(PC1=PhenotypeClass, P1=phe_id, singletrait_topvar.x=var_id, beta.x.unstandardized=slope)
    ) %&gt;%
  mutate(source.eQTL = recode(PC1, !!!c(&quot;Expression.Splicing.Subset_YRI&quot;=&quot;polyA.Expression.YRI&quot;, &quot;chRNA.Expression.Splicing&quot;=&quot;chRNA.Expression&quot;))) %&gt;%
rename(source.sQTL=PC2) %&gt;%
  mutate(intron = str_replace(P2, &quot;^(.+):(.+?):(.+?):clu.+?_([+-])$&quot;, &quot;chr\\1_\\2_\\3_\\4&quot;)) %&gt;%
  mutate(IntronAnnotation = case_when(
    intron %in% NMD.specific.introns ~ &quot;Annotated NMD&quot;,
    intron %in% Intron.Annotations.basic$intron ~ &quot;Annotated basic&quot;,
    intron %in% Introns.Annotations.comprehensive$intron ~ &quot;Annotated Not basic&quot;,
    TRUE ~ &quot;Unannotated&quot;
  )) %&gt;%
  ggplot(aes(y=beta.x.unstandardized, x=trait.x.beta.in.y
, color=IntronAnnotation)) +
  geom_point(alpha=0.5) +
  geom_vline(xintercept=0) +
  geom_hline(yintercept=0) +
  geom_smooth(method=&#39;lm&#39;) +
  facet_grid(rows = vars(source.eQTL), cols=vars(source.sQTL)) +
  theme_bw() +
  labs(title=&quot;genetic effects of NMD introns and host gene expression&quot;, x=&quot;standardized beta, sQTL&quot;, y=&quot;unstandardized beta, eQTL (Log2FC)&quot;, caption=&quot;Only FDR&lt;10% sQTLs and eQTLs plotted. sQTL effects are relative to top eQTL SNP&quot;)

P</code></pre>
<p><img src="figure/20221024_InspectUnstandardizedBetas.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>P +
  coord_cartesian(ylim=c(-1,1))</code></pre>
<p><img src="figure/20221024_InspectUnstandardizedBetas.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, I now realize I can’t really do this plot correctly (with both the x-axis and y-axis in unstandardized units) without doing some editing to the snakemake pipeline to get nominal pass QTL results in the genewise way…</p>
<p>For now, I can only do one axis unstandardized… Let’s ‘unstandardize’ the sQTLs. It’s also clear from this plot that the purple (unannotated) regression lines are driven by a few outlier points with many introns mapping to the same gene. Let’s replot with one point per gene or one point per intron cluster…</p>
<pre class="r"><code>P2.dat &lt;- TopSNPEffects.ByPairs %&gt;%
  filter(
      (PC2 %in% c(&quot;chRNA.Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;)) &amp;
      (PC1 %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;)) &amp;
      (FDR.x &lt; 0.1) &amp;
      (FDR.y &lt; 0.1)) %&gt;%
  inner_join(
    unstandardized.dat %&gt;%
      dplyr::select(PC1=PhenotypeClass, P1=phe_id, singletrait_topvar.x=var_id, beta.x.unstandardized=slope)
    ) %&gt;%
  mutate(source.eQTL = recode(PC2, !!!c(&quot;Expression.Splicing.Subset_YRI&quot;=&quot;polyA.Expression.YRI&quot;, &quot;chRNA.Expression.Splicing&quot;=&quot;chRNA.Expression&quot;))) %&gt;%
  rename(source.sQTL=PC1) %&gt;%
  mutate(intron = str_replace(P1, &quot;^(.+):(.+?):(.+?):clu.+?_([+-])$&quot;, &quot;chr\\1_\\2_\\3_\\4&quot;)) %&gt;%
  mutate(IntronAnnotation = case_when(
    intron %in% NMD.specific.introns ~ &quot;Annotated NMD&quot;,
    intron %in% Intron.Annotations.basic$intron ~ &quot;Annotated basic&quot;,
    intron %in% Introns.Annotations.comprehensive$intron ~ &quot;Annotated Not basic&quot;,
    TRUE ~ &quot;Unannotated&quot;
  ))

ggplot(P2.dat, aes(x=beta.x.unstandardized, y=trait.x.beta.in.y
, color=IntronAnnotation)) +
  geom_point(alpha=0.5) +
  geom_vline(xintercept=0) +
  geom_hline(yintercept=0) +
  geom_smooth(method=&#39;lm&#39;) +
  facet_grid(rows = vars(source.eQTL), cols=vars(source.sQTL)) +
  theme_bw() +
  labs(title=&quot;genetic effects of NMD introns and host gene expression&quot;, x=&quot;unstandardized beta, sQTL; deltapsi&quot;, y=&quot;standardized beta, eQTL&quot;, caption=&quot;Only FDR&lt;10% sQTLs and eQTLs plotted. eQTL effects are relative to top sQTL SNP&quot;)</code></pre>
<p><img src="figure/20221024_InspectUnstandardizedBetas.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Now let’s consider aggregating the intron QTL points by cluster or by gene</p>
<pre class="r"><code>P2.dat %&gt;%
  mutate(cluster = str_extract(P1, &quot;clu.+$&quot;)) %&gt;%
  distinct(GeneLocus, cluster) %&gt;%
  count(cluster) %&gt;%
  count(n)</code></pre>
<pre><code>   n  nn
1: 1 679
2: 2  21</code></pre>
<p>If we aggregated by gene, 21/700 of the genes will have multiple clusters. I think that’s acceptable. Let’s just aggregate by gene for simplicity…</p>
<pre class="r"><code>P2.dat %&gt;%
  group_by(IntronAnnotation, source.sQTL, source.eQTL, trait.x.beta.in.y) %&gt;%
  summarise(aggregated_deltapsi=sum(beta.x.unstandardized
)) %&gt;%
  ungroup() %&gt;%
  ggplot(aes(x=aggregated_deltapsi, y=trait.x.beta.in.y
, color=IntronAnnotation)) +
  geom_point(alpha=0.5) +
  geom_vline(xintercept=0) +
  geom_hline(yintercept=0) +
  geom_smooth(method=&#39;lm&#39;) +
  facet_grid(rows = vars(source.eQTL), cols=vars(source.sQTL)) +
  theme_bw() +
  labs(title=&quot;genetic effects of NMD introns and host gene expression&quot;, x=&quot;unstandardized beta, sQTL; deltapsi&quot;, y=&quot;standardized beta, eQTL&quot;, caption=&quot;Only FDR&lt;10% sQTLs and eQTLs plotted. eQTL effects are relative to top sQTL SNP&quot;)</code></pre>
<p><img src="figure/20221024_InspectUnstandardizedBetas.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>There’s so many overlapping points. Let me seperate the colors…</p>
<pre class="r"><code>P2.dat %&gt;%
  group_by(IntronAnnotation, P2, source.sQTL, source.eQTL, trait.x.beta.in.y) %&gt;%
  summarise(aggregated_deltapsi=sum(beta.x.unstandardized
)) %&gt;%
  ungroup() %&gt;%
  ggplot(aes(x=aggregated_deltapsi, y=trait.x.beta.in.y
, color=IntronAnnotation)) +
  geom_point(alpha=0.5) +
  geom_vline(xintercept=0) +
  geom_hline(yintercept=0) +
  geom_smooth(method=&#39;lm&#39;) +
  facet_grid(source.eQTL ~ source.sQTL ~ IntronAnnotation ) +
  theme_bw() +
  labs(title=&quot;genetic effects of sQTLs and host gene expression&quot;, x=&quot;unstandardized beta, sQTL; deltapsi&quot;, y=&quot;standardized beta, eQTL&quot;, caption=&quot;Only FDR&lt;10% sQTLs and eQTLs plotted. eQTL effects are relative to top sQTL SNP&quot;)</code></pre>
<p><img src="figure/20221024_InspectUnstandardizedBetas.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /> Hmm, from this where there aren’t double plotting lot’s of points because i aggregated the intron effects by gene, there isn’t an obvious correlation between increase in unannotated splicing resulting in more expression.</p>
<p>Let’s also try expressing these correlations just by the overabundance of concordant/discordanct signs between the sQTL and eQTL after this gene-wise aggregation for sQTLs</p>
<pre class="r"><code>P2.dat %&gt;%
  group_by(IntronAnnotation, P2, source.sQTL, source.eQTL, trait.x.beta.in.y) %&gt;%
  summarise(aggregated_deltapsi=sum(beta.x.unstandardized
)) %&gt;%
  ungroup() %&gt;%
  mutate(Concordance = if_else(sign(trait.x.beta.in.y) == sign(aggregated_deltapsi), &quot;Concordant&quot;, &quot;Discordant&quot;)) %&gt;%
  count(IntronAnnotation, source.sQTL, source.eQTL, Concordance) %&gt;%
  ggplot(aes(x=IntronAnnotation, y=n, fill=IntronAnnotation, color=Concordance)) +
  geom_col() +
  facet_grid(source.eQTL ~ source.sQTL) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))</code></pre>
<p><img src="figure/20221024_InspectUnstandardizedBetas.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Now let’s come up with a lower limit for how many eQTLs might be explained by these splicing effects. To do this I will count the overabundance of discordant sign eQTL/sQTLs, and ask what fraction those are out of all eQTLs.</p>
<p>Said another way, I will consider an sQTLs mapped in either chRNA or polyA, and count how many have sQTL signs consistent with my expectations on how it would effect expression. For this analysis, it might be simplest to first just combine the “Annotated NMD”, “Annotated Not basic”, and “Unannotated” categories by summing their deltapsi… Before I do that, let’s look at the deltapsi distributions…</p>
<pre class="r"><code>P2.dat %&gt;%
  group_by(IntronAnnotation, P2, source.sQTL, source.eQTL, trait.x.beta.in.y) %&gt;%
  summarise(aggregated_deltapsi=sum(beta.x.unstandardized
)) %&gt;%
  ggplot(aes(x=aggregated_deltapsi, color=IntronAnnotation)) +
  stat_ecdf() +
  facet_grid(source.eQTL ~ source.sQTL) +
  theme_bw()</code></pre>
<p><img src="figure/20221024_InspectUnstandardizedBetas.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>P2.dat %&gt;%
  mutate(IntronAnnotation = if_else(IntronAnnotation==&quot;Annotated basic&quot;, &quot;Annotated basic&quot;, &quot;Not annotated basic&quot;)) %&gt;%
  group_by(IntronAnnotation, P2, source.sQTL, source.eQTL, trait.x.beta.in.y) %&gt;%
  summarise(aggregated_deltapsi=sum(beta.x.unstandardized
)) %&gt;%
  ungroup() %&gt;%
  mutate(Concordance = if_else(sign(trait.x.beta.in.y) == sign(aggregated_deltapsi), &quot;Concordant&quot;, &quot;Discordant&quot;)) %&gt;%
  count(IntronAnnotation, source.sQTL, source.eQTL, Concordance) %&gt;%
  ggplot(aes(x=IntronAnnotation, y=n, fill=IntronAnnotation, color=Concordance)) +
  geom_col() +
  facet_grid(source.eQTL ~ source.sQTL) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))</code></pre>
<p><img src="figure/20221024_InspectUnstandardizedBetas.Rmd/unnamed-chunk-12-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Actually, below is perhaps a more intuitive graph of what I mean:</p>
<pre class="r"><code>Number_eQTLs_sQTLs_consistentWithNMD &lt;- P2.dat %&gt;%
  filter(source.eQTL == &quot;polyA.Expression.YRI&quot;) %&gt;%
  mutate(IntronAnnotation = if_else(IntronAnnotation==&quot;Annotated basic&quot;, &quot;Annotated basic&quot;, &quot;Not annotated basic&quot;)) %&gt;%
  group_by(IntronAnnotation, P2, source.sQTL, source.eQTL, trait.x.beta.in.y) %&gt;%
  summarise(aggregated_deltapsi=sum(beta.x.unstandardized
)) %&gt;%
  ungroup() %&gt;%
  mutate(Concordance = if_else(sign(trait.x.beta.in.y) == sign(aggregated_deltapsi), &quot;Concordant&quot;, &quot;Discordant&quot;)) %&gt;%
  mutate(ConcordanceConsistentWithNMD = case_when(
    IntronAnnotation == &quot;Annotated basic&quot; &amp; Concordance==&quot;Concordant&quot; ~ &quot;Consistent&quot;,
    IntronAnnotation == &quot;Not annotated basic&quot; &amp; Concordance==&quot;Discordant&quot; ~ &quot;Consistent&quot;,
    TRUE ~ &quot;Inconsistent&quot;
  )) %&gt;%
  distinct(P2, .keep_all=T) %&gt;%
  count(ConcordanceConsistentWithNMD)

Number_eQTLs_sQTLs_consistentWithNMD$n[1] - Number_eQTLs_sQTLs_consistentWithNMD$n[2]</code></pre>
<pre><code>[1] 73</code></pre>
<p>So perhaps 73 QTLs might be due to NMD mechanism… That is out of how many eQTLs…</p>
<pre class="r"><code>TopSNPEffects.ByPairs %&gt;%
  filter(
      (PC1 %in% c(&quot;Expression.Splicing.Subset_YRI&quot;)) &amp;
      (FDR.x &lt; 0.1)) %&gt;%
  distinct(P1) %&gt;% nrow()</code></pre>
<pre><code>[1] 4021</code></pre>
<pre class="r"><code>LowerBound &lt;- 73/4021 * 100
LowerBound</code></pre>
<pre><code>[1] 1.815469</code></pre>
<p>Ok, so not many. perhaps 2% of eQTLs might be explained by that mechanism. Note that here I filtered for sQTLs &lt; 10% FDR. Perhaps that strictness is part of why it is important to view this as a conservative estimate or lower bound.. Right now I’m now set up to be more lenient than that, I would have to reprocess some other datafiles…</p>
<p>Let’s also crudley verify that these QTLs are more aligned (the top eQTL and sQTL SNP is the same or close) in the NMD model cases, versus the other QTLs…</p>
<pre class="r"><code>DistanceDist.dat &lt;- P2.dat %&gt;%
  mutate(HighConfidenceNMD_eQTLs = if_else(
    source.eQTL == &quot;polyA.Expression.YRI&quot; &amp;
    source.sQTL %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;) &amp;
    IntronAnnotation == c(&quot;Annotated NMD&quot;) &amp;
    !sign(beta.x) == trait.x.beta.in.y,
    &quot;High confidence NMD sQTL --&gt; eQTL&quot;,
    &quot;Other&quot;
  )) %&gt;%
  mutate(Dist = abs(singletrait_topvar_pos.y-singletrait_topvar_pos.x)) %&gt;%
  mutate(Dist=pmax(1, Dist))

ggplot(DistanceDist.dat, aes(x=Dist)) +
  geom_histogram() +
  scale_x_continuous(trans=&#39;log10&#39;) +
  facet_wrap(~HighConfidenceNMD_eQTLs, scales=&quot;free_y&quot;) +
  theme_bw() +
  labs(title=&quot;Distance between top sQTL and top eQTL SNP&quot;, x=&quot;Dist b/n top eQTL and sQTL SNP&quot;, caption=str_wrap(&quot;High confidence NMD means polyA|chRNA sQTL and eQTL in polyA with opposite sign betas for annotated NMD introns&quot;, 30))</code></pre>
<p><img src="figure/20221024_InspectUnstandardizedBetas.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(DistanceDist.dat, aes(x=Dist, color=HighConfidenceNMD_eQTLs)) +
  stat_ecdf() +
  scale_x_continuous(trans=&#39;log10&#39;) +
  theme_bw() +
  labs(title=&quot;Distance between top sQTL and top eQTL SNP&quot;, x=&quot;Dist b/n top eQTL and sQTL SNP&quot;, caption=str_wrap(&quot;High confidence NMD means polyA|chRNA sQTL and eQTL in polyA with opposite sign betas for annotated NMD introns&quot;, 30))</code></pre>
<p><img src="figure/20221024_InspectUnstandardizedBetas.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Hmm… this makes me skeptical whether this distance between top SNPs is even a useful way to think about stuff. I don’t think there is a bug in my code, given how the beta signs make sense.</p>
<p>Anyway, what if we included the whole geuvadis dataset for splicing.. Just curious what that the results will like considering more sQTLs… Now go back to counting sQTLs/eQTLs consistent/inconsistent with NMD mechanism when we include all GEUVADIS samples.</p>
<pre class="r"><code>P2.dat &lt;- TopSNPEffects.ByPairs %&gt;%
  filter(
      (PC2 %in% c(&quot;chRNA.Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;)) &amp;
      (PC1 %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;polyA.Splicing&quot;)) &amp;
      (FDR.x &lt; 0.1) &amp;
      (FDR.y &lt; 0.1)) %&gt;%
  inner_join(
    unstandardized.dat %&gt;%
      dplyr::select(PC1=PhenotypeClass, P1=phe_id, singletrait_topvar.x=var_id, beta.x.unstandardized=slope)
    ) %&gt;%
  mutate(source.eQTL = recode(PC2, !!!c(&quot;Expression.Splicing.Subset_YRI&quot;=&quot;polyA.Expression.YRI&quot;, &quot;chRNA.Expression.Splicing&quot;=&quot;chRNA.Expression&quot;))) %&gt;%
  rename(source.sQTL=PC1) %&gt;%
  mutate(intron = str_replace(P1, &quot;^(.+):(.+?):(.+?):clu.+?_([+-])$&quot;, &quot;chr\\1_\\2_\\3_\\4&quot;)) %&gt;%
  mutate(IntronAnnotation = case_when(
    intron %in% NMD.specific.introns ~ &quot;Annotated NMD&quot;,
    intron %in% Intron.Annotations.basic$intron ~ &quot;Annotated basic&quot;,
    intron %in% Introns.Annotations.comprehensive$intron ~ &quot;Annotated Not basic&quot;,
    TRUE ~ &quot;Unannotated&quot;
  ))

P2.dat %&gt;%
  group_by(IntronAnnotation, P2, source.sQTL, source.eQTL, trait.x.beta.in.y) %&gt;%
  summarise(aggregated_deltapsi=sum(beta.x.unstandardized
)) %&gt;%
  ungroup() %&gt;%
  ggplot(aes(x=aggregated_deltapsi, y=trait.x.beta.in.y
, color=IntronAnnotation)) +
  geom_point(alpha=0.5) +
  geom_vline(xintercept=0) +
  geom_hline(yintercept=0) +
  geom_smooth(method=&#39;lm&#39;) +
  facet_grid(source.eQTL ~ source.sQTL ~ IntronAnnotation ) +
  theme_bw() +
  labs(title=&quot;genetic effects of sQTLs and host gene expression&quot;, x=&quot;unstandardized beta, sQTL; deltapsi&quot;, y=&quot;standardized beta, eQTL&quot;, caption=str_wrap(&quot;Only FDR&lt;10% sQTLs and eQTLs plotted. eQTL effects are relative to top sQTL SNP. Full geauvadis sQTLs are &#39;polyA.Splicing&#39;&quot;))</code></pre>
<p><img src="figure/20221024_InspectUnstandardizedBetas.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Number_eQTLs_sQTLs_consistentWithNMD &lt;- P2.dat %&gt;%
  filter(source.eQTL == &quot;polyA.Expression.YRI&quot;) %&gt;%
  mutate(IntronAnnotation = if_else(IntronAnnotation==&quot;Annotated basic&quot;, &quot;Annotated basic&quot;, &quot;Not annotated basic&quot;)) %&gt;%
  group_by(IntronAnnotation, P2, source.sQTL, source.eQTL, trait.x.beta.in.y) %&gt;%
  summarise(aggregated_deltapsi=sum(beta.x.unstandardized
)) %&gt;%
  ungroup() %&gt;%
  mutate(Concordance = if_else(sign(trait.x.beta.in.y) == sign(aggregated_deltapsi), &quot;Concordant&quot;, &quot;Discordant&quot;)) %&gt;%
  mutate(ConcordanceConsistentWithNMD = case_when(
    IntronAnnotation == &quot;Annotated basic&quot; &amp; Concordance==&quot;Concordant&quot; ~ &quot;Consistent&quot;,
    IntronAnnotation == &quot;Not annotated basic&quot; &amp; Concordance==&quot;Discordant&quot; ~ &quot;Consistent&quot;,
    TRUE ~ &quot;Inconsistent&quot;
  )) %&gt;%
  distinct(P2, .keep_all=T) %&gt;%
  count(ConcordanceConsistentWithNMD)

Number_eQTLs_sQTLs_consistentWithNMD</code></pre>
<pre><code># A tibble: 2 × 2
  ConcordanceConsistentWithNMD     n
  &lt;chr&gt;                        &lt;int&gt;
1 Consistent                     598
2 Inconsistent                   584</code></pre>
<p>Ok, so now when I consider all sQTLs mapped with the whole GEUVADIS the overabundance of sQTL/eQTLs signs consistent with NMD model is actually fewer. So maybe just adding power this way isn’t actually helpful.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] GGally_1.4.0      data.table_1.14.2 forcats_0.4.0     stringr_1.4.0    
 [5] dplyr_1.0.9       purrr_0.3.4       readr_1.3.1       tidyr_1.2.0      
 [9] tibble_3.1.7      ggplot2_3.3.6     tidyverse_1.3.0  

loaded via a namespace (and not attached):
 [1] httr_1.4.4         jsonlite_1.6       viridisLite_0.3.0  splines_3.6.1     
 [5] R.utils_2.9.0      modelr_0.1.8       assertthat_0.2.1   highr_0.9         
 [9] cellranger_1.1.0   yaml_2.2.0         pillar_1.7.0       backports_1.4.1   
[13] lattice_0.20-38    glue_1.6.2         digest_0.6.20      RColorBrewer_1.1-2
[17] promises_1.0.1     rvest_0.3.5        colorspace_1.4-1   htmltools_0.5.3   
[21] httpuv_1.5.1       Matrix_1.2-18      R.oo_1.22.0        plyr_1.8.4        
[25] pkgconfig_2.0.2    broom_1.0.0        haven_2.3.1        scales_1.1.0      
[29] later_0.8.0        git2r_0.26.1       mgcv_1.8-40        generics_0.1.3    
[33] farver_2.1.0       ellipsis_0.3.2     withr_2.5.0        cli_3.3.0         
[37] magrittr_1.5       crayon_1.3.4       readxl_1.3.1       evaluate_0.15     
[41] R.methodsS3_1.7.1  fs_1.5.2           fansi_0.4.0        nlme_3.1-140      
[45] xml2_1.3.2         tools_3.6.1        hms_0.5.3          lifecycle_1.0.1   
[49] munsell_0.5.0      reprex_0.3.0       compiler_3.6.1     rlang_1.0.5       
[53] grid_3.6.1         rstudioapi_0.14    labeling_0.3       rmarkdown_1.13    
[57] gtable_0.3.0       DBI_1.1.0          reshape_0.8.8      R6_2.4.0          
[61] lubridate_1.7.4    knitr_1.39         fastmap_1.1.0      utf8_1.1.4        
[65] workflowr_1.6.2    rprojroot_2.0.2    stringi_1.4.3      Rcpp_1.0.5        
[69] vctrs_0.4.1        dbplyr_1.4.2       tidyselect_1.1.2   xfun_0.31         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
