<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>CheckFinemapSNPEnrichments</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">CheckFinemapSNPEnrichments</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-10-07
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>ChromatinSplicingQTLs/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstreeedfdacc2897647bc1368ee2d56dc5b4c2d618df3targetblankedfdacca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/edfdacc2897647bc1368ee2d56dc5b4c2d618df3" target="_blank">edfdacc</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstreeedfdacc2897647bc1368ee2d56dc5b4c2d618df3targetblankedfdacca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/edfdacc2897647bc1368ee2d56dc5b4c2d618df3" target="_blank">edfdacc</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store
    Ignored:    code/.RData
    Ignored:    code/._.DS_Store
    Ignored:    code/._README.md
    Ignored:    code/._report.html
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    code/.snakemake/
    Ignored:    code/APA_Processing/
    Ignored:    code/Alignments/
    Ignored:    code/ChromHMM/
    Ignored:    code/ENCODE/
    Ignored:    code/ExpressionAnalysis/
    Ignored:    code/FastqFastp/
    Ignored:    code/FastqFastpSE/
    Ignored:    code/Genotypes/
    Ignored:    code/IntronSlopes/
    Ignored:    code/Misc/
    Ignored:    code/MiscCountTables/
    Ignored:    code/Multiqc/
    Ignored:    code/Multiqc_chRNA/
    Ignored:    code/NonCodingRNA_annotation/
    Ignored:    code/PeakCalling/
    Ignored:    code/Phenotypes/
    Ignored:    code/PlotGruberQTLs/
    Ignored:    code/PlotQTLs/
    Ignored:    code/ProCapAnalysis/
    Ignored:    code/QC/
    Ignored:    code/QTL_SNP_Enrichment/
    Ignored:    code/QTLs/
    Ignored:    code/ReferenceGenome/
    Ignored:    code/Rplots.pdf
    Ignored:    code/Session.vim
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TODO
    Ignored:    code/Tehranchi/
    Ignored:    code/bigwigs/
    Ignored:    code/bigwigs_FromNonWASPFilteredReads/
    Ignored:    code/config/.DS_Store
    Ignored:    code/config/._.DS_Store
    Ignored:    code/config/.ipynb_checkpoints/
    Ignored:    code/debug.ipynb
    Ignored:    code/debug_python.ipynb
    Ignored:    code/deepTools/
    Ignored:    code/featureCounts/
    Ignored:    code/gwas_summary_stats/
    Ignored:    code/hyprcoloc/
    Ignored:    code/igv_session.xml
    Ignored:    code/log
    Ignored:    code/logs/
    Ignored:    code/notebooks/.ipynb_checkpoints/
    Ignored:    code/rules/.QTL_SNP_Enrichment.smk.swp
    Ignored:    code/rules/.ipynb_checkpoints/
    Ignored:    code/rules/OldRules/
    Ignored:    code/rules/notebooks/
    Ignored:    code/scratch/
    Ignored:    code/scripts/.ipynb_checkpoints/
    Ignored:    code/scripts/GTFtools_0.8.0/
    Ignored:    code/scripts/__pycache__/
    Ignored:    code/scripts/liftOverBedpe/liftOverBedpe.py
    Ignored:    code/snakemake.log
    Ignored:    code/snakemake.sbatch.log
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/._20220414203249_JASPAR2022_combined_matrices_25818_jaspar.txt
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-10.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-11.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-2.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-3.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-4.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-5.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-6.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-7.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-8.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022.csv

Untracked files:
    Untracked:  code/scripts/tidyNoncodingRNA_annotations.R
    Untracked:  code/snakemake_profiles/slurm/__pycache__/

Unstaged changes:
    Modified:   analysis/20220728_CheckFinemapSNPEnrichments.Rmd
    Modified:   code/rules/QTLTools.smk
    Modified:   code/rules/QTL_SNP_Enrichment.smk
    Modified:   code/scripts/GenometracksByGenotype

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/20220728_CheckFinemapSNPEnrichments.Rmd</code>) and HTML (<code>docs/20220728_CheckFinemapSNPEnrichments.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/dbada73271ed520d2f114c8c149d5245b704bb52/analysis/20220728_CheckFinemapSNPEnrichments.Rmd" target="_blank">dbada73</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-09-02
</td>
<td>
added APA, removed redudant file
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/971cb7f2a63048d932929f70524003de0921b4e8/analysis/20220728_CheckFinemapSNPEnrichments.Rmd" target="_blank">971cb7f</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-08-22
</td>
<td>
update SNP enrichment analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/971cb7f2a63048d932929f70524003de0921b4e8/docs/20220728_CheckFinemapSNPEnrichments.html" target="_blank">971cb7f</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-08-22
</td>
<td>
update SNP enrichment analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/bfb625eb2719f3fde37af38bfa3f13d0ff2818cc/analysis/20220728_CheckFinemapSNPEnrichments.Rmd" target="_blank">bfb625e</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-08-09
</td>
<td>
added finemapping snp enrichment rules
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/bfb625eb2719f3fde37af38bfa3f13d0ff2818cc/docs/20220728_CheckFinemapSNPEnrichments.html" target="_blank">bfb625e</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-08-09
</td>
<td>
added finemapping snp enrichment rules
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="intro" class="section level2">
<h2>Intro</h2>
<p>One way to explore my QTL results is to look for enrichments in QTL SNPs. There are many ways to do this. I propose something simple: Start with the finemapping results from hyprcoloc (the posterior probability for each SNP for each hyprcoloc cluster) and ask to what degree that posterior probability mass is enriched in annotations, compared to some other set of SNPs (eg sQTLs versus eQTLs). The annotations could be anything, but here I am just going to explore this method and make some initial plots using GM12878 ChromHMM annotations from ENCODE (lifted over from hg19-&gt;hg38).</p>
<pre class="r"><code>library(tidyverse)
library(data.table)

# dat &lt;- fread(&quot;../code/QTL_SNP_Enrichment/FinemapIntersections.bed.gz&quot;, col.names=c(&quot;SNP_chrom&quot;, &quot;SNP_start&quot;, &quot;SNP_end&quot;, &quot;SNP&quot;, &quot;PosteriorPr&quot;, &quot;Annotation_chrom&quot;, &quot;Annotation_start&quot;, &quot;Annotation_stop&quot;, &quot;Annotation&quot;, &quot;Overlap&quot;)) %&gt;%
dat &lt;- fread(&quot;../code/QTL_SNP_Enrichment/FinemapIntersections/MolColocStandard.bed.gz&quot;, col.names=c(&quot;SNP_chrom&quot;, &quot;SNP_start&quot;, &quot;SNP_end&quot;, &quot;SNP&quot;, &quot;PosteriorPr&quot;, &quot;Annotation_chrom&quot;, &quot;Annotation_start&quot;, &quot;Annotation_stop&quot;, &quot;Annotation&quot;, &quot;Overlap&quot;)) %&gt;%

  separate(SNP, into=c(&quot;SNP&quot;, &quot;HyprcolocCluster&quot;, &quot;GeneLocus&quot;), sep=&quot;_&quot;)

hyprcoloc_results &lt;- read_tsv(&quot;../code/hyprcoloc/Results/ForColoc/MolColocStandard/tidy_results_OnlyColocalized.txt.gz&quot;)



hyprcoloc_results_finemap &lt;- read_tsv(&quot;../code/hyprcoloc/Results/ForColoc/MolColocStandard/snpscores.txt.gz&quot;)</code></pre>
<p>First I want to verify that I can match up the right TopSNPs by the clusterNumber. If I matching these TopSNPs correctly, the finemapping probability should be equal from both tables…</p>
<pre class="r"><code>hyprcoloc_results_finemap %&gt;%
  group_by(ColocalizedCluster, Locus) %&gt;%
  filter(FinemapPr == max(FinemapPr)) %&gt;%
  ungroup() %&gt;%
  full_join(
    hyprcoloc_results %&gt;% distinct(snp, iteration, Locus, .keep_all=T),
    by = c(&quot;Locus&quot;, &quot;snp&quot;, &quot;ColocalizedCluster&quot;=&quot;iteration&quot;)
  ) %&gt;%
  ggplot(aes(x=FinemapPr, y=TopSNPFinemapPr)) +
  geom_point()</code></pre>
<p><img src="figure/20220728_CheckFinemapSNPEnrichments.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/bfb625eb2719f3fde37af38bfa3f13d0ff2818cc/docs/figure/20220728_CheckFinemapSNPEnrichments.Rmd/unnamed-chunk-2-1.png" target="_blank">bfb625e</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-08-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>That looks right…</p>
<p>Now let’s pick clusters with only splicing phenotypes, and compare to clusters with an eQTL</p>
<pre class="r"><code>hyprcoloc_results %&gt;%
  separate(phenotype_full, into=c(&quot;PC&quot;, &quot;P&quot;), sep=&quot;;&quot;) %&gt;%
  pull(PC) %&gt;% unique()</code></pre>
<pre><code> [1] &quot;polyA.Splicing&quot;                   &quot;MetabolicLabelled.30min&quot;         
 [3] &quot;MetabolicLabelled.60min&quot;          &quot;H3K27AC&quot;                         
 [5] &quot;H3K4ME3&quot;                          &quot;H3K4ME1&quot;                         
 [7] &quot;chRNA.IER&quot;                        &quot;ProCap&quot;                          
 [9] &quot;polyA.Splicing.Subset_YRI&quot;        &quot;CTCF&quot;                            
[11] &quot;chRNA.Splicing.Order&quot;             &quot;chRNA.Expression_ncRNA&quot;          
[13] &quot;Expression.Splicing&quot;              &quot;Expression.Splicing.Subset_YRI&quot;  
[15] &quot;chRNA.Splicing&quot;                   &quot;MetabolicLabelled.30min.Splicing&quot;
[17] &quot;MetabolicLabelled.60min.Splicing&quot; &quot;chRNA.Expression.Splicing&quot;       
[19] &quot;H3K36ME3&quot;                         &quot;polyA.IER&quot;                       
[21] &quot;MetabolicLabelled.30min.IER&quot;      &quot;MetabolicLabelled.60min.IER&quot;     
[23] &quot;polyA.IER.Subset_YRI&quot;             &quot;chRNA.Slopes&quot;                    </code></pre>
<pre class="r"><code>SplicingClusters &lt;- hyprcoloc_results %&gt;%
  separate(phenotype_full, into=c(&quot;PC&quot;, &quot;P&quot;), sep=&quot;;&quot;) %&gt;%
  rowwise() %&gt;%
  mutate(PC_new = if_else(
    any(str_detect(PC, c(&quot;polyA.Splicing&quot;,&quot;chRNA.IR&quot;, &quot;polyA.IR&quot;, &quot;polyA.IR.Subset_YRI&quot;, &quot;chRNA.IER&quot;, &quot;chRNA.IRjunctions&quot;, &quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;))),
    &quot;Splicing&quot;,
    &quot;NotSplicing&quot;
  )) %&gt;%
  ungroup() %&gt;%
  group_by(Locus, iteration) %&gt;%
  filter(all(PC_new == &quot;Splicing&quot;)) %&gt;%
  ungroup() %&gt;%
  distinct(Locus,iteration) %&gt;%
  unite(Locus.iteration, Locus,iteration) %&gt;%
  mutate(ClusterGroup = &quot;SplicingClusters&quot;)



H3K27AC_AndEqtlClusters &lt;- hyprcoloc_results %&gt;%
  separate(phenotype_full, into=c(&quot;PC&quot;, &quot;P&quot;), sep=&quot;;&quot;) %&gt;%
  rowwise() %&gt;%
  mutate(PC_new = if_else(
    any(str_detect(PC, c(&quot;H3K27AC&quot;,&quot;H3K4ME1&quot;, &quot;H3K36ME3&quot;, &quot;H3K4ME3&quot;, &quot;Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;, &quot;chRNA.Expression.Splicing&quot;, &quot;MetabolicLabelled.30min&quot;, &quot;MetabolicLabelled.60min&quot;))),
    &quot;EnhancerOrExpression&quot;,
    &quot;NotEnhancerOrExpression&quot;
  )) %&gt;%
  ungroup() %&gt;%
  group_by(Locus, iteration) %&gt;%
  filter(all(PC_new == &quot;EnhancerOrExpression&quot;)) %&gt;%
  ungroup() %&gt;%
  distinct(Locus,iteration) %&gt;%
  unite(Locus.iteration, Locus,iteration) %&gt;%
  mutate(ClusterGroup = &quot;EnhancerOrExpression&quot;)

intersect(H3K27AC_AndEqtlClusters$Locus.iteration, SplicingClusters$Locus.iteration)</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>FractionPosteriorInEachAnnotation &lt;- bind_rows(H3K27AC_AndEqtlClusters, SplicingClusters) %&gt;%
  left_join(
    dat %&gt;%
    mutate(Locus.iteration = paste(GeneLocus, HyprcolocCluster, sep=&quot;_&quot;)),
    by = &quot;Locus.iteration&quot;
  ) %&gt;%
  group_by(Annotation, ClusterGroup) %&gt;%
  summarise(Sum = sum(PosteriorPr)) %&gt;%
  group_by(ClusterGroup) %&gt;%
  mutate(FractionAnnotatedPosteriorInGroup = Sum/sum(Sum, na.rm=T)) %&gt;%
  ungroup()

FractionPosteriorInEachAnnotation %&gt;%
  pivot_wider(names_from = &quot;ClusterGroup&quot;, values_from = c(&quot;FractionAnnotatedPosteriorInGroup&quot;, &quot;Sum&quot;), id_cols=&quot;Annotation&quot;) %&gt;%
  mutate(Enrichment = FractionAnnotatedPosteriorInGroup_SplicingClusters/FractionAnnotatedPosteriorInGroup_EnhancerOrExpression) %&gt;%
  mutate(Annotation = str_replace_all(Annotation, &quot;_0$&quot;, &quot;_UnannotatedButObserved&quot;)) %&gt;%
  mutate(Annotation = str_replace_all(Annotation, &quot;_1$&quot;, &quot;_Annotated&quot;)) %&gt;%
ggplot(aes(x=Annotation, y=Enrichment)) +
  geom_col() +
  theme_bw() +
  scale_y_continuous(trans=&#39;log2&#39;) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  coord_flip() +
  labs(x = &quot;Annotation&quot;, y=&quot;Finemapping posterior mass fold enrichment\n(ChromatinQTL|eQTL &lt;-- --&gt; sQTL)&quot;)</code></pre>
<p><img src="figure/20220728_CheckFinemapSNPEnrichments.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/bfb625eb2719f3fde37af38bfa3f13d0ff2818cc/docs/figure/20220728_CheckFinemapSNPEnrichments.Rmd/unnamed-chunk-3-1.png" target="_blank">bfb625e</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-08-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Let’s try to add error bars by bootstrapping. Perhaps randomly sample the QTL clusters. Let’s try 100 bootstrap replicates and make the plot twice to see how stable the confidence interval is.</p>
<pre class="r"><code>set.seed(0)

dat$Annotation %&gt;% unique()</code></pre>
<pre><code> [1] &quot;SpliceBranchpointRegion_0&quot; &quot;10_Txn_Elongation&quot;        
 [3] &quot;6_Weak_Enhancer&quot;           &quot;SpliceDonor_0&quot;            
 [5] &quot;2_Weak_Promoter&quot;           &quot;11_Weak_Txn&quot;              
 [7] &quot;1_Active_Promoter&quot;         &quot;12_Repressed&quot;             
 [9] &quot;4_Strong_Enhancer&quot;         &quot;13_Heterochrom/lo&quot;        
[11] &quot;SpliceAcceptor_0&quot;          &quot;SpliceBranchpointRegion_1&quot;
[13] &quot;ncRNA_coRNA&quot;               &quot;ncRNA_pseudo&quot;             
[15] &quot;9_Txn_Transition&quot;          &quot;SpliceAcceptor_1&quot;         
[17] &quot;ncRNA_srtRNA&quot;              &quot;8_Insulator&quot;              
[19] &quot;ncRNA_uaRNA&quot;               &quot;5_Strong_Enhancer&quot;        
[21] &quot;7_Weak_Enhancer&quot;           &quot;SpliceDonor_1&quot;            
[23] &quot;PAS_Region&quot;                &quot;ncRNA_incRNA&quot;             
[25] &quot;3_Poised_Promoter&quot;         &quot;.&quot;                        
[27] &quot;14_Repetitive/CNV&quot;         &quot;ncRNA_ctRNA&quot;              
[29] &quot;15_Repetitive/CNV&quot;         &quot;ncRNA_lncRNA&quot;             
[31] &quot;ncRNA_rtRNA&quot;               &quot;ncRNA_snoRNA&quot;             </code></pre>
<pre class="r"><code>results &lt;- list()

for (i in 1:100){
  results[[i]] &lt;-
  bind_rows(
  sample_frac(H3K27AC_AndEqtlClusters, replace=T),
  sample_frac(SplicingClusters, replace=T)) %&gt;%
  left_join(
    dat %&gt;%
    mutate(Locus.iteration = paste(GeneLocus, HyprcolocCluster, sep=&quot;_&quot;)),
    by = &quot;Locus.iteration&quot;
  ) %&gt;%
  mutate(Annotation = str_replace_all(Annotation, &quot;_0$&quot;, &quot;_UnannotatedButObserved&quot;)) %&gt;%
  mutate(Annotation = str_replace_all(Annotation, &quot;_1$&quot;, &quot;_Annotated&quot;)) %&gt;%
  mutate(Annotation = str_replace_all(Annotation, &quot;^\\d+?_&quot;, &quot;ChromHMM:&quot;)) %&gt;%
  filter(!is.na(Annotation)) %&gt;%
  mutate(Annotation = if_else(Annotation==&quot;.&quot;, &quot;No annotation&quot;, Annotation)) %&gt;%
  group_by(Annotation, ClusterGroup) %&gt;%
  summarise(Sum = sum(PosteriorPr)) %&gt;%
  group_by(ClusterGroup) %&gt;%
  mutate(FractionAnnotatedPosteriorInGroup = Sum/sum(Sum, na.rm=T)) %&gt;%
  ungroup() %&gt;%
  pivot_wider(names_from = &quot;ClusterGroup&quot;, values_from = c(&quot;FractionAnnotatedPosteriorInGroup&quot;, &quot;Sum&quot;), id_cols=&quot;Annotation&quot;) %&gt;%
  mutate(Enrichment = FractionAnnotatedPosteriorInGroup_SplicingClusters/FractionAnnotatedPosteriorInGroup_EnhancerOrExpression) %&gt;%
  mutate(i=i)
}

bind_rows(results) %&gt;%
  group_by(Annotation) %&gt;%
  summarise(mean = exp(mean(log(Enrichment), na.rm=T)),
            Percentile10 = quantile(Enrichment, probs = 0.05, na.rm=T),
            Percentile90 = quantile(Enrichment, probs = 0.95, na.rm=T)) %&gt;%
  ggplot(aes(x=Annotation, y=mean)) +
    # geom_vline(xintercept = 1) +
    geom_col() +
    geom_errorbar(aes(ymin=Percentile10, ymax=Percentile90)) +
    theme_bw() +
    scale_y_continuous(trans=&#39;log2&#39;) +
    theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
    coord_flip() +
    labs(x = &quot;Annotation&quot;, y=&quot;Finemapping posterior mass fold enrichment\n(ChromatinQTL|eQTL &lt;-- --&gt; sQTL)&quot;,
         caption = &quot;Bootsrapped 90% confidence interval&quot;)</code></pre>
<p><img src="figure/20220728_CheckFinemapSNPEnrichments.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/bfb625eb2719f3fde37af38bfa3f13d0ff2818cc/docs/figure/20220728_CheckFinemapSNPEnrichments.Rmd/unnamed-chunk-4-1.png" target="_blank">bfb625e</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-08-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>set.seed(100)
results &lt;- list()

for (i in 1:100){
  results[[i]] &lt;-
  bind_rows(
  sample_frac(H3K27AC_AndEqtlClusters, replace=T),
  sample_frac(SplicingClusters, replace=T)) %&gt;%
  left_join(
    dat %&gt;%
    mutate(Locus.iteration = paste(GeneLocus, HyprcolocCluster, sep=&quot;_&quot;)),
    by = &quot;Locus.iteration&quot;
  ) %&gt;%
  mutate(Annotation = str_replace_all(Annotation, &quot;_0$&quot;, &quot;_UnannotatedButObserved&quot;)) %&gt;%
  mutate(Annotation = str_replace_all(Annotation, &quot;_1$&quot;, &quot;_Annotated&quot;)) %&gt;%
  mutate(Annotation = str_replace_all(Annotation, &quot;^\\d+?_&quot;, &quot;ChromHMM:&quot;)) %&gt;%
  filter(!is.na(Annotation)) %&gt;%
  mutate(Annotation = if_else(Annotation==&quot;.&quot;, &quot;No annotation&quot;, Annotation)) %&gt;%
  group_by(Annotation, ClusterGroup) %&gt;%
  summarise(Sum = sum(PosteriorPr)) %&gt;%
  group_by(ClusterGroup) %&gt;%
  mutate(FractionAnnotatedPosteriorInGroup = Sum/sum(Sum, na.rm=T)) %&gt;%
  ungroup() %&gt;%
  pivot_wider(names_from = &quot;ClusterGroup&quot;, values_from = c(&quot;FractionAnnotatedPosteriorInGroup&quot;, &quot;Sum&quot;), id_cols=&quot;Annotation&quot;) %&gt;%
  mutate(Enrichment = FractionAnnotatedPosteriorInGroup_SplicingClusters/FractionAnnotatedPosteriorInGroup_EnhancerOrExpression) %&gt;%
  mutate(i=i)
}

bind_rows(results) %&gt;%
  group_by(Annotation) %&gt;%
  summarise(mean = exp(mean(log(Enrichment), na.rm=T)),
            Percentile10 = quantile(Enrichment, probs = 0.05, na.rm=T),
            Percentile90 = quantile(Enrichment, probs = 0.95, na.rm=T)) %&gt;%
  ggplot(aes(x=Annotation, y=mean)) +
    # geom_vline(xintercept = 1) +
    geom_col() +
    geom_errorbar(aes(ymin=Percentile10, ymax=Percentile90)) +
    theme_bw() +
    scale_y_continuous(trans=&#39;log2&#39;) +
    theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
    coord_flip() +
    labs(x = &quot;Annotation&quot;, y=&quot;Finemapping posterior mass fold enrichment\n(ChromatinQTL|eQTL &lt;-- --&gt; sQTL)&quot;,
         caption = &quot;Bootsrapped 90% confidence interval&quot;)</code></pre>
<p><img src="figure/20220728_CheckFinemapSNPEnrichments.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-2">
Past versions of unnamed-chunk-4-2.png
</button>
</p>
<div id="fig-unnamed-chunk-4-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/bfb625eb2719f3fde37af38bfa3f13d0ff2818cc/docs/figure/20220728_CheckFinemapSNPEnrichments.Rmd/unnamed-chunk-4-2.png" target="_blank">bfb625e</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-08-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok those two separate graphs look similar enough for me, that I think this is enough bootstrap replicates. Let’s wrap this process into a function and try making similar plots comparing different sets of QTLs…</p>
<p>Actually, first, before considering “relative enrichment” in comparing two sets of QTLs, I want to get a sense of the total mass under each annotation in both sets…</p>
<pre class="r"><code>FractionPosteriorInEachAnnotation %&gt;%
  ggplot(aes(x=Annotation, y=FractionAnnotatedPosteriorInGroup*100)) +
  geom_col() +
  facet_wrap(~ClusterGroup) +
  coord_flip() +
  theme_bw()</code></pre>
<p><img src="figure/20220728_CheckFinemapSNPEnrichments.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/bfb625eb2719f3fde37af38bfa3f13d0ff2818cc/docs/figure/20220728_CheckFinemapSNPEnrichments.Rmd/unnamed-chunk-5-1.png" target="_blank">bfb625e</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-08-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FractionPosteriorInEachAnnotation %&gt;%
  ggplot(aes(x=ClusterGroup, y=FractionAnnotatedPosteriorInGroup*100, fill=Annotation)) +
  geom_col(position=&#39;fill&#39;) +
  theme_bw()</code></pre>
<p><img src="figure/20220728_CheckFinemapSNPEnrichments.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-2">
Past versions of unnamed-chunk-5-2.png
</button>
</p>
<div id="fig-unnamed-chunk-5-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/bfb625eb2719f3fde37af38bfa3f13d0ff2818cc/docs/figure/20220728_CheckFinemapSNPEnrichments.Rmd/unnamed-chunk-5-2.png" target="_blank">bfb625e</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-08-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok, so perhaps only a 1/3 of the posterior mass for sQTLs lies in splice site regions.</p>
<p>Now, let’s go back to wrapping the enrichment plot between two sets of QTLs into a function, so I can reuse it to comapre different sets of QTLs</p>
<pre class="r"><code>iterations = 100
GeneLocus_Cluster_QTLSet1 = H3K27AC_AndEqtlClusters
GeneLocus_Cluster_QTLSet2 = SplicingClusters

#Define the function
PlotDifferentialEnrichmentBetweenQTLSets &lt;- function(dat, GeneLocus_Cluster_QTLSet1, GeneLocus_Cluster_QTLSet2, seed=0, iterations=100){
  output &lt;- list()
  results &lt;- list()
  for (i in 1:iterations){
    results[[i]] &lt;-
    bind_rows(
    sample_frac(GeneLocus_Cluster_QTLSet1, replace=T),
    sample_frac(GeneLocus_Cluster_QTLSet2, replace=T),
    .id=&quot;QTL.Set&quot;) %&gt;%
    left_join(
      dat %&gt;%
      mutate(Locus.iteration = paste(GeneLocus, HyprcolocCluster, sep=&quot;_&quot;)),
      by = &quot;Locus.iteration&quot;
    ) %&gt;%
    mutate(Annotation = str_replace_all(Annotation, &quot;_0$&quot;, &quot;_UnannotatedButObserved&quot;)) %&gt;%
    mutate(Annotation = str_replace_all(Annotation, &quot;_1$&quot;, &quot;_Annotated&quot;)) %&gt;%
    mutate(Annotation = str_replace_all(Annotation, &quot;^\\d+?_&quot;, &quot;ChromHMM:&quot;)) %&gt;%
    filter(!is.na(Annotation)) %&gt;%
    mutate(Annotation = if_else(Annotation==&quot;.&quot;, &quot;No annotation&quot;, Annotation)) %&gt;%
    group_by(Annotation, ClusterGroup, QTL.Set) %&gt;%
    summarise(Sum = sum(PosteriorPr)) %&gt;%
    group_by(ClusterGroup, QTL.Set) %&gt;%
    mutate(FractionAnnotatedPosteriorInGroup = Sum/sum(Sum, na.rm=T)) %&gt;%
    ungroup() %&gt;%
    # select(-ClusterGroup) %&gt;%
    pivot_wider(names_from = &quot;QTL.Set&quot;, values_from = c(&quot;FractionAnnotatedPosteriorInGroup&quot;, &quot;Sum&quot;), id_cols=&quot;Annotation&quot;) %&gt;%
    mutate(Enrichment = FractionAnnotatedPosteriorInGroup_1/FractionAnnotatedPosteriorInGroup_2) %&gt;%
    mutate(i=i)
  }
  output[[&quot;data&quot;]] &lt;- bind_rows(results)
  output[[&quot;plot&quot;]] &lt;- output[[&quot;data&quot;]] %&gt;%
  group_by(Annotation) %&gt;%
  summarise(mean = exp(mean(log(Enrichment), na.rm=T)),
            Percentile10 = quantile(Enrichment, probs = 0.05, na.rm=T),
            Percentile90 = quantile(Enrichment, probs = 0.95, na.rm=T)) %&gt;%
  ggplot(aes(x=Annotation, y=mean)) +
    # geom_vline(xintercept = 1) +
    geom_col() +
    geom_errorbar(aes(ymin=Percentile10, ymax=Percentile90)) +
    theme_bw() +
    scale_y_continuous(trans=&#39;log2&#39;) +
    theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
    coord_flip() +
    labs(x = &quot;Annotation&quot;, y=&quot;Finemapping posterior mass relative fold enrichment\nQTL.Set2 &lt;----&gt; QTL.Set1&quot;,
         caption = &quot;Bootsrapped 90% confidence interval&quot;)
  return(output)
}

#Test the function
test.results &lt;- PlotDifferentialEnrichmentBetweenQTLSets(dat, H3K27AC_AndEqtlClusters, SplicingClusters)

test.results$data %&gt;% head()</code></pre>
<pre><code># A tibble: 6 × 7
  Annotation    FractionAnnotat… FractionAnnotat…  Sum_1  Sum_2 Enrichment     i
  &lt;chr&gt;                    &lt;dbl&gt;            &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;      &lt;dbl&gt; &lt;int&gt;
1 ChromHMM:Act…          0.0563          0.0391    383.   98.1       1.44      1
2 ChromHMM:Het…          0.199           0.153    1355.  385.        1.30      1
3 ChromHMM:Ins…          0.00639         0.00709    43.5  17.8       0.902     1
4 ChromHMM:Poi…          0.00378         0.000789   25.7   1.98      4.80      1
5 ChromHMM:Rep…          0.00178         0.00303    12.1   7.60      0.587     1
6 ChromHMM:Rep…          0.0160          0.00790   109.   19.8       2.02      1</code></pre>
<pre class="r"><code>test.results$plot +
  labs(y=&quot;FoldEnrichment\nSplicingQTLs &lt;----&gt; Chromatin|Expression QTLs&quot;)</code></pre>
<p><img src="figure/20220728_CheckFinemapSNPEnrichments.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/bfb625eb2719f3fde37af38bfa3f13d0ff2818cc/docs/figure/20220728_CheckFinemapSNPEnrichments.Rmd/unnamed-chunk-6-1.png" target="_blank">bfb625e</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-08-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Another hypothesis I have is that the chromatin-specific sQTLs will more reflect splicing and therefore be more enriched to splice sites, whereas the poly-A specific sQTLs will more reflect isoform stability. Let’s see…</p>
<pre class="r"><code>hyprcoloc_results %&gt;%
  separate(phenotype_full, into=c(&quot;PC&quot;, &quot;P&quot;), sep=&quot;;&quot;) %&gt;%
  pull(PC) %&gt;% unique()</code></pre>
<pre><code> [1] &quot;polyA.Splicing&quot;                   &quot;MetabolicLabelled.30min&quot;         
 [3] &quot;MetabolicLabelled.60min&quot;          &quot;H3K27AC&quot;                         
 [5] &quot;H3K4ME3&quot;                          &quot;H3K4ME1&quot;                         
 [7] &quot;chRNA.IER&quot;                        &quot;ProCap&quot;                          
 [9] &quot;polyA.Splicing.Subset_YRI&quot;        &quot;CTCF&quot;                            
[11] &quot;chRNA.Splicing.Order&quot;             &quot;chRNA.Expression_ncRNA&quot;          
[13] &quot;Expression.Splicing&quot;              &quot;Expression.Splicing.Subset_YRI&quot;  
[15] &quot;chRNA.Splicing&quot;                   &quot;MetabolicLabelled.30min.Splicing&quot;
[17] &quot;MetabolicLabelled.60min.Splicing&quot; &quot;chRNA.Expression.Splicing&quot;       
[19] &quot;H3K36ME3&quot;                         &quot;polyA.IER&quot;                       
[21] &quot;MetabolicLabelled.30min.IER&quot;      &quot;MetabolicLabelled.60min.IER&quot;     
[23] &quot;polyA.IER.Subset_YRI&quot;             &quot;chRNA.Slopes&quot;                    </code></pre>
<pre class="r"><code># chRNA.Splicing.PCs &lt;- c(&quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;)
# polyA.Splicing.PCs &lt;- c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;polyA.IER&quot;)

chRNA.Splicing.PCs &lt;- c(&quot;chRNA.Splicing&quot;)
polyA.Splicing.PCs &lt;- c(&quot;polyA.Splicing.Subset_YRI&quot;)


chRNA.Specific.SplicingClusters &lt;- hyprcoloc_results %&gt;%
  separate(phenotype_full, into=c(&quot;PC&quot;, &quot;P&quot;), sep=&quot;;&quot;) %&gt;%
  group_by(Locus, iteration) %&gt;%
  filter(any(PC %in% chRNA.Splicing.PCs)) %&gt;%
  filter(any(PC %in% polyA.Splicing.PCs)) %&gt;%
  distinct(Locus,iteration) %&gt;%
  unite(Locus.iteration, Locus,iteration) %&gt;%
  mutate(ClusterGroup = &quot;chRNA.Specific.SplicingClusters&quot;)

polyA.Specific.SplicingClusters &lt;- hyprcoloc_results %&gt;%
  separate(phenotype_full, into=c(&quot;PC&quot;, &quot;P&quot;), sep=&quot;;&quot;) %&gt;%
  group_by(Locus, iteration) %&gt;%
  filter(any(PC %in% polyA.Splicing.PCs)) %&gt;%
  filter(!any(PC %in% chRNA.Splicing.PCs)) %&gt;%
  distinct(Locus,iteration) %&gt;%
  unite(Locus.iteration, Locus,iteration) %&gt;%
  mutate(ClusterGroup = &quot;polyA.Specific.SplicingClusters&quot;)

results &lt;- PlotDifferentialEnrichmentBetweenQTLSets(dat, chRNA.Specific.SplicingClusters, polyA.Specific.SplicingClusters, iterations=400)


results$plot +
  labs(y = &quot;FoldEnrichment\npolyA-specific sQTLs &lt;----&gt; chromatin-specific sQTLs&quot;)</code></pre>
<p><img src="figure/20220728_CheckFinemapSNPEnrichments.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/bfb625eb2719f3fde37af38bfa3f13d0ff2818cc/docs/figure/20220728_CheckFinemapSNPEnrichments.Rmd/unnamed-chunk-7-1.png" target="_blank">bfb625e</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-08-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Now let’s compare eQTLs with colocalizing H3K27AC|H3K4ME3|H3K4ME1, versus eQTLs that colocalize with something but not H3K27AC|H3K4ME3|H3K4ME1</p>
<pre class="r"><code>chRNA.Splicing.PCs &lt;- c(&quot;chRNA.Splicing&quot;)
polyA.Splicing.PCs &lt;- c(&quot;polyA.Splicing.Subset_YRI&quot;)


eQTL.Chromatin.Clusters &lt;- hyprcoloc_results %&gt;%
  separate(phenotype_full, into=c(&quot;PC&quot;, &quot;P&quot;), sep=&quot;;&quot;) %&gt;%
  group_by(Locus, iteration) %&gt;%
  filter(any(PC == &quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
  filter(any(PC %in% c(&quot;H3K27AC&quot;, &quot;H3K4ME3&quot;, &quot;H3K4ME1&quot;))) %&gt;%
  distinct(Locus,iteration) %&gt;%
  unite(Locus.iteration, Locus,iteration) %&gt;%
  mutate(ClusterGroup = &quot;eQTL.Chromatin.Clusters&quot;)

eQTL.NoChromatin.Clusters &lt;- hyprcoloc_results %&gt;%
  separate(phenotype_full, into=c(&quot;PC&quot;, &quot;P&quot;), sep=&quot;;&quot;) %&gt;%
  group_by(Locus, iteration) %&gt;%
  filter(any(PC == &quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
  filter(!any(PC %in% c(&quot;H3K27AC&quot;, &quot;H3K4ME3&quot;, &quot;H3K4ME1&quot;))) %&gt;%
  distinct(Locus,iteration) %&gt;%
  unite(Locus.iteration, Locus,iteration) %&gt;%
  mutate(ClusterGroup = &quot;eQTL.NoChromatin.Clusters&quot;)

intersect(eQTL.Chromatin.Clusters$Locus.iteration, eQTL.NoChromatin.Clusters$Locus.iteration)</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>results &lt;- PlotDifferentialEnrichmentBetweenQTLSets(dat, eQTL.Chromatin.Clusters, eQTL.NoChromatin.Clusters, iterations=400)


results$plot +
  labs(y = &quot;FoldEnrichment\neQTL.NoChromatin.Clusters &lt;----&gt; eQTL.Chromatin.Clusters&quot;)</code></pre>
<p><img src="figure/20220728_CheckFinemapSNPEnrichments.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/bfb625eb2719f3fde37af38bfa3f13d0ff2818cc/docs/figure/20220728_CheckFinemapSNPEnrichments.Rmd/unnamed-chunk-8-1.png" target="_blank">bfb625e</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-08-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Update: Yang suggested making sure that the enrichment for splice sites in non-chromatinQTL eQTLs is not due to the inclusion of chromatin data in the finemapping that biases results by posterior mass towards enhancers (and away from splice sites). His suggestion was to repeat this analyses, using the same sets of eQTLs to compare, but using finemapping results based solely on RNA-seq datasets (polyA, chRNA, and metabolicRNA).</p>
<pre class="r"><code>dat.onlyRNASeq &lt;- fread(&quot;../code/QTL_SNP_Enrichment/FinemapIntersections/MolColocTestOnlyExpression.bed.gz&quot;, col.names=c(&quot;SNP_chrom&quot;, &quot;SNP_start&quot;, &quot;SNP_end&quot;, &quot;SNP&quot;, &quot;PosteriorPr&quot;, &quot;Annotation_chrom&quot;, &quot;Annotation_start&quot;, &quot;Annotation_stop&quot;, &quot;Annotation&quot;, &quot;Overlap&quot;)) %&gt;%
  separate(SNP, into=c(&quot;SNP&quot;, &quot;HyprcolocCluster&quot;, &quot;GeneLocus&quot;), sep=&quot;_&quot;)

hyprcoloc_results.onlyRNASeq &lt;- read_tsv(&quot;../code/hyprcoloc/Results/ForColoc/MolColocTestOnlyExpression/tidy_results_OnlyColocalized.txt.gz&quot;)


hyprcoloc_results_finemap.onlyRNASeq &lt;- read_tsv(&quot;../code/hyprcoloc/Results/ForColoc/MolColocTestOnlyExpression/snpscores.txt.gz&quot;)</code></pre>
<p>Because the hyprcoloc clusters iteration IDs won’t match up between hyprcoloc runs, let’s hackishly just use hyprcoloc clusters in the RNASeq only results if there is only one colocalized cluster.</p>
<pre class="r"><code>hyprcoloc_results.onlyRNASeq %&gt;%
  distinct(Locus, iteration) %&gt;%
  count(iteration)</code></pre>
<pre><code># A tibble: 2 × 2
  iteration     n
      &lt;dbl&gt; &lt;int&gt;
1         1   969
2         2     7</code></pre>
<p>Notably, this only excludes at most 10 clusters</p>
<pre class="r"><code>LociWithEqtlsInRNASeqOnly &lt;- hyprcoloc_results.onlyRNASeq %&gt;%
  group_by(Locus) %&gt;%
  filter(!any(iteration == 2)) %&gt;%
  ungroup() %&gt;% pull(Locus)


results &lt;- 
  dat.onlyRNASeq %&gt;%
  filter(GeneLocus %in% LociWithEqtlsInRNASeqOnly) %&gt;%
  PlotDifferentialEnrichmentBetweenQTLSets(
    eQTL.Chromatin.Clusters %&gt;%
      mutate(Locus.iteration = str_replace(Locus.iteration, &quot;(.+?)_.+?&quot;, &quot;\\1_1&quot;)),
    eQTL.NoChromatin.Clusters %&gt;%
      mutate(Locus.iteration = str_replace(Locus.iteration, &quot;(.+?)_.+?&quot;, &quot;\\1_1&quot;)),
    iteration = 400
  )

results$plot +
  labs(y = &quot;FoldEnrichment\neQTL.NoChromatin.Clusters &lt;----&gt; eQTL.Chromatin.Clusters&quot;,
       title=&quot;Finemapping based only on RNA-seq eQTL colocalizations&quot;)</code></pre>
<p><img src="figure/20220728_CheckFinemapSNPEnrichments.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/971cb7f2a63048d932929f70524003de0921b4e8/docs/figure/20220728_CheckFinemapSNPEnrichments.Rmd/unnamed-chunk-11-1.png" target="_blank">971cb7f</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dat.onlyRNASeq %&gt;%
  mutate(Locus.iteration = paste(GeneLocus, HyprcolocCluster, sep=&quot;_&quot;)) %&gt;%
  filter(GeneLocus %in% LociWithEqtlsInRNASeqOnly) %&gt;%
  left_join(bind_rows(eQTL.Chromatin.Clusters, eQTL.NoChromatin.Clusters),
            by = &quot;Locus.iteration&quot;) %&gt;%
  distinct(GeneLocus, .keep_all=T) %&gt;%
  count(ClusterGroup)</code></pre>
<pre><code>                ClusterGroup   n
1:   eQTL.Chromatin.Clusters 378
2: eQTL.NoChromatin.Clusters 305
3:                      &lt;NA&gt; 281</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] data.table_1.14.2 forcats_0.4.0     stringr_1.4.0     dplyr_1.0.9      
 [5] purrr_0.3.4       readr_1.3.1       tidyr_1.2.0       tibble_3.1.7     
 [9] ggplot2_3.3.6     tidyverse_1.3.0  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5        lubridate_1.7.4   assertthat_0.2.1  rprojroot_2.0.2  
 [5] digest_0.6.20     utf8_1.1.4        R6_2.4.0          cellranger_1.1.0 
 [9] backports_1.4.1   reprex_0.3.0      evaluate_0.15     highr_0.9        
[13] httr_1.4.4        pillar_1.7.0      rlang_1.0.5       readxl_1.3.1     
[17] rstudioapi_0.14   whisker_0.3-2     R.utils_2.9.0     R.oo_1.22.0      
[21] rmarkdown_1.13    labeling_0.3      munsell_0.5.0     broom_1.0.0      
[25] compiler_3.6.1    httpuv_1.5.1      modelr_0.1.8      xfun_0.31        
[29] pkgconfig_2.0.2   htmltools_0.5.3   tidyselect_1.1.2  workflowr_1.6.2  
[33] fansi_0.4.0       crayon_1.3.4      dbplyr_1.4.2      withr_2.5.0      
[37] later_0.8.0       R.methodsS3_1.7.1 grid_3.6.1        jsonlite_1.6     
[41] gtable_0.3.0      lifecycle_1.0.1   DBI_1.1.0         git2r_0.26.1     
[45] magrittr_1.5      scales_1.1.0      cli_3.3.0         stringi_1.4.3    
[49] farver_2.1.0      fs_1.5.2          promises_1.0.1    xml2_1.3.2       
[53] ellipsis_0.3.2    generics_0.1.3    vctrs_0.4.1       tools_3.6.1      
[57] glue_1.6.2        hms_0.5.3         fastmap_1.1.0     yaml_2.2.0       
[61] colorspace_1.4-1  rvest_0.3.5       knitr_1.39        haven_2.3.1      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
