<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>SmallMoleculeProcessing</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">SmallMoleculeProcessing</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-01-10
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>ChromatinSplicingQTLs/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstree57404e6416d66e1d84afd4235e3c979f603823d0targetblank57404e6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/57404e6416d66e1d84afd4235e3c979f603823d0" target="_blank">57404e6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstree57404e6416d66e1d84afd4235e3c979f603823d0targetblank57404e6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/57404e6416d66e1d84afd4235e3c979f603823d0" target="_blank">57404e6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store
    Ignored:    code/.RData
    Ignored:    code/._.DS_Store
    Ignored:    code/._README.md
    Ignored:    code/._report.html
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    code/.snakemake/
    Ignored:    code/APA_Processing/
    Ignored:    code/Alignments/
    Ignored:    code/ChromHMM/
    Ignored:    code/ENCODE/
    Ignored:    code/ExpressionAnalysis/
    Ignored:    code/FastqFastp/
    Ignored:    code/FastqFastpSE/
    Ignored:    code/Genotypes/
    Ignored:    code/H3K36me3_CutAndTag.pdf
    Ignored:    code/IntronSlopes/
    Ignored:    code/Metaplots/
    Ignored:    code/Misc/
    Ignored:    code/MiscCountTables/
    Ignored:    code/Multiqc/
    Ignored:    code/Multiqc_chRNA/
    Ignored:    code/NonCodingRNA/
    Ignored:    code/NonCodingRNA_annotation/
    Ignored:    code/PeakCalling/
    Ignored:    code/Phenotypes/
    Ignored:    code/PlotGruberQTLs/
    Ignored:    code/PlotQTLs/
    Ignored:    code/ProCapAnalysis/
    Ignored:    code/QC/
    Ignored:    code/QTL_SNP_Enrichment/
    Ignored:    code/QTLs/
    Ignored:    code/RPKM_tables/
    Ignored:    code/ReferenceGenome/
    Ignored:    code/Rplots.pdf
    Ignored:    code/Session.vim
    Ignored:    code/SmallMolecule/
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TODO
    Ignored:    code/Tehranchi/
    Ignored:    code/bigwigs/
    Ignored:    code/bigwigs_FromNonWASPFilteredReads/
    Ignored:    code/config/.DS_Store
    Ignored:    code/config/._.DS_Store
    Ignored:    code/config/.ipynb_checkpoints/
    Ignored:    code/config/config.local.yaml
    Ignored:    code/dag.pdf
    Ignored:    code/dag.png
    Ignored:    code/dag.svg
    Ignored:    code/debug.ipynb
    Ignored:    code/debug_python.ipynb
    Ignored:    code/deepTools/
    Ignored:    code/featureCounts/
    Ignored:    code/gwas_summary_stats/
    Ignored:    code/hyprcoloc/
    Ignored:    code/igv_session.xml
    Ignored:    code/log
    Ignored:    code/logs/
    Ignored:    code/notebooks/.ipynb_checkpoints/
    Ignored:    code/pi1/
    Ignored:    code/rules/.ipynb_checkpoints/
    Ignored:    code/rules/OldRules/
    Ignored:    code/rules/notebooks/
    Ignored:    code/scratch/
    Ignored:    code/scripts/.SmallMoleculeModelFits.R.swp
    Ignored:    code/scripts/.ipynb_checkpoints/
    Ignored:    code/scripts/GTFtools_0.8.0/
    Ignored:    code/scripts/__pycache__/
    Ignored:    code/scripts/liftOverBedpe/liftOverBedpe.py
    Ignored:    code/snakemake.dryrun.log
    Ignored:    code/snakemake.log
    Ignored:    code/snakemake.sbatch.log
    Ignored:    code/test.introns.bed
    Ignored:    code/test.introns2.bed
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/._20220414203249_JASPAR2022_combined_matrices_25818_jaspar.txt
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-10.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-11.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-2.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-3.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-4.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-5.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-6.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-7.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-8.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022.csv
    Ignored:    data/Metaplots/.DS_Store

Untracked files:
    Untracked:  code/snakemake_profiles/slurm/__pycache__/

Unstaged changes:
    Modified:   analysis/20221202_ProcessSmallMolecule.Rmd
    Modified:   code/scripts/GenometracksByGenotype

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/20221202_ProcessSmallMolecule.Rmd</code>) and HTML (<code>docs/20221202_ProcessSmallMolecule.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/6f61b11edcd811f58478658904cc017067b5c6af/analysis/20221202_ProcessSmallMolecule.Rmd" target="_blank">6f61b11</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-01-09
</td>
<td>
further SM analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ea425ea0d4eafe82bf3fff9b1e4a6c44222f5e79/analysis/20221202_ProcessSmallMolecule.Rmd" target="_blank">ea425ea</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-12-15
</td>
<td>
add sm processing nb
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/ea425ea0d4eafe82bf3fff9b1e4a6c44222f5e79/docs/20221202_ProcessSmallMolecule.html" target="_blank">ea425ea</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-12-15
</td>
<td>
add sm processing nb
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/fd6fd963acbb3c7ca56dd8ee9c6c90b903b96676/analysis/20221202_ProcessSmallMolecule.Rmd" target="_blank">fd6fd96</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-12-09
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/d03d9d77e27c382c1fc84289351b25d50152163e/analysis/20221202_ProcessSmallMolecule.Rmd" target="_blank">d03d9d7</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-12-06
</td>
<td>
started small molecule processing
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="intro" class="section level2">
<h2>Intro</h2>
<p>From my slack thread with Yang:</p>
<blockquote>
<p>i going to think out loud here about how to pre-process the small molecule data to explore the splicing-induced NMD/expression effects, in case you have thoughts: So to recap, for risdiplam we have 3 polyA DMSO replicates, and one polyA replicate for 3.2, 10, 32, 100, 320, 1000, 3200, and 10000nM doses. Then we have 3 replicates of chRNA for DMSO, 3 replicates for 100nM, and 3 replicates for 3200nM. I’m thinking the 100nM dose and the 3200nM dose will be most useful, since we have polyA and chRNA at those doses as well. But I don’t want to just throw away the other doses in polyA, especially since splicing quantification tends to be noisy, and also because presenting full dose-response curves could be nice. So I will of course start by just mapping and counting junctions and doing the standard leafcutter clustering with all those samples. Then I will focus on GA-GT junctions, and in the polyA data I will fit a dose-response curve to the intronic PSI, and the host gene log(CPM). When I want to compare with chRNA, I will use the delta-PSI from the fit curve at 100nM and 3200nM doses (estimates of error can also be extracted from the fit), and compare to the deltaPSI (or logCPM expression) estimated from 3 chRNA replicates at that dose and the 3 chRNA DMSO replicates (using standard tools like leafcutter, or edgeR, or even just calculating manually). From those processed numbers I will explore the splicing/expression effects</p>
</blockquote>
</div>
<div id="data-tidying-and-exploration" class="section level2">
<h2>Data tidying and exploration</h2>
<p>So here I will process the conut table and splice count tables.</p>
<pre class="r"><code>library(tidyverse)
library(edgeR)
library(data.table)
library(drc)
library(broom)
library(qvalue)

sample_n_of &lt;- function(data, size, ...) {
  dots &lt;- quos(...)
  
  group_ids &lt;- data %&gt;% 
    group_by(!!! dots) %&gt;% 
    group_indices()
  
  sampled_groups &lt;- sample(unique(group_ids), size)
  
  data %&gt;% 
    filter(group_ids %in% sampled_groups)
}</code></pre>
<p>Read in count tables</p>
<pre class="r"><code>Samples &lt;- read_tsv(&quot;../code/config/SmallMoleculeRNASeq.Samples.tsv&quot;)

GeneCounts &lt;- read_tsv(&quot;../code/SmallMolecule/featureCounts/Counts.txt&quot;, comment=&quot;#&quot;) %&gt;%
  rename_at(vars(-c(1:6)), ~str_replace(.x, &quot;SmallMolecule/AlignmentsPass2/(.+?)/Aligned.sortedByCoord.out.bam&quot;, &quot;\\1&quot;))

Genes &lt;- read_tsv(&quot;../code/ExpressionAnalysis/polyA/ExpressedGeneList.txt&quot;, col_names=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;))</code></pre>
<p>One thing I will have to consider is how to best normalize libraries. Since I suspect a genomewide effect at high concentrations, it might make sense to normalize to intronless genes or some set of genes which I don’t think will be generally effected by small molecule. For now I will normalize the standard way (by total libray size, or rather by the default method of <code>calcNormFactors</code> function), and consider other ways to normalize later.</p>
<pre class="r"><code>#For now, use the same list of 14000 genes used in QTL analysis
GeneCounts.filtered &lt;- GeneCounts %&gt;%
  filter(Geneid %in% Genes$name)

# Or alternatively, use protein coding genes with mean CPM&gt;1 in polyA DMSO samples 

# function to help parse gtf file to identify protein coding genes
extract_attributes &lt;- function(gtf_attributes, att_of_interest){
  att &lt;- unlist(strsplit(gtf_attributes, &quot; &quot;))
  if(att_of_interest %in% att){
    return(gsub(&quot;\&quot;|;&quot;,&quot;&quot;, att[which(att %in% att_of_interest)+1]))
  }else{
    return(NA)}
}

#Get protein coding features, using same gtf as used in featureCounts so that gene_ids perfectly match
gtf &lt;- read_tsv(&quot;../code/ReferenceGenome/Annotations/gencode.v34.chromasomal.basic.annotation.gtf&quot;, comment=&quot;#&quot;, n_max=Inf, col_names=c(&quot;seqname&quot;, &quot;source&quot;, &quot;feature&quot;, &quot;start&quot;, &quot;end&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;frame&quot;, &quot;attribute&quot; )) %&gt;%
    filter(feature==&quot;gene&quot;) %&gt;%
    dplyr::select(attribute)
gtf$gene_id &lt;- unlist(lapply(gtf$attribute, extract_attributes, &quot;gene_id&quot;))
gtf$gene_type &lt;- unlist(lapply(gtf$attribute, extract_attributes, &quot;gene_type&quot;))
ProteinCodingGenes &lt;- filter(gtf, gene_type==&quot;protein_coding&quot;) %&gt;% pull(gene_id)

GenesToInclude &lt;- GeneCounts %&gt;%
  dplyr::select(Geneid, contains(&quot;DMSO_NA_LCL_polyA&quot;)) %&gt;%
  filter(Geneid %in% ProteinCodingGenes) %&gt;%
  column_to_rownames(&quot;Geneid&quot;) %&gt;%
  DGEList() %&gt;%
  calcNormFactors() %&gt;%
  cpm(prior.count=0.1, log=F) %&gt;%
  log2() %&gt;%
  apply(1, mean) %&gt;%
  as.data.frame() %&gt;%
  filter(`.` &gt; 1) %&gt;%
  rownames_to_column(&quot;Geneid&quot;)

GeneCounts.filtered &lt;- GeneCounts %&gt;%
  filter(Geneid %in% GenesToInclude$Geneid)

Count.table.StandardNormFactors &lt;- GeneCounts.filtered %&gt;%
  dplyr::select(-c(2:6)) %&gt;%
  column_to_rownames(&quot;Geneid&quot;) %&gt;%
  DGEList() %&gt;%
  calcNormFactors()


CPM.StandardNormFactors &lt;- Count.table.StandardNormFactors %&gt;%
  cpm(prior.count=0.1, log=F) 

CPM.StandardNormFactors %&gt;%
  log2() %&gt;%
  apply(1, mean) %&gt;% hist()</code></pre>
<p><img src="figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ea425ea0d4eafe82bf3fff9b1e4a6c44222f5e79/docs/figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-3-1.png" target="_blank">ea425ea</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>gene_names &lt;- read_tsv(&quot;/project2/yangili1/bjf79/20211209_JingxinRNAseq/data/Genes.list.txt&quot;)

gene_names %&gt;%
  filter(hgnc_symbol %in% c(&quot;HSPB3&quot;, &quot;IFNA1&quot;))</code></pre>
<pre><code># A tibble: 2 × 2
  ensembl_gene_id hgnc_symbol
  &lt;chr&gt;           &lt;chr&gt;      
1 ENSG00000197919 IFNA1      
2 ENSG00000169271 HSPB3      </code></pre>
<pre class="r"><code>CPM.StandardNormFactors.tidy &lt;- CPM.StandardNormFactors %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;Geneid&quot;) %&gt;%
  gather(&quot;Sample&quot;, &quot;CPM&quot;, -Geneid) %&gt;%
  separate(Sample, into=c(&quot;treatment&quot;, &quot;dose.nM&quot;, &quot;Cell.type&quot;, &quot;LibraryType&quot;, &quot;rep&quot;), sep=&quot;_&quot;, convert=T) %&gt;%
  replace_na(list(dose.nM=0)) %&gt;%
  mutate(ensembl_gene_id = str_replace(Geneid, &quot;^(.+?)\\..+$&quot;, &quot;\\1&quot;)) %&gt;%
  left_join(gene_names)


CPM.StandardNormFactors.tidy %&gt;%
  filter(hgnc_symbol %in% c(&quot;HTT&quot;, &quot;STAT1&quot;, &quot;HSPB3&quot;, &quot;H2AC20&quot;, &quot;IFNA1&quot;)) %&gt;%
  filter(LibraryType == &quot;polyA&quot;) %&gt;%
  ggplot(aes(x=dose.nM, y=CPM)) +
  geom_line() +
  geom_point() + 
  scale_x_continuous(trans=&quot;log1p&quot;, limits=c(0, 10000), breaks=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 1, 0.316, 0), labels=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 1, 0.316, 0)) +
  facet_wrap(~hgnc_symbol, scales = &quot;free_y&quot;) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, size=5))</code></pre>
<p><img src="figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ea425ea0d4eafe82bf3fff9b1e4a6c44222f5e79/docs/figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-4-1.png" target="_blank">ea425ea</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>set.seed(0)
CPM.StandardNormFactors.tidy %&gt;%
  filter(LibraryType == &quot;polyA&quot;) %&gt;%
  sample_n_of(20, hgnc_symbol) %&gt;% 
  ggplot(aes(x=dose.nM, y=CPM)) +
  geom_line() +
  geom_point() + 
  scale_x_continuous(trans=&quot;log1p&quot;, limits=c(0, 10000), breaks=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 1, 0.316, 0), labels=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 1, 0.316, 0)) +
  facet_wrap(~hgnc_symbol, scales = &quot;free_y&quot;) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, size=5))</code></pre>
<p><img src="figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-2">
Past versions of unnamed-chunk-4-2.png
</button>
</p>
<div id="fig-unnamed-chunk-4-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ea425ea0d4eafe82bf3fff9b1e4a6c44222f5e79/docs/figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-4-2.png" target="_blank">ea425ea</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="splicing-data" class="section level2">
<h2>Splicing data</h2>
<pre class="r"><code>leafcutter.counts &lt;- read.table(&quot;../code/SmallMolecule/leafcutter/clustering/autosomes/leafcutter_perind_numers.counts.gz&quot;, header=T, sep=&#39; &#39;) %&gt;% as.matrix()

ClusterMax.mat &lt;- leafcutter.counts %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(&quot;junc&quot;) %&gt;%
    mutate(cluster=str_replace(junc, &quot;^(.+?):.+?:.+?:(.+)$&quot;, &quot;\\1_\\2&quot;)) %&gt;%
    group_by(cluster) %&gt;%
    mutate(across(where(is.numeric), sum)) %&gt;%
    ungroup() %&gt;%
    dplyr::select(junc, everything(), -cluster) %&gt;%
    column_to_rownames(&quot;junc&quot;) %&gt;%
    as.matrix()


PSI.df &lt;- (leafcutter.counts / as.numeric(ClusterMax.mat) * 100) %&gt;%
  signif() %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;Leafcutter.ID&quot;) %&gt;%
  mutate(Intron = str_replace(Leafcutter.ID, &quot;(.+?):(.+?):(.+?):clu_.+?_([+-])$&quot;, &quot;\\1:\\2:\\3:\\4&quot;))

Intron.Donors &lt;- fread(&quot;../code/SmallMolecule/leafcutter/JuncfilesMerged.annotated.basic.bed.5ss.tab.gz&quot;, col.names = c(&quot;Intron&quot;, &quot;DonorSeq&quot;, &quot;DonorScore&quot;)) %&gt;%
  mutate(Intron = str_replace(Intron, &quot;(.+?)_(.+?)_(.+?)_(.+?)::.+?$&quot;, &quot;\\1:\\2:\\3:\\4&quot;))

PSI.tidy &lt;- PSI.df %&gt;%
  left_join(Intron.Donors) %&gt;%
  gather(&quot;Sample&quot;, &quot;PSI&quot;,contains(&quot;_&quot;)) %&gt;%
  inner_join(
    leafcutter.counts %&gt;%
      as.data.frame() %&gt;%
      rownames_to_column(&quot;Leafcutter.ID&quot;) %&gt;%
      mutate(Intron = str_replace(Leafcutter.ID, &quot;(.+?):(.+?):(.+?):clu_.+?_([+-])$&quot;, &quot;\\1:\\2:\\3:\\4&quot;)) %&gt;%
      gather(&quot;Sample&quot;, &quot;Counts&quot;,contains(&quot;_&quot;))
  ) %&gt;%
  separate(Sample, into=c(&quot;treatment&quot;, &quot;dose.nM&quot;, &quot;Cell.type&quot;, &quot;LibraryType&quot;, &quot;rep&quot;), sep=&quot;_&quot;, convert=T) %&gt;%
  replace_na(list(dose.nM=0))

PSI.tidy %&gt;%
  filter(LibraryType == &quot;polyA&quot;) %&gt;%
  # mutate(Is.GA.GT = str_detect(DonorSeq, &quot;^\\w{2}GAGT&quot;)) %&gt;%
  # filter(!is.na(Is.GA.GT)) %&gt;%
  mutate(Is.GA.GT = substr(DonorSeq, 3,4)) %&gt;%
  group_by(dose.nM, Is.GA.GT) %&gt;%
  summarise(PSI.summary = sum(Counts, na.rm=T)) %&gt;%
  ungroup() %&gt;%
  drop_na() %&gt;%
  mutate(Sum = sum(PSI.summary)) %&gt;%
  mutate(FractionJuncs = PSI.summary/Sum) %&gt;%
  ggplot(aes(x=dose.nM, y=FractionJuncs)) +
  geom_line() +
  geom_point() + 
  scale_x_continuous(trans=&quot;log1p&quot;, limits=c(0, 10000), breaks=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 1, 0.316, 0), labels=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 1, 0.316, 0)) +
  facet_wrap(~Is.GA.GT, scales = &quot;free_y&quot;) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, size=5)) +
  labs(title=&quot;Fraction of total juncs&quot;)</code></pre>
<p><img src="figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ea425ea0d4eafe82bf3fff9b1e4a6c44222f5e79/docs/figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-5-1.png" target="_blank">ea425ea</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Not sure why that doesn’t look right. let’s try looking at mean deltaPSI</p>
<pre class="r"><code>PSI.ByDonor.Summary &lt;- PSI.tidy %&gt;%
  # filter(LibraryType == &quot;polyA&quot;) %&gt;%
  # mutate(Is.GA.GT = str_detect(DonorSeq, &quot;^\\w{2}GAGT&quot;)) %&gt;%
  # filter(!is.na(Is.GA.GT)) %&gt;%
  mutate(Is.GA.GT = substr(DonorSeq, 3,4)) %&gt;%
  group_by(dose.nM, Is.GA.GT, LibraryType) %&gt;%
  summarise(PSI.summary = mean(PSI, na.rm=T)) %&gt;%
  ungroup() %&gt;%
  drop_na()

left_join(
  PSI.ByDonor.Summary,
  PSI.ByDonor.Summary %&gt;% filter(dose.nM == 0) %&gt;%
    dplyr::select(Is.GA.GT, PSI.summary, LibraryType),
  by=c(&quot;Is.GA.GT&quot;, &quot;LibraryType&quot;)
) %&gt;%
  mutate(MeanPSIFoldChange=(PSI.summary.x-PSI.summary.y)) %&gt;%
  ggplot(aes(x=dose.nM, y=MeanPSIFoldChange, color=Is.GA.GT==&quot;GA&quot;,  group=interaction(Is.GA.GT, LibraryType))) +
  geom_line(aes(linetype=LibraryType)) +
  geom_point() + 
  scale_x_continuous(trans=&quot;log1p&quot;, limits=c(0, 10000), breaks=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 0), labels=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 0)) +
  # facet_wrap(~Is.GA.GT, scales = &quot;free_y&quot;) +
  # scale_y_continuous(trans=&quot;log1p&quot;) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1)) +
  labs(y=&quot;Mean deltaPSI&quot;,caption=str_wrap(&quot;Average PSI across all introns, then calculate difference&quot;, 30))</code></pre>
<p><img src="figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ea425ea0d4eafe82bf3fff9b1e4a6c44222f5e79/docs/figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-6-1.png" target="_blank">ea425ea</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Mean.PSI &lt;- PSI.tidy %&gt;%
  group_by(Intron, dose.nM, DonorSeq, LibraryType) %&gt;%
  summarise(PSI=mean(PSI, na.rm=T)) %&gt;%
  ungroup()

inner_join(
  Mean.PSI,
  Mean.PSI %&gt;%
    filter(dose.nM==0) %&gt;%
    dplyr::select(Intron, DonorSeq, PSI, LibraryType),
  by=c(&quot;Intron&quot;, &quot;DonorSeq&quot;, &quot;LibraryType&quot;)
) %&gt;%
  mutate(Is.GA.GT = substr(DonorSeq, 3,4)) %&gt;%
  mutate(DeltaPSI=(PSI.x-PSI.y)) %&gt;%
  group_by(dose.nM, Is.GA.GT, LibraryType) %&gt;%
  summarise(PSI.summary = mean(DeltaPSI, na.rm=T)) %&gt;%
  ungroup() %&gt;%
  drop_na() %&gt;%
  ggplot(aes(x=dose.nM, y=PSI.summary, color=Is.GA.GT==&quot;GA&quot;,  group=interaction(Is.GA.GT, LibraryType))) +
  geom_line(aes(linetype=LibraryType)) +
  geom_point() + 
  scale_x_continuous(trans=&quot;log1p&quot;, limits=c(0, 10000), breaks=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 0), labels=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 0)) +
  # facet_wrap(~Is.GA.GT, scales = &quot;free_y&quot;) +
  # scale_y_continuous(trans=&quot;log1p&quot;) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1)) +
  labs(y=&quot;Mean deltaPSI&quot;, caption=str_wrap(&quot;Average DeltaPSI across all introns&quot;, 30))</code></pre>
<p><img src="figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-2">
Past versions of unnamed-chunk-6-2.png
</button>
</p>
<div id="fig-unnamed-chunk-6-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ea425ea0d4eafe82bf3fff9b1e4a6c44222f5e79/docs/figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-6-2.png" target="_blank">ea425ea</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Now make a nicer version of this figure. Yang’s suggestions to make it nicer included grouping all 16 diculeotides into three groups for simplicity: GA|GT, AG|GT, and everything else. Then also make some confidence interval, perhaps by bootstrapping.</p>
<pre class="r"><code>dat.to.plot.GAGT.enrichment &lt;- inner_join(
  Mean.PSI,
  Mean.PSI %&gt;%
    filter(dose.nM==0) %&gt;%
    dplyr::select(Intron, DonorSeq, PSI, LibraryType),
  by=c(&quot;Intron&quot;, &quot;DonorSeq&quot;, &quot;LibraryType&quot;)
) %&gt;%
  mutate(Is.GA.GT = substr(DonorSeq, 3,4)) %&gt;%
  mutate(DinculeotideCategory = case_when(
    Is.GA.GT == &quot;AG&quot; ~ &quot;AG|GU&quot;,
    Is.GA.GT == &quot;GA&quot; ~ &quot;GA|GU&quot;,
    TRUE ~ &quot;not (AG or GA)|GU&quot;
  )) %&gt;%
  mutate(DeltaPSI=(PSI.x-PSI.y))


# Bootstrap resamples for confidence intervals
NumIterations &lt;- 200
ResampleResults &lt;- list()
for (i in 1:NumIterations){
  df &lt;- dat.to.plot.GAGT.enrichment %&gt;%
    group_by(dose.nM, DinculeotideCategory, LibraryType) %&gt;%
    sample_frac(replace=T) %&gt;%
    ungroup() %&gt;%
    group_by(dose.nM, DinculeotideCategory, LibraryType) %&gt;%
    summarise(PSI.summary = mean(DeltaPSI, na.rm=T)) %&gt;%
    ungroup() %&gt;%
    drop_na()
  ResampleResults[[i]] &lt;- df
}

bind_rows(ResampleResults, .id=&quot;iteration&quot;) %&gt;%
  group_by(dose.nM, DinculeotideCategory, LibraryType) %&gt;%
  summarise(enframe(quantile(PSI.summary, c(0.025, 0.5, 0.975)), &quot;ResampledQuantile&quot;, &quot;MeanDeltaPSI&quot;)) %&gt;%
  pivot_wider(names_from=&quot;ResampledQuantile&quot;, values_from=&quot;MeanDeltaPSI&quot;) %&gt;%
  ggplot(aes(x=dose.nM, y=`50%`, color=DinculeotideCategory,linetype=LibraryType, group=interaction(DinculeotideCategory, LibraryType))) +
  geom_ribbon(aes(ymin=`2.5%`, ymax=`97.5%`, fill=DinculeotideCategory), alpha=0.1, size=0.5) +
  geom_line(size=1) +
  geom_point() + 
  scale_x_continuous(trans=&quot;log1p&quot;, limits=c(0, 10000), breaks=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 0), labels=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 0)) +
  scale_colour_brewer(type=&quot;qual&quot;, palette = &quot;Dark2&quot;) +
  scale_fill_brewer(type=&quot;qual&quot;, palette = &quot;Dark2&quot;) +
  # facet_wrap(~Is.GA.GT, scales = &quot;free_y&quot;) +
  # scale_y_continuous(trans=&quot;log1p&quot;) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1)) +
  labs(y=&quot;Mean deltaPSI&quot;, x=&quot;dose (nM)&quot;, caption=str_wrap(&quot;Average DeltaPSI across all introns&quot;, 30))</code></pre>
<p><img src="figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ea425ea0d4eafe82bf3fff9b1e4a6c44222f5e79/docs/figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-7-1.png" target="_blank">ea425ea</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="consider-intronless-gene-normalization" class="section level3">
<h3>Consider intronless gene normalization</h3>
<p>let’s look intron annotations and to help find intronless genes to check out as a seperate group.</p>
<pre class="r"><code>IntronAnnotations &lt;- read_tsv(&quot;../code/SmallMolecule/FullSpliceSiteAnnotations/JuncfilesMerged.annotated.basic.bed.gz&quot;)

JuncCountsPerGene &lt;- IntronAnnotations %&gt;%
  filter(!is.na(gene_ids)) %&gt;%
  separate_rows(gene_ids, sep=&#39;,&#39;) %&gt;%
  filter(gene_ids %in% rownames(CPM.StandardNormFactors)) %&gt;%
  group_by(gene_ids) %&gt;%
  summarise(NumUniqueJuncs = n(),
            NumTotalJuncs = sum(score)) %&gt;%
  right_join(
    data.frame(gene_ids = CPM.StandardNormFactors %&gt;% rownames())
  ) %&gt;%
  replace_na(list(NumUniqueJuncs=0, NumTotalJuncs=0)) %&gt;%
  left_join(
    data.frame(
      logCPM = CPM.StandardNormFactors %&gt;%
      log2() %&gt;%
      apply(1, mean)) %&gt;%
      rownames_to_column(&quot;gene_ids&quot;)
  ) %&gt;%
  mutate(FractionJuncs = log2(NumTotalJuncs+0.1) - logCPM) %&gt;%
  mutate(PercentileSplicing = percent_rank(FractionJuncs))

JuncCountsPerGene %&gt;%
  ggplot(aes(x=NumUniqueJuncs + 0.1, y=NumTotalJuncs + 0.1)) +
  geom_point(alpha=0.05) +
  scale_x_continuous(trans=&#39;log10&#39;) +
  scale_y_continuous(trans=&#39;log10&#39;)</code></pre>
<p><img src="figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ea425ea0d4eafe82bf3fff9b1e4a6c44222f5e79/docs/figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-8-1.png" target="_blank">ea425ea</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>JuncCountsPerGene %&gt;%
  gather(key=&quot;Metric&quot;, value=&quot;value&quot;, FractionJuncs, logCPM) %&gt;%
  ggplot(aes(x=value, fill=PercentileSplicing&lt;0.05)) +
  geom_histogram() +
    facet_wrap(~Metric, scales=&quot;free_x&quot;) +
  labs(title=&quot;Expression distribution of lowly spliced genes&quot;, caption=str_wrap(&quot;FractionJuncs is approximately RatioOfJuncReadsToTotalReads across all samples&quot;, 30))</code></pre>
<p><img src="figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-2">
Past versions of unnamed-chunk-8-2.png
</button>
</p>
<div id="fig-unnamed-chunk-8-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ea425ea0d4eafe82bf3fff9b1e4a6c44222f5e79/docs/figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-8-2.png" target="_blank">ea425ea</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>RarelySplicedGenes &lt;- JuncCountsPerGene %&gt;%
  filter(PercentileSplicing &lt; 0.05) %&gt;%
  pull(gene_ids)

set.seed(0)
CPM.StandardNormFactors.tidy %&gt;%
  filter(LibraryType == &quot;polyA&quot;) %&gt;%
  filter(Geneid %in% RarelySplicedGenes) %&gt;%
  sample_n_of(20, hgnc_symbol) %&gt;% 
  ggplot(aes(x=dose.nM, y=CPM)) +
  geom_line() +
  geom_point() + 
  scale_x_continuous(trans=&quot;log1p&quot;, limits=c(0, 10000), breaks=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 1, 0.316, 0), labels=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 1, 0.316, 0)) +
  facet_wrap(~hgnc_symbol, scales = &quot;free_y&quot;) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, size=5))</code></pre>
<p><img src="figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ea425ea0d4eafe82bf3fff9b1e4a6c44222f5e79/docs/figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-9-1.png" target="_blank">ea425ea</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Spearman.cors.CPM &lt;- CPM.StandardNormFactors.tidy %&gt;%
  mutate(IsRarelySplicedGene = Geneid %in% RarelySplicedGenes) %&gt;%
  nest(-Geneid) %&gt;% 
  mutate(cor=map(data,~cor.test(.x$dose.nM, .x$CPM, method = &quot;sp&quot;))) %&gt;%
  mutate(tidied = map(cor, tidy)) %&gt;%
  unnest(tidied, .drop = T)

Spearman.cors.CPM %&gt;%
  unnest(data) %&gt;%
  distinct(Geneid, .keep_all=T) %&gt;%
  ggplot(aes(x=estimate, color=IsRarelySplicedGene)) +
  stat_ecdf() +
  labs(title=&quot;Rarely spliced genes don&#39;t have systematically diff dose response cor coef&quot;, x=&quot;dose:CPM Spearman coef&quot;, y=&quot;ecdf&quot;)</code></pre>
<p><img src="figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-2">
Past versions of unnamed-chunk-9-2.png
</button>
</p>
<div id="fig-unnamed-chunk-9-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ea425ea0d4eafe82bf3fff9b1e4a6c44222f5e79/docs/figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-9-2.png" target="_blank">ea425ea</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok I think it’s not worth the time to do something fancy with library normalization to alleviate concerns of transcriptome wide changes obscuring normalization. I’ll just continue with the standard library size normalized data.</p>
</div>
</div>
<div id="model-fitting" class="section level2">
<h2>Model fitting</h2>
<p>As I have done before, I’m going to use <code>drc</code> package to fit 4 parameter log-logistic function to the PSI values for GA|GT introns, and the host gene logCPM values.</p>
<div id="filtering-introns" class="section level3">
<h3>Filtering introns</h3>
<p>As before, in addition to GA|GT filter, I think a reasonable pre-filter for introns worth modelling is to filter for introns with a strong positive dose:response spearman correlation coefficient.</p>
<p>Let’s start by looking at the distribution of spearman correlation coefficients for a sample of introns</p>
<pre class="r"><code>PSI.tidy %&gt;%
  filter(LibraryType == &quot;polyA&quot;) %&gt;%
  drop_na() %&gt;%
  add_count(Intron) %&gt;%
  filter(n==11) %&gt;%
  distinct(Intron, DonorSeq) %&gt;%
  mutate(Is.GA.GT = substr(DonorSeq, 3,4)) %&gt;%
  count(Is.GA.GT) %&gt;%
  ggplot(aes(x=Is.GA.GT, y=n)) +
  geom_col() +
  geom_text(aes(label=n), color=&quot;red&quot;) +
  labs(title=&quot;Total number introns with complete data&quot;)</code></pre>
<p><img src="figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ea425ea0d4eafe82bf3fff9b1e4a6c44222f5e79/docs/figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-10-1.png" target="_blank">ea425ea</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>set.seed(0)
PSI.tidy.SampleWithCor &lt;- PSI.tidy %&gt;%
  filter(LibraryType == &quot;polyA&quot;) %&gt;%
  drop_na() %&gt;%
  add_count(Intron) %&gt;%
  filter(n==11) %&gt;%
  sample_n_of(10000, Intron) %&gt;% 
  nest(-Intron) %&gt;%
  mutate(cor=map(data,~cor.test(.x$dose.nM, .x$PSI, method = &quot;sp&quot;, alternative=&quot;greater&quot;))) %&gt;%
  mutate(tidied = map(cor, tidy)) %&gt;% 
  unnest(tidied, .drop = T) %&gt;%
  dplyr::select(Intron:data, spearman=estimate, spearman.p = p.value) %&gt;%
  unnest(data)

PSI.tidy.SampleWithCor.summarised &lt;- PSI.tidy.SampleWithCor %&gt;%
  distinct(Intron, .keep_all=T) %&gt;%
  mutate(Is.GA.GT = substr(DonorSeq, 3,4)) %&gt;%
  group_by(Is.GA.GT) %&gt;%
  mutate(q = qvalue(spearman.p)$qvalues) %&gt;%
  ungroup()

PSI.tidy.SampleWithCor.summarised %&gt;%
  ggplot(aes(x=spearman, group=Is.GA.GT, color=Is.GA.GT==&quot;GA&quot;)) +
  stat_ecdf() +
  labs(y=&quot;ecdf&quot;, x=&quot;dose:response Spearman coef&quot;)</code></pre>
<p><img src="figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-2">
Past versions of unnamed-chunk-10-2.png
</button>
</p>
<div id="fig-unnamed-chunk-10-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ea425ea0d4eafe82bf3fff9b1e4a6c44222f5e79/docs/figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-10-2.png" target="_blank">ea425ea</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok, so while GA|GT introns are as a group clearly different, there are still quite a lot of introns that don’t have a positive relationship that probably aren’t even worth modelling, even if just for sake of saving computational time in this notebook. As I have done before, I will just consider GA.GT introns with high spearman correlation coef. But what threshold should I use. I could consider looking at the spearman test P values as a critera.</p>
<pre class="r"><code>PSI.tidy.SampleWithCor.summarised %&gt;%
  ggplot(aes(x=spearman.p)) +
  geom_histogram() +
  facet_wrap(~Is.GA.GT, scales=&quot;free&quot;) +
  labs(y=&quot;freq&quot;, x=&quot;dose:response Spearman coef P&quot;)</code></pre>
<p><img src="figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ea425ea0d4eafe82bf3fff9b1e4a6c44222f5e79/docs/figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-11-1.png" target="_blank">ea425ea</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>PSI.tidy.SampleWithCor.summarised %&gt;%
  ggplot(aes(x=q, group=Is.GA.GT, color=Is.GA.GT==&quot;GA&quot;)) +
  stat_ecdf() +
  scale_x_continuous(trans=&quot;log10&quot;) +
  labs(y=&quot;ecdf&quot;, x=&quot;dose:response Spearman coef FDR&quot;)</code></pre>
<p><img src="figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>PSI.tidy.SampleWithCor.summarised %&gt;%
  filter(Is.GA.GT == &quot;GA&quot; &amp; q&lt;0.01) %&gt;%
  pull(spearman) %&gt;% min()</code></pre>
<pre><code>[1] 0.7890454</code></pre>
<p>Ok so maybe one reasonable thing is to just select all GA|GT introns with spearman test FDR &lt; 0.01 which corresponds roughly to a spearman correlation coef of 0.8… This in practice is similar to what I did previously which is just model introns with spearman coef &gt; 0.9. During model fitting, we can get estimates of parameter standard errors, and if the model fit is really back the fitting process tends to throw errors (does not converge or something like that), we can always filter out bad models after attempting to fit them. So really this pre-filtering is more just to save computational time more than anything else… Ok let’s fit some models now…</p>
</div>
<div id="fit-model-to-splicing" class="section level3">
<h3>Fit model to splicing</h3>
<pre class="r"><code>chRNADoses &lt;- c(0, 100, 3160)


model.dat.df.AllGAGT &lt;- PSI.tidy %&gt;%
  filter(LibraryType == &quot;polyA&quot;) %&gt;%
  drop_na() %&gt;%
  add_count(Intron) %&gt;%
  filter(n==11) %&gt;%
  mutate(Is.GA.GT = substr(DonorSeq, 3,4)) %&gt;%
  filter(Is.GA.GT == &quot;GA&quot;) %&gt;%
  nest(-Intron) %&gt;%
  mutate(cor=map(data,~cor.test(.x$dose.nM, .x$PSI, method = &quot;sp&quot;, alternative=&quot;greater&quot;))) %&gt;%
  mutate(tidied = map(cor, tidy)) %&gt;% 
  unnest(tidied, .drop = T) %&gt;%
  dplyr::select(Intron:data, spearman=estimate, spearman.p = p.value) %&gt;%
  mutate(q = qvalue(spearman.p)$qvalues)

model.dat.df.AllGAGT %&gt;%
  filter(q&lt;0.01) %&gt;%
  pull(spearman) %&gt;% min()</code></pre>
<pre><code>[1] 0.7862256</code></pre>
<pre class="r"><code>model.dat.df.FilteredGAGT &lt;- model.dat.df.AllGAGT %&gt;%
  filter(q&lt;0.01) %&gt;%
  dplyr::select(junc=Intron, everything())

#Plot some model fits
set.seed(0)
model.dat.df.FilteredGAGT %&gt;%
  sample_n_of(30, junc) %&gt;% 
  unnest(data) %&gt;%
  ggplot(aes(x=dose.nM, y=PSI)) +
  geom_point() +
  geom_smooth(method = drm, method.args = list(
    fct = L.4(names=c(&quot;Steepness&quot;, &quot;LowerLimit&quot;, &quot;UpperLimit&quot;, &quot;ED50&quot;)),
    lowerl=c(NA,0,NA,NA),
    upperl=c(NA,NA,100,NA),
    robust = &quot;median&quot;), se = FALSE) +
  scale_x_continuous(trans=&quot;log1p&quot;, limits=c(0, 10000), breaks=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 0)) +
  facet_wrap(~junc, scales=&quot;free_y&quot;) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))</code></pre>
<p><img src="figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Going into this, I was thinking one potentially useful way to express splicing effects would be to consider the difference in the lowerLimit and upperLimit parameters of the fit model. However, I now appreciate how noisy the UpperLimit estimate is since we don’t actually approach saturation of effect in our tested doses. In later analysis I might consider other ways to quantify effect size. In any case, now let’s fit models using that actual <code>drc</code> functions where I can save the coefficients and standard errors to an data frame</p>
<pre class="r"><code>Results &lt;- list()
for(i in 1:nrow(model.dat.df.FilteredGAGT)) {
# for(i in 1:10) {
  tryCatch(
    expr = {
      junc &lt;- model.dat.df.FilteredGAGT$junc[i]
      data &lt;- model.dat.df.FilteredGAGT$data[i] %&gt;% as.data.frame()
      fit &lt;- drm(formula = PSI ~ dose.nM,
                  data = data,
                  fct = LL.4(names=c(&quot;Steepness&quot;, &quot;LowerLimit&quot;, &quot;UpperLimit&quot;, &quot;ED50&quot;)),
                  lowerl=c(NA,0,NA,NA),
                  upperl=c(NA,NA,100,NA),
                  robust = &quot;mean&quot;
                  )
      df.out &lt;- 
        bind_rows(
          coef(summary(fit)) %&gt;%
            as.data.frame() %&gt;%
            rownames_to_column(&quot;param&quot;) %&gt;%
            dplyr::select(param, Estimate, SE=`Std. Error`),
          predict(fit, data.frame(dose.nM=chRNADoses), se.fit = T) %&gt;%
            as.data.frame() %&gt;%
            dplyr::rename(&quot;Estimate&quot;=&quot;Prediction&quot;) %&gt;%
            mutate(param=paste(&quot;Pred&quot;,chRNADoses, sep=&quot;_&quot;))
        )
      Results[[junc]] &lt;- df.out
      message(&quot;Successfully fitted model.&quot;)
    },
    error=function(e){
      if (i &lt; 100){
        cat(&quot;ERROR :&quot;,conditionMessage(e), junc, &quot;\n&quot;)
      }
      })
}</code></pre>
<pre><code>Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
ERROR : Convergence failed chr1:28151059:28174529:- 
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
ERROR : Convergence failed chr1:50535457:50537413:- 
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
ERROR : Convergence failed chr1:70316691:70324853:- 
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
ERROR : Convergence failed chr1:85737505:85744105:- 
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
ERROR : Convergence failed chr1:171710814:171717073:- 
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
ERROR : Convergence failed chr1:111627526:111691334:+ 
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
ERROR : Convergence failed chr1:154262202:154263103:+ 
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
ERROR : Convergence failed chr2:9543226:9543696:- 
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  non-finite finite-difference value [4]
ERROR : Convergence failed chr2:46907969:46909023:- 
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
ERROR : Convergence failed chr2:75702423:75702691:- 
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
ERROR : Convergence failed chr2:175964589:175970199:- 
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
ERROR : Convergence failed chr2:197121544:197121675:- 
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
ERROR : Convergence failed chr2:6974023:6996916:+ 
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  non-finite finite-difference value [4]
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  non-finite finite-difference value [4]
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  non-finite finite-difference value [1]
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  non-finite finite-difference value [4]
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  non-finite finite-difference value [4]
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;
Error in optim(startVec, opfct, hessian = TRUE, method = &quot;L-BFGS-B&quot;, lower = lowerLimits,  : 
  L-BFGS-B needs finite values of &#39;fn&#39;</code></pre>
<pre class="r"><code>ModelFits.Coefficients.GAGTIntrons &lt;- bind_rows(Results, .id=&quot;junc&quot;)


ModelFits.Coefficients.GAGTIntrons %&gt;% head(10)</code></pre>
<pre><code>                     junc                  param      Estimate           SE
1  chr1:2189781:2192647:-  Steepness:(Intercept) -1.332043e+00 1.151592e-01
2  chr1:2189781:2192647:- LowerLimit:(Intercept)  2.249745e-02 2.177514e-02
3  chr1:2189781:2192647:- UpperLimit:(Intercept)  7.232108e+01 1.656313e+02
4  chr1:2189781:2192647:-       ED50:(Intercept)  1.206143e+05 2.281111e+05
5  chr1:2189781:2192647:-                 Pred_0  2.249745e-02 2.177514e-02
6  chr1:2189781:2192647:-               Pred_100  2.817997e-02 2.071654e-02
7  chr1:2189781:2192647:-              Pred_3160  5.833389e-01 5.864401e-02
8  chr1:8365974:8396444:-  Steepness:(Intercept) -2.481852e+00 3.415155e-01
9  chr1:8365974:8396444:- LowerLimit:(Intercept)  1.432810e+00 6.332343e-01
10 chr1:8365974:8396444:- UpperLimit:(Intercept)  4.784370e+01 1.247750e+00</code></pre>
</div>
<div id="fit-model-to-expression" class="section level3">
<h3>Fit model to expression</h3>
<p>Now let’s fit models to the logCPM. To speed this up for sake of this notebook, I will just look at the host genes of the modelled GA|GT introns, and an equally sized random set of other genes.</p>
<pre class="r"><code>ModelFits.Coefficients.GAGTIntrons.MappedToGenes &lt;- ModelFits.Coefficients.GAGTIntrons %&gt;%
  dplyr::select(-SE) %&gt;%
  pivot_wider(names_from=c(&quot;param&quot;), values_from=&quot;Estimate&quot;) %&gt;%
  dplyr::rename(&quot;LowerLimit&quot; =`LowerLimit:(Intercept)`, &quot;UpperLimit&quot; =`UpperLimit:(Intercept)`, &quot;Steepness&quot; =`Steepness:(Intercept)`, &quot;ED50&quot; =`ED50:(Intercept)`) %&gt;%
  inner_join(
    IntronAnnotations %&gt;%
    filter(!is.na(gene_ids)) %&gt;%
    separate_rows(gene_ids, sep=&#39;,&#39;) %&gt;%
    unite(junc, chrom, start, end, strand, sep=&quot;:&quot;)
  )

GA.GT.Modelled.HostGenes &lt;- ModelFits.Coefficients.GAGTIntrons.MappedToGenes$gene_ids

set.seed(0)
ControlHostGenes &lt;- CPM.StandardNormFactors.tidy %&gt;%
  distinct(Geneid) %&gt;%
  filter(!Geneid %in% GA.GT.Modelled.HostGenes) %&gt;%
  sample_n(size=length(GA.GT.Modelled.HostGenes)) %&gt;%
  pull(Geneid)

model.dat.df.CPM.StandardNormFactors &lt;- CPM.StandardNormFactors.tidy %&gt;%
  filter(LibraryType == &quot;polyA&quot;) %&gt;%
  filter(Geneid %in% c(GA.GT.Modelled.HostGenes, ControlHostGenes)) %&gt;%
  mutate(Is.Affected.GAGT.HostGene = if_else(Geneid %in% GA.GT.Modelled.HostGenes, &quot;GAGT&quot;, &quot;!GAGT&quot;)) %&gt;%
  mutate(Log2CPM = log2(CPM)) %&gt;%
  nest(data=-Geneid)

set.seed(0)
model.dat.df.CPM.StandardNormFactors %&gt;%
  sample_n_of(30, Geneid) %&gt;% 
  unnest(data) %&gt;%
  ggplot(aes(x=dose.nM, y=Log2CPM, color=Is.Affected.GAGT.HostGene)) +
  geom_point() +
  geom_smooth(method = drm, method.args = list(
    fct = L.4(names=c(&quot;Steepness&quot;, &quot;LowerLimit&quot;, &quot;UpperLimit&quot;, &quot;ED50&quot;)),
    robust = &quot;mean&quot;), se = FALSE) +
  scale_x_continuous(trans=&quot;log1p&quot;, limits=c(0, 10000), breaks=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 0)) +
  facet_wrap(~Geneid, scales=&quot;free_y&quot;) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))</code></pre>
<p><img src="figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ea425ea0d4eafe82bf3fff9b1e4a6c44222f5e79/docs/figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-14-1.png" target="_blank">ea425ea</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Results &lt;- list()
for(i in 1:nrow(model.dat.df.CPM.StandardNormFactors)) {
# for(i in 1:100) {
  tryCatch(
    expr = {
      Geneid &lt;- model.dat.df.CPM.StandardNormFactors$Geneid[i]
      data &lt;- model.dat.df.CPM.StandardNormFactors$data[i] %&gt;% as.data.frame()
      fit &lt;- drm(formula = Log2CPM ~ dose.nM,
                  data = data,
                  fct = LL.4(names=c(&quot;Steepness&quot;, &quot;LowerLimit&quot;, &quot;UpperLimit&quot;, &quot;ED50&quot;)),
                  robust = &quot;mean&quot;
                  )
      df.out &lt;- 
        bind_rows(
          coef(summary(fit)) %&gt;%
            as.data.frame() %&gt;%
            rownames_to_column(&quot;param&quot;) %&gt;%
            dplyr::select(param, Estimate, SE=`Std. Error`),
          predict(fit, data.frame(dose.nM=chRNADoses), se.fit = T) %&gt;%
            as.data.frame() %&gt;%
            dplyr::rename(&quot;Estimate&quot;=&quot;Prediction&quot;) %&gt;%
            mutate(param=paste(&quot;Pred&quot;,chRNADoses, sep=&quot;_&quot;))
        )
      Results[[Geneid]] &lt;- df.out
      message(&quot;Successfully fitted model.&quot;)
    },
    error=function(e){
      if (i &lt; 100){
        cat(&quot;ERROR :&quot;,conditionMessage(e), Geneid, &quot;\n&quot;)
      }
      })
}</code></pre>
<pre><code>Error in optim(startVec, opfct, hessian = TRUE, method = optMethod, control = list(maxit = maxIt,  : 
  non-finite finite-difference value [1]
ERROR : Convergence failed ENSG00000077549.19 
Error in optim(startVec, opfct, hessian = TRUE, method = optMethod, control = list(maxit = maxIt,  : 
  non-finite finite-difference value [4]
Error in optim(startVec, opfct, hessian = TRUE, method = optMethod, control = list(maxit = maxIt,  : 
  non-finite finite-difference value [4]
Error in optim(startVec, opfct, hessian = TRUE, method = optMethod, control = list(maxit = maxIt,  : 
  non-finite finite-difference value [4]
Error in optim(startVec, opfct, hessian = TRUE, method = optMethod, control = list(maxit = maxIt,  : 
  non-finite value supplied by optim
Error in optim(startVec, opfct, hessian = TRUE, method = optMethod, control = list(maxit = maxIt,  : 
  non-finite finite-difference value [4]
Error in optim(startVec, opfct, hessian = TRUE, method = optMethod, control = list(maxit = maxIt,  : 
  non-finite finite-difference value [4]
Error in optim(startVec, opfct, hessian = TRUE, method = optMethod, control = list(maxit = maxIt,  : 
  non-finite finite-difference value [4]
Error in optim(startVec, opfct, hessian = TRUE, method = optMethod, control = list(maxit = maxIt,  : 
  non-finite finite-difference value [4]</code></pre>
<pre class="r"><code>ModelFits.Coefficients.Genes &lt;- bind_rows(Results, .id=&quot;Geneid&quot;)</code></pre>
<p>Now let’s merge the gene expression fit coefficients table with the splicing table</p>
<pre class="r"><code>IntronTypeAnnotations &lt;- read_tsv(&quot;../code/SplicingAnalysis/IntronTypeAnnotations.txt.gz&quot;) %&gt;%
  unite(junc, chrom, start, stop, strand, sep=&quot;:&quot;) %&gt;%
  dplyr::select(junc, Geneid=gene_id, Annotation)



MergedTable &lt;- ModelFits.Coefficients.GAGTIntrons.MappedToGenes %&gt;%
  dplyr::select(junc:Pred_3160, Geneid=gene_ids) %&gt;%
  left_join(IntronTypeAnnotations) %&gt;%
  replace_na(list(Annotation=&quot;Unannotated&quot;)) %&gt;%
  full_join(
    ModelFits.Coefficients.Genes %&gt;%
      dplyr::select(-SE) %&gt;%
      pivot_wider(names_from=c(&quot;param&quot;), values_from=&quot;Estimate&quot;) %&gt;%
  dplyr::rename(&quot;LowerLimit&quot; =`LowerLimit:(Intercept)`, &quot;UpperLimit&quot; =`UpperLimit:(Intercept)`, &quot;Steepness&quot; =`Steepness:(Intercept)`, &quot;ED50&quot; =`ED50:(Intercept)`),
    by=&quot;Geneid&quot;, suffix=c(&quot;.Splicing&quot;, &quot;.Expression&quot;)) %&gt;%
  mutate(LimitDiff.Splicing = (LowerLimit.Splicing - UpperLimit.Splicing) * sign(Steepness.Splicing)) %&gt;%
  mutate(LimitDiff.Expression = (LowerLimit.Expression - UpperLimit.Expression) * sign(Steepness.Expression))

MergedTable %&gt;%
  mutate(Geneset = case_when(
    !is.na(Annotation) ~ Annotation,
    TRUE ~ &quot;Not Diff GA.GT Host&quot;
  )) %&gt;%
  count(Geneset) %&gt;%
  ggplot(aes(x=Geneset, y=n)) +
  geom_col() +
  geom_label(aes(label=n), color=&#39;red&#39;) +
  coord_flip() +
  labs(x=&quot;category&quot;, caption=str_wrap(&quot;Number of host genes with model fits by intron type&quot;, 30))</code></pre>
<p><img src="figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ea425ea0d4eafe82bf3fff9b1e4a6c44222f5e79/docs/figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-15-1.png" target="_blank">ea425ea</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>MergedTable %&gt;%
  mutate(Geneset = case_when(
    !is.na(Annotation) ~ Annotation,
    TRUE ~ &quot;Not Diff GA.GT Host&quot;
  )) %&gt;%
  ggplot(aes(x=LimitDiff.Expression, color=Geneset)) +
  geom_vline(xintercept=0) +
  stat_ecdf() +
  coord_cartesian(xlim=c(-5,5)) +
  labs(x=&quot;Effect size at fit limits, Log2FC&quot;, y=&quot;ecdf&quot;)</code></pre>
<p><img src="figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-2">
Past versions of unnamed-chunk-15-2.png
</button>
</p>
<div id="fig-unnamed-chunk-15-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ea425ea0d4eafe82bf3fff9b1e4a6c44222f5e79/docs/figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-15-2.png" target="_blank">ea425ea</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok as expected, the crpyic unannotated introns are generally associated with down-regulation of host gene.</p>
<p>Eventually I want to compare this to chRNA.. Maybe a more interpretable thing to do will be to express effect size in terms of the logFC at the same doses used in chRNA (extracted from the model fit), so I can then compare to differential expression analysis that I do in chRNA.</p>
<pre class="r"><code>MergedTable %&gt;%
  mutate(Geneset = case_when(
    !is.na(Annotation) ~ Annotation,
    TRUE ~ &quot;Not Diff GA.GT Host&quot;
  )) %&gt;%
  mutate(EffectSize = Pred_3160.Expression - Pred_0.Expression) %&gt;%
  ggplot(aes(x=EffectSize, color=Geneset)) +
  geom_vline(xintercept=0) +
  stat_ecdf() +
  coord_cartesian(xlim=c(-2,2)) +
  labs(x=&quot;Effect size from model fit at 3160nM\n(Log2FC)&quot;, y=&quot;ecdf&quot;)</code></pre>
<p><img src="figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>MergedTable %&gt;%
  mutate(Geneset = case_when(
    !is.na(Annotation) ~ Annotation,
    TRUE ~ &quot;Not Diff GA.GT Host&quot;
  )) %&gt;%
  mutate(ExpressionEffectSize = Pred_3160.Expression - Pred_0.Expression) %&gt;%
  mutate(SplicingEffectSize = Pred_3160.Splicing - Pred_0.Splicing) %&gt;%
  ggplot(aes(x=SplicingEffectSize, y=ExpressionEffectSize, color=Geneset)) +
  geom_hline(yintercept=0) +
  geom_vline(xintercept=0) +
  geom_point() +
  geom_smooth(method=&#39;lm&#39;) +
  facet_wrap(~Geneset)</code></pre>
<p><img src="figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-16-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  labs(x=&quot;Effect size from model fit at 3160nM\n(Log2FC)&quot;, y=&quot;ecdf&quot;)</code></pre>
<pre><code>$x
[1] &quot;Effect size from model fit at 3160nM\n(Log2FC)&quot;

$y
[1] &quot;ecdf&quot;

attr(,&quot;class&quot;)
[1] &quot;labels&quot;</code></pre>
<p>Write out a file for Yang to try to reannotate some of the unannotated ones</p>
<pre class="r"><code>IntronAnnotations.sQTL &lt;- read_tsv(&quot;../code/SplicingAnalysis/IntronTypeAnnotations.JoinedWithLeafcutterJuncList.txt.gz&quot;)


PSI.df %&gt;%
  dplyr::select(Leafcutter.ID) %&gt;%
  separate(Leafcutter.ID, into=c(&quot;#Chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;clust&quot;),remove=F, convert=T, sep=&quot;:&quot;) %&gt;%
  mutate(strand = str_extract(clust, &quot;[+-]&quot;)) %&gt;%
  mutate(gid_SM=paste(`#Chrom`, &quot;_SmallMolecule&quot;, clust, sep=&quot;&quot;)) %&gt;%
  dplyr::select(-Leafcutter.ID) %&gt;%
  full_join(IntronAnnotations.sQTL) %&gt;%
  dplyr::select(1:3, gid, strand, gene_id, Annotation, gid_SM) %&gt;%
  replace_na(list(Annotation=&quot;Unannotated&quot;)) %&gt;%
  write_tsv(&quot;../code/SplicingAnalysis/IntronTypeAnnotations.JoinedWithLeafcutterJuncList.RisdiJuncsAdded.txt.gz&quot;)</code></pre>
</div>
</div>
<div id="differential-expression-in-chrna" class="section level2">
<h2>Differential expression in chRNA</h2>
<pre class="r"><code>Count.table.StandardNormFactors$samples</code></pre>
<pre><code>                            group lib.size norm.factors
Risdiplam_3160_LCL_chRNA_1      1 15242787    1.1129660
Risdiplam_3160_LCL_chRNA_2      1 13262913    1.1593596
Risdiplam_3160_LCL_chRNA_3      1 10313945    1.1886588
Risdiplam_100_LCL_chRNA_1       1  7101348    1.4206660
Risdiplam_100_LCL_chRNA_2       1 15575980    1.1715925
Risdiplam_100_LCL_chRNA_3       1  8174969    1.3053154
DMSO_NA_LCL_chRNA_1             1 19457197    1.2273345
DMSO_NA_LCL_chRNA_2             1 16172051    1.2142297
DMSO_NA_LCL_chRNA_3             1 12813080    1.2125163
DMSO_NA_LCL_polyA_1             1 63047944    0.8407132
DMSO_NA_LCL_polyA_2             1 59264088    0.8392651
DMSO_NA_LCL_polyA_3             1 56232771    0.8471394
Risdiplam_10000_LCL_polyA_1     1 68816955    0.8380396
Risdiplam_3160_LCL_polyA_1      1 71370289    0.8491881
Risdiplam_1000_LCL_polyA_1      1 47015535    0.8232758
Risdiplam_316_LCL_polyA_1       1 60256678    0.8565196
Risdiplam_100_LCL_polyA_1       1 64039975    0.8628250
Risdiplam_31.6_LCL_polyA_1      1 53215062    0.8637613
Risdiplam_10_LCL_polyA_1        1 54901814    0.8621561
Risdiplam_3.16_LCL_polyA_1      1 59836419    0.8616918</code></pre>
<pre class="r"><code>Count.table.StandardNormFactors.chRNA &lt;- Count.table.StandardNormFactors[, c(T,T,T, F,F,F, T,T,T, F,F,F,F,F, F,F,F,F,F,F)]

Count.table.StandardNormFactors.chRNA$samples$group &lt;- factor(c(&quot;Treat&quot;,&quot;Treat&quot;,&quot;Treat&quot;,&quot;Control&quot;,&quot;Control&quot;,&quot;Control&quot;), levels=c(&quot;Control&quot;, &quot;Treat&quot;))

Count.table.StandardNormFactors.chRNA$samples</code></pre>
<pre><code>                             group lib.size norm.factors
Risdiplam_3160_LCL_chRNA_1   Treat 15242787     1.112966
Risdiplam_3160_LCL_chRNA_2   Treat 13262913     1.159360
Risdiplam_3160_LCL_chRNA_3   Treat 10313945     1.188659
DMSO_NA_LCL_chRNA_1        Control 19457197     1.227334
DMSO_NA_LCL_chRNA_2        Control 16172051     1.214230
DMSO_NA_LCL_chRNA_3        Control 12813080     1.212516</code></pre>
<pre class="r"><code>DiffExpression &lt;- Count.table.StandardNormFactors.chRNA %&gt;%
  estimateDisp() %&gt;%
  exactTest()</code></pre>
<pre><code>Design matrix not provided. Switch to the classic mode.</code></pre>
<pre class="r"><code>DiffExpression.df &lt;- DiffExpression$table %&gt;%
   as.data.frame() %&gt;%
   rownames_to_column(&quot;Geneid&quot;) %&gt;%
   mutate(q = qvalue(PValue)$qvalues)

DiffExpression.df %&gt;% pull(PValue) %&gt;% hist()</code></pre>
<p><img src="figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ea425ea0d4eafe82bf3fff9b1e4a6c44222f5e79/docs/figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-18-1.png" target="_blank">ea425ea</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DiffExpression.df %&gt;%
  mutate(FDR_10 = q &lt; 0.1,
         FDR_05 = q &lt; 0.05,
         FDR_01 = q&lt; 0.01) %&gt;%
  dplyr::select(Geneid, FDR_10:FDR_01) %&gt;%
  gather(key=&quot;Threshold&quot;, value=&quot;IsSig&quot;, FDR_10:FDR_01) %&gt;%
  count(Threshold, IsSig)</code></pre>
<pre><code>  Threshold IsSig    n
1    FDR_01 FALSE 6621
2    FDR_01  TRUE 3856
3    FDR_05 FALSE 5224
4    FDR_05  TRUE 5253
5    FDR_10 FALSE 4243
6    FDR_10  TRUE 6234</code></pre>
<pre class="r"><code>Count.table.StandardNormFactors.chRNA[&quot;ENSG00000186891.14&quot;,]$counts %&gt;%
  t() %&gt;%
  as.data.frame()</code></pre>
<pre><code>                           ENSG00000186891.14
Risdiplam_3160_LCL_chRNA_1                195
Risdiplam_3160_LCL_chRNA_2                232
Risdiplam_3160_LCL_chRNA_3                142
DMSO_NA_LCL_chRNA_1                       189
DMSO_NA_LCL_chRNA_2                       123
DMSO_NA_LCL_chRNA_3                        81</code></pre>
<p>Now integrate diff exrpession (dose=3160nM) results in polyA titration series and chRNA..</p>
<pre class="r"><code>DiffExpression.df %&gt;%
  inner_join(MergedTable) %&gt;%
  mutate(Geneset = case_when(
    !is.na(Annotation) ~ Annotation,
    TRUE ~ &quot;Not Diff GA.GT Host&quot;
  )) %&gt;%
  # pull(Geneset) %&gt;% unique()
  mutate(PolyAEffectSize = Pred_3160.Expression - Pred_0.Expression) %&gt;%
  gather(key=&quot;Metric&quot;, value=&quot;value&quot;, PolyAEffectSize, logFC) %&gt;%
  filter(Geneset %in% c(&quot;Not Diff GA.GT Host&quot;, &quot;Unannotated&quot;, &quot;Unique to nonsense_mediated_decay&quot;, &quot;In protein_coding&quot;)) %&gt;%
  mutate(Metric = recode(Metric, !!!c(&quot;PolyAEffectSize&quot;=&quot;polyA&quot;, &quot;logFC&quot;=&quot;chRNA&quot;))) %&gt;%
  mutate(Geneset = recode(Geneset, !!!c(&quot;Not Diff GA.GT Host&quot;=&quot;Control genes&quot;, &quot;Unannotated&quot;=&quot;Unannotated intron&quot;, &quot;Unique to nonsense_mediated_decay&quot;=&quot;annotated NMD intron&quot;, &quot;In protein_coding&quot;=&quot;Annotated canonical intron&quot;))) %&gt;%
  ggplot(aes(x=value, color=Geneset)) +
  geom_vline(xintercept=0) +
  stat_ecdf(aes(linetype=Metric)) +
  coord_cartesian(xlim=c(-2,2)) +
  labs(x=&quot;Host gene effect size at 3160nM, log2FC&quot;, y=&quot;ecdf&quot;, linetype=&quot;Dataset&quot;, caption=&quot;Host genes of GA-GT induced introns&quot;) +
  theme_bw()</code></pre>
<p><img src="figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DiffExpression.df %&gt;%
  inner_join(MergedTable) %&gt;%
  mutate(Geneset = case_when(
    !is.na(Annotation) ~ Annotation,
    TRUE ~ &quot;Not Diff GA.GT Host&quot;
  )) %&gt;%
  mutate(PolyAEffectSize = Pred_3160.Expression - Pred_0.Expression) %&gt;%
  filter(Geneset %in% c(&quot;Not Diff GA.GT Host&quot;, &quot;Unannotated&quot;, &quot;Unique to nonsense_mediated_decay&quot;, &quot;In protein_coding&quot;)) %&gt;%
  mutate(Geneset = recode(Geneset, !!!c(&quot;Not Diff GA.GT Host&quot;=&quot;Control genes&quot;, &quot;Unannotated&quot;=&quot;Unannotated intron&quot;, &quot;Unique to nonsense_mediated_decay&quot;=&quot;annotated NMD intron&quot;, &quot;In protein_coding&quot;=&quot;Annotated canonical intron&quot;))) %&gt;%
  # ggplot(aes(x=value, color=Geneset)) +
  ggplot(aes(x=PolyAEffectSize, y=logFC, color=Geneset)) +
  geom_hline(yintercept=0) +
  geom_vline(xintercept=0) +
  geom_point(alpha=0.5) +
  geom_smooth(method=&#39;lm&#39;) +
  facet_wrap(~Geneset) +
  labs(x=&quot;polyA Effect size&quot;, y=&quot;chRNA Effect size&quot;, caption=&quot;Host genes of GA-GT induced introns.\nSplicing effects on expression are polyA-specific&quot;)</code></pre>
<p><img src="figure/20221202_ProcessSmallMolecule.Rmd/unnamed-chunk-19-2.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] qvalue_2.16.0     broom_1.0.0       drc_3.0-1         MASS_7.3-51.4    
 [5] data.table_1.14.2 edgeR_3.26.5      limma_3.40.6      forcats_0.4.0    
 [9] stringr_1.4.0     dplyr_1.0.9       purrr_0.3.4       readr_1.3.1      
[13] tidyr_1.2.0       tibble_3.1.7      ggplot2_3.3.6     tidyverse_1.3.0  

loaded via a namespace (and not attached):
 [1] nlme_3.1-140       fs_1.5.2           lubridate_1.7.4    RColorBrewer_1.1-2
 [5] httr_1.4.4         rprojroot_2.0.2    tools_3.6.1        backports_1.4.1   
 [9] utf8_1.1.4         R6_2.4.0           mgcv_1.8-40        DBI_1.1.0         
[13] colorspace_1.4-1   withr_2.5.0        tidyselect_1.1.2   curl_3.3          
[17] compiler_3.6.1     git2r_0.26.1       cli_3.3.0          rvest_0.3.5       
[21] xml2_1.3.2         sandwich_3.0-2     labeling_0.3       scales_1.1.0      
[25] mvtnorm_1.1-3      digest_0.6.20      foreign_0.8-71     R.utils_2.9.0     
[29] rmarkdown_1.13     rio_0.5.16         pkgconfig_2.0.2    htmltools_0.5.3   
[33] plotrix_3.8-2      highr_0.9          dbplyr_1.4.2       fastmap_1.1.0     
[37] rlang_1.0.5        readxl_1.3.1       rstudioapi_0.14    farver_2.1.0      
[41] generics_0.1.3     zoo_1.8-10         jsonlite_1.6       gtools_3.9.2.2    
[45] R.oo_1.22.0        zip_2.2.1          car_3.0-5          magrittr_1.5      
[49] Matrix_1.5-3       Rcpp_1.0.5         munsell_0.5.0      fansi_0.4.0       
[53] abind_1.4-5        R.methodsS3_1.7.1  lifecycle_1.0.1    stringi_1.4.3     
[57] multcomp_1.4-19    whisker_0.3-2      yaml_2.2.0         carData_3.0-5     
[61] plyr_1.8.4         grid_3.6.1         promises_1.0.1     crayon_1.3.4      
[65] lattice_0.20-38    haven_2.3.1        splines_3.6.1      hms_0.5.3         
[69] locfit_1.5-9.1     knitr_1.39         pillar_1.7.0       reshape2_1.4.3    
[73] codetools_0.2-18   reprex_0.3.0       glue_1.6.2         evaluate_0.15     
[77] modelr_0.1.8       vctrs_0.4.1        httpuv_1.5.1       cellranger_1.1.0  
[81] gtable_0.3.0       assertthat_0.2.1   xfun_0.31          openxlsx_4.1.0.1  
[85] later_0.8.0        survival_2.44-1.1  workflowr_1.6.2    TH.data_1.1-1     
[89] ellipsis_0.3.2    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
