<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Explore GWAS colocalizations</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Explore GWAS colocalizations</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-01-24
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>ChromatinSplicingQTLs/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstreecc1c333631f88b8b65947c07e7a1032b2a016ae2targetblankcc1c333a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/cc1c333631f88b8b65947c07e7a1032b2a016ae2" target="_blank">cc1c333</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstreecc1c333631f88b8b65947c07e7a1032b2a016ae2targetblankcc1c333a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/cc1c333631f88b8b65947c07e7a1032b2a016ae2" target="_blank">cc1c333</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    ._.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store
    Ignored:    code/.Rhistory
    Ignored:    code/._.DS_Store
    Ignored:    code/._README.md
    Ignored:    code/._dag.pdf
    Ignored:    code/._report.html
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    code/.snakemake/
    Ignored:    code/Alignments/
    Ignored:    code/ENCODE/
    Ignored:    code/ExpressionAnalysis/
    Ignored:    code/Fastq/
    Ignored:    code/FastqFastp/
    Ignored:    code/FastqFastpSE/
    Ignored:    code/FastqSE/
    Ignored:    code/Gather_gwas_summary_stats/
    Ignored:    code/Genotypes/
    Ignored:    code/Multiqc/
    Ignored:    code/Multiqc_chRNA/
    Ignored:    code/PeakCalling/
    Ignored:    code/Phenotypes/
    Ignored:    code/PlotGruberQTLs/
    Ignored:    code/ProCapAnalysis/
    Ignored:    code/QC/
    Ignored:    code/QTLs/
    Ignored:    code/ReferenceGenome/
    Ignored:    code/Session.vim
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TODO
    Ignored:    code/bigwigs/
    Ignored:    code/bigwigs_FromNonWASPFilteredReads/
    Ignored:    code/config/.DS_Store
    Ignored:    code/config/._.DS_Store
    Ignored:    code/config/ExternalFastqDataAccessions/
    Ignored:    code/config/OldSamplesConfig/
    Ignored:    code/dag.pdf
    Ignored:    code/featureCounts/
    Ignored:    code/gwas_summary_stats/
    Ignored:    code/hyprcoloc/
    Ignored:    code/logs/
    Ignored:    code/notebooks/.ipynb_checkpoints/
    Ignored:    code/out.hap.ld
    Ignored:    code/out.log
    Ignored:    code/report.html
    Ignored:    code/rules/OldRules/
    Ignored:    code/scratch/
    Ignored:    code/scripts/GTFtools_0.8.0/
    Ignored:    code/scripts/__pycache__/
    Ignored:    code/scripts/liftOverBedpe/liftOverBedpe.py
    Ignored:    code/snakemake.log
    Ignored:    code/snakemake.sbatch.log
    Ignored:    data/._PRJEB1350_RunTable.Ding_etal_CTCF.txt
    Ignored:    data/._igsr_samples.tsv
    Ignored:    data/._list_gwas_summary_statistics_PMID27863252.csv
    Ignored:    data/GrowthNotes/._20210830_GrowthNotes_chRNA.ConcentrationsToReplate.txt

Untracked files:
    Untracked:  code/snakemake_profiles/slurm/__pycache__/

Unstaged changes:
    Modified:   analysis/20211217_GenewiseColocFirstLook.Rmd
    Modified:   analysis/20220114_ColocalizationEffectSizeCorrelations.Rmd
    Modified:   code/rules/report.smk
    Modified:   code/scripts/BamToBigwig.sh
    Modified:   code/scripts/GenometracksByGenotype

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/20220114_ColocalizationEffectSizeCorrelations.Rmd</code>) and HTML (<code>docs/20220114_ColocalizationEffectSizeCorrelations.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/0c8ee9c65abd3ee8c00bed55cfafcfd2caa5bdbd/analysis/20220114_ColocalizationEffectSizeCorrelations.Rmd" target="_blank">0c8ee9c</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-01-18
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/0c8ee9c65abd3ee8c00bed55cfafcfd2caa5bdbd/docs/20220114_ColocalizationEffectSizeCorrelations.html" target="_blank">0c8ee9c</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-01-18
</td>
<td>
update site
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>I have previously colocalized molecular traits gene-wise using hyprcoloc and saved the important summary stats, including beta values for each molecular trait for the top SNP for the hyprcoloc cluster, to file. Here I aim to check how those beta values correlate among different trait types for colocalized traits. For example, among H3K27Ac QTL peaks that colocalize with gene expression QTLs in polyA RNA-seq, are the beta values correlated. Since the paridigm is that H2K27Ac enahncers are activating, such that increase in enhancer signal would result in increase in target gene expression, I expect a generally positive correlation. More interstingly will be to check these kinds of correlations for splicing phenotypes. There are some reports that splicing (via the spliceosome) activates H3K4me3 writers nearby for example (<a href="https://pubmed.ncbi.nlm.nih.gov/31787377/">EMATS mechanism</a>), though Li et al did not see enrichment of sQTLs among H3K4me3 QTLs. So it would be interseting to see if we do or do not see these correlations among colocalization traits.</p>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching packages ────────────────────────────────── tidyverse 1.3.0 ──</code></pre>
<pre><code>✓ ggplot2 3.3.3     ✓ purrr   0.3.4
✓ tibble  2.1.3     ✓ dplyr   0.8.3
✓ tidyr   1.1.0     ✓ stringr 1.4.0
✓ readr   1.3.1     ✓ forcats 0.4.0</code></pre>
<pre><code>── Conflicts ───────────────────────────────────── tidyverse_conflicts() ──
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(gplots)</code></pre>
<pre><code>
Attaching package: &#39;gplots&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    lowess</code></pre>
<pre class="r"><code>library(data.table)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre class="r"><code>dat &lt;- read_tsv(&quot;../output/hyprcoloc_results/ForColoc/hyprcoloc.results.OnlyColocalized.Stats.txt.gz&quot;) %&gt;%
  select(-iteration) %&gt;%
  unite(Locus_snp, Locus, snp) %&gt;%
  mutate(pheno_class = str_replace(phenotype_full, &quot;(.+?);.+$&quot;, &quot;\\1&quot;))</code></pre>
<pre><code>Parsed with column specification:
cols(
  snp = col_character(),
  beta = col_double(),
  beta_se = col_double(),
  p = col_double(),
  Locus = col_character(),
  phenotype_full = col_character(),
  iteration = col_double(),
  ColocPr = col_double(),
  RegionalPr = col_double(),
  TopSNPFinemapPr = col_double()
)</code></pre>
<pre class="r"><code>head(dat)</code></pre>
<pre><code># A tibble: 6 x 9
  Locus_snp   beta beta_se       p phenotype_full ColocPr RegionalPr
  &lt;chr&gt;      &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;            &lt;dbl&gt;      &lt;dbl&gt;
1 ENSG0000…  0.708  0.110  1.74e-8 MetabolicLabe…   0.838      0.962
2 ENSG0000…  0.580  0.114  3.59e-6 MetabolicLabe…   0.838      0.962
3 ENSG0000… -0.549  0.0880 2.95e-8 H3K27AC;H3K27…   0.377      0.690
4 ENSG0000… -0.538  0.128  7.30e-5 H3K27AC;H3K27…   0.377      0.690
5 ENSG0000… -0.605  0.104  1.75e-7 H3K27AC;H3K27…   0.377      0.690
6 ENSG0000… -0.455  0.121  3.57e-4 H3K4ME3;H3K4M…   0.377      0.690
# … with 2 more variables: TopSNPFinemapPr &lt;dbl&gt;, pheno_class &lt;chr&gt;</code></pre>
<p>Let’s also read in the positions of these molecular phenotypes… later I will look at how effect sizes correlate for differing distances between phenotype pairs…</p>
<pre class="r"><code>Phenoclasses &lt;- dat %&gt;%
  pull(pheno_class) %&gt;% unique()

files &lt;- paste0(&quot;../code/QTLs/QTLTools/&quot;, Phenoclasses, &quot;/OnlyFirstReps.sorted.qqnorm.bed.gz&quot;)

bed.list &lt;- lapply(files, fread, sep=&#39;\t&#39;, select=2:6) %&gt;%
  setNames(Phenoclasses)</code></pre>
<pre><code>Registered S3 method overwritten by &#39;R.oo&#39;:
  method        from       
  throw.default R.methodsS3</code></pre>
<pre class="r"><code>positions &lt;- bind_rows(bed.list, .id=&quot;source&quot;) %&gt;%
  unite(phenotype_full, source, pid, sep = &quot;;&quot;) %&gt;%
  select(-gid)

head(positions)</code></pre>
<pre><code>                               phenotype_full   start     end strand
1: MetabolicLabelled.30min;ENSG00000188976.11  944202  959309      -
2: MetabolicLabelled.30min;ENSG00000187961.14  960583  965719      +
3: MetabolicLabelled.30min;ENSG00000187583.11  966481  975865      +
4: MetabolicLabelled.30min;ENSG00000188290.11  998961 1000172      -
5: MetabolicLabelled.30min;ENSG00000187608.10 1001137 1014540      +
6: MetabolicLabelled.30min;ENSG00000188157.15 1020119 1056118      +</code></pre>
<pre class="r"><code>dat &lt;- dat %&gt;%
  left_join(positions, by=&quot;phenotype_full&quot;)

dat$pheno_class %&gt;% unique()</code></pre>
<pre><code>[1] &quot;MetabolicLabelled.30min&quot;        &quot;MetabolicLabelled.60min&quot;       
[3] &quot;H3K27AC&quot;                        &quot;H3K4ME3&quot;                       
[5] &quot;Expression.Splicing&quot;            &quot;Expression.Splicing.Subset_YRI&quot;
[7] &quot;polyA.Splicing.Subset_YRI&quot;      &quot;chRNA.Expression.Splicing&quot;     </code></pre>
<p>Ok, now lets do some data tidy-ing to match up all pairs of within-cluster colocalized traits and their beta values</p>
<pre class="r"><code>#Let&#39;s recode the gene phenotypes to just be the promoter positions which is more relevent when evaluating how distance effects correlations between phenotype classes
GeneWithPromoterPhenotypes &lt;- c(&quot;Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;, &quot;chRNA.Expression.Splicing&quot;, &quot;MetabolicLabelled.60min&quot;, &quot;MetabolicLabelled.30min&quot;)

dat.tidy.for.cor &lt;- dat %&gt;%
  mutate(start = case_when(
    (pheno_class %in% GeneWithPromoterPhenotypes) &amp; (strand==&quot;-&quot;) ~ as.integer(end-1),
    TRUE ~ start
  )) %&gt;%
  mutate(end = case_when(
    (pheno_class %in% GeneWithPromoterPhenotypes) &amp; (strand==&quot;+&quot;) ~ as.integer(start+1),
    TRUE ~ end
  )) %&gt;%
  mutate(pheno_class = recode(pheno_class,
                                   Expression.Splicing = &quot;Expression (Geuvadis)&quot;,
                                   Expression.Splicing.Subset_YRI = &quot;Expression (YRI.Subset)&quot;,
                                   polyA.Splicing.Subset_YRI = &quot;Splicing (leafcutter.YRI.Subset)&quot;,
                                   chRNA.Expression.Splicing = &quot;Expression (chRNA)&quot;)) %&gt;%
  # filter(Locus_snp == &quot;ENSG00000002822.15_7:1998126:G:A&quot;) %&gt;%
  left_join(., ., by = c(&quot;Locus_snp&quot;, &quot;ColocPr&quot;, &quot;RegionalPr&quot;, &quot;TopSNPFinemapPr&quot;)) %&gt;% 
  filter(phenotype_full.x != phenotype_full.y) %&gt;% 
  rowwise() %&gt;%
  mutate(name = toString(sort(c(phenotype_full.x,phenotype_full.y)))) %&gt;% 
  distinct(Locus_snp, name, .keep_all = T) %&gt;%
  mutate(pheno_class_name = toString(sort(c(pheno_class.x,pheno_class.y)))) %&gt;%
  mutate(Distance_between = case_when(
    # If features overlap
    ((start.x &gt;= start.y) &amp; (start.x &lt;= end.y)) |
    ((end.x &gt;= start.y) &amp; (end.x &lt;= end.y)) |
    ((start.x &lt;= start.y) &amp; (end.x &gt;= end.y)) ~ as.integer(0),
    TRUE ~ min(abs(end.x - start.y), abs(start.x - end.y))
  ))


PhenotypeClassPairs &lt;- dat.tidy.for.cor %&gt;%
  pull(pheno_class_name) %&gt;% unique()
PhenotypeClassPairs</code></pre>
<pre><code> [1] &quot;MetabolicLabelled.30min, MetabolicLabelled.60min&quot;                  
 [2] &quot;H3K27AC, H3K27AC&quot;                                                  
 [3] &quot;H3K27AC, H3K4ME3&quot;                                                  
 [4] &quot;H3K4ME3, H3K4ME3&quot;                                                  
 [5] &quot;Expression (Geuvadis), MetabolicLabelled.30min&quot;                    
 [6] &quot;Expression (Geuvadis), Expression (YRI.Subset)&quot;                    
 [7] &quot;Expression (YRI.Subset), MetabolicLabelled.30min&quot;                  
 [8] &quot;Expression (Geuvadis), H3K27AC&quot;                                    
 [9] &quot;Expression (Geuvadis), H3K4ME3&quot;                                    
[10] &quot;Expression (Geuvadis), Splicing (leafcutter.YRI.Subset)&quot;           
[11] &quot;Expression (YRI.Subset), H3K27AC&quot;                                  
[12] &quot;H3K27AC, Splicing (leafcutter.YRI.Subset)&quot;                         
[13] &quot;Expression (YRI.Subset), H3K4ME3&quot;                                  
[14] &quot;H3K4ME3, Splicing (leafcutter.YRI.Subset)&quot;                         
[15] &quot;Expression (YRI.Subset), Splicing (leafcutter.YRI.Subset)&quot;         
[16] &quot;Splicing (leafcutter.YRI.Subset), Splicing (leafcutter.YRI.Subset)&quot;
[17] &quot;H3K4ME3, MetabolicLabelled.30min&quot;                                  
[18] &quot;MetabolicLabelled.30min, Splicing (leafcutter.YRI.Subset)&quot;         
[19] &quot;Expression (Geuvadis), MetabolicLabelled.60min&quot;                    
[20] &quot;H3K4ME3, MetabolicLabelled.60min&quot;                                  
[21] &quot;H3K27AC, MetabolicLabelled.30min&quot;                                  
[22] &quot;H3K27AC, MetabolicLabelled.60min&quot;                                  
[23] &quot;Expression (YRI.Subset), MetabolicLabelled.60min&quot;                  
[24] &quot;MetabolicLabelled.60min, Splicing (leafcutter.YRI.Subset)&quot;         
[25] &quot;Expression (chRNA), H3K27AC&quot;                                       
[26] &quot;Expression (chRNA), H3K4ME3&quot;                                       
[27] &quot;Expression (chRNA), MetabolicLabelled.30min&quot;                       
[28] &quot;Expression (chRNA), Splicing (leafcutter.YRI.Subset)&quot;              
[29] &quot;Expression (chRNA), Expression (Geuvadis)&quot;                         
[30] &quot;Expression (chRNA), MetabolicLabelled.60min&quot;                       
[31] &quot;Expression (chRNA), Expression (YRI.Subset)&quot;                       </code></pre>
<p>Ok, now let’s plot the correlation of beta values for certain classes of phenotypes (eg Expression QTLs and H3K27Ac QTLs), and more generally for all pairs of phenotype classes</p>
<pre class="r"><code>dat.tidy.for.cor %&gt;%
  filter(pheno_class_name == &quot;Expression (Geuvadis), H3K27AC&quot;) %&gt;%
  ggplot(aes(x=beta.x, y=beta.y)) +
  geom_point() +
  facet_wrap(~pheno_class_name) +
  theme_classic()</code></pre>
<p><img src="figure/20220114_ColocalizationEffectSizeCorrelations.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/0c8ee9c65abd3ee8c00bed55cfafcfd2caa5bdbd/docs/figure/20220114_ColocalizationEffectSizeCorrelations.Rmd/unnamed-chunk-4-1.png" target="_blank">0c8ee9c</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-01-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dat.tidy.for.cor %&gt;%
  filter(pheno_class_name == &quot;MetabolicLabelled.30min, MetabolicLabelled.60min&quot;) %&gt;%
  ggplot(aes(x=beta.x, y=beta.y)) +
  geom_point() +
  facet_wrap(~pheno_class_name) +
  theme_classic()</code></pre>
<p><img src="figure/20220114_ColocalizationEffectSizeCorrelations.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dat.tidy.for.cor %&gt;%
  filter(pheno_class_name == &quot;Splicing (leafcutter.YRI.Subset), Splicing (leafcutter.YRI.Subset)&quot;) %&gt;%
  ggplot(aes(x=beta.x, y=beta.y)) +
  geom_point() +
  facet_wrap(~pheno_class_name) +
  theme_classic()</code></pre>
<p><img src="figure/20220114_ColocalizationEffectSizeCorrelations.Rmd/unnamed-chunk-4-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dat.tidy.for.cor %&gt;%
  ggplot(aes(x=beta.x, y=beta.y)) +
  geom_point(alpha=0.05) +
  facet_wrap(~pheno_class_name) +
  theme_classic()</code></pre>
<p><img src="figure/20220114_ColocalizationEffectSizeCorrelations.Rmd/unnamed-chunk-4-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok wow. That is nice. H3K27Ac beta values are very strongly positively correlated with eQTL betas, as expected. We can look at other pairs of traits too, and if there is an effect, I think this is a very powerful way to assess it. But unsurprisingly, the way we have quantified splicing here with leafcutter, splicing is not really obviously correlated with anything, since a leafcutter intron excision ratio PSI that goes up for one intron necessarily means some other introns went down. Perhaps if we looked at intron retention, the effects would be more interpretable. That will be on the future to do list once we get better chRNA-seq data.</p>
<p>Also, let’s replot these as a clustered heatmap for matrix of correlation coefficients.</p>
<pre class="r"><code>library(RColorBrewer)
# my_palette &lt;- colorRampPalette(c(&quot;blue&quot;, &quot;black&quot;, &quot;yellow&quot;))(n = 1000)

dat.tidy.for.cor %&gt;%
  group_by(pheno_class_name) %&gt;%
  summarise(cor = cor.test(beta.x, beta.y, method=&quot;pearson&quot;)$p.value) %&gt;%
  separate(pheno_class_name, into=c(&quot;PhenotypeClass1&quot;, &quot;PhenotypeClass2&quot;), sep = &quot;, &quot;)</code></pre>
<pre><code># A tibble: 31 x 3
   PhenotypeClass1       PhenotypeClass2                        cor
   &lt;chr&gt;                 &lt;chr&gt;                                &lt;dbl&gt;
 1 Expression (chRNA)    Expression (Geuvadis)            1.27e-  5
 2 Expression (chRNA)    Expression (YRI.Subset)          3.07e-  5
 3 Expression (chRNA)    H3K27AC                          2.18e-  6
 4 Expression (chRNA)    H3K4ME3                          1.05e-  3
 5 Expression (chRNA)    MetabolicLabelled.30min          1.79e-  8
 6 Expression (chRNA)    MetabolicLabelled.60min          3.96e-  7
 7 Expression (chRNA)    Splicing (leafcutter.YRI.Subset) 6.29e-  1
 8 Expression (Geuvadis) Expression (YRI.Subset)          0.       
 9 Expression (Geuvadis) H3K27AC                          0.       
10 Expression (Geuvadis) H3K4ME3                          2.47e-177
# … with 21 more rows</code></pre>
<pre class="r"><code>dat.cor &lt;- dat.tidy.for.cor %&gt;%
  group_by(pheno_class_name) %&gt;%
  summarise(cor = cor(beta.x, beta.y, method=&quot;spearman&quot;)) %&gt;%
  separate(pheno_class_name, into=c(&quot;PhenotypeClass1&quot;, &quot;PhenotypeClass2&quot;), sep = &quot;, &quot;)
  
dat.cor %&gt;%
  rename(PhenotypeClass1=PhenotypeClass2, PhenotypeClass2=PhenotypeClass1) %&gt;%
  bind_rows(dat.cor) %&gt;%
  distinct(.keep_all=T) %&gt;%
  # mutate(cor = as.numeric(cor)) %&gt;%
  pivot_wider(names_from = &quot;PhenotypeClass1&quot;, values_from = &quot;cor&quot;, values_fill=NA, names_sort=T) %&gt;%
  column_to_rownames(&quot;PhenotypeClass2&quot;) %&gt;%
  as.matrix() %&gt;%
  # heatmap.2(trace=&quot;none&quot;, col=brewer.pal(51,&quot;Spectral&quot;))
  heatmap.2(trace=&quot;none&quot;, col=colorpanel(75, &quot;blue&quot;, &quot;black&quot;, &quot;yellow&quot;), cexRow = 1, cexCol = 1, offsetRow=0, offsetCol = 0, margins=c(11,11), dendrogram = &quot;column&quot;)</code></pre>
<p><img src="figure/20220114_ColocalizationEffectSizeCorrelations.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The slight negative correlation between leafcutter splicing and other phenotypes is interesting. I wonder if there is a technical explanation. For example, perhaps many of the colocalizing sQTLs are actually alt TSS, such that the measured decrease in expression if a downstream TSS is used (shorter isoform), results in general upregulation of within-cluster introns which would occur if the cluster had a few introns only present in the long isoform that go down, and the other introns necessarily go up.</p>
<p>Another informative thing would be to look at the distribution of distances between colocalized traits, and the effect sizes between colocalized traits that are close versus far. First let’s plot the distribution of distances for each type of pair of colocalizes traits:</p>
<pre class="r"><code>dat.tidy.for.cor %&gt;%
  #Add a psueodcount so we can still plot things that are distance=0 overlapping
  mutate(Distance_between = Distance_between+1) %&gt;%
  ggplot(aes(x=Distance_between)) +
  stat_ecdf() +
  scale_x_continuous(trans=&quot;log10&quot;) +
  facet_wrap(~pheno_class_name) +
  theme_classic()</code></pre>
<p><img src="figure/20220114_ColocalizationEffectSizeCorrelations.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok now let’s look at some examples. For example, looking at splicing and H3K4me3 traits that colocalizes, if we look at the ones that are close together (eg &lt;1000bp), do we see any positive correlation like what might be expected with EMATs?</p>
<pre class="r"><code>dat.tidy.for.cor %&gt;%
  filter(pheno_class_name == &quot;H3K4ME3, Splicing (leafcutter.YRI.Subset)&quot;) %&gt;%
  mutate(Distance_between = Distance_between+1) %&gt;%
  ggplot(aes(x=Distance_between)) +
  stat_ecdf() +
  scale_x_continuous(trans=&quot;log10&quot;) +
  facet_wrap(~pheno_class_name) +
  theme_classic()</code></pre>
<p><img src="figure/20220114_ColocalizationEffectSizeCorrelations.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dat.tidy.for.cor %&gt;%
  filter(pheno_class_name == &quot;H3K4ME3, Splicing (leafcutter.YRI.Subset)&quot;) %&gt;%
  mutate(DistanceClass = cut(Distance_between, breaks = c(0, 1, 500, 1000, 5000, 10000, Inf), include.lowest = T, right=T)) %&gt;%
  mutate(DistanceClass = recode(DistanceClass, `[0,1]` = &quot;Overlapping&quot;)) %&gt;%
  ggplot(aes(x=beta.x, y=beta.y)) +
  geom_point() +
  # geom_smooth(method=&#39;lm&#39;) +
  facet_wrap(~DistanceClass) +
  theme_classic()</code></pre>
<p><img src="figure/20220114_ColocalizationEffectSizeCorrelations.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Not so much. Again, maybe we need to look at intron retention instead of leafcutter splicing.</p>
<p>Let’s explore how leafcutter splicing phenotypes are correlated with the same plots…</p>
<pre class="r"><code>dat.tidy.for.cor %&gt;%
  filter(pheno_class_name == &quot;Splicing (leafcutter.YRI.Subset), Splicing (leafcutter.YRI.Subset)&quot;) %&gt;%
  mutate(Distance_between = Distance_between+1) %&gt;%
  ggplot(aes(x=Distance_between)) +
  stat_ecdf() +
  ylab(&quot;Cumulative fraction&quot;) +
  xlab(&quot;Distance b/n features&quot;) +
  scale_x_continuous(trans=&quot;log10&quot;) +
  facet_wrap(~pheno_class_name) +
  theme_classic()</code></pre>
<p><img src="figure/20220114_ColocalizationEffectSizeCorrelations.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dat.tidy.for.cor %&gt;%
  filter(pheno_class_name == &quot;Splicing (leafcutter.YRI.Subset), Splicing (leafcutter.YRI.Subset)&quot;) %&gt;%
  mutate(DistanceClass = cut(Distance_between, breaks = c(0, 1, 500, 1000, 5000, 10000, Inf), include.lowest = T, right=T)) %&gt;%
  mutate(DistanceClass = recode(DistanceClass, `[0,1]` = &quot;Overlapping&quot;)) %&gt;%
  ggplot(aes(x=beta.x, y=beta.y)) +
  geom_point(alpha=0.3) +
  # geom_smooth(method=&#39;lm&#39;) +
  facet_wrap(~DistanceClass) +
  theme_classic()</code></pre>
<p><img src="figure/20220114_ColocalizationEffectSizeCorrelations.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok that actually sort of makes sense… Within a single molecule, overlapping introns are mutually exclusive and we see that they are anticorrelated. Also, for introns that are 1-500 bp apart (the size of typical exons), effect sizes are actually positively correlated, which might reflect QTLs for cassette exons.</p>
<p>Lastly, let’s check H3K27AC mark and gene expression. And also H3K27AC vs H3K4ME3</p>
<pre class="r"><code>dat.tidy.for.cor %&gt;%
  filter(pheno_class_name == &quot;Expression (Geuvadis), H3K27AC&quot;) %&gt;%
  mutate(Distance_between = Distance_between+1) %&gt;%
  ggplot(aes(x=Distance_between)) +
  stat_ecdf() +
  ylab(&quot;Cumulative fraction&quot;) +
  xlab(&quot;Distance b/n features&quot;) +
  scale_x_continuous(trans=&quot;log10&quot;) +
  facet_wrap(~pheno_class_name) +
  theme_classic()</code></pre>
<p><img src="figure/20220114_ColocalizationEffectSizeCorrelations.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dat.tidy.for.cor %&gt;%
  filter(pheno_class_name == &quot;Expression (Geuvadis), H3K27AC&quot;) %&gt;%
  mutate(DistanceClass = cut(Distance_between, breaks = c(0, 1000, 5000, 10000, 50000, 100000, Inf), include.lowest = T, right=T)) %&gt;%
  mutate(DistanceClass = recode(DistanceClass, `[0,1]` = &quot;Overlapping&quot;)) %&gt;%
  ggplot(aes(x=beta.x, y=beta.y)) +
  geom_point(alpha=0.3) +
  # geom_smooth(method=&#39;lm&#39;) +
  facet_wrap(~DistanceClass) +
  theme_classic()</code></pre>
<p><img src="figure/20220114_ColocalizationEffectSizeCorrelations.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dat.tidy.for.cor %&gt;%
  filter(pheno_class_name == &quot;H3K27AC, H3K4ME3&quot;) %&gt;%
  mutate(Distance_between = Distance_between+1) %&gt;%
  ggplot(aes(x=Distance_between)) +
  stat_ecdf() +
  ylab(&quot;Cumulative fraction&quot;) +
  xlab(&quot;Distance b/n features&quot;) +
  scale_x_continuous(trans=&quot;log10&quot;) +
  facet_wrap(~pheno_class_name) +
  theme_classic()</code></pre>
<p><img src="figure/20220114_ColocalizationEffectSizeCorrelations.Rmd/unnamed-chunk-9-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dat.tidy.for.cor %&gt;%
  filter(pheno_class_name == &quot;H3K27AC, H3K4ME3&quot;) %&gt;%
  mutate(DistanceClass = cut(Distance_between, breaks = c(0, 1000, 5000, 10000, 50000, 100000, Inf), include.lowest = T, right=T)) %&gt;%
  mutate(DistanceClass = recode(DistanceClass, `[0,1]` = &quot;Overlapping&quot;)) %&gt;%
  ggplot(aes(x=beta.x, y=beta.y)) +
  geom_point(alpha=0.3) +
  # geom_smooth(method=&#39;lm&#39;) +
  facet_wrap(~DistanceClass) +
  theme_classic()</code></pre>
<p><img src="figure/20220114_ColocalizationEffectSizeCorrelations.Rmd/unnamed-chunk-9-4.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] RColorBrewer_1.1-2 data.table_1.13.2  gplots_3.0.1.1    
 [4] forcats_0.4.0      stringr_1.4.0      dplyr_0.8.3       
 [7] purrr_0.3.4        readr_1.3.1        tidyr_1.1.0       
[10] tibble_2.1.3       ggplot2_3.3.3      tidyverse_1.3.0   

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5         lubridate_1.7.4    lattice_0.20-38   
 [4] gtools_3.8.1       utf8_1.1.4         assertthat_0.2.1  
 [7] rprojroot_2.0.2    digest_0.6.20      R6_2.4.0          
[10] cellranger_1.1.0   backports_1.1.4    reprex_0.3.0      
[13] evaluate_0.14      httr_1.4.1         pillar_1.4.2      
[16] rlang_0.4.10       readxl_1.3.1       rstudioapi_0.10   
[19] gdata_2.18.0       whisker_0.3-2      R.oo_1.22.0       
[22] R.utils_2.9.0      rmarkdown_1.13     labeling_0.3      
[25] munsell_0.5.0      broom_0.5.2        compiler_3.6.1    
[28] httpuv_1.5.1       modelr_0.1.8       xfun_0.8          
[31] pkgconfig_2.0.2    htmltools_0.3.6    tidyselect_1.1.0  
[34] workflowr_1.6.2    fansi_0.4.0        crayon_1.3.4      
[37] dbplyr_1.4.2       withr_2.4.1        later_0.8.0       
[40] R.methodsS3_1.7.1  bitops_1.0-6       grid_3.6.1        
[43] nlme_3.1-140       jsonlite_1.6       gtable_0.3.0      
[46] lifecycle_0.1.0    DBI_1.1.0          git2r_0.26.1      
[49] magrittr_1.5       scales_1.1.0       KernSmooth_2.23-15
[52] cli_2.2.0          stringi_1.4.3      farver_2.1.0      
[55] fs_1.3.1           promises_1.0.1     xml2_1.3.2        
[58] ellipsis_0.2.0.1   generics_0.0.2     vctrs_0.3.1       
[61] tools_3.6.1        glue_1.3.1         hms_0.5.3         
[64] yaml_2.2.0         colorspace_1.4-1   caTools_1.17.1.2  
[67] rvest_0.3.5        knitr_1.23         haven_2.3.1       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
