<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>20221101_PlotStuffForYang</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">20221101_PlotStuffForYang</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-11-04
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>ChromatinSplicingQTLs/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstree293329bfe63df3f76fbddaf06b596049d54590b8targetblank293329ba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/293329bfe63df3f76fbddaf06b596049d54590b8" target="_blank">293329b</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstree293329bfe63df3f76fbddaf06b596049d54590b8targetblank293329ba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/293329bfe63df3f76fbddaf06b596049d54590b8" target="_blank">293329b</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/figure/
    Ignored:    code/.DS_Store
    Ignored:    code/.RData
    Ignored:    code/.Snakefile.swp
    Ignored:    code/._.DS_Store
    Ignored:    code/._README.md
    Ignored:    code/._report.html
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    code/.snakemake/
    Ignored:    code/APA_Processing/
    Ignored:    code/Alignments/
    Ignored:    code/ChromHMM/
    Ignored:    code/ENCODE/
    Ignored:    code/ExpressionAnalysis/
    Ignored:    code/FastqFastp/
    Ignored:    code/FastqFastpSE/
    Ignored:    code/Genotypes/
    Ignored:    code/H3K36me3_CutAndTag.pdf
    Ignored:    code/IntronSlopes/
    Ignored:    code/Metaplots/
    Ignored:    code/Misc/
    Ignored:    code/MiscCountTables/
    Ignored:    code/Multiqc/
    Ignored:    code/Multiqc_chRNA/
    Ignored:    code/NonCodingRNA/
    Ignored:    code/NonCodingRNA_annotation/
    Ignored:    code/PeakCalling/
    Ignored:    code/Phenotypes/
    Ignored:    code/PlotGruberQTLs/
    Ignored:    code/PlotQTLs/
    Ignored:    code/ProCapAnalysis/
    Ignored:    code/QC/
    Ignored:    code/QTL_SNP_Enrichment/
    Ignored:    code/QTLs/
    Ignored:    code/RPKM_tables/
    Ignored:    code/ReferenceGenome/
    Ignored:    code/Rplots.pdf
    Ignored:    code/Session.vim
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TODO
    Ignored:    code/Tehranchi/
    Ignored:    code/bigwigs/
    Ignored:    code/bigwigs_FromNonWASPFilteredReads/
    Ignored:    code/config/.DS_Store
    Ignored:    code/config/._.DS_Store
    Ignored:    code/config/.ipynb_checkpoints/
    Ignored:    code/dag.pdf
    Ignored:    code/dag.png
    Ignored:    code/dag.svg
    Ignored:    code/debug.ipynb
    Ignored:    code/debug_python.ipynb
    Ignored:    code/deepTools/
    Ignored:    code/featureCounts/
    Ignored:    code/gwas_summary_stats/
    Ignored:    code/hyprcoloc/
    Ignored:    code/igv_session.xml
    Ignored:    code/log
    Ignored:    code/logs/
    Ignored:    code/notebooks/.ipynb_checkpoints/
    Ignored:    code/pi1/
    Ignored:    code/rules/.ipynb_checkpoints/
    Ignored:    code/rules/OldRules/
    Ignored:    code/rules/notebooks/
    Ignored:    code/scratch/
    Ignored:    code/scripts/.ipynb_checkpoints/
    Ignored:    code/scripts/GTFtools_0.8.0/
    Ignored:    code/scripts/__pycache__/
    Ignored:    code/scripts/liftOverBedpe/liftOverBedpe.py
    Ignored:    code/snakemake.dryrun.log
    Ignored:    code/snakemake.log
    Ignored:    code/snakemake.sbatch.log
    Ignored:    code/test.introns.bed
    Ignored:    code/test.introns2.bed
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/._20220414203249_JASPAR2022_combined_matrices_25818_jaspar.txt
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-10.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-11.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-2.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-3.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-4.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-5.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-6.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-7.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-8.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022.csv

Untracked files:
    Untracked:  analysis/20221102_PlotStuffForYang.Rmd
    Untracked:  code/snakemake_profiles/slurm/__pycache__/

Unstaged changes:
    Modified:   analysis/20221012_IntronRetentionAndExpressionConcordance.Rmd
    Modified:   code/scripts/GenometracksByGenotype

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="intro" class="section level2">
<h2>Intro</h2>
<p>Yang asked for some specific plots to put in a grant or presentation or something. Here I will make those plots, possibly explore the data a little bit if something comes to mind. A lot of these plots were originally explored in <a href="20221012_IntronRetentionAndExpressionConcordance.html">this notebook</a> but that notebook is getting unweildy and disorganized and overall a pain when I want to go back and re-run parts of it so I am re-plotting the highlights that Yang asked for here.</p>
</div>
<div id="analysis" class="section level2">
<h2>Analysis</h2>
<p>Load libraries and read in a bunch of files…</p>
<pre class="r"><code>library(tidyverse)
library(data.table)
library(MASS)
library(RColorBrewer)
library(scales)


NMD.transcript.introns &lt;- fread(&quot;../code/SplicingAnalysis/Annotations/NMD/NMD_trancsript_introns.bed.gz&quot;, col.names=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;)) %&gt;%
  mutate(stop=stop+1) %&gt;%
  unite(intron, chrom:stop, strand)

Non.NMD.transcript.introns &lt;- fread(&quot;../code/SplicingAnalysis/Annotations/NMD/NonNMD_trancsript_introns.bed.gz&quot;, col.names=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;)) %&gt;%
  mutate(stop=stop+1) %&gt;%
  unite(intron, chrom:stop, strand)


NMD.specific.introns &lt;- setdiff(NMD.transcript.introns$intron, Non.NMD.transcript.introns$intron)

Intron.Annotations.basic &lt;- fread(&quot;../code/SplicingAnalysis/regtools_annotate_combined/basic.bed.gz&quot;) %&gt;%
  filter(known_junction ==1) %&gt;%
  unite(intron, chrom, start, end, strand)
Introns.Annotations.comprehensive &lt;- fread(&quot;../code/SplicingAnalysis/regtools_annotate_combined/comprehensive.bed.gz&quot;) %&gt;%
  filter(known_junction ==1) %&gt;%
  unite(intron, chrom, start, end, strand)
Introns.Annotations.all &lt;- fread(&quot;../code/SplicingAnalysis/regtools_annotate_combined/comprehensive.bed.gz&quot;) %&gt;%
  unite(intron, chrom, start, end, strand)

PhenotypeAliases &lt;- read_tsv(&quot;../data/Phenotypes_recode_for_Plotting.txt&quot;)

PC.ShortAliases &lt;- PhenotypeAliases %&gt;%
  dplyr::select(PC, ShorterAlias) %&gt;% deframe()

PeaksToTSS &lt;- Sys.glob(&quot;../code/Misc/PeaksClosestToTSS/*_assigned.tsv.gz&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/Misc/PeaksClosestToTSS/(.+?)_assigned.tsv.gz&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(read_tsv) %&gt;%
  bind_rows(.id=&quot;ChromatinMark&quot;) %&gt;%
  mutate(GenePeakPair = paste(gene, peak, sep = &quot;;&quot;)) %&gt;%
  distinct(ChromatinMark, peak, gene, .keep_all=T)

TopSNPEffects.ByPairs &lt;- fread(&quot;../code/pi1/PairwisePi1Traits.P.all.txt.gz&quot;)


coloc.tidy &lt;- fread(&quot;../output/hyprcoloc_results/ForColoc/MolColocStandard/hyprcoloc.results.OnlyColocalized.Stats.txt.gz&quot;) %&gt;%
  separate(phenotype_full, into=c(&quot;PC&quot;, &quot;P&quot;), sep=&quot;;&quot;)

coloc.tidy.pairwise &lt;- left_join(
  coloc.tidy,
  coloc.tidy %&gt;%
    dplyr::select(-iteration, -ColocPr, -RegionalPr, -TopSNPFinemapPr),
  by=c(&quot;Locus&quot;),
  suffix=c(&quot;1&quot;, &quot;2&quot;)
) %&gt;%
  filter(!(P1==P2 &amp; PC1 == PC2)) %&gt;%
  filter(snp1 == snp2) %&gt;%
  dplyr::select(ColocalizedTopSNP = snp1, GeneLocus=Locus, everything(), -snp2) %&gt;%
  unite(TraitPair, P1, PC1, P2, PC2, remove=F)</code></pre>
<p>Now let’s start to plot some stuff:</p>
<div id="sqtl-betas-vs-eqtl-beta" class="section level3">
<h3>sQTL betas vs eQTL beta</h3>
<pre class="r"><code>coloc.tidy.pairwise %&gt;%
  filter(
      (PC1 %in% c(&quot;Expression.Splicing.Subset_YRI&quot;)) &amp;
      (PC2 %in% c(&quot;polyA.Splicing&quot;))
  ) %&gt;%
  ggplot(aes(y=beta1, x=beta2)) +
  geom_point(alpha=0.5, color=&#39;black&#39;) +
  geom_vline(xintercept=0) +
  geom_hline(yintercept=0) +
  geom_text(
  data = . %&gt;%
    summarise(cor=cor.test(beta1,beta2)[[&quot;estimate&quot;]], pval=cor.test(beta1,beta2)[[&quot;p.value&quot;]]) %&gt;%
    mutate(R = signif(cor, 3), P=format.pval(pval, 3)) %&gt;%
    mutate(label = str_glue(&quot;R:{R}\nP:{P}&quot;)),
  aes(x=-Inf, y=-Inf, label=label),
  hjust=-0.1, vjust=-0.1, color=&#39;black&#39;
  ) +
  theme_bw() +
  labs(title=&quot;All colocalized eQTLs and sQTLs&quot;, x=&quot;sQTL beta\n(standardized units)&quot;, y=&quot;eQTL beta\n(standardized units)&quot;)</code></pre>
<p><img src="figure/20221102_PlotStuffForYang.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="same-plot-but-grouped-by-intron-type" class="section level3">
<h3>Same plot but grouped by intron type</h3>
<pre class="r"><code>P1Category &lt;- &quot;polyA.Splicing&quot;


sQTL.eQTL.dat.ColocalizedAndUncolocalized &lt;- TopSNPEffects.ByPairs %&gt;%
  filter(
      (PC1 %in% c(P1Category)) &amp;
      (PC2 %in% c(&quot;Expression.Splicing.Subset_YRI&quot;)) &amp;
      (FDR.x &lt; 0.1)) %&gt;%
  mutate(intron = str_replace(P1, &quot;^(.+):(.+?):(.+?):clu.+?_([+-])$&quot;, &quot;chr\\1_\\2_\\3_\\4&quot;)) %&gt;%
  mutate(IntronAnnotation = case_when(
    intron %in% NMD.specific.introns ~ &quot;Annotated NMD&quot;,
    intron %in% Intron.Annotations.basic$intron ~ &quot;Annotated basic&quot;,
    intron %in% Introns.Annotations.comprehensive$intron ~ &quot;Annotated Not basic&quot;,
    TRUE ~ &quot;Unannotated&quot;
  )) %&gt;%
  dplyr::select(beta1=beta.x, beta2=trait.x.beta.in.y, P1, PC1, P2, PC2, IntronAnnotation) %&gt;%
  unite(TraitPair, P1, PC1, P2, PC2, remove=F) %&gt;%
  filter(!TraitPair %in% coloc.tidy.pairwise$TraitPair) %&gt;%
  mutate(Colocalized = &quot;Not colocalized&quot;) %&gt;%
  bind_rows(
    coloc.tidy.pairwise %&gt;%
      filter(
          (PC2 %in% c(&quot;Expression.Splicing.Subset_YRI&quot;)) &amp;
          (PC1 %in% c(P1Category))
      ) %&gt;%
      mutate(intron = str_replace(P1, &quot;^(.+):(.+?):(.+?):clu.+?_([+-])$&quot;, &quot;chr\\1_\\2_\\3_\\4&quot;)) %&gt;%
      mutate(IntronAnnotation = case_when(
        intron %in% NMD.specific.introns ~ &quot;Annotated NMD&quot;,
        intron %in% Intron.Annotations.basic$intron ~ &quot;Annotated basic&quot;,
        intron %in% Introns.Annotations.comprehensive$intron ~ &quot;Annotated Not basic&quot;,
        TRUE ~ &quot;Unannotated&quot;
      )) %&gt;%
      dplyr::select(beta1, beta2, P1, PC1, P2, PC2, IntronAnnotation) %&gt;%
      mutate(Colocalized = &quot;Colocalized&quot;)
  ) %&gt;%
  mutate(Is.Concordant = sign(beta1)==sign(beta2)) %&gt;%
  mutate(ConsistentWithNMD = case_when(
    !IntronAnnotation == &quot;Annotated basic&quot; &amp; Is.Concordant ~ &quot;Inconsistent with NMD&quot;,
    !IntronAnnotation == &quot;Annotated basic&quot; &amp; !Is.Concordant ~ &quot;Consistent with NMD&quot;,
    IntronAnnotation == &quot;Annotated basic&quot; ~ NA_character_
  ))

P.sQTL.eQTLs &lt;- ggplot(sQTL.eQTL.dat.ColocalizedAndUncolocalized, aes(x=beta1, y=beta2, color=IntronAnnotation)) +
  stat_ellipse(
    data = mvrnorm(1000, mu=c(0,0), matrix(c(1,0.8,0.8,1),2,2)) %&gt;%
      as.data.frame() %&gt;%
      dplyr::select(beta1=V1, beta2=V2),
    type = &quot;norm&quot;, linetype = 2, color=&quot;green&quot;, alpha=0.3) +
  stat_ellipse(
    data = mvrnorm(1000, mu=c(0,0), matrix(c(1,-0.8,-0.8,1),2,2)) %&gt;%
      as.data.frame() %&gt;%
      dplyr::select(beta1=V1, beta2=V2),
    type = &quot;norm&quot;, linetype = 2, color=&quot;red&quot;, alpha=0.3) +
  geom_point(alpha=0.1) +
  geom_vline(xintercept=0) +
  geom_hline(yintercept=0) +
  # geom_smooth(method=&#39;lm&#39;) +
  geom_text(
  data = . %&gt;%
    group_by(IntronAnnotation, Colocalized) %&gt;%
    summarise(cor=cor.test(beta1,beta2)[[&quot;estimate&quot;]], pval=cor.test(beta1,beta2)[[&quot;p.value&quot;]]) %&gt;%
    mutate(R = signif(cor, 3), P=format.pval(pval, 3)) %&gt;%
    mutate(label = str_glue(&quot;R:{R}\nP:{P}&quot;)),
  aes(x=-Inf, y=-Inf, label=label),
  hjust=-.1, vjust=-0.1, color=&#39;black&#39;, size=3
  ) +
  geom_text(
  data = . %&gt;%
    group_by(IntronAnnotation, ConsistentWithNMD, Colocalized) %&gt;%
    distinct(P2, .keep_all=T) %&gt;%
    ungroup() %&gt;%
    count(IntronAnnotation, ConsistentWithNMD, Colocalized) %&gt;%
    filter(ConsistentWithNMD==&quot;Consistent with NMD&quot;),
  aes(x=-Inf, y=Inf, label=n),
  hjust=-0.2, vjust=1.5, color=&#39;red&#39;
  ) +
  geom_text(
  data = . %&gt;%
    group_by(IntronAnnotation, ConsistentWithNMD, Colocalized) %&gt;%
    distinct(P2, .keep_all=T) %&gt;%
    ungroup() %&gt;%
    count(IntronAnnotation, ConsistentWithNMD, Colocalized) %&gt;%
    filter(ConsistentWithNMD==&quot;Inconsistent with NMD&quot;),
  aes(x=Inf, y=Inf, label=n),
  hjust=1.2, vjust=1.5, color=&#39;green&#39;
  ) +
  facet_grid(Colocalized ~ IntronAnnotation) +
  theme_bw() +
  labs(title=&quot;genetic effects of NMD introns and host gene expression&quot;, x=&quot;chRNA sQTL beta\n(standardized units)&quot;, y=&quot;eQTL beta\n(standardized units)&quot;, caption=&quot;Colocalized sQTL/eQTLs betas at top co-finemapped SNP\nUncolocalized betas relative to top chRNA sQTL SNP&quot;) +
  guides(colour = guide_legend(override.aes = list(alpha = 1)))

P.sQTL.eQTLs</code></pre>
<p><img src="figure/20221102_PlotStuffForYang.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>If you sum up all the red and the green counts in the colocalized sub-plots, and then only take distinct genes, you get the following:</p>
<pre class="r"><code>CountsOfNMD.Consitent.eGene.sQTL.Effects &lt;- sQTL.eQTL.dat.ColocalizedAndUncolocalized %&gt;%
  filter(Colocalized == &quot;Colocalized&quot;) %&gt;%
  distinct(P2, .keep_all=T) %&gt;%
  count(ConsistentWithNMD) %&gt;%
  drop_na()
  
CountsOfNMD.Consitent.eGene.sQTL.Effects</code></pre>
<pre><code>       ConsistentWithNMD   n
1:   Consistent with NMD 373
2: Inconsistent with NMD  49</code></pre>
<pre class="r"><code>(CountsOfNMD.Consitent.eGene.sQTL.Effects$n[1] - CountsOfNMD.Consitent.eGene.sQTL.Effects$n[2])/3970</code></pre>
<pre><code>[1] 0.08161209</code></pre>
<p>For reference, let’s make the same plots and do the same calculation for H3K27AC, H3K4ME3, and H3K4ME1:</p>
<pre class="r"><code>P1Category &lt;- c(&quot;H3K4ME3&quot;, &quot;H3K4ME1&quot;, &quot;H3K27AC&quot;)


hQTL.eQTL.dat.ColocalizedAndUncolocalized &lt;- TopSNPEffects.ByPairs %&gt;%
  filter(
      (PC1 %in% c(P1Category)) &amp;
      (PC2 %in% c(&quot;Expression.Splicing.Subset_YRI&quot;)) &amp;
      (FDR.x &lt; 0.1)) %&gt;%
  dplyr::select(beta1=beta.x, beta2=trait.x.beta.in.y, P1, PC1, P2, PC2) %&gt;%
  unite(TraitPair, P1, PC1, P2, PC2, remove=F) %&gt;%
  filter(!TraitPair %in% coloc.tidy.pairwise$TraitPair) %&gt;%
  mutate(Colocalized = &quot;Not colocalized&quot;) %&gt;%
  bind_rows(
    coloc.tidy.pairwise %&gt;%
      filter(
          (PC2 %in% c(&quot;Expression.Splicing.Subset_YRI&quot;)) &amp;
          (PC1 %in% c(P1Category))
      ) %&gt;%
      dplyr::select(beta1, beta2, P1, PC1, P2, PC1) %&gt;%
      mutate(Colocalized = &quot;Colocalized&quot;)
  ) %&gt;%
  mutate(Is.Concordant = sign(beta1)==sign(beta2)) %&gt;%
  mutate(ConsistentWithNMD = if_else(Is.Concordant, &quot;Consistent with enhancer/promoter activation&quot;, &quot;Inconsistent with enhancer/promoter activation&quot;))

P.hQTL.eQTLs &lt;- ggplot(hQTL.eQTL.dat.ColocalizedAndUncolocalized, aes(x=beta1, y=beta2, color=PC1)) +
  stat_ellipse(
    data = mvrnorm(1000, mu=c(0,0), matrix(c(1,0.8,0.8,1),2,2)) %&gt;%
      as.data.frame() %&gt;%
      dplyr::select(beta1=V1, beta2=V2),
    type = &quot;norm&quot;, linetype = 2, color=&quot;green&quot;, alpha=0.3) +
  stat_ellipse(
    data = mvrnorm(1000, mu=c(0,0), matrix(c(1,-0.8,-0.8,1),2,2)) %&gt;%
      as.data.frame() %&gt;%
      dplyr::select(beta1=V1, beta2=V2),
    type = &quot;norm&quot;, linetype = 2, color=&quot;red&quot;, alpha=0.3) +
  geom_point(alpha=0.05) +
  geom_vline(xintercept=0) +
  geom_hline(yintercept=0) +
  # geom_smooth(method=&#39;lm&#39;) +
  geom_text(
  data = . %&gt;%
    group_by(PC1, Colocalized) %&gt;%
    summarise(cor=cor.test(beta1,beta2)[[&quot;estimate&quot;]], pval=cor.test(beta1,beta2)[[&quot;p.value&quot;]]) %&gt;%
    mutate(R = signif(cor, 3), P=format.pval(pval, 3)) %&gt;%
    mutate(label = str_glue(&quot;R:{R}\nP:{P}&quot;)),
  aes(x=-Inf, y=-Inf, label=label),
  hjust=-.1, vjust=-0.1, color=&#39;black&#39;, size=3
  ) +
  geom_text(
  data = . %&gt;%
    group_by(PC1, ConsistentWithNMD, Colocalized) %&gt;%
    distinct(P2, .keep_all=T) %&gt;%
    ungroup() %&gt;%
    count(PC1, ConsistentWithNMD, Colocalized) %&gt;%
    filter(ConsistentWithNMD==&quot;Consistent with enhancer/promoter activation&quot;),
  aes(x=Inf, y=Inf, label=n),
  hjust=1.2, vjust=1.5, color=&#39;green&#39;
  ) +
  geom_text(
  data = . %&gt;%
    group_by(PC1, ConsistentWithNMD, Colocalized) %&gt;%
    distinct(P2, .keep_all=T) %&gt;%
    ungroup() %&gt;%
    count(PC1, ConsistentWithNMD, Colocalized) %&gt;%
    filter(ConsistentWithNMD==&quot;Inconsistent with enhancer/promoter activation&quot;),
  aes(x=-Inf, y=Inf, label=n),
  hjust=-0.2, vjust=1.5, color=&#39;red&#39;
  ) +
  facet_grid(Colocalized ~ PC1) +
  theme_bw() +
  labs(title=&quot;genetic effects of activating chromatin marks on cis gene expression&quot;, x=&quot;xQTL beta\n(standardized units)&quot;, y=&quot;eQTL beta\n(standardized units)&quot;, caption=&quot;Colocalized xQTL/eQTLs betas at top co-finemapped SNP\nUncolocalized betas relative to top xQTL SNP&quot;, color=&quot;xQTL&quot;) +
  guides(colour = guide_legend(override.aes = list(alpha = 1)))
P.hQTL.eQTLs</code></pre>
<p><img src="figure/20221102_PlotStuffForYang.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>CountsOfNMD.Consitent.eGene.hQTL.Effects &lt;- hQTL.eQTL.dat.ColocalizedAndUncolocalized %&gt;%
  filter(Colocalized == &quot;Colocalized&quot;) %&gt;%
  distinct(P2, .keep_all=T) %&gt;%
  count(ConsistentWithNMD) %&gt;%
  dplyr::select(ConsistentWithPromoterEnhancerActivation=ConsistentWithNMD,n) %&gt;%
  drop_na()
  
CountsOfNMD.Consitent.eGene.hQTL.Effects</code></pre>
<pre><code>         ConsistentWithPromoterEnhancerActivation   n
1:   Consistent with enhancer/promoter activation 658
2: Inconsistent with enhancer/promoter activation 102</code></pre>
<pre class="r"><code>(CountsOfNMD.Consitent.eGene.hQTL.Effects$n[1] - CountsOfNMD.Consitent.eGene.hQTL.Effects$n[2])/3970</code></pre>
<pre><code>[1] 0.1400504</code></pre>
<pre class="r"><code>ggsave(&quot;../code/scratch/eQTL.sQTLs.Betas.pdf&quot;, P.sQTL.eQTLs, width=8)
ggsave(&quot;../code/scratch/eQTL.hQTLs.Betas.pdf&quot;, P.hQTL.eQTLs, width=8)</code></pre>
<p>I wonder if this plot might be more informative (in terms of counting the non-colocalized effects, and also not seeing a trend) if i broke up the non-colocalized facets into those with nominal eQTL P &lt; 0.01 and those &gt; 0.01</p>
<pre class="r"><code>eQTL.P.Threshold &lt;- 0.1
P1Category &lt;- &quot;polyA.Splicing&quot;


sQTL.eQTL.dat.ColocalizedAndUncolocalized &lt;- TopSNPEffects.ByPairs %&gt;%
  filter(
      (PC1 %in% c(P1Category)) &amp;
      (PC2 %in% c(&quot;Expression.Splicing.Subset_YRI&quot;)) &amp;
      (FDR.x &lt; 0.1)) %&gt;%
  mutate(intron = str_replace(P1, &quot;^(.+):(.+?):(.+?):clu.+?_([+-])$&quot;, &quot;chr\\1_\\2_\\3_\\4&quot;)) %&gt;%
  mutate(IntronAnnotation = case_when(
    intron %in% NMD.specific.introns ~ &quot;Annotated NMD&quot;,
    intron %in% Intron.Annotations.basic$intron ~ &quot;Annotated basic&quot;,
    intron %in% Introns.Annotations.comprehensive$intron ~ &quot;Annotated Not basic&quot;,
    TRUE ~ &quot;Unannotated&quot;
  )) %&gt;%
  dplyr::select(beta1=beta.x, beta2=trait.x.beta.in.y, P1, PC1, P2, PC2, IntronAnnotation, eQTL.P = trait.x.p.in.y) %&gt;%
  unite(TraitPair, P1, PC1, P2, PC2, remove=F) %&gt;%
  filter(!TraitPair %in% coloc.tidy.pairwise$TraitPair) %&gt;%
  mutate(Colocalized = if_else(
    eQTL.P &lt; 0.05,
    paste0(&quot;Not colocalized, eQTL P&lt;&quot;, eQTL.P.Threshold),
    paste0(&quot;Not colocalized, eQTL P&gt;&quot;, eQTL.P.Threshold)
  )) %&gt;%
  bind_rows(
    coloc.tidy.pairwise %&gt;%
      filter(
          (PC2 %in% c(&quot;Expression.Splicing.Subset_YRI&quot;)) &amp;
          (PC1 %in% c(P1Category))
      ) %&gt;%
      mutate(intron = str_replace(P1, &quot;^(.+):(.+?):(.+?):clu.+?_([+-])$&quot;, &quot;chr\\1_\\2_\\3_\\4&quot;)) %&gt;%
      mutate(IntronAnnotation = case_when(
        intron %in% NMD.specific.introns ~ &quot;Annotated NMD&quot;,
        intron %in% Intron.Annotations.basic$intron ~ &quot;Annotated basic&quot;,
        intron %in% Introns.Annotations.comprehensive$intron ~ &quot;Annotated Not basic&quot;,
        TRUE ~ &quot;Unannotated&quot;
      )) %&gt;%
      dplyr::select(beta1, beta2, P1, PC1, P2, PC2, IntronAnnotation) %&gt;%
      mutate(Colocalized = &quot;Colocalized&quot;)
  ) %&gt;%
  mutate(Is.Concordant = sign(beta1)==sign(beta2)) %&gt;%
  mutate(ConsistentWithNMD = case_when(
    !IntronAnnotation == &quot;Annotated basic&quot; &amp; Is.Concordant ~ &quot;Inconsistent with NMD&quot;,
    !IntronAnnotation == &quot;Annotated basic&quot; &amp; !Is.Concordant ~ &quot;Consistent with NMD&quot;,
    # IntronAnnotation == &quot;Annotated basic&quot; ~ NA_character_,
    IntronAnnotation == &quot;Annotated basic&quot; &amp; Is.Concordant ~ &quot;Inconsistent with NMD&quot;,
    IntronAnnotation == &quot;Annotated basic&quot; &amp; !Is.Concordant ~ &quot;Consistent with NMD&quot;
  )) %&gt;%
  filter(!is.na(Colocalized))

ggplot(sQTL.eQTL.dat.ColocalizedAndUncolocalized, aes(x=beta1, y=beta2, color=IntronAnnotation)) +
  stat_ellipse(
    data = mvrnorm(1000, mu=c(0,0), matrix(c(1,0.8,0.8,1),2,2)) %&gt;%
      as.data.frame() %&gt;%
      dplyr::select(beta1=V1, beta2=V2),
    type = &quot;norm&quot;, linetype = 2, color=&quot;green&quot;, alpha=0.3) +
  stat_ellipse(
    data = mvrnorm(1000, mu=c(0,0), matrix(c(1,-0.8,-0.8,1),2,2)) %&gt;%
      as.data.frame() %&gt;%
      dplyr::select(beta1=V1, beta2=V2),
    type = &quot;norm&quot;, linetype = 2, color=&quot;red&quot;, alpha=0.3) +
  geom_point(alpha=0.1) +
  geom_vline(xintercept=0) +
  geom_hline(yintercept=0) +
  geom_text(
  data = . %&gt;%
    group_by(IntronAnnotation, Colocalized) %&gt;%
    summarise(cor=cor.test(beta1,beta2)[[&quot;estimate&quot;]], pval=cor.test(beta1,beta2)[[&quot;p.value&quot;]]) %&gt;%
    mutate(R = signif(cor, 3), P=format.pval(pval, 3)) %&gt;%
    mutate(label = str_glue(&quot;R:{R}\nP:{P}&quot;)),
  aes(x=-Inf, y=-Inf, label=label),
  hjust=-.1, vjust=-0.1, color=&#39;black&#39;, size=3
  ) +
  geom_text(
  data = . %&gt;%
    group_by(IntronAnnotation, ConsistentWithNMD, Colocalized) %&gt;%
    distinct(P2, .keep_all=T) %&gt;%
    ungroup() %&gt;%
    count(IntronAnnotation, ConsistentWithNMD, Colocalized) %&gt;%
    filter(ConsistentWithNMD==&quot;Consistent with NMD&quot;),
  aes(x=-Inf, y=Inf, label=n),
  hjust=-0.2, vjust=1.5, color=&#39;red&#39;
  ) +
  geom_text(
  data = . %&gt;%
    group_by(IntronAnnotation, ConsistentWithNMD, Colocalized) %&gt;%
    distinct(P2, .keep_all=T) %&gt;%
    ungroup() %&gt;%
    count(IntronAnnotation, ConsistentWithNMD, Colocalized) %&gt;%
    filter(ConsistentWithNMD==&quot;Inconsistent with NMD&quot;),
  aes(x=Inf, y=Inf, label=n),
  hjust=1.2, vjust=1.5, color=&#39;green&#39;
  ) +
  facet_grid(Colocalized ~ IntronAnnotation) +
  theme_bw() +
  labs(title=&quot;genetic effects of NMD introns and host gene expression&quot;, x=&quot;chRNA sQTL beta\n(standardized units)&quot;, y=&quot;eQTL beta\n(standardized units)&quot;, caption=&quot;Colocalized sQTL/eQTLs betas at top co-finemapped SNP\nUncolocalized betas relative to top chRNA sQTL SNP&quot;) +
  guides(colour = guide_legend(override.aes = list(alpha = 1)))</code></pre>
<p><img src="figure/20221102_PlotStuffForYang.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparing-effects-of-those-qtls-in-polya-metabolic-labelled-and-chromtain" class="section level3">
<h3>Comparing effects of those QTLs in polyA, Metabolic labelled, and Chromtain</h3>
</div>
<div id="qq-plots-sqtls-for-poly-specific-eqtls" class="section level3">
<h3>QQ plots sQTLs for poly-specific eQTLs</h3>
<p>Let’s revisit making QQ plots to reproduce Carlos’ observation that polyA-specific eQTLs are more likely to be sQTLs…</p>
<pre class="r"><code>brewer.pal(n=6, name=&quot;Paired&quot;)</code></pre>
<pre><code>[1] &quot;#A6CEE3&quot; &quot;#1F78B4&quot; &quot;#B2DF8A&quot; &quot;#33A02C&quot; &quot;#FB9A99&quot; &quot;#E31A1C&quot;</code></pre>
<pre class="r"><code>ColorKey &lt;- c(&quot;chRNA.Expression.Splicing TRUE&quot;=&quot;#1F78B4&quot;,
                             &quot;chRNA.Expression.Splicing FALSE&quot;=&quot;#A6CEE3&quot;,
                             &quot;H3K36ME3 TRUE&quot;=&quot;#33A02C&quot;,
                             &quot;H3K36ME3 FALSE&quot;=&quot;#B2DF8A&quot;,
                             &quot;H3K27AC TRUE&quot;=&quot;#E31A1C&quot;,
                             &quot;H3K27AC FALSE&quot;=&quot;#FB9A99&quot;)
LabelKey &lt;- c(&quot;chRNA.Expression.Splicing&quot;=&quot;chRNA&quot;,
                             &quot;H3K36ME3&quot;=&quot;H3K36ME3&quot;,
                             &quot;H3K27AC&quot;=&quot;H3K27AC@TSS&quot;)


sGenes &lt;- paste0(&quot;../code/QTLs/QTLTools/&quot;, c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;polyA.Splicing&quot;,&quot;chRNA.Splicing&quot;),&quot;/GroupedPermutationPassForColoc.FDR_Added.txt.gz&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/QTLs/QTLTools/(.+?)/GroupedPermutationPassForColoc.FDR_Added.txt.gz&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(fread) %&gt;%
  bind_rows(.id=&quot;Dataset&quot;)

Dat.to.plot &lt;- TopSNPEffects.ByPairs %&gt;%
  filter(
    PC1 == &quot;Expression.Splicing.Subset_YRI&quot; &amp;
      PC2 %in% c(&quot;H3K27AC&quot;, &quot;chRNA.Expression.Splicing&quot;, &quot;H3K36ME3&quot;)
    ) %&gt;%
  filter(
    PC2 %in% c(&quot;chRNA.Expression.Splicing&quot;, &quot;H3K36ME3&quot;) |
      paste(P1, P2, sep=&quot;;&quot;) %in% PeaksToTSS$GenePeakPair
  ) %&gt;%
  group_by(GeneLocus, PC2) %&gt;%
  arrange(FDR.y) %&gt;%
  distinct(PC2, .keep_all=T) %&gt;%
  ungroup() %&gt;%
  # mutate(Is_xQTL = FDR.y&lt;0.05) %&gt;%
  mutate(Is_xQTL = p_permutation.y&lt;0.05) %&gt;%
  mutate(FacetLabel = recode(PC2, !!!LabelKey)) %&gt;%
  mutate(Group = paste(PC2, Is_xQTL)) %&gt;%
  add_count(Group, FacetLabel)</code></pre>
<pre class="r"><code>Dat.to.plot %&gt;%
  ggplot(aes(x=-log10(p_permutation.x), y=-log10(p_permutation.y), color=Group)) +
  geom_label(
    data = . %&gt;%
      distinct(Group, FacetLabel, n) %&gt;%
      group_by(FacetLabel) %&gt;%
      mutate(rn = row_number()),
    aes(label=paste0(&quot;n=&quot;, n), vjust=rn+1, color=Group),
    x=-Inf, y=Inf, hjust=-0.5) +
  geom_point(alpha=0.1) +
  scale_color_manual(values=ColorKey) +
  facet_wrap(~FacetLabel) +
  theme_bw() +
  theme(legend.position = &quot;none&quot;) +
  labs(caption=&quot;Number of eGenes that are xQTLs, 10% FDR&quot;,  x=&quot;-log10(q) (eQTL)&quot;, y=&quot;-log10(q) (xQTL)&quot;)</code></pre>
<p><img src="figure/20221102_PlotStuffForYang.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Dat.to.plot %&gt;%
  inner_join(
    sGenes %&gt;%
      dplyr::select(GeneLocus=grp_id, sGene.P = adj_beta_pval, Dataset)
  ) %&gt;%
  group_by(Dataset, Group, FacetLabel) %&gt;%
  mutate(ExpectedP = percent_rank(sGene.P)) %&gt;%
  ungroup() %&gt;%
  ggplot(aes(x=-log10(ExpectedP), y=-log10(sGene.P), color=Group)) +
  geom_abline() +
  geom_point() +
  scale_color_manual(values=c(ColorKey)) +
  facet_grid(Dataset ~ FacetLabel, scales=&quot;free_y&quot;) +
  theme_bw() +
  theme(legend.position = &quot;none&quot;) +
  labs(title=&quot;QQ plot of sQTL P-values&quot;, color=&quot;xQTL SNP&quot;, y=&quot;Observed -log10(P)&quot;, x=&quot;Theoretical -log10(P)&quot;, caption=str_wrap(&quot;&quot;, 30))</code></pre>
<p><img src="figure/20221102_PlotStuffForYang.Rmd/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Hmmm, again I cannot reproduce the results… Maybe I have to group by intron type to see a difference… Most sQTLs are in annotated basic introns, which do not effect expression. So all those sQTLs could swamp and difference that distinguished polyA-specific eQTLs.</p>
<p>Let’s try to reproduce the results more to exactly how Carlos did it…</p>
</div>
<div id="similar-sqtl-plots-yang-asked-for" class="section level3">
<h3>similar sQTL plots Yang asked for:</h3>
<ul>
<li>qqplot of sQTLs for the different types of introns with unexplained vs explained eQTLs signal (y-axis)</li>
</ul>
</div>
<div id="qq-plots-for-eqtls-for-different-categories-of-xqtl-snps" class="section level3">
<h3>QQ plots for eQTLs for different categories of xQTL SNPs</h3>
<ul>
<li>clean version of qqplot that you made, but with only unannotated, NMD, non-basic junctions, H3K27ac, and genome-wide (as black)</li>
</ul>
<pre class="r"><code>test.SNPs &lt;- paste0(&quot;../code/QTLs/QTLTools/&quot;, c(&quot;chRNA.Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;), &quot;/NominalPassForColoc.RandomSamplePvals.txt.gz&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/QTLs/QTLTools/(.+?)/NominalPassForColoc.RandomSamplePvals.txt.gz&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(read_tsv, col_names=c(&quot;trait.x.p.in.y&quot;)) %&gt;%
  bind_rows(.id=&quot;PC2&quot;) %&gt;%
  mutate(PC1 = &quot;TestSNPs&quot;) %&gt;%
  group_by(PC2) %&gt;%
  sample_n(5000) %&gt;%
  ungroup()

show_col(hue_pal()(4))</code></pre>
<p><img src="figure/20221102_PlotStuffForYang.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hue_pal()(4)</code></pre>
<pre><code>[1] &quot;#F8766D&quot; &quot;#7CAE00&quot; &quot;#00BFC4&quot; &quot;#C77CFF&quot;</code></pre>
<pre class="r"><code>Colors &lt;- c(&quot;H3K27AC&quot;=&quot;#808080&quot;, &quot;TestSNPs&quot;=&quot;#000000&quot;, &quot;Annotated NMD&quot;=&quot;#F8766D&quot;, &quot;Annotated Not basic&quot;=&quot;#7CAE00&quot;, &quot;Unannotated&quot;=&quot;#C77CFF&quot;, &quot;Annotated basic&quot;=&quot;#00BFC4&quot;)


TopSNPEffects.ByPairs %&gt;%
  filter(
      (PC2 %in% c(&quot;chRNA.Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;)) &amp;
      # (PC1 %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;H3K27AC&quot;)) &amp;
      (PC1 %in% c(&quot;polyA.Splicing&quot;, &quot;H3K27AC&quot;)) &amp;
      (FDR.x &lt; 0.1)) %&gt;%
  bind_rows(test.SNPs) %&gt;%
  mutate(intron = case_when(
    PC1 %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;polyA.Splicing&quot;) ~ str_replace(P1, &quot;^(.+):(.+?):(.+?):clu.+?_([+-])$&quot;, &quot;chr\\1_\\2_\\3_\\4&quot;),
    TRUE ~ NA_character_)) %&gt;%
  mutate(IntronAnnotation = case_when(
    intron %in% NMD.specific.introns ~ &quot;Annotated NMD&quot;,
    intron %in% Intron.Annotations.basic$intron ~ &quot;Annotated basic&quot;,
    intron %in% Introns.Annotations.comprehensive$intron ~ &quot;Annotated Not basic&quot;,
    !is.na(intron) ~ &quot;Unannotated&quot;,
    TRUE ~ &quot;&quot;
  )) %&gt;%
  # filter(!IntronAnnotation==&quot;Annotated basic&quot;) %&gt;%
  filter(PC2 == &quot;Expression.Splicing.Subset_YRI&quot;) %&gt;%
  mutate(SnpSet = case_when(
    str_detect(PC1, &quot;Splicing&quot;) ~ IntronAnnotation,
    TRUE ~ PC1
  )) %&gt;%
  group_by(SnpSet, PC2) %&gt;%
  mutate(ExpectedP = percent_rank(trait.x.p.in.y)) %&gt;%
  ungroup() %&gt;%
  # mutate(SnpSetColor = recode(SnpSet, !!!Colors)) %&gt;%
  ggplot(aes(x=-log10(ExpectedP), y=-log10(trait.x.p.in.y), color=SnpSet)) +
  geom_abline() +
  geom_point() +
  # scale_color_identity() +
  scale_color_manual(values=Colors) +
  theme_bw() +
  labs(title=&quot;QQ plot of eQTL P-values&quot;, color=&quot;xQTL SNP&quot;, y=&quot;-log10(P)&quot;)</code></pre>
<p><img src="figure/20221102_PlotStuffForYang.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(&quot;../code/scratch/QQ.eQTLs.By.xQTL.pdf&quot;, height=4, width=5)</code></pre>
<p>Let’s make the “polarized” version of this QQ plot… If the eQTL and xQTL betas have the same sign, plot them upright, if opposite site, mirror the QQ plot.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] scales_1.1.0       RColorBrewer_1.1-2 MASS_7.3-51.4      data.table_1.14.2 
 [5] forcats_0.4.0      stringr_1.4.0      dplyr_1.0.9        purrr_0.3.4       
 [9] readr_1.3.1        tidyr_1.2.0        tibble_3.1.7       ggplot2_3.3.6     
[13] tidyverse_1.3.0   

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5        lubridate_1.7.4   assertthat_0.2.1  rprojroot_2.0.2  
 [5] digest_0.6.20     utf8_1.1.4        R6_2.4.0          cellranger_1.1.0 
 [9] backports_1.4.1   reprex_0.3.0      evaluate_0.15     highr_0.9        
[13] httr_1.4.4        pillar_1.7.0      rlang_1.0.5       readxl_1.3.1     
[17] rstudioapi_0.14   R.oo_1.22.0       R.utils_2.9.0     rmarkdown_1.13   
[21] labeling_0.3      munsell_0.5.0     broom_1.0.0       compiler_3.6.1   
[25] httpuv_1.5.1      modelr_0.1.8      xfun_0.31         pkgconfig_2.0.2  
[29] htmltools_0.5.3   tidyselect_1.1.2  workflowr_1.6.2   fansi_0.4.0      
[33] crayon_1.3.4      dbplyr_1.4.2      withr_2.5.0       later_0.8.0      
[37] R.methodsS3_1.7.1 grid_3.6.1        jsonlite_1.6      gtable_0.3.0     
[41] lifecycle_1.0.1   DBI_1.1.0         git2r_0.26.1      magrittr_1.5     
[45] cli_3.3.0         stringi_1.4.3     farver_2.1.0      fs_1.5.2         
[49] promises_1.0.1    xml2_1.3.2        ellipsis_0.3.2    generics_0.1.3   
[53] vctrs_0.4.1       tools_3.6.1       glue_1.6.2        hms_0.5.3        
[57] fastmap_1.1.0     yaml_2.2.0        colorspace_1.4-1  rvest_0.3.5      
[61] knitr_1.39        haven_2.3.1      </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
