<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Check robustness of colocalization results at various thresholds</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Check robustness of colocalization results at various thresholds</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-06-08
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>ChromatinSplicingQTLs/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstree9c3089100be7ffea187f213d23c78fa3b78744c5targetblank9c30891a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/9c3089100be7ffea187f213d23c78fa3b78744c5" target="_blank">9c30891</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstree9c3089100be7ffea187f213d23c78fa3b78744c5targetblank9c30891a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.20220606_PlotColocsForIntuitions.Rmd.swp
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store
    Ignored:    code/.RData
    Ignored:    code/._.DS_Store
    Ignored:    code/._README.md
    Ignored:    code/._report.html
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    code/.snakemake/
    Ignored:    code/Alignments/
    Ignored:    code/ENCODE/
    Ignored:    code/ExpressionAnalysis/
    Ignored:    code/FastqFastp/
    Ignored:    code/FastqFastpSE/
    Ignored:    code/Genotypes/
    Ignored:    code/IntronSlopes/
    Ignored:    code/Misc/
    Ignored:    code/MiscCountTables/
    Ignored:    code/Multiqc/
    Ignored:    code/Multiqc_chRNA/
    Ignored:    code/PeakCalling/
    Ignored:    code/Phenotypes/
    Ignored:    code/PlotGruberQTLs/
    Ignored:    code/PlotQTLs/
    Ignored:    code/ProCapAnalysis/
    Ignored:    code/QC/
    Ignored:    code/QTLs/
    Ignored:    code/ReferenceGenome/
    Ignored:    code/Rplots.pdf
    Ignored:    code/Session.vim
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TODO
    Ignored:    code/Tehranchi/
    Ignored:    code/bigwigs/
    Ignored:    code/bigwigs_FromNonWASPFilteredReads/
    Ignored:    code/config/.DS_Store
    Ignored:    code/config/._.DS_Store
    Ignored:    code/config/.ipynb_checkpoints/
    Ignored:    code/debug.ipynb
    Ignored:    code/debug_python.ipynb
    Ignored:    code/deepTools/
    Ignored:    code/featureCounts/
    Ignored:    code/gwas_summary_stats/
    Ignored:    code/hyprcoloc/
    Ignored:    code/igv_session.xml
    Ignored:    code/log
    Ignored:    code/logs/
    Ignored:    code/notebooks/.ipynb_checkpoints/
    Ignored:    code/rules/.ipynb_checkpoints/
    Ignored:    code/rules/OldRules/
    Ignored:    code/rules/notebooks/
    Ignored:    code/scratch/
    Ignored:    code/scripts/.ipynb_checkpoints/
    Ignored:    code/scripts/GTFtools_0.8.0/
    Ignored:    code/scripts/liftOverBedpe/liftOverBedpe.py
    Ignored:    code/snakemake.log
    Ignored:    code/snakemake.sbatch.log
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/._20220414203249_JASPAR2022_combined_matrices_25818_jaspar.txt
    Ignored:    data/._ColorsForPhenotypes.xlsx
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-10.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-11.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-2.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-3.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-4.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-5.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-6.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-7.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-8.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022.csv

Untracked files:
    Untracked:  code/snakemake_profiles/slurm/__pycache__/

Unstaged changes:
    Modified:   analysis/20220606_PlotColocsForIntuitions.Rmd
    Modified:   code/scripts/GenometracksByGenotype

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/9c3089100be7ffea187f213d23c78fa3b78744c5/analysis/20220606_PlotColocsForIntuitions.Rmd" target="_blank">9c30891</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-06-08
</td>
<td>
update hhyprcoloc output
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>In a <a href="20211217_GenewiseColocFirstLook.html">previous notebook</a> I explored the genewise hyprcoloc output, and noted that 30min and 60min 4sU colocalize (with all default parameters/thresholds) 80% of the time that they are tested. I expect this to be closer to 100%, and we should get similarly high colocalization with eQTL from polyA RNA-seq. Perhaps just by filtering for colocalizations above some threshold we can get more believable results. I could/should technically re-run hyprcoloc with different parameters, but before I do that, to understand the results better, let’s see how these colocalization rates change after filter for different posterior probabilities for colocalization.</p>
</div>
<div id="analysis" class="section level2">
<h2>Analysis</h2>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.2.1 ──</code></pre>
<pre><code>✔ ggplot2 3.2.1     ✔ purrr   0.3.3
✔ tibble  3.0.4     ✔ dplyr   1.0.2
✔ tidyr   1.1.2     ✔ stringr 1.4.0
✔ readr   1.4.0     ✔ forcats 0.5.0</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(viridis)</code></pre>
<pre><code>Loading required package: viridisLite</code></pre>
<pre class="r"><code>library(gplots)</code></pre>
<pre><code>
Attaching package: &#39;gplots&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    lowess</code></pre>
<pre class="r"><code>library(data.table)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre class="r"><code>library(qvalue)
# library(purrr)

sample_n_of &lt;- function(data, size, ...) {
  dots &lt;- quos(...)
  
  group_ids &lt;- data %&gt;% 
    group_by(!!! dots) %&gt;% 
    group_indices()
  
  sampled_groups &lt;- sample(unique(group_ids), size)
  
  data %&gt;% 
    filter(group_ids %in% sampled_groups)
}

dat &lt;- Sys.glob(&quot;../code/hyprcoloc/Results/ForColoc/MolColocTest*_*/results.txt.gz&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/hyprcoloc/Results/ForColoc/MolColocTest(.*?)_(.+?)/results.txt.gz&quot;, &quot;\\1_0.\\2&quot;)) %&gt;%
  lapply(read_tsv) %&gt;%
  bind_rows(.id=&quot;Threshold&quot;)</code></pre>
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  GeneLocus = col_character(),
  HyprcolocIteration = col_double(),
  PosteriorColocalizationPr = col_double(),
  RegionalAssociationPr = col_double(),
  TopCandidateSNP = col_character(),
  ProportionPosteriorPrExplainedByTopSNP = col_double(),
  Trait = col_character()
)</code></pre>
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  GeneLocus = col_character(),
  HyprcolocIteration = col_double(),
  PosteriorColocalizationPr = col_double(),
  RegionalAssociationPr = col_double(),
  TopCandidateSNP = col_character(),
  ProportionPosteriorPrExplainedByTopSNP = col_double(),
  Trait = col_character()
)


── Column specification ────────────────────────────────────────────────────────
cols(
  GeneLocus = col_character(),
  HyprcolocIteration = col_double(),
  PosteriorColocalizationPr = col_double(),
  RegionalAssociationPr = col_double(),
  TopCandidateSNP = col_character(),
  ProportionPosteriorPrExplainedByTopSNP = col_double(),
  Trait = col_character()
)


── Column specification ────────────────────────────────────────────────────────
cols(
  GeneLocus = col_character(),
  HyprcolocIteration = col_double(),
  PosteriorColocalizationPr = col_double(),
  RegionalAssociationPr = col_double(),
  TopCandidateSNP = col_character(),
  ProportionPosteriorPrExplainedByTopSNP = col_double(),
  Trait = col_character()
)


── Column specification ────────────────────────────────────────────────────────
cols(
  GeneLocus = col_character(),
  HyprcolocIteration = col_double(),
  PosteriorColocalizationPr = col_double(),
  RegionalAssociationPr = col_double(),
  TopCandidateSNP = col_character(),
  ProportionPosteriorPrExplainedByTopSNP = col_double(),
  Trait = col_character()
)


── Column specification ────────────────────────────────────────────────────────
cols(
  GeneLocus = col_character(),
  HyprcolocIteration = col_double(),
  PosteriorColocalizationPr = col_double(),
  RegionalAssociationPr = col_double(),
  TopCandidateSNP = col_character(),
  ProportionPosteriorPrExplainedByTopSNP = col_double(),
  Trait = col_character()
)</code></pre>
<pre class="r"><code>PeaksToTSS &lt;- Sys.glob(&quot;../code/Misc/PeaksClosestToTSS/*_assigned.tsv.gz&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/Misc/PeaksClosestToTSS/(.+?)_assigned.tsv.gz&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(read_tsv) %&gt;%
  bind_rows(.id=&quot;ChromatinMark&quot;) %&gt;%
  mutate(GenePeakPair = paste(gene, peak, sep = &quot;;&quot;)) %&gt;%
  distinct(ChromatinMark, peak, gene, .keep_all=T)</code></pre>
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  chrom = col_character(),
  TSS_start = col_double(),
  gene = col_character(),
  strand = col_character(),
  peak = col_character(),
  distance = col_double()
)</code></pre>
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  chrom = col_character(),
  TSS_start = col_double(),
  gene = col_character(),
  strand = col_character(),
  peak = col_character(),
  distance = col_double()
)


── Column specification ────────────────────────────────────────────────────────
cols(
  chrom = col_character(),
  TSS_start = col_double(),
  gene = col_character(),
  strand = col_character(),
  peak = col_character(),
  distance = col_double()
)</code></pre>
<pre class="r"><code>dat %&gt;%
  distinct(Threshold, GeneLocus, TopCandidateSNP, .keep_all = T) %&gt;%
  pull(PosteriorColocalizationPr) %&gt;% hist()</code></pre>
<p><img src="figure/20220606_PlotColocsForIntuitions.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dat %&gt;%
  separate(Trait, into=c(&quot;PC&quot;, &quot;P&quot;), sep=&quot;;&quot;) %&gt;%
  pull(PC) %&gt;% unique() %&gt;% sort()</code></pre>
<pre><code> [1] &quot;chRNA.Expression_cheRNA&quot;             &quot;chRNA.Expression_eRNA&quot;              
 [3] &quot;chRNA.Expression_lncRNA&quot;             &quot;chRNA.Expression_snoRNA&quot;            
 [5] &quot;chRNA.Expression.Splicing&quot;           &quot;chRNA.IER&quot;                          
 [7] &quot;chRNA.IR&quot;                            &quot;chRNA.IRjunctions&quot;                  
 [9] &quot;chRNA.Slopes&quot;                        &quot;chRNA.Splicing&quot;                     
[11] &quot;CTCF&quot;                                &quot;Expression.Splicing&quot;                
[13] &quot;Expression.Splicing.Subset_YRI&quot;      &quot;H3K27AC&quot;                            
[15] &quot;H3K36ME3&quot;                            &quot;H3K4ME1&quot;                            
[17] &quot;H3K4ME3&quot;                             &quot;MetabolicLabelled.30min&quot;            
[19] &quot;MetabolicLabelled.30min.IER&quot;         &quot;MetabolicLabelled.30min.IR&quot;         
[21] &quot;MetabolicLabelled.30min.IRjunctions&quot; &quot;MetabolicLabelled.30min.Splicing&quot;   
[23] &quot;MetabolicLabelled.60min&quot;             &quot;MetabolicLabelled.60min.IER&quot;        
[25] &quot;MetabolicLabelled.60min.IR&quot;          &quot;MetabolicLabelled.60min.IRjunctions&quot;
[27] &quot;MetabolicLabelled.60min.Splicing&quot;    &quot;polyA.IER&quot;                          
[29] &quot;polyA.IR&quot;                            &quot;polyA.IR.Subset_YRI&quot;                
[31] &quot;polyA.IRjunctions&quot;                   &quot;polyA.Splicing&quot;                     
[33] &quot;polyA.Splicing.Subset_YRI&quot;           &quot;ProCap&quot;                             </code></pre>
<pre class="r"><code>dat %&gt;%
  separate(Trait, into=c(&quot;PC&quot;, &quot;P&quot;), sep=&quot;;&quot;) %&gt;%
  count(Threshold, PC) %&gt;%
  ggplot(aes(x=PC, y=n)) +
  geom_col() +
  facet_wrap(~Threshold) +
  theme_bw() +
  labs(title = &quot;Number of Loci:molQTL pairs attempted to colocalize in total&quot;) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))</code></pre>
<p><img src="figure/20220606_PlotColocsForIntuitions.Rmd/unnamed-chunk-1-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Carlos wants to know: - how many sQTL and irQTLs coloc with polyA eQTLs? And how many coloc with polyA eQTL but not polyA sQTL?</p>
<pre class="r"><code>dat.forcarlos &lt;- dat %&gt;%
    filter(!str_detect(Threshold, &quot;eQTL&quot;)) %&gt;%
    separate(Trait, into=c(&quot;PC&quot;, &quot;P&quot;), sep=&quot;;&quot;, remove = F) %&gt;%
    # pull(PC) %&gt;% unique()
    filter(PC %in% c(&quot;Expression.Splicing.Subset_YRI&quot;, &quot;polyA.Splicing.Subset_YRI&quot;, &quot;polyA.IER&quot;, &quot;chRNA.Expression.Splicing&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;))

dat.forcarlos %&gt;%
    count(PC, Threshold) %&gt;%
    ggplot(aes(x=PC, y=n)) +
    geom_col() +
    facet_wrap(~Threshold) +
    labs(title=&quot;Number features attempted for coloc at different P thresholds&quot;, y=&quot;NumFeatures&quot;, x=&quot;PhenotypeClass&quot;) +
    theme(axis.text.x=element_text(angle=45, hjust=1))</code></pre>
<p><img src="figure/20220606_PlotColocsForIntuitions.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dat.forcarlos %&gt;%
    filter(!is.na(TopCandidateSNP)) %&gt;%
    group_by(GeneLocus, TopCandidateSNP, Threshold) %&gt;%
    mutate(Contains_eQTL = any(PC == &quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
    mutate(Contains_sQTL = any(PC %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;polyA.IER&quot;,  &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;))) %&gt;%
    mutate(Contains_chRNA_specific_sQTL = any(PC %in% c(&quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;))
           &amp; !any(PC %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;polyA.IER&quot;))) %&gt;%
    ungroup() %&gt;%
    distinct(Threshold, GeneLocus, TopCandidateSNP, .keep_all=T) %&gt;%
    filter(Contains_eQTL) %&gt;%
    group_by(Threshold) %&gt;%
    summarise(
              Num_eQTL = sum(Contains_eQTL),
              Num_sQTL_coloc_eQTL = sum(Contains_sQTL),
              Num_chRNA_specific_sQTL_coloc_eQTL = sum(Contains_chRNA_specific_sQTL)
    ) %&gt;%
    gather(key=&quot;eQTL_type&quot;, value=&quot;count&quot;, -Threshold) %&gt;%
    ggplot(aes(x=eQTL_type, y=count)) +
    geom_col() +
    geom_text(aes(label=count), color=&quot;black&quot;, angle=90, hjust=1) +
    facet_wrap(~Threshold, scales=&quot;free_y&quot;) +
    labs(title=&quot;Num eQTL coloc with sQTL&quot;) +
    theme_classic() +
    theme(axis.text.x=element_text(angle=45, hjust=1))</code></pre>
<pre><code>`summarise()` ungrouping output (override with `.groups` argument)</code></pre>
<p><img src="figure/20220606_PlotColocsForIntuitions.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Below is some code for making files, and then some code for running my script to plot some colocalizations. I will plot 5 loci where metabolic labelled samples did not coloc, 5 where they did, 5 where promoterQTL/eQTL coloc, 5 where non-promoter QTL coloc, 5 where sQTL/eQTL coloc, and 5 where sQTL/eQTL don’t coloc.</p>
<pre class="r"><code>dat.ToPlotColocs &lt;- dat %&gt;%
    filter(Threshold == &quot;_0.001&quot;) %&gt;%
    separate(Trait, into=c(&quot;PC&quot;, &quot;P&quot;), sep=&quot;;&quot;, remove = F) %&gt;%
    left_join(PeaksToTSS %&gt;% select(ChromatinMark, peak, gene), by=c(&quot;PC&quot;=&quot;ChromatinMark&quot;, &quot;P&quot;=&quot;peak&quot;)) %&gt;%
    mutate( PC = case_when(
                           gene == GeneLocus ~ paste(PC, &quot;AtPromoter&quot; ,sep=&quot;_&quot;),
                           !is.na(gene) ~ paste(PC, &quot;AtDistalPromoter&quot; ,sep=&quot;_&quot;),
                           TRUE ~ PC
                           ) )

# both metabolic coloc
Targets &lt;- dat.ToPlotColocs  %&gt;%
    filter(!is.na(TopCandidateSNP)) %&gt;%
    group_by(GeneLocus, TopCandidateSNP) %&gt;%
    filter(any(str_detect(PC, &quot;MetabolicLabelled.30min&quot;)) &amp; any(str_detect(PC, &quot;MetabolicLabelled.60min&quot;))) %&gt;%
    ungroup() %&gt;%
    pull(GeneLocus) %&gt;% unique()
dat.ToPlotColocs %&gt;%
    filter(Threshold == &quot;_0.001&quot;) %&gt;%
    filter(PC %in% c(&quot;MetabolicLabelled.30min&quot;, &quot;MetabolicLabelled.60min&quot;, &quot;Expression.Splicing.Subset_YRI&quot;, &quot;H3K27AC&quot;, &quot;H3K27AC_AtPromoter&quot;, &quot;H3K27AC_AtDistalPromoter&quot;)) %&gt;%
    filter(GeneLocus %in% Targets) %&gt;%
    sample_n_of(10, GeneLocus) %&gt;%
    select(-Threshold) %&gt;%
    write_tsv(&quot;scratch/ColocExamples/GroupMetabolicColoc.tsv&quot;)


# metabolic don&#39;t coloc
Targets &lt;- dat.ToPlotColocs  %&gt;%
    filter(PC %in% c(&quot;MetabolicLabelled.30min&quot;, &quot;MetabolicLabelled.60min&quot;)) %&gt;%
    group_by(GeneLocus) %&gt;%
    filter(any(str_detect(PC, &quot;MetabolicLabelled.30min&quot;)) &amp; any(str_detect(PC, &quot;MetabolicLabelled.60min&quot;))) %&gt;%
    ungroup() %&gt;%
    group_by(GeneLocus, TopCandidateSNP) %&gt;%
    filter(
           (is.na(TopCandidateSNP) &amp; PC %in% c(&quot;MetabolicLabelled.30min&quot;, &quot;MetabolicLabelled.60min&quot;)) | (sum(str_detect(PC, &quot;MetabolicLabelled&quot;)) == 1)
    ) %&gt;%
    ungroup() %&gt;%
    pull(GeneLocus) %&gt;% unique()
Targets &lt;- dat.ToPlotColocs %&gt;%
    filter(GeneLocus %in% TestedTargets) %&gt;%
    filter(is.na(TopCandidateSNP))
dat.ToPlotColocs %&gt;%
    filter(Threshold == &quot;_0.001&quot;) %&gt;%
    filter(PC %in% c(&quot;MetabolicLabelled.30min&quot;, &quot;MetabolicLabelled.60min&quot;, &quot;Expression.Splicing.Subset_YRI&quot;, &quot;H3K27AC&quot;, &quot;H3K27AC_AtPromoter&quot;, &quot;H3K27AC_AtDistalPromoter&quot;)) %&gt;%
    filter(GeneLocus %in% Targets) %&gt;%
    sample_n_of(10, GeneLocus) %&gt;%
    select(-Threshold) %&gt;%
    write_tsv(&quot;scratch/ColocExamples/GroupMetabolicNotColoc.tsv&quot;)

# Promoter and eqtl coloc TODO
Targets &lt;- dat.ToPlotColocs %&gt;%
    filter(PC %in% c(&quot;MetabolicLabelled.30min&quot;, &quot;MetabolicLabelled.60min&quot;, &quot;Expression.Splicing.Subset_YRI&quot;, &quot;H3K27AC&quot;, &quot;H3K27AC_AtPromoter&quot;, &quot;H3K27AC_AtDistalPromoter&quot;)) %&gt;%
    filter(!is.na(TopCandidateSNP)) %&gt;%
    group_by(GeneLocus, TopCandidateSNP) %&gt;%
    filter(any(PC == &quot;Expression.Splicing.Subset_YRI&quot;) &amp; any(PC == &quot;H3K27AC_AtPromoter&quot;)) %&gt;%
    ungroup() %&gt;%
    pull(GeneLocus) %&gt;% unique()
dat.ToPlotColocs %&gt;%
    filter(Threshold == &quot;_0.001&quot;) %&gt;%
    filter(PC %in% c(&quot;MetabolicLabelled.30min&quot;, &quot;MetabolicLabelled.60min&quot;, &quot;Expression.Splicing.Subset_YRI&quot;, &quot;H3K27AC&quot;, &quot;H3K27AC_AtPromoter&quot;, &quot;H3K27AC_AtDistalPromoter&quot;)) %&gt;%
    filter(GeneLocus %in% Targets) %&gt;%
    sample_n_of(10, GeneLocus) %&gt;%
    select(-Threshold) %&gt;%
    write_tsv(&quot;scratch/ColocExamples/GroupPromoterEqtlColoc.tsv&quot;)

# Promoter and eqtl don&#39;t coloc
Targets &lt;- dat.ToPlotColocs %&gt;%
    filter(PC %in% c(&quot;MetabolicLabelled.30min&quot;, &quot;MetabolicLabelled.60min&quot;, &quot;Expression.Splicing.Subset_YRI&quot;, &quot;H3K27AC&quot;, &quot;H3K27AC_AtPromoter&quot;, &quot;H3K27AC_AtDistalPromoter&quot;)) %&gt;%
    filter(!is.na(TopCandidateSNP)) %&gt;%
    group_by(GeneLocus, TopCandidateSNP) %&gt;%
    filter(any(PC == &quot;Expression.Splicing.Subset_YRI&quot;) &amp; any(PC == &quot;H3K27AC_AtPromoter&quot;)) %&gt;%
    ungroup() %&gt;%
    pull(GeneLocus) %&gt;% unique()
dat.ToPlotColocs %&gt;%
    filter(Threshold == &quot;_0.001&quot;) %&gt;%
    filter(PC %in% c(&quot;MetabolicLabelled.30min&quot;, &quot;MetabolicLabelled.60min&quot;, &quot;Expression.Splicing.Subset_YRI&quot;, &quot;H3K27AC&quot;, &quot;H3K27AC_AtPromoter&quot;, &quot;H3K27AC_AtDistalPromoter&quot;)) %&gt;%
    filter(GeneLocus %in% Targets) %&gt;%
    sample_n_of(10, GeneLocus) %&gt;%
    select(-Threshold) %&gt;%
    write_tsv(&quot;scratch/ColocExamples/GroupPromoterEqtlColoc.tsv&quot;)

dat %&gt;%
  count(Threshold, GeneLocus) %&gt;%
  ggplot(aes(x=n, color=Threshold)) +
  stat_ecdf() +
  coord_cartesian(xlim=c(0,10)) +
  theme_bw()

dat.tidy



dat %&gt;%
  filter(Threshold == &quot;_0.01&quot;) %&gt;%
  select(-Threshold) %&gt;%
  # add_count(GeneLocus, TopCandidateSNP) %
  ) %&gt;%
  write_tsv(&quot;../code/scratch/TestHyprcolocPlots.tsv&quot;)</code></pre>
<pre class="bash"><code>conda activate r_essentials
echo &quot;hello world&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.3 (2017-11-30)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] qvalue_2.10.0     data.table_1.12.8 gplots_3.0.1      viridis_0.5.1    
 [5] viridisLite_0.3.0 forcats_0.5.0     stringr_1.4.0     dplyr_1.0.2      
 [9] purrr_0.3.3       readr_1.4.0       tidyr_1.1.2       tibble_3.0.4     
[13] ggplot2_3.2.1     tidyverse_1.2.1  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.3         lubridate_1.7.9.2  gtools_3.5.0       assertthat_0.2.1  
 [5] rprojroot_1.3-2    digest_0.6.27      plyr_1.8.4         R6_2.4.1          
 [9] cellranger_1.1.0   backports_1.1.5    evaluate_0.14      httr_1.4.2        
[13] pillar_1.4.7       rlang_0.4.9        lazyeval_0.2.2     readxl_1.3.1      
[17] rstudioapi_0.10    gdata_2.18.0       whisker_0.3-2      rmarkdown_2.6     
[21] labeling_0.3       splines_3.4.3      munsell_0.5.0      broom_0.7.3       
[25] compiler_3.4.3     httpuv_1.5.2       modelr_0.1.8       xfun_0.20         
[29] pkgconfig_2.0.3    htmltools_0.4.0    tidyselect_1.1.0   gridExtra_2.3     
[33] workflowr_1.5.0    fansi_0.4.0        crayon_1.3.4       withr_2.1.2       
[37] later_1.0.0        bitops_1.0-6       grid_3.4.3         jsonlite_1.6      
[41] gtable_0.3.0       lifecycle_0.2.0    git2r_0.26.1       magrittr_1.5      
[45] scales_1.1.0       KernSmooth_2.23-15 cli_2.0.0          stringi_1.4.3     
[49] farver_2.0.1       reshape2_1.4.3     fs_1.3.1           promises_1.1.0    
[53] xml2_1.2.0         ellipsis_0.3.0     generics_0.1.0     vctrs_0.3.6       
[57] tools_3.4.3        glue_1.4.2         hms_0.5.3          yaml_2.2.0        
[61] colorspace_2.0-0   caTools_1.17.1     rvest_0.3.6        knitr_1.26        
[65] haven_2.3.1       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
