<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Plot heatmaps many ways</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Plot heatmaps many ways</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-10-07
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>ChromatinSplicingQTLs/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstree149e3ef82b5a9e26bd4b86d40e5c057d0e726f39targetblank149e3efa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/149e3ef82b5a9e26bd4b86d40e5c057d0e726f39" target="_blank">149e3ef</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstree149e3ef82b5a9e26bd4b86d40e5c057d0e726f39targetblank149e3efa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/149e3ef82b5a9e26bd4b86d40e5c057d0e726f39" target="_blank">149e3ef</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/figure/
    Ignored:    analysis/site_libs/
    Ignored:    code/.DS_Store
    Ignored:    code/.RData
    Ignored:    code/._.DS_Store
    Ignored:    code/._README.md
    Ignored:    code/._report.html
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    code/.snakemake/
    Ignored:    code/APA_Processing/
    Ignored:    code/Alignments/
    Ignored:    code/ChromHMM/
    Ignored:    code/ENCODE/
    Ignored:    code/ExpressionAnalysis/
    Ignored:    code/FastqFastp/
    Ignored:    code/FastqFastpSE/
    Ignored:    code/Genotypes/
    Ignored:    code/IntronSlopes/
    Ignored:    code/Misc/
    Ignored:    code/MiscCountTables/
    Ignored:    code/Multiqc/
    Ignored:    code/Multiqc_chRNA/
    Ignored:    code/NonCodingRNA_annotation/
    Ignored:    code/PeakCalling/
    Ignored:    code/Phenotypes/
    Ignored:    code/PlotGruberQTLs/
    Ignored:    code/PlotQTLs/
    Ignored:    code/ProCapAnalysis/
    Ignored:    code/QC/
    Ignored:    code/QTL_SNP_Enrichment/
    Ignored:    code/QTLs/
    Ignored:    code/ReferenceGenome/
    Ignored:    code/Rplots.pdf
    Ignored:    code/Session.vim
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TODO
    Ignored:    code/Tehranchi/
    Ignored:    code/bigwigs/
    Ignored:    code/bigwigs_FromNonWASPFilteredReads/
    Ignored:    code/config/.DS_Store
    Ignored:    code/config/._.DS_Store
    Ignored:    code/config/.ipynb_checkpoints/
    Ignored:    code/debug.ipynb
    Ignored:    code/debug_python.ipynb
    Ignored:    code/deepTools/
    Ignored:    code/featureCounts/
    Ignored:    code/gwas_summary_stats/
    Ignored:    code/hyprcoloc/
    Ignored:    code/igv_session.xml
    Ignored:    code/log
    Ignored:    code/logs/
    Ignored:    code/notebooks/.ipynb_checkpoints/
    Ignored:    code/rules/.QTL_SNP_Enrichment.smk.swp
    Ignored:    code/rules/.ipynb_checkpoints/
    Ignored:    code/rules/OldRules/
    Ignored:    code/rules/notebooks/
    Ignored:    code/scratch/
    Ignored:    code/scripts/.ipynb_checkpoints/
    Ignored:    code/scripts/GTFtools_0.8.0/
    Ignored:    code/scripts/__pycache__/
    Ignored:    code/scripts/liftOverBedpe/liftOverBedpe.py
    Ignored:    code/snakemake.log
    Ignored:    code/snakemake.sbatch.log
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/._20220414203249_JASPAR2022_combined_matrices_25818_jaspar.txt
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-10.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-11.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-2.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-3.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-4.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-5.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-6.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-7.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-8.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022.csv

Untracked files:
    Untracked:  analysis/20220713_PlotHeatmapManyWays_JustPi1.Rmd
    Untracked:  code/snakemake_profiles/slurm/__pycache__/

Unstaged changes:
    Modified:   analysis/20220713_PlotHeatmapManyWays.Rmd
    Modified:   code/scripts/GenometracksByGenotype

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>I’ve been making heatmaps to communicate the fraction of xQTLs (eg eQTLs/sQTLs/chromatinQTLs) that colocalize with yQTLs. There are many slightly different methods I’ve been using to make this heatmap, with slightly different interpretations. In this notebook, I will make lots of those heatmaps to compare the different methods.</p>
<p>Here is a list of the different methods I have been using, with the names I am calling these methods. Note that for all of these methods I can consider using different statistical thresholds for calling something a QTL, so in this notebook I’ll explore that too.</p>
<ul>
<li>Pi1: “What is the pi1 statistic for yQTLs (ascertainment phenotype), in xQTLs (discovery phenotype)”. I can use different FDR thresholds for x (discovery phenotype)</li>
<li>ColocalizationRate: “What fraction of xQTLs have at least one colocalizing yQTL”.</li>
<li>ColocalizationRateAmongAttempted: “When there is both a xQTL and yQTL around the same gene, how often do they colocalize”</li>
<li>SpearmanCorrealtionThreshold: What fraction of xQTLs have at least one yQTL with correlated association signal (spearman correlation coef &gt; Treshold)</li>
<li>SpearmanCorrealtionThresholdAmongAttempted: When there is both a xQTL and yQTL around the same gene, how often do they have correlation association signals (spearman correlation coef &gt; Treshold)</li>
</ul>
<pre class="r"><code>library(tidyverse)
library(viridis)
library(gplots)
library(data.table)
library(qvalue)
library(purrr)
library(GGally)
library(pROC)

# pi0est function breaks sometimes. this wrapper function returns 1 when that happens
CalculatePi1 &lt;- function (dat.in, ...) {
  return(tryCatch(1-pi0est(dat.in$Pvals.For.Pi1, ...)$pi0, error=function(e) 1))
}</code></pre>
</div>
<div id="pi1" class="section level1">
<h1>pi1</h1>
<p>First let’s read in data for Pi1 method</p>
<pre class="r"><code>FilesChunks &lt;- paste0(&quot;../code/scratch/PairwisePi1Traits.P.&quot;, 1:10, &quot;.txt.gz&quot;)


dat &lt;- lapply(FilesChunks, fread, sep=&#39;\t&#39;) %&gt;%
  bind_rows()

RecodeDat &lt;- read_tsv(&quot;../data/Phenotypes_recode_for_Plotting.txt&quot;)

RecodeVec &lt;- RecodeDat %&gt;%
  select(PC, ShorterAlias) %&gt;%
  deframe()

RecodeIncludePCs &lt;- RecodeDat %&gt;%
  filter(Include) %&gt;%
  pull(PC)

colnames(dat)</code></pre>
<pre><code> [1] &quot;PC1&quot;                      &quot;P1&quot;                      
 [3] &quot;GeneLocus&quot;                &quot;p_permutation.x&quot;         
 [5] &quot;singletrait_topvar.x&quot;     &quot;singletrait_topvar_chr.x&quot;
 [7] &quot;singletrait_topvar_pos.x&quot; &quot;FDR.x&quot;                   
 [9] &quot;PC2&quot;                      &quot;P2&quot;                      
[11] &quot;p_permutation.y&quot;          &quot;singletrait_topvar.y&quot;    
[13] &quot;singletrait_topvar_chr.y&quot; &quot;singletrait_topvar_pos.y&quot;
[15] &quot;FDR.y&quot;                    &quot;trait.x.p.in.y&quot;          </code></pre>
<p>I also need to simulate the null distribution to deal with the one to many problem…</p>
<pre class="r"><code>MaxSampleSizeToCreateANull &lt;- 250
NSamplesToEstimateDistribution &lt;- 10000
NullSimulatedTestStats &lt;- matrix(nrow=MaxSampleSizeToCreateANull, ncol=NSamplesToEstimateDistribution)
rownames(NullSimulatedTestStats) &lt;- paste0(&quot;runif_samplesize&quot;, 1:MaxSampleSizeToCreateANull)
colnames(NullSimulatedTestStats) &lt;- paste0(&quot;Sample_&quot;, 1:NSamplesToEstimateDistribution)


for (i in 1:MaxSampleSizeToCreateANull){
  SampleSizeFromUniform &lt;- i
  SampledDat &lt;- matrix(runif(SampleSizeFromUniform*NSamplesToEstimateDistribution), nrow=NSamplesToEstimateDistribution)
  SampleDatNullTestStatistics &lt;- -log10(apply(SampledDat, 1, min))
  NullSimulatedTestStats[i,] &lt;- SampleDatNullTestStatistics
}


NullSimulatedTestStats %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;runif_samplesize&quot;) %&gt;%
  slice(1:20) %&gt;%
  # mutate(runif_samplesize = as.numeric(str_remove(runif_samplesize, &quot;runif_samplesize&quot;))) %&gt;%
  gather(key=&quot;Sample&quot;, value=&quot;value&quot;, -runif_samplesize) %&gt;%
  ggplot(aes(x=value, color=runif_samplesize)) +
  geom_density() +
  theme_bw()</code></pre>
<p><img src="figure/20220713_PlotHeatmapManyWays_JustPi1.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ecdf.functions &lt;- apply(NullSimulatedTestStats, 1, ecdf)
ecdf.functions[[1]](1)</code></pre>
<pre><code>[1] 0.9036</code></pre>
<p>Now calculate pi1</p>
<pre class="r"><code>dat.split &lt;- dat %&gt;%
  filter(PC1 %in% RecodeIncludePCs) %&gt;%
  filter(PC2 %in% RecodeIncludePCs) %&gt;%
  mutate(PC1 = recode(PC1, !!!RecodeVec)) %&gt;%
  mutate(PC2 = recode(PC2, !!!RecodeVec)) %&gt;%
  group_by(PC1, P1, PC2) %&gt;%
  mutate(test.stat.obs = -log10(min(trait.x.p.in.y))) %&gt;%
  ungroup() %&gt;%
  add_count(PC1, P1, PC2) %&gt;%
  filter(n&lt;=100) %&gt;%
  group_by(PC1, PC2) %&gt;%
  rowwise() %&gt;%
  mutate(Pvals.For.Pi1 = 1-ecdf.functions[[n]](test.stat.obs)) %&gt;%
  ungroup() %&gt;%
  select(PC1, PC2, Pvals.For.Pi1) %&gt;%
  filter(!PC1==PC2) %&gt;%
  split(paste(.$PC1, .$PC2, sep = &quot;;&quot;))

dat.pi1 &lt;- lapply(dat.split, CalculatePi1) %&gt;%
  unlist() %&gt;%
  data.frame(pi1=.) %&gt;%
  rownames_to_column(&quot;PC1_PC2&quot;) %&gt;%
  separate(PC1_PC2, into=c(&quot;PC1&quot;, &quot;PC2&quot;), sep=&#39;;&#39;)

pi.heatmap &lt;- ggplot(dat.pi1, aes(x=PC1, y=PC2, fill=pi1)) +
  geom_raster() +
  geom_text(aes(label=signif(pi1*100, 2)), color=&quot;blue&quot;) +
  scale_fill_viridis(option=&quot;A&quot;, direction = -1, limits=c(0,1)) +
  coord_flip() +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(x=&quot;Discovery QTL phenotype&quot;, y=&quot;Phenotype assessed for overlap&quot;)
pi.heatmap </code></pre>
<p><img src="figure/20220713_PlotHeatmapManyWays_JustPi1.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, there are some 0 values… Let’s check the P value distribution as histogram for these cells in the heatmap to see what’s up with pi1 estimation…</p>
<pre class="r"><code>dat.split %&gt;%
  bind_rows(.id=&quot;ID&quot;) %&gt;%
  # filter(PC1 == &quot;ProCap&quot; &amp; PC2 == &quot;H3K4ME3&quot;) %&gt;%
  ggplot(aes(x=Pvals.For.Pi1)) +
  geom_histogram() +
  facet_wrap(~ID, scales = &quot;free_y&quot;) +
  theme_bw()</code></pre>
<p><img src="figure/20220713_PlotHeatmapManyWays_JustPi1.Rmd/unnamed-chunk-5-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>Ok by in large the p value histograms look good. let’s zoom in on one of the problematic histograms.</p>
<pre class="r"><code>dat.split %&gt;%
  bind_rows(.id=&quot;ID&quot;) %&gt;%
  filter(PC1 == &quot;ProCap&quot; &amp; PC2 == &quot;H3K4ME3&quot;) %&gt;%
  pull(Pvals.For.Pi1) %&gt;%
  hist(breaks=100)</code></pre>
<p><img src="figure/20220713_PlotHeatmapManyWays_JustPi1.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, I could see how <code>pi0est</code> function might have trouble with this histogram. Let’s play with the <code>methods</code> parameter to see how it effects things.</p>
<pre class="r"><code>dat.split %&gt;%
  bind_rows(.id=&quot;ID&quot;) %&gt;%
  filter(PC1 == &quot;ProCap&quot; &amp; PC2 == &quot;H3K4ME3&quot;) %&gt;%
  pull(Pvals.For.Pi1) %&gt;%
  pi0est(pi0.method=&quot;smoother&quot;)</code></pre>
<pre><code>$pi0
[1] 1

$pi0.lambda
 [1] 0.6268863 0.6020165 0.5937425 0.5969455 0.6001775 0.6078990 0.6177524
 [8] 0.6269954 0.6424722 0.6605468 0.6846282 0.7169067 0.7496413 0.8149586
[15] 0.8954897 0.9921968 1.1268392 1.4137042 2.1635114

$lambda
 [1] 0.05 0.10 0.15 0.20 0.25 0.30 0.35 0.40 0.45 0.50 0.55 0.60 0.65 0.70 0.75
[16] 0.80 0.85 0.90 0.95

$pi0.smooth
 [1] 0.5694685 0.5696858 0.5704027 0.5723214 0.5763434 0.5835632 0.5952669
 [8] 0.6129310 0.6382021 0.6728370 0.7186174 0.7772207 0.8500475 0.9380003
[15] 1.0412060 1.1588122 1.2888099 1.4278979 1.5716760</code></pre>
<pre class="r"><code>dat.split %&gt;%
  bind_rows(.id=&quot;ID&quot;) %&gt;%
  filter(PC1 == &quot;ProCap&quot; &amp; PC2 == &quot;H3K4ME3&quot;) %&gt;%
  pull(Pvals.For.Pi1) %&gt;%
  pi0est(pi0.method=&quot;bootstrap&quot;)</code></pre>
<pre><code>$pi0
[1] 0.6001775

$pi0.lambda
 [1] 0.6268863 0.6020165 0.5937425 0.5969455 0.6001775 0.6078990 0.6177524
 [8] 0.6269954 0.6424722 0.6605468 0.6846282 0.7169067 0.7496413 0.8149586
[15] 0.8954897 0.9921968 1.1268392 1.4137042 2.1635114

$lambda
 [1] 0.05 0.10 0.15 0.20 0.25 0.30 0.35 0.40 0.45 0.50 0.55 0.60 0.65 0.70 0.75
[16] 0.80 0.85 0.90 0.95

$pi0.smooth
NULL</code></pre>
<p>Ok I think the key is to use the <code>pi0.method=&quot;bootstrap&quot;</code> parameter for the <code>pi0est</code> function. That seems to yield pi0 estimations that are more in line with what I would estimate by eye from the histogram. <a href="https://www.jstor.org/stable/3647644#metadata_info_tab_contents">This publication</a> has some detials on the difference between these two methods for estimating <span class="math inline">\(\pi_0\)</span>. My understanding in brief, is that the ‘smooth’ method uses spline fitting to estimate the <span class="math inline">\(\lambda\)</span> parameter which defines exactly what p-value to define as 100% null tests, and the p-values less than <span class="math inline">\(\lambda\)</span> are a mixture of null and non-null tests. If <span class="math inline">\(\lambda\)</span> is over-estimated, then pi0 will be downwardly biased, however, it is a bias/variance tradeoff… So the smoothing fits a smooth curve to estimate <span class="math inline">\(\lambda\)</span>, <em>assuming the the p-value distribution has the least mass at 1</em> (thereby making the <span class="math inline">\(\pi_0\)</span> very sensitive to fluctuations at the right edge of the P-value histogram), whereas the bootstrap method randomly samples from the P-value vector to find an optimal balance between variance and bias for estimating <span class="math inline">\(\lambda\)</span>. In our cases where the P-value distribution is (slightly concerningly) heavy near 1 for some trait pairs, the smooth method gives wonky results, and the bootsrap method is much closer to what a human would estimate <span class="math inline">\(\pi_0\)</span> just from eye-balling the P-value distribution.</p>
<p>I think the reason the P-value distribution is a bit odd for some histograms with some extra mass near one is because for cases where there are many-to-one traits mapping and I take the minimum, sometimes the many traits are not significant and also not independent, so the resulting test statistic is inflated compared to my simulated null which is drawn from truly independent samples. In general, this will may make results a tad conservative.</p>
<p>Anyway, let’s try remaking the heatmap and see what changed with this bootstrap method for estimating pi1.</p>
<pre class="r"><code>lapply(dat.split, CalculatePi1, pi0.method=&quot;bootstrap&quot;) %&gt;%
  unlist() %&gt;%
  data.frame(pi1=.) %&gt;%
  rownames_to_column(&quot;PC1_PC2&quot;) %&gt;%
  separate(PC1_PC2, into=c(&quot;PC1&quot;, &quot;PC2&quot;), sep=&#39;;&#39;) %&gt;%
ggplot(aes(x=PC1, y=PC2, fill=pi1)) +
  geom_raster() +
  geom_text(aes(label=signif(pi1*100, 2)), color=&quot;blue&quot;) +
  scale_fill_viridis(option=&quot;A&quot;, direction = -1, limits=c(0,1)) +
  coord_flip() +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(x=&quot;Discovery QTL phenotype&quot;, y=&quot;Phenotype assessed for overlap&quot;)</code></pre>
<p><img src="figure/20220713_PlotHeatmapManyWays_JustPi1.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>That looks better. Now let’s recode the activiting promoter marks to either distal promoter, host-gene promoter, or distal enhancer. First read in mapping of peaks to TSS</p>
<pre class="r"><code>PeaksToTSS &lt;- Sys.glob(&quot;../code/Misc/PeaksClosestToTSS/*_assigned.tsv.gz&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/Misc/PeaksClosestToTSS/(.+?)_assigned.tsv.gz&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(read_tsv) %&gt;%
  bind_rows(.id=&quot;ChromatinMark&quot;) %&gt;%
  mutate(GenePeakPair = paste(gene, peak, sep = &quot;;&quot;)) %&gt;%
  distinct(ChromatinMark, peak, gene, .keep_all=T)

GeneWindows &lt;- dat$GeneLocus %&gt;% unique()
TSSGenes &lt;- PeaksToTSS$gene %&gt;% unique()

length(GeneWindows)</code></pre>
<pre><code>[1] 13800</code></pre>
<pre class="r"><code>length(TSSGenes)</code></pre>
<pre><code>[1] 28091</code></pre>
<pre class="r"><code>intersect(GeneWindows, TSSGenes) %&gt;% length()</code></pre>
<pre><code>[1] 12903</code></pre>
<pre class="r"><code>dat.split.added.union.activating.marks &lt;- dat %&gt;%
  filter(PC1 %in% RecodeIncludePCs) %&gt;%
  filter(PC2 %in% RecodeIncludePCs) %&gt;%
  mutate(PC1 = recode(PC1, !!!RecodeVec)) %&gt;%
  mutate(PC2 = recode(PC2, !!!RecodeVec)) %&gt;%
  mutate(PC1 = case_when(
    paste(GeneLocus, P1, sep=&quot;;&quot;) %in% PeaksToTSS$GenePeakPair ~ &quot;ActivatingMark_HostGenePromoter&quot;,
    P1 %in% PeaksToTSS$peak ~ &quot;ActivatingMark_OtherGenePromoter&quot;,
    PC1 %in% c(&quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;H3K27AC&quot;) ~ &quot;ActivatingMark_Enhancer&quot;,
    TRUE ~ PC1
  )) %&gt;%
  mutate(PC2 = case_when(
    paste(GeneLocus, P2, sep=&quot;;&quot;) %in% PeaksToTSS$GenePeakPair ~ &quot;ActivatingMark_HostGenePromoter&quot;,
    P2 %in% PeaksToTSS$peak ~ &quot;ActivatingMark_OtherGenePromoter&quot;,
    PC2 %in% c(&quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;H3K27AC&quot;) ~ &quot;ActivatingMark_Enhancer&quot;,
    TRUE ~ PC2
  ))

dat.split.added.union.activating.marks &lt;-
  bind_rows(dat.split.added.union.activating.marks, 
    (dat.split.added.union.activating.marks %&gt;% filter(str_detect(PC1, &quot;ActivatingMark&quot;)) %&gt;%
    mutate(PC1 = case_when(
      str_detect(PC1, &quot;ActivatingMark&quot;) ~ &quot;AnyActivatingMark&quot;,
      TRUE ~ PC1
    ))),
    (dat.split.added.union.activating.marks %&gt;% filter(str_detect(PC2, &quot;ActivatingMark&quot;)) %&gt;%
    mutate(PC2 = case_when(
      str_detect(PC2, &quot;ActivatingMark&quot;) ~ &quot;AnyActivatingMark&quot;,
      TRUE ~ PC2
    )))) %&gt;%
  group_by(PC1, P1, PC2) %&gt;%
  mutate(test.stat.obs = -log10(min(trait.x.p.in.y))) %&gt;%
  ungroup() %&gt;%
  add_count(PC1, P1, PC2) %&gt;%
  filter(n&lt;=250) %&gt;%
  group_by(PC1, PC2) %&gt;%
  rowwise() %&gt;%
  mutate(Pvals.For.Pi1 = 1-ecdf.functions[[n]](test.stat.obs)) %&gt;%
  ungroup() %&gt;%
  select(PC1, PC2, Pvals.For.Pi1) %&gt;%
  filter(!PC1==PC2) %&gt;%
  split(paste(.$PC1, .$PC2, sep = &quot;;&quot;))


lapply(dat.split.added.union.activating.marks, CalculatePi1, pi0.method=&quot;bootstrap&quot;) %&gt;%
  unlist() %&gt;%
  data.frame(pi1=.) %&gt;%
  rownames_to_column(&quot;PC1_PC2&quot;) %&gt;%
  separate(PC1_PC2, into=c(&quot;PC1&quot;, &quot;PC2&quot;), sep=&#39;;&#39;) %&gt;%
  # filter(!(PC1==&quot;ActivatingMark&quot; | PC2 == &quot;ActivatingMark&quot;) ) %&gt;%
  mutate(TextColor = case_when(
    pi1 &gt;= 0.7 ~ &quot;white&quot;,
    TRUE ~ &quot;black&quot;
  )) %&gt;%
ggplot(aes(x=PC1, y=PC2, fill=pi1, color=TextColor)) +
  geom_raster() +
  geom_text(aes(label=signif(pi1*100, 2))) +
  scale_color_identity() +
  scale_fill_viridis(option=&quot;A&quot;, direction = -1, limits=c(0,1)) +
  coord_flip() +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(x=&quot;Discovery QTL phenotype&quot;, y=&quot;Phenotype assessed for overlap&quot;)</code></pre>
<p><img src="figure/20220713_PlotHeatmapManyWays_JustPi1.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Some of these pi1 estimates don’t make sense. When discovery QTL is “ActivatingMark_*“, pi1 assessed in”AnyActivatingMark&quot; should be 100%… Let’s check the histograms…</p>
<pre class="r"><code>dat.split.added.union.activating.marks %&gt;%
  bind_rows(.id=&quot;ID&quot;) %&gt;%
  filter(PC2 == &quot;AnyActivatingMark&quot;) %&gt;%
  split(paste(.$PC1, .$PC2, sep = &quot;;&quot;)) %&gt;%
  lapply(CalculatePi1, lambda=0.5, pi0.method=&quot;bootstrap&quot;) %&gt;%
  unlist() %&gt;%
  data.frame(pi1=.)</code></pre>
<pre><code>                                                         pi1
ActivatingMark_Enhancer;AnyActivatingMark          0.8318326
ActivatingMark_HostGenePromoter;AnyActivatingMark  0.8733038
ActivatingMark_OtherGenePromoter;AnyActivatingMark 0.7430970
Expression_Metabolic.30min;AnyActivatingMark       0.5855742
Expression_Metabolic.60min;AnyActivatingMark       0.6085198
Expression_chRNA;AnyActivatingMark                 0.5549810
Expression_polyA;AnyActivatingMark                 0.5580077
H3K36ME3;AnyActivatingMark                         0.7095374
ProCap;AnyActivatingMark                           0.5314758
Splicing.IR_chRNA;AnyActivatingMark                0.3916971
Splicing.IR_polyA;AnyActivatingMark                0.4455069
Splicing_chRNA;AnyActivatingMark                   0.3747645
Splicing_polyA;AnyActivatingMark                   0.4190953
ncRNA_chRNA;AnyActivatingMark                      0.4601645</code></pre>
<pre class="r"><code>QvalueObj &lt;- dat.split.added.union.activating.marks %&gt;%
  bind_rows(.id=&quot;ID&quot;) %&gt;%
  filter(PC1 == &quot;ActivatingMark_Enhancer&quot; &amp; PC2 == &quot;AnyActivatingMark&quot;) %&gt;%
  pull(Pvals.For.Pi1) %&gt;%
  pi0est(pi0.method=&quot;bootstrap&quot;)

plot(QvalueObj$lambda, QvalueObj$pi0.lambda)</code></pre>
<p><img src="figure/20220713_PlotHeatmapManyWays_JustPi1.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dat.split.added.union.activating.marks %&gt;%
  bind_rows(.id=&quot;ID&quot;) %&gt;%
  filter(str_detect(PC1, &quot;ActivatingMark&quot;) &amp; str_detect(PC2, &quot;ActivatingMark&quot;)) %&gt;%
  ggplot(aes(x=Pvals.For.Pi1)) +
  geom_histogram() +
  facet_wrap(~ID, scales = &quot;free_y&quot;) +
  theme_bw()</code></pre>
<p><img src="figure/20220713_PlotHeatmapManyWays_JustPi1.Rmd/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dat.split.added.union.activating.marks %&gt;%
  bind_rows(.id=&quot;ID&quot;) %&gt;%
  filter(PC1==&quot;Expression_Metabolic.30min&quot;, str_detect(PC2, &quot;ActivatingMark&quot;)) %&gt;%
  ggplot(aes(x=Pvals.For.Pi1)) +
  geom_histogram() +
  facet_wrap(~ID, scales = &quot;free_y&quot;) +
  theme_bw()</code></pre>
<p><img src="figure/20220713_PlotHeatmapManyWays_JustPi1.Rmd/unnamed-chunk-10-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Hmm, I suspect this is another problem like I described before where there are many non-signifant non-independent tests that make the Pvalue distribution a tad conservative when there is a one-to-many phenotype mapping. I note that Metabolic.30min vs Metabolic.60min don’t seem to have this problem, as the pi1 staistic is near 100%, and it doesn’t have a the one-to-many mapping problem. Well I am not too concerned for now, I will just have to remember that when there are many ascertainment phenotypes that get tested against each discovery, that the pi1 estimates may be a bit convervatice (smaller than truth).</p>
<p>Ok now make the plot with different FDR thresholds:</p>
<pre class="r"><code># Make a function to re-use

CalculatePi1Matrix &lt;- function(dat, DiscoveryFDR=0.1){
  dat.split.added.union.activating.marks &lt;- dat %&gt;%
    filter(FDR.x &lt;= DiscoveryFDR) %&gt;%
    filter(PC1 %in% RecodeIncludePCs) %&gt;%
    filter(PC2 %in% RecodeIncludePCs) %&gt;%
    mutate(PC1 = recode(PC1, !!!RecodeVec)) %&gt;%
    mutate(PC2 = recode(PC2, !!!RecodeVec)) %&gt;%
    mutate(PC1 = case_when(
      paste(GeneLocus, P1, sep=&quot;;&quot;) %in% PeaksToTSS$GenePeakPair ~ &quot;ActivatingMark_HostGenePromoter&quot;,
      P1 %in% PeaksToTSS$peak ~ &quot;ActivatingMark_OtherGenePromoter&quot;,
      PC1 %in% c(&quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;H3K27AC&quot;) ~ &quot;ActivatingMark_Enhancer&quot;,
      TRUE ~ PC1
    )) %&gt;%
    mutate(PC2 = case_when(
      paste(GeneLocus, P2, sep=&quot;;&quot;) %in% PeaksToTSS$GenePeakPair ~ &quot;ActivatingMark_HostGenePromoter&quot;,
      P2 %in% PeaksToTSS$peak ~ &quot;ActivatingMark_OtherGenePromoter&quot;,
      PC2 %in% c(&quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;H3K27AC&quot;) ~ &quot;ActivatingMark_Enhancer&quot;,
      TRUE ~ PC2
    ))
  
  dat.split.added.union.activating.marks &lt;-
    bind_rows(dat.split.added.union.activating.marks, 
      (dat.split.added.union.activating.marks %&gt;% filter(str_detect(PC1, &quot;ActivatingMark&quot;)) %&gt;%
      mutate(PC1 = case_when(
        str_detect(PC1, &quot;ActivatingMark&quot;) ~ &quot;AnyActivatingMark&quot;,
        TRUE ~ PC1
      ))),
      (dat.split.added.union.activating.marks %&gt;% filter(str_detect(PC2, &quot;ActivatingMark&quot;)) %&gt;%
      mutate(PC2 = case_when(
        str_detect(PC2, &quot;ActivatingMark&quot;) ~ &quot;AnyActivatingMark&quot;,
        TRUE ~ PC2
      )))) %&gt;%
    group_by(PC1, P1, PC2) %&gt;%
    mutate(test.stat.obs = -log10(min(trait.x.p.in.y))) %&gt;%
    ungroup() %&gt;%
    add_count(PC1, P1, PC2) %&gt;%
    filter(n&lt;=250) %&gt;%
    group_by(PC1, PC2) %&gt;%
    rowwise() %&gt;%
    mutate(Pvals.For.Pi1 = 1-ecdf.functions[[n]](test.stat.obs)) %&gt;%
    ungroup() %&gt;%
    select(PC1, PC2, Pvals.For.Pi1) %&gt;%
    filter(!PC1==PC2) %&gt;%
    split(paste(.$PC1, .$PC2, sep = &quot;;&quot;))
  
  dat.pi1 &lt;- lapply(dat.split.added.union.activating.marks, CalculatePi1, pi0.method=&quot;bootstrap&quot;) %&gt;%
    unlist() %&gt;%
    data.frame(pi1=.) %&gt;%
    rownames_to_column(&quot;PC1_PC2&quot;) %&gt;%
    separate(PC1_PC2, into=c(&quot;PC1&quot;, &quot;PC2&quot;), sep=&#39;;&#39;) %&gt;%
    mutate(DiscoveryFDR = DiscoveryFDR) %&gt;%
    mutate(TextColor = case_when(
      pi1 &gt;= 0.7 ~ &quot;white&quot;,
      TRUE ~ &quot;black&quot;
    ))
return(dat.pi1)
}

dat.pi.AtThresholds &lt;- bind_rows(
  CalculatePi1Matrix(dat, DiscoveryFDR=0.1),
  CalculatePi1Matrix(dat, DiscoveryFDR=0.05),
  CalculatePi1Matrix(dat, DiscoveryFDR=0.01)
)

pi.heatmap &lt;- ggplot(dat.pi.AtThresholds, aes(x=PC1, y=PC2, fill=pi1, color=TextColor)) +
  geom_raster() +
  geom_text(aes(label=signif(pi1*100, 2))) +
  scale_fill_viridis(option=&quot;A&quot;, direction = -1, limits=c(0,1)) +
  scale_color_identity() +
  facet_wrap(~DiscoveryFDR) +
  coord_flip() +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(x=&quot;Discovery QTL phenotype&quot;, y=&quot;Phenotype assessed for overlap&quot;, &quot;Pi1 of xQTL among Discovery QTL classes&quot;)
pi.heatmap</code></pre>
<p><img src="figure/20220713_PlotHeatmapManyWays_JustPi1.Rmd/unnamed-chunk-11-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>Consistent with the one-to-many explanation of pi1 being overly conservatice, when there are less phenotypes to map to one (ie when the 0.01 facet), the ActinvatingMark_* to AnyActivatingMark approaches 100%.</p>
<div id="update-lets-break-out-ncrna-into-some-subcategories" class="section level3">
<h3>Update: let’s break out ncRNA into some subcategories</h3>
<pre class="r"><code>dat %&gt;% pull(PC1) %&gt;% unique()</code></pre>
<pre><code> [1] &quot;chRNA.Expression_cheRNA&quot;             &quot;chRNA.Expression_eRNA&quot;              
 [3] &quot;chRNA.Expression_lncRNA&quot;             &quot;chRNA.Expression_snoRNA&quot;            
 [5] &quot;chRNA.Expression.Splicing&quot;           &quot;chRNA.IER&quot;                          
 [7] &quot;chRNA.IRjunctions&quot;                   &quot;chRNA.IR&quot;                           
 [9] &quot;chRNA.Slopes&quot;                        &quot;chRNA.Splicing&quot;                     
[11] &quot;CTCF&quot;                                &quot;Expression.Splicing&quot;                
[13] &quot;Expression.Splicing.Subset_YRI&quot;      &quot;H3K27AC&quot;                            
[15] &quot;H3K36ME3&quot;                            &quot;H3K4ME1&quot;                            
[17] &quot;H3K4ME3&quot;                             &quot;MetabolicLabelled.30min.IER&quot;        
[19] &quot;MetabolicLabelled.30min.IRjunctions&quot; &quot;MetabolicLabelled.30min.IR&quot;         
[21] &quot;MetabolicLabelled.30min&quot;             &quot;MetabolicLabelled.30min.Splicing&quot;   
[23] &quot;MetabolicLabelled.60min.IER&quot;         &quot;MetabolicLabelled.60min.IRjunctions&quot;
[25] &quot;MetabolicLabelled.60min.IR&quot;          &quot;MetabolicLabelled.60min&quot;            
[27] &quot;MetabolicLabelled.60min.Splicing&quot;    &quot;polyA.IER&quot;                          
[29] &quot;polyA.IRjunctions&quot;                   &quot;polyA.IR&quot;                           
[31] &quot;polyA.IR.Subset_YRI&quot;                 &quot;polyA.Splicing&quot;                     
[33] &quot;polyA.Splicing.Subset_YRI&quot;           &quot;ProCap&quot;                             </code></pre>
<p>Ok, I will have to update this.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] pROC_1.15.0       GGally_1.4.0      qvalue_2.16.0     data.table_1.14.2
 [5] gplots_3.0.1.1    viridis_0.5.1     viridisLite_0.3.0 forcats_0.4.0    
 [9] stringr_1.4.0     dplyr_1.0.9       purrr_0.3.4       readr_1.3.1      
[13] tidyr_1.2.0       tibble_3.1.7      ggplot2_3.3.6     tidyverse_1.3.0  

loaded via a namespace (and not attached):
 [1] httr_1.4.4         jsonlite_1.6       splines_3.6.1      R.utils_2.9.0     
 [5] modelr_0.1.8       gtools_3.9.2.2     assertthat_0.2.1   highr_0.9         
 [9] cellranger_1.1.0   yaml_2.2.0         pillar_1.7.0       backports_1.4.1   
[13] glue_1.6.2         digest_0.6.20      RColorBrewer_1.1-2 promises_1.0.1    
[17] rvest_0.3.5        colorspace_1.4-1   R.oo_1.22.0        htmltools_0.5.3   
[21] httpuv_1.5.1       plyr_1.8.4         pkgconfig_2.0.2    broom_1.0.0       
[25] haven_2.3.1        scales_1.1.0       gdata_2.18.0       later_0.8.0       
[29] git2r_0.26.1       farver_2.1.0       generics_0.1.3     ellipsis_0.3.2    
[33] withr_2.5.0        cli_3.3.0          magrittr_1.5       crayon_1.3.4      
[37] readxl_1.3.1       evaluate_0.15      R.methodsS3_1.7.1  fs_1.5.2          
[41] fansi_0.4.0        xml2_1.3.2         tools_3.6.1        hms_0.5.3         
[45] lifecycle_1.0.1    munsell_0.5.0      reprex_0.3.0       compiler_3.6.1    
[49] caTools_1.17.1.2   rlang_1.0.5        grid_3.6.1         rstudioapi_0.14   
[53] labeling_0.3       bitops_1.0-6       rmarkdown_1.13     gtable_0.3.0      
[57] DBI_1.1.0          reshape_0.8.8      reshape2_1.4.3     R6_2.4.0          
[61] gridExtra_2.3      lubridate_1.7.4    knitr_1.39         fastmap_1.1.0     
[65] utf8_1.1.4         workflowr_1.6.2    rprojroot_2.0.2    KernSmooth_2.23-15
[69] stringi_1.4.3      Rcpp_1.0.5         vctrs_0.4.1        dbplyr_1.4.2      
[73] tidyselect_1.1.2   xfun_0.31         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
