<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Does intron retention down regulate gene expression</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Does intron retention down regulate gene expression</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-10-12
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>ChromatinSplicingQTLs/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstreeae6409c23e6e7fb2acd20f0c180a593ab3a0c7f6targetblankae6409ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/ae6409c23e6e7fb2acd20f0c180a593ab3a0c7f6" target="_blank">ae6409c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstreeae6409c23e6e7fb2acd20f0c180a593ab3a0c7f6targetblankae6409ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/ae6409c23e6e7fb2acd20f0c180a593ab3a0c7f6" target="_blank">ae6409c</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    code/%
    Ignored:    code/.DS_Store
    Ignored:    code/.RData
    Ignored:    code/._.DS_Store
    Ignored:    code/._README.md
    Ignored:    code/._report.html
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    code/.snakemake/
    Ignored:    code/APA_Processing/
    Ignored:    code/Alignments/
    Ignored:    code/ChromHMM/
    Ignored:    code/ENCODE/
    Ignored:    code/ExpressionAnalysis/
    Ignored:    code/FastqFastp/
    Ignored:    code/FastqFastpSE/
    Ignored:    code/Genotypes/
    Ignored:    code/IntronSlopes/
    Ignored:    code/Metaplots/
    Ignored:    code/Misc/
    Ignored:    code/MiscCountTables/
    Ignored:    code/Multiqc/
    Ignored:    code/Multiqc_chRNA/
    Ignored:    code/NonCodingRNA_annotation/
    Ignored:    code/PeakCalling/
    Ignored:    code/Phenotypes/
    Ignored:    code/PlotGruberQTLs/
    Ignored:    code/PlotQTLs/
    Ignored:    code/ProCapAnalysis/
    Ignored:    code/QC/
    Ignored:    code/QTL_SNP_Enrichment/
    Ignored:    code/QTLs/
    Ignored:    code/ReferenceGenome/
    Ignored:    code/Rplots.pdf
    Ignored:    code/Session.vim
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TODO
    Ignored:    code/Tehranchi/
    Ignored:    code/bigwigs/
    Ignored:    code/bigwigs_FromNonWASPFilteredReads/
    Ignored:    code/config/.DS_Store
    Ignored:    code/config/._.DS_Store
    Ignored:    code/config/.ipynb_checkpoints/
    Ignored:    code/debug.ipynb
    Ignored:    code/debug_python.ipynb
    Ignored:    code/deepTools/
    Ignored:    code/featureCounts/
    Ignored:    code/gwas_summary_stats/
    Ignored:    code/hyprcoloc/
    Ignored:    code/igv_session.xml
    Ignored:    code/log
    Ignored:    code/logs/
    Ignored:    code/notebooks/.ipynb_checkpoints/
    Ignored:    code/rules/.ipynb_checkpoints/
    Ignored:    code/rules/OldRules/
    Ignored:    code/rules/notebooks/
    Ignored:    code/scratch/
    Ignored:    code/scripts/.ipynb_checkpoints/
    Ignored:    code/scripts/GTFtools_0.8.0/
    Ignored:    code/scripts/__pycache__/
    Ignored:    code/scripts/liftOverBedpe/liftOverBedpe.py
    Ignored:    code/snakemake.log
    Ignored:    code/snakemake.sbatch.log
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/._20220414203249_JASPAR2022_combined_matrices_25818_jaspar.txt
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-10.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-11.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-2.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-3.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-4.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-5.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-6.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-7.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-8.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022.csv

Untracked files:
    Untracked:  analysis/20221011_PlotHeatmapManyWays_ncRNA_Updated.Rmd
    Untracked:  analysis/20221012_IntronRetentionAndExpressionConcordance.Rmd
    Untracked:  code/rules/Metaplots.smk
    Untracked:  code/scripts/MakeBigwigList.R
    Untracked:  code/scripts/TidyDeeptoolsComputeMatrix.R
    Untracked:  code/snakemake_profiles/slurm/__pycache__/

Unstaged changes:
    Modified:   code/Snakefile
    Modified:   code/rules/CalculatePi1.smk
    Modified:   code/rules/QTLTools.smk
    Modified:   code/scripts/CalculatePi1_GetAscertainmentP_AllPairs.py
    Modified:   code/scripts/CalculatePi1_GetTraitPairs_AllTraits.R
    Modified:   code/scripts/GenometracksByGenotype
    Modified:   data/ColorsForPhenotypes.xlsx
    Modified:   data/Phenotypes_recode_for_Plotting.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="intro" class="section level2">
<h2>Intro</h2>
<p>I previously showed that effect size directions for intron retention QTLs puzzingly positively correct with eQTLs (more intron retention = more expression). Somehow I think intron retention QTLs are often picking up on chromatin effects. Let’s break up the intron retention QTLs into groups by the location of the SNP (in a enhancer/promoter vs a splice site) and reassess the concordance of expression effects. I can also look at the same idea with normal leafcutter sQTLs, comparing introns by their annotation type. For example increase in splicing of annotated or “basic” tagged introns might generally increase expression, versus increase in unannoated or “NMD” tagged introns might decrease expression.</p>
<pre class="r"><code>library(tidyverse)</code></pre>
<p>First let’s make some basic plots establishing that chRNA has more unannoated and NMD-specific splicing.</p>
<pre class="r"><code>#TODO</code></pre>
<p>Let’s</p>
<pre class="r"><code>PhenotypeAliases &lt;- read_tsv(&quot;../data/Phenotypes_recode_for_Plotting.txt&quot;)

PC.ShortAliases &lt;- PhenotypeAliases %&gt;%
  dplyr::select(PC, ShorterAlias) %&gt;% deframe()

coloc.results &lt;- read_tsv(&quot;../code/hyprcoloc/Results/ForColoc/MolColocStandard/results.txt.gz&quot;)

coloc.results.tidycolocalized &lt;- read_tsv(&quot;../code/hyprcoloc/Results/ForColoc/MolColocStandard/tidy_results_OnlyColocalized.txt.gz&quot;) %&gt;%
  separate(phenotype_full, into=c(&quot;PC&quot;, &quot;P&quot;), sep=&quot;;&quot;)

finemap.snps.annotated &lt;- read_tsv(&quot;../code/QTL_SNP_Enrichment/FinemapIntersections/MolColocStandard.bed.gz&quot;, col_names=c(&quot;SNPchrom&quot;, &quot;SNPstart&quot;, &quot;SNPstop&quot;, &quot;SNP_iteration_locus&quot;, &quot;FinemapPP&quot;, &quot;AnnotationChrom&quot;, &quot;AnnotationStart&quot;, &quot;AnnotatioStop&quot;, &quot;AnnotationClass&quot;, &quot;Overlap&quot;)) %&gt;%
  dplyr::select(-Overlap)</code></pre>
<p>Count chromatin-colocalizing and splicing-colocalizing eQTLs, and recreate previous observation about concordant effects</p>
<pre class="r"><code>coloc.results.tidycolocalized %&gt;%
  group_by(Locus, snp) %&gt;%
  filter(any(PC==&quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
  summarise(
    ContainsChromatinEqtl = any(PC %in% c(&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;)),
    ContainsSqtl = any(PC %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;))
    ) %&gt;%
  ggplot(aes(x=1, fill=paste(ContainsChromatinEqtl, ContainsSqtl))) +
  geom_bar() +
  labs(title=&quot;More chromatin localization with eQTLs than splicing&quot;, y=&quot;Number of colocalizing eQTLs&quot;) +
  theme_classic() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>coloc.results.tidycolocalized %&gt;%
  group_by(Locus, snp) %&gt;%
  filter(any(PC==&quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
  ungroup() %&gt;%
  filter(PC %in% c(&quot;Expression.Splicing.Subset_YRI&quot;,&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;)) %&gt;%
  left_join(., ., by=c(&quot;Locus&quot;, &quot;snp&quot;)) %&gt;%
  filter(!((P.x == P.y) &amp; (PC.x == PC.y))) %&gt;% 
  group_by(Locus, snp) %&gt;%
  mutate(
    ContainsChromatinEqtl = any(PC.x %in% c(&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;)),
    ContainsSqtl = any(PC.x %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;))
    ) %&gt;%
  ungroup() %&gt;%
  mutate(Contains.eQTL_Contains.sQTL = paste(ContainsChromatinEqtl, ContainsSqtl)) %&gt;%
  group_by(Contains.eQTL_Contains.sQTL, PC.x, PC.y) %&gt;%
  summarise(cor = cor(beta.x, beta.y, method=&quot;spearman&quot;)) %&gt;%
  mutate(PC.x = recode(PC.x, !!!PC.ShortAliases)) %&gt;%
  mutate(PC.y = recode(PC.y, !!!PC.ShortAliases)) %&gt;%
  ggplot(aes(x=PC.x, y=PC.y, fill=cor)) +
  geom_raster() +
  scale_fill_gradient2() +
  facet_wrap(~Contains.eQTL_Contains.sQTL) +
  scale_x_discrete(expand=c(0,0)) +
  scale_y_discrete(expand=c(0,0)) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  labs(x=&quot;TraitA Phenotype class&quot;, y=&quot;TraitB Phenotype class&quot;, fill=&quot;Spearman cor&quot;, title=&quot;Effect size correlation&quot;,
       caption = &quot;FacetTitle: IsChromatinQTL IsSplicingQTL&quot;)</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Now split QTLs by location of SNP as either in splice site, in enhancer/promoter, or neither</p>
<pre class="r"><code>#annotation types
finemap.snps.annotated$AnnotationClass %&gt;% unique()</code></pre>
<pre><code> [1] &quot;SpliceBranchpointRegion_0&quot; &quot;10_Txn_Elongation&quot;        
 [3] &quot;6_Weak_Enhancer&quot;           &quot;SpliceDonor_0&quot;            
 [5] &quot;2_Weak_Promoter&quot;           &quot;11_Weak_Txn&quot;              
 [7] &quot;1_Active_Promoter&quot;         &quot;12_Repressed&quot;             
 [9] &quot;4_Strong_Enhancer&quot;         &quot;13_Heterochrom/lo&quot;        
[11] &quot;SpliceAcceptor_0&quot;          &quot;SpliceBranchpointRegion_1&quot;
[13] &quot;ncRNA_coRNA&quot;               &quot;ncRNA_pseudo&quot;             
[15] &quot;9_Txn_Transition&quot;          &quot;SpliceAcceptor_1&quot;         
[17] &quot;ncRNA_srtRNA&quot;              &quot;8_Insulator&quot;              
[19] &quot;ncRNA_uaRNA&quot;               &quot;5_Strong_Enhancer&quot;        
[21] &quot;7_Weak_Enhancer&quot;           &quot;SpliceDonor_1&quot;            
[23] &quot;PAS_Region&quot;                &quot;ncRNA_incRNA&quot;             
[25] &quot;3_Poised_Promoter&quot;         &quot;.&quot;                        
[27] &quot;14_Repetitive/CNV&quot;         &quot;ncRNA_ctRNA&quot;              
[29] &quot;15_Repetitive/CNV&quot;         &quot;ncRNA_lncRNA&quot;             
[31] &quot;ncRNA_rtRNA&quot;               &quot;ncRNA_snoRNA&quot;             </code></pre>
<pre class="r"><code>coloc.results.tidycolocalized %&gt;%
  group_by(Locus, snp) %&gt;%
  filter(any(PC==&quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
  ungroup() %&gt;%
  filter(PC %in% c(&quot;Expression.Splicing.Subset_YRI&quot;,&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;)) %&gt;%
  left_join(., ., by=c(&quot;Locus&quot;, &quot;snp&quot;)) %&gt;%
  filter(!((P.x == P.y) &amp; (PC.x == PC.y))) %&gt;% 
  group_by(Locus, snp) %&gt;%
  mutate(
    ContainsChromatinEqtl = any(PC.x %in% c(&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;)),
    ContainsSqtl = any(PC.x %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;))
    ) %&gt;%
  ungroup() %&gt;%
  mutate(Contains.eQTL_Contains.sQTL = paste(ContainsChromatinEqtl, ContainsSqtl)) %&gt;%
  dplyr::select(-iteration.y) %&gt;%
  left_join(
    finemap.snps.annotated %&gt;%
      dplyr::select(SNP_iteration_locus, FinemapPP, AnnotationClass) %&gt;%
      separate(SNP_iteration_locus, into=c(&quot;snp&quot;, &quot;iteration.x&quot;, &quot;Locus&quot;), convert=T, sep=&quot;_&quot;)
  ) %&gt;%
  filter(FinemapPP &gt;0.25) %&gt;%
  mutate(AnnotationSuperclass = case_when(
    str_detect(AnnotationClass, &quot;Splice.+_1&quot;) ~ &quot;Annotated SS&quot;,
    str_detect(AnnotationClass, &quot;Splice.+_0$&quot;) ~ &quot;Unannotated SS&quot;,
    str_detect(AnnotationClass, &quot;Enhancer&quot;) ~ &quot;Enhancer&quot;,
    str_detect(AnnotationClass, &quot;Promoter&quot;) ~ &quot;Promoter&quot;,
    TRUE ~ &quot;Other&quot;
  )) %&gt;%
  mutate(PC.x = recode(PC.x, !!!PC.ShortAliases)) %&gt;%
  mutate(PC.y = recode(PC.y, !!!PC.ShortAliases)) %&gt;%
  mutate(Contains.eQTL_Contains.sQTL = recode(Contains.eQTL_Contains.sQTL,
                                              !!!c(&quot;TRUE TRUE&quot;=&quot;hQTL+sQTL&quot;,
                                                   &quot;TRUE FALSE&quot;=&quot;hQTL only&quot;,
                                                   &quot;FALSE TRUE&quot;=&quot;sQTL only&quot;))) %&gt;%
  group_by(Contains.eQTL_Contains.sQTL, PC.x, PC.y, AnnotationSuperclass) %&gt;%
  summarise(cor = cor(beta.x, beta.y, method=&quot;spearman&quot;), n=n()) %&gt;%
  ggplot(aes(x=PC.x, y=PC.y, fill=cor)) +
  geom_raster() +
  geom_text(aes(label=n), size=1.5) +
  scale_fill_gradient2() +
  facet_grid(rows=vars(Contains.eQTL_Contains.sQTL), cols=vars(AnnotationSuperclass)) +
  scale_x_discrete(expand=c(0,0)) +
  scale_y_discrete(expand=c(0,0)) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  labs(x=&quot;TraitA Phenotype class&quot;, y=&quot;TraitB Phenotype class&quot;, fill=&quot;Spearman cor&quot;, title=&quot;Effect size correlation for eQTLs that coloc with something&quot;,
       caption = str_wrap(&quot;FacetCols=Annotation class of SNP (Onlyfinemap PP &gt;0.25 included). FacetRows=Type of eQTL&quot;))</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>I’m quite surprised just by the numbers of hQTL eQTLs with high finemap PP in unannoated splice site regions. This is clearly a really big annotation set that also includes lots of enhancer regions. This unannoated splice site region was defined as at least one spliced read mapping across all the data. Perhaps I shouldn’t be surprised, as any region is sufficeintly transcribed (including enhancers) will at some rate have some splice sites.</p>
<p>And even among the 7 intron retention QTLs that coloc with an eQTL but not hQTL, and are finemapped to annotated splice sites, the the direction of effects is if anything opposite what I was expecting</p>
<pre class="r"><code>coloc.results.tidycolocalized %&gt;%
  group_by(Locus, snp) %&gt;%
  filter(any(PC==&quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
  ungroup() %&gt;%
  filter(PC %in% c(&quot;Expression.Splicing.Subset_YRI&quot;,&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;)) %&gt;%
  left_join(., ., by=c(&quot;Locus&quot;, &quot;snp&quot;)) %&gt;%
  filter(!((P.x == P.y) &amp; (PC.x == PC.y))) %&gt;% 
  group_by(Locus, snp) %&gt;%
  mutate(
    ContainsChromatinEqtl = any(PC.x %in% c(&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;)),
    ContainsSqtl = any(PC.x %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;))
    ) %&gt;%
  ungroup() %&gt;%
  mutate(Contains.eQTL_Contains.sQTL = paste(ContainsChromatinEqtl, ContainsSqtl)) %&gt;%
  dplyr::select(-iteration.y) %&gt;%
  left_join(
    finemap.snps.annotated %&gt;%
      dplyr::select(SNP_iteration_locus, FinemapPP, AnnotationClass) %&gt;%
      separate(SNP_iteration_locus, into=c(&quot;snp&quot;, &quot;iteration.x&quot;, &quot;Locus&quot;), convert=T, sep=&quot;_&quot;)
  ) %&gt;%
  filter(FinemapPP &gt;0.25) %&gt;%
  mutate(AnnotationSuperclass = case_when(
    str_detect(AnnotationClass, &quot;Splice.+_1&quot;) ~ &quot;Annotated SS&quot;,
    str_detect(AnnotationClass, &quot;Splice.+_0$&quot;) ~ &quot;Unannotated SS&quot;,
    str_detect(AnnotationClass, &quot;Enhancer&quot;) ~ &quot;Enhancer&quot;,
    str_detect(AnnotationClass, &quot;Promoter&quot;) ~ &quot;Promoter&quot;,
    TRUE ~ &quot;Other&quot;
  )) %&gt;%
  mutate(PC.x = recode(PC.x, !!!PC.ShortAliases)) %&gt;%
  mutate(PC.y = recode(PC.y, !!!PC.ShortAliases)) %&gt;%
  mutate(Contains.eQTL_Contains.sQTL = recode(Contains.eQTL_Contains.sQTL,
                                              !!!c(&quot;TRUE TRUE&quot;=&quot;hQTL+sQTL&quot;,
                                                   &quot;TRUE FALSE&quot;=&quot;hQTL only&quot;,
                                                   &quot;FALSE TRUE&quot;=&quot;sQTL only&quot;))) %&gt;%
  filter(Contains.eQTL_Contains.sQTL==&quot;sQTL only&quot; &amp; AnnotationSuperclass==&quot;Annotated SS&quot;) %&gt;%
  ggplot(aes(x=beta.x, y=beta.y)) +
  geom_point() +
  geom_vline(xintercept=0) +
  geom_hline(yintercept=0) +
  facet_grid(rows=vars(PC.x), cols=vars(PC.y)) +
  theme_bw() +
  labs(x=&quot;TraitA Phenotype class&quot;, y=&quot;TraitB Phenotype class&quot;, title=&quot;Effect size correlation for eQTLs/sQTLs that are not hQTLs&quot;,
       caption = str_wrap(&quot;Only SNPs with finemap PP &gt; 0.25 in annotated splice site SNP included&quot;))</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Note there are only a few points of chRNA.IR intron retention with SNPs in splice sites that coloc with eQTL.</p>
<p>Let’s also break the original plot by ncRNAs for SNP regions</p>
<pre class="r"><code>coloc.results.tidycolocalized %&gt;%
  group_by(Locus, snp) %&gt;%
  filter(any(PC==&quot;Expression.Splicing.Subset_YRI&quot;)) %&gt;%
  ungroup() %&gt;%
  filter(PC %in% c(&quot;Expression.Splicing.Subset_YRI&quot;,&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;)) %&gt;%
  left_join(., ., by=c(&quot;Locus&quot;, &quot;snp&quot;)) %&gt;%
  filter(!((P.x == P.y) &amp; (PC.x == PC.y))) %&gt;% 
  group_by(Locus, snp) %&gt;%
  mutate(
    ContainsChromatinEqtl = any(PC.x %in% c(&quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;)),
    ContainsSqtl = any(PC.x %in% c(&quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;, &quot;chRNA.IER&quot;))
    ) %&gt;%
  ungroup() %&gt;%
  mutate(Contains.eQTL_Contains.sQTL = paste(ContainsChromatinEqtl, ContainsSqtl)) %&gt;%
  dplyr::select(-iteration.y) %&gt;%
  left_join(
    finemap.snps.annotated %&gt;%
      dplyr::select(SNP_iteration_locus, FinemapPP, AnnotationClass) %&gt;%
      separate(SNP_iteration_locus, into=c(&quot;snp&quot;, &quot;iteration.x&quot;, &quot;Locus&quot;), convert=T, sep=&quot;_&quot;)
  ) %&gt;%
  filter(FinemapPP &gt;0.25) %&gt;%
  mutate(AnnotationSuperclass = case_when(
    str_detect(AnnotationClass, &quot;Splice.+_1&quot;) ~ &quot;Annotated SS&quot;,
    str_detect(AnnotationClass, &quot;Splice.+_0$&quot;) ~ &quot;Unannotated SS&quot;,
    str_detect(AnnotationClass, &quot;Enhancer&quot;) ~ &quot;Enhancer&quot;,
    str_detect(AnnotationClass, &quot;Promoter&quot;) ~ &quot;Promoter&quot;,
    str_detect(AnnotationClass, &quot;ncRNA&quot;) ~ AnnotationClass,
    TRUE ~ &quot;Other&quot;
  )) %&gt;%
  group_by(Contains.eQTL_Contains.sQTL, PC.x, PC.y, AnnotationSuperclass) %&gt;%
  summarise(cor = cor(beta.x, beta.y, method=&quot;spearman&quot;), n=n()) %&gt;%
  mutate(PC.x = recode(PC.x, !!!PC.ShortAliases)) %&gt;%
  mutate(PC.y = recode(PC.y, !!!PC.ShortAliases)) %&gt;%
  mutate(Contains.eQTL_Contains.sQTL = recode(Contains.eQTL_Contains.sQTL,
                                              !!!c(&quot;TRUE TRUE&quot;=&quot;hQTL+sQTL&quot;,
                                                   &quot;TRUE FALSE&quot;=&quot;hQTL only&quot;,
                                                   &quot;FALSE TRUE&quot;=&quot;sQTL only&quot;))) %&gt;%
  ggplot(aes(x=PC.x, y=PC.y, fill=cor)) +
  geom_raster() +
  geom_text(aes(label=n), size=1.5) +
  scale_fill_gradient2() +
  facet_grid(rows=vars(Contains.eQTL_Contains.sQTL), cols=vars(AnnotationSuperclass)) +
  scale_x_discrete(expand=c(0,0)) +
  scale_y_discrete(expand=c(0,0)) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  labs(x=&quot;TraitA Phenotype class&quot;, y=&quot;TraitB Phenotype class&quot;, fill=&quot;Spearman cor&quot;, title=&quot;Effect size correlation for eQTLs that coloc with something&quot;,
       caption = str_wrap(&quot;FacetCols=Annotation class of SNP (Onlyfinemap PP &gt;0.25 included). FacetRows=Type of eQTL&quot;))</code></pre>
<p><img src="figure/20221012_IntronRetentionAndExpressionConcordance.Rmd/unnamed-chunk-8-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>Perhaps we should plot the same ideas but breaking out the introns into classes by whether we have some good reason to think it would be an NMD target (ie unannoated, and or annotated with nonsense mediated decay transcript tag)</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] forcats_0.4.0   stringr_1.4.0   dplyr_1.0.9     purrr_0.3.4    
[5] readr_1.3.1     tidyr_1.2.0     tibble_3.1.7    ggplot2_3.3.6  
[9] tidyverse_1.3.0

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.2 xfun_0.31        haven_2.3.1      colorspace_1.4-1
 [5] vctrs_0.4.1      generics_0.1.3   htmltools_0.5.3  yaml_2.2.0      
 [9] utf8_1.1.4       rlang_1.0.5      later_0.8.0      pillar_1.7.0    
[13] withr_2.5.0      glue_1.6.2       DBI_1.1.0        dbplyr_1.4.2    
[17] readxl_1.3.1     modelr_0.1.8     lifecycle_1.0.1  cellranger_1.1.0
[21] munsell_0.5.0    gtable_0.3.0     workflowr_1.6.2  rvest_0.3.5     
[25] evaluate_0.15    labeling_0.3     knitr_1.39       fastmap_1.1.0   
[29] httpuv_1.5.1     fansi_0.4.0      highr_0.9        broom_1.0.0     
[33] Rcpp_1.0.5       promises_1.0.1   backports_1.4.1  scales_1.1.0    
[37] jsonlite_1.6     farver_2.1.0     fs_1.5.2         hms_0.5.3       
[41] digest_0.6.20    stringi_1.4.3    rprojroot_2.0.2  grid_3.6.1      
[45] cli_3.3.0        tools_3.6.1      magrittr_1.5     crayon_1.3.4    
[49] pkgconfig_2.0.2  ellipsis_0.3.2   xml2_1.3.2       reprex_0.3.0    
[53] lubridate_1.7.4  assertthat_0.2.1 rmarkdown_1.13   httr_1.4.4      
[57] rstudioapi_0.14  R6_2.4.0         git2r_0.26.1     compiler_3.6.1  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
