<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2023-03-20" />

<title>20230320_RemakeEffectSizeScatters_UpdatedAnnotations</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">20230320_RemakeEffectSizeScatters_UpdatedAnnotations</h1>
<h4 class="date">2023-03-20</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-03-23
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>ChromatinSplicingQTLs/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstreeea58d8908cc2d094ee602c35ff2563d663680c3dtargetblankea58d89a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/ea58d8908cc2d094ee602c35ff2563d663680c3d" target="_blank">ea58d89</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstreeea58d8908cc2d094ee602c35ff2563d663680c3dtargetblankea58d89a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/ea58d8908cc2d094ee602c35ff2563d663680c3d" target="_blank">ea58d89</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store
    Ignored:    code/.RData
    Ignored:    code/._report.html
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    code/.snakemake/
    Ignored:    code/APA_Processing/
    Ignored:    code/Alignments/
    Ignored:    code/ChromHMM/
    Ignored:    code/ENCODE/
    Ignored:    code/ExpressionAnalysis/
    Ignored:    code/FastqFastp/
    Ignored:    code/FastqFastpSE/
    Ignored:    code/FastqSE/
    Ignored:    code/Genotypes/
    Ignored:    code/H3K36me3_CutAndTag.pdf
    Ignored:    code/IntronSlopes/
    Ignored:    code/LR.bed
    Ignored:    code/LR.seq.bed
    Ignored:    code/LongReads/
    Ignored:    code/Metaplots/
    Ignored:    code/Misc/
    Ignored:    code/MiscCountTables/
    Ignored:    code/Multiqc/
    Ignored:    code/Multiqc_chRNA/
    Ignored:    code/NonCodingRNA/
    Ignored:    code/NonCodingRNA_annotation/
    Ignored:    code/PairwisePi1Traits.P.all.txt.gz
    Ignored:    code/PeakCalling/
    Ignored:    code/Phenotypes/
    Ignored:    code/PlotGruberQTLs/
    Ignored:    code/PlotQTLs/
    Ignored:    code/ProCapAnalysis/
    Ignored:    code/QC/
    Ignored:    code/QTL_SNP_Enrichment/
    Ignored:    code/QTLs/
    Ignored:    code/RPKM_tables/
    Ignored:    code/ReadLengthMapExperiment/
    Ignored:    code/ReadLengthMapExperimentResults/
    Ignored:    code/ReadLengthMapExperimentSpliceCounts/
    Ignored:    code/ReferenceGenome/
    Ignored:    code/Rplots.pdf
    Ignored:    code/Session.vim
    Ignored:    code/SmallMolecule/
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TODO
    Ignored:    code/Tehranchi/
    Ignored:    code/bigwigs/
    Ignored:    code/bigwigs_FromNonWASPFilteredReads/
    Ignored:    code/config/.DS_Store
    Ignored:    code/config/._.DS_Store
    Ignored:    code/config/.ipynb_checkpoints/
    Ignored:    code/config/config.local.yaml
    Ignored:    code/dag.pdf
    Ignored:    code/dag.png
    Ignored:    code/dag.svg
    Ignored:    code/debug.ipynb
    Ignored:    code/debug_python.ipynb
    Ignored:    code/deepTools/
    Ignored:    code/featureCounts/
    Ignored:    code/featureCountsBasicGtf/
    Ignored:    code/gwas_summary_stats/
    Ignored:    code/hyprcoloc/
    Ignored:    code/igv_session.xml
    Ignored:    code/isoseqbams/
    Ignored:    code/log
    Ignored:    code/logs/
    Ignored:    code/notebooks/.ipynb_checkpoints/
    Ignored:    code/rules/.QTLTools.smk.swp
    Ignored:    code/rules/.ipynb_checkpoints/
    Ignored:    code/rules/OldRules/
    Ignored:    code/rules/notebooks/
    Ignored:    code/scratch/
    Ignored:    code/scripts/.ipynb_checkpoints/
    Ignored:    code/scripts/GTFtools_0.8.0/
    Ignored:    code/scripts/__pycache__/
    Ignored:    code/scripts/liftOverBedpe/liftOverBedpe.py
    Ignored:    code/snakemake.dryrun.log
    Ignored:    code/snakemake.log
    Ignored:    code/snakemake.sbatch.log
    Ignored:    code/snakemake_profiles/slurm/__pycache__/
    Ignored:    code/test.introns.bed
    Ignored:    code/test.introns2.bed
    Ignored:    code/test.log
    Ignored:    code/tracks.xml
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/._20220414203249_JASPAR2022_combined_matrices_25818_jaspar.txt
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-10.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-11.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-2.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-3.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-4.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-5.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-6.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-7.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-8.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022.csv
    Ignored:    data/Metaplots/.DS_Store

Untracked files:
    Untracked:  analysis/20230314_ProcessSM_ForInterpretableSplicingEffectSizes.Rmd
    Untracked:  analysis/20230320_RemakeEffectSizeScatters_UpdatedAnnotations.Rmd
    Untracked:  code/scripts/SmallMoleculeParseCassetteExons.py
    Untracked:  output/SmallMoleculeGAGT_CassetteExonclusters.bed

Unstaged changes:
    Modified:   code/rules/CalculatePi1.smk
    Modified:   code/rules/ProcessSmallMoleculeData.smk
    Modified:   code/scripts/FitSmallMoleculeModels.R
    Modified:   code/scripts/GenometracksByGenotype

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE, warning = F, message = F)

library(tidyverse)</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>✔ ggplot2 3.3.6     ✔ purrr   0.3.4
✔ tibble  3.1.7     ✔ dplyr   1.0.9
✔ tidyr   1.2.0     ✔ stringr 1.4.0
✔ readr   2.1.2     ✔ forcats 0.5.1</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(RColorBrewer)
library(data.table)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre class="r"><code>library(edgeR)</code></pre>
<pre><code>Loading required package: limma</code></pre>
<pre class="r"><code># Set theme
theme_set(
  theme_classic() +
  theme(text=element_text(size=16,  family=&quot;Helvetica&quot;)))

# I use layer a lot, to rotate long x-axis labels
Rotate_x_labels &lt;- theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))

#test plot
ggplot(mtcars, aes(x=mpg, y=cyl)) +
  geom_point()</code></pre>
<p><img src="figure/20230320_RemakeEffectSizeScatters_UpdatedAnnotations.Rmd/setup-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Annotation Colors
ColorKey &lt;- c(
  &quot;AnnotatedJunc_ProductiveCodingGene&quot; = &quot;#1f78b4&quot;,
  &quot;UnannotatedJunc_ProductiveCodingGene&quot; = &quot;#a6cee3&quot;,
  &quot;AnnotatedJunc_UnproductiveCodingGene&quot; = &quot;#e31a1c&quot;,
  &quot;UnannotatedJunc_UnproductiveCodingGene&quot; = &quot;#fb9a99&quot;,
  &quot;AnnotatedJunc_NoncodingGene&quot; = &quot;#6a3d9a&quot;,
  &quot;UnannotatedJunc_NoncodingGene&quot; = &quot;#cab2d6&quot;)</code></pre>
<div id="intro" class="section level2">
<h2>Intro</h2>
<p>I previously made eQTL beta vs sQTL beta scatters, faceted by intron
annotation to highlight the pervasiveness of eQTL predictable effects
from sQTLs at non-productive junctions. Here I will remake such plots,
similar to how I envision for publication, using Yang’s new intron
annotations.</p>
<pre class="r"><code>dat &lt;- fread(&quot;../code/PairwisePi1Traits.P.all.txt.gz&quot;)

dat$PC1 %&gt;% unique()</code></pre>
<pre><code> [1] &quot;Expression.Splicing&quot;              &quot;Expression.Splicing.Subset_YRI&quot;  
 [3] &quot;chRNA.Expression.Splicing&quot;        &quot;MetabolicLabelled.30min&quot;         
 [5] &quot;MetabolicLabelled.60min&quot;          &quot;CTCF&quot;                            
 [7] &quot;H3K27AC&quot;                          &quot;H3K4ME3&quot;                         
 [9] &quot;H3K4ME1&quot;                          &quot;H3K36ME3&quot;                        
[11] &quot;ProCap&quot;                           &quot;polyA.Splicing&quot;                  
[13] &quot;polyA.Splicing.Subset_YRI&quot;        &quot;chRNA.Splicing&quot;                  
[15] &quot;MetabolicLabelled.30min.Splicing&quot; &quot;MetabolicLabelled.60min.Splicing&quot;
[17] &quot;chRNA.Expression_ncRNA&quot;           &quot;polyA.IER&quot;                       
[19] &quot;polyA.IER.Subset_YRI&quot;             &quot;chRNA.IER&quot;                       
[21] &quot;MetabolicLabelled.30min.IER&quot;      &quot;MetabolicLabelled.60min.IER&quot;     
[23] &quot;chRNA.Slopes&quot;                     &quot;chRNA.Splicing.Order&quot;            </code></pre>
<pre class="r"><code>PC1.PossibleValues &lt;- c(&quot;polyA.Splicing&quot;, &quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;)
PC2.PossibleValues &lt;-c(&quot;Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;, &quot;H3K4ME3&quot;, &quot;H3K36ME3&quot;, &quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;MetabolicLabelled.30min&quot;, &quot;MetabolicLabelled.60min&quot;, &quot;chRNA.Expression.Splicing&quot;)

Intron.Annotations &lt;- read_tsv(&quot;../data/IntronAnnotationsFromYang.tsv.gz&quot;) %&gt;%
  mutate(IntronName = paste(chrom, start, end, strand, sep=&quot;:&quot;))

dat.sQTLs.eQTLs &lt;- dat %&gt;% filter(PC1 %in% PC1.PossibleValues  &amp; PC2 %in% PC2.PossibleValues) %&gt;%
  mutate(IntronName = str_replace(P1, &quot;^(.+?:)clu_.+?([+-])$&quot;, &quot;chr\\1\\2&quot;)) %&gt;%
  mutate(ClusterName =  str_replace(P1, &quot;^(.+?:).+?(clu_.+?[+-])$&quot;, &quot;chr\\1\\2&quot;)) %&gt;%
  left_join(Intron.Annotations)


PC1.filter = c(&quot;polyA.Splicing&quot;)
PC2.filter = c(&quot;H3K36ME3&quot;, &quot;chRNA.Expression.Splicing&quot; , &quot;MetabolicLabelled.30min&quot;, &quot;MetabolicLabelled.60min&quot;, &quot;Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;)
PC2.SignificanceFilter &lt;- c(&quot;H3K4ME3&quot;, &quot;H3K27AC&quot;, &quot;H3K36ME3&quot;)


dat.sQTLs.eQTLs$SuperAnnotation %&gt;% unique()</code></pre>
<pre><code>[1] &quot;UnannotatedJunc_UnproductiveCodingGene&quot;
[2] &quot;AnnotatedJunc_ProductiveCodingGene&quot;    
[3] &quot;AnnotatedJunc_UnproductiveCodingGene&quot;  
[4] &quot;UnannotatedJunc_ProductiveCodingGene&quot;  
[5] &quot;AnnotatedJunc_NoncodingGene&quot;           
[6] NA                                      
[7] &quot;UnannotatedJunc_NoncodingGene&quot;         </code></pre>
<pre class="r"><code>dat.sQTLs.eQTLs %&gt;%
  filter(PC1 %in% PC1.filter) %&gt;%
  group_by(P1) %&gt;%
  filter(!any((PC2 %in% PC2.SignificanceFilter) &amp; (trait.x.p.in.y &lt; 0.01))) %&gt;%
  ungroup() %&gt;%
  filter(PC2 %in% PC2.filter) %&gt;%
  filter(SuperAnnotation %in% c(&quot;UnannotatedJunc_UnproductiveCodingGene&quot;, &quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;)) %&gt;%
  mutate(SuperAnnotation = str_replace_all(SuperAnnotation, &quot;_&quot;, &quot; &quot;)) %&gt;%
  filter(GeneLocus == gene) %&gt;%
  filter(FDR.x &lt; 0.1) %&gt;%
  group_by(P2, PC2, SuperAnnotation) %&gt;%
  sample_n(1) %&gt;%
  ungroup() %&gt;%
  mutate(PC2 = factor(PC2, levels=PC2.filter)) %&gt;%
  # mutate(PC2 = recode(PC2, !!!c(&quot;chRNA.Expression.Splicing&quot;=&quot;chRNA&quot; , &quot;MetabolicLabelled.30min&quot;=&quot;30min 4sU&quot;, &quot;MetabolicLabelled.60min&quot;=&quot;60min 4sU&quot;, &quot;Expression.Splicing&quot;=&quot;polyA&quot;, &quot;Expression.Splicing.Subset_YRI&quot;=&quot;polyA YRI&quot;))) %&gt;%
  # mutate(PC2 = factor(PC2, levels=c(&quot;H3K36ME3&quot;,&quot;chRNA&quot;, &quot;30min 4sU&quot;, &quot;60min 4sU&quot;, &quot;polyA&quot;, &quot;polyA YRI&quot;))) %&gt;%
  ggplot(aes(x=beta.x, y=trait.x.beta.in.y)) +
    geom_point(alpha=0.1) +
    # geom_smooth(method = &quot;tls&quot;, se = FALSE, color = &quot;red&quot;, method=&#39;bootstrap&#39;) +
    geom_smooth(method=&#39;lm&#39;) +
    geom_vline(xintercept=0) +
    geom_hline(yintercept=0) +
    geom_text(
    data = . %&gt;%
      group_by(SuperAnnotation, PC2) %&gt;%
      summarise(cor=cor.test(beta.x,trait.x.beta.in.y, method=&quot;s&quot;)[[&quot;estimate&quot;]], pval=cor.test(beta.x,trait.x.beta.in.y, method=&quot;s&quot;)[[&quot;p.value&quot;]]) %&gt;%
      mutate(R = signif(cor, 3), P=format.pval(pval, 3)) %&gt;%
      mutate(label = str_glue(&quot;rho:{R}\nP:{P}&quot;)),
    aes(x=-Inf, y=-Inf, label=label),
    hjust=-.1, vjust=-0.1, color=&#39;red&#39;, size=3
    ) +
    facet_grid(SuperAnnotation ~ PC2, labeller = label_wrap_gen(10)) +
    theme_bw() +
    theme(strip.text = element_text(size = 6)) +
  labs(caption = &quot;FDR&lt;10% sQTLs. sQTLs with nominal P&lt;0.01 for H3K36me3, H3K27Ac, or H3K4me3 removed&quot;, y=&quot;host gene eQTL&quot;, x=&quot;sQTL&quot;)</code></pre>
<p><img src="figure/20230320_RemakeEffectSizeScatters_UpdatedAnnotations.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/sQTL_eQTL_scatter_NewAnnotations_Temp.png&quot;)</code></pre>
<p>Hmm. I wonder why the unannoated stable juncs still seem unstable.
maybe they are connected to unstable junctions. Let’s explore that.</p>
<p>Let’s plot the relative fraction of sQTL jucntion types within each
cluster that is represented in each row.</p>
<pre class="r"><code>ColorKey.Spaces &lt;- setNames(ColorKey, str_replace_all(names(ColorKey), &quot;_&quot;, &quot; &quot;))

sQTLs &lt;- dat.sQTLs.eQTLs %&gt;%
  filter(PC1 %in% PC1.filter) %&gt;%
  group_by(P1) %&gt;%
  filter(!any((PC2 %in% PC2.SignificanceFilter) &amp; (trait.x.p.in.y &lt; 0.01))) %&gt;%
  ungroup() %&gt;%
  filter(PC2 %in% PC2.filter) %&gt;%
  filter(SuperAnnotation %in% c(&quot;UnannotatedJunc_UnproductiveCodingGene&quot;, &quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;)) %&gt;%
  mutate(SuperAnnotation = str_replace_all(SuperAnnotation, &quot;_&quot;, &quot; &quot;)) %&gt;%
  mutate(SuperAnnotation = factor(SuperAnnotation, levels=names(ColorKey.Spaces))) %&gt;%
  distinct(IntronName, ClusterName, SuperAnnotation)

sQTLs %&gt;%
  count(ClusterName) %&gt;%
  ggplot(aes(x=n)) +
  stat_ecdf() +
  coord_cartesian(xlim=c(0,5)) +
  labs(&quot;Num sQTLs per sQTLs cluster&quot;)</code></pre>
<p><img src="figure/20230320_RemakeEffectSizeScatters_UpdatedAnnotations.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sQTLs %&gt;%
  count(SuperAnnotation) %&gt;%
  ggplot(aes(x=SuperAnnotation, y=n)) +
  geom_col() +
  labs(&quot;Num sQTLs in each category&quot;) +
  coord_flip()</code></pre>
<p><img src="figure/20230320_RemakeEffectSizeScatters_UpdatedAnnotations.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>left_join(sQTLs, sQTLs, by=&quot;ClusterName&quot;) %&gt;%
  filter(!IntronName.x == IntronName.y) %&gt;%
  count(SuperAnnotation.x, SuperAnnotation.y) %&gt;%
  ggplot(aes(x=SuperAnnotation.x, y=n, fill=SuperAnnotation.y)) +
  geom_col(position=&quot;fill&quot;) +
  scale_x_discrete(labels = function(x) str_wrap(x, width = 10)) +
  # scale_fill_manual(values = ColorKey.Spaces) +
  Rotate_x_labels +
  labs(title = &quot;Unannotated productive sQTLs are linked to unannotated unproductive&quot;, fill=&quot;Intra cluster annotations&quot;, y=&quot;Fraction intra cluster sQTLs&quot;, x=&quot;sQTL group&quot;, caption=&quot;removed self introns from intra-cluster&quot;)</code></pre>
<p><img src="figure/20230320_RemakeEffectSizeScatters_UpdatedAnnotations.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>left_join(sQTLs, sQTLs, by=&quot;ClusterName&quot;) %&gt;%
  count(SuperAnnotation.x, SuperAnnotation.y) %&gt;%
  ggplot(aes(x=SuperAnnotation.x, y=n, fill=SuperAnnotation.y)) +
  geom_col(position=&quot;fill&quot;) +
  scale_x_discrete(labels = function(x) str_wrap(x, width = 10)) +
  # scale_fill_manual(values = ColorKey.Spaces) +
  Rotate_x_labels +
  labs(title = &quot;Unannotated productive sQTLs are linked to unannotated unproductive&quot;, fill=&quot;Intra cluster annotations&quot;, y=&quot;Fraction intra cluster sQTLs&quot;, x=&quot;sQTL group&quot;, caption=&quot;Included self introns in intra-cluster&quot;)</code></pre>
<p><img src="figure/20230320_RemakeEffectSizeScatters_UpdatedAnnotations.Rmd/unnamed-chunk-3-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>left_join(sQTLs, sQTLs, by=&quot;ClusterName&quot;) %&gt;%
  count(SuperAnnotation.x, SuperAnnotation.y) %&gt;%
  filter(!SuperAnnotation.x == SuperAnnotation.y) %&gt;%
  ggplot(aes(x=SuperAnnotation.x, y=n, fill=SuperAnnotation.y)) +
  geom_col(position=&quot;fill&quot;) +
  scale_x_discrete(labels = function(x) str_wrap(x, width = 10)) +
  # scale_fill_manual(values = ColorKey.Spaces) +
  Rotate_x_labels +
  labs(title = &quot;Unannotated productive sQTLs are linked to unannotated unproductive&quot;, fill=&quot;Intra cluster annotations&quot;, y=&quot;Fraction intra cluster sQTLs&quot;, x=&quot;sQTL group&quot;, caption=&quot;Included self type introns excluded&quot;)</code></pre>
<p><img src="figure/20230320_RemakeEffectSizeScatters_UpdatedAnnotations.Rmd/unnamed-chunk-3-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, so far it still seems plausible that the negative slope in
unannotated productive juncs is just from linked unannotated
unproductive juncs. Let’s try filtering out all unannotated productive
juncs with any intra cluster unproductive junc sQTL, and see if the
slope levels out.</p>
<pre class="r"><code>dat.sQTLs.eQTLs %&gt;%
  filter(PC1 %in% PC1.filter) %&gt;%
  group_by(P1) %&gt;%
  filter(!any((PC2 %in% PC2.SignificanceFilter) &amp; (trait.x.p.in.y &lt; 0.01))) %&gt;%
  ungroup() %&gt;%
  filter(PC2 %in% PC2.filter) %&gt;%
  filter(SuperAnnotation %in% c(&quot;UnannotatedJunc_UnproductiveCodingGene&quot;, &quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;)) %&gt;%
  group_by(PC1, ClusterName) %&gt;%
  filter(!(SuperAnnotation == &quot;UnannotatedJunc_ProductiveCodingGene&quot; &amp; any(SuperAnnotation %in% c(&quot;UnannotatedJunc_UnproductiveCodingGene&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;)))) %&gt;%
  ungroup() %&gt;%
  mutate(SuperAnnotation = str_replace_all(SuperAnnotation, &quot;_&quot;, &quot; &quot;)) %&gt;%
  filter(GeneLocus == gene) %&gt;%
  filter(FDR.x &lt; 0.1) %&gt;%
  group_by(P2, PC2, SuperAnnotation) %&gt;%
  sample_n(1) %&gt;%
  ungroup() %&gt;%
  mutate(PC2 = factor(PC2, levels=PC2.filter)) %&gt;%
  # mutate(PC2 = recode(PC2, !!!c(&quot;chRNA.Expression.Splicing&quot;=&quot;chRNA&quot; , &quot;MetabolicLabelled.30min&quot;=&quot;30min 4sU&quot;, &quot;MetabolicLabelled.60min&quot;=&quot;60min 4sU&quot;, &quot;Expression.Splicing&quot;=&quot;polyA&quot;, &quot;Expression.Splicing.Subset_YRI&quot;=&quot;polyA YRI&quot;))) %&gt;%
  # mutate(PC2 = factor(PC2, levels=c(&quot;H3K36ME3&quot;,&quot;chRNA&quot;, &quot;30min 4sU&quot;, &quot;60min 4sU&quot;, &quot;polyA&quot;, &quot;polyA YRI&quot;))) %&gt;%
  ggplot(aes(x=beta.x, y=trait.x.beta.in.y)) +
    geom_point(alpha=0.1) +
    # geom_smooth(method = &quot;tls&quot;, se = FALSE, color = &quot;red&quot;, method=&#39;bootstrap&#39;) +
    geom_smooth(method=&#39;lm&#39;) +
    geom_vline(xintercept=0) +
    geom_hline(yintercept=0) +
    geom_text(
    data = . %&gt;%
      group_by(SuperAnnotation, PC2) %&gt;%
      summarise(cor=cor.test(beta.x,trait.x.beta.in.y, method=&quot;s&quot;)[[&quot;estimate&quot;]], pval=cor.test(beta.x,trait.x.beta.in.y, method=&quot;s&quot;)[[&quot;p.value&quot;]]) %&gt;%
      mutate(R = signif(cor, 3), P=format.pval(pval, 3)) %&gt;%
      mutate(label = str_glue(&quot;rho:{R}\nP:{P}&quot;)),
    aes(x=-Inf, y=-Inf, label=label),
    hjust=-.1, vjust=-0.1, color=&#39;red&#39;, size=3
    ) +
    facet_grid(SuperAnnotation ~ PC2, labeller = label_wrap_gen(10)) +
    theme_bw() +
    theme(strip.text = element_text(size = 6)) +
  labs(caption = &quot;FDR&lt;10% sQTLs. sQTLs with nominal P&lt;0.01 for H3K36me3, H3K27Ac, or H3K4me3 removed\nUnannotated productive with intra cluster unproductive sQTLs removed&quot;, y=&quot;host gene eQTL&quot;, x=&quot;sQTL&quot;)</code></pre>
<p><img src="figure/20230320_RemakeEffectSizeScatters_UpdatedAnnotations.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/sQTL_eQTL_scatter_NewAnnotations_Temp_RemovedSomeIntraClusterEffects.png&quot;)</code></pre>
<p>Hmm, i don’t think that was all that an improvement in showing less
effect in the unannotated productive compared to annotatied unproductive
and unannotated unproductive. Just to verify and code the basically same
thing a different way, let’s filter for Unannotated productive sQTLs
with only annotated productive clusters in the same cluster.</p>
<pre class="r"><code>dat.sQTLs.eQTLs %&gt;%
  filter(PC1 %in% PC1.filter) %&gt;%
  group_by(P1) %&gt;%
  filter(!any((PC2 %in% PC2.SignificanceFilter) &amp; (trait.x.p.in.y &lt; 0.01))) %&gt;%
  ungroup() %&gt;%
  filter(PC2 %in% PC2.filter) %&gt;%
  filter(SuperAnnotation %in% c(&quot;UnannotatedJunc_UnproductiveCodingGene&quot;, &quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;)) %&gt;%
  group_by(PC1, ClusterName) %&gt;%
  filter((!SuperAnnotation == &quot;UnannotatedJunc_ProductiveCodingGene&quot;) | (SuperAnnotation == &quot;UnannotatedJunc_ProductiveCodingGene&quot; &amp; all(SuperAnnotation %in% c(&quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;)))) %&gt;%
  ungroup() %&gt;%
  mutate(SuperAnnotation = str_replace_all(SuperAnnotation, &quot;_&quot;, &quot; &quot;)) %&gt;%
  filter(GeneLocus == gene) %&gt;%
  filter(FDR.x &lt; 0.1) %&gt;%
  group_by(P2, PC2, SuperAnnotation) %&gt;%
  sample_n(1) %&gt;%
  ungroup() %&gt;%
  mutate(PC2 = factor(PC2, levels=PC2.filter)) %&gt;%
  # mutate(PC2 = recode(PC2, !!!c(&quot;chRNA.Expression.Splicing&quot;=&quot;chRNA&quot; , &quot;MetabolicLabelled.30min&quot;=&quot;30min 4sU&quot;, &quot;MetabolicLabelled.60min&quot;=&quot;60min 4sU&quot;, &quot;Expression.Splicing&quot;=&quot;polyA&quot;, &quot;Expression.Splicing.Subset_YRI&quot;=&quot;polyA YRI&quot;))) %&gt;%
  # mutate(PC2 = factor(PC2, levels=c(&quot;H3K36ME3&quot;,&quot;chRNA&quot;, &quot;30min 4sU&quot;, &quot;60min 4sU&quot;, &quot;polyA&quot;, &quot;polyA YRI&quot;))) %&gt;%
  ggplot(aes(x=beta.x, y=trait.x.beta.in.y)) +
    geom_point(alpha=0.1) +
    # geom_smooth(method = &quot;tls&quot;, se = FALSE, color = &quot;red&quot;, method=&#39;bootstrap&#39;) +
    geom_smooth(method=&#39;lm&#39;) +
    geom_vline(xintercept=0) +
    geom_hline(yintercept=0) +
    geom_text(
    data = . %&gt;%
      group_by(SuperAnnotation, PC2) %&gt;%
      summarise(cor=cor.test(beta.x,trait.x.beta.in.y, method=&quot;s&quot;)[[&quot;estimate&quot;]], pval=cor.test(beta.x,trait.x.beta.in.y, method=&quot;s&quot;)[[&quot;p.value&quot;]]) %&gt;%
      mutate(R = signif(cor, 3), P=format.pval(pval, 3)) %&gt;%
      mutate(label = str_glue(&quot;rho:{R}\nP:{P}&quot;)),
    aes(x=-Inf, y=-Inf, label=label),
    hjust=-.1, vjust=-0.1, color=&#39;red&#39;, size=3
    ) +
    facet_grid(SuperAnnotation ~ PC2, labeller = label_wrap_gen(10)) +
    theme_bw() +
    theme(strip.text = element_text(size = 6)) +
  labs(caption = &quot;FDR&lt;10% sQTLs. sQTLs with nominal P&lt;0.01 for H3K36me3, H3K27Ac, or H3K4me3 removed\nUnannotated productive with intra cluster unproductive sQTLs removed&quot;, y=&quot;host gene eQTL&quot;, x=&quot;sQTL&quot;)</code></pre>
<p><img src="figure/20230320_RemakeEffectSizeScatters_UpdatedAnnotations.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, I guess that may be the best it is going to get in short time..
If I had more time I would consider doing something more fancy, like
plotting each cluster only once and coming up with a cluster-level sQTL
effect score, eQTL score, and annotation. But, I think what I am doing
now is satisfactory… The only thing I am still not satisfied with is the
random sampling. I can select a seed for consistent results, but from
playing around with this I know different seeds change results a bit…
see below…</p>
<pre class="r"><code>set.seed(0)

dat.sQTLs.eQTLs %&gt;%
  filter(PC1 %in% PC1.filter) %&gt;%
  group_by(P1) %&gt;%
  filter(!any((PC2 %in% PC2.SignificanceFilter) &amp; (trait.x.p.in.y &lt; 0.01))) %&gt;%
  ungroup() %&gt;%
  filter(PC2 %in% PC2.filter) %&gt;%
  filter(SuperAnnotation %in% c(&quot;UnannotatedJunc_UnproductiveCodingGene&quot;, &quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;)) %&gt;%
  group_by(PC1, ClusterName) %&gt;%
  filter((!SuperAnnotation == &quot;UnannotatedJunc_ProductiveCodingGene&quot;) | (SuperAnnotation == &quot;UnannotatedJunc_ProductiveCodingGene&quot; &amp; all(SuperAnnotation %in% c(&quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;)))) %&gt;%
  ungroup() %&gt;%
  mutate(SuperAnnotation = str_replace_all(SuperAnnotation, &quot;_&quot;, &quot; &quot;)) %&gt;%
  filter(GeneLocus == gene) %&gt;%
  filter(FDR.x &lt; 0.1) %&gt;%
  group_by(P2, PC2, SuperAnnotation) %&gt;%
  sample_n(1) %&gt;%
  ungroup() %&gt;%
  mutate(PC2 = factor(PC2, levels=PC2.filter)) %&gt;%
  # mutate(PC2 = recode(PC2, !!!c(&quot;chRNA.Expression.Splicing&quot;=&quot;chRNA&quot; , &quot;MetabolicLabelled.30min&quot;=&quot;30min 4sU&quot;, &quot;MetabolicLabelled.60min&quot;=&quot;60min 4sU&quot;, &quot;Expression.Splicing&quot;=&quot;polyA&quot;, &quot;Expression.Splicing.Subset_YRI&quot;=&quot;polyA YRI&quot;))) %&gt;%
  # mutate(PC2 = factor(PC2, levels=c(&quot;H3K36ME3&quot;,&quot;chRNA&quot;, &quot;30min 4sU&quot;, &quot;60min 4sU&quot;, &quot;polyA&quot;, &quot;polyA YRI&quot;))) %&gt;%
  ggplot(aes(x=beta.x, y=trait.x.beta.in.y)) +
    geom_point(alpha=0.1) +
    # geom_smooth(method = &quot;tls&quot;, se = FALSE, color = &quot;red&quot;, method=&#39;bootstrap&#39;) +
    geom_smooth(method=&#39;lm&#39;) +
    geom_vline(xintercept=0) +
    geom_hline(yintercept=0) +
    geom_text(
    data = . %&gt;%
      group_by(SuperAnnotation, PC2) %&gt;%
      summarise(cor=cor.test(beta.x,trait.x.beta.in.y, method=&quot;s&quot;)[[&quot;estimate&quot;]], pval=cor.test(beta.x,trait.x.beta.in.y, method=&quot;s&quot;)[[&quot;p.value&quot;]]) %&gt;%
      mutate(R = signif(cor, 3), P=format.pval(pval, 3)) %&gt;%
      mutate(label = str_glue(&quot;rho:{R}\nP:{P}&quot;)),
    aes(x=-Inf, y=-Inf, label=label),
    hjust=-.1, vjust=-0.1, color=&#39;red&#39;, size=3
    ) +
    facet_grid(SuperAnnotation ~ PC2, labeller = label_wrap_gen(10)) +
    theme_bw() +
    theme(strip.text = element_text(size = 6)) +
  labs(caption = &quot;FDR&lt;10% sQTLs. sQTLs with nominal P&lt;0.01 for H3K36me3, H3K27Ac, or H3K4me3 removed\nUnannotated productive with intra cluster unproductive sQTLs removed&quot;, y=&quot;host gene eQTL&quot;, x=&quot;sQTL&quot;)</code></pre>
<p><img src="figure/20230320_RemakeEffectSizeScatters_UpdatedAnnotations.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>set.seed(1)

dat.sQTLs.eQTLs %&gt;%
  filter(PC1 %in% PC1.filter) %&gt;%
  group_by(P1) %&gt;%
  filter(!any((PC2 %in% PC2.SignificanceFilter) &amp; (trait.x.p.in.y &lt; 0.01))) %&gt;%
  ungroup() %&gt;%
  filter(PC2 %in% PC2.filter) %&gt;%
  filter(SuperAnnotation %in% c(&quot;UnannotatedJunc_UnproductiveCodingGene&quot;, &quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;)) %&gt;%
  group_by(PC1, ClusterName) %&gt;%
  filter((!SuperAnnotation == &quot;UnannotatedJunc_ProductiveCodingGene&quot;) | (SuperAnnotation == &quot;UnannotatedJunc_ProductiveCodingGene&quot; &amp; all(SuperAnnotation %in% c(&quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;)))) %&gt;%
  ungroup() %&gt;%
  mutate(SuperAnnotation = str_replace_all(SuperAnnotation, &quot;_&quot;, &quot; &quot;)) %&gt;%
  filter(GeneLocus == gene) %&gt;%
  filter(FDR.x &lt; 0.1) %&gt;%
  group_by(P2, PC2, SuperAnnotation) %&gt;%
  sample_n(1) %&gt;%
  ungroup() %&gt;%
  mutate(PC2 = factor(PC2, levels=PC2.filter)) %&gt;%
  # mutate(PC2 = recode(PC2, !!!c(&quot;chRNA.Expression.Splicing&quot;=&quot;chRNA&quot; , &quot;MetabolicLabelled.30min&quot;=&quot;30min 4sU&quot;, &quot;MetabolicLabelled.60min&quot;=&quot;60min 4sU&quot;, &quot;Expression.Splicing&quot;=&quot;polyA&quot;, &quot;Expression.Splicing.Subset_YRI&quot;=&quot;polyA YRI&quot;))) %&gt;%
  # mutate(PC2 = factor(PC2, levels=c(&quot;H3K36ME3&quot;,&quot;chRNA&quot;, &quot;30min 4sU&quot;, &quot;60min 4sU&quot;, &quot;polyA&quot;, &quot;polyA YRI&quot;))) %&gt;%
  ggplot(aes(x=beta.x, y=trait.x.beta.in.y)) +
    geom_point(alpha=0.1) +
    # geom_smooth(method = &quot;tls&quot;, se = FALSE, color = &quot;red&quot;, method=&#39;bootstrap&#39;) +
    geom_smooth(method=&#39;lm&#39;) +
    geom_vline(xintercept=0) +
    geom_hline(yintercept=0) +
    geom_text(
    data = . %&gt;%
      group_by(SuperAnnotation, PC2) %&gt;%
      summarise(cor=cor.test(beta.x,trait.x.beta.in.y, method=&quot;s&quot;)[[&quot;estimate&quot;]], pval=cor.test(beta.x,trait.x.beta.in.y, method=&quot;s&quot;)[[&quot;p.value&quot;]]) %&gt;%
      mutate(R = signif(cor, 3), P=format.pval(pval, 3)) %&gt;%
      mutate(label = str_glue(&quot;rho:{R}\nP:{P}&quot;)),
    aes(x=-Inf, y=-Inf, label=label),
    hjust=-.1, vjust=-0.1, color=&#39;red&#39;, size=3
    ) +
    facet_grid(SuperAnnotation ~ PC2, labeller = label_wrap_gen(10)) +
    theme_bw() +
    theme(strip.text = element_text(size = 6)) +
  labs(caption = &quot;FDR&lt;10% sQTLs. sQTLs with nominal P&lt;0.01 for H3K36me3, H3K27Ac, or H3K4me3 removed\nUnannotated productive with intra cluster unproductive sQTLs removed&quot;, y=&quot;host gene eQTL&quot;, x=&quot;sQTL&quot;)</code></pre>
<p><img src="figure/20230320_RemakeEffectSizeScatters_UpdatedAnnotations.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>So maybe different solution is to just pick the strongest sQTL within
each cluster.</p>
<pre class="r"><code>dat.sQTLs.eQTLs %&gt;%
  filter(PC1 %in% PC1.filter) %&gt;%
  group_by(P1) %&gt;%
  filter(!any((PC2 %in% PC2.SignificanceFilter) &amp; (trait.x.p.in.y &lt; 0.01))) %&gt;%
  ungroup() %&gt;%
  filter(PC2 %in% PC2.filter) %&gt;%
  filter(SuperAnnotation %in% c(&quot;UnannotatedJunc_UnproductiveCodingGene&quot;, &quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;)) %&gt;%
  group_by(PC1, ClusterName) %&gt;%
  filter((!SuperAnnotation == &quot;UnannotatedJunc_ProductiveCodingGene&quot;) | (SuperAnnotation == &quot;UnannotatedJunc_ProductiveCodingGene&quot; &amp; all(SuperAnnotation %in% c(&quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;)))) %&gt;%
  filter(abs(beta.x) == max(abs(beta.x))) %&gt;%
  ungroup() %&gt;%
  mutate(SuperAnnotation = str_replace_all(SuperAnnotation, &quot;_&quot;, &quot; &quot;)) %&gt;%
  filter(GeneLocus == gene) %&gt;%
  filter(FDR.x &lt; 0.1) %&gt;%
  group_by(P2, PC2, SuperAnnotation) %&gt;%
  sample_n(1) %&gt;%
  ungroup() %&gt;%
  mutate(PC2 = factor(PC2, levels=PC2.filter)) %&gt;%
  # mutate(PC2 = recode(PC2, !!!c(&quot;chRNA.Expression.Splicing&quot;=&quot;chRNA&quot; , &quot;MetabolicLabelled.30min&quot;=&quot;30min 4sU&quot;, &quot;MetabolicLabelled.60min&quot;=&quot;60min 4sU&quot;, &quot;Expression.Splicing&quot;=&quot;polyA&quot;, &quot;Expression.Splicing.Subset_YRI&quot;=&quot;polyA YRI&quot;))) %&gt;%
  # mutate(PC2 = factor(PC2, levels=c(&quot;H3K36ME3&quot;,&quot;chRNA&quot;, &quot;30min 4sU&quot;, &quot;60min 4sU&quot;, &quot;polyA&quot;, &quot;polyA YRI&quot;))) %&gt;%
  ggplot(aes(x=beta.x, y=trait.x.beta.in.y)) +
    geom_point(alpha=0.1) +
    # geom_smooth(method = &quot;tls&quot;, se = FALSE, color = &quot;red&quot;, method=&#39;bootstrap&#39;) +
    geom_smooth(method=&#39;lm&#39;) +
    geom_vline(xintercept=0) +
    geom_hline(yintercept=0) +
    geom_text(
    data = . %&gt;%
      group_by(SuperAnnotation, PC2) %&gt;%
      summarise(cor=cor.test(beta.x,trait.x.beta.in.y, method=&quot;s&quot;)[[&quot;estimate&quot;]], pval=cor.test(beta.x,trait.x.beta.in.y, method=&quot;s&quot;)[[&quot;p.value&quot;]]) %&gt;%
      mutate(R = signif(cor, 3), P=format.pval(pval, 3)) %&gt;%
      mutate(label = str_glue(&quot;rho:{R}\nP:{P}&quot;)),
    aes(x=-Inf, y=-Inf, label=label),
    hjust=-.1, vjust=-0.1, color=&#39;red&#39;, size=3
    ) +
    facet_grid(SuperAnnotation ~ PC2, labeller = label_wrap_gen(10)) +
    theme_bw() +
    theme(strip.text = element_text(size = 6)) +
  labs(caption = &quot;FDR&lt;10% sQTLs. sQTLs with nominal P&lt;0.01 for H3K36me3, H3K27Ac, or H3K4me3 removed\nUnannotated productive with intra cluster unproductive sQTLs removed\nOnly top sQTL per cluster plotted&quot;, y=&quot;host gene eQTL&quot;, x=&quot;sQTL&quot;)</code></pre>
<p><img src="figure/20230320_RemakeEffectSizeScatters_UpdatedAnnotations.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dat.sQTLs.eQTLs %&gt;%
  filter(PC1 %in% PC1.filter) %&gt;%
  group_by(P1) %&gt;%
  filter(!any((PC2 %in% PC2.SignificanceFilter) &amp; (trait.x.p.in.y &lt; 0.01))) %&gt;%
  ungroup() %&gt;%
  filter(PC2 %in% PC2.filter) %&gt;%
  filter(SuperAnnotation %in% c(&quot;UnannotatedJunc_UnproductiveCodingGene&quot;, &quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;)) %&gt;%
  group_by(PC1, ClusterName) %&gt;%
  filter(abs(beta.x) == max(abs(beta.x))) %&gt;%
  ungroup() %&gt;%
  mutate(SuperAnnotation = str_replace_all(SuperAnnotation, &quot;_&quot;, &quot; &quot;)) %&gt;%
  filter(GeneLocus == gene) %&gt;%
  filter(FDR.x &lt; 0.1) %&gt;%
  group_by(P2, PC2, SuperAnnotation) %&gt;%
  sample_n(1) %&gt;%
  ungroup() %&gt;%
  mutate(PC2 = factor(PC2, levels=PC2.filter)) %&gt;%
  # mutate(PC2 = recode(PC2, !!!c(&quot;chRNA.Expression.Splicing&quot;=&quot;chRNA&quot; , &quot;MetabolicLabelled.30min&quot;=&quot;30min 4sU&quot;, &quot;MetabolicLabelled.60min&quot;=&quot;60min 4sU&quot;, &quot;Expression.Splicing&quot;=&quot;polyA&quot;, &quot;Expression.Splicing.Subset_YRI&quot;=&quot;polyA YRI&quot;))) %&gt;%
  # mutate(PC2 = factor(PC2, levels=c(&quot;H3K36ME3&quot;,&quot;chRNA&quot;, &quot;30min 4sU&quot;, &quot;60min 4sU&quot;, &quot;polyA&quot;, &quot;polyA YRI&quot;))) %&gt;%
  ggplot(aes(x=beta.x, y=trait.x.beta.in.y)) +
    geom_point(alpha=0.1) +
    # geom_smooth(method = &quot;tls&quot;, se = FALSE, color = &quot;red&quot;, method=&#39;bootstrap&#39;) +
    geom_smooth(method=&#39;lm&#39;) +
    geom_vline(xintercept=0) +
    geom_hline(yintercept=0) +
    geom_text(
    data = . %&gt;%
      group_by(SuperAnnotation, PC2) %&gt;%
      summarise(cor=cor.test(beta.x,trait.x.beta.in.y, method=&quot;s&quot;)[[&quot;estimate&quot;]], pval=cor.test(beta.x,trait.x.beta.in.y, method=&quot;s&quot;)[[&quot;p.value&quot;]]) %&gt;%
      mutate(R = signif(cor, 3), P=format.pval(pval, 3)) %&gt;%
      mutate(label = str_glue(&quot;rho:{R}\nP:{P}&quot;)),
    aes(x=-Inf, y=-Inf, label=label),
    hjust=-.1, vjust=-0.1, color=&#39;red&#39;, size=3
    ) +
    facet_grid(SuperAnnotation ~ PC2, labeller = label_wrap_gen(10)) +
    theme_bw() +
    theme(strip.text = element_text(size = 6)) +
  labs(caption = &quot;FDR&lt;10% sQTLs. sQTLs with nominal P&lt;0.01 for H3K36me3, H3K27Ac, or H3K4me3 removed\nOnly top sQTL per cluster plotted&quot;, y=&quot;host gene eQTL&quot;, x=&quot;sQTL&quot;)</code></pre>
<p><img src="figure/20230320_RemakeEffectSizeScatters_UpdatedAnnotations.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] edgeR_3.38.4       limma_3.52.4       data.table_1.14.2  RColorBrewer_1.1-3
 [5] forcats_0.5.1      stringr_1.4.0      dplyr_1.0.9        purrr_0.3.4       
 [9] readr_2.1.2        tidyr_1.2.0        tibble_3.1.7       ggplot2_3.3.6     
[13] tidyverse_1.3.1   

loaded via a namespace (and not attached):
 [1] httr_1.4.3        sass_0.4.1        splines_4.2.0     bit64_4.0.5      
 [5] vroom_1.5.7       jsonlite_1.8.0    R.utils_2.11.0    modelr_0.1.8     
 [9] bslib_0.3.1       assertthat_0.2.1  highr_0.9         cellranger_1.1.0 
[13] yaml_2.3.5        pillar_1.7.0      backports_1.4.1   lattice_0.20-45  
[17] glue_1.6.2        digest_0.6.29     promises_1.2.0.1  rvest_1.0.2      
[21] colorspace_2.0-3  Matrix_1.5-3      htmltools_0.5.2   httpuv_1.6.5     
[25] R.oo_1.24.0       pkgconfig_2.0.3   broom_0.8.0       haven_2.5.0      
[29] scales_1.2.0      later_1.3.0       tzdb_0.3.0        git2r_0.30.1     
[33] mgcv_1.8-40       generics_0.1.2    farver_2.1.0      ellipsis_0.3.2   
[37] withr_2.5.0       cli_3.3.0         magrittr_2.0.3    crayon_1.5.1     
[41] readxl_1.4.0      evaluate_0.15     R.methodsS3_1.8.1 fs_1.5.2         
[45] fansi_1.0.3       nlme_3.1-157      xml2_1.3.3        tools_4.2.0      
[49] hms_1.1.1         lifecycle_1.0.1   munsell_0.5.0     reprex_2.0.1     
[53] locfit_1.5-9.7    compiler_4.2.0    jquerylib_0.1.4   rlang_1.0.2      
[57] grid_4.2.0        rstudioapi_0.13   labeling_0.4.2    rmarkdown_2.14   
[61] gtable_0.3.0      DBI_1.1.2         R6_2.5.1          lubridate_1.8.0  
[65] knitr_1.39        bit_4.0.4         fastmap_1.1.0     utf8_1.2.2       
[69] workflowr_1.7.0   rprojroot_2.0.3   stringi_1.7.6     parallel_4.2.0   
[73] Rcpp_1.0.8.3      vctrs_0.4.1       dbplyr_2.1.1      tidyselect_1.1.2 
[77] xfun_0.30        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
