<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2023-06-26" />

<title>20230626_betaBetaForChao</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">20230626_betaBetaForChao</h1>
<h4 class="date">2023-06-26</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-06-26
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>ChromatinSplicingQTLs/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstree4ffc2bc2961e4eabb1eabf27c3a9c8d61ce8944dtargetblank4ffc2bca">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/4ffc2bc2961e4eabb1eabf27c3a9c8d61ce8944d" target="_blank">4ffc2bc</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstree4ffc2bc2961e4eabb1eabf27c3a9c8d61ce8944dtargetblank4ffc2bca"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/4ffc2bc2961e4eabb1eabf27c3a9c8d61ce8944d" target="_blank">4ffc2bc</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store
    Ignored:    code/.RData
    Ignored:    code/._report.html
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    code/.snakemake/
    Ignored:    code/APA_Processing/
    Ignored:    code/Alignments/
    Ignored:    code/ChromHMM/
    Ignored:    code/ENCODE/
    Ignored:    code/ExpressionAnalysis/
    Ignored:    code/ExtractPhenotypeBedByGenotype.py
    Ignored:    code/FastqFastp/
    Ignored:    code/FastqFastpSE/
    Ignored:    code/FastqSE/
    Ignored:    code/FineMapping/
    Ignored:    code/GTEx/
    Ignored:    code/Genotypes/
    Ignored:    code/H3K36me3_CutAndTag.pdf
    Ignored:    code/IntronSlopes/
    Ignored:    code/LR.bed
    Ignored:    code/LR.seq.bed
    Ignored:    code/LongReads/
    Ignored:    code/Metaplots/
    Ignored:    code/Misc/
    Ignored:    code/MiscCountTables/
    Ignored:    code/Multiqc/
    Ignored:    code/Multiqc_chRNA/
    Ignored:    code/NonCodingRNA/
    Ignored:    code/NonCodingRNA_annotation/
    Ignored:    code/PairwisePi1Traits.P.all.txt.gz
    Ignored:    code/PeakCalling/
    Ignored:    code/Phenotypes/
    Ignored:    code/PlotGruberQTLs/
    Ignored:    code/PlotQTLs/
    Ignored:    code/ProCapAnalysis/
    Ignored:    code/QC/
    Ignored:    code/QTL_SNP_Enrichment/
    Ignored:    code/QTLs/
    Ignored:    code/RPKM_tables/
    Ignored:    code/ReadLengthMapExperiment/
    Ignored:    code/ReadLengthMapExperimentResults/
    Ignored:    code/ReadLengthMapExperimentSpliceCounts/
    Ignored:    code/ReferenceGenome/
    Ignored:    code/Rplots.pdf
    Ignored:    code/Session.vim
    Ignored:    code/SmallMolecule/
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TODO
    Ignored:    code/Tehranchi/
    Ignored:    code/alias/
    Ignored:    code/bigwigs/
    Ignored:    code/bigwigs_FromNonWASPFilteredReads/
    Ignored:    code/config/.DS_Store
    Ignored:    code/config/._.DS_Store
    Ignored:    code/config/.gwas_table.tsv.swp
    Ignored:    code/config/.ipynb_checkpoints/
    Ignored:    code/config/config.local.yaml
    Ignored:    code/dag.pdf
    Ignored:    code/dag.png
    Ignored:    code/dag.svg
    Ignored:    code/data/
    Ignored:    code/debug.ipynb
    Ignored:    code/debug_python.ipynb
    Ignored:    code/deepTools/
    Ignored:    code/featureCounts/
    Ignored:    code/featureCountsBasicGtf/
    Ignored:    code/genome_config.yaml
    Ignored:    code/gwas_summary_stats/
    Ignored:    code/hyprcoloc/
    Ignored:    code/igv_session.xml
    Ignored:    code/isoseqbams/
    Ignored:    code/log
    Ignored:    code/logs/
    Ignored:    code/notebooks/.ipynb_checkpoints/
    Ignored:    code/pi1/
    Ignored:    code/polyA.Splicing.Subset_YRI.NominalPassForColoc.bed.bgz
    Ignored:    code/rules/.ipynb_checkpoints/
    Ignored:    code/rules/OldRules/
    Ignored:    code/rules/notebooks/
    Ignored:    code/salmontest/
    Ignored:    code/scratch/
    Ignored:    code/scripts/.ipynb_checkpoints/
    Ignored:    code/scripts/GTFtools_0.8.0/
    Ignored:    code/scripts/__pycache__/
    Ignored:    code/scripts/liftOverBedpe/liftOverBedpe.py
    Ignored:    code/snakemake.dryrun.log
    Ignored:    code/snakemake.log
    Ignored:    code/snakemake.sbatch.log
    Ignored:    code/snakemake_profiles/slurm/__pycache__/
    Ignored:    code/test.introns.bed
    Ignored:    code/test.introns2.bed
    Ignored:    code/test.log
    Ignored:    code/tracks.xml
    Ignored:    data/.DS_Store
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-10.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-11.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-2.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-3.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-4.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-5.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-6.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-7.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-8.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022.csv
    Ignored:    data/Metaplots/.DS_Store

Untracked files:
    Untracked:  analysis/20230622_LookupCarolsSNP.Rmd
    Untracked:  analysis/20230623_BetaBetaScatterForChao.Rmd
    Untracked:  code/scripts/Tidy_GTEx_SummaryStats.py

Unstaged changes:
    Modified:   analysis/MakeFinalFigs_Fig2.Rmd
    Modified:   analysis/MakeFinalFigs_Fig3.Rmd
    Modified:   code/config/gwas_table.tsv
    Modified:   code/scripts/hyprcoloc_gwas2.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE, warning = F, message = F)

library(tidyverse)</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>✔ ggplot2 3.3.6     ✔ purrr   0.3.4
✔ tibble  3.1.7     ✔ dplyr   1.0.9
✔ tidyr   1.2.0     ✔ stringr 1.4.0
✔ readr   2.1.2     ✔ forcats 0.5.1</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(RColorBrewer)
library(data.table)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre class="r"><code>library(broom)


# Set theme
theme_set(
  theme_classic() +
  theme(text=element_text(size=16,  family=&quot;Helvetica&quot;)))

# I use layer a lot, to rotate long x-axis labels
Rotate_x_labels &lt;- theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))</code></pre>
<p>The sQTL beta vs eQTL beta scatter result hasn’t been reproduced by
Chao, at least not to the extent that I saw a strong negative
correlation (spearman rho~0.4) in polyA eQTL vs polyA sQTL for
unproductive introns. Let me reproduce that plot here so Chao can better
troubleshoot exactly what the discrepancy is.</p>
<div id="read-in-data" class="section level2">
<h2>Read in data</h2>
<p>I
<code>/project2/yangili1/bjf79/ChromatinSplicingQTLs/code/pi1/PairwisePi1Traits.P.all.txt.gz</code>
is a file where I take all the top SNPs for a trait (P1), and lookup the
corresponding summary stats for a second trait (P2). Every phenotype has
a phenotype class (PC; PC1 for P1, or PC2 for P2). Chao, feel free to
use this file and inspect it, sorry if the column labels are confusing.
From this file, the general approach will be to look for sQTLs (where P1
is splicing trait), and look at the effect sizes in expression traits
(where P2 is an expression trait), while filtering out sQTLs where any
P2 is a significant hQTL. All P1’s in this file are significant
(FDR&lt;0.1).</p>
<pre class="r"><code>PC1.PossibleValues &lt;- c(&quot;polyA.Splicing&quot;)
PC2.PossibleValues &lt;-c(&quot;Expression.Splicing&quot;, &quot;chRNA.Expression.Splicing&quot;,&quot;H3K4ME3&quot;, &quot;H3K36ME3&quot;, &quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;MetabolicLabelled.30min&quot;, &quot;MetabolicLabelled.60min&quot;)


dat &lt;- fread(&quot;../code/pi1/PairwisePi1Traits.P.all.txt.gz&quot;) %&gt;%
  filter((PC1 %in% PC1.PossibleValues) &amp; (PC2 %in% PC2.PossibleValues))

head(dat)</code></pre>
<pre><code>              PC1                            P1          GeneLocus
1: polyA.Splicing 1:11213566:11228668:clu_262_- ENSG00000198793.13
2: polyA.Splicing 1:11213566:11228668:clu_262_- ENSG00000198793.13
3: polyA.Splicing 1:11213566:11228668:clu_262_- ENSG00000198793.13
4: polyA.Splicing 1:11213566:11228668:clu_262_- ENSG00000198793.13
5: polyA.Splicing 1:11213566:11228668:clu_262_- ENSG00000198793.13
6: polyA.Splicing 1:11213566:11228668:clu_262_- ENSG00000198793.13
   p_permutation.x   beta.x beta_se.x singletrait_topvar.x
1:      0.00449478 0.371906 0.0695477      1:11264089:CA:C
2:      0.00449478 0.371906 0.0695477      1:11264089:CA:C
3:      0.00449478 0.371906 0.0695477      1:11264089:CA:C
4:      0.00449478 0.371906 0.0695477      1:11264089:CA:C
5:      0.00449478 0.371906 0.0695477      1:11264089:CA:C
6:      0.00449478 0.371906 0.0695477      1:11264089:CA:C
   singletrait_topvar_chr.x singletrait_topvar_pos.x      FDR.x
1:                     chr1                 11264085 0.04536886
2:                     chr1                 11264085 0.04536886
3:                     chr1                 11264085 0.04536886
4:                     chr1                 11264085 0.04536886
5:                     chr1                 11264085 0.04536886
6:                     chr1                 11264085 0.04536886
                         PC2                 P2 p_permutation.y    beta.y
1:       Expression.Splicing ENSG00000198793.13     0.019128200 -0.119659
2: chRNA.Expression.Splicing ENSG00000198793.13     0.000565087 -0.533056
3:   MetabolicLabelled.30min ENSG00000198793.13     0.346267000  0.235171
4:   MetabolicLabelled.60min ENSG00000198793.13     0.660299000 -0.311955
5:                   H3K27AC   H3K27AC_peak_885     0.034002300  1.076300
6:                   H3K27AC   H3K27AC_peak_901     0.065683900 -0.378248
   beta_se.y    singletrait_topvar.y singletrait_topvar_chr.y
1: 0.0283057          1:11324733:C:T                     chr1
2: 0.0999955 1:11033952:ACACACACAC:A                     chr1
3: 0.0682238          1:11035367:A:T                     chr1
4: 0.0939992          1:11341688:G:A                     chr1
5: 0.2495070          1:11312860:G:A                     chr1
6: 0.0891486          1:11335426:T:C                     chr1
   singletrait_topvar_pos.y       FDR.y trait.x.p.in.y x.beta.in.y
1:                 11324733 0.006570641       0.305897  0.03589860
2:                 11033951 0.004502949       0.926474 -0.00793427
3:                 11035367 0.419053116       0.828525  0.02134310
4:                 11341688 0.550513014       0.663203 -0.05143100
5:                 11312860 0.173421875       0.772355  0.05620940
6:                 11335426 0.242401179       0.406857 -0.12449800
   x.beta_se.in.y
1:      0.0350216
2:      0.0857210
3:      0.0981314
4:      0.1175410
5:      0.1935470
6:      0.1491950</code></pre>
<pre class="r"><code>Intron.Annotations &lt;- read_tsv(&quot;../data/IntronAnnotationsFromYang.tsv.gz&quot;) %&gt;%
  mutate(IntronName = paste(chrom, start, end, strand, sep=&quot;:&quot;))</code></pre>
<p>Let me do a little data tidying before plotting…</p>
<pre class="r"><code>dat.sQTLs.eQTLs.ForScatter &lt;- dat %&gt;%
  mutate(IntronName = str_replace(P1, &quot;^(.+?:)clu_.+?([+-])$&quot;, &quot;chr\\1\\2&quot;)) %&gt;%
  mutate(ClusterName =  str_replace(P1, &quot;^(.+?:).+?(clu_.+?[+-])$&quot;, &quot;chr\\1\\2&quot;)) %&gt;%
  left_join(Intron.Annotations)

PC1.filter = c(&quot;polyA.Splicing&quot;)
PC2.filter = c( &quot;chRNA.Expression.Splicing&quot; , &quot;MetabolicLabelled.30min&quot;, &quot;MetabolicLabelled.60min&quot;, &quot;Expression.Splicing&quot;)
PC2.SignificanceFilter &lt;- c(&quot;H3K4ME3&quot;, &quot;H3K27AC&quot;, &quot;H3K36ME3&quot;)


dat.sQTLs.eQTLs.ForScatter.ToPlot &lt;- dat.sQTLs.eQTLs.ForScatter %&gt;%
  # filter for sQTLs, where P1 is a splicing trait
  filter(PC1 %in% PC1.filter) %&gt;%
  
  # sanity check filter to make sure I am looking at junctions in the gene that I am supposed to be. The GeneLocus is the gene intersecting the junction from my file, the gene is the gene from Yangs splice junction annotations
  filter(GeneLocus == gene) %&gt;%
  
  # sanity check filter to make sure I am looking at significant sQTLs, even though they should already be filtered for that
  filter(FDR.x &lt; 0.1) %&gt;%
  
   # filter out any sQTLs where the topSNP is nominally significant for any H3K4ME3, H3K27AC, or H3K36ME3 trait
  group_by(P1) %&gt;%
  filter(!any((PC2 %in% PC2.SignificanceFilter) &amp; (trait.x.p.in.y &lt; 0.01))) %&gt;%
  ungroup() %&gt;%
  
  # filter for rows where P2 is an expression trait
  filter(PC2 %in% PC2.filter) %&gt;%

  # filter out the sQTLs in non protein coding genes
  filter(SuperAnnotation %in% c(&quot;UnannotatedJunc_UnproductiveCodingGene&quot;, &quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;)) %&gt;%
  
  # clasify sQTLs as unproductive if any of the introns in the cluster are unproductive introns.
  group_by(PC1, ClusterName) %&gt;%
  mutate(sQTL_type = case_when(
    any(SuperAnnotation %in% c(&quot;UnannotatedJunc_UnproductiveCodingGene&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;)) ~ &quot;Unproductive sQTL&quot;,
    TRUE ~ &quot;Productive sQTL&quot;
    )) %&gt;%
  ungroup() %&gt;%
  
  # For simplicity in downstream step, recode introns as either unrproductive or productive
  mutate(ProductiveOrUnproductive_junction = recode(SuperAnnotation, &quot;UnannotatedJunc_UnproductiveCodingGene&quot;=&quot;Unproductive junc&quot;, &quot;AnnotatedJunc_ProductiveCodingGene&quot;=&quot;Productive junc&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;=&quot;Unproductive junc&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;=&quot;Productive junc&quot;)) %&gt;%
  
  # Keep just the strongest unproductive junction per unproductive sQTL, and the strongest productive junction per productive sQTL
  group_by(ClusterName, P2, ProductiveOrUnproductive_junction) %&gt;%
  filter(abs(beta.x) == max(abs(beta.x))) %&gt;%
  ungroup() %&gt;%
  
  # sanity check that I am only plotting one point per gene in each facet. The previous filter for max(beta) might actually still keep more than one intron if there are ties.
  group_by(ClusterName, PC2, P2, ProductiveOrUnproductive_junction) %&gt;%
  sample_n(1) %&gt;%
  ungroup() %&gt;%

  # recode some things for better labels
  mutate(PC2 = recode(PC2, !!!c(&quot;chRNA.Expression.Splicing&quot;=&quot;chRNA&quot; , &quot;MetabolicLabelled.30min&quot;=&quot;30min 4sU&quot;, &quot;MetabolicLabelled.60min&quot;=&quot;60min 4sU&quot;, &quot;Expression.Splicing&quot;=&quot;polyA&quot;))) %&gt;%
  # redefine factor levels so order of facets is sensible
  mutate(PC2 = factor(PC2, levels=c(&quot;chRNA&quot;, &quot;30min 4sU&quot;, &quot;60min 4sU&quot;, &quot;polyA&quot;))) %&gt;%
  mutate(sQTL_int_type = str_glue(&quot;{sQTL_type}\n{ProductiveOrUnproductive_junction}&quot;))




dat.sQTLs.eQTLs.ForScatter.ToPlot.labels &lt;- dat.sQTLs.eQTLs.ForScatter.ToPlot %&gt;%
  count(sQTL_int_type, PC2) %&gt;%
  distinct(sQTL_int_type, .keep_all=T) %&gt;%
  dplyr::select(n, sQTL_int_type) %&gt;%
  mutate(AnnotationLabel = paste0(sQTL_int_type, &quot; (n=&quot;, n, &quot;)&quot;))


dat.sQTLs.eQTLs.ForScatter.ToPlot %&gt;%
  distinct(ClusterName, .keep_all=T) %&gt;%
  count(sQTL_type)</code></pre>
<pre><code># A tibble: 2 × 2
  sQTL_type             n
  &lt;chr&gt;             &lt;int&gt;
1 Productive sQTL    1160
2 Unproductive sQTL  1498</code></pre>
<pre class="r"><code>sQTL.eQTL.beta.beta &lt;- dat.sQTLs.eQTLs.ForScatter.ToPlot %&gt;%
  left_join(dat.sQTLs.eQTLs.ForScatter.ToPlot.labels, by=&quot;sQTL_int_type&quot;) %&gt;%
  nest(-sQTL_int_type, -PC2) %&gt;% 
  mutate(cor=map(data,~cor.test(.x$beta.x, .x$x.beta.in.y, method = &quot;sp&quot;))) %&gt;%
  mutate(tidied = map(cor, tidy)) %&gt;% 
  unnest(tidied, .drop = T) %&gt;%
  unnest(data) %&gt;%
ggplot(aes(x=beta.x, y=x.beta.in.y, color=ProductiveOrUnproductive_junction)) +
    geom_point(alpha=0.1) +
    # geom_smooth(method = &quot;tls&quot;, se = FALSE, color = &quot;red&quot;, method=&#39;bootstrap&#39;) +
    # geom_smooth(method=&#39;lm&#39;, color=&#39;black&#39;) +
    geom_vline(xintercept=0, linetype=&#39;dashed&#39;) +
    geom_hline(yintercept=0, linetype=&#39;dashed&#39;) +
    scale_color_manual(values=c(&quot;Productive junc&quot;=&quot;#1f78b4&quot;, &quot;Unproductive junc&quot;=&quot;#e31a1c&quot;)) +
    geom_abline(data = . %&gt;%
                  distinct(AnnotationLabel, PC2, .keep_all=T),
                aes(slope=estimate, intercept=0), size=2) +
    geom_text(
    data = . %&gt;%
        distinct(AnnotationLabel, PC2, .keep_all=T) %&gt;%
        mutate(R=signif(estimate, 3), P=format.pval(p.value, 3)) %&gt;%
      mutate(label = str_glue(&quot;rho:{R}\nP:{P}&quot;)),
    aes(x=-Inf, y=-Inf, label=label),
    hjust=-.1, vjust=-0.1, color=&#39;black&#39;, size=6
    ) +
    facet_grid(AnnotationLabel ~ PC2, labeller = label_wrap_gen(10)) +
    theme(strip.text = element_text(size = 12), legend.position=&#39;none&#39;) +
  labs(caption = &quot;FDR&lt;10% sQTLs. sQTLs with nominal P&lt;0.01 for H3K36me3, H3K27Ac, or H3K4me3 removed\nOnly top sQTL per cluster plotted&quot;, y=&quot;host gene eQTL beta&quot;, x=&quot;sQTL beta&quot;)

sQTL.eQTL.beta.beta</code></pre>
<p><img src="figure/20230623_BetaBetaScatterForChao.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The numbers of unproductive sQTLs is a bit more than my previous
plot. At the moment I don’t have time to go through my code to figure
out exactly why there are 1498 unproductive sQTL clusters here, instead
of the like 998 that i previously plotted. But I <em>think</em> this is
still right.</p>
<p>Let me write out the underlying data for Chao</p>
<pre class="r"><code>write_tsv(dat.sQTLs.eQTLs.ForScatter.ToPlot, &quot;/project2/yangili1/bjf79/scratch/DataForChao.tsv.gz&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] broom_0.8.0        data.table_1.14.2  RColorBrewer_1.1-3 forcats_0.5.1     
 [5] stringr_1.4.0      dplyr_1.0.9        purrr_0.3.4        readr_2.1.2       
 [9] tidyr_1.2.0        tibble_3.1.7       ggplot2_3.3.6      tidyverse_1.3.1   

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3      lubridate_1.8.0   assertthat_0.2.1  rprojroot_2.0.3  
 [5] digest_0.6.29     utf8_1.2.2        R6_2.5.1          cellranger_1.1.0 
 [9] backports_1.4.1   reprex_2.0.1      evaluate_0.15     highr_0.9        
[13] httr_1.4.3        pillar_1.7.0      rlang_1.0.2       readxl_1.4.0     
[17] rstudioapi_0.13   jquerylib_0.1.4   R.oo_1.24.0       R.utils_2.11.0   
[21] rmarkdown_2.14    labeling_0.4.2    bit_4.0.4         munsell_0.5.0    
[25] compiler_4.2.0    httpuv_1.6.5      modelr_0.1.8      xfun_0.30        
[29] pkgconfig_2.0.3   htmltools_0.5.2   tidyselect_1.1.2  workflowr_1.7.0  
[33] fansi_1.0.3       crayon_1.5.1      tzdb_0.3.0        dbplyr_2.1.1     
[37] withr_2.5.0       later_1.3.0       R.methodsS3_1.8.1 grid_4.2.0       
[41] jsonlite_1.8.0    gtable_0.3.0      lifecycle_1.0.1   DBI_1.1.2        
[45] git2r_0.30.1      magrittr_2.0.3    scales_1.2.0      vroom_1.5.7      
[49] cli_3.3.0         stringi_1.7.6     farver_2.1.0      fs_1.5.2         
[53] promises_1.2.0.1  xml2_1.3.3        bslib_0.3.1       ellipsis_0.3.2   
[57] generics_0.1.2    vctrs_0.4.1       tools_4.2.0       bit64_4.0.5      
[61] glue_1.6.2        hms_1.1.1         parallel_4.2.0    fastmap_1.1.0    
[65] yaml_2.3.5        colorspace_2.0-3  rvest_1.0.2       knitr_1.39       
[69] haven_2.5.0       sass_0.4.1       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
