<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2023-03-24" />

<title>20230324_MakeFigs</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">20230324_MakeFigs</h1>
<h4 class="date">2023-03-24</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-04-25
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>ChromatinSplicingQTLs/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstreeddf600dbb95a36158c06a26c5d7df9d69c969ed0targetblankddf600da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/ddf600dbb95a36158c06a26c5d7df9d69c969ed0" target="_blank">ddf600d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstreeddf600dbb95a36158c06a26c5d7df9d69c969ed0targetblankddf600da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/ddf600dbb95a36158c06a26c5d7df9d69c969ed0" target="_blank">ddf600d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/figure/
    Ignored:    code/.DS_Store
    Ignored:    code/.RData
    Ignored:    code/._report.html
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    code/.snakemake/
    Ignored:    code/APA_Processing/
    Ignored:    code/Alignments/
    Ignored:    code/ChromHMM/
    Ignored:    code/ENCODE/
    Ignored:    code/ExpressionAnalysis/
    Ignored:    code/ExtractPhenotypeBedByGenotype.py
    Ignored:    code/FastqFastp/
    Ignored:    code/FastqFastpSE/
    Ignored:    code/FastqSE/
    Ignored:    code/FineMapping/
    Ignored:    code/Genotypes/
    Ignored:    code/H3K36me3_CutAndTag.pdf
    Ignored:    code/IntronSlopes/
    Ignored:    code/LR.bed
    Ignored:    code/LR.seq.bed
    Ignored:    code/LongReads/
    Ignored:    code/Metaplots/
    Ignored:    code/Misc/
    Ignored:    code/MiscCountTables/
    Ignored:    code/Multiqc/
    Ignored:    code/Multiqc_chRNA/
    Ignored:    code/NonCodingRNA/
    Ignored:    code/NonCodingRNA_annotation/
    Ignored:    code/PairwisePi1Traits.P.all.txt.gz
    Ignored:    code/PeakCalling/
    Ignored:    code/Phenotypes/
    Ignored:    code/PlotGruberQTLs/
    Ignored:    code/PlotQTLs/
    Ignored:    code/ProCapAnalysis/
    Ignored:    code/QC/
    Ignored:    code/QTL_SNP_Enrichment/
    Ignored:    code/QTLs/
    Ignored:    code/RPKM_tables/
    Ignored:    code/ReadLengthMapExperiment/
    Ignored:    code/ReadLengthMapExperimentResults/
    Ignored:    code/ReadLengthMapExperimentSpliceCounts/
    Ignored:    code/ReferenceGenome/
    Ignored:    code/Rplots.pdf
    Ignored:    code/Session.vim
    Ignored:    code/SmallMolecule/
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TODO
    Ignored:    code/Tehranchi/
    Ignored:    code/bigwigs/
    Ignored:    code/bigwigs_FromNonWASPFilteredReads/
    Ignored:    code/config/.DS_Store
    Ignored:    code/config/._.DS_Store
    Ignored:    code/config/.ipynb_checkpoints/
    Ignored:    code/config/config.local.yaml
    Ignored:    code/dag.pdf
    Ignored:    code/dag.png
    Ignored:    code/dag.svg
    Ignored:    code/debug.ipynb
    Ignored:    code/debug_python.ipynb
    Ignored:    code/deepTools/
    Ignored:    code/featureCounts/
    Ignored:    code/featureCountsBasicGtf/
    Ignored:    code/gwas_summary_stats/
    Ignored:    code/hyprcoloc/
    Ignored:    code/igv_session.xml
    Ignored:    code/isoseqbams/
    Ignored:    code/log
    Ignored:    code/logs/
    Ignored:    code/notebooks/.ipynb_checkpoints/
    Ignored:    code/pi1/
    Ignored:    code/rules/.ipynb_checkpoints/
    Ignored:    code/rules/.molQTL_GWAS_QQ.smk.swp
    Ignored:    code/rules/OldRules/
    Ignored:    code/rules/notebooks/
    Ignored:    code/scratch/
    Ignored:    code/scripts/.ipynb_checkpoints/
    Ignored:    code/scripts/GTFtools_0.8.0/
    Ignored:    code/scripts/__pycache__/
    Ignored:    code/scripts/liftOverBedpe/liftOverBedpe.py
    Ignored:    code/snakemake.dryrun.log
    Ignored:    code/snakemake.log
    Ignored:    code/snakemake.sbatch.log
    Ignored:    code/snakemake_profiles/slurm/__pycache__/
    Ignored:    code/test.introns.bed
    Ignored:    code/test.introns2.bed
    Ignored:    code/test.log
    Ignored:    code/tracks.xml
    Ignored:    data/.DS_Store
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-10.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-11.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-2.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-3.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-4.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-5.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-6.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-7.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-8.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022.csv
    Ignored:    data/Metaplots/.DS_Store

Untracked files:
    Untracked:  analysis/20230417_SusieFinemapEnrichment.Rmd
    Untracked:  analysis/20230420_FindeQTLsQTLGWAS_Colocs.Rmd
    Untracked:  code/rules/molQTL_GWAS_QQ.smk
    Untracked:  data/Ben_ncQTL.groups.tsv
    Untracked:  output/ColocPlotData.MVP.tsv.gz
    Untracked:  output/ColocPlotData.NUDT15.tsv.gz

Unstaged changes:
    Modified:   analysis/20230324_MakeFigs.Rmd
    Modified:   code/scripts/GenometracksByGenotype

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/20230324_MakeFigs.Rmd</code>) and
HTML (<code>docs/20230324_MakeFigs.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ddf600dbb95a36158c06a26c5d7df9d69c969ed0/analysis/20230324_MakeFigs.Rmd" target="_blank">ddf600d</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-04-14
</td>
<td>
add SM splicing timing analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/2ee57a6fc6f7d5ba04c337e50199bb1c45229e86/analysis/20230324_MakeFigs.Rmd" target="_blank">2ee57a6</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-04-13
</td>
<td>
update Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/e8e4a64f33cebe0e6d0e6a1da90995a383f735fc/analysis/20230324_MakeFigs.Rmd" target="_blank">e8e4a64</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-04-11
</td>
<td>
update make figs nb
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/e8e4a64f33cebe0e6d0e6a1da90995a383f735fc/docs/20230324_MakeFigs.html" target="_blank">e8e4a64</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-04-11
</td>
<td>
update make figs nb
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/3b44dfde49612d49977fafe369d3cbf25a8f5525/analysis/20230324_MakeFigs.Rmd" target="_blank">3b44dfd</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-30
</td>
<td>
add fig nb
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/3b44dfde49612d49977fafe369d3cbf25a8f5525/docs/20230324_MakeFigs.html" target="_blank">3b44dfd</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-30
</td>
<td>
add fig nb
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ddbf2225e7f71ace4a67a2cdae69ece7e4026b66/analysis/20230324_MakeFigs.Rmd" target="_blank">ddbf222</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-27
</td>
<td>
add pi1 heatmaps nb
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/ddbf2225e7f71ace4a67a2cdae69ece7e4026b66/docs/20230324_MakeFigs.html" target="_blank">ddbf222</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-27
</td>
<td>
add pi1 heatmaps nb
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE, warning = F, message = F)

library(tidyverse)</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>✔ ggplot2 3.3.6     ✔ purrr   0.3.4
✔ tibble  3.1.7     ✔ dplyr   1.0.9
✔ tidyr   1.2.0     ✔ stringr 1.4.0
✔ readr   2.1.2     ✔ forcats 0.5.1</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(RColorBrewer)
library(data.table)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre class="r"><code>library(edgeR)</code></pre>
<pre><code>Loading required package: limma</code></pre>
<pre class="r"><code># Set theme
theme_set(
  theme_classic() +
  theme(text=element_text(size=16,  family=&quot;Helvetica&quot;)))

# I use layer a lot, to rotate long x-axis labels
Rotate_x_labels &lt;- theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))

#test plot
ggplot(mtcars, aes(x=mpg, y=cyl)) +
  geom_point()</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/setup-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-setup-1">
Past versions of setup-1.png
</button>
</p>
<div id="fig-setup-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ddbf2225e7f71ace4a67a2cdae69ece7e4026b66/docs/figure/20230324_MakeFigs.Rmd/setup-1.png" target="_blank">ddbf222</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="intro" class="section level2">
<h2>Intro</h2>
<p>I notebook to make figure panels for publication…</p>
<p>Figures to make:</p>
<ol style="list-style-type: decimal">
<li><p>Placeholder, Nascent RNA reveals abundance of NMD-destined
transcripts</p></li>
<li><p>sQTLs, eQTLs</p></li>
</ol>
<p><input type="checkbox" unchecked></input>a. pi heatmap across
promoter hQTLs, H3K36me3, chRNA, 4sU, polyA. Number of QTLs in
margin</p>
<p><input type="checkbox" unchecked></input>b. Enrichment of annotations
in transcriptional vs post-transcriptional eQTLs. Classify
transcriptional vs post-transcriptional by comparing hQTL to eQTL
significance (or effect size).</p>
<p><input type="checkbox" unchecked></input>c. QQ-plot of eQTL signal,
grouped by hQTLs, sQTLs that affect productive junctions, and sQTLs that
affect unproductive junctions</p>
<p><input type="checkbox" unchecked></input>d. polyA sQTL beta vs polyA
eQTL beta scatter, faceted by stable vs unstable.</p>
<p><input type="checkbox" unchecked></input>e. Beta vs beta spearman
slope with different lines for chRNA, 4sU, polyA, faceted by stable vs
unstable</p>
<ol start="3" style="list-style-type: decimal">
<li>Contribution of splicing-mediated decay to complex trait</li>
</ol>
<p><input type="checkbox" unchecked></input>a. A single GWAS trait (ie
MS) QQ plot, colored by splicing-mediated eQTLs, transcription-mediated
QTLs, productive-AS sQTLs</p>
<p><input type="checkbox" unchecked></input>b. Heatmap of similar
enrichment in A, but across many traits</p>
<p><input type="checkbox" unchecked></input>c. Example locus showing
colocalization eQTL/sQTL/GWAS scatter, and IGV screenshot of the
splicing and eQTL effect</p>
<ol start="4" style="list-style-type: decimal">
<li>The effect of splice-modifying drugs on gene expression</li>
</ol>
<p><input type="checkbox" unchecked></input>a. Cartoon showing inset
with risdiplam bulge repair, along with dose:response trend showing the
tested doses and the increasing response in GA|GT but not AG|GT
splicing.</p>
<p><input type="checkbox" unchecked></input>b. Scatter that shows
splicing vs expression effect, faeted by productive unproductive. Or
something like that</p>
<p><input type="checkbox" unchecked></input>c. Something that shows the
4B effect is not in chRNA</p>
<p><input type="checkbox" unchecked></input>d. An illustrative example
effect in IGV.</p>
<p>Supplemental Figures to make</p>
<ol style="list-style-type: decimal">
<li><p>Placeholder</p></li>
<li><p>sQTLs, eQTLs</p></li>
</ol>
<p><input type="checkbox" unchecked></input>a. Total number of QTLs in
each type of type of trait. Bargraph, with multiple lines for different
FDR levels</p>
<p><input type="checkbox" unchecked></input>b. Correlation of effect
sizes across promoter hQTLs, H3K36me3, chRNA, 4sU, polyA</p>
<p><input type="checkbox" unchecked></input>i. Need to select set of
eQTLs to plot… Previous versions of this plot I tend to just pick
colocalized things. That would be easiest.</p>
<p><input type="checkbox" unchecked></input>c. sQTL QQ plot, with
different colors for post-transcriptional eQTLs to transcriptional eQTLs
(like what Carlos has done in red and black)</p>
<p><input type="checkbox" unchecked></input>d. eQTL QQ plot, with
different colors for H3K27AC QTLs and sQTLs</p>
<p><input type="checkbox" unchecked></input>e. lead gene-wise sQTLs are
not strongly enriched for eQTLs compared to control SNPs. eQTL QQ-plot
with top gene-wise sQTL SNP, compared to some controls. Or do this with
a pi1, and can use various thresholds</p>
<p><input type="checkbox" unchecked></input>f. enrichment of eQTL
signals is explained by both primary and non-primary eQTL signals
(FigX). More precisely, we used the SuSiE finemapping approach which
does not assume a single causal SNP for each expression trait, and can
thus be used to finemap multiple eQTL signals per gene. We find that
approximately x% of primary signals could be overlapped with sQTLs,
compared to y% of primary signals that we could overlap to hQTLs.
Furthermore, x% and y% of non primary signals could be mapped sQTLs and
hQTLs, respectively</p>
<p><input type="checkbox" unchecked></input>g. More extensive version of
2d, so we can explain the nuances in filtering out transcriptional
effects, and there are some lingering non-NMD effects that are degraded
in cytoplasm</p>
<ol start="3" style="list-style-type: decimal">
<li>Contribution of splicing-mediated decay to complex trait</li>
</ol>
<p><input type="checkbox" unchecked></input>a. Colocalization framework
cartoon</p>
<p><input type="checkbox" unchecked></input>b. Version of 3b that uses
colocalization. Shown as barplots similar to Phoenix’s paper</p>
<ol start="4" style="list-style-type: decimal">
<li>The effect of splice-modifying drugs on gene expression</li>
</ol>
<p><input type="checkbox" unchecked></input>a. In total, we observe x%
of observed non-canonincal GA-GT splice sites are activated with x
criteria… ecdf of spearman correlation coefficient, for different groups
of 5’ss.</p>
</div>
<div id="figure2" class="section level2">
<h2>Figure2</h2>
<div id="a" class="section level3">
<h3>A</h3>
<p>pi heatmap of gene-feature QTLs. Num QTLs in margin.</p>
<p>Only include hQTLs at promoter, chRNA, 4sU, and polyA.</p>
<pre class="r"><code>library(qvalue)
library(viridis)

PeaksToTSS &lt;- Sys.glob(&quot;../code/Misc/PeaksClosestToTSS/*_assigned.tsv.gz&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/Misc/PeaksClosestToTSS/(.+?)_assigned.tsv.gz&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(read_tsv) %&gt;%
  bind_rows(.id=&quot;ChromatinMark&quot;) %&gt;%
  mutate(GenePeakPair = paste(gene, peak, sep = &quot;;&quot;)) %&gt;%
  distinct(ChromatinMark, peak, gene, .keep_all=T)

PC1.filter &lt;- c(&quot;Expression.Splicing&quot;, &quot;MetabolicLabelled.30min&quot;, &quot;MetabolicLabelled.60min&quot;, &quot;H3K36ME3&quot;, &quot;H3K27AC&quot;, &quot;H3K4ME3&quot;, &quot;H3K4ME1&quot;, &quot;Expression.Splicing.Subset_YRI&quot;, &quot;chRNA.Expression.Splicing&quot;)

dat &lt;- fread(&quot;../code/pi1/PairwisePi1Traits.P.all.txt.gz&quot;) %&gt;%
  filter((PC1 %in% PC1.filter) &amp; (PC2 %in% PC1.filter))

dat$PC1 %&gt;% unique()</code></pre>
<pre><code>[1] &quot;Expression.Splicing&quot;            &quot;Expression.Splicing.Subset_YRI&quot;
[3] &quot;chRNA.Expression.Splicing&quot;      &quot;MetabolicLabelled.30min&quot;       
[5] &quot;MetabolicLabelled.60min&quot;        &quot;H3K27AC&quot;                       
[7] &quot;H3K4ME3&quot;                        &quot;H3K4ME1&quot;                       
[9] &quot;H3K36ME3&quot;                      </code></pre>
<pre class="r"><code>PhenotypeAliases &lt;- read_tsv(&quot;../data/Phenotypes_recode_for_Plotting.txt&quot;)</code></pre>
<p>Many genes have multiple promoters that should all be accounted for
as potential transcriptional effects. So calculating pi1 can be tricky.
I want to ask the question, “What fraction of eQTLs have a promoter hQTL
in at least one of the promoters?”… I have already solved this problem…
For each gene, get lowest P value for across tests (ie all the
promoters) and use that a a test statistic to come up with a new Pvalue,
to calculate pi1. The test statistic distribution under the null can be
approximate by drawing from uniform n times.</p>
<p>First we need to approximate that null using <code>runif</code>…</p>
<pre class="r"><code>MaxSampleSizeToCreateANull &lt;- 250
NSamplesToEstimateDistribution &lt;- 10000
NullSimulatedTestStats &lt;- matrix(nrow=MaxSampleSizeToCreateANull, ncol=NSamplesToEstimateDistribution)
rownames(NullSimulatedTestStats) &lt;- paste0(&quot;runif_samplesize&quot;, 1:MaxSampleSizeToCreateANull)
colnames(NullSimulatedTestStats) &lt;- paste0(&quot;Sample_&quot;, 1:NSamplesToEstimateDistribution)


for (i in 1:MaxSampleSizeToCreateANull){
  SampleSizeFromUniform &lt;- i
  SampledDat &lt;- matrix(runif(SampleSizeFromUniform*NSamplesToEstimateDistribution), nrow=NSamplesToEstimateDistribution)
  SampleDatNullTestStatistics &lt;- -log10(apply(SampledDat, 1, min))
  NullSimulatedTestStats[i,] &lt;- SampleDatNullTestStatistics
}


NullSimulatedTestStats %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;runif_samplesize&quot;) %&gt;%
  slice(1:250) %&gt;%
  mutate(runif_samplesize = as.numeric(str_remove(runif_samplesize, &quot;runif_samplesize&quot;))) %&gt;%
  gather(key=&quot;Sample&quot;, value=&quot;value&quot;, -runif_samplesize) %&gt;%
  ggplot(aes(x=value, color=runif_samplesize, group=runif_samplesize)) +
  geom_density() +
  scale_color_viridis_c() +
  theme_bw() +
  labs(x=&quot;Test statistic under null\n-log10(MinP) across n draws from uniform&quot;, color=&quot;n&quot;)</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ddbf2225e7f71ace4a67a2cdae69ece7e4026b66/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-2-1.png" target="_blank">ddbf222</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ecdf.functions &lt;- apply(NullSimulatedTestStats, 1, ecdf)
ecdf.functions[[1]](1)</code></pre>
<pre><code>[1] 0.9034</code></pre>
<p>Make a function to calculate pi1</p>
<pre class="r"><code># pi0est function breaks sometimes. this wrapper function returns 1 when that happens
CalculatePi1 &lt;- function (dat.in, ...) {
  return(tryCatch(1-pi0est(dat.in$Pvals.For.Pi1, ...)$pi0, error=function(e) 1))
}</code></pre>
<p>And now, recode peaks as promoters, and calculate test
statistics.</p>
<pre class="r"><code>Recode.PCs = c(&quot;ActivatingMark_HostGenePromoter&quot;=&quot;Promoter marks*&quot;, &quot;H3K36ME3&quot;=&quot;H3K36ME3&quot;, &quot;chRNA.Expression.Splicing&quot;=&quot;chRNA&quot;, &quot;MetabolicLabelled.30min&quot;=&quot;4sU 30min&quot;, &quot;MetabolicLabelled.60min&quot;=&quot;4sU 60min&quot;, &quot;Expression.Splicing.Subset_YRI&quot;=&quot;polyA YRI&quot;, &quot;Expression.Splicing&quot;=&quot;polyA&quot;)

ActivatingMarksToConsider &lt;- c(&quot;H3K27AC&quot;)

ProcessPi1HeatmapDat &lt;- function(FDR.cutoff=0.1){
  dat.split &lt;- dat %&gt;%
  filter(FDR.x &lt; FDR.cutoff) %&gt;%
  mutate(PC1 = case_when(
    paste(GeneLocus, P1, sep=&quot;;&quot;) %in% PeaksToTSS$GenePeakPair &amp; PC1 %in% ActivatingMarksToConsider ~ &quot;ActivatingMark_HostGenePromoter&quot;,
    P1 %in% PeaksToTSS$peak ~ &quot;ActivatingMark_OtherGenePromoter&quot;,
    PC1 %in% c(&quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;H3K27AC&quot;) ~ &quot;ActivatingMark_Enhancer&quot;,
    TRUE ~ PC1
  )) %&gt;%
  mutate(PC2 = case_when(
    paste(GeneLocus, P2, sep=&quot;;&quot;) %in% PeaksToTSS$GenePeakPair &amp; PC2 %in% ActivatingMarksToConsider ~ &quot;ActivatingMark_HostGenePromoter&quot;,
    P2 %in% PeaksToTSS$peak ~ &quot;ActivatingMark_OtherGenePromoter&quot;,
    PC2 %in% c(&quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;H3K27AC&quot;) ~ &quot;ActivatingMark_Enhancer&quot;,
    TRUE ~ PC2
  )) %&gt;%
  group_by(PC1, P1, PC2) %&gt;%
  mutate(test.stat.obs = -log10(min(trait.x.p.in.y))) %&gt;%
  ungroup() %&gt;%
  add_count(PC1, P1, PC2) %&gt;%
  filter(n&lt;=250) %&gt;%
  group_by(PC1, PC2) %&gt;%
  rowwise() %&gt;%
  mutate(Pvals.For.Pi1 = 1-ecdf.functions[[n]](test.stat.obs)) %&gt;%
  ungroup() %&gt;%
  select(PC1, PC2, Pvals.For.Pi1) %&gt;%
  filter(!PC1==PC2) %&gt;%
  split(paste(.$PC1, .$PC2, sep = &quot;;&quot;))
dat.out &lt;- lapply(dat.split, CalculatePi1, pi0.method=&quot;bootstrap&quot;) %&gt;%
  unlist() %&gt;%
  data.frame(pi1=.) %&gt;%
  rownames_to_column(&quot;PC1_PC2&quot;) %&gt;%
  separate(PC1_PC2, into=c(&quot;PC1&quot;, &quot;PC2&quot;), sep=&#39;;&#39;) %&gt;%
  filter((PC1%in% c(PC1.filter, &quot;ActivatingMark_HostGenePromoter&quot;) &amp; (PC2%in% c(PC1.filter, &quot;ActivatingMark_HostGenePromoter&quot;) ))) %&gt;%
  mutate(TextColor = case_when(
    pi1 &gt;= 0.7 ~ &quot;white&quot;,
    TRUE ~ &quot;black&quot;
  )) %&gt;%
mutate(PC1 = recode(PC1, !!!Recode.PCs)) %&gt;%
mutate(PC2 = recode(PC2, !!!Recode.PCs)) %&gt;%
mutate(PC1 = factor(PC1, levels=Recode.PCs)) %&gt;%
mutate(PC2 = factor(PC2, levels=Recode.PCs))
P &lt;- ggplot(dat.out, aes(x=PC1, y=PC2, fill=pi1, color=TextColor)) +
  geom_raster() +
  geom_text(aes(label=signif(pi1*100, 2))) +
  scale_color_identity() +
  scale_fill_viridis_c(option=&quot;A&quot;, direction = -1, limits=c(0,1)) +
  coord_flip() +
  scale_x_discrete(limits=rev) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(x=&quot;Discovery QTL phenotype&quot;, y=&quot;Phenotype assessed for overlap&quot;, caption=paste(&quot;*Promoter marks include&quot;, paste(ActivatingMarksToConsider, collapse=&quot;,&quot;)))
return(list(dat=dat, P=P))
}

ProcessPi1HeatmapDat()$P</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/3b44dfde49612d49977fafe369d3cbf25a8f5525/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-4-1.png" target="_blank">3b44dfd</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ddbf2225e7f71ace4a67a2cdae69ece7e4026b66/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-4-1.png" target="_blank">ddbf222</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Redo, but also consider both H3K27AC and H3K4ME3 at promoter…</p>
<pre class="r"><code>ActivatingMarksToConsider &lt;- c(&quot;H3K27AC&quot;, &quot;H3K4ME3&quot;)

ProcessPi1HeatmapDat()$P</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/3b44dfde49612d49977fafe369d3cbf25a8f5525/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-5-1.png" target="_blank">3b44dfd</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ddbf2225e7f71ace4a67a2cdae69ece7e4026b66/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-5-1.png" target="_blank">ddbf222</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Redo but consider H3K4ME3, H3K4ME1, and H3K27AC</p>
<pre class="r"><code>ActivatingMarksToConsider &lt;- c(&quot;H3K27AC&quot;, &quot;H3K4ME3&quot;, &quot;H3K4ME1&quot;)

ProcessPi1HeatmapDat()$P</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/3b44dfde49612d49977fafe369d3cbf25a8f5525/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-6-1.png" target="_blank">3b44dfd</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ddbf2225e7f71ace4a67a2cdae69ece7e4026b66/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-6-1.png" target="_blank">ddbf222</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>In theory things should only go up as you include H3K4ME1, but that
is not the case. This phenomena is something I’ve noticed before, and I
think it’s because adding non-overlapping phenotypes to promoter marks
only modestly increases the lowest P-value, but more greatly can
increase the n number of features used to determine the null.</p>
<p>So, let’s now include H3K4ME1. This is more thought of as an enhancer
rather than promoter mark anyway.</p>
<p>So now let’s redo, but only include H3K27AC and H3K4ME3 but use lower
FDR cutoffs.</p>
<pre class="r"><code>ActivatingMarksToConsider &lt;- c(&quot;H3K27AC&quot;, &quot;H3K4ME3&quot;)

ProcessPi1HeatmapDat(FDR.cutoff = 0.05)$P</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/3b44dfde49612d49977fafe369d3cbf25a8f5525/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-7-1.png" target="_blank">3b44dfd</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ddbf2225e7f71ace4a67a2cdae69ece7e4026b66/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-7-1.png" target="_blank">ddbf222</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ProcessPi1HeatmapDat(FDR.cutoff = 0.01)$P</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-2">
Past versions of unnamed-chunk-7-2.png
</button>
</p>
<div id="fig-unnamed-chunk-7-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/3b44dfde49612d49977fafe369d3cbf25a8f5525/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-7-2.png" target="_blank">3b44dfd</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ddbf2225e7f71ace4a67a2cdae69ece7e4026b66/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-7-2.png" target="_blank">ddbf222</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok, cool. I mean slightly different versions. I’ll leave it to Yang
as to which version should be the main figure, and perhaps we can
include alternate versions as supplements. In every case though I notice
that the top right corner is dark, suggesting histone effects can be
picked up pretty well in downstream assays (ie polyA), but the bottom
left corner is light, because of post-transcriptional mechanisms of gene
regulation.</p>
<ul class="task-list">
<li><input type="checkbox" disabled="" />save final prettier versions,
with axes flipped (yang likes discovery on x-axis)</li>
<li><input type="checkbox" disabled="" />save alternate versions for
supplement, with different FDR thresholds</li>
</ul>
</div>
<div id="b" class="section level3">
<h3>B</h3>
<p>Now I need a way to classify eQTLs as transcriptional or
post-transcriptional. Carlos has previously used effect sizes in polyA
vs chRNA. I may do something similar. First let’s some scatter plots of
effect sizes</p>
<pre class="r"><code>FDR.cutoff &lt;- 0.1
ActivatingMarksToConsider &lt;- c(&quot;H3K27AC&quot;, &quot;H3K4ME3&quot;)

dat.recoded &lt;- dat %&gt;%
  filter(FDR.x &lt; FDR.cutoff) %&gt;%
  mutate(PC1 = case_when(
    paste(GeneLocus, P1, sep=&quot;;&quot;) %in% PeaksToTSS$GenePeakPair &amp; PC1 %in% ActivatingMarksToConsider ~ PC1,
    P1 %in% PeaksToTSS$peak ~ &quot;ActivatingMark_OtherGenePromoter&quot;,
    PC1 %in% c(&quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;H3K27AC&quot;) ~ &quot;ActivatingMark_Enhancer&quot;,
    TRUE ~ PC1
  )) %&gt;%
  mutate(PC2 = case_when(
    paste(GeneLocus, P2, sep=&quot;;&quot;) %in% PeaksToTSS$GenePeakPair &amp; PC2 %in% ActivatingMarksToConsider ~ PC2,
    P1 %in% PeaksToTSS$peak ~ &quot;ActivatingMark_OtherGenePromoter&quot;,
    PC2 %in% c(&quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;H3K27AC&quot;) ~ &quot;ActivatingMark_Enhancer&quot;,
    TRUE ~ PC2
  ))

dat.recoded %&gt;%
  filter(PC1 == &quot;Expression.Splicing&quot;) %&gt;%
  filter(!str_detect(PC2, &quot;ActivatingMark_Enhancer&quot;)) %&gt;%
  filter(!str_detect(PC2, &quot;ActivatingMark_OtherGenePromoter&quot;)) %&gt;%
  group_by(PC1, P1, PC2) %&gt;%
  slice(which.min(trait.x.p.in.y)) %&gt;%
  ungroup() %&gt;%
  ggplot(aes(x=beta.x, y=x.beta.in.y)) +
  geom_point(alpha=0.1) +
  facet_wrap(~PC2) +
  labs(x=&quot;eQTL beta&quot;, y=&quot;facetQTL beta&quot;)</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/3b44dfde49612d49977fafe369d3cbf25a8f5525/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-8-1.png" target="_blank">3b44dfd</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>I think what I want to do is to first consider eQTLs in polyA, then
perform one-sided test for significant differences in beta for
corresponding feature in chRNA/promoterMarks. For simplicity, I think I
can use Z-test, like <a
href="https://stats.stackexchange.com/questions/55501/test-a-significant-difference-between-two-slope-values">this</a>,
but first, to check that this approach is reasonable, I think I’m going
to check that Z-scores from random trait pairs closely follow normal
distribution as they should under null. Also, I note that winners curse
will bias things since I am conditioning on significant QTLs for polyA,
then assessing difference in betas between polyA vs chRNA/promoter. But
since only the polyA beta is subject to winners curse bias, it will turn
out being overly convervative for the SNP enrichments I’m trying to
highlight (because some transcriptional QTLs might erronesouly be
classified as non-trancriptional) so i think that’s acceptable.</p>
<pre class="r"><code>PC1.filter = c(&quot;Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;)

dat.ZsFromAllTraitPairs &lt;- dat.recoded %&gt;%
  filter(PC1 %in% PC1.filter) %&gt;%
  filter(!str_detect(PC2, &quot;ActivatingMark_Enhancer&quot;)) %&gt;%
  filter(!str_detect(PC2, &quot;ActivatingMark_OtherGenePromoter&quot;)) %&gt;%
  group_by(PC1, P1, PC2) %&gt;%
  slice(which.min(trait.x.p.in.y)) %&gt;%
  ungroup() %&gt;%
  filter(!is.na(x.beta.in.y)) %&gt;%
  distinct(PC1, P1, PC2, .keep_all=T) %&gt;%
  mutate(Z = (abs(beta.x)-abs(x.beta.in.y)*sign(beta.x)*sign(x.beta.in.y))/sqrt(x.beta_se.in.y**2 + beta_se.x**2)) %&gt;%
  mutate(OneSidedZTest.P = pnorm(Z, lower.tail=F)) %&gt;%
  group_by(PC1, PC2) %&gt;%
  mutate(q = qvalue(OneSidedZTest.P)$qvalues) %&gt;%
  mutate(PolyASpecific = cut(q, breaks=c(0, 0.001, 0.01, 0.1, 1), include.lowest=T, ordered_result = T))


dat.ZsFromAllTraitPairs %&gt;%
ggplot(aes(x=beta.x, y=x.beta.in.y, color=PolyASpecific)) +
  geom_point(alpha=0.1) +
  scale_color_viridis_d(direction=-1) +
  facet_grid(PC1~PC2) +
  guides(colour = guide_legend(override.aes = list(alpha = 1))) +
  labs(color=&quot;PolyA-specific FDR&quot;) +
  theme(legend.position=&quot;bottom&quot;)</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/e8e4a64f33cebe0e6d0e6a1da90995a383f735fc/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-9-1.png" target="_blank">e8e4a64</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-04-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/3b44dfde49612d49977fafe369d3cbf25a8f5525/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-9-1.png" target="_blank">3b44dfd</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dat.ZsFromAllTraitPairs %&gt;%
ggplot(aes(x=beta.x, y=x.beta.in.y, color=PolyASpecific==&quot;[0,0.001]&quot;)) +
  geom_point(alpha=0.1) +
  facet_grid(PC1~PC2) +
  guides(colour = guide_legend(override.aes = list(alpha = 1))) +
  labs(color=&quot;PolyA-specific FDR &lt; 0.001&quot;) +
  theme(legend.position=&quot;bottom&quot;)</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-2">
Past versions of unnamed-chunk-9-2.png
</button>
</p>
<div id="fig-unnamed-chunk-9-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/e8e4a64f33cebe0e6d0e6a1da90995a383f735fc/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-9-2.png" target="_blank">e8e4a64</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-04-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/3b44dfde49612d49977fafe369d3cbf25a8f5525/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-9-2.png" target="_blank">3b44dfd</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dat.ZsFromAllTraitPairs %&gt;%
  count(PC1, PC2, PolyASpecific) %&gt;%
  ggplot(aes(x=PC2, y=n, fill=fct_rev(PolyASpecific))) +
  geom_col(position=&quot;stack&quot;) +
  facet_wrap(~PC1) +
  scale_fill_viridis_d(direction=1) +
  Rotate_x_labels</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-9-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-3">
Past versions of unnamed-chunk-9-3.png
</button>
</p>
<div id="fig-unnamed-chunk-9-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/e8e4a64f33cebe0e6d0e6a1da90995a383f735fc/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-9-3.png" target="_blank">e8e4a64</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-04-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/3b44dfde49612d49977fafe369d3cbf25a8f5525/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-9-3.png" target="_blank">3b44dfd</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dat.ZsFromAllTraitPairs %&gt;%
  count(PC1, PC2, PolyASpecific) %&gt;%
  ggplot(aes(x=PC2, y=n, fill=fct_rev(PolyASpecific))) +
  geom_col(position=&quot;fill&quot;) +
  facet_wrap(~PC1) +
  scale_fill_viridis_d(direction=1) +
  Rotate_x_labels</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-9-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-4">
Past versions of unnamed-chunk-9-4.png
</button>
</p>
<div id="fig-unnamed-chunk-9-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/e8e4a64f33cebe0e6d0e6a1da90995a383f735fc/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-9-4.png" target="_blank">e8e4a64</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-04-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/3b44dfde49612d49977fafe369d3cbf25a8f5525/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-9-4.png" target="_blank">3b44dfd</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Okay, keep in mind that there is both biology and power issues at
play… Biologically, I would expect very few polyA-specific eQTLs when
comparing to 4sU, yet there are basically the same amount of
polyA-specific eQTLs in that comparison as when comparing to chRNA. One
could chalk that up to the 4sU dataset being so underpowered, that a lot
more eQTLs would be considered polyA specific. Similarly, that could
explain why so many are polyA-specific when comparing to the relatively
underpowered H3K36me3 dataset.</p>
<p>Let’s explore Carlos’ susie finemapping output</p>
<pre class="r"><code>susie.finemap &lt;- fread(&quot;/project2/yangili1/cfbuenabadn/ChromatinSplicingQTLs/code/FineMapping/susie_runs_Geuvadis/susie_output.tab&quot;)
 
susie.finemap %&gt;%
  distinct(cs_gene, cs_names) %&gt;%
  count(cs_gene) %&gt;%
  ggplot(aes(x=n)) +
  geom_bar() +
  labs(y=&quot;Num genes&quot;, x=&quot;Num eQTL signals&quot;)</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/e8e4a64f33cebe0e6d0e6a1da90995a383f735fc/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-10-1.png" target="_blank">e8e4a64</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-04-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/3b44dfde49612d49977fafe369d3cbf25a8f5525/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-10-1.png" target="_blank">3b44dfd</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>susie.finemap %&gt;%
  group_by(cs_gene, cs_names) %&gt;%
  summarise(PIP.sum = sum(cs_pip)) %&gt;%
  ungroup() %&gt;%
  ggplot(aes(x=PIP.sum)) +
  stat_ecdf() +
  labs(x=&quot;Sum PIP in CI&quot;, title=&quot;These are 95% CIs&quot;) +
  coord_cartesian(xlim=c(0.9,1))</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-2">
Past versions of unnamed-chunk-10-2.png
</button>
</p>
<div id="fig-unnamed-chunk-10-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/e8e4a64f33cebe0e6d0e6a1da90995a383f735fc/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-10-2.png" target="_blank">e8e4a64</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-04-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/3b44dfde49612d49977fafe369d3cbf25a8f5525/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-10-2.png" target="_blank">3b44dfd</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>susie.finemap %&gt;%
  count(cs_gene, cs_names) %&gt;%
  ggplot(aes(x=n)) +
  stat_ecdf() +
  coord_cartesian(xlim=c(0,20)) +
  labs(x=&quot;Num SNPs in CI&quot;)</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-10-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-3">
Past versions of unnamed-chunk-10-3.png
</button>
</p>
<div id="fig-unnamed-chunk-10-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/e8e4a64f33cebe0e6d0e6a1da90995a383f735fc/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-10-3.png" target="_blank">e8e4a64</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-04-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/3b44dfde49612d49977fafe369d3cbf25a8f5525/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-10-3.png" target="_blank">3b44dfd</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok great, next I will intersect the finemapped SNPs with annotations,
and sum up the number of SNPs in each annotation (weighted by PIP), and
compare transcriptional vs post-transcriptional eQTLs. I did some work
in the snakemake to intersect Carlos’s finemap output with SNP
annotations and now let’s explore that output.</p>
<pre class="r"><code>finemappip.annotation.intersections &lt;- fread(&quot;../code/QTL_SNP_Enrichment/FinemapIntersections_Susie/YRI.bed.gz&quot;, col.names=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;name&quot;, &quot;pip&quot;,&quot;featChrom&quot;, &quot;featStart&quot;, &quot;featStop&quot;,&quot;Annotation&quot;, &quot;Overlap&quot;)) %&gt;%
  separate(name, into=c(&quot;snp&quot;, &quot;eGene&quot;, &quot;Signal&quot;), sep = &quot;_&quot;) %&gt;%
  mutate(Annotation = str_replace_all(Annotation, &quot;_0$&quot;, &quot;_UnannotatedButObserved&quot;)) %&gt;%
  mutate(Annotation = str_replace_all(Annotation, &quot;_1$&quot;, &quot;_Annotated&quot;)) %&gt;%
  mutate(Annotation = str_replace_all(Annotation, &quot;^\\d+?_&quot;, &quot;ChromHMM:&quot;)) %&gt;%
  filter(!is.na(Annotation)) %&gt;%
  mutate(Annotation = if_else(Annotation==&quot;.&quot;, &quot;No annotation&quot;, Annotation)) %&gt;%
  filter(!str_detect(Annotation, &quot;^ncRNA&quot;)) %&gt;%
  filter(!Annotation==&quot;No annotation&quot;)

transcriptional.PC2s &lt;- c(&quot;chRNA.Expression.Splicing&quot;, &quot;H3K27AC&quot;, &quot;H3K4ME3&quot;)

transcriptional.eQTLs &lt;- dat.ZsFromAllTraitPairs %&gt;%
  filter(PC2 %in% transcriptional.PC2s) %&gt;%
  mutate(IsPolyASpecific = if_else(q &lt; 0.001, &quot;PostTxn&quot;, &quot;Txn&quot;))

count(transcriptional.eQTLs,PC1, PC2, IsPolyASpecific)</code></pre>
<pre><code># A tibble: 12 × 4
# Groups:   PC1, PC2 [6]
   PC1                            PC2                      IsPolyASpecific     n
   &lt;chr&gt;                          &lt;chr&gt;                    &lt;chr&gt;           &lt;int&gt;
 1 Expression.Splicing            H3K27AC                  PostTxn           523
 2 Expression.Splicing            H3K27AC                  Txn              7751
 3 Expression.Splicing            H3K4ME3                  PostTxn           473
 4 Expression.Splicing            H3K4ME3                  Txn              8447
 5 Expression.Splicing            chRNA.Expression.Splici… PostTxn           565
 6 Expression.Splicing            chRNA.Expression.Splici… Txn              9275
 7 Expression.Splicing.Subset_YRI H3K27AC                  PostTxn           760
 8 Expression.Splicing.Subset_YRI H3K27AC                  Txn              2610
 9 Expression.Splicing.Subset_YRI H3K4ME3                  PostTxn           643
10 Expression.Splicing.Subset_YRI H3K4ME3                  Txn              2935
11 Expression.Splicing.Subset_YRI chRNA.Expression.Splici… PostTxn           746
12 Expression.Splicing.Subset_YRI chRNA.Expression.Splici… Txn              3167</code></pre>
<pre class="r"><code>finemappip.annotation.intersections %&gt;%
  filter(Signal == &quot;L1&quot;) %&gt;%
  left_join(transcriptional.eQTLs, by=c(&quot;eGene&quot;=&quot;P1&quot;)) %&gt;%
  filter(!is.na(IsPolyASpecific)) %&gt;%
  group_by(Annotation, IsPolyASpecific, PC1, PC2) %&gt;%
  summarise(SumInAnnotation = sum(pip, na.rm=T)) %&gt;%
  ungroup() %&gt;%
  group_by(IsPolyASpecific, PC1, PC2) %&gt;%
  mutate(TotalPIP = sum(SumInAnnotation, na.rm=T)) %&gt;%
  ungroup() %&gt;%
  mutate(FractionPIP_in_Annotation = SumInAnnotation/TotalPIP ) %&gt;%
  dplyr::select(FractionPIP_in_Annotation, PC1, PC2, IsPolyASpecific, Annotation) %&gt;%
  pivot_wider(names_from = &quot;IsPolyASpecific&quot;, values_from = c(&quot;FractionPIP_in_Annotation&quot;)) %&gt;%
  mutate(Enrichment = log2(PostTxn/Txn)) %&gt;%
  ggplot(aes(x=Annotation, y=Enrichment)) +
  geom_col() +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  coord_flip() +
  facet_grid(PC2~PC1) +
  labs(x = &quot;Annotation&quot;, y=&quot;Finemapping posterior mass fold enrichment\n(Txn &lt;-- --&gt; PostTxn)&quot;)</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/e8e4a64f33cebe0e6d0e6a1da90995a383f735fc/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-11-1.png" target="_blank">e8e4a64</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-04-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/3b44dfde49612d49977fafe369d3cbf25a8f5525/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-11-1.png" target="_blank">3b44dfd</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>set.seed(0)
results &lt;- list()
for (i in 1:100){
  print(paste(&quot;Bootstrap resample&quot;, i))
  transcriptional.eQTLs.sample &lt;- transcriptional.eQTLs %&gt;%
  group_by(PC1, PC2, IsPolyASpecific) %&gt;%
  sample_frac(replace=T)
  results[[i]] &lt;-
  finemappip.annotation.intersections %&gt;%
  filter(Signal == &quot;L1&quot;) %&gt;%
  left_join(transcriptional.eQTLs.sample, by=c(&quot;eGene&quot;=&quot;P1&quot;)) %&gt;%
  filter(!is.na(IsPolyASpecific)) %&gt;%
  group_by(Annotation, IsPolyASpecific, PC1, PC2) %&gt;%
  summarise(SumInAnnotation = sum(pip, na.rm=T)) %&gt;%
  ungroup() %&gt;%
  group_by(IsPolyASpecific, PC1, PC2) %&gt;%
  mutate(TotalPIP = sum(SumInAnnotation, na.rm=T)) %&gt;%
  ungroup() %&gt;%
  mutate(FractionPIP_in_Annotation = SumInAnnotation/TotalPIP ) %&gt;%
  dplyr::select(FractionPIP_in_Annotation, PC1, PC2, IsPolyASpecific, Annotation) %&gt;%
  pivot_wider(names_from = &quot;IsPolyASpecific&quot;, values_from = c(&quot;FractionPIP_in_Annotation&quot;)) %&gt;%
  mutate(Enrichment = log2(PostTxn/Txn)) %&gt;%
  mutate(i=i)
}</code></pre>
<pre><code>[1] &quot;Bootstrap resample 1&quot;
[1] &quot;Bootstrap resample 2&quot;
[1] &quot;Bootstrap resample 3&quot;
[1] &quot;Bootstrap resample 4&quot;
[1] &quot;Bootstrap resample 5&quot;
[1] &quot;Bootstrap resample 6&quot;
[1] &quot;Bootstrap resample 7&quot;
[1] &quot;Bootstrap resample 8&quot;
[1] &quot;Bootstrap resample 9&quot;
[1] &quot;Bootstrap resample 10&quot;
[1] &quot;Bootstrap resample 11&quot;
[1] &quot;Bootstrap resample 12&quot;
[1] &quot;Bootstrap resample 13&quot;
[1] &quot;Bootstrap resample 14&quot;
[1] &quot;Bootstrap resample 15&quot;
[1] &quot;Bootstrap resample 16&quot;
[1] &quot;Bootstrap resample 17&quot;
[1] &quot;Bootstrap resample 18&quot;
[1] &quot;Bootstrap resample 19&quot;
[1] &quot;Bootstrap resample 20&quot;
[1] &quot;Bootstrap resample 21&quot;
[1] &quot;Bootstrap resample 22&quot;
[1] &quot;Bootstrap resample 23&quot;
[1] &quot;Bootstrap resample 24&quot;
[1] &quot;Bootstrap resample 25&quot;
[1] &quot;Bootstrap resample 26&quot;
[1] &quot;Bootstrap resample 27&quot;
[1] &quot;Bootstrap resample 28&quot;
[1] &quot;Bootstrap resample 29&quot;
[1] &quot;Bootstrap resample 30&quot;
[1] &quot;Bootstrap resample 31&quot;
[1] &quot;Bootstrap resample 32&quot;
[1] &quot;Bootstrap resample 33&quot;
[1] &quot;Bootstrap resample 34&quot;
[1] &quot;Bootstrap resample 35&quot;
[1] &quot;Bootstrap resample 36&quot;
[1] &quot;Bootstrap resample 37&quot;
[1] &quot;Bootstrap resample 38&quot;
[1] &quot;Bootstrap resample 39&quot;
[1] &quot;Bootstrap resample 40&quot;
[1] &quot;Bootstrap resample 41&quot;
[1] &quot;Bootstrap resample 42&quot;
[1] &quot;Bootstrap resample 43&quot;
[1] &quot;Bootstrap resample 44&quot;
[1] &quot;Bootstrap resample 45&quot;
[1] &quot;Bootstrap resample 46&quot;
[1] &quot;Bootstrap resample 47&quot;
[1] &quot;Bootstrap resample 48&quot;
[1] &quot;Bootstrap resample 49&quot;
[1] &quot;Bootstrap resample 50&quot;
[1] &quot;Bootstrap resample 51&quot;
[1] &quot;Bootstrap resample 52&quot;
[1] &quot;Bootstrap resample 53&quot;
[1] &quot;Bootstrap resample 54&quot;
[1] &quot;Bootstrap resample 55&quot;
[1] &quot;Bootstrap resample 56&quot;
[1] &quot;Bootstrap resample 57&quot;
[1] &quot;Bootstrap resample 58&quot;
[1] &quot;Bootstrap resample 59&quot;
[1] &quot;Bootstrap resample 60&quot;
[1] &quot;Bootstrap resample 61&quot;
[1] &quot;Bootstrap resample 62&quot;
[1] &quot;Bootstrap resample 63&quot;
[1] &quot;Bootstrap resample 64&quot;
[1] &quot;Bootstrap resample 65&quot;
[1] &quot;Bootstrap resample 66&quot;
[1] &quot;Bootstrap resample 67&quot;
[1] &quot;Bootstrap resample 68&quot;
[1] &quot;Bootstrap resample 69&quot;
[1] &quot;Bootstrap resample 70&quot;
[1] &quot;Bootstrap resample 71&quot;
[1] &quot;Bootstrap resample 72&quot;
[1] &quot;Bootstrap resample 73&quot;
[1] &quot;Bootstrap resample 74&quot;
[1] &quot;Bootstrap resample 75&quot;
[1] &quot;Bootstrap resample 76&quot;
[1] &quot;Bootstrap resample 77&quot;
[1] &quot;Bootstrap resample 78&quot;
[1] &quot;Bootstrap resample 79&quot;
[1] &quot;Bootstrap resample 80&quot;
[1] &quot;Bootstrap resample 81&quot;
[1] &quot;Bootstrap resample 82&quot;
[1] &quot;Bootstrap resample 83&quot;
[1] &quot;Bootstrap resample 84&quot;
[1] &quot;Bootstrap resample 85&quot;
[1] &quot;Bootstrap resample 86&quot;
[1] &quot;Bootstrap resample 87&quot;
[1] &quot;Bootstrap resample 88&quot;
[1] &quot;Bootstrap resample 89&quot;
[1] &quot;Bootstrap resample 90&quot;
[1] &quot;Bootstrap resample 91&quot;
[1] &quot;Bootstrap resample 92&quot;
[1] &quot;Bootstrap resample 93&quot;
[1] &quot;Bootstrap resample 94&quot;
[1] &quot;Bootstrap resample 95&quot;
[1] &quot;Bootstrap resample 96&quot;
[1] &quot;Bootstrap resample 97&quot;
[1] &quot;Bootstrap resample 98&quot;
[1] &quot;Bootstrap resample 99&quot;
[1] &quot;Bootstrap resample 100&quot;</code></pre>
<pre class="r"><code>finemappip.annotation.intersections$Annotation %&gt;% unique()</code></pre>
<pre><code> [1] &quot;ChromHMM:Weak_Promoter&quot;                        
 [2] &quot;ChromHMM:Active_Promoter&quot;                      
 [3] &quot;ChromHMM:Heterochrom/lo&quot;                       
 [4] &quot;ChromHMM:Repressed&quot;                            
 [5] &quot;SpliceBranchpointRegion_UnannotatedButObserved&quot;
 [6] &quot;ChromHMM:Txn_Elongation&quot;                       
 [7] &quot;SpliceAcceptor_UnannotatedButObserved&quot;         
 [8] &quot;SpliceDonor_UnannotatedButObserved&quot;            
 [9] &quot;ChromHMM:Txn_Transition&quot;                       
[10] &quot;SpliceDonor_Annotated&quot;                         
[11] &quot;ChromHMM:Strong_Enhancer&quot;                      
[12] &quot;ChromHMM:Weak_Enhancer&quot;                        
[13] &quot;ChromHMM:Weak_Txn&quot;                             
[14] &quot;SpliceAcceptor_Annotated&quot;                      
[15] &quot;SpliceBranchpointRegion_Annotated&quot;             
[16] &quot;PAS_Region&quot;                                    
[17] &quot;ChromHMM:Repetitive/CNV&quot;                       
[18] &quot;ChromHMM:Poised_Promoter&quot;                      
[19] &quot;ChromHMM:Insulator&quot;                            
[20] &quot;miRNA_BS&quot;                                      </code></pre>
<pre class="r"><code>bind_rows(results) %&gt;%
  # filter(PC1 == &quot;Expression.Splicing&quot; &amp; PC2 == &quot;H3K27AC&quot; &amp; Annotation == &quot;ChromHMM:Active_Promoter&quot;)
  group_by(Annotation, PC1, PC2) %&gt;%
  summarise(mean = mean(Enrichment, na.rm=T),
            Percentile10 = quantile(Enrichment, probs = 0.05, na.rm=T),
            Percentile90 = quantile(Enrichment, probs = 0.95, na.rm=T)) %&gt;%
  mutate(color = case_when(
    str_detect(Annotation, &quot;^Splic&quot;) ~ &quot;Splice region&quot;,
    str_detect(Annotation, &quot;Enhancer&quot;) ~ &quot;Enhancer&quot;,
    str_detect(Annotation, &quot;Promoter&quot;) ~ &quot;Promoter&quot;,
    str_detect(Annotation, &quot;Txn&quot;) ~ &quot;Transcribed&quot;,
    str_detect(Annotation, &quot;ChromHMM&quot;) ~ &quot;Other ChromHMM&quot;,
    TRUE ~ &quot;Other&quot;
  )) %&gt;%
  ggplot(aes(x=Annotation, y=mean, color=color)) +
    geom_hline(yintercept = 0) +
    geom_point() +
    geom_errorbar(aes(ymin=Percentile10, ymax=Percentile90)) +
    theme_bw() +
    theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
    coord_flip() +
    facet_grid(PC2~PC1) +
    labs(x = &quot;Annotation&quot;, y=&quot;Finemapping posterior mass fold enrichment\n(ChromatinQTL|eQTL &lt;-- --&gt; sQTL)&quot;,
         caption = &quot;Bootsrapped 90% confidence interval&quot;)</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-2">
Past versions of unnamed-chunk-11-2.png
</button>
</p>
<div id="fig-unnamed-chunk-11-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/e8e4a64f33cebe0e6d0e6a1da90995a383f735fc/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-11-2.png" target="_blank">e8e4a64</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-04-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/3b44dfde49612d49977fafe369d3cbf25a8f5525/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-11-2.png" target="_blank">3b44dfd</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Hmm, the results aren’t quite what I was expecting, and they aren’t
so consistent across the different ways to classify eQTLs as post txn vs
txn (H3K27AC, H3K4ME3, chRNA). I wonder how consistent the
classifications are…</p>
<pre class="r"><code>transcriptional.eQTLs &lt;- dat.ZsFromAllTraitPairs %&gt;%
  filter(PC2 %in% transcriptional.PC2s) %&gt;%
  mutate(IsPolyASpecific = if_else(q &lt; 0.01, &quot;PostTxn&quot;, &quot;Txn&quot;))

transcriptional.eQTLs %&gt;%
  group_by(PC1) %&gt;%
  add_count(P1) %&gt;%
  filter(n==3) %&gt;%
  ungroup() %&gt;%
  mutate(IsPolyASpecific = IsPolyASpecific==&quot;Txn&quot;) %&gt;%
  group_by(P1, PC1) %&gt;%
  summarise(TxnQTL_In_HowManyOf3 = sum(IsPolyASpecific)) %&gt;%
  ggplot(aes(x=TxnQTL_In_HowManyOf3)) +
  geom_bar() +
  facet_wrap(~PC1) +
  labs(&quot;How many of the 3 methods is a gene classified as txn eQTL&quot;)</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/e8e4a64f33cebe0e6d0e6a1da90995a383f735fc/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-12-1.png" target="_blank">e8e4a64</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-04-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/3b44dfde49612d49977fafe369d3cbf25a8f5525/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-12-1.png" target="_blank">3b44dfd</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>transcriptional.eQTLs &lt;- dat.ZsFromAllTraitPairs %&gt;%
  filter(PC2 %in% transcriptional.PC2s) %&gt;%
  mutate(IsPolyASpecific = if_else(q &lt; 0.001, &quot;PostTxn&quot;, &quot;Txn&quot;))

transcriptional.eQTLs %&gt;%
  group_by(PC1) %&gt;%
  add_count(P1) %&gt;%
  filter(n==3) %&gt;%
  ungroup() %&gt;%
  mutate(IsPolyASpecific = IsPolyASpecific==&quot;Txn&quot;) %&gt;%
  group_by(P1, PC1) %&gt;%
  summarise(TxnQTL_In_HowManyOf3 = sum(IsPolyASpecific)) %&gt;%
  ggplot(aes(x=TxnQTL_In_HowManyOf3)) +
  geom_bar() +
  facet_wrap(~PC1) +
  labs(&quot;How many of the 3 methods is a gene classified as txn eQTL&quot;)</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-12-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-2">
Past versions of unnamed-chunk-12-2.png
</button>
</p>
<div id="fig-unnamed-chunk-12-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/e8e4a64f33cebe0e6d0e6a1da90995a383f735fc/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-12-2.png" target="_blank">e8e4a64</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-04-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>transcriptional.eQTLs &lt;- dat.ZsFromAllTraitPairs %&gt;%
  filter(PC2 %in% transcriptional.PC2s) %&gt;%
  mutate(IsPolyASpecific = if_else(q &lt; 0.0001, &quot;PostTxn&quot;, &quot;Txn&quot;))

transcriptional.eQTLs %&gt;%
  group_by(PC1) %&gt;%
  add_count(P1) %&gt;%
  filter(n==3) %&gt;%
  ungroup() %&gt;%
  mutate(IsPolyASpecific = IsPolyASpecific==&quot;Txn&quot;) %&gt;%
  group_by(P1, PC1) %&gt;%
  summarise(TxnQTL_In_HowManyOf3 = sum(IsPolyASpecific)) %&gt;%
  ggplot(aes(x=TxnQTL_In_HowManyOf3)) +
  geom_bar() +
  facet_wrap(~PC1) +
  labs(&quot;How many of the 3 methods is a gene classified as txn eQTL&quot;)</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-12-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-3">
Past versions of unnamed-chunk-12-3.png
</button>
</p>
<div id="fig-unnamed-chunk-12-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/e8e4a64f33cebe0e6d0e6a1da90995a383f735fc/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-12-3.png" target="_blank">e8e4a64</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-04-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok so playing with the thresholds still doesn’t get a nice
clean/consistent classification between the different methods… It’s not
like there are a lot of genes that are in txn class in all 3 and in
non-txn class in all 3. There are a lot of in-betweens. Let’s maybe just
try reclassifying them as txn if they are significant any of the
three…</p>
<pre class="r"><code>transcriptional.eQTLs &lt;- dat.ZsFromAllTraitPairs %&gt;%
  filter(PC2 %in% transcriptional.PC2s) %&gt;%
  group_by(PC1) %&gt;%
  add_count(P1) %&gt;%
  filter(n==3) %&gt;%
  ungroup() %&gt;%
  group_by(P1, PC1) %&gt;%
  mutate(IsPolyASpecific = if_else(any(q &lt; 0.001), &quot;PostTxn&quot;, &quot;Txn&quot;)) %&gt;%
  ungroup()

count(transcriptional.eQTLs,PC1, PC2, IsPolyASpecific)</code></pre>
<pre><code># A tibble: 12 × 4
   PC1                            PC2                      IsPolyASpecific     n
   &lt;chr&gt;                          &lt;chr&gt;                    &lt;chr&gt;           &lt;int&gt;
 1 Expression.Splicing            H3K27AC                  PostTxn          1027
 2 Expression.Splicing            H3K27AC                  Txn              7107
 3 Expression.Splicing            H3K4ME3                  PostTxn          1027
 4 Expression.Splicing            H3K4ME3                  Txn              7107
 5 Expression.Splicing            chRNA.Expression.Splici… PostTxn          1027
 6 Expression.Splicing            chRNA.Expression.Splici… Txn              7107
 7 Expression.Splicing.Subset_YRI H3K27AC                  PostTxn          1274
 8 Expression.Splicing.Subset_YRI H3K27AC                  Txn              2014
 9 Expression.Splicing.Subset_YRI H3K4ME3                  PostTxn          1274
10 Expression.Splicing.Subset_YRI H3K4ME3                  Txn              2014
11 Expression.Splicing.Subset_YRI chRNA.Expression.Splici… PostTxn          1274
12 Expression.Splicing.Subset_YRI chRNA.Expression.Splici… Txn              2014</code></pre>
<pre class="r"><code>set.seed(0)
results &lt;- list()
for (i in 1:100){
  print(paste(&quot;Bootstrap resample&quot;, i))
  transcriptional.eQTLs.sample &lt;- transcriptional.eQTLs %&gt;%
  group_by(PC1, PC2, IsPolyASpecific) %&gt;%
  sample_frac(replace=T)
  results[[i]] &lt;-
  finemappip.annotation.intersections %&gt;%
  filter(Signal == &quot;L1&quot;) %&gt;%
  inner_join(transcriptional.eQTLs.sample, by=c(&quot;eGene&quot;=&quot;P1&quot;)) %&gt;%
  filter(!is.na(IsPolyASpecific)) %&gt;%
  group_by(Annotation, IsPolyASpecific, PC1, PC2) %&gt;%
  summarise(SumInAnnotation = sum(pip, na.rm=T)) %&gt;%
  ungroup() %&gt;%
  group_by(IsPolyASpecific, PC1, PC2) %&gt;%
  mutate(TotalPIP = sum(SumInAnnotation, na.rm=T)) %&gt;%
  ungroup() %&gt;%
  mutate(FractionPIP_in_Annotation = SumInAnnotation/TotalPIP ) %&gt;%
  dplyr::select(FractionPIP_in_Annotation, PC1, PC2, IsPolyASpecific, Annotation) %&gt;%
  pivot_wider(names_from = &quot;IsPolyASpecific&quot;, values_from = c(&quot;FractionPIP_in_Annotation&quot;)) %&gt;%
  mutate(Enrichment = log2(PostTxn/Txn)) %&gt;%
  mutate(i=i)
}</code></pre>
<pre><code>[1] &quot;Bootstrap resample 1&quot;
[1] &quot;Bootstrap resample 2&quot;
[1] &quot;Bootstrap resample 3&quot;
[1] &quot;Bootstrap resample 4&quot;
[1] &quot;Bootstrap resample 5&quot;
[1] &quot;Bootstrap resample 6&quot;
[1] &quot;Bootstrap resample 7&quot;
[1] &quot;Bootstrap resample 8&quot;
[1] &quot;Bootstrap resample 9&quot;
[1] &quot;Bootstrap resample 10&quot;
[1] &quot;Bootstrap resample 11&quot;
[1] &quot;Bootstrap resample 12&quot;
[1] &quot;Bootstrap resample 13&quot;
[1] &quot;Bootstrap resample 14&quot;
[1] &quot;Bootstrap resample 15&quot;
[1] &quot;Bootstrap resample 16&quot;
[1] &quot;Bootstrap resample 17&quot;
[1] &quot;Bootstrap resample 18&quot;
[1] &quot;Bootstrap resample 19&quot;
[1] &quot;Bootstrap resample 20&quot;
[1] &quot;Bootstrap resample 21&quot;
[1] &quot;Bootstrap resample 22&quot;
[1] &quot;Bootstrap resample 23&quot;
[1] &quot;Bootstrap resample 24&quot;
[1] &quot;Bootstrap resample 25&quot;
[1] &quot;Bootstrap resample 26&quot;
[1] &quot;Bootstrap resample 27&quot;
[1] &quot;Bootstrap resample 28&quot;
[1] &quot;Bootstrap resample 29&quot;
[1] &quot;Bootstrap resample 30&quot;
[1] &quot;Bootstrap resample 31&quot;
[1] &quot;Bootstrap resample 32&quot;
[1] &quot;Bootstrap resample 33&quot;
[1] &quot;Bootstrap resample 34&quot;
[1] &quot;Bootstrap resample 35&quot;
[1] &quot;Bootstrap resample 36&quot;
[1] &quot;Bootstrap resample 37&quot;
[1] &quot;Bootstrap resample 38&quot;
[1] &quot;Bootstrap resample 39&quot;
[1] &quot;Bootstrap resample 40&quot;
[1] &quot;Bootstrap resample 41&quot;
[1] &quot;Bootstrap resample 42&quot;
[1] &quot;Bootstrap resample 43&quot;
[1] &quot;Bootstrap resample 44&quot;
[1] &quot;Bootstrap resample 45&quot;
[1] &quot;Bootstrap resample 46&quot;
[1] &quot;Bootstrap resample 47&quot;
[1] &quot;Bootstrap resample 48&quot;
[1] &quot;Bootstrap resample 49&quot;
[1] &quot;Bootstrap resample 50&quot;
[1] &quot;Bootstrap resample 51&quot;
[1] &quot;Bootstrap resample 52&quot;
[1] &quot;Bootstrap resample 53&quot;
[1] &quot;Bootstrap resample 54&quot;
[1] &quot;Bootstrap resample 55&quot;
[1] &quot;Bootstrap resample 56&quot;
[1] &quot;Bootstrap resample 57&quot;
[1] &quot;Bootstrap resample 58&quot;
[1] &quot;Bootstrap resample 59&quot;
[1] &quot;Bootstrap resample 60&quot;
[1] &quot;Bootstrap resample 61&quot;
[1] &quot;Bootstrap resample 62&quot;
[1] &quot;Bootstrap resample 63&quot;
[1] &quot;Bootstrap resample 64&quot;
[1] &quot;Bootstrap resample 65&quot;
[1] &quot;Bootstrap resample 66&quot;
[1] &quot;Bootstrap resample 67&quot;
[1] &quot;Bootstrap resample 68&quot;
[1] &quot;Bootstrap resample 69&quot;
[1] &quot;Bootstrap resample 70&quot;
[1] &quot;Bootstrap resample 71&quot;
[1] &quot;Bootstrap resample 72&quot;
[1] &quot;Bootstrap resample 73&quot;
[1] &quot;Bootstrap resample 74&quot;
[1] &quot;Bootstrap resample 75&quot;
[1] &quot;Bootstrap resample 76&quot;
[1] &quot;Bootstrap resample 77&quot;
[1] &quot;Bootstrap resample 78&quot;
[1] &quot;Bootstrap resample 79&quot;
[1] &quot;Bootstrap resample 80&quot;
[1] &quot;Bootstrap resample 81&quot;
[1] &quot;Bootstrap resample 82&quot;
[1] &quot;Bootstrap resample 83&quot;
[1] &quot;Bootstrap resample 84&quot;
[1] &quot;Bootstrap resample 85&quot;
[1] &quot;Bootstrap resample 86&quot;
[1] &quot;Bootstrap resample 87&quot;
[1] &quot;Bootstrap resample 88&quot;
[1] &quot;Bootstrap resample 89&quot;
[1] &quot;Bootstrap resample 90&quot;
[1] &quot;Bootstrap resample 91&quot;
[1] &quot;Bootstrap resample 92&quot;
[1] &quot;Bootstrap resample 93&quot;
[1] &quot;Bootstrap resample 94&quot;
[1] &quot;Bootstrap resample 95&quot;
[1] &quot;Bootstrap resample 96&quot;
[1] &quot;Bootstrap resample 97&quot;
[1] &quot;Bootstrap resample 98&quot;
[1] &quot;Bootstrap resample 99&quot;
[1] &quot;Bootstrap resample 100&quot;</code></pre>
<pre class="r"><code>bind_rows(results) %&gt;%
  # filter(PC1 == &quot;Expression.Splicing&quot; &amp; PC2 == &quot;H3K27AC&quot; &amp; Annotation == &quot;ChromHMM:Active_Promoter&quot;)
  group_by(Annotation, PC1, PC2) %&gt;%
  summarise(mean = mean(Enrichment, na.rm=T),
            Percentile10 = quantile(Enrichment, probs = 0.05, na.rm=T),
            Percentile90 = quantile(Enrichment, probs = 0.95, na.rm=T)) %&gt;%
  mutate(color = case_when(
    str_detect(Annotation, &quot;^Splic&quot;) ~ &quot;Splice region&quot;,
    str_detect(Annotation, &quot;Enhancer&quot;) ~ &quot;Enhancer&quot;,
    str_detect(Annotation, &quot;Promoter&quot;) ~ &quot;Promoter&quot;,
    str_detect(Annotation, &quot;Txn&quot;) ~ &quot;Transcribed&quot;,
    str_detect(Annotation, &quot;ChromHMM&quot;) ~ &quot;Other ChromHMM&quot;,
    TRUE ~ &quot;Other&quot;
  )) %&gt;%
  ggplot(aes(x=Annotation, y=mean, color=color)) +
    geom_hline(yintercept = 0) +
    geom_point() +
    geom_errorbar(aes(ymin=Percentile10, ymax=Percentile90)) +
    theme_bw() +
    theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
    coord_flip() +
    facet_grid(PC2~PC1) +
    labs(x = &quot;Annotation&quot;, y=&quot;Finemapping posterior mass fold enrichment\n(ChromatinQTL|eQTL &lt;-- --&gt; sQTL)&quot;,
         caption = &quot;Bootsrapped 90% confidence interval&quot;)</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/e8e4a64f33cebe0e6d0e6a1da90995a383f735fc/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-13-1.png" target="_blank">e8e4a64</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-04-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="figs2c" class="section level2">
<h2>FigS2C</h2>
<p>sQTL QQ plot, with different colors for post-transcriptional eQTLs to
transcriptional eQTLs (like what Carlos has done in red and black)</p>
<pre class="r"><code>transcriptional.PC2s &lt;- c(&quot;chRNA.Expression.Splicing&quot;, &quot;H3K27AC&quot;, &quot;H3K4ME3&quot;)

transcriptional.eQTLs &lt;- dat.ZsFromAllTraitPairs %&gt;%
  filter(PC2 %in% transcriptional.PC2s) %&gt;%
  mutate(IsPolyASpecific = q &lt; 0.01 &amp; trait.x.p.in.y&gt;0.01 )

count(transcriptional.eQTLs, IsPolyASpecific, PC2)</code></pre>
<pre><code># A tibble: 12 × 4
# Groups:   PC1, PC2 [6]
   PC1                            PC2                      IsPolyASpecific     n
   &lt;chr&gt;                          &lt;chr&gt;                    &lt;lgl&gt;           &lt;int&gt;
 1 Expression.Splicing            H3K27AC                  FALSE            7294
 2 Expression.Splicing            H3K27AC                  TRUE              980
 3 Expression.Splicing            H3K4ME3                  FALSE            8021
 4 Expression.Splicing            H3K4ME3                  TRUE              899
 5 Expression.Splicing            chRNA.Expression.Splici… FALSE            8943
 6 Expression.Splicing            chRNA.Expression.Splici… TRUE              897
 7 Expression.Splicing.Subset_YRI H3K27AC                  FALSE            1965
 8 Expression.Splicing.Subset_YRI H3K27AC                  TRUE             1405
 9 Expression.Splicing.Subset_YRI H3K4ME3                  FALSE            2217
10 Expression.Splicing.Subset_YRI H3K4ME3                  TRUE             1361
11 Expression.Splicing.Subset_YRI chRNA.Expression.Splici… FALSE            2326
12 Expression.Splicing.Subset_YRI chRNA.Expression.Splici… TRUE             1587</code></pre>
<pre class="r"><code>PC1.filter &lt;- c(&quot;Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;)

PC1.sQTL.filter &lt;- c(&quot;polyA.Splicing&quot;, &quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;)


dat.forsQTLQQ &lt;- fread(&quot;../code/pi1/PairwisePi1Traits.P.all.txt.gz&quot;) %&gt;%
  filter((PC1 %in% PC1.filter) &amp; (PC2 %in% PC1.sQTL.filter))

dat.forsQTLQQ %&gt;%
  inner_join(
    transcriptional.eQTLs %&gt;%
    dplyr::select(ClassificationPC = PC2, IsPolyASpecific, P1, PC1)
  ) %&gt;%
  group_by(PC1, PC2, IsPolyASpecific, ClassificationPC ) %&gt;%
  mutate(ExpectedP = percent_rank(trait.x.p.in.y)) %&gt;%
  ungroup() %&gt;%
  ggplot(aes(x=-log10(ExpectedP), y=-log10(trait.x.p.in.y), color=IsPolyASpecific)) +
  # geom_abline(slope=1, intercept=1)
  geom_point() +
  geom_abline(slope=1) +
  facet_grid(PC2~ClassificationPC, scales = &quot;free&quot;) +
  labs(x=&quot;Expected -log10P&quot;, y=&quot;Observed -log10P&quot;, title=&quot;sQTL QQ for top eQTL SNPs&quot;, caption=&quot;Column facets: criteria to determine polyA-specific eQTLs\nRow facets: sQTL dataset&quot;)</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/e8e4a64f33cebe0e6d0e6a1da90995a383f735fc/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-14-1.png" target="_blank">e8e4a64</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-04-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="fig2c" class="section level2">
<h2>Fig2c</h2>
<p>The eQTL QQplot of stable sQTLs, unstable sQTLs, and H3K27AC QTLs</p>
<pre class="r"><code>PC2.filter &lt;- c(&quot;Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;)
PC2.SignificanceFilter &lt;- c(&quot;H3K27AC&quot;, &quot;H3K4ME3&quot;, &quot;H3K36ME3&quot;)
PC1.filter &lt;- c(&quot;polyA.Splicing&quot;, &quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;)
PC1.filter.Splicing &lt;- PC1.filter[str_detect(PC1.filter, &quot;Splicing&quot;)]
PC1.filter.NonSplicing &lt;- PC1.filter[!str_detect(PC1.filter, &quot;Splicing&quot;)]

dat.foreQTLQQ &lt;- fread(&quot;../code/pi1/PairwisePi1Traits.P.all.txt.gz&quot;) %&gt;%
  filter((PC1 %in% PC1.filter))

IntronAnnotatins &lt;- read_tsv(&quot;../data/IntronAnnotationsFromYang.tsv.gz&quot;) %&gt;%
  mutate(chrom = str_remove_all(chrom, &quot;chr&quot;)) %&gt;%
  mutate(Intron = paste(chrom, start, end, sep=&quot;:&quot;)) %&gt;%
  filter(!str_detect(SuperAnnotation, &quot;NoncodingGene&quot;))

IntronAnnotatins$SuperAnnotation %&gt;% unique()</code></pre>
<pre><code>[1] &quot;AnnotatedJunc_UnproductiveCodingGene&quot;  
[2] &quot;UnannotatedJunc_UnproductiveCodingGene&quot;
[3] &quot;AnnotatedJunc_ProductiveCodingGene&quot;    
[4] &quot;UnannotatedJunc_ProductiveCodingGene&quot;  </code></pre>
<pre class="r"><code>RandomTestSNPs &lt;- paste0(&quot;../code/QTLs/QTLTools/&quot;, PC2.filter, &quot;/NominalPassForColoc.RandomSamplePvals.txt.gz&quot;) %&gt;%
  setNames(PC2.filter) %&gt;%
  lapply(fread) %&gt;%
  bind_rows(.id=&quot;PC2&quot;) %&gt;%
  dplyr::select(trait.x.p.in.y=V1, PC2) %&gt;%
  mutate(SNP_group=&quot;Random test SNPs&quot;) %&gt;%
  group_by(PC2) %&gt;%
  sample_n(50000)


dat.foreQTLQQ.sQTLs &lt;- dat.foreQTLQQ %&gt;%
    filter(PC1 %in% PC1.filter.Splicing) %&gt;%
    group_by(PC1, P1) %&gt;%
    filter(!any((PC2 %in% PC2.SignificanceFilter) &amp; (trait.x.p.in.y &lt; 0.01))) %&gt;%
    ungroup() %&gt;%
    filter(PC2 %in% PC2.filter) %&gt;%
    separate(P1, into=c(&quot;Intron&quot;, &quot;Cluster&quot;), sep=&quot;:clu&quot;, remove=F) %&gt;%
    inner_join(
      IntronAnnotatins %&gt;%
        dplyr::select(Intron, SuperAnnotation),
      by=&quot;Intron&quot;) %&gt;%
    group_by(PC1, PC2, Cluster) %&gt;%
    mutate(SNP_group = case_when(
      all(str_detect(SuperAnnotation, &quot;Productive&quot;)) ~ &quot;Productive sQTL cluster&quot;,
      any(str_detect(SuperAnnotation, &quot;Unproductive&quot;)) ~ &quot;Unproductive sQTL cluster&quot;,
      TRUE ~ &quot;sQTL Other&quot;
    )) %&gt;%
    # sample_n(1) %&gt;%
    slice(which.min(p_permutation.x)) %&gt;%
    ungroup()

bind_rows(
  dat.foreQTLQQ.sQTLs,
  dat.foreQTLQQ %&gt;%
    filter(PC2 %in% PC2.filter) %&gt;%
    filter(PC1 %in% PC1.filter.NonSplicing) %&gt;%
    mutate(SNP_group = PC1),
  RandomTestSNPs
) %&gt;%
  group_by(SNP_group, PC2) %&gt;%
  mutate(MyRank = rank(trait.x.p.in.y, ties.method=&#39;random&#39;)) %&gt;%
  mutate(ExpectedP = MyRank/(max(MyRank) + 1)) %&gt;%
  ungroup() %&gt;%
  ggplot(aes(x=-log10(ExpectedP), y=-log10(trait.x.p.in.y), color=SNP_group)) +
  geom_abline(slope=1, intercept=0) +
  geom_point() +
  facet_wrap(~PC2, scales=&quot;free&quot;) +
  labs(y=&quot;-log10(ObservedP)&quot;, caption=&quot;sQTL clusters containing any significant unproductive sQTL intron are grouped as Unproductive sQTL\nP value from only one significant sQTL intron from cluster selected at random from cluster\nsQTLs with nominal P&lt;0.01 for H3K36me3, H3K27Ac, or H3K4me3 removed&quot;, title=&quot;eQTL QQ plot&quot;)</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/e8e4a64f33cebe0e6d0e6a1da90995a383f735fc/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-15-1.png" target="_blank">e8e4a64</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-04-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dat.foreQTLQQ.sQTLs %&gt;%
  count(SNP_group, PC2)</code></pre>
<pre><code># A tibble: 4 × 3
  SNP_group                 PC2                                n
  &lt;chr&gt;                     &lt;chr&gt;                          &lt;int&gt;
1 Productive sQTL cluster   Expression.Splicing             1472
2 Productive sQTL cluster   Expression.Splicing.Subset_YRI  1472
3 Unproductive sQTL cluster Expression.Splicing             1791
4 Unproductive sQTL cluster Expression.Splicing.Subset_YRI  1791</code></pre>
<pre class="r"><code># pi0est function breaks sometimes. this wrapper function returns 1 when that happens
CalculatePi1_2 &lt;- function (dat.in, ...) {
  return(tryCatch(1-pi0est(dat.in$trait.x.p.in.y, ...)$pi0, error=function(e) 1))
}

labels &lt;- dat.foreQTLQQ.sQTLs %&gt;%
  split(paste(.$SNP_group, .$PC2, sep=&quot;;&quot;)) %&gt;%
  lapply(CalculatePi1_2, pi0.method=&quot;bootstrap&quot;) %&gt;%
  unlist() %&gt;%
  data.frame(pi1=.) %&gt;%
  rownames_to_column(&quot;sQTLGroup_eQTLDataset&quot;) %&gt;%
  separate(sQTLGroup_eQTLDataset, into=c(&quot;SNP_group&quot;, &quot;PC2&quot;), sep=&quot;;&quot;) %&gt;%
  mutate(label=paste0(&quot;pi=&quot;, signif(pi1, digits=3)))

  
dat.foreQTLQQ.sQTLs %&gt;%
  ggplot(aes(x=trait.x.p.in.y)) +
  geom_histogram() +
  geom_text(data = labels, aes(label=label), x=Inf, y=Inf, vjust=1, hjust=1) +
  facet_grid(SNP_group ~ PC2) +
  labs(x=&quot;P&quot;, title=&quot;eQTL P value historgram&quot;, caption=&quot;Vertical facets: eQTL dataset\nHorizontal facets: sQTL cluster type&quot;) +
  theme(strip.text = element_text(size = 8))</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-2">
Past versions of unnamed-chunk-15-2.png
</button>
</p>
<div id="fig-unnamed-chunk-15-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/e8e4a64f33cebe0e6d0e6a1da90995a383f735fc/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-15-2.png" target="_blank">e8e4a64</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-04-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>save a qqplot For lab meeting…</p>
<pre class="r"><code>bind_rows(
  dat.foreQTLQQ.sQTLs,
  dat.foreQTLQQ %&gt;%
    filter(PC2 %in% PC2.filter) %&gt;%
    filter(PC1 %in% PC1.filter.NonSplicing) %&gt;%
    mutate(SNP_group = PC1),
  RandomTestSNPs
) %&gt;%
  group_by(SNP_group, PC2) %&gt;%
  mutate(MyRank = rank(trait.x.p.in.y, ties.method=&#39;random&#39;)) %&gt;%
  mutate(ExpectedP = MyRank/(max(MyRank) + 1)) %&gt;%
  ungroup() %&gt;%
  filter(PC2==&quot;Expression.Splicing&quot;) %&gt;%
  filter(!PC1 %in% c(&quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;)) %&gt;%
  ggplot(aes(x=-log10(ExpectedP), y=-log10(trait.x.p.in.y), color=SNP_group)) +
  geom_abline(slope=1, intercept=0) +
  geom_point() +
  scale_color_manual(values=
                       c(&quot;Random test SNPs&quot;=&quot;#000000&quot;, &quot;Unproductive sQTL cluster&quot;=&quot;#e31a1c&quot;, &quot;Productive sQTL cluster&quot;=&quot;#1f78b4&quot;, &quot;H3K27AC&quot;=&quot;#6a3d9a&quot;),
                     labels=c(&quot;Random test SNPs&quot;=&quot;Random test SNPs&quot;, &quot;Unproductive sQTL cluster&quot;=&quot;Unproductive sQTL&quot;, &quot;Productive sQTL cluster&quot;=&quot;Productive sQTL&quot;, &quot;H3K27AC&quot;=&quot;H3K27AC QTL w/in 100kb&quot;)) +
  labs(y=&quot;-log10(ObservedP)&quot;, caption=&quot;sQTL clusters containing any significant unproductive sQTL intron are grouped as Unproductive sQTL\nP value from only one significant sQTL intron from cluster selected at random from cluster\nsQTLs with nominal P&lt;0.01 for H3K36me3, H3K27Ac, or H3K4me3 removed&quot;, title=&quot;eQTL QQ plot&quot;) +
  theme(plot.subtitle=element_text(size=5))
ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/20230412_WIP_eQTLQQ.png&quot;, height=5, width=7)</code></pre>
<p>For lab meeting I’m also going to make those beta beta scatter
plots</p>
<pre class="r"><code>PC1.PossibleValues &lt;- c(&quot;polyA.Splicing&quot;, &quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;)
PC2.PossibleValues &lt;-c(&quot;Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;, &quot;H3K4ME3&quot;, &quot;H3K36ME3&quot;, &quot;H3K27AC&quot;, &quot;H3K4ME1&quot;, &quot;MetabolicLabelled.30min&quot;, &quot;MetabolicLabelled.60min&quot;, &quot;chRNA.Expression.Splicing&quot;)


dat &lt;- fread(&quot;../code/pi1/PairwisePi1Traits.P.all.txt.gz&quot;) %&gt;%
  filter((PC1 %in% PC1.PossibleValues) &amp; (PC2 %in% PC2.PossibleValues))

Intron.Annotations &lt;- read_tsv(&quot;../data/IntronAnnotationsFromYang.tsv.gz&quot;) %&gt;%
  mutate(IntronName = paste(chrom, start, end, strand, sep=&quot;:&quot;))

dat.sQTLs.eQTLs.ForScatter &lt;- dat %&gt;%
  mutate(IntronName = str_replace(P1, &quot;^(.+?:)clu_.+?([+-])$&quot;, &quot;chr\\1\\2&quot;)) %&gt;%
  mutate(ClusterName =  str_replace(P1, &quot;^(.+?:).+?(clu_.+?[+-])$&quot;, &quot;chr\\1\\2&quot;)) %&gt;%
  left_join(Intron.Annotations)

PC1.filter = c(&quot;polyA.Splicing&quot;)
PC2.filter = c(&quot;H3K36ME3&quot;, &quot;chRNA.Expression.Splicing&quot; , &quot;MetabolicLabelled.30min&quot;, &quot;MetabolicLabelled.60min&quot;, &quot;Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;)
PC2.SignificanceFilter &lt;- c(&quot;H3K4ME3&quot;, &quot;H3K27AC&quot;, &quot;H3K36ME3&quot;)

Intron.Annotations$SuperAnnotation %&gt;% unique()</code></pre>
<pre><code>[1] &quot;AnnotatedJunc_NoncodingGene&quot;           
[2] &quot;UnannotatedJunc_NoncodingGene&quot;         
[3] &quot;AnnotatedJunc_UnproductiveCodingGene&quot;  
[4] &quot;UnannotatedJunc_UnproductiveCodingGene&quot;
[5] &quot;AnnotatedJunc_ProductiveCodingGene&quot;    
[6] &quot;UnannotatedJunc_ProductiveCodingGene&quot;  </code></pre>
<pre class="r"><code>dat.sQTLs.eQTLs.ForScatter.ToPlot &lt;- dat.sQTLs.eQTLs.ForScatter %&gt;%
  filter(PC1 %in% PC1.filter) %&gt;%
  group_by(P1) %&gt;%
  filter(!any((PC2 %in% PC2.SignificanceFilter) &amp; (trait.x.p.in.y &lt; 0.01))) %&gt;%
  ungroup() %&gt;%
  filter(PC2 %in% PC2.filter) %&gt;%
  # pull(PC2) %&gt;% unique()
  filter(SuperAnnotation %in% c(&quot;UnannotatedJunc_UnproductiveCodingGene&quot;, &quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;)) %&gt;%
  group_by(PC1, ClusterName) %&gt;%
  filter(abs(beta.x) == max(abs(beta.x))) %&gt;%
  ungroup() %&gt;%
  mutate(SuperAnnotation = str_replace_all(SuperAnnotation, &quot;_&quot;, &quot; &quot;)) %&gt;%
  filter(GeneLocus == gene) %&gt;%
  filter(FDR.x &lt; 0.1) %&gt;%
  group_by(P2, PC2, SuperAnnotation) %&gt;%
  sample_n(1) %&gt;%
  ungroup() %&gt;%
  mutate(PC2 = factor(PC2, levels=PC2.filter)) %&gt;%
  mutate(PC2 = recode(PC2, !!!c(&quot;chRNA.Expression.Splicing&quot;=&quot;chRNA&quot; , &quot;MetabolicLabelled.30min&quot;=&quot;30min 4sU&quot;, &quot;MetabolicLabelled.60min&quot;=&quot;60min 4sU&quot;, &quot;Expression.Splicing&quot;=&quot;polyA&quot;, &quot;Expression.Splicing.Subset_YRI&quot;=&quot;polyA YRI&quot;))) %&gt;%
  mutate(PC2 = factor(PC2, levels=c(&quot;H3K36ME3&quot;,&quot;chRNA&quot;, &quot;30min 4sU&quot;, &quot;60min 4sU&quot;, &quot;polyA&quot;, &quot;polyA YRI&quot;))) %&gt;%
  filter(!PC2 %in% c(&quot;polyA YRI&quot;, &quot;H3K36ME3&quot;)) %&gt;%
  # pull(SuperAnnotation) %&gt;% unique()
  mutate(SuperAnnotation = recode(SuperAnnotation, !!!c(&quot;AnnotatedJunc UnproductiveCodingGene&quot;=&quot;Unproductive&quot;, &quot;UnannotatedJunc UnproductiveCodingGene&quot;=&quot;Unproductive&quot;, &quot;AnnotatedJunc ProductiveCodingGene&quot;=&quot;Productive&quot;, &quot;UnannotatedJunc ProductiveCodingGene&quot;=&quot;Productive&quot;)))

dat.sQTLs.eQTLs.ForScatter.ToPlot.labels &lt;- dat.sQTLs.eQTLs.ForScatter.ToPlot %&gt;%
  count(SuperAnnotation, PC2) %&gt;%
  distinct(SuperAnnotation, .keep_all=T) %&gt;%
  dplyr::select(n, SuperAnnotation) %&gt;%
  mutate(SuperAnnotationLabel = paste0(SuperAnnotation, &quot; (n=&quot;, n, &quot;)&quot;))

library(broom)

dat.sQTLs.eQTLs.ForScatter.ToPlot %&gt;%
  left_join(dat.sQTLs.eQTLs.ForScatter.ToPlot.labels, by=&quot;SuperAnnotation&quot;) %&gt;%
  nest(-SuperAnnotationLabel, -PC2) %&gt;% 
  mutate(cor=map(data,~cor.test(.x$beta.x, .x$x.beta.in.y, method = &quot;sp&quot;))) %&gt;%
  mutate(tidied = map(cor, tidy)) %&gt;% 
  unnest(tidied, .drop = T) %&gt;%
  unnest(data) %&gt;%
ggplot(aes(x=beta.x, y=x.beta.in.y, color=SuperAnnotation)) +
    geom_point(alpha=0.1) +
    # geom_smooth(method = &quot;tls&quot;, se = FALSE, color = &quot;red&quot;, method=&#39;bootstrap&#39;) +
    # geom_smooth(method=&#39;lm&#39;, color=&#39;black&#39;) +
    geom_vline(xintercept=0, linetype=&#39;dashed&#39;) +
    geom_hline(yintercept=0, linetype=&#39;dashed&#39;) +
    scale_color_manual(values=c(&quot;Productive&quot;=&quot;#1f78b4&quot;, &quot;Unproductive&quot;=&quot;#e31a1c&quot;)) +
    geom_abline(data = . %&gt;%
                  distinct(SuperAnnotationLabel, PC2, .keep_all=T),
                aes(slope=estimate, intercept=0), size=2) +
    geom_text(
    data = . %&gt;%
        distinct(SuperAnnotationLabel, PC2, .keep_all=T) %&gt;%
        mutate(R=signif(estimate, 3), P=format.pval(p.value, 3)) %&gt;%
      mutate(label = str_glue(&quot;rho:{R}\nP:{P}&quot;)),
    aes(x=-Inf, y=-Inf, label=label),
    hjust=-.1, vjust=-0.1, color=&#39;black&#39;, size=6
    ) +
    facet_grid(SuperAnnotationLabel ~ PC2, labeller = label_wrap_gen(10)) +
    theme_bw() +
    theme(strip.text = element_text(size = 12), legend.position=&#39;none&#39;) +
  labs(caption = &quot;FDR&lt;10% sQTLs. sQTLs with nominal P&lt;0.01 for H3K36me3, H3K27Ac, or H3K4me3 removed\nOnly top sQTL per cluster plotted&quot;, y=&quot;host gene eQTL beta&quot;, x=&quot;sQTL beta&quot;)</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/e8e4a64f33cebe0e6d0e6a1da90995a383f735fc/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-17-1.png" target="_blank">e8e4a64</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-04-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/20230412_WIP_eQTLsQTLbetascatter.png&quot;, height=4, width=10)</code></pre>
</div>
<div id="fig3" class="section level2">
<h2>Fig3</h2>
<p>Fig3, exploring connection to complex traits using gwas. There’s a
few ways I did this. Firstly I took all molQTL top SNPs and got the
corresponding gwas P-values, to compare to global gwas P-values and plot
enrichment in QQplot form. I currently don’t have any good control sets
of SNPs. Later I can come up with a control, and also come up with a
systematic way to quantify enrichment of molQTLs for GWAS SNPs (ie
lambda inflation factor, compared to control SNPs, or something like
that) and think of a way to present these results not just for one GWAS
but for many… The second way I am considering this problem is with
hyprcoloc… I attempt to colocalize all molQTLs (chromatinQTLs,
expressionQTLs, sQTLs, APA QTLs, etc) within a 1MB window centered
around top SNP at each GWAS loci.</p>
<p>First let’s try to make the QQ plot.</p>
<pre class="r"><code>gwas.trait.accession &lt;- &quot;IMSGC2019&quot;

molQTL.gwas.P &lt;- fread(paste0(&quot;../code/gwas_summary_stats/MolQTLIntersections/&quot;, gwas.trait.accession, &quot;.bed.gz&quot;), col.names = c(&quot;chrom&quot;, &quot;varPos&quot;, &quot;GWAS.P&quot;, &quot;molQTL.name&quot;, &quot;molQTL.P&quot;, &quot;strand&quot;, &quot;molQTL.beta&quot;, &quot;molQTL.se&quot;, &quot;molQTL.q&quot;, &quot;varID&quot;, &quot;overlap&quot;)) %&gt;%
  separate(molQTL.name, into=c(&quot;PhenotypeClass&quot;, &quot;MolPhenotypeName&quot;), sep = &quot;;&quot;)

gwas.P &lt;- fread(paste0(&quot;../code/gwas_summary_stats/sorted_index_summarystat_hg38beds/&quot;, gwas.trait.accession, &quot;.bed.gz&quot;), select = 4, col.names = &quot;GWAS.P&quot;) %&gt;%
  mutate(PhenotypeClass = &quot;All SNPs&quot;)

molQTL.gwas.P$PhenotypeClass %&gt;% unique()</code></pre>
<pre><code> [1] &quot;APA_Nuclear&quot;                      &quot;H3K27AC&quot;                         
 [3] &quot;MetabolicLabelled.30min.Splicing&quot; &quot;H3K4ME1&quot;                         
 [5] &quot;polyA.Splicing&quot;                   &quot;polyA.Splicing.Subset_YRI&quot;       
 [7] &quot;MetabolicLabelled.60min.Splicing&quot; &quot;chRNA.Splicing&quot;                  
 [9] &quot;chRNA.Splicing.Order&quot;             &quot;CTCF&quot;                            
[11] &quot;APA_Total&quot;                        &quot;chRNA.Expression_ncRNA&quot;          
[13] &quot;DNaseISensitivity&quot;                &quot;H3K36ME3_ncRNA&quot;                  
[15] &quot;chRNA.IER&quot;                        &quot;polyA.IER&quot;                       
[17] &quot;H3K36ME3&quot;                         &quot;Expression.Splicing&quot;             
[19] &quot;MetabolicLabelled.30min&quot;          &quot;polyA.IER.Subset_YRI&quot;            
[21] &quot;ProCap&quot;                           &quot;H3K4ME3&quot;                         
[23] &quot;MetabolicLabelled.60min&quot;          &quot;chRNA.Expression.Splicing&quot;       
[25] &quot;Expression.Splicing.Subset_YRI&quot;   &quot;MetabolicLabelled.60min.IER&quot;     
[27] &quot;chRNA.Slopes&quot;                     &quot;MetabolicLabelled.30min.IER&quot;     </code></pre>
<pre class="r"><code>PhenotypeClass.filter &lt;- c(&quot;Expression.Splicing.Subset_YRI&quot;, &quot;Expression.Splicing&quot;, &quot;polyA.Splicing&quot;, &quot;H3K27AC&quot;)



QQ.gwas &lt;- bind_rows(
  molQTL.gwas.P %&gt;%
    filter(PhenotypeClass %in% PhenotypeClass.filter) %&gt;%
    filter(molQTL.q &lt; 0.1) %&gt;%
    dplyr::select(PhenotypeClass, GWAS.P),
  gwas.P %&gt;%
    sample_n(1E5)
) %&gt;%
dplyr::select(SNP_group = PhenotypeClass, GWAS.P) %&gt;%
group_by(SNP_group) %&gt;%
  mutate(MyRank = rank(GWAS.P, ties.method=&#39;random&#39;)) %&gt;%
  mutate(ExpectedP = MyRank/(max(MyRank) + 1)) %&gt;%
  ungroup() %&gt;%
  mutate(SNP_group = relevel(factor(SNP_group), &quot;All SNPs&quot;)) %&gt;%
  arrange(SNP_group) %&gt;%
  ggplot(aes(x=-log10(ExpectedP), y=-log10(GWAS.P), color=SNP_group)) +
  geom_abline(slope=1, intercept=0) +
  geom_point() +
  labs(y=&quot;-log10(ObservedP)&quot;, title=&quot;GWAS QQ plot&quot;, caption=&quot;GWAS SNPs sub-sampled to 100K for plotting speed&quot;)
QQ.gwas</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>QQ.gwas +
  coord_cartesian(ylim=c(0,20))</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-19-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, it is nice to see some enrichment, though just plotting different
axis limits can really visually change the story. I will need to be
careful here. Let’s distinguish the sQTLs now, between productive and
non-productive…</p>
<pre class="r"><code># sQTL.permutationPass.Classified &lt;- read_delim(&quot;../code/QTLs/QTLTools/polyA.Splicing/PermutationPass.FDR_Added.txt.gz&quot;, delim=&#39; &#39;) %&gt;%
#   # head() %&gt;%
#   separate(phe_id, into=c(&quot;intron&quot;, &quot;cluster&quot;), sep=&quot;:clu&quot;, remove=F) %&gt;%
#   mutate(IntronName = str_replace(phe_id, &quot;^(.+?:.+?:.+?:)clu_.+?_([+-])$&quot;, &quot;chr\\1\\2&quot;)) %&gt;%
#   inner_join(Intron.Annotations, by=&quot;IntronName&quot;) %&gt;%
#   filter(q &lt; 0.1) %&gt;%
#   group_by(cluster) %&gt;%
#   mutate(SNP_group = case_when(
#       all(str_detect(SuperAnnotation, &quot;Productive&quot;)) ~ &quot;Productive sQTL cluster&quot;,
#       any(str_detect(SuperAnnotation, &quot;Unproductive&quot;)) ~ &quot;Unproductive sQTL cluster&quot;,
#       TRUE ~ &quot;sQTL Other&quot;
#     )) %&gt;%
#     slice(which.min(nom_pval)) %&gt;%
#     ungroup()
# 
# sQTL.permutationPass.Classified %&gt;%
#   count(SNP_group)

SplicingPhenotypeFilter &lt;- PhenotypeClass.filter[str_detect(PhenotypeClass.filter, &quot;Splicing&quot;) &amp; !str_detect(PhenotypeClass.filter, &quot;Expression&quot;)]

 

QQ.gwas &lt;- bind_rows(
  molQTL.gwas.P %&gt;%
    filter((PhenotypeClass %in% PhenotypeClass.filter) &amp; !(PhenotypeClass %in% SplicingPhenotypeFilter)) %&gt;%
    filter(molQTL.q &lt; 0.1) %&gt;%
    dplyr::select(PhenotypeClass, GWAS.P),
  molQTL.gwas.P %&gt;%
    filter(PhenotypeClass %in% SplicingPhenotypeFilter) %&gt;%
    filter(molQTL.q &lt; 0.1) %&gt;%
    dplyr::select(GWAS.P, MolPhenotypeName) %&gt;%
    inner_join(
      dat.foreQTLQQ.sQTLs %&gt;%
        dplyr::select(MolPhenotypeName=P1, PhenotypeClass=SNP_group) %&gt;%
        distinct(),
      by=c(&quot;MolPhenotypeName&quot;)) %&gt;%
    filter(!PhenotypeClass==&quot;sQTL Other&quot;),
  gwas.P %&gt;%
    sample_n(1E5)
) %&gt;%
dplyr::select(SNP_group = PhenotypeClass, GWAS.P) %&gt;%
group_by(SNP_group) %&gt;%
  mutate(MyRank = rank(GWAS.P, ties.method=&#39;random&#39;)) %&gt;%
  mutate(ExpectedP = MyRank/(max(MyRank) + 1)) %&gt;%
  ungroup() %&gt;%
  mutate(SNP_group = relevel(factor(SNP_group), &quot;All SNPs&quot;)) %&gt;%
  arrange(SNP_group) %&gt;%
  ggplot(aes(x=-log10(ExpectedP), y=-log10(GWAS.P), color=SNP_group)) +
  geom_abline(slope=1, intercept=0) +
  geom_point() +
  labs(y=&quot;-log10(ObservedP)&quot;, title=&quot;GWAS QQ plot&quot;, caption=&quot;GWAS SNPs sub-sampled to 100K for plotting speed&quot;)
QQ.gwas</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>QQ.gwas +
  coord_cartesian(ylim=c(0,20))</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-20-2.png" width="672" style="display: block; margin: auto;" />
So from this the enrichment is similar whether looking at productive or
unproductive sQTLs. Now let me go back and add some control SNPs. One
easy control to include would be like what Yang did in his 2016 paper:
Include molQTL test SNPs within peaks (or within introns) may work for
sQTLs. Also, no need to include the YRI subset anymore. I was just
including it in my exploratory analysis but it is
unnecessary/distracting for a main figure. Let’s also take that out…</p>
<pre class="r"><code>ControlSNPs &lt;- fread(paste0(&quot;../code/gwas_summary_stats/MolQTLIntersections_ControlSNPs/&quot;, gwas.trait.accession, &quot;/ALL.txt.gz&quot;), col.names=c(&quot;GWAS.P&quot;, &quot;PhenotypeClassControl&quot;))

ControlSNPs$PhenotypeClassControl %&gt;% unique()</code></pre>
<pre><code> [1] &quot;Expression.Splicing&quot;              &quot;Expression.Splicing.Subset_YRI&quot;  
 [3] &quot;chRNA.Expression.Splicing&quot;        &quot;MetabolicLabelled.30min&quot;         
 [5] &quot;MetabolicLabelled.60min&quot;          &quot;CTCF&quot;                            
 [7] &quot;H3K27AC&quot;                          &quot;H3K4ME3&quot;                         
 [9] &quot;H3K4ME1&quot;                          &quot;H3K36ME3&quot;                        
[11] &quot;H3K36ME3_ncRNA&quot;                   &quot;ProCap&quot;                          
[13] &quot;polyA.Splicing&quot;                   &quot;polyA.Splicing.Subset_YRI&quot;       
[15] &quot;chRNA.Splicing&quot;                   &quot;MetabolicLabelled.30min.Splicing&quot;
[17] &quot;MetabolicLabelled.60min.Splicing&quot; &quot;chRNA.Expression_ncRNA&quot;          
[19] &quot;APA_Nuclear&quot;                      &quot;APA_Total&quot;                       
[21] &quot;polyA.IER&quot;                        &quot;polyA.IER.Subset_YRI&quot;            
[23] &quot;chRNA.IER&quot;                        &quot;MetabolicLabelled.30min.IER&quot;     
[25] &quot;MetabolicLabelled.60min.IER&quot;      &quot;chRNA.Slopes&quot;                    
[27] &quot;chRNA.Splicing.Order&quot;             &quot;DNaseISensitivity&quot;               </code></pre>
<pre class="r"><code>QQ.gwas &lt;- bind_rows(
  molQTL.gwas.P %&gt;%
    filter((PhenotypeClass %in% PhenotypeClass.filter) &amp; !(PhenotypeClass %in% SplicingPhenotypeFilter)) %&gt;%
    filter(!PhenotypeClass == &quot;Expression.Splicing.Subset_YRI&quot;) %&gt;%
    filter(molQTL.q &lt; 0.1) %&gt;%
    dplyr::select(PhenotypeClass, GWAS.P),
  molQTL.gwas.P %&gt;%
    filter(PhenotypeClass %in% SplicingPhenotypeFilter) %&gt;%
    filter(molQTL.q &lt; 0.1) %&gt;%
    dplyr::select(GWAS.P, MolPhenotypeName) %&gt;%
    inner_join(
      dat.foreQTLQQ.sQTLs %&gt;%
        dplyr::select(MolPhenotypeName=P1, PhenotypeClass=SNP_group) %&gt;%
        distinct(),
      by=c(&quot;MolPhenotypeName&quot;)) %&gt;%
    filter(!PhenotypeClass==&quot;sQTL Other&quot;),
  ControlSNPs %&gt;%
    filter(PhenotypeClassControl %in% c(&quot;polyA.Splicing&quot;, &quot;H3K27AC&quot;, &quot;Expression.Splicing&quot;)) %&gt;%
    mutate(PhenotypeClass = recode(PhenotypeClassControl, &quot;polyA.Splicing&quot;=&quot;Intronic SNPS&quot;, &quot;H3K27AC&quot;=&quot;H3K27AC peak SNPs&quot;, &quot;Expression.Splicing&quot;=&quot;genic SNPs&quot;)) %&gt;%
    filter(!PhenotypeClass==&quot;Intronic SNPS&quot;) %&gt;%
    dplyr::select(-PhenotypeClassControl),
  gwas.P %&gt;%
    sample_n(1E5)
) %&gt;%
dplyr::select(SNP_group = PhenotypeClass, GWAS.P) %&gt;%
  # pull(SNP_group) %&gt;% unique()
group_by(SNP_group) %&gt;%
  mutate(MyRank = rank(GWAS.P, ties.method=&#39;random&#39;)) %&gt;%
  mutate(ExpectedP = MyRank/(max(MyRank) + 1)) %&gt;%
  ungroup() %&gt;%
  mutate(SNP_group = relevel(factor(SNP_group), &quot;All SNPs&quot;)) %&gt;%
  arrange(SNP_group) %&gt;%
  ggplot(aes(x=-log10(ExpectedP), y=-log10(GWAS.P), color=SNP_group)) +
  geom_abline(slope=1, intercept=0) +
  geom_point() +
  # scale_color_brewer(palette = &quot;Dark2&quot;) +
  scale_color_manual(values=
                       c(&quot;genic SNPs&quot;=&quot;#969696&quot;, &quot;All SNPs&quot;=&quot;#000000&quot;, &quot;Unproductive sQTL cluster&quot;=&quot;#e31a1c&quot;, &quot;Productive sQTL cluster&quot;=&quot;#1f78b4&quot;, &quot;H3K27AC&quot;=&quot;#6a3d9a&quot;, &quot;H3K27AC peak SNPs&quot;=&quot;#cab2d6&quot;, &quot;Expression.Splicing&quot;=&quot;#ff7f00&quot;),
                     labels=c(&quot;genic SNPs&quot;=&quot;genic SNPs&quot;, &quot;All SNPs&quot;=&quot;All SNPs&quot;, &quot;Unproductive sQTL cluster&quot;=&quot;Unproductive sQTL&quot;, &quot;Productive sQTL cluster&quot;=&quot;Productive sQTL&quot;, &quot;H3K27AC&quot;=&quot;H3K27AC QTL&quot;, &quot;H3K27AC peak SNPs&quot;=&quot;H3K27AC peak SNPs&quot;, &quot;Expression.Splicing&quot;=&quot;eQTL&quot;)) +
  labs(y=&quot;-log10(ObservedP)&quot;, title=&quot;MS GWAS QQ plot&quot;, caption=&quot;GWAS SNPs sub-sampled to 100K for plotting speed&quot;, fill=&quot;SNP category&quot;)
QQ.gwas</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>QQ.gwas +
  coord_cartesian(ylim=c(0,20))</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-21-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/20230412_WIP_GWAS_QQ.png&quot;)</code></pre>
<p>Hmm, ok there seems to be subtle shift compared to the control SNPs.
These may not be the best controls. Will have to think about this more
later.</p>
<p>Now Let’s expore the hypercoloc results.</p>
<p>Note that I did the hyprcoloc a number of different ways, each with
different sets of molQTLs attempted… Let’s explore the most inclusive
hyprcoloc iteration.</p>
<pre class="r"><code>gwas.traits &lt;- read_tsv(&quot;../code/config/gwas_table.tsv&quot;) %&gt;%
  dplyr::rename(GWAS.accession=gwas, gwas.trait=trait)


hyprcoloc.results &lt;- read_tsv(&quot;../code/hyprcoloc/Results/ForGWASColoc/GWASColoc_ChromatinAPAAndRNA/results.txt.gz&quot;) %&gt;%
  dplyr::rename(GWAS.Loci = GWASLeadSnpChrom_Pos_RefAllele_AltAllele_rsID_trait) %&gt;%
  separate(GWAS.Loci, into=c(&quot;GWAS.LeadSNP.Chrom&quot;, &quot;GWAS.LeadSNP.Pos&quot;, &quot;GWAS.accession&quot;), sep=&quot;_&quot;, remove=F) %&gt;%
  separate_rows(ColocalizedTraits, sep = &quot;,&quot;) %&gt;%
  mutate(IsColocalizedWithSomething = !ColocalizedTraits == &quot;None&quot;) %&gt;%
  mutate(Trait = if_else(IsColocalizedWithSomething, ColocalizedTraits, DroppedTrait)) %&gt;%
  dplyr::select(-DroppedTrait, -ColocalizedTraits) %&gt;%
  mutate(Trait = str_replace_all(Trait, &quot; &quot;, &quot;&quot;)) %&gt;%
  mutate(GWAS.Loci = str_replace_all(GWAS.Loci, &quot; &quot;, &quot;&quot;)) %&gt;%
  mutate(Trait = if_else(Trait == GWAS.Loci, paste(&quot;GWAS&quot;,GWAS.Loci,sep = &quot;;&quot;),Trait)) %&gt;%
  separate(Trait, into=c(&quot;PhenotypeClass&quot;, &quot;Phenotype&quot;), sep=&quot;;&quot;, remove=F) %&gt;%
  group_by(GWAS.Loci, HyprcolocIteration) %&gt;%
  mutate(ColocalizedClusterContainsGWASTrait = any(PhenotypeClass==&quot;GWAS&quot;) &amp; IsColocalizedWithSomething) %&gt;%
  ungroup() %&gt;%
  inner_join(gwas.traits %&gt;%
               dplyr::select(1:2))

hyprcoloc.results$PhenotypeClass %&gt;% unique()</code></pre>
<pre><code> [1] &quot;Expression.Splicing&quot;       &quot;H3K4ME1&quot;                  
 [3] &quot;H3K4ME3&quot;                   &quot;polyA.Splicing&quot;           
 [5] &quot;chRNA.Splicing&quot;            &quot;H3K27AC&quot;                  
 [7] &quot;GWAS&quot;                      &quot;H3K36ME3&quot;                 
 [9] &quot;chRNA.Expression.Splicing&quot; &quot;APA_Nuclear&quot;              
[11] &quot;APA_Total&quot;                 NA                         
[13] &quot;chr19_47135282_IMSGC2019&quot;  &quot;chr1_92687078_IMSGC2019&quot;  
[15] &quot;chr1_85264137_IMSGC2019&quot;   &quot;chr1_91756532_IMSGC2019&quot;  
[17] &quot;chr1_100824876_IMSGC2019&quot;  &quot;chr2_43128185_IMSGC2019&quot;  
[19] &quot;chr1_157716547_IMSGC2019&quot;  &quot;chr12_57713053_IMSGC2019&quot; 
[21] &quot;chr11_118872577_IMSGC2019&quot; &quot;chr12_123119506_IMSGC2019&quot;
[23] &quot;chr14_75547955_IMSGC2019&quot;  &quot;chr14_88057144_IMSGC2019&quot; 
[25] &quot;chr6_135512325_IMSGC2019&quot;  &quot;chr5_40396323_IMSGC2019&quot;  
[27] &quot;chr16_85987899_IMSGC2019&quot;  &quot;chr7_50288743_IMSGC2019&quot;  
[29] &quot;chr11_61026179_IMSGC2019&quot;  &quot;chr19_49333989_IMSGC2019&quot; 
[31] &quot;chr3_102030178_IMSGC2019&quot;  &quot;chr1_192571891_IMSGC2019&quot; 
[33] &quot;chr2_61015275_IMSGC2019&quot;   &quot;chr2_230257114_IMSGC2019&quot; 
[35] &quot;chr14_102799507_IMSGC2019&quot; &quot;chr20_54173204_IMSGC2019&quot; 
[37] &quot;chr3_28030595_IMSGC2019&quot;   &quot;chr22_50532837_IMSGC2019&quot; 
[39] &quot;chr6_27099878_IMSGC2019&quot;   &quot;chr6_159044945_IMSGC2019&quot; 
[41] &quot;chr6_137116920_IMSGC2019&quot;  &quot;chr6_137638318_IMSGC2019&quot; 
[43] &quot;chr5_159332892_IMSGC2019&quot;  &quot;chr15_90344352_IMSGC2019&quot; 
[45] &quot;chr6_27893892_IMSGC2019&quot;   &quot;chr16_79077400_IMSGC2019&quot; 
[47] &quot;chr8_127801845_IMSGC2019&quot;  &quot;chr7_149592373_IMSGC2019&quot; 
[49] &quot;chr18_58680812_IMSGC2019&quot; </code></pre>
<pre class="r"><code>hyprcoloc.results %&gt;%
  filter(!GWAS.accession==&quot;IMSGC2019&quot;) %&gt;%
  pull(PhenotypeClass) %&gt;% unique()</code></pre>
<pre><code> [1] &quot;Expression.Splicing&quot;       &quot;H3K4ME1&quot;                  
 [3] &quot;H3K4ME3&quot;                   &quot;polyA.Splicing&quot;           
 [5] &quot;chRNA.Splicing&quot;            &quot;H3K27AC&quot;                  
 [7] &quot;GWAS&quot;                      &quot;H3K36ME3&quot;                 
 [9] &quot;chRNA.Expression.Splicing&quot; &quot;APA_Nuclear&quot;              
[11] &quot;APA_Total&quot;                 NA                         </code></pre>
<pre class="r"><code>PhenotypeRecodes = c(&quot;H3K36ME3&quot;=&quot;hQTL&quot;, &quot;H3K27AC&quot;=&quot;hQTL&quot;, &quot;H3K4ME3&quot;=&quot;hQTL&quot;, &quot;H3K4ME1&quot;=&quot;hQTL&quot;,
                     &quot;Expression.Splicing&quot;=&quot;eQTL&quot;, &quot;chRNA.Expression.Splicing&quot;=&quot;chRNA eQTL&quot;,
                     &quot;APA_Nuclear&quot;=&quot;APA QTL&quot;, &quot;APA_Total&quot;=&quot;APA QTL&quot;, &quot;polyA.Splicing&quot;=&quot;sQTL&quot;, &quot;GWAS&quot;=&quot;GWAS&quot;)

PhenotypeRecodes.df &lt;- data.frame(PhenotypeRecodes) %&gt;%
  rownames_to_column(&quot;PhenotypeClass&quot;)

# hyprcoloc.results %&gt;%
#   filter(GWAS.Loci == &quot;chr9_79683075_GCST004627&quot;)
#   filter(is.na(PhenotypeClass))
  
  

hyprcoloc.results.toplot &lt;- hyprcoloc.results %&gt;%
  filter(!GWAS.accession==&quot;IMSGC2019&quot;) %&gt;%
  left_join(PhenotypeRecodes.df) %&gt;%
  mutate(PhenotypeRecodes = if_else(is.na(PhenotypeRecodes), PhenotypeClass, PhenotypeRecodes)) %&gt;%
  filter(!PhenotypeRecodes == &quot;APA QTL&quot;) %&gt;%
  group_by(GWAS.Loci, HyprcolocIteration) %&gt;%
  filter(any(ColocalizedClusterContainsGWASTrait) | PhenotypeClass==&quot;GWAS&quot;) %&gt;%
  mutate(Category = case_when(
    all(ColocalizedClusterContainsGWASTrait==FALSE) | all(is.na(HyprcolocIteration))  ~ &quot;No molQTL colocs&quot;,
    all(PhenotypeRecodes %in% c(&quot;GWAS&quot;, &quot;hQTL&quot;)) ~ &quot;Only hQTL colocs&quot;,
    all(PhenotypeRecodes %in% c(&quot;GWAS&quot;, &quot;eQTL&quot;)) ~ &quot;Only eQTL colocs&quot;,
    all(PhenotypeRecodes %in% c(&quot;GWAS&quot;, &quot;eQTL&quot;, &quot;hQTL&quot;, &quot;chRNA eQTL&quot;)) ~ &quot;hQTL+eQTL colocs&quot;,
    all(PhenotypeRecodes %in% c(&quot;GWAS&quot;, &quot;sQTL&quot;)) ~ &quot;sQTL colocs&quot;,
    all(PhenotypeRecodes %in% c(&quot;GWAS&quot;, &quot;sQTL&quot;, &quot;chRNA eQTL&quot;, &quot;eQTL&quot;)) ~ &quot;sQTL+eQTL colocs&quot;,
    # all(PhenotypeRecodes %in% c(&quot;GWAS&quot;, &quot;sQTL&quot;, &quot;chRNA eQTL&quot;, &quot;eQTL&quot;, &quot;hQTL&quot;)) ~ &quot;sQTL+eQTL+hQTL colocs&quot;,
    TRUE ~ &quot;Other&quot;
  )) %&gt;%
  ungroup()
  # filter(Category==&quot;Other&quot;)

Num.Gwas.loci &lt;- hyprcoloc.results.toplot %&gt;%
  distinct(GWAS.Loci, .keep_all=T) %&gt;%
  count(GWAS.accession, gwas.trait)

hyprcoloc.results.toplot %&gt;%
  distinct(GWAS.Loci, .keep_all=T) %&gt;%
  count(Category,GWAS.accession, gwas.trait) %&gt;%
  ggplot(aes(x=gwas.trait)) +
  geom_col(position=&quot;fill&quot;, aes( y=n, fill=Category)) +
  geom_text(data=Num.Gwas.loci, aes(x=gwas.trait, label=n), y=Inf, hjust=1.1, size=4, angle=90) +
  scale_fill_brewer(palette = &quot;Dark2&quot;) +
  Rotate_x_labels +
  theme(axis.text.x = element_text(size=8)) +
  labs(fill=&quot;Exclusive\nCategory&quot;, caption=&quot;Categories are exclusive; sQTL means no hQTL or eQTL\nsQTL + eQTL means no hQTLs\nhQTL + sQTL + eQTL would be Other\nTotal number GWAS loci marked at top&quot;, y=&quot;Fraction of colocalizations&quot;) +
  scale_y_continuous(expand=c(0,0))</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-22-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/20230412_WIP_Colocs.png&quot;, height=6, width=12)</code></pre>
<pre class="r"><code>test.results &lt;- hyprcoloc.results.toplot %&gt;%
  filter(Category %in% c(&quot;sQTL colocs&quot;, &quot;sQTL+eQTL colocs&quot;)) %&gt;%
  filter(PhenotypeClass == &quot;polyA.Splicing&quot;) %&gt;%
  distinct(GWAS.Loci, .keep_all=T) %&gt;%
  inner_join(dat.foreQTLQQ.sQTLs %&gt;%
               filter(PC1==&quot;polyA.Splicing&quot;), by=c(&quot;Phenotype&quot;=&quot;P1&quot;)) %&gt;%
  count(SNP_group, Category) %&gt;%
  pivot_wider(names_from=&quot;Category&quot;, values_from =&quot;n&quot;) %&gt;%
  column_to_rownames(&quot;SNP_group&quot;) %&gt;%
  as.matrix() %&gt;%
  fisher.test(alternative=&quot;greater&quot;)

hyprcoloc.results.toplot %&gt;%
  filter(Category %in% c(&quot;sQTL colocs&quot;, &quot;sQTL+eQTL colocs&quot;)) %&gt;%
  filter(PhenotypeClass == &quot;polyA.Splicing&quot;) %&gt;%
  distinct(GWAS.Loci, .keep_all=T) %&gt;%
  inner_join(dat.foreQTLQQ.sQTLs %&gt;%
               filter(PC1==&quot;polyA.Splicing&quot;), by=c(&quot;Phenotype&quot;=&quot;P1&quot;)) %&gt;%
  count(SNP_group, Category) %&gt;%
  ggplot(aes(x=Category, y=n, fill=SNP_group)) +
  geom_col() +
  scale_y_continuous(expand=c(0,0)) +
  scale_fill_manual(values=c(&quot;Productive sQTL cluster&quot;=&quot;#377eb8&quot;, &quot;Unproductive sQTL cluster&quot;=&quot;#e41a1c&quot;), labels=c(&quot;Productive sQTL cluster&quot;=&quot;Productive sQTL&quot;, &quot;Unproductive sQTL cluster&quot;=&quot;Unproductive sQTL&quot;)) +
  labs(caption=paste(&quot;One sided Fisher test P:&quot;, format.pval(test.results$p.value, 3), &quot;\nOR:&quot;, round(test.results$estimate, 3)), y=&quot;Number of\nGWAS/sQTL colocalizations&quot;, fill=&quot;sQTL type&quot;, x=&quot;&quot;) +
  Rotate_x_labels +
  theme(legend.position=&#39;bottom&#39;, legend.box=&quot;vertical&quot;, legend.margin=margin()) +
  guides(fill=guide_legend(nrow=2,byrow=TRUE))</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/20230412_WIP_Colocs_ProductiveOrUnproductive.png&quot;, height=6, width=3)</code></pre>
<p>Next I want to find some nice examples to highlight. I think I’ll
filter for eQTL/sQTL/GWAS colocs, plot the systematically, then pick one
that looks nice and also highlight it as a coloc plot.</p>
<div id="small-molecule-stuff" class="section level3">
<h3>Small molecule stuff</h3>
<pre class="r"><code>small.molecule.dat.genes &lt;- read_tsv(&quot;../code/SmallMolecule/FitModels/polyA_genes.tsv.gz&quot;)

small.molecule.dat.introns &lt;- read_tsv(&quot;../code/SmallMolecule/FitModels/polyA_GAGTIntrons.tsv.gz&quot;) %&gt;%
  mutate(Intron = str_replace(junc, &quot;^chr(.+?):[+-]$&quot;, &quot;\\1&quot;)) %&gt;%
  inner_join(IntronAnnotatins)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] broom_0.8.0        viridis_0.6.2      viridisLite_0.4.0  qvalue_2.28.0     
 [5] edgeR_3.38.4       limma_3.52.4       data.table_1.14.2  RColorBrewer_1.1-3
 [9] forcats_0.5.1      stringr_1.4.0      dplyr_1.0.9        purrr_0.3.4       
[13] readr_2.1.2        tidyr_1.2.0        tibble_3.1.7       ggplot2_3.3.6     
[17] tidyverse_1.3.1   

loaded via a namespace (and not attached):
 [1] fs_1.5.2          lubridate_1.8.0   bit64_4.0.5       httr_1.4.3       
 [5] rprojroot_2.0.3   tools_4.2.0       backports_1.4.1   bslib_0.3.1      
 [9] utf8_1.2.2        R6_2.5.1          DBI_1.1.2         colorspace_2.0-3 
[13] withr_2.5.0       tidyselect_1.1.2  gridExtra_2.3     bit_4.0.4        
[17] compiler_4.2.0    git2r_0.30.1      cli_3.3.0         rvest_1.0.2      
[21] xml2_1.3.3        labeling_0.4.2    sass_0.4.1        scales_1.2.0     
[25] digest_0.6.29     rmarkdown_2.14    R.utils_2.11.0    pkgconfig_2.0.3  
[29] htmltools_0.5.2   dbplyr_2.1.1      fastmap_1.1.0     highr_0.9        
[33] rlang_1.0.2       readxl_1.4.0      rstudioapi_0.13   jquerylib_0.1.4  
[37] farver_2.1.0      generics_0.1.2    jsonlite_1.8.0    vroom_1.5.7      
[41] R.oo_1.24.0       magrittr_2.0.3    Rcpp_1.0.8.3      munsell_0.5.0    
[45] fansi_1.0.3       lifecycle_1.0.1   R.methodsS3_1.8.1 stringi_1.7.6    
[49] whisker_0.4       yaml_2.3.5        plyr_1.8.7        grid_4.2.0       
[53] parallel_4.2.0    promises_1.2.0.1  crayon_1.5.1      lattice_0.20-45  
[57] haven_2.5.0       splines_4.2.0     hms_1.1.1         locfit_1.5-9.7   
[61] knitr_1.39        pillar_1.7.0      reshape2_1.4.4    reprex_2.0.1     
[65] glue_1.6.2        evaluate_0.15     modelr_0.1.8      vctrs_0.4.1      
[69] tzdb_0.3.0        httpuv_1.6.5      cellranger_1.1.0  gtable_0.3.0     
[73] assertthat_0.2.1  xfun_0.30         later_1.3.0       workflowr_1.7.0  
[77] ellipsis_0.3.2   </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
