<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2023-03-24" />

<title>20230324_MakeFigs</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">20230324_MakeFigs</h1>
<h4 class="date">2023-03-24</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-03-30
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>ChromatinSplicingQTLs/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstreeddbf2225e7f71ace4a67a2cdae69ece7e4026b66targetblankddbf222a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/ddbf2225e7f71ace4a67a2cdae69ece7e4026b66" target="_blank">ddbf222</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstreeddbf2225e7f71ace4a67a2cdae69ece7e4026b66targetblankddbf222a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/ddbf2225e7f71ace4a67a2cdae69ece7e4026b66" target="_blank">ddbf222</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store
    Ignored:    code/.RData
    Ignored:    code/._report.html
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    code/.snakemake/
    Ignored:    code/APA_Processing/
    Ignored:    code/Alignments/
    Ignored:    code/ChromHMM/
    Ignored:    code/ENCODE/
    Ignored:    code/ExpressionAnalysis/
    Ignored:    code/FastqFastp/
    Ignored:    code/FastqFastpSE/
    Ignored:    code/FastqSE/
    Ignored:    code/FineMapping/
    Ignored:    code/Genotypes/
    Ignored:    code/H3K36me3_CutAndTag.pdf
    Ignored:    code/IntronSlopes/
    Ignored:    code/LR.bed
    Ignored:    code/LR.seq.bed
    Ignored:    code/LongReads/
    Ignored:    code/Metaplots/
    Ignored:    code/Misc/
    Ignored:    code/MiscCountTables/
    Ignored:    code/Multiqc/
    Ignored:    code/Multiqc_chRNA/
    Ignored:    code/NonCodingRNA/
    Ignored:    code/NonCodingRNA_annotation/
    Ignored:    code/PairwisePi1Traits.P.all.txt.gz
    Ignored:    code/PeakCalling/
    Ignored:    code/Phenotypes/
    Ignored:    code/PlotGruberQTLs/
    Ignored:    code/PlotQTLs/
    Ignored:    code/ProCapAnalysis/
    Ignored:    code/QC/
    Ignored:    code/QTL_SNP_Enrichment/
    Ignored:    code/QTLs/
    Ignored:    code/RPKM_tables/
    Ignored:    code/ReadLengthMapExperiment/
    Ignored:    code/ReadLengthMapExperimentResults/
    Ignored:    code/ReadLengthMapExperimentSpliceCounts/
    Ignored:    code/ReferenceGenome/
    Ignored:    code/Rplots.pdf
    Ignored:    code/Session.vim
    Ignored:    code/SmallMolecule/
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TODO
    Ignored:    code/Tehranchi/
    Ignored:    code/bigwigs/
    Ignored:    code/bigwigs_FromNonWASPFilteredReads/
    Ignored:    code/config/.DS_Store
    Ignored:    code/config/._.DS_Store
    Ignored:    code/config/.ipynb_checkpoints/
    Ignored:    code/config/config.local.yaml
    Ignored:    code/dag.pdf
    Ignored:    code/dag.png
    Ignored:    code/dag.svg
    Ignored:    code/debug.ipynb
    Ignored:    code/debug_python.ipynb
    Ignored:    code/deepTools/
    Ignored:    code/featureCounts/
    Ignored:    code/featureCountsBasicGtf/
    Ignored:    code/gwas_summary_stats/
    Ignored:    code/hyprcoloc/
    Ignored:    code/igv_session.xml
    Ignored:    code/isoseqbams/
    Ignored:    code/log
    Ignored:    code/logs/
    Ignored:    code/notebooks/.ipynb_checkpoints/
    Ignored:    code/pi1/
    Ignored:    code/rules/.ipynb_checkpoints/
    Ignored:    code/rules/OldRules/
    Ignored:    code/rules/notebooks/
    Ignored:    code/scratch/
    Ignored:    code/scripts/.ipynb_checkpoints/
    Ignored:    code/scripts/GTFtools_0.8.0/
    Ignored:    code/scripts/__pycache__/
    Ignored:    code/scripts/liftOverBedpe/liftOverBedpe.py
    Ignored:    code/snakemake.dryrun.log
    Ignored:    code/snakemake.log
    Ignored:    code/snakemake.sbatch.log
    Ignored:    code/snakemake_profiles/slurm/__pycache__/
    Ignored:    code/test.introns.bed
    Ignored:    code/test.introns2.bed
    Ignored:    code/test.log
    Ignored:    code/tracks.xml
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/._20220414203249_JASPAR2022_combined_matrices_25818_jaspar.txt
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-10.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-11.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-2.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-3.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-4.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-5.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-6.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-7.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-8.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022.csv
    Ignored:    data/Metaplots/.DS_Store

Untracked files:
    Untracked:  analysis/test_rmd_format.Rmd

Unstaged changes:
    Modified:   analysis/20230324_MakeFigs.Rmd
    Modified:   code/Snakefile
    Modified:   code/rules/QTL_SNP_Enrichment.smk
    Modified:   code/rules/common.py
    Modified:   code/scripts/CalculatePi1_GetAscertainmentP_AllPairs.py
    Modified:   code/scripts/CalculatePi1_GetTraitPairs_AllTraits.R
    Modified:   code/scripts/GenometracksByGenotype

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/20230324_MakeFigs.Rmd</code>) and
HTML (<code>docs/20230324_MakeFigs.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ddbf2225e7f71ace4a67a2cdae69ece7e4026b66/analysis/20230324_MakeFigs.Rmd" target="_blank">ddbf222</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-27
</td>
<td>
add pi1 heatmaps nb
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/ddbf2225e7f71ace4a67a2cdae69ece7e4026b66/docs/20230324_MakeFigs.html" target="_blank">ddbf222</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-27
</td>
<td>
add pi1 heatmaps nb
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE, warning = F, message = F)

library(tidyverse)</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>✔ ggplot2 3.3.6     ✔ purrr   0.3.4
✔ tibble  3.1.7     ✔ dplyr   1.0.9
✔ tidyr   1.2.0     ✔ stringr 1.4.0
✔ readr   2.1.2     ✔ forcats 0.5.1</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(RColorBrewer)
library(data.table)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre class="r"><code>library(edgeR)</code></pre>
<pre><code>Loading required package: limma</code></pre>
<pre class="r"><code># Set theme
theme_set(
  theme_classic() +
  theme(text=element_text(size=16,  family=&quot;Helvetica&quot;)))

# I use layer a lot, to rotate long x-axis labels
Rotate_x_labels &lt;- theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))

#test plot
ggplot(mtcars, aes(x=mpg, y=cyl)) +
  geom_point()</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/setup-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-setup-1">
Past versions of setup-1.png
</button>
</p>
<div id="fig-setup-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ddbf2225e7f71ace4a67a2cdae69ece7e4026b66/docs/figure/20230324_MakeFigs.Rmd/setup-1.png" target="_blank">ddbf222</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="intro" class="section level2">
<h2>Intro</h2>
<p>I notebook to make figure panels for publication…</p>
<p>Figures to make:</p>
<ol style="list-style-type: decimal">
<li><p>Placeholder, Nascent RNA reveals abundance of NMD-destined
transcripts</p></li>
<li><p>sQTLs, eQTLs</p></li>
</ol>
<p><input type="checkbox" unchecked></input>a. pi heatmap across
promoter hQTLs, H3K36me3, chRNA, 4sU, polyA. Number of QTLs in
margin</p>
<p><input type="checkbox" unchecked></input>b. Enrichment of annotations
in transcriptional vs post-transcriptional eQTLs. Classify
transcriptional vs post-transcriptional by comparing hQTL to eQTL
significance (or effect size).</p>
<p><input type="checkbox" unchecked></input>c. QQ-plot of eQTL signal,
grouped by hQTLs, sQTLs that affect productive junctions, and sQTLs that
affect unproductive junctions</p>
<p><input type="checkbox" unchecked></input>d. polyA sQTL beta vs polyA
eQTL beta scatter, faceted by stable vs unstable.</p>
<p><input type="checkbox" unchecked></input>e. Beta vs beta spearman
slope with different lines for chRNA, 4sU, polyA, faceted by stable vs
unstable</p>
<ol start="3" style="list-style-type: decimal">
<li>Contribution of splicing-mediated decay to complex trait</li>
</ol>
<p><input type="checkbox" unchecked></input>a. A single GWAS trait (ie
MS) QQ plot, colored by splicing-mediated eQTLs, transcription-mediated
QTLs, productive-AS sQTLs</p>
<p><input type="checkbox" unchecked></input>b. Heatmap of similar
enrichment in A, but across many traits</p>
<p><input type="checkbox" unchecked></input>c. Example locus showing
colocalization eQTL/sQTL/GWAS scatter, and IGV screenshot of the
splicing and eQTL effect</p>
<ol start="4" style="list-style-type: decimal">
<li>The effect of splice-modifying drugs on gene expression</li>
</ol>
<p><input type="checkbox" unchecked></input>a. Cartoon showing inset
with risdiplam bulge repair, along with dose:response trend showing the
tested doses and the increasing response in GA|GT but not AG|GT
splicing.</p>
<p><input type="checkbox" unchecked></input>b. Scatter that shows
splicing vs expression effect, faeted by productive unproductive. Or
something like that</p>
<p><input type="checkbox" unchecked></input>c. Something that shows the
4B effect is not in chRNA</p>
<p><input type="checkbox" unchecked></input>d. An illustrative example
effect in IGV.</p>
<p>Supplemental Figures to make</p>
<ol style="list-style-type: decimal">
<li><p>Placeholder</p></li>
<li><p>sQTLs, eQTLs</p></li>
</ol>
<p><input type="checkbox" unchecked></input>a. Total number of QTLs in
each type of type of trait. Bargraph, with multiple lines for different
FDR levels</p>
<p><input type="checkbox" unchecked></input>b. Correlation of effect
sizes across promoter hQTLs, H3K36me3, chRNA, 4sU, polyA</p>
<p><input type="checkbox" unchecked></input>i. Need to select set of
eQTLs to plot… Previous versions of this plot I tend to just pick
colocalized things. That would be easiest.</p>
<p><input type="checkbox" unchecked></input>c. sQTL QQ plot, with
different colors for post-transcriptional eQTLs to transcriptional eQTLs
(like what Carlos has done in red and black)</p>
<p><input type="checkbox" unchecked></input>d. eQTL QQ plot, with
different colors for H3K27AC QTLs and sQTLs</p>
<p><input type="checkbox" unchecked></input>e. lead gene-wise sQTLs are
not strongly enriched for eQTLs compared to control SNPs. eQTL QQ-plot
with top gene-wise sQTL SNP, compared to some controls. Or do this with
a pi1, and can use various thresholds</p>
<p><input type="checkbox" unchecked></input>f. enrichment of eQTL
signals is explained by both primary and non-primary eQTL signals
(FigX). More precisely, we used the SuSiE finemapping approach which
does not assume a single causal SNP for each expression trait, and can
thus be used to finemap multiple eQTL signals per gene. We find that
approximately x% of primary signals could be overlapped with sQTLs,
compared to y% of primary signals that we could overlap to hQTLs.
Furthermore, x% and y% of non primary signals could be mapped sQTLs and
hQTLs, respectively</p>
<p><input type="checkbox" unchecked></input>g. More extensive version of
2d, so we can explain the nuances in filtering out transcriptional
effects, and there are some lingering non-NMD effects that are degraded
in cytoplasm</p>
<ol start="3" style="list-style-type: decimal">
<li>Contribution of splicing-mediated decay to complex trait</li>
</ol>
<p><input type="checkbox" unchecked></input>a. Colocalization framework
cartoon</p>
<p><input type="checkbox" unchecked></input>b. Version of 3b that uses
colocalization. Shown as barplots similar to Phoenix’s paper</p>
<ol start="4" style="list-style-type: decimal">
<li>The effect of splice-modifying drugs on gene expression</li>
</ol>
<p><input type="checkbox" unchecked></input>a. In total, we observe x%
of observed non-canonincal GA-GT splice sites are activated with x
criteria… ecdf of spearman correlation coefficient, for different groups
of 5’ss.</p>
</div>
<div id="figure2" class="section level2">
<h2>Figure2</h2>
<div id="a" class="section level3">
<h3>A</h3>
<p>pi heatmap of gene-feature QTLs. Num QTLs in margin.</p>
<p>Only include hQTLs at promoter, chRNA, 4sU, and polyA.</p>
<pre class="r"><code>library(qvalue)
library(viridis)

PeaksToTSS &lt;- Sys.glob(&quot;../code/Misc/PeaksClosestToTSS/*_assigned.tsv.gz&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/Misc/PeaksClosestToTSS/(.+?)_assigned.tsv.gz&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(read_tsv) %&gt;%
  bind_rows(.id=&quot;ChromatinMark&quot;) %&gt;%
  mutate(GenePeakPair = paste(gene, peak, sep = &quot;;&quot;)) %&gt;%
  distinct(ChromatinMark, peak, gene, .keep_all=T)

PC1.filter &lt;- c(&quot;Expression.Splicing&quot;, &quot;MetabolicLabelled.30min&quot;, &quot;MetabolicLabelled.60min&quot;, &quot;H3K36ME3&quot;, &quot;H3K27AC&quot;, &quot;H3K4ME3&quot;, &quot;H3K4ME1&quot;, &quot;Expression.Splicing.Subset_YRI&quot;, &quot;chRNA.Expression.Splicing&quot;)

dat &lt;- fread(&quot;../code/pi1/PairwisePi1Traits.P.all.txt.gz&quot;) %&gt;%
  filter((PC1 %in% PC1.filter) &amp; (PC2 %in% PC1.filter))

dat$PC1 %&gt;% unique()</code></pre>
<pre><code>[1] &quot;Expression.Splicing&quot;            &quot;Expression.Splicing.Subset_YRI&quot;
[3] &quot;chRNA.Expression.Splicing&quot;      &quot;MetabolicLabelled.30min&quot;       
[5] &quot;MetabolicLabelled.60min&quot;        &quot;H3K27AC&quot;                       
[7] &quot;H3K4ME3&quot;                        &quot;H3K4ME1&quot;                       
[9] &quot;H3K36ME3&quot;                      </code></pre>
<pre class="r"><code>PhenotypeAliases &lt;- read_tsv(&quot;../data/Phenotypes_recode_for_Plotting.txt&quot;)</code></pre>
<p>Many genes have multiple promoters that should all be accounted for
as potential transcriptional effects. So calculating pi1 can be tricky.
I want to ask the question, “What fraction of eQTLs have a promoter hQTL
in at least one of the promoters?”… I have already solved this problem…
For each gene, get lowest P value for across tests (ie all the
promoters) and use that a a test statistic to come up with a new Pvalue,
to calculate pi1. The test statistic distribution under the null can be
approximate by drawing from uniform n times.</p>
<p>First we need to approximate that null using <code>runif</code>…</p>
<pre class="r"><code>MaxSampleSizeToCreateANull &lt;- 250
NSamplesToEstimateDistribution &lt;- 10000
NullSimulatedTestStats &lt;- matrix(nrow=MaxSampleSizeToCreateANull, ncol=NSamplesToEstimateDistribution)
rownames(NullSimulatedTestStats) &lt;- paste0(&quot;runif_samplesize&quot;, 1:MaxSampleSizeToCreateANull)
colnames(NullSimulatedTestStats) &lt;- paste0(&quot;Sample_&quot;, 1:NSamplesToEstimateDistribution)


for (i in 1:MaxSampleSizeToCreateANull){
  SampleSizeFromUniform &lt;- i
  SampledDat &lt;- matrix(runif(SampleSizeFromUniform*NSamplesToEstimateDistribution), nrow=NSamplesToEstimateDistribution)
  SampleDatNullTestStatistics &lt;- -log10(apply(SampledDat, 1, min))
  NullSimulatedTestStats[i,] &lt;- SampleDatNullTestStatistics
}


NullSimulatedTestStats %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;runif_samplesize&quot;) %&gt;%
  slice(1:250) %&gt;%
  mutate(runif_samplesize = as.numeric(str_remove(runif_samplesize, &quot;runif_samplesize&quot;))) %&gt;%
  gather(key=&quot;Sample&quot;, value=&quot;value&quot;, -runif_samplesize) %&gt;%
  ggplot(aes(x=value, color=runif_samplesize, group=runif_samplesize)) +
  geom_density() +
  scale_color_viridis_c() +
  theme_bw() +
  labs(x=&quot;Test statistic under null\n-log10(MinP) across n draws from uniform&quot;, color=&quot;n&quot;)</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ddbf2225e7f71ace4a67a2cdae69ece7e4026b66/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-2-1.png" target="_blank">ddbf222</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ecdf.functions &lt;- apply(NullSimulatedTestStats, 1, ecdf)
ecdf.functions[[1]](1)</code></pre>
<pre><code>[1] 0.9034</code></pre>
<p>Make a function to calculate pi1</p>
<pre class="r"><code># pi0est function breaks sometimes. this wrapper function returns 1 when that happens
CalculatePi1 &lt;- function (dat.in, ...) {
  return(tryCatch(1-pi0est(dat.in$Pvals.For.Pi1, ...)$pi0, error=function(e) 1))
}</code></pre>
<p>And now, recode peaks as promoters, and calculate test
statistics.</p>
<pre class="r"><code>Recode.PCs = c(&quot;ActivatingMark_HostGenePromoter&quot;=&quot;Promoter marks*&quot;, &quot;H3K36ME3&quot;=&quot;H3K36ME3&quot;, &quot;chRNA.Expression.Splicing&quot;=&quot;chRNA&quot;, &quot;MetabolicLabelled.30min&quot;=&quot;4sU 30min&quot;, &quot;MetabolicLabelled.60min&quot;=&quot;4sU 60min&quot;, &quot;Expression.Splicing.Subset_YRI&quot;=&quot;polyA YRI&quot;, &quot;Expression.Splicing&quot;=&quot;polyA&quot;)

ActivatingMarksToConsider &lt;- c(&quot;H3K27AC&quot;)

ProcessPi1HeatmapDat &lt;- function(FDR.cutoff=0.1){
  dat.split &lt;- dat %&gt;%
  filter(FDR.x &lt; FDR.cutoff) %&gt;%
  mutate(PC1 = case_when(
    paste(GeneLocus, P1, sep=&quot;;&quot;) %in% PeaksToTSS$GenePeakPair &amp; PC1 %in% ActivatingMarksToConsider ~ &quot;ActivatingMark_HostGenePromoter&quot;,
    P1 %in% PeaksToTSS$peak ~ &quot;ActivatingMark_OtherGenePromoter&quot;,
    PC1 %in% c(&quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;H3K27AC&quot;) ~ &quot;ActivatingMark_Enhancer&quot;,
    TRUE ~ PC1
  )) %&gt;%
  mutate(PC2 = case_when(
    paste(GeneLocus, P2, sep=&quot;;&quot;) %in% PeaksToTSS$GenePeakPair &amp; PC2 %in% ActivatingMarksToConsider ~ &quot;ActivatingMark_HostGenePromoter&quot;,
    P2 %in% PeaksToTSS$peak ~ &quot;ActivatingMark_OtherGenePromoter&quot;,
    PC2 %in% c(&quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;H3K27AC&quot;) ~ &quot;ActivatingMark_Enhancer&quot;,
    TRUE ~ PC2
  )) %&gt;%
  group_by(PC1, P1, PC2) %&gt;%
  mutate(test.stat.obs = -log10(min(trait.x.p.in.y))) %&gt;%
  ungroup() %&gt;%
  add_count(PC1, P1, PC2) %&gt;%
  filter(n&lt;=250) %&gt;%
  group_by(PC1, PC2) %&gt;%
  rowwise() %&gt;%
  mutate(Pvals.For.Pi1 = 1-ecdf.functions[[n]](test.stat.obs)) %&gt;%
  ungroup() %&gt;%
  select(PC1, PC2, Pvals.For.Pi1) %&gt;%
  filter(!PC1==PC2) %&gt;%
  split(paste(.$PC1, .$PC2, sep = &quot;;&quot;))
dat.out &lt;- lapply(dat.split, CalculatePi1, pi0.method=&quot;bootstrap&quot;) %&gt;%
  unlist() %&gt;%
  data.frame(pi1=.) %&gt;%
  rownames_to_column(&quot;PC1_PC2&quot;) %&gt;%
  separate(PC1_PC2, into=c(&quot;PC1&quot;, &quot;PC2&quot;), sep=&#39;;&#39;) %&gt;%
  filter((PC1%in% c(PC1.filter, &quot;ActivatingMark_HostGenePromoter&quot;) &amp; (PC2%in% c(PC1.filter, &quot;ActivatingMark_HostGenePromoter&quot;) ))) %&gt;%
  mutate(TextColor = case_when(
    pi1 &gt;= 0.7 ~ &quot;white&quot;,
    TRUE ~ &quot;black&quot;
  )) %&gt;%
mutate(PC1 = recode(PC1, !!!Recode.PCs)) %&gt;%
mutate(PC2 = recode(PC2, !!!Recode.PCs)) %&gt;%
mutate(PC1 = factor(PC1, levels=Recode.PCs)) %&gt;%
mutate(PC2 = factor(PC2, levels=Recode.PCs))
P &lt;- ggplot(dat.out, aes(x=PC1, y=PC2, fill=pi1, color=TextColor)) +
  geom_raster() +
  geom_text(aes(label=signif(pi1*100, 2))) +
  scale_color_identity() +
  scale_fill_viridis_c(option=&quot;A&quot;, direction = -1, limits=c(0,1)) +
  coord_flip() +
  scale_x_discrete(limits=rev) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(x=&quot;Discovery QTL phenotype&quot;, y=&quot;Phenotype assessed for overlap&quot;, caption=paste(&quot;*Promoter marks include&quot;, paste(ActivatingMarksToConsider, collapse=&quot;,&quot;)))
return(list(dat=dat, P=P))
}

ProcessPi1HeatmapDat()$P</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ddbf2225e7f71ace4a67a2cdae69ece7e4026b66/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-4-1.png" target="_blank">ddbf222</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Redo, but also consider both H3K27AC and H3K4ME3 at promoter…</p>
<pre class="r"><code>ActivatingMarksToConsider &lt;- c(&quot;H3K27AC&quot;, &quot;H3K4ME3&quot;)

ProcessPi1HeatmapDat()$P</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ddbf2225e7f71ace4a67a2cdae69ece7e4026b66/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-5-1.png" target="_blank">ddbf222</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Redo but consider H3K4ME3, H3K4ME1, and H3K27AC</p>
<pre class="r"><code>ActivatingMarksToConsider &lt;- c(&quot;H3K27AC&quot;, &quot;H3K4ME3&quot;, &quot;H3K4ME1&quot;)

ProcessPi1HeatmapDat()$P</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ddbf2225e7f71ace4a67a2cdae69ece7e4026b66/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-6-1.png" target="_blank">ddbf222</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>In theory things should only go up as you include H3K4ME1, but that
is not the case. This phenomena is something I’ve noticed before, and I
think it’s because adding non-overlapping phenotypes to promoter marks
only modestly increases the lowest P-value, but more greatly can
increase the n number of features used to determine the null.</p>
<p>So, let’s now include H3K4ME1. This is more thought of as an enhancer
rather than promoter mark anyway.</p>
<p>So now let’s redo, but only include H3K27AC and H3K4ME3 but use lower
FDR cutoffs.</p>
<pre class="r"><code>ActivatingMarksToConsider &lt;- c(&quot;H3K27AC&quot;, &quot;H3K4ME3&quot;)

ProcessPi1HeatmapDat(FDR.cutoff = 0.05)$P</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ddbf2225e7f71ace4a67a2cdae69ece7e4026b66/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-7-1.png" target="_blank">ddbf222</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ProcessPi1HeatmapDat(FDR.cutoff = 0.01)$P</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-2">
Past versions of unnamed-chunk-7-2.png
</button>
</p>
<div id="fig-unnamed-chunk-7-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ddbf2225e7f71ace4a67a2cdae69ece7e4026b66/docs/figure/20230324_MakeFigs.Rmd/unnamed-chunk-7-2.png" target="_blank">ddbf222</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok, cool. I mean slightly different versions. I’ll leave it to Yang
as to which version should be the main figure, and perhaps we can
include alternate versions as supplements. In every case though I notice
that the top right corner is dark, suggesting histone effects can be
picked up pretty well in downstream assays (ie polyA), but the bottom
left corner is light, because of post-transcriptional mechanisms of gene
regulation.</p>
<ul class="task-list">
<li><input type="checkbox" disabled="" />save final prettier versions,
with axes flipped (yang likes discovery on x-axis)</li>
<li><input type="checkbox" disabled="" />save alternate versions for
supplement, with different FDR thresholds</li>
</ul>
</div>
<div id="b" class="section level3">
<h3>B</h3>
<p>Now I need a way to classify eQTLs as transcriptional or
post-transcriptional. Carlos has previously used effect sizes in polyA
vs chRNA. I may do something similar. First let’s some scatter plots of
effect sizes</p>
<pre class="r"><code>FDR.cutoff &lt;- 0.1
ActivatingMarksToConsider &lt;- c(&quot;H3K27AC&quot;, &quot;H3K4ME3&quot;)

dat.recoded &lt;- dat %&gt;%
  filter(FDR.x &lt; FDR.cutoff) %&gt;%
  mutate(PC1 = case_when(
    paste(GeneLocus, P1, sep=&quot;;&quot;) %in% PeaksToTSS$GenePeakPair &amp; PC1 %in% ActivatingMarksToConsider ~ PC1,
    P1 %in% PeaksToTSS$peak ~ &quot;ActivatingMark_OtherGenePromoter&quot;,
    PC1 %in% c(&quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;H3K27AC&quot;) ~ &quot;ActivatingMark_Enhancer&quot;,
    TRUE ~ PC1
  )) %&gt;%
  mutate(PC2 = case_when(
    paste(GeneLocus, P2, sep=&quot;;&quot;) %in% PeaksToTSS$GenePeakPair &amp; PC2 %in% ActivatingMarksToConsider ~ PC2,
    P1 %in% PeaksToTSS$peak ~ &quot;ActivatingMark_OtherGenePromoter&quot;,
    PC2 %in% c(&quot;H3K4ME1&quot;, &quot;H3K4ME3&quot;, &quot;H3K27AC&quot;) ~ &quot;ActivatingMark_Enhancer&quot;,
    TRUE ~ PC2
  ))

dat.recoded %&gt;%
  filter(PC1 == &quot;Expression.Splicing&quot;) %&gt;%
  filter(!str_detect(PC2, &quot;ActivatingMark_Enhancer&quot;)) %&gt;%
  filter(!str_detect(PC2, &quot;ActivatingMark_OtherGenePromoter&quot;)) %&gt;%
  group_by(PC1, P1, PC2) %&gt;%
  slice(which.min(trait.x.p.in.y)) %&gt;%
  ungroup() %&gt;%
  ggplot(aes(x=beta.x, y=x.beta.in.y)) +
  geom_point(alpha=0.1) +
  facet_wrap(~PC2) +
  labs(x=&quot;eQTL beta&quot;, y=&quot;facetQTL beta&quot;)</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>I think what I want to do is to first consider eQTLs in polyA, then
perform one-sided test for significant differences in beta for
corresponding feature in chRNA/promoterMarks. For simplicity, I think I
can use Z-test, like <a
href="https://stats.stackexchange.com/questions/55501/test-a-significant-difference-between-two-slope-values">this</a>,
but first, to check that this approach is reasonable, I think I’m going
to check that Z-scores from random trait pairs closely follow normal
distribution as they should under null. Also, I note that winners curse
will bias things since I am conditioning on significant QTLs for polyA,
then assessing difference in betas between polyA vs chRNA/promoter. But
since only the polyA beta is subject to winners curse bias, it will turn
out being overly convervative for the SNP enrichments I’m trying to
highlight (because some transcriptional QTLs might erronesouly be
classified as non-trancriptional) so i think that’s acceptable.</p>
<pre class="r"><code>PC1.filter = &quot;Expression.Splicing&quot;

dat.ZsFromAllTraitPairs &lt;- dat.recoded %&gt;%
  filter(PC1 == PC1.filter) %&gt;%
  filter(!str_detect(PC2, &quot;ActivatingMark_Enhancer&quot;)) %&gt;%
  filter(!str_detect(PC2, &quot;ActivatingMark_OtherGenePromoter&quot;)) %&gt;%
  group_by(PC1, P1, PC2) %&gt;%
  slice(which.min(trait.x.p.in.y)) %&gt;%
  ungroup() %&gt;%
  filter(!is.na(x.beta.in.y)) %&gt;%
  distinct(PC1, P1, PC2, .keep_all=T) %&gt;%
  mutate(Z = (abs(beta.x)-abs(x.beta.in.y)*sign(beta.x)*sign(x.beta.in.y))/sqrt(x.beta_se.in.y**2 + beta_se.x**2)) %&gt;%
  mutate(OneSidedZTest.P = pnorm(Z, lower.tail=F)) %&gt;%
  group_by(PC1, PC2) %&gt;%
  mutate(q = qvalue(OneSidedZTest.P)$qvalues) %&gt;%
  mutate(PolyASpecific = cut(q, breaks=c(0, 0.01, 0.05, 0.1, 1), include.lowest=T, ordered_result = T))


dat.ZsFromAllTraitPairs %&gt;%
ggplot(aes(x=beta.x, y=x.beta.in.y, color=PolyASpecific)) +
  geom_point(alpha=0.1) +
  scale_color_manual(values = rev(brewer.pal(4, &quot;Reds&quot;))) +
  facet_wrap(~PC2) +
  guides(colour = guide_legend(override.aes = list(alpha = 1))) +
  labs(color=&quot;PolyA-specific FDR&quot;)</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dat.ZsFromAllTraitPairs %&gt;%
ggplot(aes(x=beta.x, y=x.beta.in.y, color=PolyASpecific==&quot;[0,0.01]&quot;)) +
  geom_point(alpha=0.1) +
  facet_wrap(~PC2) +
  guides(colour = guide_legend(override.aes = list(alpha = 1))) +
  labs(color=&quot;PolyA-specific FDR &lt; 0.01&quot;)</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dat.ZsFromAllTraitPairs %&gt;%
  count(PC1, PC2, PolyASpecific) %&gt;%
  ggplot(aes(x=PC2, y=n, fill=fct_rev(PolyASpecific))) +
  geom_col(position=&quot;stack&quot;) +
  scale_fill_manual(values = brewer.pal(4, &quot;Reds&quot;)) +
  Rotate_x_labels</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-9-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dat.ZsFromAllTraitPairs %&gt;%
  count(PC1, PC2, PolyASpecific) %&gt;%
  ggplot(aes(x=PC2, y=n, fill=fct_rev(PolyASpecific))) +
  geom_col(position=&quot;fill&quot;) +
  scale_fill_manual(values = brewer.pal(4, &quot;Reds&quot;)) +
  Rotate_x_labels</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-9-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>Okay, keep in mind that there is both biology and power issues at
play… Biologically, I would expect very few polyA-specific eQTLs when
comparing to 4sU, yet there are basically the same amount of
polyA-specific eQTLs in that comparison as when comparing to chRNA. One
could chalk that up to the 4sU dataset being so underpowered, that a lot
more eQTLs would be considered polyA specific. Similarly, that could
explain why so many are polyA-specific when comparing to the relatively
underpowered H3K36me3 dataset.</p>
<p>Let’s do the same but anchor on YRI eQTLs..</p>
<pre class="r"><code>PC1.filter = &quot;Expression.Splicing.Subset_YRI&quot;

dat.ZsFromAllTraitPairs &lt;- dat.recoded %&gt;%
  filter(PC1 == PC1.filter) %&gt;%
  filter(!str_detect(PC2, &quot;ActivatingMark_Enhancer&quot;)) %&gt;%
  filter(!str_detect(PC2, &quot;ActivatingMark_OtherGenePromoter&quot;)) %&gt;%
  group_by(PC1, P1, PC2) %&gt;%
  slice(which.min(trait.x.p.in.y)) %&gt;%
  ungroup() %&gt;%
  filter(!is.na(x.beta.in.y)) %&gt;%
  distinct(PC1, P1, PC2, .keep_all=T) %&gt;%
  mutate(Z = (abs(beta.x)-abs(x.beta.in.y)*sign(beta.x)*sign(x.beta.in.y))/sqrt(x.beta_se.in.y**2 + beta_se.x**2)) %&gt;%
  mutate(OneSidedZTest.P = pnorm(Z, lower.tail=F)) %&gt;%
  group_by(PC1, PC2) %&gt;%
  mutate(q = qvalue(OneSidedZTest.P)$qvalues)

dat.ZsFromAllTraitPairs %&gt;%
  mutate(PolyASpecific = cut(q, breaks=c(0, 0.01, 0.05, 0.1, 1), include.lowest=T, ordered_result = T)) %&gt;%
ggplot(aes(x=beta.x, y=x.beta.in.y, color=PolyASpecific)) +
  geom_point(alpha=0.1) +
  scale_color_manual(values = rev(brewer.pal(4, &quot;Reds&quot;))) +
  facet_wrap(~PC2) +
  guides(colour = guide_legend(override.aes = list(alpha = 1))) +
  labs(color=&quot;PolyA-specific FDR&quot;)</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dat.ZsFromAllTraitPairs %&gt;%
  mutate(PolyASpecific = cut(q, breaks=c(0, 0.01, 0.05, 0.1, 1), include.lowest=T, ordered_result = T)) %&gt;%
  # mutate(PolyASpecific = cut(OneSidedZTest.P, breaks=c(0, 0.01, 0.05, 0.1, 1), include.lowest=T, ordered_result = T)) %&gt;%
  count(PC1, PC2, PolyASpecific) %&gt;%
  ggplot(aes(x=PC2, y=n, fill=fct_rev(PolyASpecific))) +
  geom_col(position=&quot;stack&quot;) +
  scale_fill_manual(values = brewer.pal(4, &quot;Reds&quot;)) +
  Rotate_x_labels</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dat.ZsFromAllTraitPairs %&gt;%
  mutate(PolyASpecific = cut(q, breaks=c(0, 0.01, 0.05, 0.1, 1), include.lowest=T, ordered_result = T)) %&gt;%
  # mutate(PolyASpecific = cut(OneSidedZTest.P, breaks=c(0, 0.01, 0.05, 0.1, 1), include.lowest=T, ordered_result = T)) %&gt;%
  count(PC1, PC2, PolyASpecific) %&gt;%
  ggplot(aes(x=PC2, y=n, fill=fct_rev(PolyASpecific))) +
  geom_col(position=&quot;fill&quot;) +
  scale_fill_manual(values = brewer.pal(4, &quot;Reds&quot;)) +
  Rotate_x_labels</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-10-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Let’s explore Carlos’ susie finemapping output</p>
<pre class="r"><code>susie.finemap &lt;- fread(&quot;/project2/yangili1/cfbuenabadn/ChromatinSplicingQTLs/code/FineMapping/susie_runs_Geuvadis/susie_output.tab&quot;)
 
susie.finemap %&gt;%
  distinct(cs_gene, cs_names) %&gt;%
  count(cs_gene) %&gt;%
  ggplot(aes(x=n)) +
  geom_bar() +
  labs(y=&quot;Num genes&quot;, x=&quot;Num eQTL signals&quot;)</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>susie.finemap %&gt;%
  group_by(cs_gene, cs_names) %&gt;%
  summarise(PIP.sum = sum(cs_pip)) %&gt;%
  ungroup() %&gt;%
  ggplot(aes(x=PIP.sum)) +
  stat_ecdf() +
  labs(x=&quot;Sum PIP in CI&quot;, title=&quot;These are 95% CIs&quot;) +
  coord_cartesian(xlim=c(0.9,1))</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>susie.finemap %&gt;%
  count(cs_gene, cs_names) %&gt;%
  ggplot(aes(x=n)) +
  stat_ecdf() +
  coord_cartesian(xlim=c(0,20)) +
  labs(x=&quot;Num SNPs in CI&quot;)</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-11-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok great, next I will intersect the finemapped SNPs with annotations,
and sum up the number of SNPs in each annotation (weighted by PIP), and
compare transcriptional vs post-transcriptional eQTLs. I think I’ll do
some work in the snakemake to accomplish this and come back to it…</p>
</div>
</div>
<div id="figs2c" class="section level2">
<h2>FigS2C</h2>
<p>sQTL QQ plot, with different colors for post-transcriptional eQTLs to
transcriptional eQTLs (like what Carlos has done in red and black)</p>
<pre class="r"><code>transcriptional.PC2s &lt;- c(&quot;chRNA.Expression.Splicing&quot;, &quot;H3K27AC&quot;, &quot;H3K4ME3&quot;)

transcriptional.eQTLs &lt;- dat.ZsFromAllTraitPairs %&gt;%
  filter(PC2 %in% transcriptional.PC2s) %&gt;%
  mutate(IsPolyASpecific = q &lt; 0.01 &amp; trait.x.p.in.y&gt;0.01 )

count(transcriptional.eQTLs, IsPolyASpecific, PC2)</code></pre>
<pre><code># A tibble: 6 × 4
# Groups:   PC1, PC2 [3]
  PC1                            PC2                       IsPolyASpecific     n
  &lt;chr&gt;                          &lt;chr&gt;                     &lt;lgl&gt;           &lt;int&gt;
1 Expression.Splicing.Subset_YRI H3K27AC                   FALSE            1965
2 Expression.Splicing.Subset_YRI H3K27AC                   TRUE             1405
3 Expression.Splicing.Subset_YRI H3K4ME3                   FALSE            2217
4 Expression.Splicing.Subset_YRI H3K4ME3                   TRUE             1361
5 Expression.Splicing.Subset_YRI chRNA.Expression.Splicing FALSE            2326
6 Expression.Splicing.Subset_YRI chRNA.Expression.Splicing TRUE             1587</code></pre>
<pre class="r"><code>PC1.filter &lt;- c(&quot;Expression.Splicing&quot;, &quot;Expression.Splicing.Subset_YRI&quot;)

PC1.sQTL.filter &lt;- c(&quot;polyA.Splicing&quot;, &quot;polyA.Splicing.Subset_YRI&quot;, &quot;chRNA.Splicing&quot;)


dat.forsQTLQQ &lt;- fread(&quot;../code/pi1/PairwisePi1Traits.P.all.txt.gz&quot;) %&gt;%
  filter((PC1 %in% PC1.filter) &amp; (PC2 %in% PC1.sQTL.filter))

dat.forsQTLQQ %&gt;%
  inner_join(
    transcriptional.eQTLs %&gt;%
    dplyr::select(ClassificationPC = PC2, IsPolyASpecific, P1, PC1)
  ) %&gt;%
  group_by(PC1, PC2, IsPolyASpecific, ClassificationPC ) %&gt;%
  mutate(ExpectedP = percent_rank(trait.x.p.in.y)) %&gt;%
  ungroup() %&gt;%
  ggplot(aes(x=-log10(ExpectedP), y=-log10(trait.x.p.in.y), color=IsPolyASpecific)) +
  # geom_abline(slope=1, intercept=1)
  geom_point() +
  geom_abline(slope=1) +
  facet_grid(PC2~ClassificationPC, scales = &quot;free&quot;) +
  labs(x=&quot;Expected -log10P&quot;, y=&quot;Observed -log10P&quot;, title=&quot;sQTL QQ for top eQTL SNPs&quot;, caption=&quot;Column facets: criteria to determine polyA-specific eQTLs\nRow facets: sQTL dataset&quot;)</code></pre>
<p><img src="figure/20230324_MakeFigs.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="fig2c" class="section level2">
<h2>Fig2c</h2>
<p>The eQTL QQplot of transcriptional</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] viridis_0.6.2      viridisLite_0.4.0  qvalue_2.28.0      edgeR_3.38.4      
 [5] limma_3.52.4       data.table_1.14.2  RColorBrewer_1.1-3 forcats_0.5.1     
 [9] stringr_1.4.0      dplyr_1.0.9        purrr_0.3.4        readr_2.1.2       
[13] tidyr_1.2.0        tibble_3.1.7       ggplot2_3.3.6      tidyverse_1.3.1   

loaded via a namespace (and not attached):
 [1] fs_1.5.2          lubridate_1.8.0   bit64_4.0.5       httr_1.4.3       
 [5] rprojroot_2.0.3   tools_4.2.0       backports_1.4.1   bslib_0.3.1      
 [9] utf8_1.2.2        R6_2.5.1          DBI_1.1.2         colorspace_2.0-3 
[13] withr_2.5.0       tidyselect_1.1.2  gridExtra_2.3     bit_4.0.4        
[17] compiler_4.2.0    git2r_0.30.1      cli_3.3.0         rvest_1.0.2      
[21] xml2_1.3.3        labeling_0.4.2    sass_0.4.1        scales_1.2.0     
[25] digest_0.6.29     rmarkdown_2.14    R.utils_2.11.0    pkgconfig_2.0.3  
[29] htmltools_0.5.2   dbplyr_2.1.1      fastmap_1.1.0     highr_0.9        
[33] rlang_1.0.2       readxl_1.4.0      rstudioapi_0.13   jquerylib_0.1.4  
[37] farver_2.1.0      generics_0.1.2    jsonlite_1.8.0    vroom_1.5.7      
[41] R.oo_1.24.0       magrittr_2.0.3    Rcpp_1.0.8.3      munsell_0.5.0    
[45] fansi_1.0.3       lifecycle_1.0.1   R.methodsS3_1.8.1 stringi_1.7.6    
[49] whisker_0.4       yaml_2.3.5        plyr_1.8.7        grid_4.2.0       
[53] parallel_4.2.0    promises_1.2.0.1  crayon_1.5.1      lattice_0.20-45  
[57] haven_2.5.0       splines_4.2.0     hms_1.1.1         locfit_1.5-9.7   
[61] knitr_1.39        pillar_1.7.0      reshape2_1.4.4    reprex_2.0.1     
[65] glue_1.6.2        evaluate_0.15     modelr_0.1.8      vctrs_0.4.1      
[69] tzdb_0.3.0        httpuv_1.6.5      cellranger_1.1.0  gtable_0.3.0     
[73] assertthat_0.2.1  xfun_0.30         broom_0.8.0       later_1.3.0      
[77] workflowr_1.7.0   ellipsis_0.3.2   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
