<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2024-01-17" />

<title>2024-01-17_CheckYangsJunctionAnnotator</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">2024-01-17_CheckYangsJunctionAnnotator</h1>
<h4 class="date">2024-01-17</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-06-06
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong>
<code>ChromatinSplicingQTLs/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/yangili1/bjf79/ChromatinSplicingQTLs/data/igsr_samples.tsv.gz
</td>
<td style="text-align:left;">
../data/igsr_samples.tsv.gz
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstree209c2d5e42d8949e4919e540f4cfed87ff4e0cbatargetblank209c2d5a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/209c2d5e42d8949e4919e540f4cfed87ff4e0cba" target="_blank">209c2d5</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstree209c2d5e42d8949e4919e540f4cfed87ff4e0cbatargetblank209c2d5a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/209c2d5e42d8949e4919e540f4cfed87ff4e0cba" target="_blank">209c2d5</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store
    Ignored:    code/.RData
    Ignored:    code/._report.html
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    code/.snakemake/
    Ignored:    code/APA_Processing/
    Ignored:    code/Alignments/
    Ignored:    code/ChromHMM/
    Ignored:    code/ENCODE/
    Ignored:    code/ExpressionAnalysis/
    Ignored:    code/ExtractPhenotypeBedByGenotype.py
    Ignored:    code/FastqFastp/
    Ignored:    code/FastqFastpSE/
    Ignored:    code/FastqSE/
    Ignored:    code/FineMapping/
    Ignored:    code/GTEx/
    Ignored:    code/Gencode.v34.6Colors.bed.gz
    Ignored:    code/Genotypes/
    Ignored:    code/H3K36me3_CutAndTag.pdf
    Ignored:    code/IntronSlopes/
    Ignored:    code/LR.bed
    Ignored:    code/LR.seq.bed
    Ignored:    code/LongReads/
    Ignored:    code/MYB.tracks.ini
    Ignored:    code/Metaplots/
    Ignored:    code/Misc/
    Ignored:    code/MiscCountTables/
    Ignored:    code/Multiqc/
    Ignored:    code/Multiqc_chRNA/
    Ignored:    code/NonCodingRNA/
    Ignored:    code/NonCodingRNA_annotation/
    Ignored:    code/PairwisePi1Traits.P.all.txt.gz
    Ignored:    code/PeakCalling/
    Ignored:    code/Phenotypes/
    Ignored:    code/PlotGruberQTLs/
    Ignored:    code/PlotQTLs/
    Ignored:    code/ProCapAnalysis/
    Ignored:    code/QC/
    Ignored:    code/QTL_SNP_Enrichment/
    Ignored:    code/QTLs/
    Ignored:    code/RPKM_tables/
    Ignored:    code/ReadLengthMapExperiment/
    Ignored:    code/ReadLengthMapExperimentResults/
    Ignored:    code/ReadLengthMapExperimentSpliceCounts/
    Ignored:    code/ReferenceGenome/
    Ignored:    code/Rplots.pdf
    Ignored:    code/Session.vim
    Ignored:    code/SmallMolecule/
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TODO
    Ignored:    code/Tehranchi/
    Ignored:    code/alias/
    Ignored:    code/bigwigs/
    Ignored:    code/bigwigs_FromNonWASPFilteredReads/
    Ignored:    code/config/.DS_Store
    Ignored:    code/config/._.DS_Store
    Ignored:    code/config/.ipynb_checkpoints/
    Ignored:    code/config/config.local.yaml
    Ignored:    code/dag.pdf
    Ignored:    code/dag.png
    Ignored:    code/dag.svg
    Ignored:    code/data/
    Ignored:    code/debug.ipynb
    Ignored:    code/debug_python.ipynb
    Ignored:    code/deepTools/
    Ignored:    code/featureCounts/
    Ignored:    code/featureCountsBasicGtf/
    Ignored:    code/genome_config.yaml
    Ignored:    code/gwas_summary_stats/
    Ignored:    code/hyprcoloc/
    Ignored:    code/igv_session.xml
    Ignored:    code/isoseqbams/
    Ignored:    code/log
    Ignored:    code/logs/
    Ignored:    code/notebooks/.ipynb_checkpoints/
    Ignored:    code/pi1/
    Ignored:    code/polyA.Splicing.Subset_YRI.NominalPassForColoc.bed.bgz
    Ignored:    code/rules/.ipynb_checkpoints/
    Ignored:    code/rules/OldRules/
    Ignored:    code/rules/notebooks/
    Ignored:    code/salmontest/
    Ignored:    code/scratch/
    Ignored:    code/scripts/.ipynb_checkpoints/
    Ignored:    code/scripts/GTFtools_0.8.0/
    Ignored:    code/scripts/__pycache__/
    Ignored:    code/scripts/liftOverBedpe/liftOverBedpe.py
    Ignored:    code/snakemake.dryrun.log
    Ignored:    code/snakemake.log
    Ignored:    code/snakemake.sbatch.log
    Ignored:    code/snakemake_profiles/slurm/__pycache__/
    Ignored:    code/test.introns.bed
    Ignored:    code/test.introns2.bed
    Ignored:    code/test.log
    Ignored:    code/tracks.xml
    Ignored:    data/.DS_Store
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-10.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-11.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-2.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-3.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-4.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-5.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-6.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-7.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-8.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022.csv
    Ignored:    data/Metaplots/.DS_Store
    Ignored:    output/~$20240223_NumQTLs_edited.xlsx

Untracked files:
    Untracked:  analysis/2024-04-26_eQTL_subtypePlieotropy.Rmd
    Untracked:  analysis/2024-04-30_MakeExampleConfigForPlotting.Rmd
    Untracked:  analysis/2024-05-16_IntronSizes.Rmd
    Untracked:  analysis/2024-06-06_Recreate_naRNA_v_steadyState_juncScatter.Rmd
    Untracked:  analysis/20240520_ScratchForEditorRequestedFigChanges.Rmd
    Untracked:  code/scripts/Tidy_GTEx_SummaryStats_AllCisWindowGenes.py

Unstaged changes:
    Modified:   analysis/2024-01-17_CheckYangsJunctionAnnotator.Rmd
    Modified:   analysis/2024-04-18_RelativeNMDEfficiency.Rmd
    Modified:   analysis/MakeFinalFigs_Fig3.Rmd
    Modified:   analysis/MakeFinalFigs_Fig4.Rmd
    Modified:   code/rules/GTEx.smk
    Modified:   code/scripts/GenometracksByGenotype
    Modified:   code/scripts/SmallMolecule_fitDoseResponseSplicing_AsCassetteExons.R
    Modified:   code/scripts/Tidy_GTEx_SummaryStats.py
    Modified:   data/FigS23_Join_Fig5DE.dat.tsv.gz

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/2024-01-17_CheckYangsJunctionAnnotator.Rmd</code>) and
HTML (<code>docs/2024-01-17_CheckYangsJunctionAnnotator.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/2eb27cddb069557b9f1c5c7dcad4627d4128976e/analysis/2024-01-17_CheckYangsJunctionAnnotator.Rmd" target="_blank">2eb27cd</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2024-04-24
</td>
<td>
Update nb after revisions
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/59cbb0e894b09a42f56a18648f26bbe2139b5c0d/analysis/2024-01-17_CheckYangsJunctionAnnotator.Rmd" target="_blank">59cbb0e</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2024-03-02
</td>
<td>
add tranlsate long reads script
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/39b1eede3a77c6df3e39302e21b51102b3da0101/analysis/2024-01-17_CheckYangsJunctionAnnotator.Rmd" target="_blank">39b1eed</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2024-02-13
</td>
<td>
add nb for Fig4
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="intro" class="section level2">
<h2>Intro</h2>
<p>Yang wrote a new script to annotate junctions. basically, as I
understand it, for each gene he first selects a single functional stop
and start codon. Then he considers every set of observed junctions (that
are mutually compatible that is) to create every possible transcript and
annotates whether each junction creates a PTC in any of those
transcripts. I ran the script, using all the junctions oberseved in our
sets of experiments (steady state RNA-seq, metabolic labelled, and
naRNA-seq), and now I will explore the output.</p>
<pre class="r"><code>library(tidyverse)
library(data.table)
library(scattermore)

# Set theme
theme_set(
  theme_classic() +
  theme(text=element_text(size=16,  family=&quot;Helvetica&quot;)))

# I use layer a lot, to rotate long x-axis labels
Rotate_x_labels &lt;- theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))


Annotations &lt;- read_tsv(&quot;/project/yangili1/bjf79/2024_NMD_Junction_Classifier/Leaf2_junction_classifications.txt&quot;) %&gt;%
  separate(Intron_coord, into=c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;), sep=&quot;[:-]&quot;, convert=T, remove = F) %&gt;%
  add_count(Intron_coord, name=&quot;NumberOfEntriesWithSameCoords&quot;)</code></pre>
<p>Let’s quickly explore some things about this data frame…</p>
<p>Yang mentioned some introns are duplicated if they overlap multiple
genes… He suggested just filtering these out of analysis for simplicity.
Let’s count these</p>
<pre class="r"><code>Annotations %&gt;%
  count(NumberOfEntriesWithSameCoords)</code></pre>
<pre><code># A tibble: 13 × 2
   NumberOfEntriesWithSameCoords      n
                           &lt;int&gt;  &lt;int&gt;
 1                             1 297799
 2                             2  12770
 3                             3    858
 4                             4     64
 5                             5     20
 6                             6     24
 7                             7     14
 8                             8      8
 9                             9     45
10                            15     30
11                            17     17
12                            20     60
13                            22    352</code></pre>
<p>The vast majority of introns are listed just once. But as he
suggested, let’s filter out the other ones for simplicity.</p>
<pre class="r"><code>Annotations.filtered &lt;- Annotations %&gt;%
  filter(NumberOfEntriesWithSameCoords==1)


Annotations.filtered %&gt;%
  dplyr::select(Intron_coord, Annot, Coding, UTR) %&gt;%
  pivot_longer(names_to = &quot;Column&quot;, values_to = &quot;BooleanValue&quot;, -Intron_coord) %&gt;%
  count(Column, BooleanValue) %&gt;%
  ggplot(aes(x=Column, y=n, fill=BooleanValue)) +
  geom_col(position=&#39;fill&#39;) +
  labs(y=&quot;fraction&quot;)</code></pre>
<p><img src="figure/2024-01-17_CheckYangsJunctionAnnotator.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>And reminder to self that all UTR junctions (UTR==TRUE) are
considered non-coding. Not sure if this is the precise terminology we
want in the end… Like UTR junctions are non-coding, but don’t
necessarily imply they create a NMD-targeted transcript.</p>
<pre class="r"><code>Annotations.filtered %&gt;%
  filter(Coding==F) %&gt;%
  count(UTR)</code></pre>
<pre><code># A tibble: 2 × 2
  UTR       n
  &lt;lgl&gt; &lt;int&gt;
1 FALSE 91883
2 TRUE  47312</code></pre>
<pre class="r"><code>Annotations.filtered %&gt;%
  filter(Coding) %&gt;%
  count(UTR)</code></pre>
<pre><code># A tibble: 1 × 2
  UTR        n
  &lt;lgl&gt;  &lt;int&gt;
1 FALSE 158604</code></pre>
<pre class="r"><code>Annotations.filtered %&gt;%
  filter(UTR) %&gt;%
  count(Coding)</code></pre>
<pre><code># A tibble: 1 × 2
  Coding     n
  &lt;lgl&gt;  &lt;int&gt;
1 FALSE  47312</code></pre>
<p>Ok, let’s read in our previous intron annotations and compare…</p>
<pre class="r"><code>Previous.Annotations &lt;- read_tsv(&quot;../data/IntronAnnotationsFromYang.Updated.tsv.gz&quot;)

Annotations.filtered.joined &lt;- Annotations.filtered %&gt;%
  inner_join(Previous.Annotations)

Annotations.filtered.joined %&gt;%
  distinct(NewAnnotation)</code></pre>
<pre><code># A tibble: 31 × 1
   NewAnnotation                     
   &lt;chr&gt;                             
 1 retained_intron.gencode           
 2 protein_coding.gencode            
 3 nonsense_mediated_decay.YN        
 4 processed_transcript.gencode      
 5 nonsense_mediated_decay.pstopcodon
 6 stable.YY                         
 7 nonsense_mediated_decay.far3p     
 8 stable.UTR_junction               
 9 nonsense_mediated_decay.far5p     
10 nonsense_mediated_decay.gencode   
# … with 21 more rows</code></pre>
<pre class="r"><code>Annotations.filtered.joined %&gt;%
  distinct(SuperAnnotation)</code></pre>
<pre><code># A tibble: 6 × 1
  SuperAnnotation                       
  &lt;chr&gt;                                 
1 AnnotatedJunc_UnproductiveCodingGene  
2 AnnotatedJunc_ProductiveCodingGene    
3 UnannotatedJunc_UnproductiveCodingGene
4 UnannotatedJunc_ProductiveCodingGene  
5 AnnotatedJunc_NoncodingGene           
6 UnannotatedJunc_NoncodingJunc         </code></pre>
<pre class="r"><code>Annotations.filtered.joined %&gt;%
  count(NewAnnotation, SuperAnnotation,  Annot, Coding, UTR) %&gt;%
  group_by(NewAnnotation) %&gt;%
  mutate(Group_n = sum(n)) %&gt;%
  ungroup() %&gt;%
  mutate(Color = recode(SuperAnnotation, AnnotatedJunc_NoncodingGene=&quot;#6a3d9a&quot;, UnannotatedJunc_NoncodingJunc=&quot;#cab2d6&quot;, AnnotatedJunc_UnproductiveCodingGene=&quot;#e31a1c&quot;, UnannotatedJunc_UnproductiveCodingGene=&quot;#fb9a99&quot;, AnnotatedJunc_ProductiveCodingGene=&quot;#1f78b4&quot;, UnannotatedJunc_ProductiveCodingGene=&quot;#a6cee3&quot;)) %&gt;%
  mutate(groupName = str_glue(&quot;n={Group_n}\n{NewAnnotation}&quot;)) %&gt;%
  filter(Group_n&gt;50) %&gt;%
  arrange(SuperAnnotation, Group_n) %&gt;%
  pivot_longer(names_to = &quot;Column&quot;, values_to = &quot;BooleanValue&quot;,  c(&quot;Annot&quot;, &quot;Coding&quot;, &quot;UTR&quot;)) %&gt;%
  ggplot() +
    geom_col(aes(x=Column, y=n, fill=BooleanValue), position=&#39;fill&#39;) +
    geom_rect(data = . %&gt;%
                distinct(groupName, Color,.keep_all=T), alpha=0,
              xmin=-Inf, xmax=Inf, ymin=-Inf, ymax=Inf, size=3, aes(color=Color)) +
   scale_color_identity() +
    labs(y=&quot;fraction&quot;) +
    facet_wrap(~groupName, scales=&quot;free&quot;)</code></pre>
<p><img src="figure/2024-01-17_CheckYangsJunctionAnnotator.Rmd/unnamed-chunk-5-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>Ok I think those make sense generally. Let’s also look at relative
junction usage in naRNA vs steady-state, similar to what we did in a
supplement figure with scatter plots of junction RPM.</p>
<pre class="r"><code>juncs.long &lt;- fread(&quot;../code/SplicingAnalysis/CombinedJuncTables/All.tsv.gz&quot;)
dat.KD &lt;- fread(&quot;/project2/yangili1/cfbuenabadn/ChromatinSplicingQTLs/code/SplicingAnalysis/CombinedJuncTables/NMD_KD.tsv.gz&quot;)


juncs.long.summary &lt;- 
  bind_rows(
    juncs.long,
    dat.KD) %&gt;%
  dplyr::select(chrom, start, stop, strand, Dataset, Count) %&gt;%
  group_by(Dataset, chrom, start, stop) %&gt;%
  summarise(Sum=sum(Count)) %&gt;%
  ungroup()

juncs.long.summary.joined &lt;- juncs.long.summary %&gt;%
  group_by(Dataset) %&gt;%
  mutate(DatasetSum = sum(Sum)) %&gt;%
  ungroup() %&gt;%
  mutate(RPM = Sum/DatasetSum*1E6) %&gt;%
  mutate(stop = stop+1) %&gt;%
  inner_join(Annotations.filtered.joined, by=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;=&quot;end&quot;))

juncs.long.summary.joined %&gt;%
  distinct(Dataset)</code></pre>
<pre><code># A tibble: 9 × 1
  Dataset                  
  &lt;chr&gt;                    
1 Expression.Splicing      
2 HeLa.SMG6.KD             
3 HeLa.SMG7.KD             
4 HeLa.UPF1.KD             
5 HeLa.dKD                 
6 HeLa.scr                 
7 MetabolicLabelled.30min  
8 MetabolicLabelled.60min  
9 chRNA.Expression.Splicing</code></pre>
<pre class="r"><code>juncs.long.summary.joined %&gt;%
  filter(!UTR) %&gt;%
  filter(Dataset %in% c(&quot;Expression.Splicing&quot;, &quot;chRNA.Expression.Splicing&quot;)) %&gt;%
  dplyr::select(-DatasetSum, -Sum) %&gt;%
  pivot_wider(names_from=&quot;Dataset&quot;, values_from=&quot;RPM&quot;) %&gt;%
  mutate(Color = recode(SuperAnnotation, AnnotatedJunc_NoncodingGene=&quot;#6a3d9a&quot;, UnannotatedJunc_NoncodingJunc=&quot;#cab2d6&quot;, AnnotatedJunc_UnproductiveCodingGene=&quot;#e31a1c&quot;, UnannotatedJunc_UnproductiveCodingGene=&quot;#fb9a99&quot;, AnnotatedJunc_ProductiveCodingGene=&quot;#1f78b4&quot;, UnannotatedJunc_ProductiveCodingGene=&quot;#a6cee3&quot;)) %&gt;%
  mutate(SuperAnnotation = factor(SuperAnnotation, levels=c(&quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_UnproductiveCodingGene&quot;, &quot;AnnotatedJunc_NoncodingGene&quot;, &quot;UnannotatedJunc_NoncodingJunc&quot;))) %&gt;%
  ggplot(aes(x=Expression.Splicing, y=chRNA.Expression.Splicing)) +
  geom_point(alpha=0.01, color=&#39;black&#39;) +
  geom_density2d(aes(color=Color)) +
  geom_text(data = . %&gt;%
              replace_na(list(Expression.Splicing=1E-5, chRNA.Expression.Splicing=1E-5)) %&gt;%
              group_by(Coding, SuperAnnotation) %&gt;%
              summarise(med = round(median(chRNA.Expression.Splicing/Expression.Splicing, na.rm=F), 3),
                        n = n()) %&gt;%
              ungroup(),
            aes(label=str_glue(&quot;med={med}\nn={n}&quot;)),
            x=-Inf, y=Inf, vjust=1, hjust=0.) +
  scale_color_identity() +
  scale_x_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  scale_y_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  geom_abline(slope = 1, color=&#39;red&#39;) +
  facet_grid(Coding~SuperAnnotation) +
  theme_bw() +
  labs(x=&quot;Junction RPM, steady-state&quot;, y=&quot;Junction RPM, chRNA&quot;, caption=&quot;Filtered out UTR juncs.\nHorizontal facets are &#39;Coding&#39; value&quot;)</code></pre>
<p><img src="figure/2024-01-17_CheckYangsJunctionAnnotator.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>juncs.long.summary.joined %&gt;%
  filter(!UTR) %&gt;%
  filter(Dataset %in% c(&quot;HeLa.scr&quot;, &quot;HeLa.dKD&quot;)) %&gt;%
  dplyr::select(-DatasetSum, -Sum) %&gt;%
  pivot_wider(names_from=&quot;Dataset&quot;, values_from=&quot;RPM&quot;) %&gt;%
  mutate(Color = recode(SuperAnnotation, AnnotatedJunc_NoncodingGene=&quot;#6a3d9a&quot;, UnannotatedJunc_NoncodingJunc=&quot;#cab2d6&quot;, AnnotatedJunc_UnproductiveCodingGene=&quot;#e31a1c&quot;, UnannotatedJunc_UnproductiveCodingGene=&quot;#fb9a99&quot;, AnnotatedJunc_ProductiveCodingGene=&quot;#1f78b4&quot;, UnannotatedJunc_ProductiveCodingGene=&quot;#a6cee3&quot;)) %&gt;%
  mutate(SuperAnnotation = factor(SuperAnnotation, levels=c(&quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_UnproductiveCodingGene&quot;, &quot;AnnotatedJunc_NoncodingGene&quot;, &quot;UnannotatedJunc_NoncodingJunc&quot;))) %&gt;%
  ggplot(aes(x=HeLa.scr, y=HeLa.dKD)) +
  geom_point(alpha=0.01, color=&#39;black&#39;) +
  geom_density2d(aes(color=Color)) +
  geom_text(data = . %&gt;%
              replace_na(list(HeLa.scr=1E-5, HeLa.dKD=1E-5)) %&gt;%
              group_by(Coding, SuperAnnotation) %&gt;%
              summarise(med = round(median(HeLa.dKD/HeLa.scr, na.rm=F), 3),
                        n = n()) %&gt;%
              ungroup(),
            aes(label=str_glue(&quot;med={med}\nn={n}&quot;)),
            x=-Inf, y=Inf, vjust=1, hjust=0.) +
  scale_color_identity() +
  scale_x_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  scale_y_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  geom_abline(slope = 1, color=&#39;red&#39;) +
  facet_grid(Coding~SuperAnnotation) +
  theme_bw() +
  labs(x=&quot;Junction RPM, control&quot;, y=&quot;Junction RPM, dKD&quot;, caption=&quot;Filtered out UTR juncs.\nHorizontal facets are &#39;Coding&#39; value&quot;)</code></pre>
<p><img src="figure/2024-01-17_CheckYangsJunctionAnnotator.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<div id="response-to-review" class="section level3">
<h3>Response to review</h3>
<p>Let’s make similar versions of these plots but now from the long
read-determined context of juncs</p>
<pre class="r"><code>JuncAnnotationsFromLongReadContext &lt;- read_tsv(&quot;../output/20240322_ResponseToReviewerMostCommonJuncContexts.tsv.gz&quot;) %&gt;%
  separate(Introns, into=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;strand&quot;), sep=&quot;_&quot;, convert=T)

juncs.long.summary.joined.LongReadContext &lt;- juncs.long.summary %&gt;%
  group_by(Dataset) %&gt;%
  mutate(DatasetSum = sum(Sum)) %&gt;%
  ungroup() %&gt;%
  mutate(RPM = Sum/DatasetSum*1E6) %&gt;%
  mutate(stop = stop+1) %&gt;%
  inner_join(JuncAnnotationsFromLongReadContext, by=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;))

juncs.long.summary.joined.LongReadContext %&gt;%
  distinct(Dataset)</code></pre>
<pre><code># A tibble: 9 × 1
  Dataset                  
  &lt;chr&gt;                    
1 Expression.Splicing      
2 HeLa.SMG6.KD             
3 HeLa.SMG7.KD             
4 HeLa.UPF1.KD             
5 HeLa.dKD                 
6 HeLa.scr                 
7 MetabolicLabelled.30min  
8 MetabolicLabelled.60min  
9 chRNA.Expression.Splicing</code></pre>
<pre class="r"><code>juncs.long.summary.joined.LongReadContext %&gt;%
  filter(Dataset %in% c(&quot;Expression.Splicing&quot;, &quot;chRNA.Expression.Splicing&quot;)) %&gt;%
  dplyr::select(-DatasetSum, -Sum) %&gt;%
  pivot_wider(names_from=&quot;Dataset&quot;, values_from=&quot;RPM&quot;) %&gt;%
  mutate(Color = recode(SuperAnnotation, AnnotatedJunc_NoncodingGene=&quot;#6a3d9a&quot;, UnannotatedJunc_NoncodingJunc=&quot;#cab2d6&quot;, AnnotatedJunc_UnproductiveCodingGene=&quot;#e31a1c&quot;, UnannotatedJunc_UnproductiveCodingGene=&quot;#fb9a99&quot;, AnnotatedJunc_ProductiveCodingGene=&quot;#1f78b4&quot;, UnannotatedJunc_ProductiveCodingGene=&quot;#a6cee3&quot;)) %&gt;%
  mutate(SuperAnnotation = factor(SuperAnnotation, levels=c(&quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_UnproductiveCodingGene&quot;, &quot;AnnotatedJunc_NoncodingGene&quot;, &quot;UnannotatedJunc_NoncodingJunc&quot;))) %&gt;%
  ggplot(aes(x=Expression.Splicing, y=chRNA.Expression.Splicing)) +
  geom_point(alpha=0.01, color=&#39;black&#39;) +
  geom_density2d() +
  geom_text(data = . %&gt;%
              replace_na(list(Expression.Splicing=1E-5, chRNA.Expression.Splicing=1E-5)) %&gt;%
              group_by(ModeNMDFinder, SuperAnnotation) %&gt;%
              summarise(med = round(median(chRNA.Expression.Splicing/Expression.Splicing, na.rm=F), 3),
                        n = n()) %&gt;%
              ungroup(),
            aes(label=str_glue(&quot;med={med}\nn={n}&quot;)),
            x=-Inf, y=Inf, vjust=1, hjust=0.) +
  scale_color_identity() +
  scale_x_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  scale_y_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  geom_abline(slope = 1, color=&#39;red&#39;) +
  facet_grid(ModeNMDFinder~SuperAnnotation) +
  theme_bw() +
  labs(x=&quot;Junction RPM, steady-state&quot;, y=&quot;Junction RPM, chRNA&quot;)</code></pre>
<p><img src="figure/2024-01-17_CheckYangsJunctionAnnotator.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>juncs.long.summary.joined.LongReadContext %&gt;%
  filter(Dataset %in% c(&quot;HeLa.scr&quot;, &quot;HeLa.dKD&quot;)) %&gt;%
  dplyr::select(-DatasetSum, -Sum) %&gt;%
  pivot_wider(names_from=&quot;Dataset&quot;, values_from=&quot;RPM&quot;) %&gt;%
  mutate(Color = recode(SuperAnnotation, AnnotatedJunc_NoncodingGene=&quot;#6a3d9a&quot;, UnannotatedJunc_NoncodingJunc=&quot;#cab2d6&quot;, AnnotatedJunc_UnproductiveCodingGene=&quot;#e31a1c&quot;, UnannotatedJunc_UnproductiveCodingGene=&quot;#fb9a99&quot;, AnnotatedJunc_ProductiveCodingGene=&quot;#1f78b4&quot;, UnannotatedJunc_ProductiveCodingGene=&quot;#a6cee3&quot;)) %&gt;%
  mutate(SuperAnnotation = factor(SuperAnnotation, levels=c(&quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_UnproductiveCodingGene&quot;, &quot;AnnotatedJunc_NoncodingGene&quot;, &quot;UnannotatedJunc_NoncodingJunc&quot;))) %&gt;%
  ggplot(aes(x=HeLa.scr, y=HeLa.dKD)) +
  geom_point(alpha=0.01, color=&#39;black&#39;) +
  geom_density2d() +
  geom_text(data = . %&gt;%
              replace_na(list(HeLa.scr=1E-5, HeLa.dKD=1E-5)) %&gt;%
              group_by(ModeNMDFinder, SuperAnnotation) %&gt;%
              summarise(med = round(median(HeLa.dKD/HeLa.scr, na.rm=F), 3),
                        n = n()) %&gt;%
              ungroup(),
            aes(label=str_glue(&quot;med={med}\nn={n}&quot;)),
            x=-Inf, y=Inf, vjust=1, hjust=0.) +
  scale_color_identity() +
  scale_x_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  scale_y_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  geom_abline(slope = 1, color=&#39;red&#39;) +
  facet_grid(ModeNMDFinder~SuperAnnotation) +
  theme_bw() +
  labs(x=&quot;Junction RPM, control&quot;, y=&quot;Junction RPM, dKD&quot;)</code></pre>
<p><img src="figure/2024-01-17_CheckYangsJunctionAnnotator.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>…same thing but now just disregard the old categories</p>
<pre class="r"><code>juncs.long.summary.joined.LongReadContext %&gt;%
  filter(Dataset %in% c(&quot;Expression.Splicing&quot;, &quot;chRNA.Expression.Splicing&quot;)) %&gt;%
  dplyr::select(-DatasetSum, -Sum) %&gt;%
  pivot_wider(names_from=&quot;Dataset&quot;, values_from=&quot;RPM&quot;) %&gt;%
  ggplot(aes(x=Expression.Splicing, y=chRNA.Expression.Splicing)) +
  geom_point(alpha=0.01, color=&#39;black&#39;) +
  geom_density2d() +
  geom_text(data = . %&gt;%
              replace_na(list(Expression.Splicing=1E-5, chRNA.Expression.Splicing=1E-5)) %&gt;%
              group_by(ModeNMDFinder) %&gt;%
              summarise(med = round(median(chRNA.Expression.Splicing/Expression.Splicing, na.rm=F), 3),
                        n = n()) %&gt;%
              ungroup(),
            aes(label=str_glue(&quot;med={med}\nn={n}&quot;)),
            x=-Inf, y=Inf, vjust=1, hjust=0.) +
  scale_color_identity() +
  scale_x_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  scale_y_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  geom_abline(slope = 1, color=&#39;red&#39;) +
  facet_wrap(~ModeNMDFinder) +
  theme_bw() +
  labs(x=&quot;Junction RPM, steady-state&quot;, y=&quot;Junction RPM, chRNA&quot;)</code></pre>
<p><img src="figure/2024-01-17_CheckYangsJunctionAnnotator.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>juncs.long.summary.joined.LongReadContext %&gt;%
  filter(Dataset %in% c(&quot;HeLa.scr&quot;, &quot;HeLa.dKD&quot;)) %&gt;%
  dplyr::select(-DatasetSum, -Sum) %&gt;%
  pivot_wider(names_from=&quot;Dataset&quot;, values_from=&quot;RPM&quot;) %&gt;%
  ggplot(aes(x=HeLa.scr, y=HeLa.dKD)) +
  geom_point(alpha=0.01, color=&#39;black&#39;) +
  geom_density2d() +
  geom_text(data = . %&gt;%
              replace_na(list(HeLa.scr=1E-5, HeLa.dKD=1E-5)) %&gt;%
              group_by(ModeNMDFinder) %&gt;%
              summarise(med = round(median(HeLa.dKD/HeLa.scr, na.rm=F), 3),
                        n = n()) %&gt;%
              ungroup(),
            aes(label=str_glue(&quot;med={med}\nn={n}&quot;)),
            x=-Inf, y=Inf, vjust=1, hjust=0.) +
  scale_color_identity() +
  scale_x_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  scale_y_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  geom_abline(slope = 1, color=&#39;red&#39;) +
  facet_wrap(~ModeNMDFinder) +
  theme_bw() +
  labs(x=&quot;Junction RPM, control&quot;, y=&quot;Junction RPM, dKD&quot;)</code></pre>
<p><img src="figure/2024-01-17_CheckYangsJunctionAnnotator.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" />
Ok let’s plot these median fold enrichments as a heatmap, since it is
totally consistent with the levels of NMD efficiency observed in
Lindeboom et al.</p>
<pre class="r"><code>NMD.Efficiency.P.dat &lt;- bind_rows(
  juncs.long.summary.joined.LongReadContext %&gt;%
    filter(Dataset %in% c(&quot;Expression.Splicing&quot;, &quot;chRNA.Expression.Splicing&quot;)) %&gt;%
    dplyr::select(-DatasetSum, -Sum) %&gt;%
    pivot_wider(names_from=&quot;Dataset&quot;, values_from=&quot;RPM&quot;) %&gt;%
    replace_na(list(Expression.Splicing=1E-5, chRNA.Expression.Splicing=1E-5)) %&gt;%
    group_by(ModeNMDFinder) %&gt;%
    summarise(med = log2(median(chRNA.Expression.Splicing/Expression.Splicing, na.rm=F))) %&gt;%
    ungroup() %&gt;%
    mutate(Comparison = &quot;log2(naRNA/SteadyState)&quot;),
  juncs.long.summary.joined.LongReadContext %&gt;%
  filter(Dataset %in% c(&quot;HeLa.scr&quot;, &quot;HeLa.dKD&quot;)) %&gt;%
    dplyr::select(-DatasetSum, -Sum) %&gt;%
    pivot_wider(names_from=&quot;Dataset&quot;, values_from=&quot;RPM&quot;) %&gt;%
    replace_na(list(HeLa.scr=1E-5, HeLa.dKD=1E-5)) %&gt;%
    group_by(ModeNMDFinder) %&gt;%
    summarise(med = log2(median(HeLa.dKD/HeLa.scr, na.rm=F))) %&gt;%
    ungroup() %&gt;%
    mutate(Comparison = &quot;log2(HeLa.dKD/HeLa.scr)&quot;)
)

NMD.Efficiency.P.dat %&gt;%
  inner_join(NMD.Efficiency.P.dat %&gt;%
               filter(ModeNMDFinder == &quot;Last exon&quot;) %&gt;%
               dplyr::select(LastExonMed = med, Comparison)) %&gt;%
  mutate(MedEffectRelativeToLastExonMed = med - LastExonMed) %&gt;%
  mutate(ModeNMDFinder = factor(ModeNMDFinder, levels=c(&quot;Last exon&quot;, &quot;50 nt rule&quot;, &quot;Start proximal&quot;,  &quot;Long exon&quot;, &quot;Trigger NMD&quot;, &quot;No stop&quot;, &quot;No CDS&quot;))) %&gt;%
  mutate(Comparison = recode(Comparison, &quot;log2(HeLa.dKD/HeLa.scr)&quot;=str_wrap(&quot;shRNA control\nvs\nshRNA dKD NMD factors&quot;, 10), &quot;log2(naRNA/SteadyState)&quot;=str_wrap(&quot;steady-state vs naRNA&quot;, 20))) %&gt;%
  ggplot(aes(x=Comparison, y=ModeNMDFinder, fill=MedEffectRelativeToLastExonMed)) +
  geom_raster() +
  geom_text(aes(label=round(MedEffectRelativeToLastExonMed, 3))) +
  scale_fill_viridis_c() +
  scale_x_discrete(expand=c(0,0)) +
  scale_y_discrete(expand=c(0,0)) +
  labs(y=&quot;Most common full-\ntranscript context&quot;, x=&quot;Short read datasets for degredation efficiency estimate&quot;, fill=&quot;Degredation\nefficiency\nestimate&quot;)</code></pre>
<p><img src="figure/2024-01-17_CheckYangsJunctionAnnotator.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/202403_ResponseToReviewersDegredationEfficiency.pdf&quot;, width=5, height=4)</code></pre>
</div>
<div id="head-to-head-plots" class="section level3">
<h3>Head to head plots</h3>
<p>Let’s more directly plot the old annotations vs the new annotations
to see which method is better (has more enrichment in dKD or chRNA)</p>
<pre class="r"><code>juncs.long.summary.joined.toplot &lt;- juncs.long.summary.joined %&gt;%
  filter(!UTR) %&gt;%
  filter(Dataset %in% c(&quot;HeLa.scr&quot;, &quot;HeLa.dKD&quot;)) %&gt;%
  dplyr::select(-DatasetSum, -Sum) %&gt;%
  pivot_wider(names_from=&quot;Dataset&quot;, values_from=&quot;RPM&quot;) %&gt;%
  mutate(Color = recode(SuperAnnotation, AnnotatedJunc_NoncodingGene=&quot;#6a3d9a&quot;, UnannotatedJunc_NoncodingJunc=&quot;#cab2d6&quot;, AnnotatedJunc_UnproductiveCodingGene=&quot;#e31a1c&quot;, UnannotatedJunc_UnproductiveCodingGene=&quot;#fb9a99&quot;, AnnotatedJunc_ProductiveCodingGene=&quot;#1f78b4&quot;, UnannotatedJunc_ProductiveCodingGene=&quot;#a6cee3&quot;)) %&gt;%
  mutate(SuperAnnotation = factor(SuperAnnotation, levels=c(&quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_UnproductiveCodingGene&quot;, &quot;AnnotatedJunc_NoncodingGene&quot;, &quot;UnannotatedJunc_NoncodingJunc&quot;))) %&gt;%
  filter(!SuperAnnotation %in% c(&quot;AnnotatedJunc_NoncodingGene&quot;, &quot;UnannotatedJunc_NoncodingJunc&quot;))

bind_rows(
  juncs.long.summary.joined.toplot %&gt;%
    mutate(method=&quot;Old&quot;),
  juncs.long.summary.joined.toplot %&gt;%
    mutate(method=&quot;New&quot;), 
) %&gt;%
  mutate(ProductiveOrUnproductive = case_when(
    method == &quot;New&quot; ~ Coding,
    method == &quot;Old&quot; &amp; SuperAnnotation %in% c(&quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;) ~ TRUE,
    method == &quot;Old&quot; &amp; SuperAnnotation %in% c(&quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_UnproductiveCodingGene&quot;) ~ FALSE
  )) %&gt;%
  ggplot(aes(x=HeLa.scr, y=HeLa.dKD)) +
  geom_point(alpha=0.01, color=&#39;black&#39;) +
  geom_density2d(color=&#39;red&#39;) +
  geom_text(data = . %&gt;%
              replace_na(list(HeLa.scr=1E-5, HeLa.dKD=1E-5)) %&gt;%
              group_by(method, ProductiveOrUnproductive) %&gt;%
              summarise(med = round(median(HeLa.dKD/HeLa.scr, na.rm=F), 3),
                        n = n()) %&gt;%
              ungroup(),
            aes(label=str_glue(&quot;med={med}\nn={n}&quot;)),
            x=-Inf, y=Inf, vjust=1, hjust=0.) +
  scale_color_identity() +
  scale_x_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  scale_y_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  geom_abline(slope = 1, color=&#39;red&#39;) +
  facet_grid(method~ProductiveOrUnproductive) +
  theme_bw() +
  labs(x=&quot;Junction RPM, control&quot;, y=&quot;Junction RPM, dKD&quot;, caption=&quot;Filtered out UTR juncs.\nVertical facets are &#39;Coding&#39; value&quot;)</code></pre>
<p><img src="figure/2024-01-17_CheckYangsJunctionAnnotator.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Same plot but steady state vs chRNA…</p>
<pre class="r"><code>juncs.long.summary.joined.toplot &lt;- juncs.long.summary.joined %&gt;%
  filter(!UTR) %&gt;%
  filter(Dataset %in% c(&quot;Expression.Splicing&quot;, &quot;chRNA.Expression.Splicing&quot;)) %&gt;%
  dplyr::select(-DatasetSum, -Sum) %&gt;%
  pivot_wider(names_from=&quot;Dataset&quot;, values_from=&quot;RPM&quot;) %&gt;%
  mutate(Color = recode(SuperAnnotation, AnnotatedJunc_NoncodingGene=&quot;#6a3d9a&quot;, UnannotatedJunc_NoncodingJunc=&quot;#cab2d6&quot;, AnnotatedJunc_UnproductiveCodingGene=&quot;#e31a1c&quot;, UnannotatedJunc_UnproductiveCodingGene=&quot;#fb9a99&quot;, AnnotatedJunc_ProductiveCodingGene=&quot;#1f78b4&quot;, UnannotatedJunc_ProductiveCodingGene=&quot;#a6cee3&quot;)) %&gt;%
  mutate(SuperAnnotation = factor(SuperAnnotation, levels=c(&quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_UnproductiveCodingGene&quot;, &quot;AnnotatedJunc_NoncodingGene&quot;, &quot;UnannotatedJunc_NoncodingJunc&quot;))) %&gt;%
  filter(!SuperAnnotation %in% c(&quot;AnnotatedJunc_NoncodingGene&quot;, &quot;UnannotatedJunc_NoncodingJunc&quot;))

bind_rows(
  juncs.long.summary.joined.toplot %&gt;%
    mutate(method=&quot;Old&quot;),
  juncs.long.summary.joined.toplot %&gt;%
    mutate(method=&quot;New&quot;), 
) %&gt;%
  mutate(ProductiveOrUnproductive = case_when(
    method == &quot;New&quot; ~ Coding,
    method == &quot;Old&quot; &amp; SuperAnnotation %in% c(&quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;) ~ TRUE,
    method == &quot;Old&quot; &amp; SuperAnnotation %in% c(&quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_UnproductiveCodingGene&quot;) ~ FALSE
  )) %&gt;%
  ggplot(aes(x=Expression.Splicing, y=chRNA.Expression.Splicing)) +
  geom_point(alpha=0.01, color=&#39;black&#39;) +
  geom_density2d(color=&#39;red&#39;) +
  geom_text(data = . %&gt;%
              replace_na(list(Expression.Splicing=1E-5, chRNA.Expression.Splicing=1E-5)) %&gt;%
              group_by(method, ProductiveOrUnproductive) %&gt;%
              summarise(med = round(median(chRNA.Expression.Splicing/Expression.Splicing, na.rm=F), 3),
                        n = n()) %&gt;%
              ungroup(),
            aes(label=str_glue(&quot;med={med}\nn={n}&quot;)),
            x=-Inf, y=Inf, vjust=1, hjust=0.) +
  scale_color_identity() +
  scale_x_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  scale_y_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  geom_abline(slope = 1, color=&#39;red&#39;) +
  facet_grid(method~ProductiveOrUnproductive) +
  theme_bw() +
  labs(x=&quot;Junction RPM, steady-state&quot;, y=&quot;Junction RPM, chRNA&quot;, caption=&quot;Filtered out UTR juncs.\nVertical facets are &#39;Coding&#39; value&quot;)</code></pre>
<p><img src="figure/2024-01-17_CheckYangsJunctionAnnotator.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="head-to-head-with-more-inclusive-for-new-coding-method"
class="section level3">
<h3>Head to head, with more inclusive for new coding method</h3>
<p>Yang suggested tweaking the New classification to also include all
annotated productive juncs and productive. So an or statement… here is
pseudocode: IsProductiveInNewMethod = (Coding==True OR
AnnotatedProductive==TRUE). Now let’s remake those head to head
plots…</p>
<pre class="r"><code>juncs.long.summary.joined.toplot &lt;- juncs.long.summary.joined %&gt;%
  filter(!UTR) %&gt;%
  filter(Dataset %in% c(&quot;HeLa.scr&quot;, &quot;HeLa.dKD&quot;)) %&gt;%
  dplyr::select(-DatasetSum, -Sum) %&gt;%
  pivot_wider(names_from=&quot;Dataset&quot;, values_from=&quot;RPM&quot;) %&gt;%
  mutate(Color = recode(SuperAnnotation, AnnotatedJunc_NoncodingGene=&quot;#6a3d9a&quot;, UnannotatedJunc_NoncodingJunc=&quot;#cab2d6&quot;, AnnotatedJunc_UnproductiveCodingGene=&quot;#e31a1c&quot;, UnannotatedJunc_UnproductiveCodingGene=&quot;#fb9a99&quot;, AnnotatedJunc_ProductiveCodingGene=&quot;#1f78b4&quot;, UnannotatedJunc_ProductiveCodingGene=&quot;#a6cee3&quot;)) %&gt;%
  mutate(SuperAnnotation = factor(SuperAnnotation, levels=c(&quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_UnproductiveCodingGene&quot;, &quot;AnnotatedJunc_NoncodingGene&quot;, &quot;UnannotatedJunc_NoncodingJunc&quot;))) %&gt;%
  filter(!SuperAnnotation %in% c(&quot;AnnotatedJunc_NoncodingGene&quot;, &quot;UnannotatedJunc_NoncodingJunc&quot;))

bind_rows(
  juncs.long.summary.joined.toplot %&gt;%
    mutate(method=&quot;Old&quot;),
  juncs.long.summary.joined.toplot %&gt;%
    mutate(method=&quot;New&quot;), 
) %&gt;%
  mutate(ProductiveOrUnproductive = case_when(
    method == &quot;New&quot; &amp; (Coding | SuperAnnotation==&quot;AnnotatedJunc_ProductiveCodingGene&quot;) ~ TRUE,
    method == &quot;New&quot; &amp; !(Coding | SuperAnnotation==&quot;AnnotatedJunc_ProductiveCodingGene&quot;) ~ FALSE,
    method == &quot;Old&quot; &amp; SuperAnnotation %in% c(&quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;) ~ TRUE,
    method == &quot;Old&quot; &amp; SuperAnnotation %in% c(&quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_UnproductiveCodingGene&quot;) ~ FALSE
  )) %&gt;%
  ggplot(aes(x=HeLa.scr, y=HeLa.dKD)) +
  geom_point(alpha=0.01, color=&#39;black&#39;) +
  geom_density2d(color=&#39;red&#39;) +
  geom_text(data = . %&gt;%
              replace_na(list(HeLa.scr=1E-5, HeLa.dKD=1E-5)) %&gt;%
              group_by(method, ProductiveOrUnproductive) %&gt;%
              summarise(med = round(median(HeLa.dKD/HeLa.scr, na.rm=F), 3),
                        n = n()) %&gt;%
              ungroup(),
            aes(label=str_glue(&quot;med={med}\nn={n}&quot;)),
            x=-Inf, y=Inf, vjust=1, hjust=0.) +
  scale_color_identity() +
  scale_x_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  scale_y_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  geom_abline(slope = 1, color=&#39;red&#39;) +
  facet_grid(method~ProductiveOrUnproductive) +
  theme_bw() +
  labs(x=&quot;Junction RPM, control&quot;, y=&quot;Junction RPM, dKD&quot;, caption=&quot;Filtered out UTR juncs.\nVertical facets are &#39;Coding&#39; value&quot;)</code></pre>
<p><img src="figure/2024-01-17_CheckYangsJunctionAnnotator.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Same plot but steady state vs chRNA…</p>
<pre class="r"><code>juncs.long.summary.joined.toplot &lt;- juncs.long.summary.joined %&gt;%
  filter(!UTR) %&gt;%
  filter(Dataset %in% c(&quot;Expression.Splicing&quot;, &quot;chRNA.Expression.Splicing&quot;)) %&gt;%
  dplyr::select(-DatasetSum, -Sum) %&gt;%
  pivot_wider(names_from=&quot;Dataset&quot;, values_from=&quot;RPM&quot;) %&gt;%
  mutate(Color = recode(SuperAnnotation, AnnotatedJunc_NoncodingGene=&quot;#6a3d9a&quot;, UnannotatedJunc_NoncodingJunc=&quot;#cab2d6&quot;, AnnotatedJunc_UnproductiveCodingGene=&quot;#e31a1c&quot;, UnannotatedJunc_UnproductiveCodingGene=&quot;#fb9a99&quot;, AnnotatedJunc_ProductiveCodingGene=&quot;#1f78b4&quot;, UnannotatedJunc_ProductiveCodingGene=&quot;#a6cee3&quot;)) %&gt;%
  mutate(SuperAnnotation = factor(SuperAnnotation, levels=c(&quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_UnproductiveCodingGene&quot;, &quot;AnnotatedJunc_NoncodingGene&quot;, &quot;UnannotatedJunc_NoncodingJunc&quot;))) %&gt;%
  filter(!SuperAnnotation %in% c(&quot;AnnotatedJunc_NoncodingGene&quot;, &quot;UnannotatedJunc_NoncodingJunc&quot;))

bind_rows(
  juncs.long.summary.joined.toplot %&gt;%
    mutate(method=&quot;Old&quot;),
  juncs.long.summary.joined.toplot %&gt;%
    mutate(method=&quot;New&quot;), 
) %&gt;%
  mutate(ProductiveOrUnproductive = case_when(
    method == &quot;New&quot; &amp; (Coding | SuperAnnotation==&quot;AnnotatedJunc_ProductiveCodingGene&quot;) ~ TRUE,
    method == &quot;New&quot; &amp; !(Coding | SuperAnnotation==&quot;AnnotatedJunc_ProductiveCodingGene&quot;) ~ FALSE,
    method == &quot;Old&quot; &amp; SuperAnnotation %in% c(&quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;) ~ TRUE,
    method == &quot;Old&quot; &amp; SuperAnnotation %in% c(&quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_UnproductiveCodingGene&quot;) ~ FALSE
  )) %&gt;%
  ggplot(aes(x=Expression.Splicing, y=chRNA.Expression.Splicing)) +
  geom_point(alpha=0.01, color=&#39;black&#39;) +
  geom_density2d(color=&#39;red&#39;) +
  geom_text(data = . %&gt;%
              replace_na(list(Expression.Splicing=1E-5, chRNA.Expression.Splicing=1E-5)) %&gt;%
              group_by(method, ProductiveOrUnproductive) %&gt;%
              summarise(med = round(median(chRNA.Expression.Splicing/Expression.Splicing, na.rm=F), 3),
                        n = n()) %&gt;%
              ungroup(),
            aes(label=str_glue(&quot;med={med}\nn={n}&quot;)),
            x=-Inf, y=Inf, vjust=1, hjust=0.) +
  scale_color_identity() +
  scale_x_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  scale_y_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  geom_abline(slope = 1, color=&#39;red&#39;) +
  facet_grid(method~ProductiveOrUnproductive) +
  theme_bw() +
  labs(x=&quot;Junction RPM, steady-state&quot;, y=&quot;Junction RPM, naRNA&quot;)</code></pre>
<p><img src="figure/2024-01-17_CheckYangsJunctionAnnotator.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>…same plot but just new method</p>
<pre class="r"><code>dat.to.Plot &lt;- juncs.long.summary.joined.toplot %&gt;%
  mutate(ProductiveOrUnproductive = case_when(
    (Coding | SuperAnnotation==&quot;AnnotatedJunc_ProductiveCodingGene&quot;) ~ &quot;Productive\nSplice juncs&quot;,
    !(Coding | SuperAnnotation==&quot;AnnotatedJunc_ProductiveCodingGene&quot;) ~ &quot;Unproductive\nSplice juncs&quot;,
  ))

ggplot(dat.to.Plot, aes(x=Expression.Splicing, y=chRNA.Expression.Splicing)) +
  geom_point(alpha=0.01, color=&#39;black&#39;) +
  geom_density2d(color=&#39;red&#39;) +
  geom_abline(slope = 1, color=&#39;red&#39;) +
  geom_text(data = . %&gt;%
              replace_na(list(Expression.Splicing=1E-5, chRNA.Expression.Splicing=1E-5)) %&gt;%
              group_by(ProductiveOrUnproductive) %&gt;%
              summarise(med = round(median(chRNA.Expression.Splicing/Expression.Splicing, na.rm=F), 3),
                        n = n()) %&gt;%
              ungroup(),
            aes(label=str_glue(&quot;median FC={med}\nn={n}&quot;)),
            x=-Inf, y=Inf, vjust=1.1, hjust=0.) +
  scale_color_identity() +
  scale_x_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  scale_y_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  facet_wrap(~ProductiveOrUnproductive) +
  theme_bw() +
  coord_fixed() +
  labs(x=&quot;Junction RPM, steady-state&quot;, y=&quot;Junction RPM, naRNA&quot;)</code></pre>
<p><img src="figure/2024-01-17_CheckYangsJunctionAnnotator.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(&quot;../code/scratch/PlotsForYangGrant.JunctionScatter.Points.pdf&quot;, height=3, width=7)

ggplot(dat.to.Plot, aes(x=Expression.Splicing, y=chRNA.Expression.Splicing)) +
  # geom_point(alpha=0.01, color=&#39;black&#39;) +
  # geom_density2d(color=&#39;red&#39;) +
  geom_hex(bins=70) +
  geom_abline(slope = 1, color=&#39;red&#39;) +
  geom_text(data = . %&gt;%
              replace_na(list(Expression.Splicing=1E-5, chRNA.Expression.Splicing=1E-5)) %&gt;%
              group_by(ProductiveOrUnproductive) %&gt;%
              summarise(med = round(median(chRNA.Expression.Splicing/Expression.Splicing, na.rm=F), 3),
                        n = n()) %&gt;%
              ungroup(),
            aes(label=str_glue(&quot; median FC={med}\n n={n}&quot;)),
            x=-Inf, y=Inf, vjust=1.1, hjust=0.) +
  scale_color_identity() +
  scale_fill_viridis_c(trans=&#39;log10&#39;, option = &#39;B&#39;) + 
  scale_x_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  scale_y_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  facet_wrap(~ProductiveOrUnproductive) +
  theme_bw() +
  coord_fixed() +
  labs(x=&quot;Junction RPM, steady-state&quot;, y=&quot;Junction RPM, naRNA&quot;)</code></pre>
<p><img src="figure/2024-01-17_CheckYangsJunctionAnnotator.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(&quot;../code/scratch/PlotsForYangGrant.JunctionScatter.Hexbin.pdf&quot;, height=2, width=4.25)</code></pre>
<p>Plots for Yang’s grant.</p>
<pre class="r"><code>juncs.long.summary.joined.toplot &lt;- juncs.long.summary.joined %&gt;%
  filter(!UTR) %&gt;%
  filter(Dataset %in% c(&quot;HeLa.scr&quot;, &quot;HeLa.dKD&quot;)) %&gt;%
  dplyr::select(-DatasetSum, -Sum) %&gt;%
  pivot_wider(names_from=&quot;Dataset&quot;, values_from=&quot;RPM&quot;) %&gt;%
  mutate(Color = recode(SuperAnnotation, AnnotatedJunc_NoncodingGene=&quot;#6a3d9a&quot;, UnannotatedJunc_NoncodingJunc=&quot;#cab2d6&quot;, AnnotatedJunc_UnproductiveCodingGene=&quot;#e31a1c&quot;, UnannotatedJunc_UnproductiveCodingGene=&quot;#fb9a99&quot;, AnnotatedJunc_ProductiveCodingGene=&quot;#1f78b4&quot;, UnannotatedJunc_ProductiveCodingGene=&quot;#a6cee3&quot;)) %&gt;%
  mutate(SuperAnnotation = factor(SuperAnnotation, levels=c(&quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_UnproductiveCodingGene&quot;, &quot;AnnotatedJunc_NoncodingGene&quot;, &quot;UnannotatedJunc_NoncodingJunc&quot;))) %&gt;%
  filter(!SuperAnnotation %in% c(&quot;AnnotatedJunc_NoncodingGene&quot;, &quot;UnannotatedJunc_NoncodingJunc&quot;))

bind_rows(
  juncs.long.summary.joined.toplot %&gt;%
    mutate(method=&quot;Old&quot;),
  juncs.long.summary.joined.toplot %&gt;%
    mutate(method=&quot;New&quot;), 
) %&gt;%
  mutate(ProductiveOrUnproductive = case_when(
    method == &quot;New&quot; &amp; (Coding | SuperAnnotation==&quot;AnnotatedJunc_ProductiveCodingGene&quot;) ~ TRUE,
    method == &quot;New&quot; &amp; !(Coding | SuperAnnotation==&quot;AnnotatedJunc_ProductiveCodingGene&quot;) ~ FALSE,
    method == &quot;Old&quot; &amp; SuperAnnotation %in% c(&quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;) ~ TRUE,
    method == &quot;Old&quot; &amp; SuperAnnotation %in% c(&quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_UnproductiveCodingGene&quot;) ~ FALSE
  )) %&gt;%
  mutate(ProductiveOrUnproductive = if_else(ProductiveOrUnproductive, &quot;Productive&quot;, &quot;Unproductive&quot;)) %&gt;%
  ggplot(aes(x=HeLa.scr, y=HeLa.dKD)) +
  # geom_scattermore(alpha=0.01, color=&#39;black&#39;) +
  # geom_hex() +
  # scale_fill_viridis_c()
  geom_point(alpha=0.01, color=&#39;black&#39;) +
  geom_density2d(color=&#39;red&#39;) +
  geom_text(data = . %&gt;%
              replace_na(list(HeLa.scr=1E-5, HeLa.dKD=1E-5)) %&gt;%
              group_by(method, ProductiveOrUnproductive) %&gt;%
              summarise(med = round(median(HeLa.dKD/HeLa.scr, na.rm=F), 3),
                        n = n()) %&gt;%
              ungroup(),
            aes(label=str_glue(&quot;Med={med}\nn={n}&quot;)),
            x=-Inf, y=Inf, vjust=1, hjust=0.) +
  scale_color_identity() +
  scale_x_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  scale_y_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  geom_abline(slope = 1, color=&#39;red&#39;) +
  facet_grid(method~ProductiveOrUnproductive) +
  theme_classic() +
  labs(x=&quot;Junction RPM, control&quot;, y=&quot;Junction RPM, dKD&quot;, caption=&quot;Filtered out UTR juncs.\nVertical facets are &#39;Coding&#39; value&quot;)
ggsave(&quot;/project2/yangili1/bjf79/scratch/EvaluateJuncClassification.shRNAKD.pdf&quot;, height=6, width=3)

bind_rows(
  juncs.long.summary.joined.toplot %&gt;%
    mutate(method=&quot;Old&quot;),
  juncs.long.summary.joined.toplot %&gt;%
    mutate(method=&quot;New&quot;), 
) %&gt;%
  mutate(ProductiveOrUnproductive = case_when(
    method == &quot;New&quot; &amp; (Coding | SuperAnnotation==&quot;AnnotatedJunc_ProductiveCodingGene&quot;) ~ TRUE,
    method == &quot;New&quot; &amp; !(Coding | SuperAnnotation==&quot;AnnotatedJunc_ProductiveCodingGene&quot;) ~ FALSE,
    method == &quot;Old&quot; &amp; SuperAnnotation %in% c(&quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;) ~ TRUE,
    method == &quot;Old&quot; &amp; SuperAnnotation %in% c(&quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_UnproductiveCodingGene&quot;) ~ FALSE
  )) %&gt;%
  mutate(ProductiveOrUnproductive = if_else(ProductiveOrUnproductive, &quot;Productive&quot;, &quot;Unproductive&quot;)) %&gt;%
  filter(method == &quot;New&quot;) %&gt;%
  ggplot(aes(x=HeLa.scr, y=HeLa.dKD)) +
  geom_point(alpha=0.01, color=&#39;black&#39;) +
  geom_density2d(color=&#39;red&#39;) +
  geom_text(data = . %&gt;%
              replace_na(list(HeLa.scr=1E-5, HeLa.dKD=1E-5)) %&gt;%
              group_by(ProductiveOrUnproductive) %&gt;%
              summarise(med = round(median(HeLa.dKD/HeLa.scr, na.rm=F), 3),
                        n = n()) %&gt;%
              ungroup(),
            aes(label=str_glue(&quot;Med={med}\nn={n}&quot;)),
            x=-Inf, y=Inf, vjust=1, hjust=0.) +
  scale_color_identity() +
  scale_x_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  scale_y_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  geom_abline(slope = 1, color=&#39;red&#39;) +
  facet_wrap(~ProductiveOrUnproductive) +
  theme_classic() +
  labs(x=&quot;Junction RPM, control&quot;, y=&quot;Junction RPM, dKD&quot;, caption=&quot;Filtered out UTR juncs.\nVertical facets are &#39;Coding&#39; value&quot;)
# ggsave(&quot;/project2/yangili1/bjf79/scratch/EvaluateJuncClassification.shRNAKD.NoOld.pdf&quot;, height=3, width=3)





juncs.long.summary.joined.toplot &lt;- juncs.long.summary.joined %&gt;%
  filter(!UTR) %&gt;%
  filter(Dataset %in% c(&quot;Expression.Splicing&quot;, &quot;chRNA.Expression.Splicing&quot;)) %&gt;%
  dplyr::select(-DatasetSum, -Sum) %&gt;%
  pivot_wider(names_from=&quot;Dataset&quot;, values_from=&quot;RPM&quot;) %&gt;%
  mutate(Color = recode(SuperAnnotation, AnnotatedJunc_NoncodingGene=&quot;#6a3d9a&quot;, UnannotatedJunc_NoncodingJunc=&quot;#cab2d6&quot;, AnnotatedJunc_UnproductiveCodingGene=&quot;#e31a1c&quot;, UnannotatedJunc_UnproductiveCodingGene=&quot;#fb9a99&quot;, AnnotatedJunc_ProductiveCodingGene=&quot;#1f78b4&quot;, UnannotatedJunc_ProductiveCodingGene=&quot;#a6cee3&quot;)) %&gt;%
  mutate(SuperAnnotation = factor(SuperAnnotation, levels=c(&quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_UnproductiveCodingGene&quot;, &quot;AnnotatedJunc_NoncodingGene&quot;, &quot;UnannotatedJunc_NoncodingJunc&quot;))) %&gt;%
  filter(!SuperAnnotation %in% c(&quot;AnnotatedJunc_NoncodingGene&quot;, &quot;UnannotatedJunc_NoncodingJunc&quot;))

bind_rows(
  juncs.long.summary.joined.toplot %&gt;%
    mutate(method=&quot;Old&quot;),
  juncs.long.summary.joined.toplot %&gt;%
    mutate(method=&quot;New&quot;), 
) %&gt;%
  mutate(ProductiveOrUnproductive = case_when(
    method == &quot;New&quot; &amp; (Coding | SuperAnnotation==&quot;AnnotatedJunc_ProductiveCodingGene&quot;) ~ TRUE,
    method == &quot;New&quot; &amp; !(Coding | SuperAnnotation==&quot;AnnotatedJunc_ProductiveCodingGene&quot;) ~ FALSE,
    method == &quot;Old&quot; &amp; SuperAnnotation %in% c(&quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;) ~ TRUE,
    method == &quot;Old&quot; &amp; SuperAnnotation %in% c(&quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_UnproductiveCodingGene&quot;) ~ FALSE
  )) %&gt;%
  mutate(ProductiveOrUnproductive = if_else(ProductiveOrUnproductive, &quot;Productive&quot;, &quot;Unproductive&quot;)) %&gt;%
  ggplot(aes(x=Expression.Splicing, y=chRNA.Expression.Splicing)) +
  geom_point(alpha=0.01, color=&#39;black&#39;) +
  geom_density2d(color=&#39;red&#39;) +
  geom_text(data = . %&gt;%
              replace_na(list(Expression.Splicing=1E-5, chRNA.Expression.Splicing=1E-5)) %&gt;%
              group_by(method, ProductiveOrUnproductive) %&gt;%
              summarise(med = round(median(chRNA.Expression.Splicing/Expression.Splicing, na.rm=F), 3),
                        n = n()) %&gt;%
              ungroup(),
            aes(label=str_glue(&quot;Med={med}\nn={n}&quot;)),
            x=-Inf, y=Inf, vjust=1, hjust=0.) +
  scale_color_identity() +
  scale_x_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  scale_y_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  geom_abline(slope = 1, color=&#39;red&#39;) +
  facet_grid(method~ProductiveOrUnproductive) +
  theme_classic() +
  labs(x=&quot;Junction RPM, steady-state&quot;, y=&quot;Junction RPM, naRNA&quot;, caption=&quot;Filtered out UTR juncs.\nVertical facets are &#39;Coding&#39; value&quot;)
# ggsave(&quot;/project2/yangili1/bjf79/scratch/EvaluateJuncClassification.naRNA.pdf&quot;, height=6, width=3)

bind_rows(
  juncs.long.summary.joined.toplot %&gt;%
    mutate(method=&quot;Old&quot;),
  juncs.long.summary.joined.toplot %&gt;%
    mutate(method=&quot;New&quot;), 
) %&gt;%
  mutate(ProductiveOrUnproductive = case_when(
    method == &quot;New&quot; &amp; (Coding | SuperAnnotation==&quot;AnnotatedJunc_ProductiveCodingGene&quot;) ~ TRUE,
    method == &quot;New&quot; &amp; !(Coding | SuperAnnotation==&quot;AnnotatedJunc_ProductiveCodingGene&quot;) ~ FALSE,
    method == &quot;Old&quot; &amp; SuperAnnotation %in% c(&quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;) ~ TRUE,
    method == &quot;Old&quot; &amp; SuperAnnotation %in% c(&quot;AnnotatedJunc_UnproductiveCodingGene&quot;, &quot;UnannotatedJunc_UnproductiveCodingGene&quot;) ~ FALSE
  )) %&gt;%
  mutate(ProductiveOrUnproductive = if_else(ProductiveOrUnproductive, &quot;Productive&quot;, &quot;Unproductive&quot;)) %&gt;%
  filter(method == &quot;New&quot;) %&gt;%
  ggplot(aes(x=Expression.Splicing, y=chRNA.Expression.Splicing)) +
  geom_point(alpha=0.01, color=&#39;black&#39;) +
  geom_density2d(color=&#39;red&#39;) +
  geom_text(data = . %&gt;%
              replace_na(list(Expression.Splicing=1E-5, chRNA.Expression.Splicing=1E-5)) %&gt;%
              group_by(ProductiveOrUnproductive) %&gt;%
              summarise(med = round(median(chRNA.Expression.Splicing/Expression.Splicing, na.rm=F), 3),
                        n = n()) %&gt;%
              ungroup(),
            aes(label=str_glue(&quot;Med={med}\nn={n}&quot;)),
            x=-Inf, y=Inf, vjust=1, hjust=0.) +
  scale_color_identity() +
  scale_x_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  scale_y_continuous(trans=&quot;log10&quot;, breaks=c(1E-1, 10, 1000), labels=c(&quot;0.1&quot;, &quot;10&quot;, &quot;1000&quot;)) +
  geom_abline(slope = 1, color=&#39;red&#39;) +
  facet_wrap(~ProductiveOrUnproductive) +
  theme_classic() +
  labs(x=&quot;Junction RPM, steady-state&quot;, y=&quot;Junction RPM, naRNA&quot;, caption=&quot;Filtered out UTR juncs.\nVertical facets are &#39;Coding&#39; value&quot;)
# ggsave(&quot;/project2/yangili1/bjf79/scratch/EvaluateJuncClassification.naRNA.NoOld.pdf&quot;, height=3, width=3)</code></pre>
</div>
<div id="test-updated-script" class="section level3">
<h3>Test updated script</h3>
<p>Yang updated his script to reflect this new change… I reran the
script and now let’s verify the new annotations.</p>
<pre class="r"><code>NewScript.Annotations &lt;- read_tsv(&quot;/project/yangili1/bjf79/2024_NMD_Junction_Classifier/Leaf2_updated_junction_classifications.txt&quot;) %&gt;%
  separate(Intron_coord, into=c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;), sep=&quot;[:-]&quot;, convert=T, remove = F) %&gt;%
  add_count(Intron_coord, name=&quot;NumberOfEntriesWithSameCoords&quot;)

juncs.long.summary.joined.toplot %&gt;%
  inner_join(
    NewScript.Annotations %&gt;%
      dplyr::select(Intron_coord, CodingUpdated = Coding)
  ) %&gt;%
  mutate(ProductiveOrUnproductiveMyTest = case_when(
    Coding | SuperAnnotation==&quot;AnnotatedJunc_ProductiveCodingGene&quot; ~ TRUE,
    TRUE ~ FALSE)) %&gt;%
  count(CodingUpdated, ProductiveOrUnproductiveMyTest)</code></pre>
<pre><code># A tibble: 3 × 3
  CodingUpdated ProductiveOrUnproductiveMyTest      n
  &lt;lgl&gt;         &lt;lgl&gt;                           &lt;int&gt;
1 FALSE         FALSE                           63886
2 FALSE         TRUE                               11
3 TRUE          TRUE                           164980</code></pre>
<p>The script is correctly updated, such that the Coding column in the
new srcipt output is true if it can create a transcript that reaches a
stop codon, <em>OR</em> if it is already in an annotated functional
transcript.</p>
</div>
<div id="unrelated-code" class="section level3">
<h3>Unrelated code</h3>
<p>…To make new metadata file for different project with new sequencing
data.</p>
<pre class="r"><code>Metadata &lt;- read_tsv(&quot;/project2/yangili1/bjf79/sm_splicingmodulators/code/config/samples.tsv&quot;)

NewFastq &lt;- read_tsv(&quot;/project2/yangili1/bjf79/sm_splicingmodulators/code/scratch/NewSamples.tsv&quot;, col_names=c(&quot;R1&quot;, &quot;R2&quot;)) %&gt;%
  mutate(old.sample.name = str_replace(R1, &quot;/cds/yangili1/bjf79/Fastq/20240118_AdditionalSMSM74Replicates/Sample_(.+?)_L00.+$&quot;, &quot;\\1&quot;)) %&gt;%
  mutate(old.sample.name = case_when(
    old.sample.name == &quot;DMSO-3_101823&quot; ~ &quot;DMSO-1_101823&quot;,
    old.sample.name == &quot;DMSO-4_101823&quot; ~ &quot;DMSO-2_101823&quot;,
    TRUE ~ str_replace(old.sample.name, &quot;^(.+?)-2$&quot;, &quot;\\1-1&quot;)
  ))

bind_rows(
  Metadata,
  Metadata %&gt;%
    filter(Experiment == 6) %&gt;%
    dplyr::select(-R1, -R2) %&gt;%
    distinct(sample, .keep_all=T) %&gt;%
    inner_join(NewFastq)
) %&gt;%
  arrange(sample) %&gt;%
  write_tsv(&quot;/project2/yangili1/bjf79/sm_splicingmodulators/code/config/samples.tsv&quot;)

# also another samples.tsv file for a snakemake to confirm similarity between batches and proper sample matching
bind_rows(
  Metadata %&gt;%
    filter(Experiment == 6),
  Metadata %&gt;%
    filter(Experiment == 6) %&gt;%
    dplyr::select(-R1, -R2) %&gt;%
    distinct(sample, .keep_all=T) %&gt;%
    inner_join(NewFastq) %&gt;%
    mutate(old.sample.name = str_replace(R1, &quot;/cds/yangili1/bjf79/Fastq/20240118_AdditionalSMSM74Replicates/Sample_(.+?)_L00.+$&quot;, &quot;\\1&quot;))
) %&gt;%
  mutate(sample = old.sample.name) %&gt;%
  write_tsv(&quot;/project2/yangili1/bjf79/20240118_CheckNewSMSM74_SeqLane_/code/config/samples.tsv&quot;)</code></pre>
</div>
<div id="write-organized-metadata-to-help-carlos-upload-data-to-geo"
class="section level3">
<h3>Write organized metadata to help Carlos upload data to GEO</h3>
<pre class="r"><code>Metadata &lt;- read_tsv(&quot;../code/config/samples.tsv&quot;)

Metadata %&gt;%
  filter(Phenotype == &quot;H3K36ME3&quot;) %&gt;%
  distinct(IndID) %&gt;%
  inner_join(
    read_tsv(&quot;/project2/yangili1/bjf79/ChromatinSplicingQTLs/data/igsr_samples.tsv.gz&quot;),
    by=c(&quot;IndID&quot;=&quot;Sample name&quot;)
  ) %&gt;%
  count(Sex)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] scattermore_0.8   data.table_1.14.2 forcats_0.5.1     stringr_1.4.0    
 [5] dplyr_1.0.9       purrr_0.3.4       readr_2.1.2       tidyr_1.2.0      
 [9] tibble_3.1.7      ggplot2_3.3.6     tidyverse_1.3.1  

loaded via a namespace (and not attached):
 [1] fs_1.5.2          lubridate_1.8.0   bit64_4.0.5       httr_1.4.3       
 [5] rprojroot_2.0.3   tools_4.2.0       backports_1.4.1   bslib_0.3.1      
 [9] utf8_1.2.2        R6_2.5.1          DBI_1.1.2         colorspace_2.0-3 
[13] withr_2.5.0       tidyselect_1.1.2  bit_4.0.4         compiler_4.2.0   
[17] git2r_0.30.1      textshaping_0.3.6 cli_3.3.0         rvest_1.0.2      
[21] xml2_1.3.3        isoband_0.2.5     labeling_0.4.2    sass_0.4.1       
[25] scales_1.2.0      hexbin_1.28.3     systemfonts_1.0.4 digest_0.6.29    
[29] rmarkdown_2.14    R.utils_2.11.0    pkgconfig_2.0.3   htmltools_0.5.2  
[33] dbplyr_2.1.1      fastmap_1.1.0     highr_0.9         rlang_1.0.2      
[37] readxl_1.4.0      rstudioapi_0.13   jquerylib_0.1.4   farver_2.1.0     
[41] generics_0.1.2    jsonlite_1.8.0    vroom_1.5.7       R.oo_1.24.0      
[45] magrittr_2.0.3    Rcpp_1.0.8.3      munsell_0.5.0     fansi_1.0.3      
[49] lifecycle_1.0.1   R.methodsS3_1.8.1 stringi_1.7.6     whisker_0.4      
[53] yaml_2.3.5        MASS_7.3-56       grid_4.2.0        parallel_4.2.0   
[57] promises_1.2.0.1  crayon_1.5.1      lattice_0.20-45   haven_2.5.0      
[61] hms_1.1.1         knitr_1.39        pillar_1.7.0      reprex_2.0.1     
[65] glue_1.6.2        evaluate_0.15     modelr_0.1.8      vctrs_0.4.1      
[69] tzdb_0.3.0        httpuv_1.6.5      cellranger_1.1.0  gtable_0.3.0     
[73] assertthat_0.2.1  xfun_0.30         broom_0.8.0       later_1.3.0      
[77] ragg_1.2.5        viridisLite_0.4.0 workflowr_1.7.0   ellipsis_0.3.2   </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
