<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2024-04-18" />

<title>2024-04-18_RelativeNMDEfficiency</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">2024-04-18_RelativeNMDEfficiency</h1>
<h4 class="date">2024-04-18</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-04-24
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>ChromatinSplicingQTLs/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstree420d8ddd276030bf2ad083d2f68a8fd22981956ctargetblank420d8dda">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/420d8ddd276030bf2ad083d2f68a8fd22981956c" target="_blank">420d8dd</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstree420d8ddd276030bf2ad083d2f68a8fd22981956ctargetblank420d8dda"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/420d8ddd276030bf2ad083d2f68a8fd22981956c" target="_blank">420d8dd</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store
    Ignored:    code/.RData
    Ignored:    code/._report.html
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    code/.snakemake/
    Ignored:    code/APA_Processing/
    Ignored:    code/Alignments/
    Ignored:    code/ChromHMM/
    Ignored:    code/ENCODE/
    Ignored:    code/ExpressionAnalysis/
    Ignored:    code/ExtractPhenotypeBedByGenotype.py
    Ignored:    code/FastqFastp/
    Ignored:    code/FastqFastpSE/
    Ignored:    code/FastqSE/
    Ignored:    code/FineMapping/
    Ignored:    code/GTEx/
    Ignored:    code/Gencode.v34.6Colors.bed.gz
    Ignored:    code/Genotypes/
    Ignored:    code/H3K36me3_CutAndTag.pdf
    Ignored:    code/IntronSlopes/
    Ignored:    code/LR.bed
    Ignored:    code/LR.seq.bed
    Ignored:    code/LongReads/
    Ignored:    code/MYB.tracks.ini
    Ignored:    code/Metaplots/
    Ignored:    code/Misc/
    Ignored:    code/MiscCountTables/
    Ignored:    code/Multiqc/
    Ignored:    code/Multiqc_chRNA/
    Ignored:    code/NonCodingRNA/
    Ignored:    code/NonCodingRNA_annotation/
    Ignored:    code/PairwisePi1Traits.P.all.txt.gz
    Ignored:    code/PeakCalling/
    Ignored:    code/Phenotypes/
    Ignored:    code/PlotGruberQTLs/
    Ignored:    code/PlotQTLs/
    Ignored:    code/ProCapAnalysis/
    Ignored:    code/QC/
    Ignored:    code/QTL_SNP_Enrichment/
    Ignored:    code/QTLs/
    Ignored:    code/RPKM_tables/
    Ignored:    code/ReadLengthMapExperiment/
    Ignored:    code/ReadLengthMapExperimentResults/
    Ignored:    code/ReadLengthMapExperimentSpliceCounts/
    Ignored:    code/ReferenceGenome/
    Ignored:    code/Rplots.pdf
    Ignored:    code/Session.vim
    Ignored:    code/SmallMolecule/
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TODO
    Ignored:    code/Tehranchi/
    Ignored:    code/alias/
    Ignored:    code/bigwigs/
    Ignored:    code/bigwigs_FromNonWASPFilteredReads/
    Ignored:    code/config/.DS_Store
    Ignored:    code/config/._.DS_Store
    Ignored:    code/config/.ipynb_checkpoints/
    Ignored:    code/config/config.local.yaml
    Ignored:    code/dag.pdf
    Ignored:    code/dag.png
    Ignored:    code/dag.svg
    Ignored:    code/data/
    Ignored:    code/debug.ipynb
    Ignored:    code/debug_python.ipynb
    Ignored:    code/deepTools/
    Ignored:    code/featureCounts/
    Ignored:    code/featureCountsBasicGtf/
    Ignored:    code/genome_config.yaml
    Ignored:    code/gwas_summary_stats/
    Ignored:    code/hyprcoloc/
    Ignored:    code/igv_session.xml
    Ignored:    code/isoseqbams/
    Ignored:    code/log
    Ignored:    code/logs/
    Ignored:    code/notebooks/.ipynb_checkpoints/
    Ignored:    code/pi1/
    Ignored:    code/polyA.Splicing.Subset_YRI.NominalPassForColoc.bed.bgz
    Ignored:    code/rules/.ipynb_checkpoints/
    Ignored:    code/rules/OldRules/
    Ignored:    code/rules/notebooks/
    Ignored:    code/salmontest/
    Ignored:    code/scratch/
    Ignored:    code/scripts/.ipynb_checkpoints/
    Ignored:    code/scripts/GTFtools_0.8.0/
    Ignored:    code/scripts/__pycache__/
    Ignored:    code/scripts/liftOverBedpe/liftOverBedpe.py
    Ignored:    code/snakemake.dryrun.log
    Ignored:    code/snakemake.log
    Ignored:    code/snakemake.sbatch.log
    Ignored:    code/snakemake_profiles/slurm/__pycache__/
    Ignored:    code/test.introns.bed
    Ignored:    code/test.introns2.bed
    Ignored:    code/test.log
    Ignored:    code/tracks.xml
    Ignored:    data/.DS_Store
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-10.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-11.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-2.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-3.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-4.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-5.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-6.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-7.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-8.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022.csv
    Ignored:    data/Metaplots/.DS_Store
    Ignored:    output/~$20240223_NumQTLs_edited.xlsx

Untracked files:
    Untracked:  analysis/2024-04-18_RelativeNMDEfficiency.Rmd
    Untracked:  analysis/test.rmd
    Untracked:  output/20240322_ResponseToReviewerMostCommonJuncContexts.tsv.gz

Unstaged changes:
    Modified:   analysis/2024-01-17_CheckYangsJunctionAnnotator.Rmd
    Modified:   analysis/2024-03-07_InvestigateTranslatedLongReads.Rmd
    Modified:   analysis/MakeFinalFigs_Fig2.Rmd
    Modified:   analysis/MakeFinalFigs_Fig4.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="intro" class="section level2">
<h2>Intro</h2>
<p>Similar to Lindeboom 2019, I see some groups of unproductive
junctions are more unstable than others as assessed by
(naRNA/steady-state) or (dKD/control)… Here I want to reproduce that
analysis, but also incorporate some significance filters to not consider
the juncs with fold changes that are not significant based on just a
couple counts.</p>
</div>
<div id="analysis" class="section level2">
<h2>Analysis</h2>
<pre class="r"><code>library(data.table)
library(tidyverse)</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>✔ ggplot2 3.3.6     ✔ purrr   0.3.4
✔ tibble  3.1.7     ✔ dplyr   1.0.9
✔ tidyr   1.2.0     ✔ stringr 1.4.0
✔ readr   2.1.2     ✔ forcats 0.5.1</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::between()   masks data.table::between()
✖ dplyr::filter()    masks stats::filter()
✖ dplyr::first()     masks data.table::first()
✖ dplyr::lag()       masks stats::lag()
✖ dplyr::last()      masks data.table::last()
✖ purrr::transpose() masks data.table::transpose()</code></pre>
<pre class="r"><code>library(edgeR)</code></pre>
<pre><code>Loading required package: limma</code></pre>
<pre class="r"><code>MostCommonContexts &lt;- read_tsv(&quot;../output/20240322_ResponseToReviewerMostCommonJuncContexts.tsv.gz&quot;) %&gt;%
    separate(Introns, into=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;strand&quot;), sep=&quot;_&quot;, convert=T, remove=F)</code></pre>
<pre><code>Rows: 28166 Columns: 8</code></pre>
<pre><code>── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (8): Introns, NewAnnotation, gene, symbol, SuperAnnotation, SemiSupergro...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>juncs.long &lt;- fread(&quot;../code/SplicingAnalysis/CombinedJuncTables/All.tsv.gz&quot;)

dat.KD &lt;- fread(&quot;/project2/yangili1/cfbuenabadn/ChromatinSplicingQTLs/code/SplicingAnalysis/CombinedJuncTables/NMD_KD.tsv.gz&quot;)

juncs.long &lt;- bind_rows(juncs.long, dat.KD)

juncs.long %&gt;%
    distinct(Dataset, IndID)</code></pre>
<pre><code>                 Dataset      IndID
  1: Expression.Splicing    HG00096
  2: Expression.Splicing    HG00097
  3: Expression.Splicing    HG00099
  4: Expression.Splicing    HG00100
  5: Expression.Splicing    HG00101
 ---                               
695:        HeLa.UPF1.KD SRR4081226
696:        HeLa.UPF1.KD SRR4081227
697:            HeLa.dKD SRR4081246
698:            HeLa.dKD SRR4081247
699:            HeLa.dKD SRR4081248</code></pre>
<p>To differential junction count analysis for NMD vs dKD… I will use
edgeR, just like a differential expression analysis… I think this makes
sense (as opposed to testing PSI or something like leafcutter does)
because I want to capture differences due to host transcript
degredation.</p>
<p>Let’s start with the NMD vs dKD contrast</p>
<pre class="r"><code>dat.mat &lt;- dat.KD %&gt;%
    # distinct(Dataset)
    filter(Dataset %in% c(&quot;HeLa.scr&quot;, &quot;HeLa.dKD&quot;)) %&gt;%
    unite(Introns, chrom, start, stop, strand, sep=&quot;_&quot;) %&gt;%
    unite(sample, Dataset, IndID, sep=&quot;_&quot;) %&gt;%
    dplyr::select(Introns, sample, Count) %&gt;%
    pivot_wider(names_from = &quot;sample&quot;, values_from=&quot;Count&quot;, values_fill=0) %&gt;%
    column_to_rownames(&quot;Introns&quot;) %&gt;%
    DGEList(group = relevel(factor(str_extract(colnames(.), &quot;^.+_&quot;)), &quot;HeLa.scr_&quot;))

keep &lt;- filterByExpr(dat.mat)
y &lt;- dat.mat[keep,] %&gt;%
    calcNormFactors() %&gt;%
    estimateDisp()</code></pre>
<pre><code>Using classic mode.</code></pre>
<pre class="r"><code>fit &lt;- y %&gt;%
    glmQLFit()
qlf &lt;- glmQLFTest(fit)
results &lt;- topTags(qlf,   n=Inf) %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(&quot;Intron&quot;)

results %&gt;%
    sample_n(10000) %&gt;%
ggplot(aes(x=logFC, y=-log10(PValue))) +
    geom_point(alpha=0.1)</code></pre>
<p><img src="figure/2024-04-18_RelativeNMDEfficiency.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>SidePoints &lt;- results %&gt;%
    filter(logFC &gt; 6 &amp; -log10(PValue)&gt;4 &amp; -log10(PValue)&lt;7) 

results %&gt;%
    sample_n(10000) %&gt;%
    mutate(SidePoints = Intron %in% SidePoints$Intron) %&gt;%
ggplot(aes(x=logFC, y=-log10(PValue), color=SidePoints)) +
    geom_point(alpha=0.1)</code></pre>
<p><img src="figure/2024-04-18_RelativeNMDEfficiency.Rmd/unnamed-chunk-1-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cpm &lt;- cpm(y, log=T, prior.count=0.001)

Mean.Junc.CPM &lt;- cpm %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(&quot;Intron&quot;) %&gt;%
    pivot_longer(names_to=&quot;sample&quot;, -Intron) %&gt;%
    mutate(group = if_else(str_detect(sample, &quot;dKD&quot;), &quot;dKD&quot;, &quot;control&quot;)) %&gt;%
    group_by(Intron, group) %&gt;%
    summarise(MeanJuncCPM = mean(value)) %&gt;%
    ungroup() %&gt;%
    distinct() </code></pre>
<pre><code>`summarise()` has grouped output by &#39;Intron&#39;. You can override using the
`.groups` argument.</code></pre>
<pre class="r"><code>Mean.Junc.CPM %&gt;%
    mutate(SidePoints = Intron %in% SidePoints$Intron) %&gt;%
    ggplot(aes(x=MeanJuncCPM)) +
    geom_histogram() +
    facet_grid(SidePoints~group, scales=&quot;free&quot;)</code></pre>
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="figure/2024-04-18_RelativeNMDEfficiency.Rmd/unnamed-chunk-1-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, so I have a differential junction expression analysis, and even
though the volcano plot looks a bit weird, I think it makes sense… The
streak of points off to the side are the junctions that are basically
non-existent in the control dataset but exist in the dKD dataset.</p>
<p>Now let’s just select for significant junctions, and then group them
by my annotations categories for different susceptibilities to NMD.</p>
<pre class="r"><code>results %&gt;%
    inner_join(
        MostCommonContexts %&gt;%
            mutate(Intron = str_glue(&quot;{chrom}_{start}_{stop-1}_{strand}&quot;))
    ) %&gt;%
    mutate(Signif = FDR&lt;0.1) %&gt;%
    ggplot(aes(x=logFC, color=ModeNMDFinder)) +
    stat_ecdf() +
    facet_wrap(~Signif, nrow=2) +
    coord_cartesian(xlim=c(-2,2))</code></pre>
<pre><code>Joining, by = &quot;Intron&quot;</code></pre>
<p><img src="figure/2024-04-18_RelativeNMDEfficiency.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>results %&gt;%
    inner_join(
        MostCommonContexts %&gt;%
            mutate(Intron = str_glue(&quot;{chrom}_{start}_{stop-1}_{strand}&quot;))
    ) %&gt;%
    mutate(Signif = FDR&lt;0.1) %&gt;%
    ggplot(aes(x=logFC, color=ModeNMDFinder)) +
    stat_ecdf() +
    coord_cartesian(xlim=c(-2,2)) +
    theme_bw()</code></pre>
<pre><code>Joining, by = &quot;Intron&quot;</code></pre>
<p><img src="figure/2024-04-18_RelativeNMDEfficiency.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>results %&gt;%
    filter(logCPM&gt;0) %&gt;%
    inner_join(
        MostCommonContexts %&gt;%
            mutate(Intron = str_glue(&quot;{chrom}_{start}_{stop-1}_{strand}&quot;))
    ) %&gt;%
    mutate(Signif = FDR&lt;0.1) %&gt;%
    ggplot(aes(x=logFC, color=ModeNMDFinder)) +
    stat_ecdf() +
    coord_cartesian(xlim=c(-2,2)) +
    theme_bw()</code></pre>
<pre><code>Joining, by = &quot;Intron&quot;</code></pre>
<p><img src="figure/2024-04-18_RelativeNMDEfficiency.Rmd/unnamed-chunk-2-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Mean.Junc.CPM %&gt;%
    pivot_wider(names_from = &quot;group&quot;, values_from = &quot;MeanJuncCPM&quot;) %&gt;%
    inner_join(
        MostCommonContexts %&gt;%
            mutate(Intron = str_glue(&quot;{chrom}_{start}_{stop-1}_{strand}&quot;))
    ) %&gt;%
    ggplot(aes(x=dKD, y=control)) +
        geom_point(alpha=0.1) +
        geom_abline(slope=1, intercept=0, color=&#39;#760d0d&#39;) +
        geom_text( data = . %&gt;%
            group_by(ModeNMDFinder) %&gt;%
                summarise(med = str_glue(round(2**median(dKD-control),3 )), n=n()) %&gt;%
                ungroup() %&gt;%
                mutate(label = str_glue(&quot;med={med}\nn={n}&quot;)),
            aes(label=label),
            x=-Inf, y=Inf, hjust=0, vjust=1
            ) +
        facet_wrap(~ModeNMDFinder) +
        theme_bw()</code></pre>
<pre><code>Joining, by = &quot;Intron&quot;</code></pre>
<p><img src="figure/2024-04-18_RelativeNMDEfficiency.Rmd/unnamed-chunk-2-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Mean.Junc.CPM %&gt;%
    pivot_wider(names_from = &quot;group&quot;, values_from = &quot;MeanJuncCPM&quot;) %&gt;%
    inner_join(
        MostCommonContexts %&gt;%
            mutate(Intron = str_glue(&quot;{chrom}_{start}_{stop-1}_{strand}&quot;))) %&gt;%
    group_by(ModeNMDFinder) %&gt;%
    summarise(med = round(2**median(dKD-control), 3)) %&gt;%
    ungroup()</code></pre>
<pre><code>Joining, by = &quot;Intron&quot;</code></pre>
<pre><code># A tibble: 7 × 2
  ModeNMDFinder    med
  &lt;chr&gt;          &lt;dbl&gt;
1 50 nt rule     0.728
2 Last exon      0.718
3 Long exon      1.04 
4 No CDS         0.833
5 No stop        0.797
6 Start proximal 0.789
7 Trigger NMD    1.06 </code></pre>
<pre class="r"><code>results %&gt;%
    inner_join(
        MostCommonContexts %&gt;%
            mutate(Intron = str_glue(&quot;{chrom}_{start}_{stop-1}_{strand}&quot;))
    ) %&gt;%
    mutate(Signif = FDR&lt;0.1) %&gt;%
    ggplot(aes(x=logFC, y=-log10(FDR), color=Signif)) +
    geom_point(alpha=0.2) +
    geom_vline(data = . %&gt;%
        group_by(Signif, ModeNMDFinder) %&gt;%
        summarise(med = median(logFC)) %&gt;%
        ungroup(),
        aes(xintercept = med, color=Signif)
        ) +
    coord_cartesian(xlim=c(-2,2)) +
    theme_bw() +
    facet_wrap(~ModeNMDFinder) +
    labs(x=&quot;logFC (dKD/control)&quot;, caption=&quot;vertical lines is median affect by group color\nSignf means FDR&lt;10%&quot;)</code></pre>
<pre><code>Joining, by = &quot;Intron&quot;
`summarise()` has grouped output by &#39;Signif&#39;. You can override using the
`.groups` argument.</code></pre>
<p><img src="figure/2024-04-18_RelativeNMDEfficiency.Rmd/unnamed-chunk-2-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>I want to next check out the scatter plot of junc RPM but while
distinguishing between the junctions that we classified as productive or
unproductive in the nat gen manuscript.</p>
<pre class="r"><code>Mean.Junc.CPM %&gt;%
    pivot_wider(names_from = &quot;group&quot;, values_from = &quot;MeanJuncCPM&quot;) %&gt;%
    inner_join(
        MostCommonContexts %&gt;%
            mutate(Intron = str_glue(&quot;{chrom}_{start}_{stop-1}_{strand}&quot;))
    ) %&gt;%
  mutate(ProductiveOrUnproductive = case_when(
    SuperAnnotation %in% c(&quot;AnnotatedJunc_ProductiveCodingGene&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;) ~ &quot;Productive&quot;,
    SuperAnnotation %in% c(&quot;UnannotatedJunc_UnproductiveCodingGene&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;) ~ &quot;Unproductive&quot;,
    TRUE ~ NA_character_
  )) %&gt;%
  mutate(ProductiveOrUnproductive = SuperAnnotation) %&gt;%
  filter(!str_detect(SuperAnnotation, &quot;Noncoding&quot;)) %&gt;%
    ggplot(aes(x=dKD, y=control, color=ProductiveOrUnproductive)) +
        geom_point(alpha=0.1) +
        geom_abline(slope=1, intercept=0, color=&#39;#760d0d&#39;) +
        geom_text( data = . %&gt;%
            group_by(ModeNMDFinder, ProductiveOrUnproductive) %&gt;%
                summarise(med = str_glue(round(2**median(dKD-control),3 )), n=n()) %&gt;%
                ungroup() %&gt;%
                mutate(label = str_glue(&quot;med={med}; n={n}&quot;)) %&gt;%
              group_by(ModeNMDFinder) %&gt;%
              mutate(vjust = row_number()) %&gt;%
              ungroup(),
            aes(label=label, vjust=vjust),
            x=-Inf, y=Inf, hjust=0
            ) +
        scale_color_manual(values=c(&quot;AnnotatedJunc_ProductiveCodingGene&quot;=&quot;#1f78b4&quot;, &quot;UnannotatedJunc_ProductiveCodingGene&quot;=&quot;#a6cee3&quot;, &quot;UnannotatedJunc_UnproductiveCodingGene&quot;=&quot;#fb9a99&quot;, &quot;AnnotatedJunc_UnproductiveCodingGene&quot;=&quot;#e31a1c&quot;)) +
        facet_wrap(~ModeNMDFinder) +
        theme_bw()</code></pre>
<pre><code>Joining, by = &quot;Intron&quot;
`summarise()` has grouped output by &#39;ModeNMDFinder&#39;. You can override using the
`.groups` argument.</code></pre>
<p><img src="figure/2024-04-18_RelativeNMDEfficiency.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The presence of many Gencode annotated productive juncs in the
“Trigger NMD” category that are also unaffected by dKD, highlights that
many of these junction classifications are wrong. This could explain
some of the variance between categories - bad classification. The reason
for bad classification may be that combination of low read count noise
and real biology in long read dataset leads to presence of some genes
for which the most common context is “Trigger NMD”, result in all of
those introns being classified as Trigger NMD in my methodology.</p>
<pre class="r"><code>IntronAnnotations &lt;- read_tsv(&quot;../data/IntronAnnotationsFromYang.Updated.tsv.gz&quot;)</code></pre>
<pre><code>Rows: 458713 Columns: 10
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (8): chrom, strand, NewAnnotation, gene, symbol, SuperAnnotation, SemiSu...
dbl (2): start, end

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>juncs.long.summarised &lt;- juncs.long %&gt;%
    filter(Dataset %in% c(&quot;chRNA.Expression.Splicing&quot;, &quot;Expression.Splicing&quot;,&quot;HeLa.dKD&quot;, &quot;HeLa.scr&quot;, &quot;MetabolicLabelled.30min&quot;, &quot;MetabolicLabelled.60min&quot;)) %&gt;%
    mutate(Dataset = if_else(str_detect(Dataset, &quot;Metabolic&quot;), &quot;Metabolic&quot;, Dataset)) %&gt;%
    group_by(chrom, start, stop, strand, Dataset) %&gt;%
    summarise(count = sum(Count)) %&gt;%
    ungroup() %&gt;%
    distinct()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;chrom&#39;, &#39;start&#39;, &#39;stop&#39;, &#39;strand&#39;. You can
override using the `.groups` argument.</code></pre>
<pre class="r"><code>juncs.long.summarised %&gt;%
  distinct(Dataset)</code></pre>
<pre><code># A tibble: 5 × 1
  Dataset                  
  &lt;chr&gt;                    
1 chRNA.Expression.Splicing
2 HeLa.scr                 
3 Expression.Splicing      
4 Metabolic                
5 HeLa.dKD                 </code></pre>
<pre class="r"><code>JuncCountMin &lt;- 100

juncs.summarised.Ratios &lt;- bind_rows(
  full_join(
    juncs.long.summarised %&gt;%
      filter(Dataset == &quot;chRNA.Expression.Splicing&quot;) %&gt;%
      dplyr::select(-Dataset),
    juncs.long.summarised %&gt;%
      filter(Dataset == &quot;Expression.Splicing&quot;)  %&gt;%
      dplyr::select(-Dataset),
    by=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;strand&quot;),
    suffix=c(&quot;.num&quot;, &quot;.den&quot;)) %&gt;%
    mutate(Comparison = &quot;chRNA/steadyState&quot;),
  full_join(
    juncs.long.summarised %&gt;%
      filter(Dataset == &quot;HeLa.dKD&quot;) %&gt;%
      dplyr::select(-Dataset),
    juncs.long.summarised %&gt;%
      filter(Dataset == &quot;HeLa.scr&quot;)  %&gt;%
      dplyr::select(-Dataset),
    by=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;strand&quot;),
    suffix=c(&quot;.num&quot;, &quot;.den&quot;)) %&gt;%
    mutate(Comparison = &quot;dKD/control&quot;),
  full_join(
    juncs.long.summarised %&gt;%
      filter(Dataset == &quot;Metabolic&quot;) %&gt;%
      dplyr::select(-Dataset),
    juncs.long.summarised %&gt;%
      filter(Dataset == &quot;Expression.Splicing&quot;)  %&gt;%
      dplyr::select(-Dataset),
    by=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;strand&quot;),
    suffix=c(&quot;.num&quot;, &quot;.den&quot;)) %&gt;%
    mutate(Comparison = &quot;Metabolic/SteadyState&quot;)
  ) %&gt;%
  replace_na(list(count.den = 0, count.num=0)) %&gt;%
  group_by(Comparison) %&gt;%
    mutate(count.num.sum = sum(count.num), count.den.sum=sum(count.den)) %&gt;%
    ungroup() %&gt;%
    mutate(Ratio = (count.num/count.num.sum)/(count.den/count.den.sum)) %&gt;%
  drop_na()

juncs.summarised.Ratios %&gt;%
  filter(count.num &gt;= JuncCountMin | count.den &gt;= JuncCountMin) %&gt;%
  pivot_wider(names_from=&quot;Comparison&quot;, values_from=&quot;Ratio&quot;, id_cols = c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;strand&quot;)) %&gt;%
  inner_join(IntronAnnotations) %&gt;%
  ggplot(aes(x=`chRNA/steadyState`, y=`dKD/control`)) +
  geom_hex(bins=100) +
  # geom_point(alpha=0.1) +
  scale_fill_viridis_c() +
  scale_y_continuous(trans=&#39;log10&#39;) +
  scale_x_continuous(trans=&#39;log10&#39;) +
  theme_bw() +
  facet_wrap(~SuperAnnotation) +
  geom_vline(color=&#39;red&#39;, xintercept = 1, linetype=&#39;dashed&#39;) +
  geom_hline(color=&#39;red&#39;, yintercept = 1, linetype=&#39;dashed&#39;) +
  coord_fixed() +
  labs(title=&quot;correlation of junctionRPM ratios&quot;, caption=&quot;Only included juncs with &gt;100 reads in at least one of the datasets&quot;)</code></pre>
<pre><code>Joining, by = c(&quot;chrom&quot;, &quot;start&quot;, &quot;strand&quot;)</code></pre>
<pre><code>Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
<pre><code>Warning: Transformation introduced infinite values in continuous x-axis</code></pre>
<pre><code>Warning: Removed 320134 rows containing non-finite values (stat_binhex).</code></pre>
<p><img src="figure/2024-04-18_RelativeNMDEfficiency.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>juncs.summarised.Ratios %&gt;%
  filter(count.num &gt;= JuncCountMin | count.den &gt;= JuncCountMin) %&gt;%
  pivot_wider(names_from=&quot;Comparison&quot;, values_from=&quot;Ratio&quot;, id_cols = c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;strand&quot;)) %&gt;%
  mutate(end=stop+1) %&gt;% dplyr::select(-stop) %&gt;%
  inner_join(IntronAnnotations) %&gt;%
  ggplot(aes(x=`chRNA/steadyState`, y=`Metabolic/SteadyState`)) +
  geom_hex(bins=100) +
  # geom_point(alpha=0.1) +
  scale_fill_viridis_c() +
  scale_y_continuous(trans=&#39;log10&#39;) +
  scale_x_continuous(trans=&#39;log10&#39;) +
  theme_bw() +
  facet_wrap(~SuperAnnotation) +
  geom_vline(color=&#39;red&#39;, xintercept = 1, linetype=&#39;dashed&#39;) +
  geom_hline(color=&#39;red&#39;, yintercept = 1, linetype=&#39;dashed&#39;) +
  coord_fixed() +
  labs(title=&quot;correlation of junctionRPM ratios&quot;, caption=&quot;Only included juncs with &gt;100 reads in at least one of the datasets&quot;)</code></pre>
<pre><code>Joining, by = c(&quot;chrom&quot;, &quot;start&quot;, &quot;strand&quot;, &quot;end&quot;)</code></pre>
<pre><code>Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
<pre><code>Warning: Transformation introduced infinite values in continuous x-axis</code></pre>
<pre><code>Warning: Removed 33661 rows containing non-finite values (stat_binhex).</code></pre>
<p><img src="figure/2024-04-18_RelativeNMDEfficiency.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>juncs.summarised.Ratios %&gt;%
  filter(count.num &gt;= JuncCountMin | count.den &gt;= JuncCountMin) %&gt;%
  pivot_wider(names_from=&quot;Comparison&quot;, values_from=&quot;Ratio&quot;, id_cols = c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;strand&quot;)) %&gt;%
  mutate(end=stop+1) %&gt;% dplyr::select(-stop) %&gt;%
  inner_join(IntronAnnotations) %&gt;% 
  ggplot(aes(x=`Metabolic/SteadyState`, y=`dKD/control`)) +
  geom_hex(bins=100) +
  # geom_point(alpha=0.1) +
  scale_fill_viridis_c() +
  scale_y_continuous(trans=&#39;log10&#39;) +
  scale_x_continuous(trans=&#39;log10&#39;) +
  theme_bw() +
  facet_wrap(~SuperAnnotation) +
  geom_vline(color=&#39;red&#39;, xintercept = 1, linetype=&#39;dashed&#39;) +
  geom_hline(color=&#39;red&#39;, yintercept = 1, linetype=&#39;dashed&#39;) +
  coord_fixed() +
  labs(title=&quot;correlation of junctionRPM ratios&quot;, caption=&quot;Only included juncs with &gt;100 reads in at least one of the datasets&quot;)</code></pre>
<pre><code>Joining, by = c(&quot;chrom&quot;, &quot;start&quot;, &quot;strand&quot;, &quot;end&quot;)</code></pre>
<pre><code>Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
<pre><code>Warning: Transformation introduced infinite values in continuous x-axis</code></pre>
<pre><code>Warning: Removed 127074 rows containing non-finite values (stat_binhex).</code></pre>
<p><img src="figure/2024-04-18_RelativeNMDEfficiency.Rmd/unnamed-chunk-4-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>juncs.summarised.Ratios %&gt;%
  filter(count.num &gt;= JuncCountMin | count.den &gt;= JuncCountMin) %&gt;%
  pivot_wider(names_from=&quot;Comparison&quot;, values_from=&quot;Ratio&quot;, id_cols = c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;strand&quot;)) %&gt;%
  dplyr::select(5:7) %&gt;% cor(method=&#39;s&#39;, use=&#39;pairwise.complete.obs&#39;) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;Comparison&quot;) %&gt;%
  pivot_longer(-c(&quot;Comparison&quot;)) %&gt;%
  ggplot(aes(x=Comparison, y=name, fill=value)) +
  geom_raster() +
  geom_text(aes(label=round(value, 2)), color=&#39;red&#39;) +
  scale_fill_viridis_c() +
  labs(x=NULL, y=NULL, title=&quot;spearman cor coef&quot;)</code></pre>
<p><img src="figure/2024-04-18_RelativeNMDEfficiency.Rmd/unnamed-chunk-4-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>It doesn’t make much sesne to me that there is decent correlation in
the logFC in naRNA/steadyState vs dKD/control even for the junctions in
non-coding. Maybe I should also make these plots with PSI or something
like that (which uses a local junction count for the denominator) to
corroborate these findings.</p>
<pre class="r"><code># juncs.summarised.Ratios %&gt;%
#   mutate(end=stop+1) %&gt;% dplyr::select(-stop) %&gt;%
#   inner_join(IntronAnnotations) %&gt;%
#   filter(!str_detect(gene, &quot;,&quot;)) %&gt;%
#   group_by(Comparison, gene) %&gt;%
#   mutate(count.num.PSInum = )


juncs.summarised.Ratios.NormalizedToGeneJuncs &lt;- juncs.summarised.Ratios %&gt;%
  mutate(end=stop+1) %&gt;% dplyr::select(-stop) %&gt;%
  inner_join(IntronAnnotations) %&gt;%
  filter(!str_detect(gene, &quot;,&quot;)) %&gt;%
  group_by(gene, Comparison) %&gt;%
  mutate(count.num.sum = sum(count.num), count.den.sum=sum(count.den)) %&gt;%
    ungroup() %&gt;%
    mutate(Ratio = (count.num/count.num.sum)/(count.den/count.den.sum)) %&gt;%
  drop_na() %&gt;%
  filter(count.num &gt;= JuncCountMin | count.den &gt;= JuncCountMin)</code></pre>
<pre><code>Joining, by = c(&quot;chrom&quot;, &quot;start&quot;, &quot;strand&quot;, &quot;end&quot;)</code></pre>
<pre class="r"><code>juncs.summarised.Ratios.NormalizedToGeneJuncs %&gt;%
  pivot_wider(names_from=&quot;Comparison&quot;, values_from=&quot;Ratio&quot;, id_cols = c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;strand&quot;)) %&gt;%
  inner_join(IntronAnnotations) %&gt;%
  ggplot(aes(x=`chRNA/steadyState`, y=`dKD/control`)) +
  geom_hex(bins=100) +
  # geom_point(alpha=0.1) +
  scale_fill_viridis_c() +
  scale_y_continuous(trans=&#39;log10&#39;) +
  scale_x_continuous(trans=&#39;log10&#39;) +
  theme_bw() +
  facet_wrap(~SuperAnnotation) +
  geom_vline(color=&#39;red&#39;, xintercept = 1, linetype=&#39;dashed&#39;) +
  geom_hline(color=&#39;red&#39;, yintercept = 1, linetype=&#39;dashed&#39;) +
  coord_fixed() +
  labs(title=&quot;correlation of junction/GenewiseSum(junction) ratios&quot;, caption=&quot;Only included juncs with &gt;100 reads in at least one of the datasets&quot;)</code></pre>
<pre><code>Joining, by = c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;strand&quot;)</code></pre>
<pre><code>Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
<pre><code>Warning: Transformation introduced infinite values in continuous x-axis</code></pre>
<pre><code>Warning: Removed 125808 rows containing non-finite values (stat_binhex).</code></pre>
<p><img src="figure/2024-04-18_RelativeNMDEfficiency.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>juncs.summarised.Ratios.NormalizedToGeneJuncs %&gt;%
  pivot_wider(names_from=&quot;Comparison&quot;, values_from=&quot;Ratio&quot;, id_cols = c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;strand&quot;)) %&gt;%
  inner_join(IntronAnnotations) %&gt;%
  ggplot(aes(x=`chRNA/steadyState`, y=`Metabolic/SteadyState`)) +
  geom_hex(bins=100) +
  # geom_point(alpha=0.1) +
  scale_fill_viridis_c() +
  scale_y_continuous(trans=&#39;log10&#39;) +
  scale_x_continuous(trans=&#39;log10&#39;) +
  theme_bw() +
  facet_wrap(~SuperAnnotation) +
  geom_vline(color=&#39;red&#39;, xintercept = 1, linetype=&#39;dashed&#39;) +
  geom_hline(color=&#39;red&#39;, yintercept = 1, linetype=&#39;dashed&#39;) +
  coord_fixed() +
  labs(title=&quot;correlation of junction/GenewiseSum(junction) ratios&quot;, caption=&quot;Only included juncs with &gt;100 reads in at least one of the datasets&quot;)</code></pre>
<pre><code>Joining, by = c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;strand&quot;)</code></pre>
<pre><code>Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
<pre><code>Warning: Transformation introduced infinite values in continuous x-axis</code></pre>
<pre><code>Warning: Removed 33616 rows containing non-finite values (stat_binhex).</code></pre>
<p><img src="figure/2024-04-18_RelativeNMDEfficiency.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>juncs.summarised.Ratios.NormalizedToGeneJuncs %&gt;%
  pivot_wider(names_from=&quot;Comparison&quot;, values_from=&quot;Ratio&quot;, id_cols = c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;strand&quot;)) %&gt;%
  inner_join(IntronAnnotations) %&gt;%
  ggplot(aes(x=`Metabolic/SteadyState`, y=`dKD/control`)) +
  geom_hex(bins=100) +
  # geom_point(alpha=0.1) +
  scale_fill_viridis_c() +
  scale_y_continuous(trans=&#39;log10&#39;) +
  scale_x_continuous(trans=&#39;log10&#39;) +
  theme_bw() +
  facet_wrap(~SuperAnnotation) +
  geom_vline(color=&#39;red&#39;, xintercept = 1, linetype=&#39;dashed&#39;) +
  geom_hline(color=&#39;red&#39;, yintercept = 1, linetype=&#39;dashed&#39;) +
  coord_fixed() +
  labs(title=&quot;correlation of junction/GenewiseSum(junction) ratios&quot;, caption=&quot;Only included juncs with &gt;100 reads in at least one of the datasets&quot;)</code></pre>
<pre><code>Joining, by = c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;strand&quot;)</code></pre>
<pre><code>Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
<pre><code>Warning: Transformation introduced infinite values in continuous x-axis</code></pre>
<pre><code>Warning: Removed 127029 rows containing non-finite values (stat_binhex).</code></pre>
<p><img src="figure/2024-04-18_RelativeNMDEfficiency.Rmd/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>juncs.summarised.Ratios.NormalizedToGeneJuncs %&gt;%
  pivot_wider(names_from=&quot;Comparison&quot;, values_from=&quot;Ratio&quot;, id_cols = c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;strand&quot;)) %&gt;%
  dplyr::select(5:7) %&gt;% cor(method=&#39;s&#39;, use=&#39;pairwise.complete.obs&#39;) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;Comparison&quot;) %&gt;%
  pivot_longer(-c(&quot;Comparison&quot;)) %&gt;%
  ggplot(aes(x=Comparison, y=name, fill=value)) +
  geom_raster() +
  geom_text(aes(label=round(value, 2)), color=&#39;red&#39;) +
  scale_fill_viridis_c() +
  labs(x=NULL, y=NULL, title=&quot;spearman cor coef&quot;)</code></pre>
<p><img src="figure/2024-04-18_RelativeNMDEfficiency.Rmd/unnamed-chunk-5-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>juncs.summarised.Ratios.NormalizedToGeneJuncs %&gt;%
  pivot_wider(names_from=&quot;Comparison&quot;, values_from=&quot;Ratio&quot;, id_cols = c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;strand&quot;)) %&gt;%
  inner_join(IntronAnnotations) %&gt;%
  group_by(SuperAnnotation) %&gt;%
  summarise(cor = cor(`chRNA/steadyState`, `dKD/control`, method=&#39;s&#39;, use=&#39;pairwise.complete.obs&#39;))</code></pre>
<pre><code>Joining, by = c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;strand&quot;)</code></pre>
<pre><code># A tibble: 6 × 2
  SuperAnnotation                            cor
  &lt;chr&gt;                                    &lt;dbl&gt;
1 AnnotatedJunc_NoncodingGene             0.0547
2 AnnotatedJunc_ProductiveCodingGene     -0.0471
3 AnnotatedJunc_UnproductiveCodingGene    0.358 
4 UnannotatedJunc_NoncodingJunc           0.187 
5 UnannotatedJunc_ProductiveCodingGene    0.0274
6 UnannotatedJunc_UnproductiveCodingGene  0.325 </code></pre>
<p>Ok that is nice, this makes me feel a bit more confident that
naRNA/steadyState is a decent proxy for dKD/control for unproductive
junctions: It’s the same plot that I shared a few minutes ago but rather
than junctionRPM (that is, junction / sum(AllJunctions), i used the
junction/sum(AllJunctionsWithinTheSameGene)… The previous correlation in
the Noncoding facets and the productive facets was probably just
reflecting some similar differential expression effects that are
(somewhat suprisingly) correlated… While that largely goes away in the
Noncoding facets in this version, the correlation in the unproductive
facet remains.</p>
<p>Now let’s see if I can better classify the unproductive junctions
from the long read data…</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] edgeR_3.38.4      limma_3.52.4      forcats_0.5.1     stringr_1.4.0    
 [5] dplyr_1.0.9       purrr_0.3.4       readr_2.1.2       tidyr_1.2.0      
 [9] tibble_3.1.7      ggplot2_3.3.6     tidyverse_1.3.1   data.table_1.14.2

loaded via a namespace (and not attached):
 [1] httr_1.4.3        sass_0.4.1        viridisLite_0.4.0 bit64_4.0.5      
 [5] vroom_1.5.7       jsonlite_1.8.0    splines_4.2.0     R.utils_2.11.0   
 [9] modelr_0.1.8      bslib_0.3.1       assertthat_0.2.1  highr_0.9        
[13] cellranger_1.1.0  yaml_2.3.5        pillar_1.7.0      backports_1.4.1  
[17] lattice_0.20-45   glue_1.6.2        digest_0.6.29     promises_1.2.0.1 
[21] rvest_1.0.2       colorspace_2.0-3  htmltools_0.5.2   httpuv_1.6.5     
[25] R.oo_1.24.0       pkgconfig_2.0.3   broom_0.8.0       haven_2.5.0      
[29] scales_1.2.0      later_1.3.0       tzdb_0.3.0        git2r_0.30.1     
[33] farver_2.1.0      generics_0.1.2    ellipsis_0.3.2    withr_2.5.0      
[37] hexbin_1.28.3     cli_3.3.0         magrittr_2.0.3    crayon_1.5.1     
[41] readxl_1.4.0      evaluate_0.15     R.methodsS3_1.8.1 fs_1.5.2         
[45] fansi_1.0.3       xml2_1.3.3        tools_4.2.0       hms_1.1.1        
[49] lifecycle_1.0.1   munsell_0.5.0     reprex_2.0.1      locfit_1.5-9.7   
[53] compiler_4.2.0    jquerylib_0.1.4   rlang_1.0.2       grid_4.2.0       
[57] rstudioapi_0.13   labeling_0.4.2    rmarkdown_2.14    gtable_0.3.0     
[61] DBI_1.1.2         R6_2.5.1          lubridate_1.8.0   knitr_1.39       
[65] fastmap_1.1.0     bit_4.0.4         utf8_1.2.2        workflowr_1.7.0  
[69] rprojroot_2.0.3   stringi_1.7.6     parallel_4.2.0    Rcpp_1.0.8.3     
[73] vctrs_0.4.1       dbplyr_2.1.1      tidyselect_1.1.2  xfun_0.30        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
