<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2024-04-26" />

<title>2024-04-26_eQTL_subtypePlieotropy</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">2024-04-26_eQTL_subtypePlieotropy</h1>
<h4 class="date">2024-04-26</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-04-29
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>ChromatinSplicingQTLs/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstree209c2d5e42d8949e4919e540f4cfed87ff4e0cbatargetblank209c2d5a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/209c2d5e42d8949e4919e540f4cfed87ff4e0cba" target="_blank">209c2d5</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstree209c2d5e42d8949e4919e540f4cfed87ff4e0cbatargetblank209c2d5a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/209c2d5e42d8949e4919e540f4cfed87ff4e0cba" target="_blank">209c2d5</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/figure/
    Ignored:    code/.DS_Store
    Ignored:    code/.RData
    Ignored:    code/._report.html
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    code/.snakemake/
    Ignored:    code/APA_Processing/
    Ignored:    code/Alignments/
    Ignored:    code/ChromHMM/
    Ignored:    code/ENCODE/
    Ignored:    code/ExpressionAnalysis/
    Ignored:    code/ExtractPhenotypeBedByGenotype.py
    Ignored:    code/FastqFastp/
    Ignored:    code/FastqFastpSE/
    Ignored:    code/FastqSE/
    Ignored:    code/FineMapping/
    Ignored:    code/GTEx/
    Ignored:    code/Gencode.v34.6Colors.bed.gz
    Ignored:    code/Genotypes/
    Ignored:    code/H3K36me3_CutAndTag.pdf
    Ignored:    code/IntronSlopes/
    Ignored:    code/LR.bed
    Ignored:    code/LR.seq.bed
    Ignored:    code/LongReads/
    Ignored:    code/MYB.tracks.ini
    Ignored:    code/Metaplots/
    Ignored:    code/Misc/
    Ignored:    code/MiscCountTables/
    Ignored:    code/Multiqc/
    Ignored:    code/Multiqc_chRNA/
    Ignored:    code/NonCodingRNA/
    Ignored:    code/NonCodingRNA_annotation/
    Ignored:    code/PairwisePi1Traits.P.all.txt.gz
    Ignored:    code/PeakCalling/
    Ignored:    code/Phenotypes/
    Ignored:    code/PlotGruberQTLs/
    Ignored:    code/PlotQTLs/
    Ignored:    code/ProCapAnalysis/
    Ignored:    code/QC/
    Ignored:    code/QTL_SNP_Enrichment/
    Ignored:    code/QTLs/
    Ignored:    code/RPKM_tables/
    Ignored:    code/ReadLengthMapExperiment/
    Ignored:    code/ReadLengthMapExperimentResults/
    Ignored:    code/ReadLengthMapExperimentSpliceCounts/
    Ignored:    code/ReferenceGenome/
    Ignored:    code/Rplots.pdf
    Ignored:    code/Session.vim
    Ignored:    code/SmallMolecule/
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TODO
    Ignored:    code/Tehranchi/
    Ignored:    code/alias/
    Ignored:    code/bigwigs/
    Ignored:    code/bigwigs_FromNonWASPFilteredReads/
    Ignored:    code/config/.DS_Store
    Ignored:    code/config/._.DS_Store
    Ignored:    code/config/.ipynb_checkpoints/
    Ignored:    code/config/config.local.yaml
    Ignored:    code/dag.pdf
    Ignored:    code/dag.png
    Ignored:    code/dag.svg
    Ignored:    code/data/
    Ignored:    code/debug.ipynb
    Ignored:    code/debug_python.ipynb
    Ignored:    code/deepTools/
    Ignored:    code/featureCounts/
    Ignored:    code/featureCountsBasicGtf/
    Ignored:    code/genome_config.yaml
    Ignored:    code/gwas_summary_stats/
    Ignored:    code/hyprcoloc/
    Ignored:    code/igv_session.xml
    Ignored:    code/isoseqbams/
    Ignored:    code/log
    Ignored:    code/logs/
    Ignored:    code/notebooks/.ipynb_checkpoints/
    Ignored:    code/pi1/
    Ignored:    code/polyA.Splicing.Subset_YRI.NominalPassForColoc.bed.bgz
    Ignored:    code/rules/.ipynb_checkpoints/
    Ignored:    code/rules/OldRules/
    Ignored:    code/rules/notebooks/
    Ignored:    code/salmontest/
    Ignored:    code/scratch/
    Ignored:    code/scripts/.ipynb_checkpoints/
    Ignored:    code/scripts/GTFtools_0.8.0/
    Ignored:    code/scripts/__pycache__/
    Ignored:    code/scripts/liftOverBedpe/liftOverBedpe.py
    Ignored:    code/snakemake.dryrun.log
    Ignored:    code/snakemake.log
    Ignored:    code/snakemake.sbatch.log
    Ignored:    code/snakemake_profiles/slurm/__pycache__/
    Ignored:    code/test.introns.bed
    Ignored:    code/test.introns2.bed
    Ignored:    code/test.log
    Ignored:    code/tracks.xml
    Ignored:    data/.DS_Store
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-10.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-11.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-2.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-3.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-4.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-5.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-6.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-7.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-8.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022.csv
    Ignored:    data/Metaplots/.DS_Store
    Ignored:    output/~$20240223_NumQTLs_edited.xlsx

Untracked files:
    Untracked:  analysis/2024-04-26_eQTL_subtypePlieotropy.Rmd
    Untracked:  code/scripts/Tidy_GTEx_SummaryStats_AllCisWindowGenes.py

Unstaged changes:
    Modified:   analysis/2024-04-18_RelativeNMDEfficiency.Rmd
    Modified:   code/rules/GTEx.smk
    Modified:   code/scripts/GenometracksByGenotype
    Modified:   code/scripts/Tidy_GTEx_SummaryStats.py

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="pleiotropy-of-u-sqtlseqtls-vs-hqtleqtls"
class="section level3">
<h3>pleiotropy of u-sQTLs/eQTLs vs hQTL/eQTLs</h3>
<p>I predict that u-sQTL/eQTLs will be eQTLs less genes than
hQTL/eQTLs…</p>
<blockquote>
<p>About that horizontal pleiotropy of transcription based eQTLs vs
splicing based eQTLs… Here I used the same red/purple groups of eQTL
SNPs that I used for the manuscript where we looked at effect sizes
across GTEx tissues. In this plot I looked for associations in GTEx LCL
data, and counted the number of protein coding genes (100kb window from
the SNP, among the 14000 genes we considered for many of our other
analyses throughout the paper) with a nominally significant association
(P&lt;0.01) with the SNP.</p>
</blockquote>
<pre class="r"><code>library(tidyverse)
GTEx.dat &lt;- read_tsv(&quot;../code/scratch/Tidy.SummaryStats.eQTLsQTLhQTL_AllGenes.tsv.gz&quot;)

GTEx.dat %&gt;% distinct(sQTL_or_hQTL, BensClassification)</code></pre>
<pre><code># A tibble: 3 × 2
  sQTL_or_hQTL BensClassification
  &lt;chr&gt;        &lt;chr&gt;             
1 sQTL         sQTL              
2 hQTL         hQTL_Distal       
3 hQTL         hQTL_Proximal     </code></pre>
<pre class="r"><code>GTEx.dat %&gt;%
  group_by(TopSNP, sQTL_or_hQTL, gene) %&gt;%
  summarise(IsSigInAnyTissue = any(eQTL_nomP &lt; 0.01)) %&gt;%
  ungroup() %&gt;%
  count(TopSNP, sQTL_or_hQTL) %&gt;%
  ggplot(aes(x=n, color=sQTL_or_hQTL)) +
  stat_ecdf()</code></pre>
<p><img src="figure/2024-04-26_eQTL_subtypePlieotropy.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>GTEx.dat %&gt;%
  distinct(tissue)</code></pre>
<pre><code># A tibble: 54 × 1
   tissue                     
   &lt;chr&gt;                      
 1 ovary                      
 2 brain_cerebellar_hemisphere
 3 nerve_tibial               
 4 breast_mammary_tissue      
 5 testis                     
 6 colon_transverse           
 7 fallopian_tube             
 8 brain_caudate_basal_ganglia
 9 lung                       
10 esophagus_muscularis       
# … with 44 more rows</code></pre>
<pre class="r"><code>GTEx.dat %&gt;%
  filter(tissue == &quot;cells_ebv-transformed_lymphocytes&quot;) %&gt;%
  # mutate(sQTL_or_hQTL = BensClassification) %&gt;%
  count(nom_pval &lt; 0.01, TopSNP, sQTL_or_hQTL) %&gt;%
  filter(`nom_pval &lt; 0.01`) %&gt;%
  add_count(sQTL_or_hQTL) %&gt;%
  mutate(sQTL_or_hQTL = recode(sQTL_or_hQTL, &quot;sQTL&quot;=&quot;u-sQTL/eQTL&quot;, &quot;hQTL&quot;=&quot;hQTL/eQTL&quot;)) %&gt;%
  mutate(sQTL_or_hQTL = str_glue(&quot;{sQTL_or_hQTL}; n={nn}&quot;)) %&gt;%
  ggplot(aes(x=n, color=sQTL_or_hQTL)) +
  stat_ecdf() +
  scale_color_manual(values=c(&quot;#e31a1c&quot;, &quot;#6a3d9a&quot;), name=NULL) +
  coord_cartesian(xlim=c(1,10)) +
  scale_x_continuous(breaks=seq(0, 10, 2)) +
  labs(y=&quot;ecdf&quot;, x=&quot;horizontal pleiotropy\n[Number eGenes per SNP\n in validation dataset]&quot;, caption=&quot;P=2.2E-16&quot;) +
  theme(
    legend.position = c(1, .5),
    legend.justification = c(&quot;right&quot;, &quot;top&quot;),
    legend.box.just = &quot;right&quot;,
    legend.margin = margin(6, 6, 6, 6)
    )</code></pre>
<p><img src="figure/2024-04-26_eQTL_subtypePlieotropy.Rmd/unnamed-chunk-1-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(&quot;../code/scratch/sQTL_hQTL_Pleiotropy.pdf&quot;, width=3.5, height=3.5)

GTEx.dat %&gt;%
  filter(tissue == &quot;cells_ebv-transformed_lymphocytes&quot;) %&gt;%
  count(nom_pval &lt; 0.01, TopSNP, sQTL_or_hQTL) %&gt;%
  filter(`nom_pval &lt; 0.01`) %&gt;%
  wilcox.test(n~sQTL_or_hQTL, data=.)</code></pre>
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  n by sQTL_or_hQTL
W = 79914, p-value &lt; 2.2e-16
alternative hypothesis: true location shift is not equal to 0</code></pre>
<p>What’s up with the 25% or so of sQTLs apparently with more than one
eQTL?</p>
<pre class="r"><code>Genes &lt;- read_tsv(&quot;../code/GTEx/QTLs/cells_ebv-transformed_lymphocytes/qqnorm.sorted.bed.gz&quot;) %&gt;%
  dplyr::select(1:6)

GTEx.dat %&gt;%
  filter(tissue == &quot;cells_ebv-transformed_lymphocytes&quot;) %&gt;%
  filter(sQTL_or_hQTL == &quot;sQTL&quot;) %&gt;%
  filter(nom_pval &lt; 0.01) %&gt;%
  add_count(TopSNP) %&gt;%
  inner_join(Genes, by=c(&quot;gene&quot;=&quot;gid&quot;)) %&gt;%
  mutate(IsPleitropic = n &gt; 1) %&gt;%
  dplyr::select(`#Chr`:pid, nom_pval, strand, IsPleitropic) %&gt;%
  group_by(IsPleitropic) %&gt;%
  group_walk(~ write_tsv(.x, paste0(&quot;../code/scratch/sQTLs_IsPleitropic_&quot;, .y$IsPleitropic, &quot;.bed&quot;), col_names = F))</code></pre>
<p>Let’s repeat some of the above analysis but after I recalled QTLtools
use use 500kb cis windows…</p>
<pre class="r"><code>dat.500kbwindow &lt;- Sys.glob(&quot;../code/GTEx/BenSubsetSNPs_QTLs/*/NominalPass.qqnorm.txt.tabix.gz&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/GTEx/BenSubsetSNPs_QTLs/(.+?)/NominalPass.qqnorm.txt.tabix.gz&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(read_tsv) %&gt;%
  bind_rows(.id=&quot;tissue&quot;) %&gt;%
  inner_join(GTEx.dat %&gt;%
               dplyr::select(-c(1:5)) %&gt;%
               distinct() %&gt;%
               separate(TopSNP, into=c(&quot;SNP_chrom&quot;, &quot;var_from&quot;, &quot;SNP_ref&quot;, &quot;SNP_alt&quot;), sep=&quot;:&quot;, remove=F, convert=T) %&gt;%
               mutate(var_chr = paste0(&quot;chr&quot;, SNP_chrom)),
             by=c(&quot;var_chr&quot;, &quot;var_from&quot;)) 



dat.500kbwindow %&gt;%
  distinct(TopSNP)</code></pre>
<pre><code># A tibble: 913 × 1
   TopSNP        
   &lt;chr&gt;         
 1 1:1661169:C:A 
 2 1:2556224:C:A 
 3 1:6205308:A:G 
 4 1:9943262:G:A 
 5 1:10211630:C:G
 6 1:11635653:T:G
 7 1:13769535:C:A
 8 1:15586903:A:T
 9 1:16206527:C:T
10 1:19801518:G:T
# … with 903 more rows</code></pre>
<pre class="r"><code>GTEx.dat %&gt;%
  distinct(TopSNP)</code></pre>
<pre><code># A tibble: 913 × 1
   TopSNP            
   &lt;chr&gt;             
 1 2:75660970:G:A    
 2 7:30500783:G:T    
 3 19:41586462:A:T   
 4 12:6321571:G:A    
 5 12:4607406:C:A    
 6 12:6184753:G:A    
 7 1:41105645:A:T    
 8 4:6609511:TG:T    
 9 19:54962842:G:A   
10 6:155314147:G:GTTC
# … with 903 more rows</code></pre>
<pre class="r"><code>dat.500kbwindow %&gt;%
  filter(tissue == &quot;cells_ebv-transformed_lymphocytes&quot;) %&gt;%
  mutate(sQTL_or_hQTL = recode(sQTL_or_hQTL, &quot;sQTL&quot;=&quot;u-sQTL/eQTL&quot;, &quot;hQTL&quot;=&quot;hQTL/eQTL&quot;)) %&gt;%
  mutate(IsDiscovery.eGene = if_else(`#phe_id` == eGene, &quot;Original discovery gene&quot;, &quot;Other cis-genes&quot;)) %&gt;%
  group_by(sQTL_or_hQTL, IsDiscovery.eGene) %&gt;%
  mutate(expected_p = percent_rank(nom_pval)) %&gt;%
  ggplot(aes(x=-log10(expected_p), y=-log10(nom_pval), color=sQTL_or_hQTL)) +
  geom_point() +
  geom_abline(slope=1, color=&#39;red&#39;) +
  scale_color_manual(values=c(&quot;#e31a1c&quot;, &quot;#6a3d9a&quot;)) +
  facet_wrap(~IsDiscovery.eGene) +
  geom_hline(yintercept =c(2, 5, 8), linetype=&#39;dashed&#39;)</code></pre>
<p><img src="figure/2024-04-26_eQTL_subtypePlieotropy.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dat.500kbwindow %&gt;%
  filter(tissue == &quot;cells_ebv-transformed_lymphocytes&quot;) %&gt;%
  mutate(`1E-2`=nom_pval&lt;0.01,
         `1E-5`=nom_pval&lt;1E-5,
         `1E-8`=nom_pval&lt;1E-8,
         ) %&gt;%
  dplyr::select(sQTL_or_hQTL, TopSNP, `1E-2`:`1E-8`) %&gt;%
  pivot_longer(names_to = &quot;Threshold&quot;, values_to = &quot;PassThreshold&quot;, `1E-2`:`1E-8`) %&gt;%
  group_by(sQTL_or_hQTL, TopSNP, Threshold) %&gt;%
  summarise(Num_eGenes = sum(PassThreshold)) %&gt;%
  ungroup() %&gt;%
  add_count(sQTL_or_hQTL, Threshold) %&gt;%
  mutate(sQTL_or_hQTL = recode(sQTL_or_hQTL, &quot;sQTL&quot;=&quot;u-sQTL/eQTL&quot;, &quot;hQTL&quot;=&quot;hQTL/eQTL&quot;)) %&gt;%
  mutate(sQTL_or_hQTL = str_glue(&quot;{sQTL_or_hQTL}; n={n}&quot;)) %&gt;%
  ggplot() +
  stat_ecdf(aes(x=Num_eGenes, color=sQTL_or_hQTL)) +
  scale_color_manual(values=c(&quot;#e31a1c&quot;, &quot;#6a3d9a&quot;)) +
  coord_cartesian(xlim=c(0,10)) +
  facet_wrap(~Threshold) +
  geom_text( data = . %&gt;%
                group_by(Threshold) %&gt;%
                summarize(results = format.pval(wilcox.test(Num_eGenes ~ sQTL_or_hQTL)$p.value, 2)),
             aes(label=str_glue(&quot;P: {results}&quot;), x=Inf, y=-Inf, vjust=0, hjust=1)) +
  labs(y=&quot;ecdf&quot;, x=&quot;horizontal pleiotropy\n(Number eGenes per SNP in GTEx LCLs)&quot;, title=&quot;facets are nominal Pval threshold&quot;)</code></pre>
<p><img src="figure/2024-04-26_eQTL_subtypePlieotropy.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dat.500kbwindow %&gt;%
  filter(tissue == &quot;cells_ebv-transformed_lymphocytes&quot;) %&gt;%
  mutate(`1E-2`=nom_pval&lt;0.01,
         `1E-5`=nom_pval&lt;1E-5,
         `1E-8`=nom_pval&lt;1E-8,
         ) %&gt;%
  dplyr::select(sQTL_or_hQTL, TopSNP, `1E-2`:`1E-8`) %&gt;%
  pivot_longer(names_to = &quot;Threshold&quot;, values_to = &quot;PassThreshold&quot;, `1E-2`:`1E-8`) %&gt;%
  group_by(sQTL_or_hQTL, TopSNP, Threshold) %&gt;%
  summarise(Num_eGenes = sum(PassThreshold)) %&gt;%
  ungroup() %&gt;%
  group_by(Threshold) %&gt;%
  summarize(results = wilcox.test(Num_eGenes ~ sQTL_or_hQTL)$p.value)</code></pre>
<pre><code># A tibble: 3 × 2
  Threshold  results
  &lt;chr&gt;        &lt;dbl&gt;
1 1E-2      3.22e-17
2 1E-5      1.08e-11
3 1E-8      4.96e- 8</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] forcats_0.5.1   stringr_1.4.0   dplyr_1.0.9     purrr_0.3.4    
[5] readr_2.1.2     tidyr_1.2.0     tibble_3.1.7    ggplot2_3.3.6  
[9] tidyverse_1.3.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3      lubridate_1.8.0   assertthat_0.2.1  rprojroot_2.0.3  
 [5] digest_0.6.29     utf8_1.2.2        R6_2.5.1          cellranger_1.1.0 
 [9] backports_1.4.1   reprex_2.0.1      evaluate_0.15     highr_0.9        
[13] httr_1.4.3        pillar_1.7.0      rlang_1.0.2       readxl_1.4.0     
[17] rstudioapi_0.13   jquerylib_0.1.4   rmarkdown_2.14    textshaping_0.3.6
[21] labeling_0.4.2    bit_4.0.4         munsell_0.5.0     broom_0.8.0      
[25] compiler_4.2.0    httpuv_1.6.5      modelr_0.1.8      xfun_0.30        
[29] systemfonts_1.0.4 pkgconfig_2.0.3   htmltools_0.5.2   tidyselect_1.1.2 
[33] workflowr_1.7.0   fansi_1.0.3       crayon_1.5.1      tzdb_0.3.0       
[37] dbplyr_2.1.1      withr_2.5.0       later_1.3.0       grid_4.2.0       
[41] jsonlite_1.8.0    gtable_0.3.0      lifecycle_1.0.1   DBI_1.1.2        
[45] git2r_0.30.1      magrittr_2.0.3    scales_1.2.0      cli_3.3.0        
[49] stringi_1.7.6     vroom_1.5.7       farver_2.1.0      fs_1.5.2         
[53] promises_1.2.0.1  xml2_1.3.3        bslib_0.3.1       ragg_1.2.5       
[57] ellipsis_0.3.2    generics_0.1.2    vctrs_0.4.1       tools_4.2.0      
[61] bit64_4.0.5       glue_1.6.2        hms_1.1.1         parallel_4.2.0   
[65] fastmap_1.1.0     yaml_2.3.5        colorspace_2.0-3  rvest_1.0.2      
[69] knitr_1.39        haven_2.5.0       sass_0.4.1       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
