<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Home</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Home</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-02-13
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>ChromatinSplicingQTLs/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstreec56b87de8f928faf3146c93e34a4289b8e07d271targetblankc56b87da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/c56b87de8f928faf3146c93e34a4289b8e07d271" target="_blank">c56b87d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstreec56b87de8f928faf3146c93e34a4289b8e07d271targetblankc56b87da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/c56b87de8f928faf3146c93e34a4289b8e07d271" target="_blank">c56b87d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    code/-
    Ignored:    code/.DS_Store
    Ignored:    code/.RData
    Ignored:    code/._report.html
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    code/.snakemake/
    Ignored:    code/APA_Processing/
    Ignored:    code/Alignments/
    Ignored:    code/ChromHMM/
    Ignored:    code/ENCODE/
    Ignored:    code/ExpressionAnalysis/
    Ignored:    code/ExtractPhenotypeBedByGenotype.py
    Ignored:    code/FastqFastp/
    Ignored:    code/FastqFastpSE/
    Ignored:    code/FastqSE/
    Ignored:    code/FineMapping/
    Ignored:    code/GTEx/
    Ignored:    code/Gencode.v34.6Colors.bed.gz
    Ignored:    code/Genotypes/
    Ignored:    code/H3K36me3_CutAndTag.pdf
    Ignored:    code/IntronSlopes/
    Ignored:    code/LR.bed
    Ignored:    code/LR.seq.bed
    Ignored:    code/LongReads/
    Ignored:    code/MYB.tracks.ini
    Ignored:    code/Metaplots/
    Ignored:    code/Misc/
    Ignored:    code/MiscCountTables/
    Ignored:    code/Multiqc/
    Ignored:    code/Multiqc_chRNA/
    Ignored:    code/NonCodingRNA/
    Ignored:    code/NonCodingRNA_annotation/
    Ignored:    code/PairwisePi1Traits.P.all.txt.gz
    Ignored:    code/PeakCalling/
    Ignored:    code/Phenotypes/
    Ignored:    code/PlotGruberQTLs/
    Ignored:    code/PlotQTLs/
    Ignored:    code/ProCapAnalysis/
    Ignored:    code/QC/
    Ignored:    code/QTL_SNP_Enrichment/
    Ignored:    code/QTLs/
    Ignored:    code/RPKM_tables/
    Ignored:    code/ReadLengthMapExperiment/
    Ignored:    code/ReadLengthMapExperimentResults/
    Ignored:    code/ReadLengthMapExperimentSpliceCounts/
    Ignored:    code/ReferenceGenome/
    Ignored:    code/Rplots.pdf
    Ignored:    code/Session.vim
    Ignored:    code/SmallMolecule/
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TODO
    Ignored:    code/Tehranchi/
    Ignored:    code/alias/
    Ignored:    code/bigwigs/
    Ignored:    code/bigwigs_FromNonWASPFilteredReads/
    Ignored:    code/config/.DS_Store
    Ignored:    code/config/._.DS_Store
    Ignored:    code/config/.ipynb_checkpoints/
    Ignored:    code/config/config.local.yaml
    Ignored:    code/dag.pdf
    Ignored:    code/dag.png
    Ignored:    code/dag.svg
    Ignored:    code/data/
    Ignored:    code/debug.ipynb
    Ignored:    code/debug_python.ipynb
    Ignored:    code/deepTools/
    Ignored:    code/featureCounts/
    Ignored:    code/featureCountsBasicGtf/
    Ignored:    code/genome_config.yaml
    Ignored:    code/gwas_summary_stats/
    Ignored:    code/hyprcoloc/
    Ignored:    code/igv_session.xml
    Ignored:    code/isoseqbams/
    Ignored:    code/log
    Ignored:    code/logs/
    Ignored:    code/notebooks/.ipynb_checkpoints/
    Ignored:    code/pi1/
    Ignored:    code/polyA.Splicing.Subset_YRI.NominalPassForColoc.bed.bgz
    Ignored:    code/rules/.ipynb_checkpoints/
    Ignored:    code/rules/OldRules/
    Ignored:    code/rules/notebooks/
    Ignored:    code/salmontest/
    Ignored:    code/scratch/
    Ignored:    code/scripts/.ipynb_checkpoints/
    Ignored:    code/scripts/GTFtools_0.8.0/
    Ignored:    code/scripts/__pycache__/
    Ignored:    code/scripts/liftOverBedpe/liftOverBedpe.py
    Ignored:    code/snakemake.dryrun.log
    Ignored:    code/snakemake.log
    Ignored:    code/snakemake.sbatch.log
    Ignored:    code/snakemake_profiles/slurm/__pycache__/
    Ignored:    code/test.introns.bed
    Ignored:    code/test.introns2.bed
    Ignored:    code/test.log
    Ignored:    code/tracks.xml
    Ignored:    data/.DS_Store
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-10.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-11.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-2.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-3.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-4.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-5.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-6.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-7.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-8.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022.csv
    Ignored:    data/Metaplots/.DS_Store

Untracked files:
    Untracked:  analysis/2024-01-17_CheckYangsJunctionAnnotator.Rmd

Unstaged changes:
    Modified:   analysis/MakeFinalFigs_Fig2.Rmd
    Modified:   analysis/MakeFinalFigs_Fig4.Rmd
    Modified:   code/scripts/GenometracksByGenotype

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML
(<code>docs/index.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/c6e8be3f584002a3eccefe5c2fd7d7975276eb1f/docs/index.html" target="_blank">c6e8be3</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-06-26
</td>
<td>
added nb for chao
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/1650a0301d581e5ba8f4fc58947506d23c989a2e/analysis/index.Rmd" target="_blank">1650a03</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-05-29
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/1650a0301d581e5ba8f4fc58947506d23c989a2e/docs/index.html" target="_blank">1650a03</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-05-29
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/5946b5ad4bcc1f248d89293af6839c2e5a2c1fbb/docs/index.html" target="_blank">5946b5a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-05-22
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/1fbb2dc857c60e51cb7656fca6b7375065bfa5b9/docs/index.html" target="_blank">1fbb2dc</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-05-02
</td>
<td>
add notebooks for sm PSI
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/6c60c18d86f538cf04f01f094fe200eb6da55140/docs/index.html" target="_blank">6c60c18</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-04-25
</td>
<td>
update nb
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/ddf600dbb95a36158c06a26c5d7df9d69c969ed0/docs/index.html" target="_blank">ddf600d</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-04-14
</td>
<td>
add SM splicing timing analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/ddbf2225e7f71ace4a67a2cdae69ece7e4026b66/docs/index.html" target="_blank">ddbf222</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-27
</td>
<td>
add pi1 heatmaps nb
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/de90abf6c2b1a988631ef11c2edd5100a5cf39c9/docs/index.html" target="_blank">de90abf</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-23
</td>
<td>
update sQTL eQTL scatter nb
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/5e6b611a1abfadaea59daf6b1965d9fcd6e756df/docs/index.html" target="_blank">5e6b611</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-01-10
</td>
<td>
update site with SM analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/ea425ea0d4eafe82bf3fff9b1e4a6c44222f5e79/docs/index.html" target="_blank">ea425ea</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-12-15
</td>
<td>
add sm processing nb
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/e34a204b7c5fb9e22cc20c47eff428e5c869699d/docs/index.html" target="_blank">e34a204</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-12-09
</td>
<td>
update homepage
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/cad3bd6912c55814bfc120da736f4f55b3fbc1f6/analysis/index.Rmd" target="_blank">cad3bd6</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-11-16
</td>
<td>
misc updates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/cad3bd6912c55814bfc120da736f4f55b3fbc1f6/docs/index.html" target="_blank">cad3bd6</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-11-16
</td>
<td>
misc updates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/e706dbd0b4b728a5e349ab62bcf83ffd9488ce83/docs/index.html" target="_blank">e706dbd</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-11-01
</td>
<td>
update nb
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/8df7e43cc493d35d443d132d58c6f327ee406d9f/docs/index.html" target="_blank">8df7e43</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-26
</td>
<td>
update nb
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/61a245fa779318e64d6bb6154b21c4a792509781/docs/index.html" target="_blank">61a245f</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-24
</td>
<td>
update nb
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/c17b6b2fcd112a08cc7e7a2fb5bb45538dc3d339/docs/index.html" target="_blank">c17b6b2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-19
</td>
<td>
add un qqnorm phenotypes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/ef85e5320603653a3919d55d51162a3ada4b35d8/docs/index.html" target="_blank">ef85e53</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-12
</td>
<td>
update index
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/ae6409c23e6e7fb2acd20f0c180a593ab3a0c7f6/docs/index.html" target="_blank">ae6409c</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-07
</td>
<td>
update pi1 nb
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/149e3ef82b5a9e26bd4b86d40e5c057d0e726f39/docs/index.html" target="_blank">149e3ef</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-10-07
</td>
<td>
update enrichment nb
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/311db7d5d72412460463be9dc5e60d4668b2b8e4/docs/index.html" target="_blank">311db7d</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-09-29
</td>
<td>
explore intron sum nb
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/f94d014c256456e08c5d6bff2c16d4f1da8093a8/docs/index.html" target="_blank">f94d014</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-09-22
</td>
<td>
edit index
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/a80e66e6d993523519eab2c54c0db7e8decc0dab/docs/index.html" target="_blank">a80e66e</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-09-22
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/bfb625eb2719f3fde37af38bfa3f13d0ff2818cc/docs/index.html" target="_blank">bfb625e</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-08-09
</td>
<td>
added finemapping snp enrichment rules
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/125a5830bc86f4326934c0e0548c83afb03167ee/docs/index.html" target="_blank">125a583</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-06-08
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/fe6333bb3982433069d51165ab0291340093976d/docs/index.html" target="_blank">fe6333b</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-05-25
</td>
<td>
update site and molQTL coloc
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/0c8ee9c65abd3ee8c00bed55cfafcfd2caa5bdbd/docs/index.html" target="_blank">0c8ee9c</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-01-18
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/237b54974507ae639dd505ca0019ae3c6c854af2/docs/index.html" target="_blank">237b549</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2021-12-28
</td>
<td>
rebuilt some site analyses
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/2f0181fdb772c14d30a0d7af03e5b3576d95903a/docs/index.html" target="_blank">2f0181f</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2021-12-15
</td>
<td>
Added genewise hyprcoloc rules
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/50ad81859f449c21a41bd45eac46c7208976dca3/docs/index.html" target="_blank">50ad818</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2021-12-07
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/a166e86d926af346299e8ec527cd15b96a2ac932/docs/index.html" target="_blank">a166e86</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2021-10-06
</td>
<td>
added test notebook
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/24331d7a50d3f935aeb26b9060bc539f194eabc2/analysis/index.Rmd" target="_blank">24331d7</a>
</td>
<td>
cfbuenabadn
</td>
<td>
2021-09-23
</td>
<td>
adding chRNA-seq processing smk
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/24331d7a50d3f935aeb26b9060bc539f194eabc2/docs/index.html" target="_blank">24331d7</a>
</td>
<td>
cfbuenabadn
</td>
<td>
2021-09-23
</td>
<td>
adding chRNA-seq processing smk
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/ee655cce83b515ff67db7b95a4a7158019356acf/docs/index.html" target="_blank">ee655cc</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2021-09-23
</td>
<td>
fixed site links
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/afc10fbdc9952caeecd2639f070f357cef548aeb/docs/index.html" target="_blank">afc10fb</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2021-09-23
</td>
<td>
fixed ipynb links on site homepage
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/91f067a6c30ae94dca690c73bd30d82f56d237e0/docs/index.html" target="_blank">91f067a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2021-09-23
</td>
<td>
added example ipynb to site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/53ba92f62702392ecd229616e12d4c4be32e2472/analysis/index.Rmd" target="_blank">53ba92f</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2021-06-07
</td>
<td>
update side and check sample swapping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/53ba92f62702392ecd229616e12d4c4be32e2472/docs/index.html" target="_blank">53ba92f</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2021-06-07
</td>
<td>
update side and check sample swapping
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/dd0eadbb209caea66e98ef74baf13cd7a9268eca/analysis/index.Rmd" target="_blank">dd0eadb</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2020-08-21
</td>
<td>
initial commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ChromatinSplicingQTLs/dd0eadbb209caea66e98ef74baf13cd7a9268eca/docs/index.html" target="_blank">dd0eadb</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2020-08-21
</td>
<td>
initial commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Welcome to my research website.</p>
<p>This project will investigate the correlation of genetic effects on
chromatin, splicing, transcription, and complex phenotypes, using
naturally occuring human genetic variation.</p>
<p>We are still collecting data, but I have also been perusing around
published data a bit.</p>
<p>Here are links to all of my rendered analysis Rmarkdowns (code <a
href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/master/analysis">here</a>):</p>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>✔ ggplot2 3.3.6     ✔ purrr   0.3.4
✔ tibble  3.1.7     ✔ dplyr   1.0.9
✔ tidyr   1.2.0     ✔ stringr 1.4.0
✔ readr   2.1.2     ✔ forcats 0.5.1</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>rmds &lt;- list.files(path = &quot;../analysis/&quot;, pattern=&quot;^\\d+.+&quot;)
rmd_htmls &lt;- str_replace(rmds, &quot;Rmd$&quot;, &quot;html&quot;)

for(i in rmd_htmls){
  cat(&quot;[&quot;, i, &quot;](&quot;, i,&quot;)\n\n&quot;, sep=&quot;&quot;)
}</code></pre>
<p><a
href="20121206_Explore_sQTLs_AndAttemptAnnotation.html">20121206_Explore_sQTLs_AndAttemptAnnotation.html</a></p>
<p><a
href="20200110_PickSamplesForGrowth.html">20200110_PickSamplesForGrowth.html</a></p>
<p><a
href="20200210_PickSamplesToOrder.html">20200210_PickSamplesToOrder.html</a></p>
<p><a
href="20210604_CheckSampleGenotypes.html">20210604_CheckSampleGenotypes.html</a></p>
<p><a
href="20211207_ExploreColocalizations.html">20211207_ExploreColocalizations.html</a></p>
<p><a
href="20211217_GenewiseColocFirstLook.html">20211217_GenewiseColocFirstLook.html</a></p>
<p><a
href="20220114_ColocalizationEffectSizeCorrelations.html">20220114_ColocalizationEffectSizeCorrelations.html</a></p>
<p><a
href="20220228_PickCutAndTagSamples.html">20220228_PickCutAndTagSamples.html</a></p>
<p><a
href="202204012_Cluster_TestGwasHarmonisation.html">202204012_Cluster_TestGwasHarmonisation.html</a></p>
<p><a
href="20220401_Cluster_CheckOutNA18855chRNA.html">20220401_Cluster_CheckOutNA18855chRNA.html</a></p>
<p><a
href="20220415_Cluster_CheckTehranchiConcordance.html">20220415_Cluster_CheckTehranchiConcordance.html</a></p>
<p><a
href="20220424_CheckH3K36me3SampleGenotypes.html">20220424_CheckH3K36me3SampleGenotypes.html</a></p>
<p><a
href="20220511_Cluster_CheckH3K36me3QC.html">20220511_Cluster_CheckH3K36me3QC.html</a></p>
<p><a
href="20220511_ExploreColocsAtDifferentThresholds.html">20220511_ExploreColocsAtDifferentThresholds.html</a></p>
<p><a
href="20220518_Explore_eQTLColocsAtDifferentThresholds.html">20220518_Explore_eQTLColocsAtDifferentThresholds.html</a></p>
<p><a
href="20220524_CheckClosestPeakToTSS.html">20220524_CheckClosestPeakToTSS.html</a></p>
<p><a
href="20220527_ExploreCarlosChromatinSplicingData.html">20220527_ExploreCarlosChromatinSplicingData.html</a></p>
<p><a
href="20220606_PlotColocsForIntuitions.html">20220606_PlotColocsForIntuitions.html</a></p>
<p><a
href="20220622_QuantifyColocRateMoreInterpretableWay.html">20220622_QuantifyColocRateMoreInterpretableWay.html</a></p>
<p><a
href="20220623_CalculatePi1.html">20220623_CalculatePi1.html</a></p>
<p><a
href="20220623_CalculatePi1_AllTraits.html">20220623_CalculatePi1_AllTraits.html</a></p>
<p><a
href="20220627_CalculatePi1_AllTraits_AndCompareToColoc.html">20220627_CalculatePi1_AllTraits_AndCompareToColoc.html</a></p>
<p><a
href="20220628_exploretiertwomethods.html">20220628_exploretiertwomethods.html</a></p>
<p><a
href="20220713_PlotHeatmapManyWays.html">20220713_PlotHeatmapManyWays.html</a></p>
<p><a
href="20220713_PlotHeatmapManyWays_JustPi1.html">20220713_PlotHeatmapManyWays_JustPi1.html</a></p>
<p><a href="20220713_RPKM_v_TPM.html">20220713_RPKM_v_TPM.html</a></p>
<p><a
href="20220728_CheckFinemapSNPEnrichments.html">20220728_CheckFinemapSNPEnrichments.html</a></p>
<p><a
href="20220920_Explore_chRNA_sQTLs.html">20220920_Explore_chRNA_sQTLs.html</a></p>
<p><a
href="20220928_ExploreIntronSum.html">20220928_ExploreIntronSum.html</a></p>
<p><a
href="20220930_CheckSNP_QTL_Positions.html">20220930_CheckSNP_QTL_Positions.html</a></p>
<p><a
href="20221011_PlotHeatmapManyWays_ncRNA_Updated.html">20221011_PlotHeatmapManyWays_ncRNA_Updated.html</a></p>
<p><a
href="20221012_IntronRetentionAndExpressionConcordance.html">20221012_IntronRetentionAndExpressionConcordance.html</a></p>
<p><a
href="20221016_ExplorePSI_alternatives.html">20221016_ExplorePSI_alternatives.html</a></p>
<p><a href="20221019_PlotFigs1.html">20221019_PlotFigs1.html</a></p>
<p><a
href="20221024_InspectUnstandardizedBetas.html">20221024_InspectUnstandardizedBetas.html</a></p>
<p><a
href="20221028_CheckDeltaPWM.html">20221028_CheckDeltaPWM.html</a></p>
<p><a
href="20221030_PlotsForLabMeeting.html">20221030_PlotsForLabMeeting.html</a></p>
<p><a
href="20221102_PlotStuffForYang.html">20221102_PlotStuffForYang.html</a></p>
<p><a
href="20221108_ExploreGtfTags.html">20221108_ExploreGtfTags.html</a></p>
<p><a
href="20221116_SmallMoleculeConfigSetup.html">20221116_SmallMoleculeConfigSetup.html</a></p>
<p><a
href="20221202_ProcessSmallMolecule.html">20221202_ProcessSmallMolecule.html</a></p>
<p><a
href="20221214_CheckYangsUnannotatedAnnotations.html">20221214_CheckYangsUnannotatedAnnotations.html</a></p>
<p><a
href="20230110_eQTL_sQTL_hQTL_Colocalization.html">20230110_eQTL_sQTL_hQTL_Colocalization.html</a></p>
<p><a
href="20230112_AreGenesWithMoreUnannotatedLessInPolyA.html">20230112_AreGenesWithMoreUnannotatedLessInPolyA.html</a></p>
<p><a
href="20230120_CheckYangsUnannotatedAnnotations.html">20230120_CheckYangsUnannotatedAnnotations.html</a></p>
<p><a
href="20230202_GenewiseFractionUnrproductiveSplicing.html">20230202_GenewiseFractionUnrproductiveSplicing.html</a></p>
<p><a
href="20230202_Pick_Unannotated_sQTLs_to_plot.html">20230202_Pick_Unannotated_sQTLs_to_plot.html</a></p>
<p><a
href="20230202_PlotForLabMeeting.html">20230202_PlotForLabMeeting.html</a></p>
<p><a
href="20230210_CheckOldNMDIntrons.html">20230210_CheckOldNMDIntrons.html</a></p>
<p><a
href="20230214_PlotsWithNewAnnotations.html">20230214_PlotsWithNewAnnotations.html</a></p>
<p><a
href="20230314_ProcessSM_ForInterpretableSplicingEffectSizes.html">20230314_ProcessSM_ForInterpretableSplicingEffectSizes.html</a></p>
<p><a
href="20230320_RemakeEffectSizeScatters_UpdatedAnnotations.html">20230320_RemakeEffectSizeScatters_UpdatedAnnotations.html</a></p>
<p><a href="20230324_MakeFigs.html">20230324_MakeFigs.html</a></p>
<p><a
href="20230413_CheckSplicingTimingOfSM_sensitiveIntrons.html">20230413_CheckSplicingTimingOfSM_sensitiveIntrons.html</a></p>
<p><a
href="20230417_SusieFinemapEnrichment.html">20230417_SusieFinemapEnrichment.html</a></p>
<p><a
href="20230420_FindeQTLsQTLGWAS_Colocs.html">20230420_FindeQTLsQTLGWAS_Colocs.html</a></p>
<p><a
href="20230502_GetColocsForCarlos.html">20230502_GetColocsForCarlos.html</a></p>
<p><a
href="20230504_SmallMoleculeFig.html">20230504_SmallMoleculeFig.html</a></p>
<p><a
href="20230511_coloc_sQTLs_hQTLs_eQTLs_GWAS.ipynb">20230511_coloc_sQTLs_hQTLs_eQTLs_GWAS.ipynb</a></p>
<p><a
href="20230516_ExploreChaosAnnotations.html">20230516_ExploreChaosAnnotations.html</a></p>
<p><a
href="20230522_ResolvePercentNMD_Discrepancy.html">20230522_ResolvePercentNMD_Discrepancy.html</a></p>
<p><a
href="20230523_FindCassettteExons.html">20230523_FindCassettteExons.html</a></p>
<p><a
href="20230525_Intron_retention_supp_plots.ipynb">20230525_Intron_retention_supp_plots.ipynb</a></p>
<p><a
href="20230528_IdentifyBrianaPacBio.html">20230528_IdentifyBrianaPacBio.html</a></p>
<p><a
href="20230622_LookupCarolsSNP.html">20230622_LookupCarolsSNP.html</a></p>
<p><a
href="20230623_BetaBetaScatterForChao.html">20230623_BetaBetaScatterForChao.html</a></p>
<p><a href="20230627_Regression.ipynb">20230627_Regression.ipynb</a></p>
<p><a
href="20230707_MostAbundantUnrproductiveJuncPerGene.html">20230707_MostAbundantUnrproductiveJuncPerGene.html</a></p>
<p><a
href="20230723_hQTL_sQTL_GwasConcordance.html">20230723_hQTL_sQTL_GwasConcordance.html</a></p>
<p><a
href="20230808_GWAS_sQTLs_QQ_and_ColocExample.html">20230808_GWAS_sQTLs_QQ_and_ColocExample.html</a></p>
<p><a
href="20230907_ColocExamplePlotsWithLD.html">20230907_ColocExamplePlotsWithLD.html</a></p>
<p><a
href="20237003_ExplorePSIMethods.html">20237003_ExplorePSIMethods.html</a></p>
<p><a
href="2024-01-17_CheckYangsJunctionAnnotator.html">2024-01-17_CheckYangsJunctionAnnotator.html</a></p>
<p>Also, a list of links to ipynb files that were incorporated into the
snakemake. Since github renders ipynbs directly, this is not a relative
link from the site folder, but rather links directly to the file to view
on github (code <a
href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/master/code/notebooks">here</a>):</p>
<pre class="r"><code>ipynbs &lt;- list.files(path = &quot;../docs/&quot;, pattern=&quot;^\\d+.+ipynb$&quot;)

for(i in ipynbs){
  cat(&quot;[&quot;, i, &quot;](&quot;,&quot;https://github.com/bfairkun/ChromatinSplicingQTLs/blob/master/docs/&quot;, i,&quot;)\n\n&quot;, sep=&quot;&quot;)
}</code></pre>
<p><a
href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/master/docs/20210921_CountSpliceSiteSNPsInSQTLs.py.ipynb">20210921_CountSpliceSiteSNPsInSQTLs.py.ipynb</a></p>
<div id="notebooks-for-figures" class="section level2">
<h2>Notebooks for figures</h2>
<pre class="r"><code>library(tidyverse)
rmds &lt;- c(list.files(path = &quot;../analysis/&quot;, pattern=&quot;^Figures.+&quot;), list.files(path = &quot;../analysis/&quot;, pattern=&quot;^MakeFi.+&quot;))
rmd_htmls &lt;- str_replace(rmds, &quot;Rmd$&quot;, &quot;html&quot;)

for(i in rmd_htmls){
  cat(&quot;[&quot;, i, &quot;](&quot;, i,&quot;)\n\n&quot;, sep=&quot;&quot;)
}</code></pre>
<p><a
href="Figures_BensTasksFromOutline.html">Figures_BensTasksFromOutline.html</a></p>
<p><a
href="Figures_sQTL_eQTL_effects.html">Figures_sQTL_eQTL_effects.html</a></p>
<p><a href="MakeFinalFigs_Fig1.html">MakeFinalFigs_Fig1.html</a></p>
<p><a href="MakeFinalFigs_Fig1.ipynb">MakeFinalFigs_Fig1.ipynb</a></p>
<p><a href="MakeFinalFigs_Fig1A.ipynb">MakeFinalFigs_Fig1A.ipynb</a></p>
<p><a href="MakeFinalFigs_Fig2.html">MakeFinalFigs_Fig2.html</a></p>
<p><a href="MakeFinalFigs_Fig2.ipynb">MakeFinalFigs_Fig2.ipynb</a></p>
<p><a
href="MakeFinalFigs_Fig2D_long_reads.ipynb">MakeFinalFigs_Fig2D_long_reads.ipynb</a></p>
<p><a href="MakeFinalFigs_Fig3.html">MakeFinalFigs_Fig3.html</a></p>
<p><a href="MakeFinalFigs_Fig4.html">MakeFinalFigs_Fig4.html</a></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] forcats_0.5.1   stringr_1.4.0   dplyr_1.0.9     purrr_0.3.4    
[5] readr_2.1.2     tidyr_1.2.0     tibble_3.1.7    ggplot2_3.3.6  
[9] tidyverse_1.3.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3     lubridate_1.8.0  assertthat_0.2.1 rprojroot_2.0.3 
 [5] digest_0.6.29    utf8_1.2.2       R6_2.5.1         cellranger_1.1.0
 [9] backports_1.4.1  reprex_2.0.1     evaluate_0.15    httr_1.4.3      
[13] pillar_1.7.0     rlang_1.0.2      readxl_1.4.0     rstudioapi_0.13 
[17] whisker_0.4      jquerylib_0.1.4  rmarkdown_2.14   munsell_0.5.0   
[21] broom_0.8.0      compiler_4.2.0   httpuv_1.6.5     modelr_0.1.8    
[25] xfun_0.30        pkgconfig_2.0.3  htmltools_0.5.2  tidyselect_1.1.2
[29] workflowr_1.7.0  fansi_1.0.3      crayon_1.5.1     tzdb_0.3.0      
[33] dbplyr_2.1.1     withr_2.5.0      later_1.3.0      grid_4.2.0      
[37] jsonlite_1.8.0   gtable_0.3.0     lifecycle_1.0.1  DBI_1.1.2       
[41] git2r_0.30.1     magrittr_2.0.3   scales_1.2.0     cli_3.3.0       
[45] stringi_1.7.6    fs_1.5.2         promises_1.2.0.1 xml2_1.3.3      
[49] bslib_0.3.1      ellipsis_0.3.2   generics_0.1.2   vctrs_0.4.1     
[53] tools_4.2.0      glue_1.6.2       hms_1.1.1        fastmap_1.1.0   
[57] yaml_2.3.5       colorspace_2.0-3 rvest_1.0.2      knitr_1.39      
[61] haven_2.5.0      sass_0.4.1      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
