<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2023-05-11" />

<title>MakeFinalFigs_Fig4</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">MakeFinalFigs_Fig4</h1>
<h4 class="date">2023-05-11</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-02-13
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>ChromatinSplicingQTLs/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstreec56b87de8f928faf3146c93e34a4289b8e07d271targetblankc56b87da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/c56b87de8f928faf3146c93e34a4289b8e07d271" target="_blank">c56b87d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunChromatinSplicingQTLstreec56b87de8f928faf3146c93e34a4289b8e07d271targetblankc56b87da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/tree/c56b87de8f928faf3146c93e34a4289b8e07d271" target="_blank">c56b87d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    code/-
    Ignored:    code/.DS_Store
    Ignored:    code/.RData
    Ignored:    code/._report.html
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    code/.snakemake/
    Ignored:    code/APA_Processing/
    Ignored:    code/Alignments/
    Ignored:    code/ChromHMM/
    Ignored:    code/ENCODE/
    Ignored:    code/ExpressionAnalysis/
    Ignored:    code/ExtractPhenotypeBedByGenotype.py
    Ignored:    code/FastqFastp/
    Ignored:    code/FastqFastpSE/
    Ignored:    code/FastqSE/
    Ignored:    code/FineMapping/
    Ignored:    code/GTEx/
    Ignored:    code/Gencode.v34.6Colors.bed.gz
    Ignored:    code/Genotypes/
    Ignored:    code/H3K36me3_CutAndTag.pdf
    Ignored:    code/IntronSlopes/
    Ignored:    code/LR.bed
    Ignored:    code/LR.seq.bed
    Ignored:    code/LongReads/
    Ignored:    code/MYB.tracks.ini
    Ignored:    code/Metaplots/
    Ignored:    code/Misc/
    Ignored:    code/MiscCountTables/
    Ignored:    code/Multiqc/
    Ignored:    code/Multiqc_chRNA/
    Ignored:    code/NonCodingRNA/
    Ignored:    code/NonCodingRNA_annotation/
    Ignored:    code/PairwisePi1Traits.P.all.txt.gz
    Ignored:    code/PeakCalling/
    Ignored:    code/Phenotypes/
    Ignored:    code/PlotGruberQTLs/
    Ignored:    code/PlotQTLs/
    Ignored:    code/ProCapAnalysis/
    Ignored:    code/QC/
    Ignored:    code/QTL_SNP_Enrichment/
    Ignored:    code/QTLs/
    Ignored:    code/RPKM_tables/
    Ignored:    code/ReadLengthMapExperiment/
    Ignored:    code/ReadLengthMapExperimentResults/
    Ignored:    code/ReadLengthMapExperimentSpliceCounts/
    Ignored:    code/ReferenceGenome/
    Ignored:    code/Rplots.pdf
    Ignored:    code/Session.vim
    Ignored:    code/SmallMolecule/
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TODO
    Ignored:    code/Tehranchi/
    Ignored:    code/alias/
    Ignored:    code/bigwigs/
    Ignored:    code/bigwigs_FromNonWASPFilteredReads/
    Ignored:    code/config/.DS_Store
    Ignored:    code/config/._.DS_Store
    Ignored:    code/config/.ipynb_checkpoints/
    Ignored:    code/config/config.local.yaml
    Ignored:    code/dag.pdf
    Ignored:    code/dag.png
    Ignored:    code/dag.svg
    Ignored:    code/data/
    Ignored:    code/debug.ipynb
    Ignored:    code/debug_python.ipynb
    Ignored:    code/deepTools/
    Ignored:    code/featureCounts/
    Ignored:    code/featureCountsBasicGtf/
    Ignored:    code/genome_config.yaml
    Ignored:    code/gwas_summary_stats/
    Ignored:    code/hyprcoloc/
    Ignored:    code/igv_session.xml
    Ignored:    code/isoseqbams/
    Ignored:    code/log
    Ignored:    code/logs/
    Ignored:    code/notebooks/.ipynb_checkpoints/
    Ignored:    code/pi1/
    Ignored:    code/polyA.Splicing.Subset_YRI.NominalPassForColoc.bed.bgz
    Ignored:    code/rules/.ipynb_checkpoints/
    Ignored:    code/rules/OldRules/
    Ignored:    code/rules/notebooks/
    Ignored:    code/salmontest/
    Ignored:    code/scratch/
    Ignored:    code/scripts/.ipynb_checkpoints/
    Ignored:    code/scripts/GTFtools_0.8.0/
    Ignored:    code/scripts/__pycache__/
    Ignored:    code/scripts/liftOverBedpe/liftOverBedpe.py
    Ignored:    code/snakemake.dryrun.log
    Ignored:    code/snakemake.log
    Ignored:    code/snakemake.sbatch.log
    Ignored:    code/snakemake_profiles/slurm/__pycache__/
    Ignored:    code/test.introns.bed
    Ignored:    code/test.introns2.bed
    Ignored:    code/test.log
    Ignored:    code/tracks.xml
    Ignored:    data/.DS_Store
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-10.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-11.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-2.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-3.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-4.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-5.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-6.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-7.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022-8.csv
    Ignored:    data/GWAS_catalog_summary_stats_sources/._list_gwas_summary_statistics_6_Apr_2022.csv
    Ignored:    data/Metaplots/.DS_Store

Untracked files:
    Untracked:  analysis/2024-01-17_CheckYangsJunctionAnnotator.Rmd

Unstaged changes:
    Modified:   analysis/MakeFinalFigs_Fig2.Rmd
    Modified:   analysis/MakeFinalFigs_Fig4.Rmd
    Modified:   code/scripts/GenometracksByGenotype

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/MakeFinalFigs_Fig4.Rmd</code>)
and HTML (<code>docs/MakeFinalFigs_Fig4.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/42d9a7b8d8fe83369cdb1e6bb7fa739b0aedd6b1/analysis/MakeFinalFigs_Fig4.Rmd" target="_blank">42d9a7b</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-09-14
</td>
<td>
ben updates
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/ada72d2d880d3117b83ca2df2d07e8c8c9f4bf3d/analysis/MakeFinalFigs_Fig4.Rmd" target="_blank">ada72d2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-07-24
</td>
<td>
updates Ben
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/3d0e9c15ac64b7677bc283ecf022dc9787894d51/analysis/MakeFinalFigs_Fig4.Rmd" target="_blank">3d0e9c1</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-06-21
</td>
<td>
update intron annotations
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ChromatinSplicingQTLs/blob/5946b5ad4bcc1f248d89293af6839c2e5a2c1fbb/analysis/MakeFinalFigs_Fig4.Rmd" target="_blank">5946b5a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-05-22
</td>
<td>
update site
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE, warning = F, message = F)

library(tidyverse)</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>✔ ggplot2 3.3.6     ✔ purrr   0.3.4
✔ tibble  3.1.7     ✔ dplyr   1.0.9
✔ tidyr   1.2.0     ✔ stringr 1.4.0
✔ readr   2.1.2     ✔ forcats 0.5.1</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(RColorBrewer)
library(data.table)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre class="r"><code>library(edgeR)</code></pre>
<pre><code>Loading required package: limma</code></pre>
<pre class="r"><code>library(readxl)
library(qvalue)
library(ggrepel)
library(knitr)
library(drc)</code></pre>
<pre><code>Loading required package: MASS</code></pre>
<pre><code>
Attaching package: &#39;MASS&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:dplyr&#39;:

    select</code></pre>
<pre><code>
&#39;drc&#39; has been loaded.</code></pre>
<pre><code>Please cite R and &#39;drc&#39; if used for a publication,</code></pre>
<pre><code>for references type &#39;citation()&#39; and &#39;citation(&#39;drc&#39;)&#39;.</code></pre>
<pre><code>
Attaching package: &#39;drc&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    gaussian, getInitial</code></pre>
<pre class="r"><code>library(ggbreak)</code></pre>
<pre><code>ggbreak v0.1.1

If you use ggbreak in published research, please cite the following
paper:

S Xu, M Chen, T Feng, L Zhan, L Zhou, G Yu. Use ggbreak to effectively
utilize plotting space to deal with large datasets and outliers.
Frontiers in Genetics. 2021, 12:774846. doi: 10.3389/fgene.2021.774846</code></pre>
<pre class="r"><code># Set theme
theme_set(
  theme_classic() +
  theme(text=element_text(size=16,  family=&quot;Helvetica&quot;)))

# I use layer a lot, to rotate long x-axis labels
Rotate_x_labels &lt;- theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))</code></pre>
<div id="list-of-figs-to-make" class="section level2">
<h2>List of figs to make</h2>
<ul>
<li>Plot of induction of GAGT introns specifically in chRNA and polyA
(main)</li>
<li>spearman correlation distribution of GAGT introns compared to all
others (supplement)</li>
<li>A barplots of expression effect in chRNA and polyA, grouped by
whether induced exons are productive or unproductive. (main)</li>
<li>barplot of induced exons and proportion which are “poison” and why.
(Supplement, possibly main)</li>
<li>The druggability gene enrichment barplots comparing genes w/
ris-induced poison exons (not based on expression effects) (main)</li>
<li>the druggability gene enrichment barplot comparing genes w/
post-txnal downregulation (Supplement)</li>
<li>the polyA vs chRNA gene expression effect scatter, highlighting
various categories (Supplement)</li>
<li>polyA volcano at high dose, at low dose (Supplement)</li>
<li>dose response and model fits of handfuls of genes of interest</li>
<li>the splicing vs expression beta faceted by predicted splicing effect
(ie frame shift, in-frame PTC, frame-preserving, etc) (Supplement)</li>
</ul>
</div>
<div id="read-in-data" class="section level2">
<h2>Read in data</h2>
<pre class="r"><code>IntronAnnotatins &lt;- read_tsv(&quot;../data/IntronAnnotationsFromYang.tsv.gz&quot;) %&gt;%
  mutate(chrom = str_remove_all(chrom, &quot;chr&quot;)) %&gt;%
  mutate(Intron = paste(chrom, start, end, sep=&quot;:&quot;)) %&gt;%
  filter(!str_detect(SuperAnnotation, &quot;NoncodingGene&quot;))

small.molecule.dat.genes &lt;- read_tsv(&quot;../code/SmallMolecule/FitModels/polyA_genes.tsv.gz&quot;)

# Model fits using PSI using intron trios
small.molecule.dat.introns &lt;- read_tsv(&quot;../code/SmallMolecule/FitModels/polyA_GAGTIntrons_asPSI.tsv.gz&quot;) %&gt;%
  mutate(Intron = str_replace(junc, &quot;^chr(.+?):[+-]$&quot;, &quot;\\1&quot;)) %&gt;%
  left_join(IntronAnnotatins)


ClusterSignificance &lt;- Sys.glob(&quot;../code/SmallMolecule/leafcutter/ds/chRNA_risdiplam_*_cluster_significance.txt&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/SmallMolecule/leafcutter/ds/chRNA_risdiplam_(.+?)_cluster_significance.txt&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(read_tsv) %&gt;%
  bind_rows(.id=&quot;dose.nM&quot;)

EffectSizes &lt;- Sys.glob(&quot;../code/SmallMolecule/leafcutter/ds/chRNA_risdiplam_*_effect_sizes.txt&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/SmallMolecule/leafcutter/ds/chRNA_risdiplam_(.+?)_effect_sizes.txt&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(read_tsv) %&gt;%
  bind_rows(.id=&quot;dose.nM&quot;) %&gt;%
  mutate(cluster = str_replace(intron, &quot;^(.+?):.+?:.+?(:clu_.+$)&quot;, &quot;\\1\\2&quot;))

chRNA.splicing.dat &lt;- inner_join(EffectSizes, ClusterSignificance)



Translated.GAGT_CassetteExons &lt;- read_tsv(&quot;../code/SmallMolecule/CassetteExons/ExonsToTranslate.Translated.tsv.gz&quot;) %&gt;%
  dplyr::select(-chrom, -Strand) %&gt;%
  mutate(TranslationDiff = str_length(IncludedTranslation) - str_length(SkippedTranslation)) %&gt;%
  mutate(InFrame = str_length(IncludedTranslation)-str_length(SkippedTranslation)==(ExonStop_Cassette-ExonStart_Cassette)/3) %&gt;%
  mutate(ExonRemainder = (ExonStop_Cassette - ExonStart_Cassette)%%3) %&gt;%
  mutate(Effect = case_when(
    is.na(InFrame) ~ &quot;Cassette in UTR&quot;,
    InFrame ~ &quot;Preserves frame&quot;,
    ExonRemainder %in% c(1,2) ~ &quot;Frame shifting&quot;,
    InFrame == F ~ &quot;In-frame PTC&quot;,
  )) %&gt;%
  left_join(
    IntronAnnotatins %&gt;%
      unite(Intron, chrom, start, end, strand, sep=&quot;:&quot;) %&gt;%
      mutate(GAGTInt = paste0(&quot;chr&quot;, Intron))
    )

leafcutter.counts &lt;- read.table(&quot;../code/SmallMolecule/leafcutter/clustering/autosomes/leafcutter_perind_numers.counts.gz&quot;, header=T, sep=&#39; &#39;) %&gt;% as.matrix()

library(biomaRt)
ensembl = useMart(&quot;ensembl&quot;,dataset=&quot;hsapiens_gene_ensembl&quot;)

ensembl_to_symbols &lt;- getBM(attributes= c(&quot;ensembl_gene_id&quot;,&quot;hgnc_symbol&quot;),mart=ensembl)

Expression.tidyForDoseResponse &lt;- read_tsv(&quot;/project2/yangili1/bjf79/20211209_JingxinRNAseq/code/DoseResponseData/LCL/TidyExpressionDoseData.txt.gz&quot;) %&gt;%
  mutate(gene = str_replace(Geneid, &quot;(^.+?)\\..+?$&quot;, &quot;\\1&quot;)) %&gt;%
   left_join(ensembl_to_symbols, by=c(&quot;gene&quot;=&quot;ensembl_gene_id&quot;))

chRNA.DE.dat &lt;- read_tsv(&quot;../code/SmallMolecule/chRNA/DE.results.tsv.gz&quot;)</code></pre>
</div>
<div id="volcanoes" class="section level2">
<h2>volcanoes</h2>
<p>First I need due testing to get pvalue from dose response fits in
polyA data at the low and high dose</p>
<pre class="r"><code>polyA.GeneEffects &lt;- small.molecule.dat.genes %&gt;%
  filter(str_detect(param, &quot;Pred&quot;)) %&gt;%
  mutate(gene = str_replace(Geneid, &quot;(^.+?)\\..+?$&quot;, &quot;\\1&quot;)) %&gt;%
  separate(param, into=c(&quot;Dummy&quot;, &quot;dose.nM&quot;), convert=T, sep=&quot;_&quot;) %&gt;%
  dplyr::select(-Dummy)

GeneEffects.Changes &lt;- 
  inner_join(
    polyA.GeneEffects %&gt;%
      filter(dose.nM == 0) %&gt;%
      dplyr::select(-Geneid, -dose.nM),
    polyA.GeneEffects %&gt;%
      filter(!dose.nM == 0),
    by=c(&quot;gene&quot;),
    suffix=c(&quot;.expression.untreated&quot;, &quot;.expression.treated&quot;)
  ) %&gt;%
  distinct() %&gt;%
  mutate(Estimate.Expression.FC = Estimate.expression.treated - Estimate.expression.untreated) %&gt;%
  mutate(z = Estimate.Expression.FC/sqrt((SE.expression.treated + SE.expression.untreated))) %&gt;%
  mutate(abs.z = abs(z)) %&gt;%
  # filter(dose.nM == 100) %&gt;%
  mutate(polyA.P = 2*pnorm(abs.z, lower.tail = F)) %&gt;%
  group_by(dose.nM) %&gt;%
  add_count(gene) %&gt;%
  filter(n==1) %&gt;%
  mutate(FDR = qvalue(polyA.P)$qvalues) %&gt;%
  ungroup() %&gt;%
  left_join(ensembl_to_symbols, by=c(&quot;gene&quot;=&quot;ensembl_gene_id&quot;))

GeneEffects.Changes %&gt;%
  ggplot(aes(x=polyA.P)) +
  geom_histogram() +
  facet_wrap(~dose.nM)</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>GeneEffects.Changes %&gt;%
  filter(dose.nM==100 &amp; FDR&lt;0.1) %&gt;%
  arrange(polyA.P)</code></pre>
<pre><code># A tibble: 3 × 14
  Estimate.expression.un… SE.expression.u… gene  Geneid dose.nM Estimate.expres…
                    &lt;dbl&gt;            &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;    &lt;int&gt;            &lt;dbl&gt;
1                    3.05           0.0621 ENSG… ENSG0…     100            0.458
2                    7.25           0.0495 ENSG… ENSG0…     100            5.44 
3                    2.26           0.211  ENSG… ENSG0…     100            4.75 
# … with 8 more variables: SE.expression.treated &lt;dbl&gt;,
#   Estimate.Expression.FC &lt;dbl&gt;, z &lt;dbl&gt;, abs.z &lt;dbl&gt;, polyA.P &lt;dbl&gt;, n &lt;int&gt;,
#   FDR &lt;dbl&gt;, hgnc_symbol &lt;chr&gt;</code></pre>
<pre class="r"><code>SigLowDoseGenes &lt;- GeneEffects.Changes %&gt;%
  filter(dose.nM==100 &amp; FDR&lt;0.1) %&gt;%
  arrange(polyA.P) %&gt;% pull(gene)

Expression.tidyForDoseResponse %&gt;%
  filter(gene %in% SigLowDoseGenes | hgnc_symbol %in% c(&quot;HTT&quot;, &quot;SMN2&quot;, &quot;GALC&quot;, &quot;STAT1&quot;, &quot;KCNT2&quot;, &quot;FHOD3&quot;, &quot;SRSF10&quot;)) %&gt;%
  ggplot(aes(x=dose.nM, y=CPM, color=treatment)) +
  geom_line() +
  scale_y_continuous(trans=&quot;log2&quot;) +
  scale_x_continuous(trans=&quot;log1p&quot;, limits=c(0, 10000), breaks=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 1, 0.316, 0)) +
  facet_wrap(~hgnc_symbol, scales=&quot;free_y&quot;) +
  Rotate_x_labels</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, the test actually seems sort of reasonably calibrated at the high
dose, but is overly conservative at the low dose. This issue might stem
from the fact that the exact model i fit too (the log-logistic), might
not be the best choice of sigmoidal model, especially around the elbows
at low doses where an effect is just starting to be measurable. But
fixing that could take a bit of work. There may be other workarounds as
well to get a better calibrated test. Intuitively I feel what I’m doing
(fitting model, then using model to output prediction with standard
error at dose=100nM and dose=0nM) isn’t really using all the data very
efficiently. Well I guess I don’t really mind being overly conservative,
and given how the message of the paper doesn’t at all revolve around
this point. So rather than figure out how to better calibrate this test,
I think I’ll just be content with this methodology. Especially since the
few significant hits actually do seem quite believable when I look at
the full dose response set. And FOS is actually in interesting gene. I’m
surprised I have missed that thus far.</p>
<p>Ok, so now I will make the final looking volcano plots</p>
<pre class="r"><code>volcano.dat &lt;- bind_rows(
  chRNA.DE.dat %&gt;%
    separate(Geneid, into=c(&quot;Geneid&quot;, &quot;symbol&quot;), sep=&quot;\\_&quot;) %&gt;%
    mutate(gene = str_replace(Geneid, &quot;(^.+?)\\..+?$&quot;, &quot;\\1&quot;)) %&gt;%
    dplyr::select(logFC, PValue, FDR, gene, symbol, dose.nM=risdiplam_conc) %&gt;%
    mutate(dataset = &quot;chRNA&quot;),
  GeneEffects.Changes %&gt;%
    dplyr::select(logFC=Estimate.Expression.FC, PValue=polyA.P, gene, dose.nM, symbol=hgnc_symbol, FDR, Estimate.expression.untreated) %&gt;%
    mutate(dataset = &quot;polyA&quot;)
)

volcano.dat %&gt;% filter(dose.nM == 100 &amp; FDR&lt;0.1) %&gt;%
  arrange(dataset, FDR)</code></pre>
<pre><code># A tibble: 11 × 8
    logFC   PValue           FDR gene    symbol dose.nM dataset Estimate.expres…
    &lt;dbl&gt;    &lt;dbl&gt;         &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;    &lt;dbl&gt; &lt;chr&gt;              &lt;dbl&gt;
 1  2.22  7.19e-10 0.00000806    ENSG00… TEX21P     100 chRNA              NA   
 2  1.72  1.01e- 9 0.00000806    ENSG00… CAND2      100 chRNA              NA   
 3  0.998 1.95e- 7 0.00104       ENSG00… ANP32A     100 chRNA              NA   
 4  0.776 1.80e- 6 0.00647       ENSG00… GALC       100 chRNA              NA   
 5  1.03  2.02e- 6 0.00647       ENSG00… POLN       100 chRNA              NA   
 6  1.02  8.20e- 6 0.0218        ENSG00… C19or…     100 chRNA              NA   
 7  0.865 1.92e- 5 0.0438        ENSG00… WDR91      100 chRNA              NA   
 8 -0.448 4.82e- 5 0.0962        ENSG00… MZB1       100 chRNA              NA   
 9 -2.60  2.33e-13 0.00000000232 ENSG00… TTC12      100 polyA               3.05
10 -1.82  1.17e- 6 0.00583       ENSG00… PDXDC1     100 polyA               7.25
11  2.48  2.90e- 5 0.0963        ENSG00… FOS        100 polyA               2.26</code></pre>
<pre class="r"><code>volcano.dat %&gt;%
  filter(!is.na(FDR)) %&gt;%
ggplot( aes(x=logFC, y=-log10(PValue), color=FDR&lt;0.1)) +
  geom_point(alpha=0.1) +
  facet_wrap(dataset~dose.nM, scales = &quot;free_y&quot;)</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>volcano.dat %&gt;%
  filter(symbol %in% c(&quot;HTT&quot;, &quot;SMN2&quot;, &quot;GALC&quot;, &quot;STAT1&quot;, &quot;KCNT2&quot;, &quot;FHOD3&quot;)) %&gt;%
  arrange(dataset, dose.nM)</code></pre>
<pre><code># A tibble: 18 × 8
      logFC   PValue      FDR gene       symbol dose.nM dataset Estimate.expres…
      &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;      &lt;chr&gt;    &lt;dbl&gt; &lt;chr&gt;              &lt;dbl&gt;
 1  0.776   1.80e- 6 6.47e- 3 ENSG00000… GALC       100 chRNA              NA   
 2  0.410   2.10e- 3 6.06e- 1 ENSG00000… FHOD3      100 chRNA              NA   
 3  0.0236  8.14e- 1 9.23e- 1 ENSG00000… STAT1      100 chRNA              NA   
 4 -0.0218  8.56e- 1 9.41e- 1 ENSG00000… SMN2       100 chRNA              NA   
 5  0.998   1.62e- 7 1.52e- 5 ENSG00000… GALC      3160 chRNA              NA   
 6  0.493   5.22e- 4 5.93e- 3 ENSG00000… STAT1     3160 chRNA              NA   
 7  0.375   3.50e- 3 2.59e- 2 ENSG00000… FHOD3     3160 chRNA              NA   
 8  0.0800  5.08e- 1 8.74e- 1 ENSG00000… SMN2      3160 chRNA              NA   
 9 -0.266   4.08e- 1 1   e+ 0 ENSG00000… STAT1      100 polyA              10.9 
10 -0.204   7.90e- 1 1   e+ 0 ENSG00000… HTT        100 polyA               8.31
11 -0.00590 9.66e- 1 1   e+ 0 ENSG00000… SMN2       100 polyA               5.99
12 -1.94    4.02e- 3 1   e+ 0 ENSG00000… GALC       100 polyA               3.38
13 -1.06    1.64e- 3 1   e+ 0 ENSG00000… FHOD3      100 polyA               3.65
14 -4.36    1.42e-36 2.87e-34 ENSG00000… STAT1     3160 polyA              10.9 
15 -1.84    1.78e- 3 9.96e- 3 ENSG00000… HTT       3160 polyA               8.31
16 -0.294   9.95e- 2 1.90e- 1 ENSG00000… SMN2      3160 polyA               5.99
17 -2.26    7.45e- 5 6.72e- 4 ENSG00000… GALC      3160 polyA               3.38
18 -2.88    2.07e-18 1.36e-16 ENSG00000… FHOD3     3160 polyA               3.65</code></pre>
<pre class="r"><code>volcano.dat %&gt;%
  filter(!is.na(FDR)) %&gt;%
  mutate(Effect = if_else(FDR&lt;0.1, sign(logFC), NaN )) %&gt;%
  count(Effect, dose.nM, dataset) %&gt;%
  arrange(dataset, dose.nM, Effect)</code></pre>
<pre><code># A tibble: 12 × 4
   Effect dose.nM dataset     n
    &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;   &lt;int&gt;
 1     -1     100 chRNA       1
 2      1     100 chRNA       7
 3    NaN     100 chRNA   15970
 4     -1    3160 chRNA    2025
 5      1    3160 chRNA    1165
 6    NaN    3160 chRNA   12788
 7     -1     100 polyA       2
 8      1     100 polyA       1
 9    NaN     100 polyA    9964
10     -1    3160 polyA    1399
11      1    3160 polyA     780
12    NaN    3160 polyA    7769</code></pre>
<pre class="r"><code>volcano.dat %&gt;%
  filter(!is.na(FDR)) %&gt;%
ggplot( aes(x=logFC, y=-log10(PValue), color=FDR&lt;0.1)) +
  geom_point(alpha=0.5) +
  facet_wrap(dataset~dose.nM, scales = &quot;free_y&quot;)</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>volcano.dat %&gt;%
  filter(!is.na(FDR)) %&gt;%
  ggplot(aes(x=logFC)) +
  stat_ecdf() +
  geom_vline(xintercept = 0) +
  coord_cartesian(xlim=c(-2,2)) +
  facet_wrap(dataset~dose.nM, scales = &quot;free_y&quot;)</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>volcano.dat %&gt;%
  filter(!is.na(FDR)) %&gt;%
ggplot( aes(x=logFC, y=-log10(PValue), color=FDR&lt;0.1)) +
  geom_point(alpha=0.5) +
  facet_wrap(dataset~dose.nM, scales = &quot;free_y&quot;)</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-3-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>P.MA.count.labels &lt;- volcano.dat %&gt;%
  filter(!is.na(FDR)) %&gt;%
  filter(dataset == &quot;polyA&quot;) %&gt;%
  mutate(dose.nM = recode(dose.nM, `100`=&quot;100nM risdiplam&quot;, `3160`=&quot;3160nM risdiplam&quot;)) %&gt;%
  filter(FDR&lt;0.01) %&gt;%
  mutate(sign = sign(logFC)) %&gt;%
  mutate(UpOrDown = recode(sign, `1`=&quot;Up&quot;, `-1`=&quot;Down&quot;)) %&gt;%
  count(dose.nM, sign, UpOrDown, .drop=F) %&gt;%
  mutate(y = Inf * sign) %&gt;%
  mutate(label = str_glue(&quot;{n} {UpOrDown}&quot;))
  

P.MA &lt;- volcano.dat %&gt;%
  filter(!is.na(FDR)) %&gt;%
  filter(dataset == &quot;polyA&quot;) %&gt;%
  mutate(dose.nM = recode(dose.nM, `100`=&quot;100nM risdiplam&quot;, `3160`=&quot;3160nM risdiplam&quot;))  %&gt;%
ggplot( aes(x=Estimate.expression.untreated, y=logFC, color=FDR&lt;0.01)) +
  geom_point(alpha=0.5) +
  geom_hline(yintercept=0, linetype=&#39;dashed&#39;) +
  scale_color_manual(values=c(`TRUE`=&quot;red&quot;, `FALSE`=&quot;gray&quot;)) +
  geom_text(data=P.MA.count.labels, aes(y=y, label=label, vjust=sign), x=Inf, color=&quot;red&quot;, hjust=1) +
  facet_wrap(~dose.nM, scales = &quot;free_y&quot;) +
  labs(x=&quot;Expression in DMSO control (log2CPM)&quot;, y=&quot;log2FoldChange,\nestimated from\ndose-response curve fit&quot;)
P.MA</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-3-5.png" width="672" style="display: block; margin: auto;" /></p>
<div id="bar-plots-of-predicted-effects" class="section level3">
<h3>bar plots of predicted effects</h3>
<pre class="r"><code>#First see how my annotations match up with Gencode and Yang&#39;s annotations
Translated.GAGT_CassetteExons %&gt;%
  mutate(IsAnnotated = if_else(str_detect(SuperAnnotation,&quot;Annotated&quot;), &quot;Annotated by Gencode&quot;, &quot;Annotated by Yang&quot;)) %&gt;%
  replace_na(list(SemiSupergroupAnnotations=&quot;Unannotated by either&quot;, IsAnnotated=&quot;Unannotated by either&quot;)) %&gt;%
  count(Effect, IsAnnotated, SemiSupergroupAnnotations) %&gt;%
  ggplot(aes(x=Effect, y=n, fill=SemiSupergroupAnnotations)) +
    geom_col(position=&quot;stack&quot;) +
    facet_wrap(~IsAnnotated, scales=&quot;free_y&quot;, nrow = 3) +
    Rotate_x_labels +
  labs(title=&quot;Comparing my annotations to Gencode/Yang&quot;, x=&quot;My annotation by comparing translations&quot;, fill=&quot;Gencode/Yang&#39;s annotation&quot;)</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, I feel good enough about my annotations which I think were
slightly more careful than Yang’s but nonetheless mostly yield the same
functional result - that the vast majority of unannotated junctions will
be non-productive.</p>
<p>Now let’s plot something that I would want to include in
manuscript</p>
<pre class="r"><code>Translated.GAGT_CassetteExons$Effect %&gt;% unique()</code></pre>
<pre><code>[1] &quot;Frame shifting&quot;  &quot;In-frame PTC&quot;    &quot;Cassette in UTR&quot; &quot;Preserves frame&quot;</code></pre>
<pre class="r"><code>P.bar.dat &lt;- Translated.GAGT_CassetteExons %&gt;%
   mutate(IsAnnotated = if_else(str_detect(SuperAnnotation,&quot;Annotated&quot;), &quot;Annotated by Gencode&quot;, &quot;Unannotated&quot;, missing=&quot;Unannotated&quot;)) %&gt;%
  mutate(MergedEffect = case_when(
    IsAnnotated == &quot;Annotated by Gencode&quot; &amp; SemiSupergroupAnnotations==&quot;basic tag&quot; ~ &quot;Productive&quot;,
    IsAnnotated == &quot;Annotated by Gencode&quot; &amp; !SemiSupergroupAnnotations==&quot;basic tag&quot; ~ &quot;NMD&quot;,
    TRUE ~ Effect
  )) %&gt;%
  mutate(ColorEffect = if_else(MergedEffect %in% c(&quot;Productive&quot;, &quot;Preserves frame&quot;, &quot;Cassette in UTR&quot;), &quot;Productive&quot;, &quot;NMD&quot;) )

#Check MYB
P.bar.dat %&gt;%
  filter(str_detect(gene, &quot;ENSG00000118513&quot;))</code></pre>
<pre><code># A tibble: 1 × 24
  GAGTInt    ExonStart_Upstr… ExonStop_Upstre… ExonStart_Casse… ExonStop_Casset…
  &lt;chr&gt;                 &lt;dbl&gt;            &lt;dbl&gt;            &lt;dbl&gt;            &lt;dbl&gt;
1 chr6:1351…        135198907        135199050        135199525        135199581
# … with 19 more variables: ExonStart_Downstream &lt;dbl&gt;,
#   ExonStop_Downstream &lt;dbl&gt;, gene &lt;chr&gt;, transcript &lt;chr&gt;,
#   SkipJuncName &lt;chr&gt;, IncludedTranslation &lt;chr&gt;, SkippedTranslation &lt;chr&gt;,
#   TranslationDiff &lt;int&gt;, InFrame &lt;lgl&gt;, ExonRemainder &lt;dbl&gt;, Effect &lt;chr&gt;,
#   Intron &lt;chr&gt;, NewAnnotation &lt;chr&gt;, symbol &lt;chr&gt;, SuperAnnotation &lt;chr&gt;,
#   SemiSupergroupAnnotations &lt;chr&gt;, IsAnnotated &lt;chr&gt;, MergedEffect &lt;chr&gt;,
#   ColorEffect &lt;chr&gt;</code></pre>
<pre class="r"><code>P.bar &lt;- P.bar.dat %&gt;%
  count(MergedEffect, ColorEffect, IsAnnotated) %&gt;%
  mutate(MergedEffect=factor(MergedEffect, levels=c(&quot;Productive&quot;, &quot;Cassette in UTR&quot;,&quot;Preserves frame&quot;, &quot;NMD&quot;, &quot;In-frame PTC&quot;, &quot;Frame shifting&quot;)))  %&gt;%
  mutate(IsAnnotated = recode(IsAnnotated, &quot;Annotated by Gencode&quot;=&quot;Annotated&quot;)) %&gt;%
  ggplot(aes(y=IsAnnotated, x=n, fill=MergedEffect, color=ColorEffect)) +
    geom_col(position=&quot;stack&quot;, size=3) +
   geom_text(aes(label=n), color=&#39;black&#39;, position=position_stack(vjust=0.5)) +
    facet_wrap(~IsAnnotated, nrow=2, scales = &quot;free&quot;) +
    scale_color_manual(values=c(
      &quot;Productive&quot; = &quot;#1f78b4&quot;,
      &quot;NMD&quot; = &quot;#e31a1c&quot;
    )) +
   scale_fill_manual(values=c(
     &quot;Productive&quot;=&quot;#1f78b4&quot;,
     &quot;NMD&quot;=&quot;#e31a1c&quot;,
     &quot;Frame shifting&quot;=&quot;#fb6a4a&quot;,
     &quot;Cassette in UTR&quot;=&quot;#deebf7&quot;,
     &quot;In-frame PTC&quot;=&quot;#fcbba1&quot;,
      &quot;Preserves frame&quot;=&quot;#a6cee3&quot;
   ),
   labels=c(
      &quot;Productive&quot;=&quot;In annotated productive transcript&quot;,
     &quot;NMD&quot;=&quot;In annotated NMD transcript&quot;,
     &quot;Frame shifting&quot;=&quot;Frame shifting&quot;,
     &quot;Cassette in UTR&quot;=&quot;Cassette in UTR&quot;,
     &quot;In-frame PTC&quot;=&quot;In-frame PTC&quot;,
      &quot;Preserves frame&quot;=&quot;Preserves frame&quot;
   )) +
  scale_x_continuous(expand=c(0,0)) +
    Rotate_x_labels +
    guides(fill = guide_legend(order = 2),
         color = guide_legend(order = 1)) +
  theme(legend.box=&quot;vertical&quot;, legend.position = &quot;bottom&quot;) +
    theme(
    strip.background = element_blank(),
    strip.text.x = element_blank()
  ) +
  labs(title=&quot;Predicted transcript effects of induced cassete exons&quot;, x=&quot;Number risdiplam-induced cassette exons&quot;, fill=&quot;Reason&quot;, y=NULL, color=&quot;Transcript effect&quot;)

P.bar</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>P.bar.dat %&gt;%
  filter(str_detect(gene, &quot;ENSG00000188529&quot;))</code></pre>
<pre><code># A tibble: 0 × 24
# … with 24 variables: GAGTInt &lt;chr&gt;, ExonStart_Upstream &lt;dbl&gt;,
#   ExonStop_Upstream &lt;dbl&gt;, ExonStart_Cassette &lt;dbl&gt;, ExonStop_Cassette &lt;dbl&gt;,
#   ExonStart_Downstream &lt;dbl&gt;, ExonStop_Downstream &lt;dbl&gt;, gene &lt;chr&gt;,
#   transcript &lt;chr&gt;, SkipJuncName &lt;chr&gt;, IncludedTranslation &lt;chr&gt;,
#   SkippedTranslation &lt;chr&gt;, TranslationDiff &lt;int&gt;, InFrame &lt;lgl&gt;,
#   ExonRemainder &lt;dbl&gt;, Effect &lt;chr&gt;, Intron &lt;chr&gt;, NewAnnotation &lt;chr&gt;,
#   symbol &lt;chr&gt;, SuperAnnotation &lt;chr&gt;, SemiSupergroupAnnotations &lt;chr&gt;, …</code></pre>
<p>I want to make one more version of the bar plot because it might more
compact to manually annotate labels over the sub-bars instead of using
fill colors</p>
<pre class="r"><code>P.bar.simpler &lt;- P.bar.dat %&gt;%
  count(MergedEffect, ColorEffect, IsAnnotated) %&gt;%
  mutate(MergedEffect=factor(MergedEffect, levels=c(&quot;Productive&quot;, &quot;Cassette in UTR&quot;,&quot;Preserves frame&quot;, &quot;NMD&quot;, &quot;In-frame PTC&quot;, &quot;Frame shifting&quot;)))  %&gt;%
  mutate(IsAnnotated = recode(IsAnnotated, &quot;Annotated by Gencode&quot;=&quot;Annotated&quot;)) %&gt;%
  ggplot(aes(y=IsAnnotated, x=n, fill=MergedEffect)) +
    geom_col(position=&quot;stack&quot;, color=&#39;black&#39;) +
   # geom_text(aes(label=str_glue(&quot;{MergedEffect}; n={n}&quot;)), color=&#39;black&#39;, position=position_stack(vjust=0.5)) +
    facet_wrap(~IsAnnotated, nrow=2, scales = &quot;free&quot;) +
   scale_fill_manual(values=c(
     &quot;Productive&quot;=&quot;#1f78b4&quot;,
     &quot;NMD&quot;=&quot;#e31a1c&quot;,
     &quot;Frame shifting&quot;=&quot;#fb9a99&quot;,
     &quot;Cassette in UTR&quot;=&quot;#a6cee3&quot;,
     &quot;In-frame PTC&quot;=&quot;#fb9a99&quot;,
      &quot;Preserves frame&quot;=&quot;#a6cee3&quot;
   ),
   labels=c(
      &quot;Productive&quot;=&quot;In annotated productive transcript&quot;,
     &quot;NMD&quot;=&quot;In annotated NMD transcript&quot;,
     &quot;Frame shifting&quot;=&quot;Frame shifting&quot;,
     &quot;Cassette in UTR&quot;=&quot;Cassette in UTR&quot;,
     &quot;In-frame PTC&quot;=&quot;In-frame PTC&quot;,
      &quot;Preserves frame&quot;=&quot;Preserves frame&quot;
   )) +
  scale_x_continuous(expand=c(0,0)) +
    Rotate_x_labels +
    guides(fill = guide_legend(order = 2),
         color = guide_legend(order = 1)) +
  theme(legend.position = &quot;none&quot;) +
    theme(
    strip.background = element_blank(),
    strip.text.x = element_blank()
  ) +
  labs(title=&quot;Predicted transcript effects of induced cassete exons&quot;, x=&quot;Number risdiplam-induced cassette exons&quot;, fill=&quot;Reason&quot;, y=NULL, color=&quot;Transcript effect&quot;)
P.bar.simpler</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>One more version… with fixed scale… borrowed code from <a
href="https://github.com/slowkow/ggrepel/issues/161">here</a></p>
<pre class="r"><code>position_stack_and_nudge &lt;- function(x = 0, y = 0, vjust = 1, reverse = FALSE) {
  ggproto(NULL, PositionStackAndNudge,
    x = x,
    y = y,
    vjust = vjust,
    reverse = reverse
  )
}

#&#39; @rdname ggplot2-ggproto
#&#39; @format NULL
#&#39; @usage NULL
#&#39; @noRd
PositionStackAndNudge &lt;- ggproto(&quot;PositionStackAndNudge&quot;, PositionStack,
  x = 0,
  y = 0,

  setup_params = function(self, data) {
    c(
        list(x = self$x, y = self$y),
        ggproto_parent(PositionStack, self)$setup_params(data)
    )
  },

  compute_layer = function(self, data, params, panel) {
    # operate on the stacked positions (updated in August 2020)
    data = ggproto_parent(PositionStack, self)$compute_layer(data, params, panel)

    x_orig &lt;- data$x
    y_orig &lt;- data$y
    # transform only the dimensions for which non-zero nudging is requested
    if (any(params$x != 0)) {
      if (any(params$y != 0)) {
        data &lt;- transform_position(data, function(x) x + params$x, function(y) y + params$y)
      } else {
        data &lt;- transform_position(data, function(x) x + params$x, NULL)
      }
    } else if (any(params$y != 0)) {
      data &lt;- transform_position(data, function(x) x, function(y) y + params$y)
    }
    data$nudge_x &lt;- data$x
    data$nudge_y &lt;- data$y
    data$x &lt;- x_orig
    data$y &lt;- y_orig

    data
  },

  compute_panel = function(self, data, params, scales) {
      ggproto_parent(PositionStack, self)$compute_panel(data, params, scales)
  }
)</code></pre>
<pre class="r"><code>P.bar.simpler.fixed &lt;- P.bar.dat %&gt;%
  count(MergedEffect, ColorEffect, IsAnnotated) %&gt;%
  mutate(MergedEffect=factor(MergedEffect, levels=c(&quot;Productive&quot;, &quot;Cassette in UTR&quot;,&quot;Preserves frame&quot;, &quot;NMD&quot;, &quot;In-frame PTC&quot;, &quot;Frame shifting&quot;)))  %&gt;%
  mutate(IsAnnotated = recode(IsAnnotated, &quot;Annotated by Gencode&quot;=&quot;Annotated&quot;)) %&gt;%
  ggplot(aes(y=IsAnnotated, x=n, fill=MergedEffect)) +
    geom_col(position=&quot;stack&quot;, color=&#39;black&#39;) +
    # geom_text(aes(label=n), color=&#39;black&#39;, position=position_stack(vjust=0.5)) +
    # geom_text_repel(aes(label=MergedEffect), color=&#39;black&#39;, position = position_stack_and_nudge(vjust = 0.5, y = 0.2),
    #   direction = &#39;x&#39;) +
   scale_fill_manual(values=c(
     &quot;Productive&quot;=&quot;#1f78b4&quot;,
     &quot;NMD&quot;=&quot;#e31a1c&quot;,
     &quot;Frame shifting&quot;=&quot;#fb9a99&quot;,
     &quot;Cassette in UTR&quot;=&quot;#a6cee3&quot;,
     &quot;In-frame PTC&quot;=&quot;#fb9a99&quot;,
      &quot;Preserves frame&quot;=&quot;#a6cee3&quot;
   ),
   labels=c(
      &quot;Productive&quot;=&quot;In annotated productive transcript&quot;,
     &quot;NMD&quot;=&quot;In annotated NMD transcript&quot;,
     &quot;Frame shifting&quot;=&quot;Frame shifting&quot;,
     &quot;Cassette in UTR&quot;=&quot;Cassette in UTR&quot;,
     &quot;In-frame PTC&quot;=&quot;In-frame PTC&quot;,
      &quot;Preserves frame&quot;=&quot;Preserves frame&quot;
   )) +
  scale_x_continuous(expand=c(0,0)) +
    Rotate_x_labels +
    guides(fill = guide_legend(order = 2),
         color = guide_legend(order = 1)) +
  theme(legend.position = &quot;none&quot;) +
    theme(
    strip.background = element_blank(),
    strip.text.x = element_blank()
  ) +
  labs(title=&quot;Predicted transcript effects of induced cassete exons&quot;, x=&quot;Number risdiplam-induced cassette exons&quot;, fill=&quot;Reason&quot;, y=NULL, color=&quot;Transcript effect&quot;) +
    facet_wrap(~IsAnnotated, nrow=2, scales = &quot;free_y&quot;)

P.bar.simpler.fixed</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="a-barplots-of-expression-effect-in-chrna-and-polya"
class="section level3">
<h3>A barplots of expression effect in chRNA and polyA</h3>
<pre class="r"><code>P.boxplot.dat &lt;- P.bar.dat %&gt;%
  dplyr::select(-symbol) %&gt;%
  mutate(gene = str_replace(gene, &quot;(^.+?)\\..+?$&quot;, &quot;\\1&quot;)) %&gt;%
  left_join(
    volcano.dat %&gt;%
      filter(dose.nM == 3160) %&gt;%
      distinct(gene, dose.nM, dataset, .keep_all=T) %&gt;%
      dplyr::select(gene, dose.nM, dataset, logFC) %&gt;%
      pivot_wider(names_from=&quot;dataset&quot;, values_from=&quot;logFC&quot;),
    by=&quot;gene&quot;
  ) %&gt;%
  left_join(ensembl_to_symbols, by=c(&quot;gene&quot;=&quot;ensembl_gene_id&quot;)) %&gt;%
  # filter(hgnc_symbol %in% c(&quot;HTT&quot;, &quot;SMN2&quot;, &quot;GALC&quot;, &quot;STAT1&quot;, &quot;KCNT2&quot;, &quot;FHOD3&quot;, &quot;FOXM1))
  dplyr::select(GAGTInt, ExonStart_Cassette, ExonStop_Cassette, gene, hgnc_symbol, chRNA, polyA, ColorEffect, MergedEffect, IsAnnotated) %&gt;%
  gather(&quot;dataset&quot;, &quot;log2FC&quot;, chRNA, polyA)

P.box.small &lt;- P.boxplot.dat %&gt;%
  mutate(ColorEffect = factor(ColorEffect, levels=c(&quot;Productive&quot;, &quot;NMD&quot;))) %&gt;%
  ggplot(aes(y=log2FC, x=ColorEffect, color=ColorEffect)) +
  geom_hline(yintercept=0) +
  geom_boxplot(outlier.shape=NA, alpha=1, fill=&quot;white&quot;) +
  geom_segment(x=&quot;Productive&quot;, xend=&quot;NMD&quot;, y=2.5, yend=2.5, color=&#39;black&#39;) +
  geom_point(position=position_jitterdodge(jitter.width=0.5), alpha=0.3) +
  geom_text(data = . %&gt;%
              group_by(dataset) %&gt;%
              do(w = wilcox.test(log2FC~ColorEffect, data=., paired=FALSE)) %&gt;%
              summarise(dataset, Wilcox = format.pval(w$p.value, digits=3)) %&gt;%
              ungroup() %&gt;%
              mutate(label=str_glue(&quot;P={Wilcox}&quot;)),
            aes(label=label, group=NULL),
            x=1.5, y=2.5, color=&#39;black&#39;, vjust=-1) +
  scale_color_manual(values=c( &quot;#1f78b4&quot;,&quot;#e31a1c&quot;)) +
  facet_wrap(~dataset) +
  labs(y=expression(&quot;Host gene &quot;*log[&quot;2&quot;]*&quot;FC&quot;), x=&quot;Predicted transcript effect&quot;) +
  Rotate_x_labels +
  theme(legend.position=&quot;none&quot;)

P.box.small</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>P.box &lt;- P.boxplot.dat %&gt;%
  mutate(ColorEffect = factor(ColorEffect, levels=c(&quot;Productive&quot;, &quot;NMD&quot;))) %&gt;%
  ggplot(aes(y=log2FC, x=ColorEffect, color=interaction(IsAnnotated, ColorEffect), group=interaction(IsAnnotated, ColorEffect))) +
  geom_hline(yintercept=0) +
  geom_boxplot(outlier.shape=NA, alpha=1, fill=&quot;white&quot;) +
  geom_segment(x=&quot;Productive&quot;, xend=&quot;NMD&quot;, y=2.5, yend=2.5, color=&#39;black&#39;) +
  geom_point(position=position_jitterdodge(jitter.width=0.5), alpha=0.3) +
  geom_text(data = . %&gt;%
              group_by(dataset) %&gt;%
              do(w = wilcox.test(log2FC~ColorEffect, data=., paired=FALSE)) %&gt;%
              summarise(dataset, Wilcox = format.pval(w$p.value, digits=3)) %&gt;%
              ungroup() %&gt;%
              mutate(label=str_glue(&quot;P={Wilcox}&quot;)),
            aes(label=label, group=NULL),
            x=1.5, y=2.5, color=&#39;black&#39;, vjust=-1) +
  scale_color_manual(values=c( &quot;#1f78b4&quot;, &quot;#a6cee3&quot;, &quot;#e31a1c&quot;, &quot;#fb9a99&quot;)) +
  facet_wrap(~dataset) +
  labs(y=expression(&quot;Host gene &quot;*log[&quot;2&quot;]*&quot;FC&quot;), x=&quot;Predicted transcript effect&quot;) +
  Rotate_x_labels +
  theme(legend.position=&quot;none&quot;)
P.box</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Let’s quickly check some genes of interest in this set of
NMD-predicted cassette exon targets</p>
<pre class="r"><code>P.boxplot.dat %&gt;%
  filter(ColorEffect == &quot;NMD&quot;) %&gt;%
  filter(hgnc_symbol %in% c(&quot;HTT&quot;, &quot;SMN2&quot;, &quot;GALC&quot;, &quot;STAT1&quot;, &quot;KCNT2&quot;, &quot;FHOD3&quot;, &quot;FOXM1&quot;, &quot;FOS&quot;, &quot;PIEZO2&quot;, &quot;MYB&quot;, &quot;SRSF10&quot;))</code></pre>
<pre><code># A tibble: 12 × 10
   GAGTInt       ExonStart_Casse… ExonStop_Casset… gene  hgnc_symbol ColorEffect
   &lt;chr&gt;                    &lt;dbl&gt;            &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;       &lt;chr&gt;      
 1 chr12:285966…          2861378          2861412 ENSG… FOXM1       NMD        
 2 chr18:107021…         10702545         10702669 ENSG… PIEZO2      NMD        
 3 chr18:109112…         10963754         10963891 ENSG… PIEZO2      NMD        
 4 chr2:1909770…        190978527        190978606 ENSG… STAT1       NMD        
 5 chr4:3213736…          3213621          3213736 ENSG… HTT         NMD        
 6 chr6:1351995…        135199525        135199581 ENSG… MYB         NMD        
 7 chr12:285966…          2861378          2861412 ENSG… FOXM1       NMD        
 8 chr18:107021…         10702545         10702669 ENSG… PIEZO2      NMD        
 9 chr18:109112…         10963754         10963891 ENSG… PIEZO2      NMD        
10 chr2:1909770…        190978527        190978606 ENSG… STAT1       NMD        
11 chr4:3213736…          3213621          3213736 ENSG… HTT         NMD        
12 chr6:1351995…        135199525        135199581 ENSG… MYB         NMD        
# … with 4 more variables: MergedEffect &lt;chr&gt;, IsAnnotated &lt;chr&gt;,
#   dataset &lt;chr&gt;, log2FC &lt;dbl&gt;</code></pre>
<pre class="r"><code>P.boxplot.dat %&gt;%
  filter(ColorEffect == &quot;NMD&quot;) %&gt;%
  pull(hgnc_symbol) %&gt;% unique()</code></pre>
<pre><code>  [1] &quot;RERE&quot;       &quot;PIK3CD&quot;     &quot;SPEN&quot;       &quot;FUCA1&quot;      &quot;TOE1&quot;      
  [6] &quot;NSUN4&quot;      &quot;IL12RB2&quot;    &quot;COL24A1&quot;    &quot;DIPK1A&quot;     &quot;POGZ&quot;      
 [11] &quot;VAMP4&quot;      &quot;COP1&quot;       &quot;SOAT1&quot;      &quot;DHX9&quot;       &quot;NIBAN1&quot;    
 [16] &quot;PLA2G4A&quot;    &quot;DENND1B&quot;    &quot;MDM4&quot;       &quot;INTS7&quot;      &quot;MLLT10&quot;    
 [21] &quot;ARHGAP12&quot;   &quot;MARCHF8&quot;    &quot;VDAC2&quot;      &quot;MMS19&quot;      &quot;DNMBP&quot;     
 [26] &quot;NT5C2&quot;      &quot;PDCD4&quot;      &quot;CASP7&quot;      &quot;WDR11&quot;      &quot;DENND5A&quot;   
 [31] &quot;BTBD10&quot;     &quot;CSTF3&quot;      &quot;APIP&quot;       &quot;MARK2&quot;      &quot;UVRAG&quot;     
 [36] &quot;EMSY&quot;       &quot;ALG8&quot;       &quot;HIKESHI&quot;    &quot;CEP57&quot;      &quot;CASP4&quot;     
 [41] &quot;SIK3&quot;       &quot;ERC1&quot;       &quot;FOXM1&quot;      &quot;PLEKHG6&quot;    &quot;KIF21A&quot;    
 [46] &quot;GXYLT1&quot;     &quot;PPHLN1&quot;     &quot;TMBIM6&quot;     &quot;GTSF1&quot;      &quot;DGKA&quot;      
 [51] &quot;PLXNC1&quot;     &quot;PPP1CC&quot;     &quot;HECTD4&quot;     &quot;RBM19&quot;      &quot;MED13L&quot;    
 [56] &quot;SPPL3&quot;      &quot;ZCCHC8&quot;     &quot;DDX51&quot;      &quot;POLR1D&quot;     &quot;SUPT20H&quot;   
 [61] &quot;COG3&quot;       &quot;FNDC3A&quot;     &quot;SPRYD7&quot;     &quot;TDRD3&quot;      &quot;MZT1&quot;      
 [66] &quot;MYCBP2&quot;     &quot;BIVM-ERCC5&quot; &quot;ERCC5&quot;      &quot;RAB2B&quot;      &quot;SOS2&quot;      
 [71] &quot;CDKL1&quot;      &quot;PPP2R5E&quot;    &quot;COQ6&quot;       &quot;ITPK1&quot;      &quot;NIPA1&quot;     
 [76] &quot;SNAP23&quot;     &quot;TCF12&quot;      &quot;CSNK1G1&quot;    &quot;&quot;           &quot;CLPX&quot;      
 [81] &quot;SCAPER&quot;     &quot;PDE8A&quot;      &quot;PDXDC1&quot;     &quot;ABCC1&quot;      &quot;TNRC6A&quot;    
 [86] &quot;CYLD&quot;       &quot;OGFOD1&quot;     &quot;SLC7A6&quot;     &quot;SF3B3&quot;      &quot;PIEZO1&quot;    
 [91] &quot;SCO1&quot;       &quot;BLTP2&quot;      &quot;NSRP1&quot;      &quot;UTP6&quot;       &quot;MYO1D&quot;     
 [96] &quot;HDAC5&quot;      &quot;LUC7L3&quot;     &quot;TRIM25&quot;     &quot;DDX42&quot;      &quot;GRB2&quot;      
[101] &quot;PIEZO2&quot;     &quot;KDSR&quot;       &quot;AKAP8L&quot;     &quot;DDX49&quot;      &quot;FBL&quot;       
[106] &quot;SNRPD2&quot;     &quot;BICRA&quot;      &quot;ADAM17&quot;     &quot;LDAH&quot;       &quot;ATAD2B&quot;    
[111] &quot;MRPL33&quot;     &quot;BABAM2&quot;     &quot;MEMO1&quot;      &quot;FAM161A&quot;    &quot;GCFC2&quot;     
[116] &quot;KCMF1&quot;      &quot;C2orf49&quot;    &quot;DARS1&quot;      &quot;MARCHF7&quot;    &quot;TANK&quot;      
[121] &quot;DCAF17&quot;     &quot;STAT1&quot;      &quot;ANKRD44&quot;    &quot;TYW5&quot;       &quot;STRADB&quot;    
[126] &quot;IDH1&quot;       &quot;COPS7B&quot;     &quot;TASP1&quot;      &quot;XRN2&quot;       &quot;PIGU&quot;      
[131] &quot;DNTTIP1&quot;    &quot;SLC9A8&quot;     &quot;PTPN1&quot;      &quot;IFNGR2&quot;     &quot;TNRC6B&quot;    
[136] &quot;SLC25A17&quot;   &quot;EP300&quot;      &quot;NUP50&quot;      &quot;TAMM41&quot;     &quot;GOLGA4&quot;    
[141] &quot;LARS2&quot;      &quot;PXK&quot;        &quot;CBLB&quot;       &quot;CD47&quot;       &quot;CIP2A&quot;     
[146] &quot;TMEM39A&quot;    &quot;SLC41A3&quot;    &quot;DNAJC13&quot;    &quot;RNF13&quot;      &quot;TFRC&quot;      
[151] &quot;NELFA&quot;      &quot;HTT&quot;        &quot;ARAP2&quot;      &quot;SEC31A&quot;     &quot;NFKB1&quot;     
[156] &quot;IRF2&quot;       &quot;EXOC3&quot;      &quot;LPCAT1&quot;     &quot;CCT5&quot;       &quot;SKP2&quot;      
[161] &quot;OXCT1&quot;      &quot;ERCC8&quot;      &quot;PIK3R1&quot;     &quot;GTF2H2&quot;     &quot;ANKRA2&quot;    
[166] &quot;TBCA&quot;       &quot;TENT2&quot;      &quot;DHFR&quot;       &quot;MCTP1&quot;      &quot;FAM174A&quot;   
[171] &quot;FER&quot;        &quot;HSD17B4&quot;    &quot;SNX24&quot;      &quot;DIAPH1&quot;     &quot;JARID2&quot;    
[176] &quot;C6orf62&quot;    &quot;KIFC1&quot;      &quot;CUL7&quot;       &quot;EFHC1&quot;      &quot;RARS2&quot;     
[181] &quot;MDN1&quot;       &quot;USP45&quot;      &quot;CRYBG1&quot;     &quot;RPF2&quot;       &quot;REV3L&quot;     
[186] &quot;MYB&quot;        &quot;WDR27&quot;      &quot;HYCC1&quot;      &quot;IGF2BP3&quot;    &quot;AVL9&quot;      
[191] &quot;NT5C3A&quot;     &quot;ELMO1&quot;      &quot;PSMA2&quot;      &quot;ZNF138&quot;     &quot;CPSF4&quot;     
[196] &quot;MDFIC&quot;      &quot;SSBP1&quot;      &quot;MCPH1&quot;      &quot;PCM1&quot;       &quot;CHRNA6&quot;    
[201] &quot;PRKDC&quot;      &quot;PDE7A&quot;      &quot;MTERF3&quot;     &quot;UBR5&quot;       &quot;NUDCD1&quot;    
[206] &quot;TAF2&quot;       &quot;FOCAD&quot;      &quot;TJP2&quot;       &quot;GNG10&quot;      &quot;FBXW2&quot;     
[211] &quot;DENND1A&quot;    &quot;PBX3&quot;       &quot;DPM2&quot;       &quot;FNBP1&quot;      &quot;RALGDS&quot;    
[216] &quot;EHMT1&quot;     </code></pre>
</div>
<div id="polya-vs-chrna-gene-expression-effect-scatter"
class="section level3">
<h3>polyA vs chRNA gene expression effect scatter</h3>
<pre class="r"><code>Threshold &lt;- log2(1.5)

Scatter.dat &lt;- volcano.dat %&gt;%
  filter(dose.nM == 3160) %&gt;%
  distinct(gene, dose.nM, dataset, .keep_all=T) %&gt;%
  dplyr::select(gene, dose.nM, dataset, logFC, PValue, FDR) %&gt;%
  pivot_wider(names_from=&quot;dataset&quot;, values_from=c(&quot;logFC&quot;, &quot;PValue&quot;, &quot;FDR&quot;)) %&gt;%
  mutate(Category = case_when(
    # FDR_polyA &lt; 0.1 &amp; !(FDR_chRNA &lt; 0.1 &amp; sign(logFC_chRNA)==sign(logFC_polyA)) ~ &quot;polyA-specific change&quot;,
    FDR_polyA &lt;0.1 &amp; FDR_chRNA &lt;0.1 &amp; logFC_polyA-logFC_chRNA&lt;Threshold &amp; logFC_polyA-logFC_chRNA &gt; -Threshold ~ &quot;txn&quot;,
    FDR_polyA &lt; 0.1 &amp; logFC_polyA-logFC_chRNA&gt;Threshold ~ &quot;post-txn up&quot;,
    FDR_polyA &lt; 0.1 &amp; logFC_polyA-logFC_chRNA &lt; -Threshold ~ &quot;post-txn down&quot;,
    TRUE ~ &quot;Other&quot;
  ))

Scatter.dat %&gt;%
  filter(!is.na(logFC_polyA) &amp; !is.na(logFC_chRNA)) %&gt;%
  count(Category)</code></pre>
<pre><code># A tibble: 4 × 2
  Category          n
  &lt;chr&gt;         &lt;int&gt;
1 Other          6928
2 post-txn down   628
3 post-txn up     340
4 txn             544</code></pre>
<pre class="r"><code>Enrichment.test &lt;- Scatter.dat %&gt;%
  filter(!is.na(logFC_polyA) &amp; !is.na(logFC_chRNA)) %&gt;%
  left_join(
    P.boxplot.dat %&gt;%
      dplyr::select(gene, ColorEffect) %&gt;%
      filter(ColorEffect == &quot;NMD&quot;),
    by=&quot;gene&quot;
    ) %&gt;%
  mutate(IsPostTxn = Category == &quot;post-txn down&quot;) %&gt;%
  mutate(IsNMD = ColorEffect == &quot;NMD&quot;) %&gt;%
  replace_na(list(IsNMD=F)) %&gt;%
  count(IsPostTxn, IsNMD) %&gt;%
  pivot_wider(names_from=&quot;IsPostTxn&quot;, values_from=&quot;n&quot;) %&gt;%
  column_to_rownames(&quot;IsNMD&quot;) %&gt;%
  fisher.test()

library(ggnewscale)

P.scatter &lt;- Scatter.dat %&gt;%
  ggplot(aes(x=logFC_chRNA, y=logFC_polyA)) +
  geom_abline(color=&#39;gray&#39;, slope=1, intercept=0) +
  geom_point(alpha=0.3, aes(color=Category)) +
  scale_color_manual(
    values=c(
      &quot;Other&quot;=&quot;#bdbdbd&quot;,
      &quot;post-txn up&quot;=&quot;#6a3d9a&quot;,
      &quot;post-txn down&quot;=&quot;#e31a1c&quot;,
      &quot;txn&quot;=&quot;#b15928&quot;
    ),
    labels=c(
      &quot;Other&quot;=&quot;Unchanged; n=6928)&quot;,
      &quot;post-txn up&quot;=&quot;Post txn up; n=340&quot;,
      &quot;post-txn down&quot;=&quot;Post txn down; n=628&quot;,
      &quot;txn&quot;=&quot;Txn change; n=544&quot;
    ),
    name=&quot;Expression change type&quot;
  ) +
  guides(colour = guide_legend(override.aes = list(alpha = 1))) +
  new_scale_color() +
  geom_point(
    data = Scatter.dat %&gt;%
      inner_join(
        P.boxplot.dat %&gt;%
        dplyr::select(gene, ColorEffect) %&gt;%
          filter(ColorEffect == &quot;NMD&quot;),
      by=&quot;gene&quot;
      ) %&gt;%
      distinct(gene, .keep_all=T),
    aes(color=&#39;NMD targets predicted from splicing changes&#39;),
    fill=&quot;#e31a1c&quot;,
    shape=21
  ) +
  scale_color_manual(values=c(&#39;NMD targets predicted from splicing changes&#39;=&quot;black&quot;), labels=c(&quot;NMD targets predicted from splicing changes&quot;=str_wrap(&quot;NMD targets predicted from splicing changes&quot;, 20)), name=NULL) +
  scale_x_continuous(limits=c(-3,3)) +
  scale_y_continuous(limits=c(-3,3)) +
  labs(x=expression(&quot;chRNA &quot;*log[&quot;2&quot;]*&quot;FC&quot;), y=expression(&quot;polyA &quot;*log[&quot;2&quot;]*&quot;FC&quot;), caption=str_wrap(str_glue(&quot;post-txn down 14.9 fold enriched for set of NMD targets predicted from splicing, P&lt;2x10-16, hypergeometric test&quot;),40)) +
  coord_fixed() +
  theme_bw()

P.scatter</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Scatter.dat %&gt;%
  filter(gene == &quot;ENSG00000188529&quot;)</code></pre>
<pre><code># A tibble: 1 × 9
  gene       dose.nM logFC_chRNA logFC_polyA PValue_chRNA PValue_polyA FDR_chRNA
  &lt;chr&gt;        &lt;dbl&gt;       &lt;dbl&gt;       &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;     &lt;dbl&gt;
1 ENSG00000…    3160      -0.415      -0.596     0.000936     0.000414   0.00930
# … with 2 more variables: FDR_polyA &lt;dbl&gt;, Category &lt;chr&gt;</code></pre>
</div>
<div id="induction-of-gagt-introns-plots" class="section level3">
<h3>Induction of GAGT introns plots</h3>
<p>first read in some data</p>
<pre class="r"><code>ClusterMax.mat &lt;- leafcutter.counts %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(&quot;junc&quot;) %&gt;%
    mutate(cluster=str_replace(junc, &quot;^(.+?):.+?:.+?:(.+)$&quot;, &quot;\\1_\\2&quot;)) %&gt;%
    group_by(cluster) %&gt;%
    mutate(across(where(is.numeric), sum)) %&gt;%
    ungroup() %&gt;%
    dplyr::select(junc, everything(), -cluster) %&gt;%
    column_to_rownames(&quot;junc&quot;) %&gt;%
    as.matrix()


PSI.df &lt;- (leafcutter.counts / as.numeric(ClusterMax.mat) * 100) %&gt;%
  signif() %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;Leafcutter.ID&quot;) %&gt;%
  mutate(Intron = str_replace(Leafcutter.ID, &quot;(.+?):(.+?):(.+?):clu_.+?_([+-])$&quot;, &quot;\\1:\\2:\\3:\\4&quot;))

Intron.Donors &lt;- fread(&quot;../code/SmallMolecule/leafcutter/JuncfilesMerged.annotated.basic.bed.5ss.tab.gz&quot;, col.names = c(&quot;Intron&quot;, &quot;DonorSeq&quot;, &quot;DonorScore&quot;)) %&gt;%
  mutate(Intron = str_replace(Intron, &quot;(.+?)_(.+?)_(.+?)_(.+?)::.+?$&quot;, &quot;\\1:\\2:\\3:\\4&quot;))

PSI.tidy &lt;- PSI.df %&gt;%
  left_join(Intron.Donors) %&gt;%
  gather(&quot;Sample&quot;, &quot;PSI&quot;,contains(&quot;_&quot;)) %&gt;%
  inner_join(
    leafcutter.counts %&gt;%
      as.data.frame() %&gt;%
      rownames_to_column(&quot;Leafcutter.ID&quot;) %&gt;%
      mutate(Intron = str_replace(Leafcutter.ID, &quot;(.+?):(.+?):(.+?):clu_.+?_([+-])$&quot;, &quot;\\1:\\2:\\3:\\4&quot;)) %&gt;%
      gather(&quot;Sample&quot;, &quot;Counts&quot;,contains(&quot;_&quot;))
  ) %&gt;%
  separate(Sample, into=c(&quot;treatment&quot;, &quot;dose.nM&quot;, &quot;Cell.type&quot;, &quot;LibraryType&quot;, &quot;rep&quot;), sep=&quot;_&quot;, convert=T) %&gt;%
  replace_na(list(dose.nM=0))</code></pre>
<ul>
<li>Plot of induction of GAGT introns specifically in chRNA and polyA
(main)</li>
</ul>
<pre class="r"><code>Mean.PSI &lt;- PSI.tidy %&gt;%
  group_by(Intron, dose.nM, DonorSeq, LibraryType) %&gt;%
  summarise(PSI=mean(PSI, na.rm=T)) %&gt;%
  ungroup()

inner_join(
  Mean.PSI,
  Mean.PSI %&gt;%
    filter(dose.nM==0) %&gt;%
    dplyr::select(Intron, DonorSeq, PSI, LibraryType),
  by=c(&quot;Intron&quot;, &quot;DonorSeq&quot;, &quot;LibraryType&quot;)
) %&gt;%
  mutate(Is.GA.GT = substr(DonorSeq, 3,4)) %&gt;%
  mutate(DeltaPSI=(PSI.x-PSI.y)) %&gt;%
  group_by(dose.nM, Is.GA.GT, LibraryType) %&gt;%
  summarise(PSI.summary = mean(DeltaPSI, na.rm=T)) %&gt;%
  ungroup() %&gt;%
  drop_na() %&gt;%
  ggplot(aes(x=dose.nM, y=PSI.summary, color=Is.GA.GT==&quot;GA&quot;,  group=interaction(Is.GA.GT, LibraryType))) +
  geom_line(aes(linetype=LibraryType)) +
  geom_point() + 
  scale_x_continuous(trans=&quot;log1p&quot;, limits=c(0, 10000), breaks=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 0), labels=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 0)) +
  # facet_wrap(~Is.GA.GT, scales = &quot;free_y&quot;) +
  # scale_y_continuous(trans=&quot;log1p&quot;) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1)) +
  labs(y=&quot;Mean deltaPSI&quot;, caption=str_wrap(&quot;Average DeltaPSI across all introns&quot;, 30))</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok now let’s bootstrap some confidence intervals…</p>
<pre class="r"><code>dat.to.plot.GAGT.enrichment &lt;- inner_join(
  Mean.PSI,
  Mean.PSI %&gt;%
    filter(dose.nM==0) %&gt;%
    dplyr::select(Intron, DonorSeq, PSI, LibraryType),
  by=c(&quot;Intron&quot;, &quot;DonorSeq&quot;, &quot;LibraryType&quot;)
) %&gt;%
  mutate(Is.GA.GT = substr(DonorSeq, 3,4)) %&gt;%
  mutate(DinculeotideCategory = case_when(
    Is.GA.GT == &quot;AG&quot; ~ &quot;AG|GU&quot;,
    Is.GA.GT == &quot;GA&quot; ~ &quot;GA|GU&quot;,
    TRUE ~ &quot;not (AG or GA)|GU&quot;
  )) %&gt;%
  mutate(DeltaPSI=(PSI.x-PSI.y))


# Bootstrap resamples for confidence intervals
NumIterations &lt;- 200
ResampleResults &lt;- list()
for (i in 1:NumIterations){
  df &lt;- dat.to.plot.GAGT.enrichment %&gt;%
    group_by(dose.nM, DinculeotideCategory, LibraryType) %&gt;%
    sample_frac(replace=T) %&gt;%
    ungroup() %&gt;%
    group_by(dose.nM, DinculeotideCategory, LibraryType) %&gt;%
    summarise(PSI.summary = mean(DeltaPSI, na.rm=T)) %&gt;%
    ungroup() %&gt;%
    drop_na()
  ResampleResults[[i]] &lt;- df
}

dat.to.plot.GAGT.enrichment %&gt;% distinct(Intron, .keep_all=T) %&gt;%
  count(DinculeotideCategory)

labels &lt;- c(&quot;AG|GU&quot;=&quot;AG|GU; n=90011&quot;, &quot;GA|GU&quot;=&quot;GA|GU; n=3263&quot;, &quot;not (AG or GA)|GU&quot;= &quot;not (AG or GA)|GU; n=63548&quot;)

P.MeanDeltaPSI &lt;- bind_rows(ResampleResults, .id=&quot;iteration&quot;) %&gt;%
  group_by(dose.nM, DinculeotideCategory, LibraryType) %&gt;%
  summarise(enframe(quantile(PSI.summary, c(0.025, 0.5, 0.975)), &quot;ResampledQuantile&quot;, &quot;MeanDeltaPSI&quot;)) %&gt;%
  pivot_wider(names_from=&quot;ResampledQuantile&quot;, values_from=&quot;MeanDeltaPSI&quot;) %&gt;%
  ggplot(aes(x=dose.nM, y=`50%`, color=DinculeotideCategory,linetype=LibraryType, group=interaction(DinculeotideCategory, LibraryType))) +
  geom_ribbon(aes(ymin=`2.5%`, ymax=`97.5%`, fill=DinculeotideCategory), alpha=0.1, size=0.5) +
  geom_line(size=1) +
  geom_point() + 
  scale_colour_brewer(type=&quot;qual&quot;, palette = &quot;Dark2&quot;, labels=labels, name=&quot;5&#39;ss motif&quot;) +
  scale_fill_brewer(type=&quot;qual&quot;, palette = &quot;Dark2&quot;, labels=labels, name=&quot;5&#39;ss motif&quot;) +
  # facet_wrap(~Is.GA.GT, scales = &quot;free_y&quot;) +
  # scale_y_continuous(trans=&quot;log1p&quot;) +
  scale_x_continuous(trans=&quot;log1p&quot;, limits=c(0, 10000), breaks=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 0), labels=rev(c(&quot;0&quot;, &quot;3.16&quot;,&quot;10&quot;,&quot;31.6&quot;,&quot;100&quot;, &quot;316&quot;, &quot;1000&quot;, &quot;3160&quot;, &quot;10000&quot;))) + 
  # scale_x_break(c(2,2)) +
  # theme(axis.text.x.top = element_blank(),
  #       axis.ticks.x.top = element_blank(),
  #       axis.line.x.top = element_blank(),
  #       plot.background = element_blank()) +
  Rotate_x_labels +
  labs(y=expression(&quot;Mean &quot;*Delta*&quot;PSI&quot;), x=&quot;[risdiplam] (nanomolar)&quot;)
P.MeanDeltaPSI</code></pre>
<ul>
<li>spearman correlation distribution of GAGT introns compared to all
others (supplement)</li>
</ul>
<pre class="r"><code>SpearmanCorr &lt;- read_tsv(&quot;../code/SmallMolecule/FitModels/Data/polyA_GAGTIntrons.tsv.gz&quot;)

SpearmanCorr %&gt;%
  distinct(Intron, .keep_all=T) %&gt;%
  filter(substr(DonorSeq, 3,4)==&quot;GA&quot;) %&gt;%
  filter(q&lt;0.1) %&gt;%
  mutate(numberInts = n()) %&gt;%
  filter(spearman == min(spearman))</code></pre>
<pre><code># A tibble: 1 × 16
  Intron           Leafcutter.ID DonorSeq DonorScore treatment dose.nM Cell.type
  &lt;chr&gt;            &lt;chr&gt;         &lt;chr&gt;         &lt;dbl&gt; &lt;chr&gt;       &lt;dbl&gt; &lt;chr&gt;    
1 chr19:58571429:… chr19:585714… AGGAGTA…       2.87 DMSO            0 LCL      
# … with 9 more variables: LibraryType &lt;chr&gt;, rep &lt;dbl&gt;, PSI &lt;dbl&gt;,
#   Counts &lt;dbl&gt;, n &lt;dbl&gt;, spearman &lt;dbl&gt;, spearman.p &lt;dbl&gt;, q &lt;dbl&gt;,
#   numberInts &lt;int&gt;</code></pre>
<pre class="r"><code>Translated.GAGT_CassetteExons %&gt;% nrow()</code></pre>
<pre><code>[1] 316</code></pre>
<pre class="r"><code>P.spearman.dist &lt;- SpearmanCorr %&gt;%
  distinct(Intron, .keep_all=T) %&gt;%
  mutate(Is.GA.GT = substr(DonorSeq, 3,4)) %&gt;%
  mutate(DinculeotideCategory = case_when(
    Is.GA.GT == &quot;AG&quot; ~ &quot;AG|GU&quot;,
    Is.GA.GT == &quot;GA&quot; ~ &quot;GA|GU&quot;,
    TRUE ~ &quot;not (AG or GA)|GU&quot;
  )) %&gt;%
  filter(!is.na(spearman)) %&gt;%
  # mutate(DinculeotideCategory = factor(DinculeotideCategory, levels=c(&quot;GA|GU&quot;, &quot;AG|GU&quot;, &quot;not (AG or GA)|GU&quot;))) %&gt;%
  arrange(desc(DinculeotideCategory)) %&gt;%
  ggplot(aes(x=spearman, group=Is.GA.GT, color=DinculeotideCategory)) +
  geom_vline(xintercept=0) +
  scale_colour_brewer(type=&quot;qual&quot;, palette = &quot;Dark2&quot;, labels=labels, name=&quot;5&#39;ss motif&quot;) +
  stat_ecdf() +
  labs(y=&quot;Cumulative fraction&quot;, x=&quot;dose:PSI spearman coef&quot;, caption=str_wrap(&quot;641 GA|GU induced introns with significant spearman correlation (FDR&lt;10%, corresponding to spearman coef &gt; 0.775) were used for futher interpretion of effects. 305 of which were classified as cassette exons, translated, and PSI was recalculated from junction trios&quot;, 40))

P.spearman.dist </code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div
id="splicing-vs-expression-beta-faceted-by-predicted-splicing-effect"
class="section level3">
<h3>splicing vs expression beta faceted by predicted splicing
effect</h3>
<pre class="r"><code>##NEEDS FIXING
PolyA.SplicingEffects.Annotationed &lt;- small.molecule.dat.introns %&gt;%
  dplyr::select(1:4) %&gt;%
  inner_join(
    Translated.GAGT_CassetteExons,
    by=c(&quot;junc&quot;=&quot;GAGTInt&quot;)
  )


P.scatter.byPrediction.dat &lt;- Scatter.dat %&gt;%
  filter(dose.nM==3160) %&gt;%
  inner_join(
    P.boxplot.dat %&gt;%
      filter(dataset==&quot;polyA&quot;) %&gt;%
      dplyr::select(gene, ColorEffect, MergedEffect),
    by=&quot;gene&quot;
    )  %&gt;%
  inner_join(
    PolyA.SplicingEffects.Annotationed %&gt;%
    filter(param == &quot;Pred_3160&quot;) %&gt;%
    dplyr::select(junc, gene, Estimate) %&gt;%
    mutate(gene = str_replace(gene, &quot;(^.+?)\\..+?$&quot;, &quot;\\1&quot;)),
    by=&quot;gene&quot;
  )

MergedEffectLabels &lt;- c(
      &quot;Productive&quot;=&quot;In annotated productive transcript&quot;,
     &quot;NMD&quot;=&quot;In annotated NMD transcript&quot;,
     &quot;Frame shifting&quot;=&quot;Frame shifting&quot;,
     &quot;Cassette in UTR&quot;=&quot;Cassette in UTR&quot;,
     &quot;In-frame PTC&quot;=&quot;In-frame PTC&quot;,
      &quot;Preserves frame&quot;=&quot;Preserves frame&quot;,
     &quot;All NMD&quot;=&quot;All predicted NMD exons&quot;,
     &quot;All productive&quot;=&quot;All predicted productive exons&quot;
                    )

P.scatter.byPrediction &lt;- bind_rows(
  P.scatter.byPrediction.dat %&gt;%
    mutate(color=MergedEffect),
  P.scatter.byPrediction.dat %&gt;%
    mutate(color=MergedEffect) %&gt;%
    mutate(MergedEffect=recode(ColorEffect, !!!c(&quot;NMD&quot;=&quot;All NMD&quot;, &quot;Productive&quot;=&quot;All productive&quot;)))
  ) %&gt;% 
  mutate(MergedEffect = factor(MergedEffect, levels=c(&quot;Productive&quot;, &quot;Preserves frame&quot;, &quot;Cassette in UTR&quot;, &quot;All productive&quot;,&quot;NMD&quot;, &quot;Frame shifting&quot;, &quot;In-frame PTC&quot;, &quot;All NMD&quot;))) %&gt;%
  ggplot(aes(x=Estimate, y=logFC_polyA, color=color)) +
  geom_hline(yintercept=0) +
  geom_point() +
   scale_color_manual(values=c(
     &quot;Productive&quot;=&quot;#1f78b4&quot;,
     &quot;NMD&quot;=&quot;#e31a1c&quot;,
     &quot;Frame shifting&quot;=&quot;#fb9a99&quot;,
     &quot;Cassette in UTR&quot;=&quot;#a6cee3&quot;,
     &quot;In-frame PTC&quot;=&quot;#fb9a99&quot;,
      &quot;Preserves frame&quot;=&quot;#a6cee3&quot;
   ),
   labels=MergedEffectLabels) +
  geom_text( data = . %&gt;%
    group_by(MergedEffect) %&gt;%
    summarise(cor=cor.test(Estimate,logFC_polyA)[[&quot;estimate&quot;]], pval=cor.test(Estimate,logFC_polyA, method=&#39;s&#39;)[[&quot;p.value&quot;]]) %&gt;%
    mutate(R = signif(cor, 3), P=format.pval(pval, 3)) %&gt;%
    mutate(label = str_glue(&quot;rho:{R}\nP:{P}&quot;)),
    aes(x=-Inf, y=-Inf, label=label),
  hjust=-.1, vjust=-0.1, color=&#39;black&#39;
  ) +
  # facet_wrap(~MergedEffect, labeller = as_labeller(MergedEffectLabels)) +
  facet_wrap(~MergedEffect, nrow=2, labeller = as_labeller(MergedEffectLabels, default=label_wrap_gen(20))) +
  theme(legend.position=&quot;none&quot;) +
  labs(y=expression(&quot;host gene &quot;*log[&quot;2&quot;]*&quot;FC, polyA RNA&quot;), x=expression(Delta*&quot;PSI&quot;), caption=&quot;[risdiplam]=3160nM, Effects measured in polyA RNA&quot;) +
  Rotate_x_labels
P.scatter.byPrediction</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Same thing but use expression effects in chRNA.</p>
<pre class="r"><code>P.scatter.byPrediction.dat.chRNA &lt;- Scatter.dat %&gt;%
  filter(dose.nM==3160) %&gt;%
  inner_join(
    P.boxplot.dat %&gt;%
      filter(dataset==&quot;polyA&quot;) %&gt;%
      dplyr::select(gene, ColorEffect, MergedEffect),
    by=&quot;gene&quot;
    )  %&gt;%
  inner_join(
    PolyA.SplicingEffects.Annotationed %&gt;%
    filter(param==&quot;Pred_3160&quot;) %&gt;%
    dplyr::select(junc, gene, Estimate) %&gt;%
    mutate(gene = str_replace(gene, &quot;(^.+?)\\..+?$&quot;, &quot;\\1&quot;)),
    by=&quot;gene&quot;
  )

MergedEffectLabels &lt;- c(
      &quot;Productive&quot;=&quot;In annotated productive transcript&quot;,
     &quot;NMD&quot;=&quot;In annotated NMD transcript&quot;,
     &quot;Frame shifting&quot;=&quot;Frame shifting&quot;,
     &quot;Cassette in UTR&quot;=&quot;Cassette in UTR&quot;,
     &quot;In-frame PTC&quot;=&quot;In-frame PTC&quot;,
      &quot;Preserves frame&quot;=&quot;Preserves frame&quot;,
     &quot;All NMD&quot;=&quot;All predicted NMD exons&quot;,
     &quot;All productive&quot;=&quot;All predicted productive exons&quot;
                    )

P.scatter.byPrediction.chRNA &lt;- bind_rows(
  P.scatter.byPrediction.dat.chRNA %&gt;%
    mutate(color=MergedEffect),
  P.scatter.byPrediction.dat.chRNA %&gt;%
    mutate(color=MergedEffect) %&gt;%
    mutate(MergedEffect=recode(ColorEffect, !!!c(&quot;NMD&quot;=&quot;All NMD&quot;, &quot;Productive&quot;=&quot;All productive&quot;)))
  ) %&gt;% 
  mutate(MergedEffect = factor(MergedEffect, levels=c(&quot;Productive&quot;, &quot;Preserves frame&quot;, &quot;Cassette in UTR&quot;, &quot;All productive&quot;,&quot;NMD&quot;, &quot;Frame shifting&quot;, &quot;In-frame PTC&quot;, &quot;All NMD&quot;))) %&gt;%
  ggplot(aes(x=Estimate, y=logFC_chRNA, color=color)) +
  geom_hline(yintercept=0) +
  geom_point() +
   scale_color_manual(values=c(
     &quot;Productive&quot;=&quot;#1f78b4&quot;,
     &quot;NMD&quot;=&quot;#e31a1c&quot;,
     &quot;Frame shifting&quot;=&quot;#fb9a99&quot;,
     &quot;Cassette in UTR&quot;=&quot;#a6cee3&quot;,
     &quot;In-frame PTC&quot;=&quot;#fb9a99&quot;,
      &quot;Preserves frame&quot;=&quot;#a6cee3&quot;
   ),
   labels=MergedEffectLabels) +
  geom_text( data = . %&gt;%
    group_by(MergedEffect) %&gt;%
    summarise(cor=cor.test(Estimate,logFC_chRNA)[[&quot;estimate&quot;]], pval=cor.test(Estimate,logFC_chRNA, method=&#39;s&#39;)[[&quot;p.value&quot;]]) %&gt;%
    mutate(R = signif(cor, 3), P=format.pval(pval, 3)) %&gt;%
    mutate(label = str_glue(&quot;rho:{R}\nP:{P}&quot;)),
    aes(x=-Inf, y=Inf, label=label),
  hjust=-.1, vjust=1.1, color=&#39;black&#39;
  ) +
  # facet_wrap(~MergedEffect, labeller = as_labeller(MergedEffectLabels)) +
  facet_wrap(~MergedEffect, nrow=2, labeller = as_labeller(MergedEffectLabels, default=label_wrap_gen(20))) +
  theme(legend.position=&quot;none&quot;) +
  labs(y=expression(&quot;host gene &quot;*log[&quot;2&quot;]*&quot;FC, chRNA&quot;), x=expression(Delta*&quot;PSI&quot;), caption=&quot;[risdiplam]=3160nM, Splicing measured in polyA&quot;) +
  Rotate_x_labels
P.scatter.byPrediction.chRNA</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="druggability-enrichment-plots" class="section level3">
<h3>Druggability enrichment plots</h3>
<ul>
<li>The druggability gene enrichment barplots comparing genes w/
ris-induced poison exons (not based on expression effects) (main)</li>
<li>the druggability gene enrichment barplot comparing genes w/
post-txnal downregulation (Supplement)</li>
</ul>
<pre class="r"><code>SplicingDruggableGenes &lt;- P.boxplot.dat %&gt;%
  filter(dataset == &#39;polyA&#39;) %&gt;%
  filter(ColorEffect == &#39;NMD&#39;) %&gt;%
  distinct(gene, .keep_all=T)

nrow(SplicingDruggableGenes)</code></pre>
<pre><code>[1] 219</code></pre>
<pre class="r"><code>exons &lt;- read_tsv(&quot;../code/ReferenceGenome/Annotations/GTFTools_BasicAnnotations/gencode.v34.chromasomal.exons.sorted.bed&quot;, col_names=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;gene_transcript&quot;, &quot;score&quot;, &quot;strand&quot;))

transcripts_to_genes &lt;- exons %&gt;%
  distinct(gene_transcript) %&gt;%
  separate(gene_transcript, into=c(&quot;gene&quot;, &quot;transcript&quot;), sep=&quot;_&quot;) %&gt;%
  mutate(gene = str_replace(gene, &quot;(^.+?)\\..+?$&quot;, &quot;\\1&quot;))


Expression.table.transcripts &lt;- read_tsv(&quot;../code/SmallMolecule/salmon.DMSO.merged.txt&quot;)
Expression.table.genes &lt;- Expression.table.transcripts %&gt;%
  inner_join(transcripts_to_genes, by=c(&quot;Name&quot;=&quot;transcript&quot;)) %&gt;%
  group_by(gene) %&gt;%
  summarise_at(vars(contains(&quot;DMSO&quot;)), sum) %&gt;%
  gather(&quot;sample&quot;, &quot;TPM&quot;, -gene) %&gt;%
  group_by(gene) %&gt;%
  summarise(medianTPM = median(TPM))

hist(log2(Expression.table.genes$medianTPM))</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Expression.table.genes %&gt;%
  filter(gene == &quot;ENSG00000111640&quot;)</code></pre>
<pre><code># A tibble: 1 × 2
  gene            medianTPM
  &lt;chr&gt;               &lt;dbl&gt;
1 ENSG00000111640     4502.</code></pre>
<pre class="r"><code>Expression.tidyForMatchingGenes &lt;- Expression.table.genes %&gt;%
  mutate(IsSplicingDruggable = gene %in% SplicingDruggableGenes$gene) %&gt;%
  arrange(medianTPM) %&gt;%
  mutate(LaggingGeneGroup = lag(IsSplicingDruggable)) %&gt;%
  mutate(LeadingGeneGroup = lead(IsSplicingDruggable)) %&gt;%
  ungroup()

Merged.WithExpressionMatchedControlGenes &lt;- 
bind_rows(
  Expression.tidyForMatchingGenes %&gt;%
    filter(IsSplicingDruggable) %&gt;%
    mutate(Group = &quot;SplicingPerturbed&quot;),
  Expression.tidyForMatchingGenes %&gt;%
    filter(!IsSplicingDruggable &amp; LaggingGeneGroup) %&gt;%
    mutate(Group = &quot;SplicingUnperturbedControl&quot;),
  Expression.tidyForMatchingGenes %&gt;%
    filter(!IsSplicingDruggable &amp; LeadingGeneGroup) %&gt;%
    mutate(Group = &quot;SplicingUnperturbedControl&quot;),
) %&gt;%
  dplyr::select(gene, medianTPM, Group)

Merged.WithExpressionMatchedControlGenes %&gt;%
  ggplot(aes(medianTPM)) +
  stat_ecdf() +
  scale_x_continuous(trans=&#39;log10&#39;) +
  facet_wrap(~Group)</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-18-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Now download the supplemental table from <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6321762/#SD1">This
paper</a>…</p>
<pre class="r"><code>url1&lt;-&#39;https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6321762/bin/NIHMS80906-supplement-Table_S1.xlsx&#39;
p1f &lt;- tempfile()
download.file(url1, p1f, mode=&quot;wb&quot;)
p1&lt;-read_excel(path = p1f, sheet = 1)

head(p1)</code></pre>
<pre><code># A tibble: 6 × 12
  ensembl_gene_id druggability_tier hgnc_names chr_b37 start_b37  end_b37 strand
  &lt;chr&gt;           &lt;chr&gt;             &lt;chr&gt;      &lt;chr&gt;       &lt;dbl&gt;    &lt;dbl&gt;  &lt;dbl&gt;
1 ENSG00000000938 Tier 1            FGR        1        27938575   2.80e7     -1
2 ENSG00000001626 Tier 1            CFTR       7       117105838   1.17e8      1
3 ENSG00000001630 Tier 1            CYP51A1    7        91741465   9.18e7     -1
4 ENSG00000002549 Tier 1            LAP3       4        17578815   1.76e7      1
5 ENSG00000004468 Tier 1            CD38       4        15779898   1.59e7      1
6 ENSG00000004478 Tier 1            FKBP4      12        2904119   2.91e6      1
# … with 5 more variables: description &lt;chr&gt;, no_of_gwas_regions &lt;dbl&gt;,
#   small_mol_druggable &lt;chr&gt;, bio_druggable &lt;chr&gt;, adme_gene &lt;chr&gt;</code></pre>
<p>Also, make sure to subset all gene sets to only include protein
coding ones.</p>
<pre class="r"><code>CodingGenes &lt;- read_tsv(&quot;../data/mart_export.txt.gz&quot;) %&gt;%
  distinct(`Gene stable ID`) %&gt;%
  dplyr::select(gene = &quot;Gene stable ID&quot;)

GeneCategories &lt;- p1 %&gt;%
  filter(small_mol_druggable==&quot;Y&quot;) %&gt;%
  left_join(Expression.table.genes, by=c(&quot;ensembl_gene_id&quot;=&quot;gene&quot;)) %&gt;%
  dplyr::select(Group=druggability_tier, gene=ensembl_gene_id, medianTPM) %&gt;%
  bind_rows(
    Merged.WithExpressionMatchedControlGenes,
    Expression.table.genes %&gt;%
      mutate(Group = &quot;Whole protein coding genome&quot;)
    ) %&gt;%
  filter(gene %in% CodingGenes$gene)</code></pre>
<p>Now check number of exons and gene length of different druggability
categories</p>
<pre class="r"><code>TopTranscriptPerGene &lt;- Expression.table.transcripts %&gt;%
  # # summarise_at(vars(contains(&quot;DMSO&quot;)), sum) %&gt;%
  gather(&quot;sample&quot;, &quot;TPM&quot;, -Name) %&gt;%
  group_by(Name) %&gt;%
  summarise(medianTPM = median(TPM)) %&gt;%
  inner_join(transcripts_to_genes, by=c(&quot;Name&quot;=&quot;transcript&quot;)) %&gt;%
  group_by(gene) %&gt;%
  filter(medianTPM == max(medianTPM)) %&gt;%
  ungroup() %&gt;%
  distinct(gene, .keep_all=T)

NumExonsAndLength &lt;- exons %&gt;%
  group_by(gene_transcript) %&gt;%
  summarise(NumExons = n(),
            Min = min(start),
            Max = max(stop)) %&gt;%
  mutate(GeneLength = Max - Min) %&gt;%
  separate(gene_transcript, into=c(&quot;gene&quot;, &quot;transcript&quot;), sep=&quot;_&quot;) %&gt;%
  filter(transcript %in% TopTranscriptPerGene$Name) %&gt;%
  mutate(gene = str_replace(gene, &quot;(^.+?)\\..+?$&quot;, &quot;\\1&quot;)) %&gt;%
  dplyr::select(-Min, -Max)

NumExonsAndLength %&gt;%
  filter(gene %in% CodingGenes$gene) %&gt;%
  ggplot(aes(x=GeneLength)) +
  stat_ecdf() +
  coord_cartesian(xlim=c(1E3, 1E6)) +
  scale_x_continuous(trans=&#39;log10&#39;) +
  annotation_logticks() +
  theme_bw()</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>GeneCategories %&gt;%
  left_join(NumExonsAndLength) %&gt;%
  filter(!Group %in% c(&quot;Tier 2&quot;, &quot;Tier 3A&quot;, &quot;Tier 3B&quot;)) %&gt;%
  gather(key=&quot;Feature&quot;, value=&quot;value&quot;, NumExons, GeneLength, medianTPM) %&gt;%
  ggplot(aes(x=value, color=Group)) +
  stat_ecdf() +
  scale_x_continuous(trans=&#39;log10&#39;) +
  annotation_logticks() +
  facet_wrap(~Feature, scales = &quot;free&quot;, ncol=1) +
  labs(y=&quot;ecdf&quot;, color=&quot;Druggability gene group&quot;)</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-21-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>GeneCategories %&gt;%
  left_join(NumExonsAndLength) %&gt;%
  gather(key=&quot;Feature&quot;, value=&quot;value&quot;, NumExons, GeneLength, medianTPM) %&gt;%
  filter(Feature == &quot;NumExons&quot;) %&gt;%
  ggplot(aes(x=value, color=Group)) +
  stat_ecdf() +
  scale_x_continuous(trans=&#39;log10&#39;) +
  annotation_logticks() +
  facet_wrap(~Feature, scales = &quot;free&quot;, ncol=1) +
  labs(y=&quot;ecdf&quot;, color=&quot;Druggability gene group&quot;)</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-21-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Save plots for number of exons and host intron length</p>
<pre class="r"><code>GeneCategories %&gt;%
  left_join(NumExonsAndLength) %&gt;%
  gather(key=&quot;Feature&quot;, value=&quot;value&quot;, NumExons, GeneLength, medianTPM) %&gt;%
  filter(Feature == &quot;NumExons&quot;) %&gt;%
  group_by(Group, Feature) %&gt;%
  summarise(Med = median(value, na.rm=T))</code></pre>
<pre><code># A tibble: 7 × 3
# Groups:   Group [7]
  Group                       Feature    Med
  &lt;chr&gt;                       &lt;chr&gt;    &lt;dbl&gt;
1 SplicingPerturbed           NumExons    14
2 SplicingUnperturbedControl  NumExons     9
3 Tier 1                      NumExons    11
4 Tier 2                      NumExons    11
5 Tier 3A                     NumExons    13
6 Tier 3B                     NumExons     6
7 Whole protein coding genome NumExons     8</code></pre>
<pre class="r"><code>Introns.Annotated.Lengths &lt;- read_tsv(&quot;../data/IntronAnnotationsFromYang.tsv.gz&quot;) %&gt;%
  mutate(gene = str_replace(gene, &quot;^(.+?)\\..+$&quot;, &quot;\\1&quot;)) %&gt;%
  filter(SemiSupergroupAnnotations == &quot;basic tag&quot;) %&gt;%
  inner_join(
    GeneCategories %&gt;%
      filter(Group == &quot;SplicingPerturbed&quot;)) %&gt;%
  mutate(Len = end - start) %&gt;%
  mutate(Group = &quot;AllOtherIntronInGenes&quot;) %&gt;%
  dplyr::select(Len, Group)


HostIntrons &lt;- read_tsv(&quot;../output/SmallMoleculeGAGT_CassetteExonclusters.bed&quot;, col_names=paste0(&quot;V&quot;, 1:9)) %&gt;%
  filter(str_detect(V4, &quot;junc.skipping&quot;)) %&gt;%
  mutate(Len = V3-V2) %&gt;%
  mutate(Group = &quot;Host introns&quot;) %&gt;%
  dplyr::select(Len, Group)

bind_rows(
  HostIntrons, Introns.Annotated.Lengths
) %&gt;%
  wilcox.test(Len ~ Group, data=.)</code></pre>
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  Len by Group
W = 564921, p-value &lt; 2.2e-16
alternative hypothesis: true location shift is not equal to 0</code></pre>
<pre class="r"><code>bind_rows(
  HostIntrons, Introns.Annotated.Lengths
) %&gt;%
  group_by(Group) %&gt;%
  summarise(medLen = median(Len))</code></pre>
<pre><code># A tibble: 2 × 2
  Group                 medLen
  &lt;chr&gt;                  &lt;dbl&gt;
1 AllOtherIntronInGenes  2232.
2 Host introns           6138.</code></pre>
<pre class="r"><code>P.IntLen &lt;- bind_rows(
  HostIntrons, Introns.Annotated.Lengths
) %&gt;%
  ggplot(aes(x=Len, color=Group)) +
  stat_ecdf() +
  scale_x_continuous(trans=&#39;log10&#39;) +
  coord_cartesian(xlim=c(100, 1E5)) +
  scale_color_manual(values=c(&quot;Host introns&quot;=&quot;#e31a1c&quot;, &quot;AllOtherIntronInGenes&quot;=&quot;#969696&quot;)) +
  labs(x=&quot;Intron length (bp)&quot;, y=&quot;Cumulative fraction&quot;, caption=&quot;2233 vs 6139bp medians, P&lt;2.2E-16&quot;)
P.IntLen</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>GeneCategories$Group %&gt;% unique()</code></pre>
<pre><code>[1] &quot;Tier 1&quot;                      &quot;Tier 2&quot;                     
[3] &quot;Tier 3A&quot;                     &quot;Tier 3B&quot;                    
[5] &quot;SplicingPerturbed&quot;           &quot;SplicingUnperturbedControl&quot; 
[7] &quot;Whole protein coding genome&quot;</code></pre>
<pre class="r"><code>GeneCategories %&gt;%
  left_join(NumExonsAndLength) %&gt;%
  gather(key=&quot;Feature&quot;, value=&quot;value&quot;, NumExons, GeneLength, medianTPM) %&gt;%
  filter(Feature == &quot;NumExons&quot;) %&gt;%
  filter(Group %in% c(&quot;SplicingPerturbed&quot;, &quot;SplicingUnperturbedControl&quot;)) %&gt;%
  wilcox.test(value~Group, data=.)</code></pre>
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  value by Group
W = 57906, p-value = 5.186e-12
alternative hypothesis: true location shift is not equal to 0</code></pre>
<pre class="r"><code>P.NumExons &lt;- 
  GeneCategories %&gt;%
  left_join(NumExonsAndLength) %&gt;%
  gather(key=&quot;Feature&quot;, value=&quot;value&quot;, NumExons, GeneLength, medianTPM) %&gt;%
  filter(Feature == &quot;NumExons&quot;) %&gt;%
  filter(!Group %in% c(&quot;Tier 2&quot;, &quot;Tier 3A&quot;, &quot;Tier 3B&quot;, &quot;Tier 1&quot;)) %&gt;%
  ggplot(aes(x=value, color=Group)) +
  stat_ecdf() +
  # scale_x_continuous(trans=&#39;log10&#39;) +
  # annotation_logticks(sides=&#39;b&#39;) +
  coord_cartesian(xlim=c(1, 30)) +
  scale_color_manual(
    labels=c(
      &quot;SplicingPerturbed&quot;=&quot;219 Risdiplam-induced NMD targets&quot;,
      &quot;SplicingUnperturbedControl&quot;=&quot;389 expression-matched genes&quot;,
      &quot;Whole protein coding genome&quot;=&quot;Whole genome&quot;,
      &quot;Tier 1&quot;=&quot;1059 targets of approved small molecules&quot;),
    values=c(
      &quot;SplicingPerturbed&quot;=&quot;#e31a1c&quot;,
      &quot;SplicingUnperturbedControl&quot;=&quot;#1f78b4&quot;,
      &quot;Whole protein coding genome&quot;=&quot;#000000&quot;,
      &quot;Tier 1&quot;=&quot;#737373&quot;)
  ) +
  labs(y=&quot;Cumulative fraction&quot;, color=&quot;Druggability gene group&quot;, x=&quot;Number exons in gene&quot;, caption=&quot;median 14 vs 9, P=5.2x10-12&quot;)
P.NumExons</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-22-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, that is one way to show that the risdiplam-perturbed genes tend
to be in longer genes with more exons, compared to a matched set of
control genes or compared to previously druggable genes. But for sake of
compactness, since I already know I also want to do an enrichment
analysis including GO categories (ie GPCRs, etc), let’s also look make
gene categories by length and present for enrichment alongside the GO
enrichment analysis…</p>
<pre class="r"><code>library(biomaRt)
ensembl = useMart(&quot;ensembl&quot;,dataset=&quot;hsapiens_gene_ensembl&quot;)

ensembl_to_symbols &lt;- getBM(attributes= c(&quot;ensembl_gene_id&quot;,&quot;hgnc_symbol&quot;),mart=ensembl)

#Manually copy paste links from MsigDB
go=c(
  &quot;GO:0015075&quot;=&quot;https://www.gsea-msigdb.org/gsea/msigdb/human/download_geneset.jsp?geneSetName=GOMF_MONOATOMIC_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY&amp;fileType=grp&quot;,
  &quot;GO:0004930&quot;=&quot;https://www.gsea-msigdb.org/gsea/msigdb/human/download_geneset.jsp?geneSetName=GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY&amp;fileType=grp&quot;,
  &quot;GO:0004984&quot;=&quot;https://www.gsea-msigdb.org/gsea/msigdb/human/download_geneset.jsp?geneSetName=GOMF_OLFACTORY_RECEPTOR_ACTIVITY&amp;fileType=grp&quot;,
  &quot;GO:0004879&quot;=&quot;https://www.gsea-msigdb.org/gsea/msigdb/human/download_geneset.jsp?geneSetName=GOMF_NUCLEAR_RECEPTOR_ACTIVITY&amp;fileType=grp&quot;,
  &quot;GO:0016301&quot;=&quot;https://www.gsea-msigdb.org/gsea/msigdb/human/download_geneset.jsp?geneSetName=GOMF_KINASE_ACTIVITY&amp;fileType=grp&quot;,
  &quot;GO:0003700&quot;=&quot;https://www.gsea-msigdb.org/gsea/msigdb/human/download_geneset.jsp?geneSetName=GOMF_DNA_BINDING_TRANSCRIPTION_FACTOR_ACTIVITY&amp;fileType=grp&quot;)

GO.genes.of.interest &lt;- lapply(go, read.table, skip=2, col.names=c(&quot;hgnc_symbol&quot;)) %&gt;%
  bind_rows(.id=&quot;GO&quot;) %&gt;%
  mutate(GO_Name = recode(GO, !!!go)) %&gt;%
  mutate(GO_Name = str_replace(GO_Name, &quot;.+?geneSetName=(.+?)&amp;fileType=grp$&quot;, &quot;\\1&quot;)) %&gt;%
  left_join(ensembl_to_symbols)

GO.genes.of.interest %&gt;%
  distinct(hgnc_symbol, GO_Name) %&gt;%
  count(GO_Name)</code></pre>
<pre><code>                                                 GO_Name    n
1         GOMF_DNA_BINDING_TRANSCRIPTION_FACTOR_ACTIVITY 1454
2               GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY  871
3                                   GOMF_KINASE_ACTIVITY  743
4 GOMF_MONOATOMIC_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY  726
5                         GOMF_NUCLEAR_RECEPTOR_ACTIVITY   52
6                       GOMF_OLFACTORY_RECEPTOR_ACTIVITY  429</code></pre>
<pre class="r"><code>GPCRs &lt;- GO.genes.of.interest %&gt;%
  filter(GO_Name == &quot;GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY&quot;) %&gt;%
  pull(hgnc_symbol) %&gt;% unique() %&gt;%
  setdiff(
    GO.genes.of.interest %&gt;%
      filter(GO_Name == &quot;GOMF_OLFACTORY_RECEPTOR_ACTIVITY&quot;) %&gt;%
      pull(hgnc_symbol) %&gt;% unique()
  )</code></pre>
<p>Now let’s do the GO enrichment tests. I’ll actually just do them “by
hand” with the <code>fisher.test</code> (hypergeometric test) function
rather than using a dedicated package to systematically look for
enrichment across all GO categories, which I am less interested in.</p>
<pre class="r"><code>library(broom)

GO.genes.of.interest.totest &lt;- 
  bind_rows(
    # filter out olfactor GPCRs from GPCR group
    GO.genes.of.interest %&gt;%
      filter(ensembl_gene_id %in% CodingGenes$gene) %&gt;%
      filter(!GO_Name==&quot;GOMF_OLFACTORY_RECEPTOR_ACTIVITY&quot;) %&gt;%
      filter(!(GO_Name==&quot;GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY&quot; &amp; !hgnc_symbol %in% GPCRs)),
  NumExonsAndLength %&gt;%
    filter(gene %in% CodingGenes$gene) %&gt;%
    # mutate(GO_Name = cut_number(GeneLength, 4, ordered_result=T)) %&gt;%
    # distinct(GO_Name)
    mutate(GO_Name = factor(ntile(GeneLength, 4))) %&gt;%
    # mutate(GO_Name = cut(GeneLength, breaks=c(-Inf,1E3, 1E4, 1E5, Inf), include.lowest=T)) %&gt;%
    filter(!is.na(GO_Name)) %&gt;%
    dplyr::select(ensembl_gene_id=gene, GO_Name)
  )
GO_Categories &lt;- GO.genes.of.interest.totest %&gt;%
  pull(GO_Name) %&gt;% unique()

count(GO.genes.of.interest.totest, GO_Name)</code></pre>
<pre><code>                                                 GO_Name    n
1                                                      1 4621
2                                                      2 4621
3                                                      3 4621
4                                                      4 4621
5         GOMF_DNA_BINDING_TRANSCRIPTION_FACTOR_ACTIVITY 1537
6               GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY  489
7                                   GOMF_KINASE_ACTIVITY  793
8 GOMF_MONOATOMIC_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY  778
9                         GOMF_NUCLEAR_RECEPTOR_ACTIVITY   57</code></pre>
<pre class="r"><code>DruggabilityCategories &lt;- GeneCategories %&gt;%
  filter(!Group==&quot;Whole protein coding genome&quot;) %&gt;% pull(Group) %&gt;% unique()
DruggabilityCategories</code></pre>
<pre><code>[1] &quot;Tier 1&quot;                     &quot;Tier 2&quot;                    
[3] &quot;Tier 3A&quot;                    &quot;Tier 3B&quot;                   
[5] &quot;SplicingPerturbed&quot;          &quot;SplicingUnperturbedControl&quot;</code></pre>
<pre class="r"><code>GeneCategories.totest &lt;- GeneCategories %&gt;%
  sample_frac() %&gt;%
  group_by(Group) %&gt;%
  mutate(n = row_number()) %&gt;%
  ungroup()
  # filter(!Group == &quot;Tier 1&quot; | (Group == &quot;Tier 1&quot; &amp; n &lt;= 116))
count(GeneCategories.totest, Group)</code></pre>
<pre><code># A tibble: 7 × 2
  Group                           n
  &lt;chr&gt;                       &lt;int&gt;
1 SplicingPerturbed             219
2 SplicingUnperturbedControl    396
3 Tier 1                       1059
4 Tier 2                        648
5 Tier 3A                        67
6 Tier 3B                       516
7 Whole protein coding genome 18484</code></pre>
<pre class="r"><code>NumExonsAndLength %&gt;%
    filter(gene %in% CodingGenes$gene) %&gt;% 
    mutate(N = cut_number(GeneLength, 4)) %&gt;%
    distinct(N)</code></pre>
<pre><code># A tibble: 4 × 1
  N                  
  &lt;fct&gt;              
1 (8.21e+03,2.25e+04]
2 (2.25e+04,5.88e+04]
3 (5.88e+04,2.3e+06] 
4 [117,8.21e+03]     </code></pre>
<pre class="r"><code>results &lt;- list()
for (DruggabilitySetName in DruggabilityCategories){
  for (GO_CategoryName in GO_Categories){
    print(paste(DruggabilitySetName, GO_CategoryName))
    DruggabilitySet &lt;- GeneCategories.totest %&gt;%
      filter(Group==DruggabilitySetName) %&gt;% pull(gene)
    GO_CategorySet &lt;- GO.genes.of.interest.totest %&gt;%
      filter(GO_Name==GO_CategoryName) %&gt;% pull(ensembl_gene_id)
    test.results &lt;- data.frame(gene=CodingGenes$gene) %&gt;%
      mutate(IsDruggable = gene %in% DruggabilitySet, IsInGO.Set = gene %in% GO_CategorySet) %&gt;%
      mutate(IsDruggable = factor(IsDruggable),
             IsInGO.Set = factor(IsInGO.Set)) %&gt;%
      count(IsDruggable, IsInGO.Set, .drop=F) %&gt;%
      pivot_wider(names_from=&quot;IsInGO.Set&quot;, values_from=&quot;n&quot;) %&gt;%
      column_to_rownames(&quot;IsDruggable&quot;) %&gt;%
      fisher.test() %&gt;% glance()
    results[[paste(DruggabilitySetName, GO_CategoryName, sep=&quot;;&quot;)]] &lt;- test.results %&gt;%
      as.data.frame()
  }
}</code></pre>
<pre><code>[1] &quot;Tier 1 GOMF_MONOATOMIC_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY&quot;
[1] &quot;Tier 1 GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY&quot;
[1] &quot;Tier 1 GOMF_NUCLEAR_RECEPTOR_ACTIVITY&quot;
[1] &quot;Tier 1 GOMF_KINASE_ACTIVITY&quot;
[1] &quot;Tier 1 GOMF_DNA_BINDING_TRANSCRIPTION_FACTOR_ACTIVITY&quot;
[1] &quot;Tier 1 2&quot;
[1] &quot;Tier 1 3&quot;
[1] &quot;Tier 1 4&quot;
[1] &quot;Tier 1 1&quot;
[1] &quot;Tier 2 GOMF_MONOATOMIC_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY&quot;
[1] &quot;Tier 2 GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY&quot;
[1] &quot;Tier 2 GOMF_NUCLEAR_RECEPTOR_ACTIVITY&quot;
[1] &quot;Tier 2 GOMF_KINASE_ACTIVITY&quot;
[1] &quot;Tier 2 GOMF_DNA_BINDING_TRANSCRIPTION_FACTOR_ACTIVITY&quot;
[1] &quot;Tier 2 2&quot;
[1] &quot;Tier 2 3&quot;
[1] &quot;Tier 2 4&quot;
[1] &quot;Tier 2 1&quot;
[1] &quot;Tier 3A GOMF_MONOATOMIC_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY&quot;
[1] &quot;Tier 3A GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY&quot;
[1] &quot;Tier 3A GOMF_NUCLEAR_RECEPTOR_ACTIVITY&quot;
[1] &quot;Tier 3A GOMF_KINASE_ACTIVITY&quot;
[1] &quot;Tier 3A GOMF_DNA_BINDING_TRANSCRIPTION_FACTOR_ACTIVITY&quot;
[1] &quot;Tier 3A 2&quot;
[1] &quot;Tier 3A 3&quot;
[1] &quot;Tier 3A 4&quot;
[1] &quot;Tier 3A 1&quot;
[1] &quot;Tier 3B GOMF_MONOATOMIC_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY&quot;
[1] &quot;Tier 3B GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY&quot;
[1] &quot;Tier 3B GOMF_NUCLEAR_RECEPTOR_ACTIVITY&quot;
[1] &quot;Tier 3B GOMF_KINASE_ACTIVITY&quot;
[1] &quot;Tier 3B GOMF_DNA_BINDING_TRANSCRIPTION_FACTOR_ACTIVITY&quot;
[1] &quot;Tier 3B 2&quot;
[1] &quot;Tier 3B 3&quot;
[1] &quot;Tier 3B 4&quot;
[1] &quot;Tier 3B 1&quot;
[1] &quot;SplicingPerturbed GOMF_MONOATOMIC_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY&quot;
[1] &quot;SplicingPerturbed GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY&quot;
[1] &quot;SplicingPerturbed GOMF_NUCLEAR_RECEPTOR_ACTIVITY&quot;
[1] &quot;SplicingPerturbed GOMF_KINASE_ACTIVITY&quot;
[1] &quot;SplicingPerturbed GOMF_DNA_BINDING_TRANSCRIPTION_FACTOR_ACTIVITY&quot;
[1] &quot;SplicingPerturbed 2&quot;
[1] &quot;SplicingPerturbed 3&quot;
[1] &quot;SplicingPerturbed 4&quot;
[1] &quot;SplicingPerturbed 1&quot;
[1] &quot;SplicingUnperturbedControl GOMF_MONOATOMIC_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY&quot;
[1] &quot;SplicingUnperturbedControl GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY&quot;
[1] &quot;SplicingUnperturbedControl GOMF_NUCLEAR_RECEPTOR_ACTIVITY&quot;
[1] &quot;SplicingUnperturbedControl GOMF_KINASE_ACTIVITY&quot;
[1] &quot;SplicingUnperturbedControl GOMF_DNA_BINDING_TRANSCRIPTION_FACTOR_ACTIVITY&quot;
[1] &quot;SplicingUnperturbedControl 2&quot;
[1] &quot;SplicingUnperturbedControl 3&quot;
[1] &quot;SplicingUnperturbedControl 4&quot;
[1] &quot;SplicingUnperturbedControl 1&quot;</code></pre>
<pre class="r"><code>results %&gt;%
  bind_rows(.id=&quot;Druggability_GO&quot;) %&gt;%
  separate(Druggability_GO, into=c(&quot;Druggability&quot;, &quot;GO category&quot;), sep=&quot;;&quot;) %&gt;%
  mutate(GO = recode(`GO category`, &quot;GOMF_KINASE_ACTIVITY&quot;=&quot;Kinases&quot;, &quot;GOMF_NUCLEAR_RECEPTOR_ACTIVITY&quot;=&quot;Nuc receptors&quot;, &quot;GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY&quot;=&quot;GPCRs&quot;, &quot;GOMF_MONOATOMIC_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY&quot;=&quot;Ion transporters&quot;, &quot;GOMF_DNA_BINDING_TRANSCRIPTION_FACTOR_ACTIVITY&quot;=&quot;TFs&quot;)) %&gt;%
  # filter(!GO==&quot;TFs&quot;) %&gt;%
  # pull(GO) %&gt;% unique()
  # mutate(GO = factor(GO, levels=c(&quot;1-10kb&quot;, &quot;10-50kb&quot;, &quot;50-100kb&quot;, &quot;&gt;100kb&quot;, &quot;GPCRs&quot;, &quot;Nuc receptors&quot;, &quot;Kinases&quot;, &quot;Ion transporters&quot;, &quot;TFs&quot;, &quot;BrainEnriched_HumanProteinAtlas&quot;))) %&gt;%
  filter(Druggability %in% c(&quot;Tier 1&quot;, &quot;SplicingPerturbed&quot;)) %&gt;%
  mutate(Druggability = recode(Druggability, &quot;Tier 1&quot;=&quot;1059 targets of approved small molecules&quot;, &quot;SplicingPerturbed&quot;=&quot;219 Risdiplam-induced NMD targets&quot;)) %&gt;%
  ggplot(aes(x=GO, y=log2(estimate), fill=GO)) +
  geom_col() +
  geom_errorbar(aes(ymin=log2(conf.low), ymax=log2(conf.high)), width=.2) +
  geom_hline(yintercept = 0) +
  Rotate_x_labels +
  scale_fill_manual(
    values=c(
      &quot;1&quot;=&quot;#b2e2e2&quot;,
      &quot;2&quot;=&quot;#66c2a4&quot;,
      &quot;3&quot;=&quot;#2ca25f&quot;,
      &quot;4&quot;=&quot;#006d2c&quot;,
      &quot;GPCRs&quot;=&quot;#ffffb3&quot;,
      &quot;Ion transporters&quot;=&quot;#bebada&quot;,
      &quot;Kinases&quot;=&quot;#fb8072&quot;,
      &quot;Nuc receptors&quot;=&quot;#80b1d3&quot;,
      &quot;TFs&quot;=&quot;#fdb462&quot;),
    labels=c(
      &quot;1&quot;=&quot;Q1 gene length; &lt;8.2kb&quot;,
      &quot;2&quot;=&quot;Q2 gene length; 8-23kb&quot;,
      &quot;3&quot;=&quot;Q3 gene length; 23-59kb&quot;,
      &quot;4&quot;=&quot;Q4 gene length; &gt;59kb&quot;,
      &quot;GPCRs&quot;=&quot;GPCRs&quot;,
      &quot;Ion transporters&quot;,
      &quot;Kinases&quot;,
      &quot;Nuc receptors&quot;=&quot;Nuclear receptors&quot;,
      &quot;TFs&quot;=&quot;Transcription factors&quot;
    )) +
  # scale_y_continuous(trans=&quot;log2&quot;) +
  facet_wrap(~Druggability, labeller = label_wrap_gen(25)) +
  labs(y=&quot;Enrichment; log2(OddsRatio)&quot;, x=&quot;Gene category&quot;, fill=&quot;Gene category&quot;)</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, I think I want to repeat that now but also add two more facets:
one for “disease genes” (ie OMIM), and another facet for the larger set
of post-txn regulated genes. First let’s explore the OMIM dataset
quickly…</p>
<pre class="r"><code>OMIM &lt;- read_tsv(&quot;/project2/yangili1/bjf79/20211209_JingxinRNAseq/code/OMIM/genemap2.txt&quot;, skip=3)

#Just dominant genes, search of Huntington&#39;s HTT gene, an expected result
OMIM %&gt;%
  filter(!is.na(Phenotypes)) %&gt;%
  filter(!is.na(`Ensembl Gene ID`)) %&gt;%
  filter(str_detect(Phenotypes, &quot;dominant&quot;)) %&gt;%
  filter(str_detect(Phenotypes, &quot;untington&quot;))</code></pre>
<pre><code># A tibble: 3 × 14
  `# Chromosome` `Genomic Position Start` `Genomic Position End` `Cyto Location`
  &lt;chr&gt;                             &lt;dbl&gt;                  &lt;dbl&gt; &lt;chr&gt;          
1 chr4                            3074680                3243959 4p16.3         
2 chr16                          87601834               87698155 16q24.3        
3 chr20                           4686455                4701587 20pter-p12     
# … with 10 more variables: `Computed Cyto Location` &lt;chr&gt;, `MIM Number` &lt;dbl&gt;,
#   `Gene Symbols` &lt;chr&gt;, `Gene Name` &lt;chr&gt;, `Approved Gene Symbol` &lt;chr&gt;,
#   `Entrez Gene ID` &lt;dbl&gt;, `Ensembl Gene ID` &lt;chr&gt;, Comments &lt;chr&gt;,
#   Phenotypes &lt;chr&gt;, `Mouse Gene Symbol/ID` &lt;chr&gt;</code></pre>
<pre class="r"><code># How many dominant OMIM genes are there?
OMIM %&gt;%
  filter(!is.na(Phenotypes)) %&gt;%
  filter(!is.na(`Ensembl Gene ID`)) %&gt;%
  filter(str_detect(Phenotypes, &quot;dominant&quot;)) %&gt;%
  distinct(`Ensembl Gene ID`) %&gt;% 
  nrow()</code></pre>
<pre><code>[1] 1834</code></pre>
<pre class="r"><code>GO.genes.of.interest.totest &lt;- 
  bind_rows(
    # filter out olfactor GPCRs from GPCR group
    GO.genes.of.interest %&gt;%
      filter(ensembl_gene_id %in% CodingGenes$gene) %&gt;%
      filter(!GO_Name==&quot;GOMF_OLFACTORY_RECEPTOR_ACTIVITY&quot;) %&gt;%
      filter(!(GO_Name==&quot;GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY&quot; &amp; !hgnc_symbol %in% GPCRs)),
  NumExonsAndLength %&gt;%
    filter(gene %in% CodingGenes$gene) %&gt;%
    # mutate(GO_Name = cut_number(GeneLength, 5, ordered_result=T)) %&gt;%
    mutate(GO_Name = factor(ntile(GeneLength, 4))) %&gt;%
    # mutate(GO_Name = cut(GeneLength, breaks=c(-Inf,1E3, 1E4, 1E5, Inf), include.lowest=T)) %&gt;%
    filter(!is.na(GO_Name)) %&gt;%
    dplyr::select(ensembl_gene_id=gene, GO_Name),
    OMIM %&gt;%
      filter(!is.na(Phenotypes)) %&gt;%
      filter(!is.na(`Ensembl Gene ID`)) %&gt;%
      filter(str_detect(Phenotypes, &quot;dominant&quot;)) %&gt;%
      dplyr::select(ensembl_gene_id = `Ensembl Gene ID`) %&gt;%
      distinct() %&gt;%
      filter(ensembl_gene_id %in% CodingGenes$gene) %&gt;%
      mutate(GO_Name = &quot;OMIM_DomNegative&quot;)
  )
GO_Categories &lt;- GO.genes.of.interest.totest %&gt;%
  pull(GO_Name) %&gt;% unique()

count(GO.genes.of.interest.totest, GO_Name)</code></pre>
<pre><code>                                                  GO_Name    n
1                                                       1 4621
2                                                       2 4621
3                                                       3 4621
4                                                       4 4621
5          GOMF_DNA_BINDING_TRANSCRIPTION_FACTOR_ACTIVITY 1537
6                GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY  489
7                                    GOMF_KINASE_ACTIVITY  793
8  GOMF_MONOATOMIC_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY  778
9                          GOMF_NUCLEAR_RECEPTOR_ACTIVITY   57
10                                       OMIM_DomNegative 1823</code></pre>
<pre class="r"><code>DruggabilityCategories.totest.df &lt;- GeneCategories %&gt;%
  sample_frac() %&gt;%
  group_by(Group) %&gt;%
  mutate(n = row_number()) %&gt;%
  ungroup() %&gt;%
  # filter(!Group == &quot;Tier 1&quot; | (Group == &quot;Tier 1&quot; &amp; n &lt;= 116))
  bind_rows(
    Scatter.dat %&gt;%
      filter(Category == &quot;post-txn down&quot;) %&gt;%
      filter(gene %in% CodingGenes$gene) %&gt;%
      dplyr::select(gene) %&gt;%
      mutate(Group = &quot;Larger set of risdiplam-effected&quot;),
    OMIM %&gt;%
      filter(!is.na(Phenotypes)) %&gt;%
      filter(!is.na(`Ensembl Gene ID`)) %&gt;%
      filter(str_detect(Phenotypes, &quot;dominant&quot;)) %&gt;%
      dplyr::select(gene = `Ensembl Gene ID`) %&gt;%
      distinct() %&gt;%
      filter(gene %in% CodingGenes$gene) %&gt;%
      mutate(Group = &quot;OMIM_DomNegative&quot;)
  )
count(DruggabilityCategories.totest.df, Group)</code></pre>
<pre><code># A tibble: 9 × 2
  Group                                n
  &lt;chr&gt;                            &lt;int&gt;
1 Larger set of risdiplam-effected   627
2 OMIM_DomNegative                  1823
3 SplicingPerturbed                  219
4 SplicingUnperturbedControl         396
5 Tier 1                            1059
6 Tier 2                             648
7 Tier 3A                             67
8 Tier 3B                            516
9 Whole protein coding genome      18484</code></pre>
<pre class="r"><code>DruggabilityCategories &lt;- DruggabilityCategories.totest.df %&gt;%
  filter(!Group==&quot;Whole protein coding genome&quot;) %&gt;% pull(Group) %&gt;% unique()
DruggabilityCategories</code></pre>
<pre><code>[1] &quot;Tier 3B&quot;                          &quot;Tier 1&quot;                          
[3] &quot;Tier 2&quot;                           &quot;SplicingUnperturbedControl&quot;      
[5] &quot;SplicingPerturbed&quot;                &quot;Tier 3A&quot;                         
[7] &quot;Larger set of risdiplam-effected&quot; &quot;OMIM_DomNegative&quot;                </code></pre>
<pre class="r"><code>results &lt;- list()
for (DruggabilitySetName in DruggabilityCategories){
  for (GO_CategoryName in GO_Categories){
    print(paste(DruggabilitySetName, GO_CategoryName))
    DruggabilitySet &lt;- DruggabilityCategories.totest.df %&gt;%
      filter(Group==DruggabilitySetName) %&gt;% pull(gene)
    GO_CategorySet &lt;- GO.genes.of.interest.totest %&gt;%
      filter(GO_Name==GO_CategoryName) %&gt;% pull(ensembl_gene_id)
    test.results &lt;- data.frame(gene=CodingGenes$gene) %&gt;%
      mutate(IsDruggable = gene %in% DruggabilitySet, IsInGO.Set = gene %in% GO_CategorySet) %&gt;%
      mutate(IsDruggable = factor(IsDruggable),
             IsInGO.Set = factor(IsInGO.Set)) %&gt;%
      count(IsDruggable, IsInGO.Set, .drop=F) %&gt;%
      pivot_wider(names_from=&quot;IsInGO.Set&quot;, values_from=&quot;n&quot;) %&gt;%
      column_to_rownames(&quot;IsDruggable&quot;) %&gt;%
      fisher.test() %&gt;% glance()
    results[[paste(DruggabilitySetName, GO_CategoryName, sep=&quot;;&quot;)]] &lt;- test.results %&gt;%
      as.data.frame()
  }
}</code></pre>
<pre><code>[1] &quot;Tier 3B GOMF_MONOATOMIC_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY&quot;
[1] &quot;Tier 3B GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY&quot;
[1] &quot;Tier 3B GOMF_NUCLEAR_RECEPTOR_ACTIVITY&quot;
[1] &quot;Tier 3B GOMF_KINASE_ACTIVITY&quot;
[1] &quot;Tier 3B GOMF_DNA_BINDING_TRANSCRIPTION_FACTOR_ACTIVITY&quot;
[1] &quot;Tier 3B 2&quot;
[1] &quot;Tier 3B 3&quot;
[1] &quot;Tier 3B 4&quot;
[1] &quot;Tier 3B 1&quot;
[1] &quot;Tier 3B OMIM_DomNegative&quot;
[1] &quot;Tier 1 GOMF_MONOATOMIC_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY&quot;
[1] &quot;Tier 1 GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY&quot;
[1] &quot;Tier 1 GOMF_NUCLEAR_RECEPTOR_ACTIVITY&quot;
[1] &quot;Tier 1 GOMF_KINASE_ACTIVITY&quot;
[1] &quot;Tier 1 GOMF_DNA_BINDING_TRANSCRIPTION_FACTOR_ACTIVITY&quot;
[1] &quot;Tier 1 2&quot;
[1] &quot;Tier 1 3&quot;
[1] &quot;Tier 1 4&quot;
[1] &quot;Tier 1 1&quot;
[1] &quot;Tier 1 OMIM_DomNegative&quot;
[1] &quot;Tier 2 GOMF_MONOATOMIC_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY&quot;
[1] &quot;Tier 2 GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY&quot;
[1] &quot;Tier 2 GOMF_NUCLEAR_RECEPTOR_ACTIVITY&quot;
[1] &quot;Tier 2 GOMF_KINASE_ACTIVITY&quot;
[1] &quot;Tier 2 GOMF_DNA_BINDING_TRANSCRIPTION_FACTOR_ACTIVITY&quot;
[1] &quot;Tier 2 2&quot;
[1] &quot;Tier 2 3&quot;
[1] &quot;Tier 2 4&quot;
[1] &quot;Tier 2 1&quot;
[1] &quot;Tier 2 OMIM_DomNegative&quot;
[1] &quot;SplicingUnperturbedControl GOMF_MONOATOMIC_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY&quot;
[1] &quot;SplicingUnperturbedControl GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY&quot;
[1] &quot;SplicingUnperturbedControl GOMF_NUCLEAR_RECEPTOR_ACTIVITY&quot;
[1] &quot;SplicingUnperturbedControl GOMF_KINASE_ACTIVITY&quot;
[1] &quot;SplicingUnperturbedControl GOMF_DNA_BINDING_TRANSCRIPTION_FACTOR_ACTIVITY&quot;
[1] &quot;SplicingUnperturbedControl 2&quot;
[1] &quot;SplicingUnperturbedControl 3&quot;
[1] &quot;SplicingUnperturbedControl 4&quot;
[1] &quot;SplicingUnperturbedControl 1&quot;
[1] &quot;SplicingUnperturbedControl OMIM_DomNegative&quot;
[1] &quot;SplicingPerturbed GOMF_MONOATOMIC_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY&quot;
[1] &quot;SplicingPerturbed GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY&quot;
[1] &quot;SplicingPerturbed GOMF_NUCLEAR_RECEPTOR_ACTIVITY&quot;
[1] &quot;SplicingPerturbed GOMF_KINASE_ACTIVITY&quot;
[1] &quot;SplicingPerturbed GOMF_DNA_BINDING_TRANSCRIPTION_FACTOR_ACTIVITY&quot;
[1] &quot;SplicingPerturbed 2&quot;
[1] &quot;SplicingPerturbed 3&quot;
[1] &quot;SplicingPerturbed 4&quot;
[1] &quot;SplicingPerturbed 1&quot;
[1] &quot;SplicingPerturbed OMIM_DomNegative&quot;
[1] &quot;Tier 3A GOMF_MONOATOMIC_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY&quot;
[1] &quot;Tier 3A GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY&quot;
[1] &quot;Tier 3A GOMF_NUCLEAR_RECEPTOR_ACTIVITY&quot;
[1] &quot;Tier 3A GOMF_KINASE_ACTIVITY&quot;
[1] &quot;Tier 3A GOMF_DNA_BINDING_TRANSCRIPTION_FACTOR_ACTIVITY&quot;
[1] &quot;Tier 3A 2&quot;
[1] &quot;Tier 3A 3&quot;
[1] &quot;Tier 3A 4&quot;
[1] &quot;Tier 3A 1&quot;
[1] &quot;Tier 3A OMIM_DomNegative&quot;
[1] &quot;Larger set of risdiplam-effected GOMF_MONOATOMIC_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY&quot;
[1] &quot;Larger set of risdiplam-effected GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY&quot;
[1] &quot;Larger set of risdiplam-effected GOMF_NUCLEAR_RECEPTOR_ACTIVITY&quot;
[1] &quot;Larger set of risdiplam-effected GOMF_KINASE_ACTIVITY&quot;
[1] &quot;Larger set of risdiplam-effected GOMF_DNA_BINDING_TRANSCRIPTION_FACTOR_ACTIVITY&quot;
[1] &quot;Larger set of risdiplam-effected 2&quot;
[1] &quot;Larger set of risdiplam-effected 3&quot;
[1] &quot;Larger set of risdiplam-effected 4&quot;
[1] &quot;Larger set of risdiplam-effected 1&quot;
[1] &quot;Larger set of risdiplam-effected OMIM_DomNegative&quot;
[1] &quot;OMIM_DomNegative GOMF_MONOATOMIC_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY&quot;
[1] &quot;OMIM_DomNegative GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY&quot;
[1] &quot;OMIM_DomNegative GOMF_NUCLEAR_RECEPTOR_ACTIVITY&quot;
[1] &quot;OMIM_DomNegative GOMF_KINASE_ACTIVITY&quot;
[1] &quot;OMIM_DomNegative GOMF_DNA_BINDING_TRANSCRIPTION_FACTOR_ACTIVITY&quot;
[1] &quot;OMIM_DomNegative 2&quot;
[1] &quot;OMIM_DomNegative 3&quot;
[1] &quot;OMIM_DomNegative 4&quot;
[1] &quot;OMIM_DomNegative 1&quot;
[1] &quot;OMIM_DomNegative OMIM_DomNegative&quot;</code></pre>
<pre class="r"><code>results %&gt;%
  bind_rows(.id=&quot;Druggability_GO&quot;) %&gt;%
  separate(Druggability_GO, into=c(&quot;Druggability&quot;, &quot;GO category&quot;), sep=&quot;;&quot;) %&gt;%
  mutate(GO = recode(`GO category`, &quot;GOMF_KINASE_ACTIVITY&quot;=&quot;Kinases&quot;, &quot;GOMF_NUCLEAR_RECEPTOR_ACTIVITY&quot;=&quot;Nuc receptors&quot;, &quot;GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY&quot;=&quot;GPCRs&quot;, &quot;GOMF_MONOATOMIC_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY&quot;=&quot;Ion transporters&quot;, &quot;GOMF_DNA_BINDING_TRANSCRIPTION_FACTOR_ACTIVITY&quot;=&quot;TFs&quot;)) %&gt;%
  # filter(!GO==&quot;TFs&quot;) %&gt;%
  # pull(GO) %&gt;% unique()
  # mutate(GO = factor(GO, levels=c(&quot;1-10kb&quot;, &quot;10-50kb&quot;, &quot;50-100kb&quot;, &quot;&gt;100kb&quot;, &quot;GPCRs&quot;, &quot;Nuc receptors&quot;, &quot;Kinases&quot;, &quot;Ion transporters&quot;, &quot;TFs&quot;, &quot;BrainEnriched_HumanProteinAtlas&quot;))) %&gt;%
  filter(Druggability %in% c(&quot;Tier 1&quot;, &quot;SplicingPerturbed&quot;, &quot;Larger set of risdiplam-effected&quot;, &quot;OMIM_DomNegative&quot;)) %&gt;%
  mutate(Druggability = recode(Druggability, &quot;Tier 1&quot;=&quot;1059 targets of approved small molecules&quot;, &quot;SplicingPerturbed&quot;=&quot;219 Risdiplam-induced NMD targets&quot;, &quot;Larger set of risdiplam-effected&quot;=&quot;627 post-txn down genes&quot;)) %&gt;%
  ggplot(aes(x=GO, y=log2(estimate), fill=GO)) +
  geom_col() +
  geom_errorbar(aes(ymin=log2(conf.low), ymax=log2(conf.high)), width=.2) +
  geom_hline(yintercept = 0) +
  scale_y_continuous(limits=c(-5,5)) +
  Rotate_x_labels +
  scale_fill_manual(
    values=c(
      &quot;1&quot;=&quot;#b2e2e2&quot;,
      &quot;2&quot;=&quot;#66c2a4&quot;,
      &quot;3&quot;=&quot;#2ca25f&quot;,
      &quot;4&quot;=&quot;#006d2c&quot;,
      &quot;GPCRs&quot;=&quot;#ffffb3&quot;,
      &quot;Ion transporters&quot;=&quot;#bebada&quot;,
      &quot;Kinases&quot;=&quot;#fb8072&quot;,
      &quot;Nuc receptors&quot;=&quot;#80b1d3&quot;,
      &quot;TFs&quot;=&quot;#fdb462&quot;,
      &quot;OMIM_DomNegative&quot;=&quot;#737373&quot;),
    labels=c(
      &quot;1&quot;=&quot;Q1 gene length; &lt;8.2kb&quot;,
      &quot;2&quot;=&quot;Q2 gene length; 8-23kb&quot;,
      &quot;3&quot;=&quot;Q3 gene length; 23-59kb&quot;,
      &quot;4&quot;=&quot;Q4 gene length; &gt;59kb&quot;,
      &quot;GPCRs&quot;=&quot;GPCRs&quot;,
      &quot;Ion transporters&quot;,
      &quot;Kinases&quot;,
      &quot;Nuc receptors&quot;=&quot;Nuclear receptors&quot;,
      &quot;TFs&quot;=&quot;Transcription factors&quot;,
      &quot;OMIM_DomNegative&quot;=&quot;OMIM_DomNegative&quot;
    )) +
  # scale_y_continuous(trans=&quot;log2&quot;) +
  facet_wrap(~Druggability, labeller = label_wrap_gen(25)) +
  labs(y=&quot;Enrichment; log2(OddsRatio)&quot;, x=&quot;Gene category&quot;, fill=&quot;Gene category&quot;)</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-26-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, so the conclusion is that risdiplam targets are particularly
enriched in long genes, and not particularly enriched in GPCRs, kinases,
etc. However, interestingly, the small molecule targets are also
somewhat enriched in long genes, though not to the extent of risdiplam
targets. Looking at OMIM dominant negative genes they are enriched in
OMIM doinant negative genes (obviously circular), and perhaps as
expected the small molecule targets are also enriched in them, and
interestingly risdiplam targets are slightly enriched. But also note
that the dominant negative OMIM genes are also slightly enriched in long
genes and gpcrs, kinases, etc. So in a way the existing small molecule
set is already more ‘matched’ for OMIM disease gene types… Maybe a
different way to present this is just as a stacked bar plots with
different categories…</p>
<p>But first let’s save a version of this plot for final fig.</p>
<pre class="r"><code>P.genesetEnrichment &lt;- results %&gt;%
  bind_rows(.id=&quot;Druggability_GO&quot;) %&gt;%
  separate(Druggability_GO, into=c(&quot;Druggability&quot;, &quot;GO category&quot;), sep=&quot;;&quot;) %&gt;%
  mutate(GO = recode(`GO category`, &quot;GOMF_KINASE_ACTIVITY&quot;=&quot;Kinases&quot;, &quot;GOMF_NUCLEAR_RECEPTOR_ACTIVITY&quot;=&quot;Nuc receptors&quot;, &quot;GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY&quot;=&quot;GPCRs&quot;, &quot;GOMF_MONOATOMIC_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY&quot;=&quot;Ion transporters&quot;, &quot;GOMF_DNA_BINDING_TRANSCRIPTION_FACTOR_ACTIVITY&quot;=&quot;TFs&quot;)) %&gt;%
  # filter(!GO==&quot;TFs&quot;) %&gt;%
  # pull(GO) %&gt;% unique()
  # mutate(GO = factor(GO, levels=c(&quot;1-10kb&quot;, &quot;10-50kb&quot;, &quot;50-100kb&quot;, &quot;&gt;100kb&quot;, &quot;GPCRs&quot;, &quot;Nuc receptors&quot;, &quot;Kinases&quot;, &quot;Ion transporters&quot;, &quot;TFs&quot;, &quot;BrainEnriched_HumanProteinAtlas&quot;))) %&gt;%
  filter(Druggability %in% c(&quot;Tier 1&quot;, &quot;SplicingPerturbed&quot;, &quot;Larger set of risdiplam-effected&quot;)) %&gt;%
  mutate(Druggability = recode(Druggability, &quot;Tier 1&quot;=&quot;1059 targets of approved small molecules&quot;, &quot;SplicingPerturbed&quot;=&quot;219 Risdiplam-induced NMD targets&quot;, &quot;Larger set of risdiplam-effected&quot;=&quot;627 post-txn down genes&quot;)) %&gt;%
  filter(!GO %in% c(&quot;OMIM_DomNegative&quot;)) %&gt;%
  ggplot(aes(x=GO, y=log2(estimate), fill=GO)) +
  geom_col() +
  geom_errorbar(aes(ymin=log2(conf.low), ymax=log2(conf.high)), width=.2) +
  geom_hline(yintercept = 0) +
  scale_y_continuous(limits=c(-5,5)) +
  Rotate_x_labels +
  scale_fill_manual(
    values=c(
      &quot;1&quot;=&quot;#b2e2e2&quot;,
      &quot;2&quot;=&quot;#66c2a4&quot;,
      &quot;3&quot;=&quot;#2ca25f&quot;,
      &quot;4&quot;=&quot;#006d2c&quot;,
      &quot;GPCRs&quot;=&quot;#ffffb3&quot;,
      &quot;Ion transporters&quot;=&quot;#bebada&quot;,
      &quot;Kinases&quot;=&quot;#fb8072&quot;,
      &quot;Nuc receptors&quot;=&quot;#80b1d3&quot;,
      &quot;TFs&quot;=&quot;#fdb462&quot;),
    labels=c(
      &quot;1&quot;=&quot;Q1 gene length; &lt;8.2kb&quot;,
      &quot;2&quot;=&quot;Q2 gene length; 8-23kb&quot;,
      &quot;3&quot;=&quot;Q3 gene length; 23-59kb&quot;,
      &quot;4&quot;=&quot;Q4 gene length; &gt;59kb&quot;,
      &quot;GPCRs&quot;=&quot;GPCRs&quot;,
      &quot;Ion transporters&quot;,
      &quot;Kinases&quot;,
      &quot;Nuc receptors&quot;=&quot;Nuclear receptors&quot;,
      &quot;TFs&quot;=&quot;Transcription factors&quot;
    )) +
  # scale_y_continuous(trans=&quot;log2&quot;) +
  scale_x_discrete(labels=c(&quot;Q1 gene length&quot;, &quot;Q2 gene length&quot;, &quot;Q3 gene length&quot;, &quot;Q4 gene length&quot;, &quot;GPCRs&quot;,
      &quot;Ion transporters&quot;,
      &quot;Kinases&quot;,
      &quot;Nuc receptors&quot;,
      &quot;TFs&quot;)) +
  facet_wrap(~Druggability, labeller = label_wrap_gen(25)) +
  labs(y=&quot;Enrichment; log2(OddsRatio)&quot;, x=&quot;Gene category&quot;, fill=&quot;Gene category&quot;)

P.genesetEnrichment</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>And also let’s print the table of p values and stuff for
reference…</p>
<pre class="r"><code>results %&gt;%
  bind_rows(.id=&quot;Druggability_GO&quot;) %&gt;%
  separate(Druggability_GO, into=c(&quot;Druggability&quot;, &quot;GO category&quot;), sep=&quot;;&quot;) %&gt;%
  mutate(GO = recode(`GO category`, &quot;GOMF_KINASE_ACTIVITY&quot;=&quot;Kinases&quot;, &quot;GOMF_NUCLEAR_RECEPTOR_ACTIVITY&quot;=&quot;Nuc receptors&quot;, &quot;GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY&quot;=&quot;GPCRs&quot;, &quot;GOMF_MONOATOMIC_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY&quot;=&quot;Ion transporters&quot;, &quot;GOMF_DNA_BINDING_TRANSCRIPTION_FACTOR_ACTIVITY&quot;=&quot;TFs&quot;)) %&gt;%
  dplyr::select(-method, -`GO category`) %&gt;%
  kable()</code></pre>
<table>
<colgroup>
<col width="30%" />
<col width="10%" />
<col width="9%" />
<col width="12%" />
<col width="10%" />
<col width="11%" />
<col width="15%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Druggability</th>
<th align="right">estimate</th>
<th align="right">p.value</th>
<th align="right">conf.low</th>
<th align="right">conf.high</th>
<th align="left">alternative</th>
<th align="left">GO</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Tier 3B</td>
<td align="right">9.1584316</td>
<td align="right">0.0000000</td>
<td align="right">7.2645120</td>
<td align="right">11.4817690</td>
<td align="left">two.sided</td>
<td align="left">Ion transporters</td>
</tr>
<tr class="even">
<td align="left">Tier 3B</td>
<td align="right">17.8318534</td>
<td align="right">0.0000000</td>
<td align="right">14.0484213</td>
<td align="right">22.5384382</td>
<td align="left">two.sided</td>
<td align="left">GPCRs</td>
</tr>
<tr class="odd">
<td align="left">Tier 3B</td>
<td align="right">7.1777300</td>
<td align="right">0.0000393</td>
<td align="right">2.9207456</td>
<td align="right">15.3880137</td>
<td align="left">two.sided</td>
<td align="left">Nuc receptors</td>
</tr>
<tr class="even">
<td align="left">Tier 3B</td>
<td align="right">4.2068880</td>
<td align="right">0.0000000</td>
<td align="right">3.1530710</td>
<td align="right">5.5375236</td>
<td align="left">two.sided</td>
<td align="left">Kinases</td>
</tr>
<tr class="odd">
<td align="left">Tier 3B</td>
<td align="right">0.2148774</td>
<td align="right">0.0000000</td>
<td align="right">0.0920764</td>
<td align="right">0.4277921</td>
<td align="left">two.sided</td>
<td align="left">TFs</td>
</tr>
<tr class="even">
<td align="left">Tier 3B</td>
<td align="right">1.0360823</td>
<td align="right">0.7393335</td>
<td align="right">0.8273790</td>
<td align="right">1.2882130</td>
<td align="left">two.sided</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">Tier 3B</td>
<td align="right">0.9381724</td>
<td align="right">0.6175399</td>
<td align="right">0.7439220</td>
<td align="right">1.1736752</td>
<td align="left">two.sided</td>
<td align="left">3</td>
</tr>
<tr class="even">
<td align="left">Tier 3B</td>
<td align="right">1.1252871</td>
<td align="right">0.2674937</td>
<td align="right">0.9034546</td>
<td align="right">1.3925325</td>
<td align="left">two.sided</td>
<td align="left">4</td>
</tr>
<tr class="odd">
<td align="left">Tier 3B</td>
<td align="right">1.5953272</td>
<td align="right">0.0000049</td>
<td align="right">1.3049055</td>
<td align="right">1.9423057</td>
<td align="left">two.sided</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">Tier 3B</td>
<td align="right">1.5695556</td>
<td align="right">0.0017286</td>
<td align="right">1.1759211</td>
<td align="right">2.0639941</td>
<td align="left">two.sided</td>
<td align="left">OMIM_DomNegative</td>
</tr>
<tr class="odd">
<td align="left">Tier 1</td>
<td align="right">4.5775591</td>
<td align="right">0.0000000</td>
<td align="right">3.7197576</td>
<td align="right">5.6022199</td>
<td align="left">two.sided</td>
<td align="left">Ion transporters</td>
</tr>
<tr class="even">
<td align="left">Tier 1</td>
<td align="right">10.0233817</td>
<td align="right">0.0000000</td>
<td align="right">8.1091459</td>
<td align="right">12.3478953</td>
<td align="left">two.sided</td>
<td align="left">GPCRs</td>
</tr>
<tr class="odd">
<td align="left">Tier 1</td>
<td align="right">19.0191977</td>
<td align="right">0.0000000</td>
<td align="right">10.8351038</td>
<td align="right">33.2253463</td>
<td align="left">two.sided</td>
<td align="left">Nuc receptors</td>
</tr>
<tr class="even">
<td align="left">Tier 1</td>
<td align="right">18.6713523</td>
<td align="right">0.0000000</td>
<td align="right">15.8462886</td>
<td align="right">21.9655357</td>
<td align="left">two.sided</td>
<td align="left">Kinases</td>
</tr>
<tr class="odd">
<td align="left">Tier 1</td>
<td align="right">0.4073325</td>
<td align="right">0.0000000</td>
<td align="right">0.2742487</td>
<td align="right">0.5844331</td>
<td align="left">two.sided</td>
<td align="left">TFs</td>
</tr>
<tr class="even">
<td align="left">Tier 1</td>
<td align="right">1.1224319</td>
<td align="right">0.1361653</td>
<td align="right">0.9627033</td>
<td align="right">1.3047237</td>
<td align="left">two.sided</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">Tier 1</td>
<td align="right">1.4346905</td>
<td align="right">0.0000011</td>
<td align="right">1.2415966</td>
<td align="right">1.6542055</td>
<td align="left">two.sided</td>
<td align="left">3</td>
</tr>
<tr class="even">
<td align="left">Tier 1</td>
<td align="right">2.0315724</td>
<td align="right">0.0000000</td>
<td align="right">1.7744165</td>
<td align="right">2.3226804</td>
<td align="left">two.sided</td>
<td align="left">4</td>
</tr>
<tr class="odd">
<td align="left">Tier 1</td>
<td align="right">0.6106577</td>
<td align="right">0.0000000</td>
<td align="right">0.5070732</td>
<td align="right">0.7310714</td>
<td align="left">two.sided</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">Tier 1</td>
<td align="right">4.6770866</td>
<td align="right">0.0000000</td>
<td align="right">4.0250797</td>
<td align="right">5.4242085</td>
<td align="left">two.sided</td>
<td align="left">OMIM_DomNegative</td>
</tr>
<tr class="odd">
<td align="left">Tier 2</td>
<td align="right">3.3065440</td>
<td align="right">0.0000000</td>
<td align="right">2.4867452</td>
<td align="right">4.3360520</td>
<td align="left">two.sided</td>
<td align="left">Ion transporters</td>
</tr>
<tr class="even">
<td align="left">Tier 2</td>
<td align="right">3.7628017</td>
<td align="right">0.0000000</td>
<td align="right">2.6857913</td>
<td align="right">5.1671221</td>
<td align="left">two.sided</td>
<td align="left">GPCRs</td>
</tr>
<tr class="odd">
<td align="left">Tier 2</td>
<td align="right">6.5159324</td>
<td align="right">0.0000294</td>
<td align="right">2.7985334</td>
<td align="right">13.5045955</td>
<td align="left">two.sided</td>
<td align="left">Nuc receptors</td>
</tr>
<tr class="even">
<td align="left">Tier 2</td>
<td align="right">7.2893233</td>
<td align="right">0.0000000</td>
<td align="right">5.8415702</td>
<td align="right">9.0413296</td>
<td align="left">two.sided</td>
<td align="left">Kinases</td>
</tr>
<tr class="odd">
<td align="left">Tier 2</td>
<td align="right">0.5740051</td>
<td align="right">0.0040124</td>
<td align="right">0.3706802</td>
<td align="right">0.8527182</td>
<td align="left">two.sided</td>
<td align="left">TFs</td>
</tr>
<tr class="even">
<td align="left">Tier 2</td>
<td align="right">1.2090959</td>
<td align="right">0.0469397</td>
<td align="right">0.9980052</td>
<td align="right">1.4579726</td>
<td align="left">two.sided</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">Tier 2</td>
<td align="right">1.6053487</td>
<td align="right">0.0000002</td>
<td align="right">1.3420347</td>
<td align="right">1.9141582</td>
<td align="left">two.sided</td>
<td align="left">3</td>
</tr>
<tr class="even">
<td align="left">Tier 2</td>
<td align="right">1.4228886</td>
<td align="right">0.0001555</td>
<td align="right">1.1836140</td>
<td align="right">1.7040751</td>
<td align="left">two.sided</td>
<td align="left">4</td>
</tr>
<tr class="odd">
<td align="left">Tier 2</td>
<td align="right">0.7421310</td>
<td align="right">0.0053825</td>
<td align="right">0.5937480</td>
<td align="right">0.9201179</td>
<td align="left">two.sided</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">Tier 2</td>
<td align="right">2.4238705</td>
<td align="right">0.0000000</td>
<td align="right">1.9429889</td>
<td align="right">3.0018796</td>
<td align="left">two.sided</td>
<td align="left">OMIM_DomNegative</td>
</tr>
<tr class="odd">
<td align="left">SplicingUnperturbedControl</td>
<td align="right">0.3653269</td>
<td align="right">0.0156537</td>
<td align="right">0.1176105</td>
<td align="right">0.8637518</td>
<td align="left">two.sided</td>
<td align="left">Ion transporters</td>
</tr>
<tr class="even">
<td align="left">SplicingUnperturbedControl</td>
<td align="right">0.1160682</td>
<td align="right">0.0039433</td>
<td align="right">0.0029283</td>
<td align="right">0.6534926</td>
<td align="left">two.sided</td>
<td align="left">GPCRs</td>
</tr>
<tr class="odd">
<td align="left">SplicingUnperturbedControl</td>
<td align="right">0.0000000</td>
<td align="right">1.0000000</td>
<td align="right">0.0000000</td>
<td align="right">3.8681414</td>
<td align="left">two.sided</td>
<td align="left">Nuc receptors</td>
</tr>
<tr class="even">
<td align="left">SplicingUnperturbedControl</td>
<td align="right">1.5204465</td>
<td align="right">0.0908983</td>
<td align="right">0.9126993</td>
<td align="right">2.3993212</td>
<td align="left">two.sided</td>
<td align="left">Kinases</td>
</tr>
<tr class="odd">
<td align="left">SplicingUnperturbedControl</td>
<td align="right">0.7482665</td>
<td align="right">0.2601520</td>
<td align="right">0.4503874</td>
<td align="right">1.1769642</td>
<td align="left">two.sided</td>
<td align="left">TFs</td>
</tr>
<tr class="even">
<td align="left">SplicingUnperturbedControl</td>
<td align="right">1.2800803</td>
<td align="right">0.0415533</td>
<td align="right">1.0025661</td>
<td align="right">1.6223144</td>
<td align="left">two.sided</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">SplicingUnperturbedControl</td>
<td align="right">1.5278082</td>
<td align="right">0.0003469</td>
<td align="right">1.2093794</td>
<td align="right">1.9186500</td>
<td align="left">two.sided</td>
<td align="left">3</td>
</tr>
<tr class="even">
<td align="left">SplicingUnperturbedControl</td>
<td align="right">1.5680826</td>
<td align="right">0.0001267</td>
<td align="right">1.2430408</td>
<td align="right">1.9668070</td>
<td align="left">two.sided</td>
<td align="left">4</td>
</tr>
<tr class="odd">
<td align="left">SplicingUnperturbedControl</td>
<td align="right">0.9730301</td>
<td align="right">0.8987106</td>
<td align="right">0.7466031</td>
<td align="right">1.2551631</td>
<td align="left">two.sided</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">SplicingUnperturbedControl</td>
<td align="right">1.1430022</td>
<td align="right">0.4497635</td>
<td align="right">0.7808881</td>
<td align="right">1.6262270</td>
<td align="left">two.sided</td>
<td align="left">OMIM_DomNegative</td>
</tr>
<tr class="odd">
<td align="left">SplicingPerturbed</td>
<td align="right">0.9386204</td>
<td align="right">1.0000000</td>
<td align="right">0.3715479</td>
<td align="right">1.9773767</td>
<td align="left">two.sided</td>
<td align="left">Ion transporters</td>
</tr>
<tr class="even">
<td align="left">SplicingPerturbed</td>
<td align="right">0.0000000</td>
<td align="right">0.0163396</td>
<td align="right">0.0000000</td>
<td align="right">0.7741248</td>
<td align="left">two.sided</td>
<td align="left">GPCRs</td>
</tr>
<tr class="odd">
<td align="left">SplicingPerturbed</td>
<td align="right">0.0000000</td>
<td align="right">1.0000000</td>
<td align="right">0.0000000</td>
<td align="right">6.9654439</td>
<td align="left">two.sided</td>
<td align="left">Nuc receptors</td>
</tr>
<tr class="even">
<td align="left">SplicingPerturbed</td>
<td align="right">1.9206458</td>
<td align="right">0.0245324</td>
<td align="right">1.0268418</td>
<td align="right">3.3166145</td>
<td align="left">two.sided</td>
<td align="left">Kinases</td>
</tr>
<tr class="odd">
<td align="left">SplicingPerturbed</td>
<td align="right">0.5246647</td>
<td align="right">0.0763309</td>
<td align="right">0.2231702</td>
<td align="right">1.0557286</td>
<td align="left">two.sided</td>
<td align="left">TFs</td>
</tr>
<tr class="even">
<td align="left">SplicingPerturbed</td>
<td align="right">0.5538893</td>
<td align="right">0.0029441</td>
<td align="right">0.3552937</td>
<td align="right">0.8326365</td>
<td align="left">two.sided</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">SplicingPerturbed</td>
<td align="right">1.8721591</td>
<td align="right">0.0000414</td>
<td align="right">1.3860027</td>
<td align="right">2.5082922</td>
<td align="left">two.sided</td>
<td align="left">3</td>
</tr>
<tr class="even">
<td align="left">SplicingPerturbed</td>
<td align="right">4.5451953</td>
<td align="right">0.0000000</td>
<td align="right">3.4485599</td>
<td align="right">5.9978310</td>
<td align="left">two.sided</td>
<td align="left">4</td>
</tr>
<tr class="odd">
<td align="left">SplicingPerturbed</td>
<td align="right">0.1103190</td>
<td align="right">0.0000000</td>
<td align="right">0.0399805</td>
<td align="right">0.2443146</td>
<td align="left">two.sided</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">SplicingPerturbed</td>
<td align="right">2.1468543</td>
<td align="right">0.0002109</td>
<td align="right">1.4387333</td>
<td align="right">3.1190947</td>
<td align="left">two.sided</td>
<td align="left">OMIM_DomNegative</td>
</tr>
<tr class="odd">
<td align="left">Tier 3A</td>
<td align="right">19.8449432</td>
<td align="right">0.0000000</td>
<td align="right">11.6405941</td>
<td align="right">33.3450155</td>
<td align="left">two.sided</td>
<td align="left">Ion transporters</td>
</tr>
<tr class="even">
<td align="left">Tier 3A</td>
<td align="right">5.4098961</td>
<td align="right">0.0005572</td>
<td align="right">2.0754929</td>
<td align="right">11.9337986</td>
<td align="left">two.sided</td>
<td align="left">GPCRs</td>
</tr>
<tr class="odd">
<td align="left">Tier 3A</td>
<td align="right">19.7590682</td>
<td align="right">0.0006250</td>
<td align="right">3.8539321</td>
<td align="right">63.4995983</td>
<td align="left">two.sided</td>
<td align="left">Nuc receptors</td>
</tr>
<tr class="even">
<td align="left">Tier 3A</td>
<td align="right">5.5396300</td>
<td align="right">0.0000175</td>
<td align="right">2.6064320</td>
<td align="right">10.7419787</td>
<td align="left">two.sided</td>
<td align="left">Kinases</td>
</tr>
<tr class="odd">
<td align="left">Tier 3A</td>
<td align="right">0.6510472</td>
<td align="right">0.6269370</td>
<td align="right">0.1306695</td>
<td align="right">1.9937555</td>
<td align="left">two.sided</td>
<td align="left">TFs</td>
</tr>
<tr class="even">
<td align="left">Tier 3A</td>
<td align="right">1.1419718</td>
<td align="right">0.6476907</td>
<td align="right">0.5965028</td>
<td align="right">2.0619601</td>
<td align="left">two.sided</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">Tier 3A</td>
<td align="right">0.8630289</td>
<td align="right">0.7608965</td>
<td align="right">0.4202778</td>
<td align="right">1.6327580</td>
<td align="left">two.sided</td>
<td align="left">3</td>
</tr>
<tr class="even">
<td align="left">Tier 3A</td>
<td align="right">2.8520798</td>
<td align="right">0.0000493</td>
<td align="right">1.6883620</td>
<td align="right">4.7608384</td>
<td align="left">two.sided</td>
<td align="left">4</td>
</tr>
<tr class="odd">
<td align="left">Tier 3A</td>
<td align="right">0.5357802</td>
<td align="right">0.1253387</td>
<td align="right">0.2209096</td>
<td align="right">1.1271720</td>
<td align="left">two.sided</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">Tier 3A</td>
<td align="right">3.0707135</td>
<td align="right">0.0006864</td>
<td align="right">1.5702242</td>
<td align="right">5.6275106</td>
<td align="left">two.sided</td>
<td align="left">OMIM_DomNegative</td>
</tr>
<tr class="odd">
<td align="left">Larger set of risdiplam-effected</td>
<td align="right">0.6431856</td>
<td align="right">0.1165465</td>
<td align="right">0.3478389</td>
<td align="right">1.0943909</td>
<td align="left">two.sided</td>
<td align="left">Ion transporters</td>
</tr>
<tr class="even">
<td align="left">Larger set of risdiplam-effected</td>
<td align="right">0.3620125</td>
<td align="right">0.0160427</td>
<td align="right">0.1165987</td>
<td align="right">0.8552979</td>
<td align="left">two.sided</td>
<td align="left">GPCRs</td>
</tr>
<tr class="odd">
<td align="left">Larger set of risdiplam-effected</td>
<td align="right">0.0000000</td>
<td align="right">0.4098360</td>
<td align="right">0.0000000</td>
<td align="right">2.3765859</td>
<td align="left">two.sided</td>
<td align="left">Nuc receptors</td>
</tr>
<tr class="even">
<td align="left">Larger set of risdiplam-effected</td>
<td align="right">1.6256628</td>
<td align="right">0.0103286</td>
<td align="right">1.1067487</td>
<td align="right">2.3176707</td>
<td align="left">two.sided</td>
<td align="left">Kinases</td>
</tr>
<tr class="odd">
<td align="left">Larger set of risdiplam-effected</td>
<td align="right">0.4746449</td>
<td align="right">0.0002571</td>
<td align="right">0.2907583</td>
<td align="right">0.7349793</td>
<td align="left">two.sided</td>
<td align="left">TFs</td>
</tr>
<tr class="even">
<td align="left">Larger set of risdiplam-effected</td>
<td align="right">0.9349390</td>
<td align="right">0.5449357</td>
<td align="right">0.7576516</td>
<td align="right">1.1461573</td>
<td align="left">two.sided</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">Larger set of risdiplam-effected</td>
<td align="right">1.8919248</td>
<td align="right">0.0000000</td>
<td align="right">1.5862322</td>
<td align="right">2.2508137</td>
<td align="left">two.sided</td>
<td align="left">3</td>
</tr>
<tr class="even">
<td align="left">Larger set of risdiplam-effected</td>
<td align="right">2.1640447</td>
<td align="right">0.0000000</td>
<td align="right">1.8214016</td>
<td align="right">2.5655820</td>
<td align="left">two.sided</td>
<td align="left">4</td>
</tr>
<tr class="odd">
<td align="left">Larger set of risdiplam-effected</td>
<td align="right">0.4578311</td>
<td align="right">0.0000000</td>
<td align="right">0.3484867</td>
<td align="right">0.5929993</td>
<td align="left">two.sided</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">Larger set of risdiplam-effected</td>
<td align="right">1.0247280</td>
<td align="right">0.8809431</td>
<td align="right">0.7508390</td>
<td align="right">1.3722556</td>
<td align="left">two.sided</td>
<td align="left">OMIM_DomNegative</td>
</tr>
<tr class="odd">
<td align="left">OMIM_DomNegative</td>
<td align="right">3.0516993</td>
<td align="right">0.0000000</td>
<td align="right">2.5247921</td>
<td align="right">3.6715647</td>
<td align="left">two.sided</td>
<td align="left">Ion transporters</td>
</tr>
<tr class="even">
<td align="left">OMIM_DomNegative</td>
<td align="right">1.1765324</td>
<td align="right">0.3101460</td>
<td align="right">0.8425683</td>
<td align="right">1.6078825</td>
<td align="left">two.sided</td>
<td align="left">GPCRs</td>
</tr>
<tr class="odd">
<td align="left">OMIM_DomNegative</td>
<td align="right">5.3809014</td>
<td align="right">0.0000002</td>
<td align="right">2.8908877</td>
<td align="right">9.6572225</td>
<td align="left">two.sided</td>
<td align="left">Nuc receptors</td>
</tr>
<tr class="even">
<td align="left">OMIM_DomNegative</td>
<td align="right">2.3184865</td>
<td align="right">0.0000000</td>
<td align="right">1.8913974</td>
<td align="right">2.8247566</td>
<td align="left">two.sided</td>
<td align="left">Kinases</td>
</tr>
<tr class="odd">
<td align="left">OMIM_DomNegative</td>
<td align="right">2.0352455</td>
<td align="right">0.0000000</td>
<td align="right">1.7399303</td>
<td align="right">2.3727446</td>
<td align="left">two.sided</td>
<td align="left">TFs</td>
</tr>
<tr class="even">
<td align="left">OMIM_DomNegative</td>
<td align="right">0.9246512</td>
<td align="right">0.2124116</td>
<td align="right">0.8163187</td>
<td align="right">1.0452143</td>
<td align="left">two.sided</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">OMIM_DomNegative</td>
<td align="right">1.2483949</td>
<td align="right">0.0001601</td>
<td align="right">1.1121383</td>
<td align="right">1.3993429</td>
<td align="left">two.sided</td>
<td align="left">3</td>
</tr>
<tr class="even">
<td align="left">OMIM_DomNegative</td>
<td align="right">2.0875980</td>
<td align="right">0.0000000</td>
<td align="right">1.8789638</td>
<td align="right">2.3174869</td>
<td align="left">two.sided</td>
<td align="left">4</td>
</tr>
<tr class="odd">
<td align="left">OMIM_DomNegative</td>
<td align="right">0.8581283</td>
<td align="right">0.0162214</td>
<td align="right">0.7556052</td>
<td align="right">0.9723993</td>
<td align="left">two.sided</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">OMIM_DomNegative</td>
<td align="right">Inf</td>
<td align="right">0.0000000</td>
<td align="right">8402.0512839</td>
<td align="right">Inf</td>
<td align="left">two.sided</td>
<td align="left">OMIM_DomNegative</td>
</tr>
</tbody>
</table>
<pre class="r"><code>DruggabilityCategories</code></pre>
<pre><code>[1] &quot;Tier 3B&quot;                          &quot;Tier 1&quot;                          
[3] &quot;Tier 2&quot;                           &quot;SplicingUnperturbedControl&quot;      
[5] &quot;SplicingPerturbed&quot;                &quot;Tier 3A&quot;                         
[7] &quot;Larger set of risdiplam-effected&quot; &quot;OMIM_DomNegative&quot;                </code></pre>
<pre class="r"><code>GroupLabels &lt;- c(&quot;OMIM_DomNegative&quot;=&quot;Disease genes&quot;, &quot;Whole genome&quot;=&quot;Whole genome&quot;, &quot;Tier 1&quot;=&quot;1059 targets of approved small molecules&quot;, &quot;SplicingPerturbed&quot;=&quot;219 Risdiplam-induced NMD targets&quot;, &quot;Larger set of risdiplam-effected&quot;=&quot;627 post-txn down genes&quot;)

P.enrichmentByBarplot &lt;- bind_rows(
  DruggabilityCategories.totest.df %&gt;%
    filter(Group %in% c(&quot;Tier 1&quot;, &quot;OMIM_DomNegative&quot;, &quot;SplicingPerturbed&quot;, &quot;Larger set of risdiplam-effected&quot;)),
  CodingGenes %&gt;%
    mutate(Group = &quot;Whole genome&quot;)
) %&gt;%
  left_join(
    GO.genes.of.interest.totest %&gt;%
      filter(str_detect(GO_Name, &quot;GOMF&quot;)),
    by=c(&quot;gene&quot;=&quot;ensembl_gene_id&quot;)
  ) %&gt;%
  distinct(gene, Group, .keep_all=T) %&gt;%
  replace_na(list(GO_Name=&quot;Other&quot;)) %&gt;%
  mutate(GO = recode(`GO_Name`, &quot;GOMF_KINASE_ACTIVITY&quot;=&quot;Kinases&quot;, &quot;GOMF_NUCLEAR_RECEPTOR_ACTIVITY&quot;=&quot;Nuc receptors&quot;, &quot;GOMF_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY&quot;=&quot;GPCRs&quot;, &quot;GOMF_MONOATOMIC_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY&quot;=&quot;Ion transporters&quot;, &quot;GOMF_DNA_BINDING_TRANSCRIPTION_FACTOR_ACTIVITY&quot;=&quot;TFs&quot;)) %&gt;%
  count(Group, GO) %&gt;%
  mutate(Group = recode(Group, !!!GroupLabels)) %&gt;%
  mutate(Group = factor(Group, levels=c(&quot;1059 targets of approved small molecules&quot;, &quot;219 Risdiplam-induced NMD targets&quot;, &quot;627 post-txn down genes&quot;, &quot;Whole genome&quot;, &quot;Disease genes&quot;))) %&gt;%
  filter(!Group==&quot;627 post-txn down genes&quot;) %&gt;%
  ggplot(aes(x=Group, y=n, fill=GO)) +
  geom_col(position=&quot;fill&quot;) +
    scale_fill_manual(
    values=c(
      &quot;GPCRs&quot;=&quot;#ffffb3&quot;,
      &quot;Ion transporters&quot;=&quot;#bebada&quot;,
      &quot;Kinases&quot;=&quot;#fb8072&quot;,
      &quot;Nuc receptors&quot;=&quot;#80b1d3&quot;,
      &quot;TFs&quot;=&quot;#fdb462&quot;,
      &quot;Other&quot;=&quot;#737373&quot;),
    labels=c(
      &quot;GPCRs&quot;=&quot;GPCRs&quot;,
      &quot;Ion transporters&quot;,
      &quot;Kinases&quot;,
      &quot;Nuc receptors&quot;=&quot;Nuclear receptors&quot;,
      &quot;TFs&quot;=&quot;Transcription factors&quot;,
      &quot;Other&quot;=&quot;Other&quot;
    )) +
  scale_x_discrete(labels = function(x) str_wrap(x, width = 22), limits=rev) +
  scale_y_continuous(expand=c(0,0)) +
  labs(x=NULL, y=&quot;Fraction&quot;, fill=&quot;Gene category&quot;, caption=str_wrap(&quot;Disease genes are 1823 genes with dominant negative phenotypes in OMIM&quot;, 30)) +
  coord_flip()
P.enrichmentByBarplot</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>And also for reference, let’s print a list of the intersection of the
OMIM genes and the NMD target sets… This isn’t the strongest filtering
(for example, to really follow up on a potential target I would
definitely manually inspect the data a lot closer), it’s just a quick
reference.</p>
<pre class="r"><code>OMIM.risdiplam.genes &lt;- P.scatter.byPrediction.dat %&gt;%
  filter(FDR_polyA &lt; 0.1) %&gt;%
  filter(logFC_polyA &lt; -1) %&gt;%
  filter(ColorEffect == &quot;NMD&quot;) %&gt;%
  dplyr::rename(DeltaPSI=Estimate) %&gt;%
  inner_join(
    OMIM %&gt;%
      filter(!is.na(Phenotypes)) %&gt;%
      filter(!is.na(`Ensembl Gene ID`)) %&gt;%
      filter(str_detect(Phenotypes, &quot;dominant&quot;)) %&gt;%
      dplyr::select(&quot;MIM Number&quot;:&quot;Phenotypes&quot;),
    by=c(&quot;gene&quot;=&quot;Ensembl Gene ID&quot;)
  )

kable(OMIM.risdiplam.genes)</code></pre>
<table>
<colgroup>
<col width="2%" />
<col width="1%" />
<col width="1%" />
<col width="1%" />
<col width="1%" />
<col width="1%" />
<col width="1%" />
<col width="1%" />
<col width="2%" />
<col width="1%" />
<col width="2%" />
<col width="4%" />
<col width="1%" />
<col width="1%" />
<col width="7%" />
<col width="7%" />
<col width="3%" />
<col width="2%" />
<col width="5%" />
<col width="46%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">gene</th>
<th align="right">dose.nM</th>
<th align="right">logFC_chRNA</th>
<th align="right">logFC_polyA</th>
<th align="right">PValue_chRNA</th>
<th align="right">PValue_polyA</th>
<th align="right">FDR_chRNA</th>
<th align="right">FDR_polyA</th>
<th align="left">Category</th>
<th align="left">ColorEffect</th>
<th align="left">MergedEffect</th>
<th align="left">junc</th>
<th align="right">DeltaPSI</th>
<th align="right">MIM Number</th>
<th align="left">Gene Symbols</th>
<th align="left">Gene Name</th>
<th align="left">Approved Gene Symbol</th>
<th align="right">Entrez Gene ID</th>
<th align="left">Comments</th>
<th align="left">Phenotypes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000078403</td>
<td align="right">3160</td>
<td align="right">0.6755900</td>
<td align="right">-1.288923</td>
<td align="right">0.0000034</td>
<td align="right">0.0000000</td>
<td align="right">0.0001318</td>
<td align="right">0.0000002</td>
<td align="left">post-txn down</td>
<td align="left">NMD</td>
<td align="left">Frame shifting</td>
<td align="left">chr10:21728675:21730900:+</td>
<td align="right">0.7762386</td>
<td align="right">602409</td>
<td align="left">AF10</td>
<td align="left">ALL1 fused gene from chromosome 10</td>
<td align="left">MLLT10</td>
<td align="right">8028</td>
<td align="left">fuses with MLL and HEAB</td>
<td align="left">Leukemia, acute myeloid, 601626 (3), Somatic mutation,
Autosomal dominant</td>
</tr>
<tr class="even">
<td align="left">ENSG00000033030</td>
<td align="right">3160</td>
<td align="right">0.6288341</td>
<td align="right">-1.961917</td>
<td align="right">0.0000090</td>
<td align="right">0.0000000</td>
<td align="right">0.0002713</td>
<td align="right">0.0000000</td>
<td align="left">post-txn down</td>
<td align="left">NMD</td>
<td align="left">Frame shifting</td>
<td align="left">chr12:122478292:122478664:-</td>
<td align="right">0.7376685</td>
<td align="right">616381</td>
<td align="left">ZCCHC8, PFBMFT5</td>
<td align="left">Zinc finger CCHC domain-containing protein 8</td>
<td align="left">ZCCHC8</td>
<td align="right">55596</td>
<td align="left">mutation identified in 1 PFBMFT5 family</td>
<td align="left">?Pulmonary fibrosis and/or bone marrow failure,
telomere-related, 5, 618674 (3), Autosomal dominant</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000115415</td>
<td align="right">3160</td>
<td align="right">0.4927000</td>
<td align="right">-4.357511</td>
<td align="right">0.0005217</td>
<td align="right">0.0000000</td>
<td align="right">0.0059284</td>
<td align="right">0.0000000</td>
<td align="left">post-txn down</td>
<td align="left">NMD</td>
<td align="left">NMD</td>
<td align="left">chr2:190977025:190978528:-</td>
<td align="right">0.9243423</td>
<td align="right">600555</td>
<td align="left">STAT1, CANDF7, IMD31A, IMD31B, IMD31C</td>
<td align="left">Signal transducer and activator of transcription-1</td>
<td align="left">STAT1</td>
<td align="right">6772</td>
<td align="left">NA</td>
<td align="left">Immunodeficiency 31C, chronic mucocutaneous
candidiasis, autosomal dominant, 614162 (3), Autosomal dominant;
Immunodeficiency 31A, mycobacteriosis, autosomal dominant, 614892 (3),
Autosomal dominant; Immunodeficiency 31B, mycobacterial and viral
infections, autosomal recessive, 613796 (3), Autosomal recessive</td>
</tr>
<tr class="even">
<td align="left">ENSG00000103335</td>
<td align="right">3160</td>
<td align="right">0.5567517</td>
<td align="right">-1.099484</td>
<td align="right">0.0005570</td>
<td align="right">0.0015190</td>
<td align="right">0.0062411</td>
<td align="right">0.0087129</td>
<td align="left">post-txn down</td>
<td align="left">NMD</td>
<td align="left">Frame shifting</td>
<td align="left">chr16:88720532:88720660:-</td>
<td align="right">1.0025870</td>
<td align="right">611184</td>
<td align="left">PIEZO1, FAM38A, MIB, DHS, LMPHM6</td>
<td align="left">PIEZO1 ion channel</td>
<td align="left">PIEZO1</td>
<td align="right">9780</td>
<td align="left">NA</td>
<td align="left">Lymphatic malformation 6, 616843 (3), Autosomal
recessive; Dehydrated hereditary stomatocytosis with or without
pseudohyperkalemia and/or perinatal edema, 194380 (3), Autosomal
dominant</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000100485</td>
<td align="right">3160</td>
<td align="right">0.3876746</td>
<td align="right">-1.104077</td>
<td align="right">0.0009561</td>
<td align="right">0.0086641</td>
<td align="right">0.0094387</td>
<td align="right">0.0339973</td>
<td align="left">post-txn down</td>
<td align="left">NMD</td>
<td align="left">Frame shifting</td>
<td align="left">chr14:50130762:50133808:-</td>
<td align="right">0.9987109</td>
<td align="right">601247</td>
<td align="left">SOS2, NS9</td>
<td align="left">SOS Ras/Rac guanine nucleotide exchange factor 2</td>
<td align="left">SOS2</td>
<td align="right">6655</td>
<td align="left">NA</td>
<td align="left">Noonan syndrome 9, 616559 (3), Autosomal dominant</td>
</tr>
<tr class="even">
<td align="left">ENSG00000096093</td>
<td align="right">3160</td>
<td align="right">0.0248003</td>
<td align="right">-1.004033</td>
<td align="right">0.8927372</td>
<td align="right">0.0278749</td>
<td align="right">0.9771466</td>
<td align="right">0.0810435</td>
<td align="left">post-txn down</td>
<td align="left">NMD</td>
<td align="left">Frame shifting</td>
<td align="left">chr6:52428243:52438304:+</td>
<td align="right">0.2162461</td>
<td align="right">608815</td>
<td align="left">EFHC1, FLJ10466, EJM1, JAE, EJA1</td>
<td align="left">EF hand domain (C-terminal)-containing 1</td>
<td align="left">EFHC1</td>
<td align="right">114327</td>
<td align="left">NA</td>
<td align="left">{Epilepsy, juvenile absence, susceptibility to, 1},
607631 (3), Autosomal dominant; {Myoclonic epilepsy, juvenile,
susceptibility to, 1}, 254770 (3), Autosomal dominant</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000197386</td>
<td align="right">3160</td>
<td align="right">NA</td>
<td align="right">-1.836470</td>
<td align="right">NA</td>
<td align="right">0.0017826</td>
<td align="right">NA</td>
<td align="right">0.0099649</td>
<td align="left">Other</td>
<td align="left">NMD</td>
<td align="left">Frame shifting</td>
<td align="left">chr4:3213736:3213958:+</td>
<td align="right">0.9595212</td>
<td align="right">613004</td>
<td align="left">HTT, HD, IT15, LOMARS</td>
<td align="left">Huntingtin</td>
<td align="left">HTT</td>
<td align="right">3064</td>
<td align="left">distal to D4S10</td>
<td align="left">Lopes-Maciel-Rodan syndrome, 617435 (3), Autosomal
recessive; Huntington disease, 143100 (3), Autosomal dominant</td>
</tr>
<tr class="even">
<td align="left">ENSG00000159063</td>
<td align="right">3160</td>
<td align="right">NA</td>
<td align="right">-2.969738</td>
<td align="right">NA</td>
<td align="right">0.0000000</td>
<td align="right">NA</td>
<td align="right">0.0000000</td>
<td align="left">Other</td>
<td align="left">NMD</td>
<td align="left">Frame shifting</td>
<td align="left">chr11:78106946:78108970:-</td>
<td align="right">0.2456783</td>
<td align="right">608103</td>
<td align="left">ALG8, CDG1H, PCLD3</td>
<td align="left">ALG8 alpha-1,3-glucosyltransferase</td>
<td align="left">ALG8</td>
<td align="right">79053</td>
<td align="left">NA</td>
<td align="left">Congenital disorder of glycosylation, type Ih, 608104
(3), Autosomal recessive; Polycystic liver disease 3 with or without
kidney cysts, 617874 (3), Autosomal dominant</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000083799</td>
<td align="right">3160</td>
<td align="right">NA</td>
<td align="right">-1.214614</td>
<td align="right">NA</td>
<td align="right">0.0000207</td>
<td align="right">NA</td>
<td align="right">0.0002209</td>
<td align="left">Other</td>
<td align="left">NMD</td>
<td align="left">In-frame PTC</td>
<td align="left">chr16:50775855:50776179:+</td>
<td align="right">0.8050513</td>
<td align="right">605018</td>
<td align="left">CYLD, CDMT, EAC, MFT1, KIAA0849, BRSS, FTDALS8</td>
<td align="left">CYLD lysine-63 deubiquitinase</td>
<td align="left">CYLD</td>
<td align="right">1540</td>
<td align="left">mutation identified in 1 FTDALS8 family</td>
<td align="left">Brooke-Spiegler syndrome, 605041 (3), Autosomal
dominant; Cylindromatosis, familial, 132700 (3), Autosomal dominant;
Trichoepithelioma, multiple familial, 1, 601606 (3), Autosomal dominant;
?Frontotemporal dementia and/or amyotrophic lateral sclerosis 8, 619132
(3), Autosomal dominant</td>
</tr>
<tr class="even">
<td align="left">ENSG00000154864</td>
<td align="right">3160</td>
<td align="right">NA</td>
<td align="right">-1.163771</td>
<td align="right">NA</td>
<td align="right">0.0057753</td>
<td align="right">NA</td>
<td align="right">0.0250624</td>
<td align="left">Other</td>
<td align="left">NMD</td>
<td align="left">Frame shifting</td>
<td align="left">chr18:10702171:10702546:-</td>
<td align="right">0.9428523</td>
<td align="right">613629</td>
<td align="left">PIEZO2, FAM38B, DA5, DA3, MWKS, DAIPT</td>
<td align="left">PIEZO-type mechanosensitive ion channel component
2</td>
<td align="left">PIEZO2</td>
<td align="right">63895</td>
<td align="left">mutation identified in 1 MWKS patient</td>
<td align="left">Arthrogryposis, distal, type 5, 108145 (3), Autosomal
dominant; Arthrogryposis, distal, with impaired proprioception and
touch, 617146 (3), Autosomal recessive; Arthrogryposis, distal, type 3,
114300 (3), Autosomal dominant; ?Marden-Walker syndrome, 248700 (3),
Autosomal dominant</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000154864</td>
<td align="right">3160</td>
<td align="right">NA</td>
<td align="right">-1.163771</td>
<td align="right">NA</td>
<td align="right">0.0057753</td>
<td align="right">NA</td>
<td align="right">0.0250624</td>
<td align="left">Other</td>
<td align="left">NMD</td>
<td align="left">Frame shifting</td>
<td align="left">chr18:10911228:10963755:-</td>
<td align="right">0.0729154</td>
<td align="right">613629</td>
<td align="left">PIEZO2, FAM38B, DA5, DA3, MWKS, DAIPT</td>
<td align="left">PIEZO-type mechanosensitive ion channel component
2</td>
<td align="left">PIEZO2</td>
<td align="right">63895</td>
<td align="left">mutation identified in 1 MWKS patient</td>
<td align="left">Arthrogryposis, distal, type 5, 108145 (3), Autosomal
dominant; Arthrogryposis, distal, with impaired proprioception and
touch, 617146 (3), Autosomal recessive; Arthrogryposis, distal, type 3,
114300 (3), Autosomal dominant; ?Marden-Walker syndrome, 248700 (3),
Autosomal dominant</td>
</tr>
<tr class="even">
<td align="left">ENSG00000154864</td>
<td align="right">3160</td>
<td align="right">NA</td>
<td align="right">-1.163771</td>
<td align="right">NA</td>
<td align="right">0.0057753</td>
<td align="right">NA</td>
<td align="right">0.0250624</td>
<td align="left">Other</td>
<td align="left">NMD</td>
<td align="left">Frame shifting</td>
<td align="left">chr18:10702171:10702546:-</td>
<td align="right">0.9428523</td>
<td align="right">613629</td>
<td align="left">PIEZO2, FAM38B, DA5, DA3, MWKS, DAIPT</td>
<td align="left">PIEZO-type mechanosensitive ion channel component
2</td>
<td align="left">PIEZO2</td>
<td align="right">63895</td>
<td align="left">mutation identified in 1 MWKS patient</td>
<td align="left">Arthrogryposis, distal, type 5, 108145 (3), Autosomal
dominant; Arthrogryposis, distal, with impaired proprioception and
touch, 617146 (3), Autosomal recessive; Arthrogryposis, distal, type 3,
114300 (3), Autosomal dominant; ?Marden-Walker syndrome, 248700 (3),
Autosomal dominant</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000154864</td>
<td align="right">3160</td>
<td align="right">NA</td>
<td align="right">-1.163771</td>
<td align="right">NA</td>
<td align="right">0.0057753</td>
<td align="right">NA</td>
<td align="right">0.0250624</td>
<td align="left">Other</td>
<td align="left">NMD</td>
<td align="left">Frame shifting</td>
<td align="left">chr18:10911228:10963755:-</td>
<td align="right">0.0729154</td>
<td align="right">613629</td>
<td align="left">PIEZO2, FAM38B, DA5, DA3, MWKS, DAIPT</td>
<td align="left">PIEZO-type mechanosensitive ion channel component
2</td>
<td align="left">PIEZO2</td>
<td align="right">63895</td>
<td align="left">mutation identified in 1 MWKS patient</td>
<td align="left">Arthrogryposis, distal, type 5, 108145 (3), Autosomal
dominant; Arthrogryposis, distal, with impaired proprioception and
touch, 617146 (3), Autosomal recessive; Arthrogryposis, distal, type 3,
114300 (3), Autosomal dominant; ?Marden-Walker syndrome, 248700 (3),
Autosomal dominant</td>
</tr>
</tbody>
</table>
<p>Now let’s plot the ecdf of gene length nicer for possible
publication…</p>
<pre class="r"><code>GeneLength.dat &lt;- GeneCategories %&gt;%
  left_join(NumExonsAndLength) %&gt;%
  filter(!Group %in% c(&quot;Tier 2&quot;, &quot;Tier 3A&quot;, &quot;Tier 3B&quot;)) %&gt;%
  mutate(GeneLength = GeneLength/1E3)


GeneLength.dat %&gt;% count(Group)</code></pre>
<pre><code># A tibble: 4 × 2
  Group                           n
  &lt;chr&gt;                       &lt;int&gt;
1 SplicingPerturbed             219
2 SplicingUnperturbedControl    396
3 Tier 1                       1059
4 Whole protein coding genome 18484</code></pre>
<pre class="r"><code>GeneLength.dat %&gt;%
  filter(Group %in% c(&quot;SplicingPerturbed&quot;, &quot;SplicingUnperturbedControl&quot;)) %&gt;%
  wilcox.test(GeneLength~Group,data = .)</code></pre>
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  GeneLength by Group
W = 60572, p-value = 3.458e-16
alternative hypothesis: true location shift is not equal to 0</code></pre>
<pre class="r"><code>P.geneLength &lt;- ggplot(GeneLength.dat, aes(x=GeneLength, color=Group)) +
  stat_ecdf() +
  scale_x_continuous(trans=&#39;log10&#39;) +
  annotation_logticks() +
  coord_cartesian(xlim=c(1, 500)) +
  scale_color_manual(
    labels=c(
      &quot;SplicingPerturbed&quot;=&quot;219 Risdiplam-induced NMD targets&quot;,
      &quot;SplicingUnperturbedControl&quot;=&quot;389 expression-matched genes&quot;,
      &quot;Whole protein coding genome&quot;=&quot;Whole genome&quot;,
      &quot;Tier 1&quot;=&quot;1059 targets of approved small molecules&quot;),
    values=c(
      &quot;SplicingPerturbed&quot;=&quot;#e31a1c&quot;,
      &quot;SplicingUnperturbedControl&quot;=&quot;#1f78b4&quot;,
      &quot;Whole protein coding genome&quot;=&quot;#000000&quot;,
      &quot;Tier 1&quot;=&quot;#737373&quot;)
  ) +
  labs(y=&quot;ecdf&quot;, color=&quot;Druggability gene group&quot;, x=&quot;Gene length (kb)&quot;, caption=&quot;Splicing perturbed to control P: 3.4x10^-15&quot;)
P.geneLength</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Even though the point is that longer genes are more likely to have a
splicing change due to chance, I kind of want to repeat this analysis in
a way that isn’t dependent on splicing and has no testing difference in
long vs short genes… That is, by looking at post-txn down genes, they
should be longer than expression matched controls…</p>
<pre class="r"><code>SplicingDruggableGenes_ByPostTxn &lt;- Scatter.dat %&gt;%
  filter(Category == &quot;post-txn down&quot;) %&gt;%
  filter(gene %in% CodingGenes$gene) %&gt;%
  dplyr::select(gene) %&gt;%
  distinct(gene, .keep_all=T)

Expression.tidyForMatchingGenes &lt;- Expression.table.genes %&gt;%
  mutate(IsSplicingDruggable = gene %in% SplicingDruggableGenes_ByPostTxn$gene) %&gt;%
  arrange(medianTPM) %&gt;%
  mutate(LaggingGeneGroup = lag(IsSplicingDruggable)) %&gt;%
  mutate(LeadingGeneGroup = lead(IsSplicingDruggable)) %&gt;%
  ungroup()

Merged.WithExpressionMatchedControlGenes &lt;- 
bind_rows(
  Expression.tidyForMatchingGenes %&gt;%
    filter(IsSplicingDruggable) %&gt;%
    mutate(Group = &quot;SplicingPerturbed&quot;),
  Expression.tidyForMatchingGenes %&gt;%
    filter(!IsSplicingDruggable &amp; LaggingGeneGroup) %&gt;%
    mutate(Group = &quot;SplicingUnperturbedControl&quot;),
  Expression.tidyForMatchingGenes %&gt;%
    filter(!IsSplicingDruggable &amp; LeadingGeneGroup) %&gt;%
    mutate(Group = &quot;SplicingUnperturbedControl&quot;),
) %&gt;%
  dplyr::select(gene, medianTPM, Group)

Merged.WithExpressionMatchedControlGenes %&gt;%
  ggplot(aes(medianTPM, color=Group)) +
  stat_ecdf() +
  scale_x_continuous(trans=&#39;log10&#39;)</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>GeneLength.dat &lt;- GeneCategories %&gt;%
  filter(Group %in% c(&quot;Tier 1&quot;, &quot;Whole protein coding genome&quot;)) %&gt;%
  bind_rows(Merged.WithExpressionMatchedControlGenes) %&gt;%
  left_join(NumExonsAndLength) %&gt;%
  mutate(GeneLength = GeneLength/1E3) 

GeneLength.dat %&gt;%
  count(Group)</code></pre>
<pre><code># A tibble: 4 × 2
  Group                           n
  &lt;chr&gt;                       &lt;int&gt;
1 SplicingPerturbed             604
2 SplicingUnperturbedControl   1142
3 Tier 1                       1059
4 Whole protein coding genome 18484</code></pre>
<pre class="r"><code>GeneLength.dat %&gt;%
  filter(Group %in% c(&quot;SplicingPerturbed&quot;, &quot;SplicingUnperturbedControl&quot;)) %&gt;%
  wilcox.test(GeneLength~Group,data = .)</code></pre>
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  GeneLength by Group
W = 427220, p-value &lt; 2.2e-16
alternative hypothesis: true location shift is not equal to 0</code></pre>
<pre class="r"><code>P.geneLength.FromPostTxnSet &lt;- ggplot(GeneLength.dat, aes(x=GeneLength, color=Group)) +
  stat_ecdf() +
  scale_x_continuous(trans=&#39;log10&#39;) +
  annotation_logticks() +
  coord_cartesian(xlim=c(1, 500)) +
  scale_color_manual(
    labels=c(
      &quot;SplicingPerturbed&quot;=&quot;604 Risdiplam-induced post-txn down genes&quot;,
      &quot;SplicingUnperturbedControl&quot;=&quot;1142 expression-matched genes&quot;,
      &quot;Whole protein coding genome&quot;=&quot;Whole genome&quot;,
      &quot;Tier 1&quot;=&quot;1059 targets of approved small molecules&quot;),
    values=c(
      &quot;SplicingPerturbed&quot;=&quot;#e31a1c&quot;,
      &quot;SplicingUnperturbedControl&quot;=&quot;#1f78b4&quot;,
      &quot;Whole protein coding genome&quot;=&quot;#000000&quot;,
      &quot;Tier 1&quot;=&quot;#737373&quot;)
  ) +
  labs(y=&quot;ecdf&quot;, color=&quot;Druggability gene group&quot;, x=&quot;Gene length (kb)&quot;, caption=&quot;Splicing perturbed to control P&lt;2.2x10^-16&quot;)
P.geneLength.FromPostTxnSet</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-32-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>I think the last thing I want to show are some examples of dose
response curves…</p>
<p>Let’s start by doing all the genes that intersect the OMIM short
list.</p>
<pre class="r"><code>OMIM.risdiplam.genes</code></pre>
<pre><code># A tibble: 13 × 20
   gene      dose.nM logFC_chRNA logFC_polyA PValue_chRNA PValue_polyA FDR_chRNA
   &lt;chr&gt;       &lt;dbl&gt;       &lt;dbl&gt;       &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;     &lt;dbl&gt;
 1 ENSG0000…    3160      0.676        -1.29   0.00000339     9.04e- 9  0.000132
 2 ENSG0000…    3160      0.629        -1.96   0.00000902     9.49e-12  0.000271
 3 ENSG0000…    3160      0.493        -4.36   0.000522       1.42e-36  0.00593 
 4 ENSG0000…    3160      0.557        -1.10   0.000557       1.52e- 3  0.00624 
 5 ENSG0000…    3160      0.388        -1.10   0.000956       8.66e- 3  0.00944 
 6 ENSG0000…    3160      0.0248       -1.00   0.893          2.79e- 2  0.977   
 7 ENSG0000…    3160     NA            -1.84  NA              1.78e- 3 NA       
 8 ENSG0000…    3160     NA            -2.97  NA              1.38e-18 NA       
 9 ENSG0000…    3160     NA            -1.21  NA              2.07e- 5 NA       
10 ENSG0000…    3160     NA            -1.16  NA              5.78e- 3 NA       
11 ENSG0000…    3160     NA            -1.16  NA              5.78e- 3 NA       
12 ENSG0000…    3160     NA            -1.16  NA              5.78e- 3 NA       
13 ENSG0000…    3160     NA            -1.16  NA              5.78e- 3 NA       
# … with 13 more variables: FDR_polyA &lt;dbl&gt;, Category &lt;chr&gt;, ColorEffect &lt;chr&gt;,
#   MergedEffect &lt;chr&gt;, junc &lt;chr&gt;, DeltaPSI &lt;dbl&gt;, `MIM Number` &lt;dbl&gt;,
#   `Gene Symbols` &lt;chr&gt;, `Gene Name` &lt;chr&gt;, `Approved Gene Symbol` &lt;chr&gt;,
#   `Entrez Gene ID` &lt;dbl&gt;, Comments &lt;chr&gt;, Phenotypes &lt;chr&gt;</code></pre>
<pre class="r"><code>Expression.tidyForDose_ActualDataForModel &lt;- read_tsv(&quot;../code/SmallMolecule/FitModels/Data/polyA_genes.tsv.gz&quot;)

#Calculate factors to visually normalize chRNA and polyA to the same value at dose.nM==0
DiffFactors &lt;- Expression.tidyForDose_ActualDataForModel %&gt;%
  filter(dose.nM == 0) %&gt;%
  group_by(Geneid, LibraryType) %&gt;%
  summarise(MeanLog2CPM=mean(Log2CPM)) %&gt;%
  ungroup() %&gt;%
  pivot_wider(names_from=&quot;LibraryType&quot;, values_from=&quot;MeanLog2CPM&quot;) %&gt;%
  mutate(DiffFactor = polyA - chRNA)

Splicing.tidyForDose_ActualDataForModel &lt;- read_tsv(&quot;../code/SmallMolecule/CassetteExons/TidyPSI.tsv.gz&quot;) %&gt;%
  inner_join(
    P.boxplot.dat %&gt;%
      filter(dataset==&quot;polyA&quot;) %&gt;%
      dplyr::select(GAGTjunc=GAGTInt, gene, ColorEffect, MergedEffect, ExonStart_Cassette, ExonStop_Cassette) %&gt;%
      unite(Exon, ExonStart_Cassette,ExonStop_Cassette, sep=&quot;-&quot;)
    )



P.OMIM.splicing.Effections &lt;- Splicing.tidyForDose_ActualDataForModel %&gt;%
  filter(gene %in% OMIM.risdiplam.genes$gene) %&gt;%
  left_join(ensembl_to_symbols, by=c(&quot;gene&quot;=&quot;ensembl_gene_id&quot;)) %&gt;%
  mutate(chrom = str_extract(GAGTjunc, &quot;^chr.+?:&quot;)) %&gt;%
  mutate(facet = str_glue(&quot;{hgnc_symbol}\n{chrom}{Exon}&quot;)) %&gt;%
  # filter(libType==&quot;polyA&quot;) %&gt;%
  # mutate(dose.nM = dose.nM + 0.1) %&gt;%
  ggplot(aes(x=dose.nM, y=PSI, color=libType)) +
  geom_point() +
  scale_color_manual(values=c(&quot;chRNA&quot;=&quot;#969696&quot;, &quot;polyA&quot;=&quot;#252525&quot;)) +
  geom_smooth(
    data = . %&gt;%
      filter(libType==&quot;polyA&quot;),
    method = drm, method.args = list(fct = L.4(names=c(&quot;Steepness&quot;, &quot;LowerLimit&quot;, &quot;UpperLimit&quot;, &quot;ED50&quot;))), se = FALSE) +
  scale_x_continuous(trans=&quot;log1p&quot;, limits=c(0, 10000), breaks=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 0) + 0.1, labels=rev(c(&quot;0&quot;, &quot;3.16&quot;,&quot;10&quot;,&quot;31.6&quot;,&quot;100&quot;, &quot;316&quot;, &quot;1000&quot;, &quot;3160&quot;, &quot;10000&quot;))) +  facet_wrap(~facet, scales=&quot;free_y&quot;) +
  Rotate_x_labels +
  labs(x=&quot;[risdiplam] (nanomolar)&quot;, y=&quot;Splicing of indicated NMD exon; PSI&quot;)
P.OMIM.splicing.Effections</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-33-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>P.OMIM.expression.Effections &lt;- Expression.tidyForDose_ActualDataForModel %&gt;%
  dplyr::rename(libType = &quot;LibraryType&quot;) %&gt;%
  filter(ensembl_gene_id %in% OMIM.risdiplam.genes$gene) %&gt;%
  left_join(ensembl_to_symbols) %&gt;%
  left_join(DiffFactors) %&gt;%
  mutate(Log2CPM = case_when(
    libType == &quot;polyA&quot; ~ Log2CPM,
    libType == &quot;chRNA&quot; ~ Log2CPM + DiffFactor
  )) %&gt;%
  # mutate(dose.nM = dose.nM + 0.1) %&gt;%
  ggplot(aes(x=dose.nM, y=Log2CPM, color=libType)) +
  geom_point() +
  scale_color_manual(values=c(&quot;chRNA&quot;=&quot;#969696&quot;, &quot;polyA&quot;=&quot;#252525&quot;)) +
  geom_smooth(
    data = . %&gt;%
      filter(libType==&quot;polyA&quot;),
    method = drm, method.args = list(fct = L.4(names=c(&quot;Steepness&quot;, &quot;LowerLimit&quot;, &quot;UpperLimit&quot;, &quot;ED50&quot;))), se = FALSE) +
  scale_x_continuous(trans=&quot;log1p&quot;, limits=c(0, 10000), breaks=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 0), labels=rev(c(&quot;0&quot;, &quot;3.16&quot;,&quot;10&quot;,&quot;31.6&quot;,&quot;100&quot;, &quot;316&quot;, &quot;1000&quot;, &quot;3160&quot;, &quot;10000&quot;))) +
  facet_wrap(~hgnc_symbol, scales=&quot;free_y&quot;) +
  Rotate_x_labels +
  # scale_x_break(c(1,2)) +
  labs(x=&quot;[risdiplam] (nanomolar)&quot;, y=expression(&quot;Expression; &quot;*log[&quot;2&quot;]*&quot;CPM&quot;))
P.OMIM.expression.Effections</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-33-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Expression.tidyForDose_ActualDataForModel %&gt;%
  dplyr::rename(libType = &quot;LibraryType&quot;) %&gt;%
  # filter(ensembl_gene_id %in% OMIM.risdiplam.genes$gene) %&gt;%
  left_join(ensembl_to_symbols) %&gt;%
  filter(hgnc_symbol %in% c(&quot;HTT&quot;, &quot;MYB&quot;, &quot;STAT1&quot;)) %&gt;%
  left_join(DiffFactors) %&gt;%
  mutate(Log2CPM = case_when(
    libType == &quot;polyA&quot; ~ Log2CPM,
    libType == &quot;chRNA&quot; ~ Log2CPM + DiffFactor
  )) %&gt;%
  # mutate(dose.nM = dose.nM + 0.1) %&gt;%
  ggplot(aes(x=dose.nM, y=Log2CPM, color=libType)) +
  geom_point() +
  scale_color_manual(values=c(&quot;chRNA&quot;=&quot;#969696&quot;, &quot;polyA&quot;=&quot;#252525&quot;)) +
  geom_smooth(
    data = . %&gt;%
      filter(libType==&quot;polyA&quot;),
    method = drm, method.args = list(fct = L.4(names=c(&quot;Steepness&quot;, &quot;LowerLimit&quot;, &quot;UpperLimit&quot;, &quot;ED50&quot;))), se = FALSE) +
  scale_x_continuous(trans=&quot;log1p&quot;, limits=c(0, 10000), breaks=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 0), labels=rev(c(&quot;0&quot;, &quot;3.16&quot;,&quot;10&quot;,&quot;31.6&quot;,&quot;100&quot;, &quot;316&quot;, &quot;1000&quot;, &quot;3160&quot;, &quot;10000&quot;))) +
  facet_wrap(~hgnc_symbol, scales=&quot;free_y&quot;) +
  Rotate_x_labels +
  # scale_x_break(c(1,2)) +
  labs(x=&quot;[risdiplam] (nanomolar)&quot;, y=expression(&quot;Expression; &quot;*log[&quot;2&quot;]*&quot;CPM&quot;))</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-33-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>One more version with just log2FC on y axis so we don’t have to
artifiially/misleadingly set different CPM scale for chRNA.</p>
<pre class="r"><code>Expression.tidyForDose_ActualDataForModel.Relative &lt;- Expression.tidyForDose_ActualDataForModel %&gt;%
  left_join(
    Expression.tidyForDose_ActualDataForModel %&gt;%
      filter(dose.nM == 0) %&gt;%
      group_by(LibraryType, Geneid) %&gt;%
      summarise(MeanNormalCPM = mean(Log2CPM, na.rm=T))
  ) %&gt;%
  mutate()

P.OMIM.expression.Effections &lt;- Expression.tidyForDose_ActualDataForModel.Relative %&gt;%
  dplyr::rename(libType = &quot;LibraryType&quot;) %&gt;%
  filter(ensembl_gene_id %in% OMIM.risdiplam.genes$gene) %&gt;%
  left_join(ensembl_to_symbols) %&gt;%
  # mutate(dose.nM = dose.nM + 0.1) %&gt;%
  ggplot(aes(x=dose.nM, y=Log2CPM-MeanNormalCPM, color=libType)) +
  geom_point() +
  scale_color_manual(values=c(&quot;chRNA&quot;=&quot;#969696&quot;, &quot;polyA&quot;=&quot;#252525&quot;)) +
  geom_smooth(
    data = . %&gt;%
      filter(libType==&quot;polyA&quot;),
    method = drm, method.args = list(fct = L.4(names=c(&quot;Steepness&quot;, &quot;LowerLimit&quot;, &quot;UpperLimit&quot;, &quot;ED50&quot;))), se = FALSE) +
  scale_x_continuous(trans=&quot;log1p&quot;, limits=c(0, 10000), breaks=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 0), labels=rev(c(&quot;0&quot;, &quot;3.16&quot;,&quot;10&quot;,&quot;31.6&quot;,&quot;100&quot;, &quot;316&quot;, &quot;1000&quot;, &quot;3160&quot;, &quot;10000&quot;))) +  facet_wrap(~hgnc_symbol, scales=&quot;free_y&quot;) +
  Rotate_x_labels +
  labs(x=&quot;[risdiplam] (nanomolar)&quot;, y=expression(&quot;Expression &quot;*log[&quot;2&quot;]*&quot;FC&quot;))
P.OMIM.expression.Effections</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Get just the stat1 data and save</p>
<pre class="r"><code>P.stat1.expression &lt;- Expression.tidyForDose_ActualDataForModel.Relative %&gt;%
  dplyr::rename(libType = &quot;LibraryType&quot;) %&gt;%
  left_join(ensembl_to_symbols) %&gt;%
  filter(hgnc_symbol == &quot;STAT1&quot;) %&gt;%
  # mutate(dose.nM = dose.nM + 0.1) %&gt;%
  ggplot(aes(x=dose.nM, y=Log2CPM-MeanNormalCPM, color=libType)) +
  geom_point() +
  scale_color_manual(values=c(&quot;chRNA&quot;=&quot;#969696&quot;, &quot;polyA&quot;=&quot;#252525&quot;)) +
  geom_smooth(
    data = . %&gt;%
      filter(libType==&quot;polyA&quot;),
    method = drm, method.args = list(fct = L.4(names=c(&quot;Steepness&quot;, &quot;LowerLimit&quot;, &quot;UpperLimit&quot;, &quot;ED50&quot;))), se = FALSE) +
  scale_x_continuous(trans=&quot;log1p&quot;, limits=c(0, 10000), breaks=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 0), labels=rev(c(&quot;0&quot;, &quot;3.16&quot;,&quot;10&quot;,&quot;31.6&quot;,&quot;100&quot;, &quot;316&quot;, &quot;1000&quot;, &quot;3160&quot;, &quot;10000&quot;))) + 
  # Rotate_x_labels +
  scale_x_break(c(2,2)) +
  theme(axis.text.x.top = element_blank(),
        axis.ticks.x.top = element_blank(),
        axis.line.x.top = element_blank(),
        plot.background = element_blank()) +
  Rotate_x_labels +
  labs(x=&quot;[risdiplam] (nanomolar)&quot;, y=expression(&quot;STAT1 expression; &quot;*log[&quot;2&quot;]*&quot;FC&quot;))
P.stat1.expression</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-35-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>P.stat1.splicing &lt;- Splicing.tidyForDose_ActualDataForModel %&gt;%
  left_join(ensembl_to_symbols, by=c(&quot;gene&quot;=&quot;ensembl_gene_id&quot;)) %&gt;%
  filter(hgnc_symbol == &quot;STAT1&quot;) %&gt;%
  mutate(chrom = str_extract(GAGTjunc, &quot;^chr.+?:&quot;)) %&gt;%
  mutate(facet = str_glue(&quot;{hgnc_symbol}\n{chrom}{Exon}&quot;)) %&gt;%
  # filter(libType==&quot;polyA&quot;) %&gt;%
  # mutate(dose.nM = dose.nM + 0.1) %&gt;%
  ggplot(aes(x=dose.nM, y=PSI, color=libType)) +
  geom_point() +
  scale_color_manual(values=c(&quot;chRNA&quot;=&quot;#969696&quot;, &quot;polyA&quot;=&quot;#252525&quot;)) +
  geom_smooth(
    data = . %&gt;%
      filter(libType==&quot;polyA&quot;),
    method = drm, method.args = list(fct = L.4(names=c(&quot;Steepness&quot;, &quot;LowerLimit&quot;, &quot;UpperLimit&quot;, &quot;ED50&quot;))), se = FALSE) +
  labs(x=&quot;[risdiplam] (nanomolar)&quot;, y=&quot;Splicing of\nNMD-targeting exon; PSI&quot;) +
  scale_x_continuous(trans=&quot;log1p&quot;, limits=c(0, 10000), breaks=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 0), labels=rev(c(&quot;0&quot;, &quot;3.16&quot;,&quot;10&quot;,&quot;31.6&quot;,&quot;100&quot;, &quot;316&quot;, &quot;1000&quot;, &quot;3160&quot;, &quot;10000&quot;))) + 
  # Rotate_x_labels +
  scale_x_break(c(2,2)) +
  theme(axis.text.x.top = element_blank(),
        axis.ticks.x.top = element_blank(),
        axis.line.x.top = element_blank(),
        plot.background = element_blank()) +
  Rotate_x_labels
P.stat1.splicing</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-35-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Get just myb data and save</p>
<pre class="r"><code>P.myb.expression &lt;- Expression.tidyForDose_ActualDataForModel.Relative %&gt;%
  dplyr::rename(libType = &quot;LibraryType&quot;) %&gt;%
  left_join(ensembl_to_symbols) %&gt;%
  filter(hgnc_symbol == &quot;MYB&quot;) %&gt;%
  # mutate(dose.nM = dose.nM + 0.1) %&gt;%
  ggplot(aes(x=dose.nM, y=Log2CPM-MeanNormalCPM, color=libType)) +
  geom_point() +
  scale_color_manual(values=c(&quot;chRNA&quot;=&quot;#969696&quot;, &quot;polyA&quot;=&quot;#252525&quot;), labels=c(&quot;chRNA&quot;=&quot;naRNA&quot;, &quot;polyA&quot;=&quot;steady-state&quot;)) +
  geom_smooth(
    data = . %&gt;%
      filter(libType==&quot;polyA&quot;),
    method = drm, method.args = list(fct = L.4(names=c(&quot;Steepness&quot;, &quot;LowerLimit&quot;, &quot;UpperLimit&quot;, &quot;ED50&quot;))), se = FALSE) +
  scale_x_continuous(trans=&quot;log1p&quot;, limits=c(0, 10000), breaks=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 0), labels=rev(c(&quot;0&quot;, &quot;3.16&quot;,&quot;10&quot;,&quot;31.6&quot;,&quot;100&quot;, &quot;316&quot;, &quot;1000&quot;, &quot;3160&quot;, &quot;10000&quot;))) + 
  # Rotate_x_labels +
  # scale_x_break(c(2,2)) +
  theme(axis.text.x.top = element_blank(),
        axis.ticks.x.top = element_blank(),
        axis.line.x.top = element_blank(),
        plot.background = element_blank()) +
  Rotate_x_labels +
  labs(x=&quot;[risdiplam] (nanomolar)&quot;, y=expression(&quot;MYB expression; &quot;*log[&quot;2&quot;]*&quot;FC&quot;))
P.myb.expression</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-36-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>P.myb.splicing &lt;- Splicing.tidyForDose_ActualDataForModel %&gt;%
  left_join(ensembl_to_symbols, by=c(&quot;gene&quot;=&quot;ensembl_gene_id&quot;)) %&gt;%
  filter(gene == &quot;ENSG00000142599&quot;) %&gt;%
  # filter(hgnc_symbol == &quot;MYB&quot;) %&gt;%
  mutate(chrom = str_extract(GAGTjunc, &quot;^chr.+?:&quot;)) %&gt;%
  mutate(facet = str_glue(&quot;{hgnc_symbol}\n{chrom}{Exon}&quot;)) %&gt;%
  # filter(libType==&quot;polyA&quot;) %&gt;%
  # mutate(dose.nM = dose.nM + 0.1) %&gt;%
  ggplot(aes(x=dose.nM, y=PSI, color=libType)) +
  geom_point() +
  scale_color_manual(values=c(&quot;chRNA&quot;=&quot;#969696&quot;, &quot;polyA&quot;=&quot;#252525&quot;), labels=c(&quot;chRNA&quot;=&quot;naRNA&quot;, &quot;polyA&quot;=&quot;steady-state&quot;)) +
  geom_smooth(
    data = . %&gt;%
      filter(libType==&quot;polyA&quot;),
    method = drm, method.args = list(fct = L.4(names=c(&quot;Steepness&quot;, &quot;LowerLimit&quot;, &quot;UpperLimit&quot;, &quot;ED50&quot;))), se = FALSE) +
  labs(x=&quot;[risdiplam] (nanomolar)&quot;, y=&quot;Splicing of\nNMD-targeting exon; PSI&quot;) +
  scale_x_continuous(trans=&quot;log1p&quot;, limits=c(0, 10000), breaks=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 0), labels=rev(c(&quot;0&quot;, &quot;3.16&quot;,&quot;10&quot;,&quot;31.6&quot;,&quot;100&quot;, &quot;316&quot;, &quot;1000&quot;, &quot;3160&quot;, &quot;10000&quot;))) + 
  # Rotate_x_labels +
  # scale_x_break(c(2,2)) +
  theme(axis.text.x.top = element_blank(),
        axis.ticks.x.top = element_blank(),
        axis.line.x.top = element_blank(),
        plot.background = element_blank()) +
  Rotate_x_labels
P.myb.splicing</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-36-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>P.myb.splicing &lt;- ggplotify::as.ggplot(print(P.myb.splicing))</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-36-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>P.myb.expression &lt;- ggplotify::as.ggplot(print(P.myb.expression))</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-36-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>Check SRSF10 for Jon</p>
<pre class="r"><code>SpearmanCorr %&gt;%
  distinct(Intron, .keep_all=T) %&gt;%
  filter(substr(DonorSeq, 3,4)==&quot;GA&quot;) %&gt;%
  filter(q&lt;0.1) </code></pre>
<pre><code># A tibble: 641 × 15
   Intron          Leafcutter.ID DonorSeq DonorScore treatment dose.nM Cell.type
   &lt;chr&gt;           &lt;chr&gt;         &lt;chr&gt;         &lt;dbl&gt; &lt;chr&gt;       &lt;dbl&gt; &lt;chr&gt;    
 1 chr1:2189781:2… chr1:2189781… CAGAGTA…       4.53 DMSO            0 LCL      
 2 chr1:8365974:8… chr1:8365974… AAGAGTA…       5.52 DMSO            0 LCL      
 3 chr1:18921121:… chr1:1892112… AGGAGTA…       6.27 DMSO            0 LCL      
 4 chr1:23863271:… chr1:2386327… AGGAGTA…       2.87 DMSO            0 LCL      
 5 chr1:23875662:… chr1:2387566… AAGAGTA…       4.83 DMSO            0 LCL      
 6 chr1:28151059:… chr1:2815105… AGGAGTA…       5.30 DMSO            0 LCL      
 7 chr1:28172527:… chr1:2817252… AGGAGTA…       5.30 DMSO            0 LCL      
 8 chr1:37493814:… chr1:3749381… GAGAGTA…       5.59 DMSO            0 LCL      
 9 chr1:41028726:… chr1:4102872… AAGAGTA…       6.42 DMSO            0 LCL      
10 chr1:44313228:… chr1:4431322… TAGAGTA…       4.50 DMSO            0 LCL      
# … with 631 more rows, and 8 more variables: LibraryType &lt;chr&gt;, rep &lt;dbl&gt;,
#   PSI &lt;dbl&gt;, Counts &lt;dbl&gt;, n &lt;dbl&gt;, spearman &lt;dbl&gt;, spearman.p &lt;dbl&gt;, q &lt;dbl&gt;</code></pre>
<pre class="r"><code>SpearmanCorr %&gt;%
  distinct(Intron, .keep_all=T) %&gt;%
  filter(substr(DonorSeq, 3,4)==&quot;GA&quot;) %&gt;%
  separate(Intron, into=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;strand&quot;), sep=&quot;:&quot;) %&gt;%
  dplyr::select(chrom, start, stop, strand, Leafcutter.ID, spearman, strand) %&gt;%
  arrange(chrom, start, stop) %&gt;%
  write_tsv(&quot;../code/scratch/AllGAGTInts.bed&quot;, col_names = F)

Expression.tidyForDose_ActualDataForModel %&gt;%
  filter(ensembl_gene_id == &quot;ENSG00000188529&quot;) %&gt;%
  dplyr::rename(libType=LibraryType) %&gt;%
  ggplot(aes(x=dose.nM, y=Log2CPM, color=libType)) +
  geom_point() +
  scale_color_manual(values=c(&quot;chRNA&quot;=&quot;#969696&quot;, &quot;polyA&quot;=&quot;#252525&quot;), labels=c(&quot;chRNA&quot;=&quot;naRNA&quot;, &quot;polyA&quot;=&quot;steady-state&quot;)) +
  geom_smooth(
    data = . %&gt;%
      filter(libType==&quot;polyA&quot;),
    method = drm, method.args = list(fct = L.4(names=c(&quot;Steepness&quot;, &quot;LowerLimit&quot;, &quot;UpperLimit&quot;, &quot;ED50&quot;))), se = FALSE) +
  labs(x=&quot;[risdiplam] (nanomolar)&quot;, y=&quot;Splicing of\nNMD-targeting exon; PSI&quot;) +
  scale_x_continuous(trans=&quot;log1p&quot;, limits=c(0, 10000), breaks=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 0), labels=rev(c(&quot;0&quot;, &quot;3.16&quot;,&quot;10&quot;,&quot;31.6&quot;,&quot;100&quot;, &quot;316&quot;, &quot;1000&quot;, &quot;3160&quot;, &quot;10000&quot;))) + 
  # Rotate_x_labels +
  # scale_x_break(c(2,2)) +
  theme(axis.text.x.top = element_blank(),
        axis.ticks.x.top = element_blank(),
        axis.line.x.top = element_blank(),
        plot.background = element_blank()) +
  Rotate_x_labels</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-37-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig4_Bar.pdf&quot;, P.bar, height=4, width=9)
ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig4_BarSimpler.pdf&quot;, P.bar.simpler, height=3, width=8)

ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig4_BarSimpler.Fixed.pdf&quot;, P.bar.simpler.fixed, height=1.5, width=4)


ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig4_BoxLarge.pdf&quot;, P.box.small, height=3, width=4)

ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig4_BoxSmall.pdf&quot;, P.box, height=3, width=4)
ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig4_chRNA_polyA_DE_scatter.pdf&quot;,P.scatter, height=6, width=8)

ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig4_MeanDeltaPSI.pdf&quot;,P.MeanDeltaPSI, height=3, width=6)

ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig4_DosePSI_SpearmanDist.pdf&quot;,P.spearman.dist, height=6, width=6)

ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig4_PSI_v_DE_ByPrediction.pdf&quot;,P.scatter.byPrediction, height=5, width=8)
ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig4_PSI_v_DE_ByPrediction_chRNA.pdf&quot;,P.scatter.byPrediction.chRNA, height=5, width=8)

ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig4_GeneSetEnrichment.pdf&quot;,P.genesetEnrichment, height=5, width=10)

ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig4_GeneSetEnrichment_QualitativeBarplot.pdf&quot;,P.enrichmentByBarplot, height=4, width=6)

ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig4_GeneLength.pdf&quot;,P.geneLength, height=3, width=7)

ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig4_GeneLength_NumExons.pdf&quot;,P.NumExons, height=3, width=7)
ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig4_GeneLength_LengthHostIntrons.pdf&quot;,P.IntLen, height=3, width=5)



ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig4_GeneLength_FromPostTxnSet.pdf&quot;,P.geneLength.FromPostTxnSet, height=3, width=7)

ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig4_stat1_SplicingDoseResponse.pdf&quot;,P.stat1.splicing, height=3.5, width=4.5)
ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig4_stat1_ExpressionDoseResponse.pdf&quot;,P.stat1.expression, height=3.5, width=4.5)

ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig4_myb_SplicingDoseResponse.pdf&quot;,P.myb.splicing, height=3.5, width=4.5)
ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig4_myb_ExpressionDoseResponse.pdf&quot;,P.myb.expression, height=3.5, width=4.5)

ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig4_OMIM_SplicingDoseResponse.pdf&quot;,P.OMIM.splicing.Effections, height=9, width=15)
ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig4_OMIM_ExpresionDoseResponse.pdf&quot;,P.OMIM.expression.Effections, height=9, width=15)

ggsave(&quot;/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig4_MA.pdf&quot;,P.MA, height=3.5, width=6)</code></pre>
</div>
</div>
<div id="update-ignore-gencode-annotations" class="section level2">
<h2>Update; ignore Gencode annotations</h2>
<p>I kind of think it will be simpler presentation (and possibly cleaner
results) if i just use my own interpretations of cassette exons instead
of using Gencode. Let’s try…</p>
<div id="remake-bar-plot" class="section level3">
<h3>remake bar plot</h3>
<pre class="r"><code>P.bar.dat &lt;- Translated.GAGT_CassetteExons %&gt;%
   mutate(IsAnnotated = if_else(str_detect(SuperAnnotation,&quot;Annotated&quot;), &quot;Annotated by Gencode&quot;, &quot;Unannotated&quot;, missing=&quot;Unannotated&quot;)) %&gt;%
  mutate(MergedEffect = case_when(
    IsAnnotated == &quot;Annotated by Gencode&quot; &amp; SemiSupergroupAnnotations==&quot;basic tag&quot; ~ &quot;Productive&quot;,
    IsAnnotated == &quot;Annotated by Gencode&quot; &amp; !SemiSupergroupAnnotations==&quot;basic tag&quot; ~ &quot;NMD&quot;,
    TRUE ~ Effect
  )) %&gt;%
  #HERE IS WHERE I RECODE TO DISREGARD GENCODE ANNOTATIONS
  mutate(OldMergedEffect = MergedEffect) %&gt;%
  mutate(MergedEffect = Effect) %&gt;%
  mutate(ColorEffect = if_else(MergedEffect %in% c(&quot;Productive&quot;, &quot;Preserves frame&quot;, &quot;Cassette in UTR&quot;), &quot;Productive&quot;, &quot;NMD&quot;) )
  

P.bar &lt;- P.bar.dat %&gt;%
  count(MergedEffect, ColorEffect) %&gt;%
  mutate(MergedEffect=factor(MergedEffect, levels=c(&quot;Cassette in UTR&quot;,&quot;Preserves frame&quot;, &quot;In-frame PTC&quot;, &quot;Frame shifting&quot;)))  %&gt;%
  mutate(y=&quot;y&quot;) %&gt;%
  # mutate(IsAnnotated = recode(IsAnnotated, &quot;Annotated by Gencode&quot;=&quot;Annotated&quot;)) %&gt;%
  ggplot(aes(y=y, x=n, fill=MergedEffect, color=ColorEffect)) +
    geom_col(position=&quot;stack&quot;, size=3) +
   geom_text( aes(label=n), color=&#39;black&#39;, position=position_stack(vjust=0.5)) +
    scale_color_manual(values=c(
      &quot;Productive&quot; = &quot;#1f78b4&quot;,
      &quot;NMD&quot; = &quot;#e31a1c&quot;
    )) +
   scale_fill_manual(values=c(
     # &quot;Productive&quot;=&quot;#1f78b4&quot;,
     # &quot;NMD&quot;=&quot;#e31a1c&quot;,
     &quot;Frame shifting&quot;=&quot;#fb6a4a&quot;,
     &quot;Cassette in UTR&quot;=&quot;#deebf7&quot;,
     &quot;In-frame PTC&quot;=&quot;#fcbba1&quot;,
      &quot;Preserves frame&quot;=&quot;#a6cee3&quot;
   ),
   labels=c(
     #  &quot;Productive&quot;=&quot;In annotated productive transcript&quot;,
     # &quot;NMD&quot;=&quot;In annotated NMD transcript&quot;,
     &quot;Frame shifting&quot;=&quot;Frame shifting&quot;,
     &quot;Cassette in UTR&quot;=&quot;Cassette in UTR&quot;,
     &quot;In-frame PTC&quot;=&quot;In-frame PTC&quot;,
      &quot;Preserves frame&quot;=&quot;Preserves frame&quot;
   )) +
  scale_x_continuous(expand=c(0,0)) +
    Rotate_x_labels +
    guides(fill = guide_legend(order = 2),
         color = guide_legend(order = 1)) +
  theme(legend.box=&quot;vertical&quot;, legend.position = &quot;bottom&quot;) +
    theme(
    strip.background = element_blank(),
    strip.text.x = element_blank()
  ) +
  labs(title=&quot;Predicted transcript effects of induced cassete exons&quot;, x=&quot;Number risdiplam-induced cassette exons&quot;, fill=&quot;Reason&quot;, y=NULL, color=&quot;Transcript effect&quot;)

P.bar</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-39-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="redo-boxplot" class="section level2">
<h2>redo boxplot</h2>
<pre class="r"><code>P.boxplot.dat &lt;- P.bar.dat %&gt;%
  dplyr::select(-symbol) %&gt;%
  mutate(gene = str_replace(gene, &quot;(^.+?)\\..+?$&quot;, &quot;\\1&quot;)) %&gt;%
  left_join(
    volcano.dat %&gt;%
      filter(dose.nM == 3160) %&gt;%
      distinct(gene, dose.nM, dataset, .keep_all=T) %&gt;%
      dplyr::select(gene, dose.nM, dataset, logFC) %&gt;%
      pivot_wider(names_from=&quot;dataset&quot;, values_from=&quot;logFC&quot;),
    by=&quot;gene&quot;
  ) %&gt;%
  left_join(ensembl_to_symbols, by=c(&quot;gene&quot;=&quot;ensembl_gene_id&quot;)) %&gt;%
  # filter(hgnc_symbol %in% c(&quot;HTT&quot;, &quot;SMN2&quot;, &quot;GALC&quot;, &quot;STAT1&quot;, &quot;KCNT2&quot;, &quot;FHOD3&quot;, &quot;FOXM1))
  dplyr::select(GAGTInt, ExonStart_Cassette, ExonStop_Cassette, gene, hgnc_symbol, chRNA, polyA, ColorEffect, MergedEffect, IsAnnotated, OldMergedEffect) %&gt;%
  gather(&quot;dataset&quot;, &quot;log2FC&quot;, chRNA, polyA)

P.box.small &lt;- P.boxplot.dat %&gt;%
  mutate(ColorEffect = factor(ColorEffect, levels=c(&quot;Productive&quot;, &quot;NMD&quot;))) %&gt;%
  ggplot(aes(y=log2FC, x=ColorEffect, color=ColorEffect)) +
  geom_hline(yintercept=0) +
  geom_boxplot(outlier.shape=NA, alpha=1, fill=&quot;white&quot;) +
  geom_segment(x=&quot;Productive&quot;, xend=&quot;NMD&quot;, y=2.5, yend=2.5, color=&#39;black&#39;) +
  geom_point(position=position_jitterdodge(jitter.width=0.5), alpha=0.3) +
  geom_text(data = . %&gt;%
              group_by(dataset) %&gt;%
              do(w = wilcox.test(log2FC~ColorEffect, data=., paired=FALSE)) %&gt;%
              summarise(dataset, Wilcox = format.pval(w$p.value, digits=3)) %&gt;%
              ungroup() %&gt;%
              mutate(label=str_glue(&quot;P={Wilcox}&quot;)),
            aes(label=label, group=NULL),
            x=1.5, y=2.5, color=&#39;black&#39;, vjust=-1) +
  scale_color_manual(values=c( &quot;#1f78b4&quot;,&quot;#e31a1c&quot;)) +
  facet_wrap(~dataset) +
  labs(y=expression(&quot;Host gene &quot;*log[&quot;2&quot;]*&quot;FC&quot;), x=&quot;Predicted transcript effect&quot;) +
  Rotate_x_labels +
  theme(legend.position=&quot;none&quot;)
P.box.small</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-40-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="redo-psi-vs-logfc-scatters" class="section level3">
<h3>redo PSI vs logFC scatters</h3>
<pre class="r"><code>P.scatter.byPrediction.dat &lt;- Scatter.dat %&gt;%
  filter(dose.nM==3160) %&gt;%
  inner_join(
    P.boxplot.dat %&gt;%
      filter(dataset==&quot;polyA&quot;) %&gt;%
      dplyr::select(gene, ColorEffect, MergedEffect, OldMergedEffect),
    by=&quot;gene&quot;
    )  %&gt;%
  inner_join(
    PolyA.SplicingEffects.Annotationed %&gt;%
    filter(param==&quot;Pred_3160&quot;) %&gt;%
    dplyr::select(junc, gene, Estimate) %&gt;%
    mutate(gene = str_replace(gene, &quot;(^.+?)\\..+?$&quot;, &quot;\\1&quot;)),
    by=&quot;gene&quot;
  )

MergedEffectLabels &lt;- c(
      &quot;Productive&quot;=&quot;In annotated productive transcript&quot;,
     &quot;NMD&quot;=&quot;In annotated NMD transcript&quot;,
     &quot;Frame shifting&quot;=&quot;Frame shifting&quot;,
     &quot;Cassette in UTR&quot;=&quot;Cassette in UTR&quot;,
     &quot;In-frame PTC&quot;=&quot;In-frame PTC&quot;,
      &quot;Preserves frame&quot;=&quot;Preserves frame&quot;,
     &quot;All NMD&quot;=&quot;All predicted NMD exons&quot;,
     &quot;All productive&quot;=&quot;All predicted productive exons&quot;
                    )

P.scatter.byPrediction &lt;- bind_rows(
  P.scatter.byPrediction.dat %&gt;%
    mutate(color=OldMergedEffect),
  P.scatter.byPrediction.dat %&gt;%
    mutate(color=OldMergedEffect) %&gt;%
    mutate(MergedEffect=recode(ColorEffect, !!!c(&quot;NMD&quot;=&quot;All NMD&quot;, &quot;Productive&quot;=&quot;All productive&quot;)))
  ) %&gt;% 
  mutate(MergedEffect = factor(MergedEffect, levels=c(&quot;Preserves frame&quot;, &quot;Cassette in UTR&quot;, &quot;All productive&quot;, &quot;Frame shifting&quot;, &quot;In-frame PTC&quot;, &quot;All NMD&quot;))) %&gt;%
  distinct(gene,MergedEffect, .keep_all=T) %&gt;%
  ggplot(aes(x=Estimate, y=logFC_polyA, color=color)) +
  geom_hline(yintercept=0) +
  geom_point() +
   scale_color_manual(values=c(
     &quot;Productive&quot;=&quot;#1f78b4&quot;,
     &quot;NMD&quot;=&quot;#e31a1c&quot;,
     &quot;Frame shifting&quot;=&quot;#fb9a99&quot;,
     &quot;Cassette in UTR&quot;=&quot;#a6cee3&quot;,
     &quot;In-frame PTC&quot;=&quot;#fb9a99&quot;,
      &quot;Preserves frame&quot;=&quot;#a6cee3&quot;
   ),
   labels=MergedEffectLabels) +
  geom_text( data = . %&gt;%
    group_by(MergedEffect) %&gt;%
    summarise(cor=cor.test(Estimate,logFC_polyA)[[&quot;estimate&quot;]], pval=cor.test(Estimate,logFC_polyA, method=&#39;s&#39;)[[&quot;p.value&quot;]]) %&gt;%
    mutate(R = signif(cor, 3), P=format.pval(pval, 3)) %&gt;%
    mutate(label = str_glue(&quot;rho:{R}\nP:{P}&quot;)),
    aes(x=-Inf, y=-Inf, label=label),
  hjust=-.1, vjust=-0.1, color=&#39;black&#39;
  ) +
  # facet_wrap(~MergedEffect, labeller = as_labeller(MergedEffectLabels)) +
  facet_wrap(~MergedEffect, nrow=2, labeller = as_labeller(MergedEffectLabels, default=label_wrap_gen(20))) +
  theme(legend.position=&quot;none&quot;) +
  labs(y=expression(&quot;host gene &quot;*log[&quot;2&quot;]*&quot;FC, polyA RNA&quot;), x=expression(Delta*&quot;PSI&quot;), caption=&quot;[risdiplam]=3160nM, Effects measured in polyA RNA&quot;, title=&quot;Colored by annotations when available, faceted by my predictions&quot;) +
  Rotate_x_labels
P.scatter.byPrediction</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-41-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#First see how my annotations match up with Gencode and Yang&#39;s annotations
Translated.GAGT_CassetteExons %&gt;%
  mutate(IsAnnotated = if_else(str_detect(SuperAnnotation,&quot;Annotated&quot;), &quot;Annotated by Gencode&quot;, &quot;Annotated by Yang&quot;)) %&gt;%
  replace_na(list(SemiSupergroupAnnotations=&quot;Unannotated by either&quot;, IsAnnotated=&quot;Unannotated by either&quot;)) %&gt;%
  count(Effect, IsAnnotated, SemiSupergroupAnnotations) %&gt;%
  ggplot(aes(x=Effect, y=n, fill=SemiSupergroupAnnotations)) +
    geom_col(position=&quot;stack&quot;) +
    facet_wrap(~IsAnnotated, scales=&quot;free_y&quot;, nrow = 3) +
    Rotate_x_labels +
  labs(title=&quot;Comparing my annotations to Gencode/Yang&quot;, x=&quot;My annotation by comparing translations&quot;, fill=&quot;Gencode/Yang&#39;s annotation&quot;)</code></pre>
<p><img src="figure/MakeFinalFigs_Fig4.Rmd/unnamed-chunk-41-2.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] broom_0.8.0        ggnewscale_0.4.8   biomaRt_2.52.0     ggbreak_0.1.1     
 [5] drc_3.0-1          MASS_7.3-56        knitr_1.39         ggrepel_0.9.1     
 [9] qvalue_2.28.0      readxl_1.4.0       edgeR_3.38.4       limma_3.52.4      
[13] data.table_1.14.2  RColorBrewer_1.1-3 forcats_0.5.1      stringr_1.4.0     
[17] dplyr_1.0.9        purrr_0.3.4        readr_2.1.2        tidyr_1.2.0       
[21] tibble_3.1.7       ggplot2_3.3.6      tidyverse_1.3.1   

loaded via a namespace (and not attached):
  [1] backports_1.4.1        workflowr_1.7.0        BiocFileCache_2.4.0   
  [4] plyr_1.8.7             splines_4.2.0          GenomeInfoDb_1.32.2   
  [7] TH.data_1.1-1          digest_0.6.29          yulab.utils_0.0.6     
 [10] htmltools_0.5.2        fansi_1.0.3            magrittr_2.0.3        
 [13] memoise_2.0.1          tzdb_0.3.0             Biostrings_2.64.0     
 [16] modelr_0.1.8           R.utils_2.11.0         vroom_1.5.7           
 [19] sandwich_3.0-1         prettyunits_1.1.1      colorspace_2.0-3      
 [22] blob_1.2.3             rvest_1.0.2            rappdirs_0.3.3        
 [25] haven_2.5.0            xfun_0.30              crayon_1.5.1          
 [28] RCurl_1.98-1.7         jsonlite_1.8.0         survival_3.3-1        
 [31] zoo_1.8-10             glue_1.6.2             gtable_0.3.0          
 [34] zlibbioc_1.42.0        XVector_0.36.0         car_3.1-1             
 [37] BiocGenerics_0.42.0    abind_1.4-5            scales_1.2.0          
 [40] mvtnorm_1.1-3          DBI_1.1.2              Rcpp_1.0.8.3          
 [43] plotrix_3.8-2          progress_1.2.2         gridGraphics_0.5-1    
 [46] bit_4.0.4              stats4_4.2.0           httr_1.4.3            
 [49] ellipsis_0.3.2         R.methodsS3_1.8.1      pkgconfig_2.0.3       
 [52] XML_3.99-0.14          farver_2.1.0           sass_0.4.1            
 [55] dbplyr_2.1.1           locfit_1.5-9.7         utf8_1.2.2            
 [58] ggplotify_0.1.0        tidyselect_1.1.2       labeling_0.4.2        
 [61] rlang_1.0.2            reshape2_1.4.4         later_1.3.0           
 [64] AnnotationDbi_1.58.0   munsell_0.5.0          cellranger_1.1.0      
 [67] tools_4.2.0            cachem_1.0.6           cli_3.3.0             
 [70] generics_0.1.2         RSQLite_2.2.14         evaluate_0.15         
 [73] fastmap_1.1.0          yaml_2.3.5             bit64_4.0.5           
 [76] fs_1.5.2               KEGGREST_1.36.2        nlme_3.1-157          
 [79] whisker_0.4            R.oo_1.24.0            aplot_0.1.10          
 [82] xml2_1.3.3             compiler_4.2.0         rstudioapi_0.13       
 [85] filelock_1.0.2         curl_4.3.2             png_0.1-7             
 [88] reprex_2.0.1           bslib_0.3.1            stringi_1.7.6         
 [91] highr_0.9              lattice_0.20-45        Matrix_1.5-3          
 [94] vctrs_0.4.1            pillar_1.7.0           lifecycle_1.0.1       
 [97] jquerylib_0.1.4        bitops_1.0-7           httpuv_1.6.5          
[100] patchwork_1.1.1        R6_2.5.1               promises_1.2.0.1      
[103] IRanges_2.30.0         codetools_0.2-18       gtools_3.9.2          
[106] assertthat_0.2.1       rprojroot_2.0.3        withr_2.5.0           
[109] multcomp_1.4-19        S4Vectors_0.34.0       GenomeInfoDbData_1.2.8
[112] mgcv_1.8-40            parallel_4.2.0         hms_1.1.1             
[115] grid_4.2.0             ggfun_0.0.9            rmarkdown_2.14        
[118] carData_3.0-5          git2r_0.30.1           Biobase_2.56.0        
[121] lubridate_1.8.0       </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
