---
title: "Make small molecule config"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = F, message = F)
```

## Intro

Making snakemake config file for samples with risdiplam treatment.

```{r}
library(tidyverse)
library(data.table)
```

```{r}
sample.list <- read_tsv("/project2/yangili1/bjf79/20211209_JingxinRNAseq/code/bigwigs/BigwigList.tsv",
                        col_names = c("SampleName", "bigwig", "group", "strand")) %>%
  filter(strand==".") %>%
  dplyr::select(-strand) %>%
  mutate(old.sample.name = str_replace(bigwig, "/project2/yangili1/bjf79/20211209_JingxinRNAseq/code/bigwigs/unstranded/(.+?).bw", "\\1")) %>%
  separate(SampleName, into=c("treatment", "dose.nM", "cell.type", "libType", "rep"), convert=T, remove=F, sep="_") %>%
  left_join(
    read_tsv("/project2/yangili1/bjf79/20211209_JingxinRNAseq/code/bigwigs/BigwigList.groups.tsv", col_names = c("group", "color", "bed", "supergroup")),
    by="group"
  )


FastqFiles <-
  bind_rows(
    read_tsv("/project2/yangili1/bjf79/20211209_JingxinRNAseq/code/config/samples.titrationseries.tsv"),
    read_tsv("/project2/yangili1/bjf79/20211209_JingxinRNAseq/code/config/samples.chRNAPilotSequencing.tsv")) %>%
  dplyr::select(old.sample.name=sample, R1, R2)

sample.list %>%
  inner_join(FastqFiles) %>%
  filter(
    treatment == "Risdiplam" | (treatment == "DMSO" & cell.type == "LCL")
    ) %>%
  dplyr::select(-bigwig, -bed) %>%
  write_tsv("../code/config/SmallMoleculeRNASeq.Samples.tsv")
```

#Check size distr of exon

```{r}
exons <- fread("../code/ReferenceGenome/Annotations/GTFTools_BasicAnnotations/gencode.v34.chromasomal.exons.bed")

exons %>%
  add_count(V5) %>%
  filter(n > 1) %>%
  distinct(V1, V2, V3) %>%
  mutate(len = V3-V2) %>%
  count(len, name="Freq") %>%
  ggplot(aes(x=len, y=Freq)) +
  geom_col() +
  coord_cartesian(xlim=c(180, 220))

exons %>%
  add_count(V5) %>%
  filter(n > 1) %>%
  distinct(V1, V2, V3) %>%
  mutate(len = (V3-V2)%%3) %>%
  count(len, name="Freq") %>%
  ggplot(aes(x=len, y=Freq)) +
  geom_col()
```

One useful way to break up 'unannotated' introns are into those that are new splice site combinations that preserve frame (using basic annotation exons) versus those that don't (and those that can't easily be determined from short read data).

```{r}
UTRs <- fread("../code/ReferenceGenome/Annotations/GTFTools_BasicAnnotations/gencode.v34.chromasomal.utr.bed", col.names = c("chrom", "start", "stop", ))
```

