---
title: "Check DeltaPWM"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = F, message = F)
```

## Intro


Here I want to investigate the effect of splice site SNPs on expression...

Carlos already some of the brute work - and in fact did a similar analysis himself... He previously quantified 5'ss usage QTLs, and based on some intermediate files he made, I further processed those files to calculate 5'ss motif scores for ref and alt allele (something Carlos has also done, and confirmed a strong correlation b/n motif score change and splicing change). Here I want to check that I correctly calculated SpliceSiteScore changes (based on simple Position weight matrix), and that these splice site score changes correlate with splicing changes... eventually i will check the effect of the splice site SNPs on expression...

```{r}
library(tidyverse)

dat.scratch <- read_tsv("../code/scratch/SpliceSiteEffects.txt.gz")

dat.scratch %>%
  distinct(phe_id, New, .keep_all=T) %>%
  # filter(nom_pval < 0.05) %>%
  ggplot(aes(x=DeltaPWM, y=slope, color=phe_strd)) +
  geom_point() +
  facet_wrap(~New) +
  theme_bw() +
  labs(x="Delta 5'ss score (PWM)", y="5'ss usage, Standardized beta")
```

Ok that looks great... Note I feel confident I calculated the 5'ss motif scores properly, and for both + and - strands, since there is clear correlation as expected... Now let's check the effects on expression...


```{r}
dat.tidy <- dat.scratch %>%
  pivot_longer(polyA_eQTL_P:chRNA_eQTL_beta, names_pattern="^(.+)_(.+)$", names_to=c("Dataset", "stat")) %>%
  pivot_wider(names_from="stat", values_from="value")

dat.tidy %>%
  filter(P < 0.05) %>%
  ggplot(aes(x=DeltaPWM, y=beta)) +
  geom_point(alpha=0.1) +
  facet_wrap(New ~ Dataset) +
  theme_bw() +
  labs(x="Delta 5'ss score (PWM)", y="expression, Standardized beta", caption="chRNA_5ss and polyA_5ss refer to 5'ss tested in each dataset", title="Effect of splice site mutations on host gene")
```

Perhaps first we should look at QQ plots to start...

```{r}
test.SNPs <- paste0("../code/QTLs/QTLTools/", c("chRNA.Expression.Splicing", "Expression.Splicing.Subset_YRI"), "/NominalPassForColoc.RandomSamplePvals.txt.gz") %>%
  setNames(c("chRNA_eQTL", "polyA_eQTL")) %>%
  lapply(read_tsv, col_names=c("P")) %>%
  bind_rows(.id="Dataset") %>%
  mutate(SnpSet = "TestSNPs") %>%
  group_by(Dataset) %>%
  sample_n(5000) %>%
  ungroup()

dat.tidy %>%
  drop_na() %>%
  mutate(SnpSet = cut(DeltaPWM, 5)) %>%
  bind_rows(test.SNPs) %>%
  group_by(Dataset, SnpSet) %>%
  mutate(ExpectedP = percent_rank(P)) %>%
  ungroup() %>%
  ggplot(aes(x=-log10(ExpectedP), y=-log10(P), color=SnpSet)) +
  geom_abline() +
  geom_point() +
  facet_wrap(~Dataset) +
  theme_bw() +
  labs(title="QQ plot of eQTL P-values", color="SpliceSiteSeverity", y="-log10(P)")
  
  

```

