---
title: "20230502_GetColocsForCarlos"
output: html_document
date: '2023-05-02'
---

```{r setup, include=TRUE}
knitr::opts_chunk$set(echo = TRUE, warning = F, message = F)

library(tidyverse)
library(RColorBrewer)
library(data.table)
library(edgeR)

# Set theme
theme_set(
  theme_classic() +
  theme(text=element_text(size=16,  family="Helvetica")))

# I use layer a lot, to rotate long x-axis labels
Rotate_x_labels <- theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))

#test plot
ggplot(mtcars, aes(x=mpg, y=cyl)) +
  geom_point()

```


## Intro

To bolster the idea that some meaningful fraction of sQTL/GWAS colocalizations are really driven by eQTL effects, we had the idea of looking at the concordance in eQTL-->GWAS effects for secondary eQTL/GWAS signals. For example, if a sQTL colocalizes with GWAS because it is the primary eQTL and it downregulates MyFavoriteGene, then the sign of the complex trait effect should be the same for secondary eQTLs of the same sign.

So here is our proposed approach to investigate this...

>I would first check the independent susie signals alongside the marginal association Manhattan plots just to make sure they sort of make sense... I haven't done this. Anyway, then I can give you a list of ~13 sQTL/eQTL/GWAS colocalizations (without a hQTL colocalization). Keep in mind colocalization was done with the marginal eQTL signal, which may be made up of multiple independent eQTLs, and colocalization assumes only one QTL so basically we are colocalizing primary signals. There aren't many of eQTL/sQTL/GWAS colocalizations (only 13 or so genes), but I have pretty high confidence in these being sQTL-driven effects for the primary eQTL. And direction of effects are again by in large consistent with sQTL increase in unproductive junction results in downregulation of gene. I can supplement that with a list of hQTL/eQTL/GWAS colocalizations, which will probably yield another 15 or so genes, for which the primary effect is through chromatin. Then we can intersect that list of 30 or so genes with the susie genes with secondary/tertiary signals. We can then check all the secondary signals to see if they map to a significant hQTL/sQTL... I would probably also verify the signals are in low-LD (can just use geuvadis genotypes to calculate LD), then I would check the concordance between the eQTL effect direction at both the primary and secondary signals and gwas.

Carlos has offered to do this analysis, but to get him started I am going to give him a list of all GWAS colocalizations clusters that contain an hQTL/eQTL, or a sQTL/eQTL.

I'm going to do this twice actually, once using the colocalization results where the polyA data used all geauvadis, and again where it's just YRI.

```{r}
gwas.traits <- read_tsv("../code/config/gwas_table.tsv") %>%
  dplyr::rename(GWAS.accession=gwas, gwas.trait=trait)


hyprcoloc.results <- read_tsv("../code/hyprcoloc/Results/ForGWASColoc/GWASColoc_ChromatinAPAAndRNA/results.txt.gz") %>%
# hyprcoloc.results <- read_tsv("../code/hyprcoloc/Results/ForGWASColoc/GWASColoc_ChromatinAPAAndRNAYRI/results.txt.gz") %>%
  dplyr::rename(GWAS.Loci = GWASLeadSnpChrom_Pos_RefAllele_AltAllele_rsID_trait) %>%
  separate(GWAS.Loci, into=c("GWAS.LeadSNP.Chrom", "GWAS.LeadSNP.Pos", "GWAS.accession"), sep="_", remove=F) %>%
  separate_rows(ColocalizedTraits, sep = ",") %>%
  mutate(IsColocalizedWithSomething = !ColocalizedTraits == "None") %>%
  mutate(Trait = if_else(IsColocalizedWithSomething, ColocalizedTraits, DroppedTrait)) %>%
  dplyr::select(-DroppedTrait, -ColocalizedTraits) %>%
  mutate(Trait = str_replace_all(Trait, " ", "")) %>%
  mutate(GWAS.Loci = str_replace_all(GWAS.Loci, " ", "")) %>%
  mutate(Trait = if_else(Trait == GWAS.Loci, paste("GWAS",GWAS.Loci,sep = ";"),Trait)) %>%
  separate(Trait, into=c("PhenotypeClass", "Phenotype"), sep=";", remove=F) %>%
  group_by(GWAS.Loci, HyprcolocIteration) %>%
  mutate(ColocalizedClusterContainsGWASTrait = any(PhenotypeClass=="GWAS") & IsColocalizedWithSomething) %>%
  ungroup() %>%
  inner_join(gwas.traits %>%
               dplyr::select(1:2))

PhenotypeRecodes = c("H3K36ME3"="hQTL", "H3K27AC"="hQTL", "H3K4ME3"="hQTL", "H3K4ME1"="hQTL",
                     "Expression.Splicing"="eQTL", "Expression.Splicing.Subset_YRI"="eQTL", "polyA.Splicing.Subset_YRI"="sQTL", "chRNA.Expression.Splicing"="chRNA eQTL",
                     "APA_Nuclear"="APA QTL", "APA_Total"="APA QTL", "polyA.Splicing"="sQTL", "GWAS"="GWAS")

PhenotypeRecodes.df <- data.frame(PhenotypeRecodes) %>%
  rownames_to_column("PhenotypeClass")


hyprcoloc.results.toplot <- hyprcoloc.results %>%
  filter(!GWAS.accession=="IMSGC2019") %>%
  left_join(PhenotypeRecodes.df) %>%
  mutate(PhenotypeRecodes = if_else(is.na(PhenotypeRecodes), PhenotypeClass, PhenotypeRecodes)) %>%
  filter(!PhenotypeRecodes == "APA QTL") %>%
  group_by(GWAS.Loci, HyprcolocIteration) %>%
  filter(any(ColocalizedClusterContainsGWASTrait) | PhenotypeClass=="GWAS") %>%
  mutate(Category = case_when(
    all(ColocalizedClusterContainsGWASTrait==FALSE) | all(is.na(HyprcolocIteration))  ~ "No molQTL colocs",
    all(PhenotypeRecodes %in% c("GWAS", "hQTL")) ~ "Only hQTL colocs",
    all(PhenotypeRecodes %in% c("GWAS", "eQTL")) ~ "Only eQTL colocs",
    all(PhenotypeRecodes %in% c("GWAS", "eQTL", "hQTL", "chRNA eQTL")) ~ "hQTL+eQTL colocs",
    all(PhenotypeRecodes %in% c("GWAS", "sQTL")) ~ "sQTL colocs",
    all(PhenotypeRecodes %in% c("GWAS", "sQTL", "eQTL", "eQTL")) ~ "sQTL+eQTL colocs",
    # all(PhenotypeRecodes %in% c("GWAS", "sQTL", "chRNA eQTL", "eQTL", "hQTL")) ~ "sQTL+eQTL+hQTL colocs",
    TRUE ~ "Other"
  )) %>%
  mutate(Category2 = case_when(
    all(ColocalizedClusterContainsGWASTrait==FALSE) | all(is.na(HyprcolocIteration))  ~ "No molQTL colocs",
    any(PhenotypeRecodes == "hQTL") & any(PhenotypeRecodes=="eQTL") & any(PhenotypeRecodes=="sQTL") ~ "Other",
    !any(PhenotypeRecodes == "hQTL") & any(PhenotypeRecodes=="eQTL") & any(PhenotypeRecodes=="sQTL") ~ "sQTL+eQTL colocs",
    any(PhenotypeRecodes == "hQTL") & any(PhenotypeRecodes=="eQTL") & !any(PhenotypeRecodes=="sQTL") ~ "hQTL+eQTL colocs",
    !any(PhenotypeRecodes == "hQTL") & any(PhenotypeRecodes=="eQTL") & !any(PhenotypeRecodes=="sQTL") ~ "eQTL only colocs",
    TRUE ~ "Filter out"
  )) %>%
  ungroup()

hyprcoloc.results.toplot %>%
  filter(!Category2=="Filter out") %>%
  ggplot(aes(x=Category, fill=Category2)) +
  geom_bar(position="fill") +
  Rotate_x_labels

hyprcoloc.results.toplot %>%
  distinct(GWAS.Loci, .keep_all=T) %>%
  count(Category2)

hyprcoloc.results.toplot %>%
  distinct(GWAS.Loci, .keep_all=T) %>%
  count(Category)

hyprcoloc.results.toplot %>%
  filter(Category2 == "sQTL+eQTL colocs") %>%
  filter(PhenotypeClass == "Expression.Splicing") %>%
  distinct(Trait)

hyprcoloc.results.toplot %>%
  filter(Category == "sQTL+eQTL colocs") %>%
  filter(PhenotypeClass == "Expression.Splicing") %>%
  distinct(Trait)

hyprcoloc.results.toplot$PhenotypeClass %>% unique()
hyprcoloc.results.toplot$PhenotypeRecodes %>% unique()

```

Ok part of the discrepancy is that previously, if there was a gwas coloc with chRNA sQTL and polyA sQTL that was getting bunched into the "Other" category... That doesn't make good sense... Let's keep with the Category2... And also check the previous beta beta scatter 

```{r}
PC2.filter <- c("Expression.Splicing", "Expression.Splicing.Subset_YRI")
PC2.SignificanceFilter <- c("H3K27AC", "H3K4ME3", "H3K36ME3")
PC1.filter <- c("polyA.Splicing", "H3K27AC", "H3K4ME1", "H3K4ME3")
PC1.filter.Splicing <- PC1.filter[str_detect(PC1.filter, "Splicing")]
PC1.filter.NonSplicing <- PC1.filter[!str_detect(PC1.filter, "Splicing")]

dat.foreQTLQQ <- fread("../code/pi1/PairwisePi1Traits.P.all.txt.gz") %>%
  filter((PC1 %in% PC1.filter))

IntronAnnotatins <- read_tsv("../data/IntronAnnotationsFromYang.tsv.gz") %>%
  mutate(chrom = str_remove_all(chrom, "chr")) %>%
  mutate(Intron = paste(chrom, start, end, sep=":")) %>%
  filter(!str_detect(SuperAnnotation, "NoncodingGene"))

dat.foreQTLQQ.sQTLs <- dat.foreQTLQQ %>%
    filter(PC1 %in% PC1.filter.Splicing) %>%
    group_by(PC1, P1) %>%
    filter(!any((PC2 %in% PC2.SignificanceFilter) & (trait.x.p.in.y < 0.01))) %>%
    ungroup() %>%
    filter(PC2 %in% PC2.filter) %>%
    separate(P1, into=c("Intron", "Cluster"), sep=":clu", remove=F) %>%
    inner_join(
      IntronAnnotatins %>%
        dplyr::select(Intron, SuperAnnotation),
      by="Intron") %>%
    group_by(PC1, PC2, Cluster) %>%
    mutate(SNP_group = case_when(
      all(str_detect(SuperAnnotation, "Productive")) ~ "Productive sQTL cluster",
      any(str_detect(SuperAnnotation, "Unproductive")) ~ "Unproductive sQTL cluster",
      TRUE ~ "sQTL Other"
    )) 


CategoriesWith_sQTLs <- c("sQTL+eQTL colocs", "Other", "sQTL")


inner_join(
  hyprcoloc.results.toplot %>%
    filter(Category2 %in% CategoriesWith_sQTLs) %>%
    filter(PhenotypeClass %in% c("polyA.Splicing")) %>%
    dplyr::select(GWAS.Loci, sQTL=Trait),
  hyprcoloc.results.toplot %>%
    filter(Category2 %in% CategoriesWith_sQTLs) %>%
    filter(PhenotypeClass %in% c("Expression.Splicing")) %>%
    dplyr::select(GWAS.Loci, eQTL=Trait, everything())
) %>%
  inner_join(
    dat.foreQTLQQ.sQTLs %>%
      mutate(sQTL=paste(PC1, P1, sep=";"))
  ) %>%
  mutate(IntronGroup = if_else(str_detect(SuperAnnotation, "Unproductive"), "Unproductive", "Productive")) %>%
  group_by(GWAS.Loci, PC2) %>%
  filter(p_permutation.x == min(p_permutation.x)) %>%
  ungroup() %>%
  mutate(Category2 = recode(Category2, "Other"="...+hQTL")) %>%
  mutate(PC2 = recode(PC2, "Expression.Splicing"="eQTL beta from all GEU","Expression.Splicing.Subset_YRI"="eQTL beta from YRI" )) %>%
  ggplot(aes(x=beta.x, y=x.beta.in.y, color=SNP_group)) +
  geom_point() +
  facet_grid(IntronGroup ~ Category2 ~ PC2) +
  theme_bw() +
  # geom_smooth(method='lm') +
  labs(x="sQTL beta", y="eQTL beta", title="sQTL and eQTL effects for eQTL/sQTL/GWAS colocs", caption="Only top sQTL intron per GWAS colocalization plotted\nEffects relative to top sQTL SNP\nFacets are intron level category category, gwas coloc category (Other includes sQTL+eQTL+hQTL) and whether eQTL effect measured in YRI or all")
```

Ok that still makes sense... Now let's write out some results for Carlos...

```{r}

ColocCategories <- c("eQTL only colocs", "hQTL+eQTL colocs", "sQTL+eQTL colocs", "Other")

hyprcoloc.results.toplot %>%
  filter(Category2 %in% ColocCategories) %>%
  mutate(Category2=recode(Category2, Other="hQTL+sQTL+eQTL colocs")) %>%
  dplyr::select(ColocCategory=Category2, Trait, PhenotypeClass, GWAS.Loci, GWAS.LeadSNP.Chrom, GWAS.LeadSNP.Pos, GWAS.accession, gwas.trait, TopCandidateSNP, PosteriorColocalizationPr) %>%
  distinct(GWAS.Loci, .keep_all=T) %>%
  count(ColocCategory)


```

```{r, eval=F}
hyprcoloc.results.toplot %>%
  filter(Category2 %in% ColocCategories) %>%
  mutate(Category2=recode(Category2, Other="hQTL+sQTL+eQTL colocs")) %>%
  dplyr::select(ColocCategory=Category2, Trait, PhenotypeClass, GWAS.Loci, GWAS.LeadSNP.Chrom, GWAS.LeadSNP.Pos, GWAS.accession, gwas.trait, TopCandidateSNP, PosteriorColocalizationPr) %>%
  write_tsv("../output/GwasColocs_by_MolQTLCategory_ForCarlos.YRI_Only.tsv.gz")
```

