---
title: "MakeFinalFigs_Fig2"
output: html_document
date: '2023-05-18'
---


```{r setup, include=TRUE}
knitr::opts_chunk$set(echo = TRUE, warning = F, message = F)

library(tidyverse)
library(RColorBrewer)
library(data.table)
library(readxl)
library(qvalue)
library(ggrepel)
library(knitr)
library(ggbreak)



# Set theme
theme_set(
  theme_classic() +
  theme(text=element_text(size=16,  family="Helvetica")))

# I use layer a lot, to rotate long x-axis labels
Rotate_x_labels <- theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
```


## List

#### Number QTLs

```{r}
PhenotypeWildcardsAll <- Sys.glob("../code/QTLs/QTLTools/*/PermutationPass.FDR_Added.txt.gz") %>%
  str_replace("../code/QTLs/QTLTools/(.+?)/PermutationPass.FDR_Added.txt.gz", "\\1")

PhenotypeWildcards_ToIncluide <- c("APA_Nuclear", "APA_Total", "Expression.Splicing.Subset_YRI", "Expression.Splicing", "H3K27AC", "H3K36ME3", "H3K4ME3", "H3K4ME1", "MetabolicLabelled.30min", "MetabolicLabelled.60min", "chRNA.Expression.Splicing", "chRNA.Splicing", "polyA.Splicing", "polyA.Splicing.Subset_YRI", "MetabolicLabelled.30min.Splicing", "MetabolicLabelled.60min.Splicing")

GroupedPhenotypeWildcards_ToIncluide <- c( "chRNA.Splicing", "polyA.Splicing.Subset_YRI", "polyA.Splicing",  "MetabolicLabelled.30min.Splicing", "MetabolicLabelled.60min.Splicing")

UngroupedPhenotypeWildcards_ToIncluide <- setdiff(PhenotypeWildcards_ToIncluide, GroupedPhenotypeWildcards_ToIncluide)

Grouped.dat <- setNames(paste0("../code/QTLs/QTLTools/",GroupedPhenotypeWildcards_ToIncluide,"/GroupedPermutationPass.FDR_Added.txt.gz"), GroupedPhenotypeWildcards_ToIncluide) %>%
  lapply(fread) %>%
  bind_rows(.id="PC")

Ungrouped.dat <- setNames(paste0("../code/QTLs/QTLTools/",UngroupedPhenotypeWildcards_ToIncluide,"/PermutationPass.FDR_Added.txt.gz"), UngroupedPhenotypeWildcards_ToIncluide) %>%
  lapply(fread) %>%
  bind_rows(.id="PC")

Full.dat <- bind_rows(
  Grouped.dat, Ungrouped.dat
)

```

First plot number of individuals for each assay

```{r}
SampleSize <- read_tsv("../code/QC/SampleSize.PerPhenotype.tsv", col_names=c("fn", "dummy", "n")) %>%
  mutate(PC = str_replace(fn,"QTLs/QTLTools/(.+?)/OnlyFirstReps.sorted.qqnorm.bed.gz", "\\1")) %>%
  filter(PC %in% c("Expression.Splicing", "Expression.Splicing.Subset_YRI", "chRNA.Expression.Splicing", "MetabolicLabelled.30min", "MetabolicLabelled.60min", "H3K27AC", "H3K4ME3", "H3K4ME1", "H3K36ME3", "APA_Nuclear", "APA_Total"))

SampleSize %>%
  dplyr::select(PC, n) %>%
  kable()



SourcePublications <- setNames(c("Lappalainen et al", "Lappalainen et al", "This study", "Li et al", "Li et al", "Grubert et al", "Grubert et al", "Grubert et al", "This study", "Mittleman et al", "Mittleman et al"), SampleSize$PC)

Labels =  setNames(c("polyA RNA-seq*", "polyA RNA-seq", "chRNA-seq", "30m 4sU RNA-seq", "60m 4sU RNA-seq", "H3K4me1 ChIP-seq", "H3K27Ac ChIP-seq", "H3K4me3 ChIP-seq", "H3K36me3 Cut&Tag", "nuclear RNA 3'seq", "Total RNA 3'seq"), SampleSize$PC)

SampleSizes <- SampleSize %>%
  filter(!PC %in% "Expression.Splicing.Subset_YRI") %>%
  mutate(publication = recode(PC, !!!SourcePublications)) %>%
  mutate(Assay = factor(PC, levels=c("H3K4ME1", "H3K27AC", "H3K4ME3", "H3K36ME3", "chRNA.Expression.Splicing", "APA_Nuclear", "APA_Total", "MetabolicLabelled.30min","MetabolicLabelled.60min", "Expression.Splicing"))) %>%
  ggplot(aes(x=Assay, y=n)) +
  geom_col() +
  # geom_text(aes(label=n), color="black") +
  geom_text(aes(label=publication),angle=90, color="black", y=0, hjust=-0.05) +
  scale_x_discrete(labels=Labels) +
  scale_y_continuous(expand=c(0,0)) +
  scale_y_break(c(100,425), scales="fixed", ticklabels=c(425, 450, 475), expand=F, space=0.4) +
  Rotate_x_labels +
  labs(x="Assay", y="Sample size", caption=str_wrap("*Includes samples non-YRI donors, though some supplementary analyses only utilize the 89 YRI samples in this dataset", 30)) +
  theme(axis.text.x=element_text(size=rel(0.5)))

P.SampleSizes <- ggplotify::as.ggplot(print(SampleSizes))
P.SampleSizes
```
Now number of QTLs

```{r}
relabel.df <- read_tsv("../data/Phenotypes_recode_for_Plotting.txt")

relabel.df %>% head()

Full.dat$PC %>% unique()

MolQTL_Labels <- relabel.df %>% dplyr::select(PC, Shortest) %>% deframe()
MolQTL_AssayLabels <- relabel.df %>% dplyr::select(PC, Assay) %>% deframe()
Factor_Order <- relabel.df %>% dplyr::select(PC, OrderFactor) %>% deframe()

NumQTLs.dat <- Full.dat %>%
  filter(PC %in% c("APA_Total", "APA_Nuclear")) %>%
  separate(phe_id, into=c("gene", "region", "peak"), sep="_") %>%
  arrange(adj_emp_pval) %>% 
  group_by(gene) %>% slice(1) %>%
  bind_rows(
    Full.dat %>%
      filter(!PC %in% c("APA_Total", "APA_Nuclear"))
  ) %>%
  group_by(PC) %>%
  summarise(
    FDR_10 = sum(q<0.1, na.rm=T),
    FDR_05 = sum(q<0.05,  na.rm=T),
    FDR_01 = sum(q<0.01,  na.rm=T),
    TestFeats=n()) %>%
  gather("FDR", "n", -PC, -TestFeats) %>%
  mutate(PC_relabeled = recode(PC, !!!MolQTL_Labels)) %>%
  mutate(Assay_relabeled = recode(PC, !!!MolQTL_AssayLabels)) %>%
  mutate(FinalLabels = paste(PC_relabeled, Assay_relabeled, sep="\n")) %>%
  mutate(OrderFactor = recode(PC, !!!Factor_Order)) %>%
  mutate(FinalLabels = fct_reorder(FinalLabels, OrderFactor))

Num.QTLs <- ggplot(NumQTLs.dat, aes(x=FinalLabels, y=n, fill=FDR)) +
  geom_col(position="identity", width=0.8) +
  Rotate_x_labels +
  scale_y_continuous(trans="log10", breaks=10**(1:5), labels=c(10, 100, "1K", "10K", "100K")) +
  # geom_text_repel(
  #   data = . %>%
  #     gather("FDR","n",-PC),
  #     aes(label=n, y=n), angle=90, hjust=-0.05,
  #   min.segment.length = unit(0, 'lines'),
  #   color="black") +
  geom_text(
    data = . %>%
      filter(FDR== "FDR_10"),
      aes(label=str_glue("{n} / {TestFeats}")), angle=90, hjust=0,
    color="black") +
  coord_cartesian(ylim=c(100,1E5)) +
  scale_fill_manual(values=c("FDR_10"="#ffeda0", "FDR_05"="#feb24c", "FDR_01"="#f03b20"), labels=c("FDR_10"="10%", "FDR_05"="5%", "FDR_01"="1%")) +
  # scale_x_discrete(labels=c())
  labs(x="molQTL type, assay", y="Number QTLs", fill="FDR", caption=str_wrap("*Each sQTL/APA-QTL count here represents a likely independent genetic effect that may effect multiple introns/APA-sites within a local leafcutter cluster of introns/APA-sites", 40)) +
  theme(axis.text.x = element_text(size=9, hjust=1, angle=70))

Num.QTLs

NumQTLs.dat %>% kable()

NumQTLs.dat %>%
  filter(FDR=="FDR_10") %>%
  pull(n) %>% sum()

NumQTLs.dat %>%
  filter(FDR=="FDR_10") %>%
  pull(TestFeats) %>% sum()
```


### Box plots for TTC3

first do some quick data searching to figure out exactly the names of the phenotypes I want to make boxpltos for... 

```{r, eval=F}
dat <- fread("../code/pi1/PairwisePi1Traits.P.all.txt.gz")

PeaksToTSS <- Sys.glob("../code/Misc/PeaksClosestToTSS/*_assigned.tsv.gz") %>%
  setNames(str_replace(., "../code/Misc/PeaksClosestToTSS/(.+?)_assigned.tsv.gz", "\\1")) %>%
  lapply(read_tsv) %>%
  bind_rows(.id="ChromatinMark") %>%
  mutate(GenePeakPair = paste(gene, peak, sep = ";")) %>%
  distinct(ChromatinMark, peak, gene, .keep_all=T)

PeaksToTSS %>%
  filter(ChromatinMark == "H3K4ME3") %>%
  filter(str_detect(gene, "ENSG00000075234"))

dat %>%
  filter(PC1 == "Expression.Splicing") %>%
  filter(str_detect(P1, "ENSG00000075234")) %>%
  # pull(PC2) %>% unique()
  filter(PC2 == "polyA.Splicing")

dat %>%
  filter(PC1 == "Expression.Splicing") %>%
  filter(str_detect(P1, "ENSG00000075234")) %>%
  filter(P2 %in% c("H3K4ME3_peak_31179", "22:46292328:46292791:clu_48515_+"))
  
```

So I'm looking to make boxplots for H3K4ME3_peak_31179 (hQTL), ENSG00000075234.17 (polyA eQTL), and chRNA sQTL and polyA sQTL for 22:46292328:46292791:clu_48515_+

TopSNP: "22:46291323:G:A"

Now use my helper script to get genotype/phenotype data for the top eSNP for those traits..

```{bash, eval=F}
cd ../code/scripts/GenometracksByGenotype

python ExtractPhenotypeBedByGenotype.py --Bed ../../QTLs/QTLTools/Expression.Splicing/OnlyFirstReps.sorted.qqnorm.bed.gz --VCF ../../Genotypes/1KG_GRCh38/Autosomes.vcf.gz --FeatureName ENSG00000075234.17 --SnpPos chr22:46291323 > ../../scratch/Boxplot.dat.eQTL.tsv

python ExtractPhenotypeBedByGenotype.py --Bed ../../QTLs/QTLTools/polyA.Splicing/OnlyFirstReps.sorted.qqnorm.bed.gz --VCF ../../Genotypes/1KG_GRCh38/Autosomes.vcf.gz --FeatureName 22:46292328:46292791:clu_48515_+  --SnpPos chr22:46291323 > ../../scratch/Boxplot.dat.polyAsQTL.tsv

python ExtractPhenotypeBedByGenotype.py --Bed ../../QTLs/QTLTools/polyA.Splicing/OnlyFirstReps.sorted.qqnorm.bed.gz --VCF ../../Genotypes/1KG_GRCh38/Autosomes.vcf.gz --FeatureName 22:46292328:46292791:clu_48515_+  --SnpPos chr22:46291323 > ../../scratch/Boxplot.dat.polyAsQTL.tsv

python ExtractPhenotypeBedByGenotype.py --Bed ../../QTLs/QTLTools/H3K4ME3/OnlyFirstReps.sorted.qqnorm.bed.gz --VCF ../../Genotypes/1KG_GRCh38/Autosomes.vcf.gz --FeatureName H3K4ME3_peak_31179  --SnpPos chr22:46291323 > ../../scratch/Boxplot.dat.PromoterhQTL.tsv
```

Now let's read in that data and do boxplots...

```{r}
dat.for.boxplots <- Sys.glob("../code/scratch/Boxplot.dat.*.tsv") %>%
  setNames(str_replace(., "../code/scratch/Boxplot.dat.(.+?).tsv", "\\1")) %>%
  lapply(read_tsv, col_names=c("SampleID", "P", "genotype", "Ref", "Alt", "ID"), skip=1) %>%
  bind_rows(.id="PC")

dat.for.boxplots %>%
  filter(!genotype==3) %>%
  mutate(genotype= factor(genotype, levels=c(0,1,2))) %>%
  ggplot(aes(x=genotype, y=P, group=genotype)) +
  geom_boxplot(outlier.shape = NA, aes(color=genotype)) +
  # geom_jitter(width=0.25, alpha=0.1) +
  geom_smooth(method='lm', aes(group=1), se=F, color="gray") +
  scale_x_discrete(labels=c("0"="GG", "1"="GA", "2"="AA")) +
  scale_color_manual(values=c("0"="#f03b20", "1"="#7a0177", "2"="#08519c")) +
  # scale_x_discrete() +
  facet_wrap(~PC) +
  theme(legend.position="none") +
  labs(x=NULL, color=NULL)
```

```{r, eval=F}
dat.for.boxplots$PC %>% unique()

dat.for.boxplots %>%
  filter(!genotype==3) %>%
  filter(PC=="PromoterhQTL") %>%
  mutate(genotype= factor(genotype, levels=c(0,1,2))) %>%
  ggplot(aes(x=genotype, y=P, group=genotype)) +
  geom_boxplot(outlier.shape = NA, aes(color=genotype)) +
  # geom_jitter(width=0.25, alpha=0.1) +
  geom_smooth(method='lm', aes(group=1), se=F, color="gray") +
  scale_x_discrete(labels=c("0"="GG", "1"="GA", "2"="AA")) +
  scale_color_manual(values=c("0"="#f03b20", "1"="#7a0177", "2"="#08519c")) +
  geom_text(x=-Inf, y=Inf, label=" beta=0.15\n P=0.48", hjust=0, vjust=1, size=5.5) +
  # scale_x_discrete() +
  theme(legend.position="none") +
  Rotate_x_labels +
  labs(x=NULL, color=NULL, y="H3K4me3 @ promoter")
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig2_ExampleBoxplot_hQTL.pdf", height=3, width=2)

dat.for.boxplots %>%
  filter(!genotype==3) %>%
  filter(PC=="eQTL") %>%
  mutate(genotype= factor(genotype, levels=c(0,1,2))) %>%
  ggplot(aes(x=genotype, y=P, group=genotype)) +
  geom_boxplot(outlier.shape = NA, aes(color=genotype)) +
  # geom_jitter(width=0.25, alpha=0.1) +
  geom_smooth(method='lm', aes(group=1), se=F, color="gray") +
  scale_x_discrete(labels=c("0"="GG", "1"="GA", "2"="AA")) +
  scale_color_manual(values=c("0"="#f03b20", "1"="#7a0177", "2"="#08519c")) +
  geom_text(x=-Inf, y=Inf, label=" beta=1.1\n P<5E-43", hjust=0, vjust=1, size=5.5) +
  # scale_x_discrete() +
  theme(legend.position="none") +
  Rotate_x_labels +
  labs(x=NULL, color=NULL, y="TTC38 expression")
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig2_ExampleBoxplot_eQTL.pdf", height=3, width=2)

dat.for.boxplots %>%
  filter(!genotype==3) %>%
  filter(PC=="polyAsQTL") %>%
  mutate(genotype= factor(genotype, levels=c(0,1,2))) %>%
  ggplot(aes(x=genotype, y=P, group=genotype)) +
  geom_boxplot(outlier.shape = NA, aes(color=genotype)) +
  # geom_jitter(width=0.25, alpha=0.1) +
  geom_smooth(method='lm', aes(group=1), se=F, color="gray") +
  scale_x_discrete(labels=c("0"="GG", "1"="GA", "2"="AA")) +
  scale_color_manual(values=c("0"="#f03b20", "1"="#7a0177", "2"="#08519c")) +
  geom_text(x=-Inf, y=Inf, label=" beta=-1.1\n P<3e-24", hjust=0, vjust=1, size=5.5) +
  # scale_x_discrete() +
  theme(legend.position="none") +
  Rotate_x_labels +
  labs(x=NULL, color=NULL, y="Splicing of cassette exon")
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig2_ExampleBoxplot_polyAsQTL.pdf", height=3, width=2)

dat.for.boxplots %>%
  filter(!genotype==3) %>%
  filter(PC=="chRNAsQTL") %>%
  mutate(genotype= factor(genotype, levels=c(0,1,2))) %>%
  ggplot(aes(x=genotype, y=P, group=genotype)) +
  geom_boxplot(outlier.shape = NA, aes(color=genotype)) +
  # geom_jitter(width=0.25, alpha=0.1) +
  geom_smooth(method='lm', aes(group=1), se=F, color="gray") +
  scale_x_discrete(labels=c("0"="GG", "1"="GA", "2"="AA")) +
  scale_color_manual(values=c("0"="#f03b20", "1"="#7a0177", "2"="#08519c")) +
  geom_text(x=-Inf, y=Inf, label=" beta=-1.3\n P<4e-19", hjust=0, vjust=1, size=5.5) +
  # scale_x_discrete() +
  theme(legend.position="none") +
  Rotate_x_labels +
  labs(x=NULL, color=NULL, y="Splicing of cassette exon")
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig2_ExampleBoxplot_chRNAsQTL.pdf", height=3, width=2)
```

### Classification of unannotated and unproductive juncs

Actually a supplement related to fig1

```{r}
juncs.long <- fread("../code/SplicingAnalysis/CombinedJuncTables/YRI.tsv.gz")

IntronAnnotations <- read_tsv("../data/IntronAnnotationsFromYang.tsv.gz") %>%
  mutate(stop = end-1)

juncs.long.summary <- juncs.long %>%
  dplyr::select(chrom, start, stop, strand, Dataset, Count) %>%
  inner_join(IntronAnnotations) %>%
  group_by(Dataset, SemiSupergroupAnnotations, SuperAnnotation) %>%
  summarise(Sum=sum(Count))


P.PercentAnnotated <- juncs.long.summary %>%
  ungroup() %>%
  filter(Dataset %in% c("Expression.Splicing", "chRNA.Expression.Splicing")) %>%
  filter(!str_detect(SuperAnnotation, "Noncoding")) %>%
  group_by(Dataset, SuperAnnotation) %>%
  summarise(TotalSum = sum(Sum)) %>%
  ungroup() %>%
  mutate(PlotGroup = if_else(SuperAnnotation=="AnnotatedJunc_ProductiveCodingGene", "Annotated productive (basic tag)", "Other")) %>%
  group_by(Dataset) %>%
  mutate(DatasetSum = sum(TotalSum)) %>%
  ungroup() %>%
  group_by(Dataset, PlotGroup) %>%
  summarise(Percent=sum(TotalSum)/DatasetSum) %>%
  ungroup() %>%
  distinct() %>%
  mutate(Dataset = factor(Dataset, levels=c("Expression.Splicing", "chRNA.Expression.Splicing"))) %>%
  ggplot(aes(x=Dataset, y=Percent*100, fill=PlotGroup)) +
  geom_col() +
  scale_x_discrete(labels=c("Expression.Splicing"="polyA RNA", "chRNA.Expression.Splicing"="chRNA")) +
  scale_y_continuous(expand=c(0,0)) +
  scale_fill_manual(values=c("Annotated productive (basic tag)"="#1f78b4", "Other"="#969696")) + 
  scale_y_break(c(5, 95), expand=F, space=0.5) +
  labs(y="Percent of splice junction reads", fill="Intron classification") +
  theme(legend.position="none")
P.PercentAnnotated <- ggplotify::as.ggplot(print(P.PercentAnnotated))
P.PercentAnnotated

P.PercentOfUnannotated <- juncs.long.summary %>%
  ungroup() %>%
  filter(Dataset %in% c("chRNA.Expression.Splicing")) %>%
  filter(!str_detect(SuperAnnotation, "Noncoding")) %>%
  filter(SuperAnnotation!="AnnotatedJunc_ProductiveCodingGene") %>%
  mutate(PercentOfNon_AnnotatedProductive = Sum/sum(Sum)*100) %>%
  arrange(desc(SuperAnnotation), PercentOfNon_AnnotatedProductive) %>%
  mutate(n=row_number()) %>%
  mutate(SemiSupergroupAnnotations=fct_reorder(SemiSupergroupAnnotations, n)) %>%
  ggplot(aes(x=Dataset, y=PercentOfNon_AnnotatedProductive, fill=SemiSupergroupAnnotations)) +
  geom_col()
P.PercentOfUnannotated
```

One more version... with fixed scale... borrowed code from [here](https://github.com/slowkow/ggrepel/issues/161)

```{r}
position_stack_and_nudge <- function(x = 0, y = 0, vjust = 1, reverse = FALSE) {
  ggproto(NULL, PositionStackAndNudge,
    x = x,
    y = y,
    vjust = vjust,
    reverse = reverse
  )
}

#' @rdname ggplot2-ggproto
#' @format NULL
#' @usage NULL
#' @noRd
PositionStackAndNudge <- ggproto("PositionStackAndNudge", PositionStack,
  x = 0,
  y = 0,

  setup_params = function(self, data) {
    c(
        list(x = self$x, y = self$y),
        ggproto_parent(PositionStack, self)$setup_params(data)
    )
  },

  compute_layer = function(self, data, params, panel) {
    # operate on the stacked positions (updated in August 2020)
    data = ggproto_parent(PositionStack, self)$compute_layer(data, params, panel)

    x_orig <- data$x
    y_orig <- data$y
    # transform only the dimensions for which non-zero nudging is requested
    if (any(params$x != 0)) {
      if (any(params$y != 0)) {
        data <- transform_position(data, function(x) x + params$x, function(y) y + params$y)
      } else {
        data <- transform_position(data, function(x) x + params$x, NULL)
      }
    } else if (any(params$y != 0)) {
      data <- transform_position(data, function(x) x, function(y) y + params$y)
    }
    data$nudge_x <- data$x
    data$nudge_y <- data$y
    data$x <- x_orig
    data$y <- y_orig

    data
  },

  compute_panel = function(self, data, params, scales) {
      ggproto_parent(PositionStack, self)$compute_panel(data, params, scales)
  }
)
```

```{r}

AnnotationColors <- IntronAnnotations %>%
  distinct(SemiSupergroupAnnotations, SuperAnnotation) %>%
  mutate(Color = recode(SuperAnnotation, !!!c(
    "AnnotatedJunc_NoncodingGene"="#6a3d9a",
    "UnannotatedJunc_NoncodingGene"="#cab2d6",
    "AnnotatedJunc_ProductiveCodingGene"="#1f78b4",
    "UnannotatedJunc_ProductiveCodingGene"="#a6cee3",
    "AnnotatedJunc_UnproductiveCodingGene"="#e31a1c",
    "UnannotatedJunc_UnproductiveCodingGene"="#fb9a99")))

P.PercentOfUnannotated <- juncs.long.summary %>%
  ungroup() %>%
  filter(Dataset %in% c("chRNA.Expression.Splicing")) %>%
  filter(!str_detect(SuperAnnotation, "Noncoding")) %>%
  filter(SuperAnnotation!="AnnotatedJunc_ProductiveCodingGene") %>%
  mutate(PercentOfNon_AnnotatedProductive = Sum/sum(Sum)*100) %>%
  mutate(SemiSupergroupAnnotations=if_else(Sum <= 996286 & SuperAnnotation=="UnannotatedJunc_UnproductiveCodingGene", "Other predicted non-productive", SemiSupergroupAnnotations)) %>%
  group_by(Dataset, SemiSupergroupAnnotations, SuperAnnotation) %>%
  summarise(PercentOfNon_AnnotatedProductive=sum(PercentOfNon_AnnotatedProductive)) %>%
  ungroup() %>%
  arrange(desc(SuperAnnotation), PercentOfNon_AnnotatedProductive) %>%
  mutate(n=row_number()) %>%
  mutate(SemiSupergroupAnnotations=fct_reorder(SemiSupergroupAnnotations, n)) %>%
  ggplot(aes(x=Dataset, y=PercentOfNon_AnnotatedProductive, fill=SemiSupergroupAnnotations)) +
  geom_col(color='black') +
  geom_text_repel(
      aes(label = SemiSupergroupAnnotations, x=1.4),
      max.overlaps = Inf,
      min.segment.length=0,
      label.padding=1,
      position = position_stack_and_nudge(vjust = 0.5, x = 1),
      direction = 'both',
      arrow=NULL
  ) +
  scale_y_continuous(expand=c(0,0)) +
  scale_x_discrete(expand=c(0,0)) +
  scale_fill_manual(values=AnnotationColors %>% dplyr::select(SemiSupergroupAnnotations, Color) %>% deframe(), na.value="#fb9a99") +
  theme(legend.position='none') +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  labs(y="Percent junction reads amongst other")

P.PercentOfUnannotated

P.PercentJuncsLegend.dat <- 
  AnnotationColors %>%
  distinct(SuperAnnotation, .keep_all=T) %>%
  filter(!str_detect(SuperAnnotation, "Noncoding"))

P.PercentJuncsLegend <- ggplot(P.PercentJuncsLegend.dat, aes(x=SuperAnnotation, y=1, fill=SuperAnnotation)) +
  geom_col() +
  scale_fill_manual(values=P.PercentJuncsLegend.dat %>%
                      dplyr::select(SuperAnnotation, Color) %>% deframe(), labels=c("AnnotatedJunc_UnproductiveCodingGene"="Annotated unproductive", "AnnotatedJunc_ProductiveCodingGene"="Annotated productive", "UnannotatedJunc_ProductiveCodingGene"="Unannotated unproductive", "UnannotatedJunc_UnproductiveCodingGene"="Unannotated productive")) +
  labs(fill="Intron classification") 

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_Final_Fig_S_ScatterReasonsForJuncClassification.png",Scatter.P, height=12, width=10)

```

Figure supplement that goes along with this... chRNA vs polyA scatter for junction RPM...


```{r}
juncs.long.summary.ForReasonsPlot <- juncs.long %>%
  filter(Dataset %in% c("Expression.Splicing", "chRNA.Expression.Splicing")) %>%
  dplyr::select(chrom, start, stop, strand, Dataset, Count) %>%
  inner_join(IntronAnnotations) %>%
  group_by(chrom, start, stop, strand, Dataset, SemiSupergroupAnnotations, SuperAnnotation) %>%
  summarise(Sum=sum(Count)) %>%
  ungroup() %>%
  group_by(Dataset) %>%
  mutate(TotalSum= sum(Sum)) %>%
  mutate(RPM = Sum/TotalSum*1E6) %>%
  ungroup() %>%
  dplyr::select(-TotalSum) %>%
  pivot_wider(names_from="Dataset", values_from=c("RPM", "Sum"), values_fill = 0)

juncs.long.summary.ForReasonsPlot %>%
  distinct(SuperAnnotation, SemiSupergroupAnnotations)

juncs.long.summary.ForReasonsPlot %>%
  distinct(SuperAnnotation)

SupergroupLabels = c(
  "AnnotatedJunc_ProductiveCodingGene"="Annotated\nProductive",
  "UnannotatedJunc_ProductiveCodingGene"="Unannotated\nProductive",
  "AnnotatedJunc_UnproductiveCodingGene"="Annotated\nUnproductive",
  "UnannotatedJunc_UnproductiveCodingGene"="Unannotated\nUnproductive",
  "AnnotatedJunc_NoncodingGene"="Annotated\nNoncoding or hypervariable gene",
  "UnannotatedJunc_NoncodingGene"="Unannotated\nNoncoding or hypervariable gene")

Orders <- juncs.long.summary.ForReasonsPlot %>%
  distinct(SuperAnnotation, SemiSupergroupAnnotations) %>%
  mutate(SuperAnnotation = factor(SuperAnnotation, levels=names(SupergroupLabels))) %>%
  arrange(SuperAnnotation, SemiSupergroupAnnotations) %>%
  mutate(rn = row_number())

Scatter.P.dat <- juncs.long.summary.ForReasonsPlot %>%
  add_count(SemiSupergroupAnnotations) %>%
  filter(n>100) %>%
  inner_join(Orders) %>%
  mutate(SemiSupergroupAnnotations=fct_reorder(SemiSupergroupAnnotations, rn))


Scatter.P.dat.summary <- Scatter.P.dat %>%
  filter(Sum_Expression.Splicing + Sum_chRNA.Expression.Splicing >= 10) %>%
  mutate(Ratio = RPM_chRNA.Expression.Splicing/RPM_Expression.Splicing) %>%
  group_by(SemiSupergroupAnnotations, SuperAnnotation, n) %>%
  summarise(median=round(median(Ratio, na.rm=F), digits=2)) %>%
  mutate(label = str_glue("n:{n}\nmedian:{median}"))



Scatter.P <- Scatter.P.dat %>%
  ggplot(aes(x=RPM_chRNA.Expression.Splicing, y=RPM_Expression.Splicing)) +
  geom_point(alpha=0.1) +
  geom_abline(color='red') +
  geom_density_2d(aes(color=SuperAnnotation)) +
  geom_text(data = Scatter.P.dat.summary, aes(label=label),x=-Inf, y=Inf, size=3, hjust=0, vjust=1.1) +
  scale_x_continuous(trans='log10') +
  scale_y_continuous(trans='log10') +
  scale_color_manual(values=AnnotationColors %>% dplyr::select(SuperAnnotation, Color) %>% deframe(), na.value="#fb9a99", labels=SupergroupLabels) +
  facet_wrap(~SemiSupergroupAnnotations, labeller = label_wrap_gen(width=14)) +
  coord_fixed() +
  theme(strip.text.x = element_text(size = 7), legend.position="bottom") +
  guides(colour = guide_legend(nrow = 2)) +
  labs(title="Reason for splice junction classification", x="chRNA\nRPM across junctions", y="steady-state RNA\nRPM across junctions", color="Intron classification") +
  Rotate_x_labels

# ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_Final_Fig_S_ScatterReasonsForJuncClassification.pdf",Scatter.P, height=12, width=10)


Scatter.P

Scatter.P.NoPoints <- Scatter.P.dat %>%
  ggplot(aes(x=RPM_chRNA.Expression.Splicing, y=RPM_Expression.Splicing)) +
  # geom_point(alpha=0.1) +
  geom_abline(color='red') +
  geom_density_2d(aes(color=SuperAnnotation)) +
  geom_text(data = Scatter.P.dat.summary, aes(label=label),x=-Inf, y=Inf, size=3, hjust=0, vjust=1.1) +
  scale_x_continuous(trans='log10') +
  scale_y_continuous(trans='log10') +
  scale_color_manual(values=AnnotationColors %>% dplyr::select(SuperAnnotation, Color) %>% deframe(), na.value="#fb9a99", labels=SupergroupLabels) +
  facet_wrap(~SemiSupergroupAnnotations, labeller = label_wrap_gen(width=14)) +
  coord_fixed() +
  theme(strip.text.x = element_text(size = 7), legend.position="bottom") +
  guides(colour = guide_legend(nrow = 2)) +
  labs(title="Reason for splice junction classification", x="chRNA\nRPM across junctions", y="steady-state RNA\nRPM across junctions", color="Intron classification") +
  Rotate_x_labels

# ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_Final_Fig_S_ScatterReasonsForJuncClassificationNoPoints.pdf",Scatter.P.NoPoints, height=12, width=10)

Scatter.P.dat %>%
  filter(SemiSupergroupAnnotations=='basic tag') %>%
  ggplot(aes(x=RPM_chRNA.Expression.Splicing, y=RPM_Expression.Splicing)) +
  geom_point(alpha=0.1) +
  geom_abline(color='red') +
  geom_density_2d(aes(color=SuperAnnotation)) +
  geom_text(data = Scatter.P.dat.summary %>%
              filter(SemiSupergroupAnnotations=='basic tag'), aes(label=label),x=-Inf, y=Inf, size=3, hjust=0, vjust=1.1) +
  scale_x_continuous(trans='log10') +
  scale_y_continuous(trans='log10') +
  scale_color_manual(values=AnnotationColors %>% dplyr::select(SuperAnnotation, Color) %>% deframe(), na.value="#fb9a99", labels=SupergroupLabels) +
  # facet_wrap(~SemiSupergroupAnnotations, labeller = label_wrap_gen(width=14)) +
  coord_fixed() +
  theme(strip.text.x = element_text(size = 7), legend.position="none") +
  guides(colour = guide_legend(nrow = 2)) +
  labs(x="chRNA\nRPM across junctions", y="steady-state RNA\nRPM across junctions", color="Intron classification") +
  Rotate_x_labels

# ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_Final_Fig_S_ScatterReasonsForJuncClassification_basicOnly.pdf", height=5, width=5)

# ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_Final_Fig_S_ScatterReasonsForJuncClassification_basicOnly.png", height=5, width=5)

Scatter.P.dat %>%
  filter(SuperAnnotation%in% c("UnannotatedJunc_UnproductiveCodingGene", "AnnotatedJunc_UnproductiveCodingGene")) %>%
  ggplot(aes(x=RPM_chRNA.Expression.Splicing, y=RPM_Expression.Splicing)) +
  geom_point(alpha=0.05) +
  # geom_hex(bins=100) +
  geom_abline(color='red') +
  scale_fill_viridis_c(trans = "log") +
  # geom_density_2d(color='red') +
  geom_text(data = . %>%
              filter(Sum_Expression.Splicing + Sum_chRNA.Expression.Splicing >= 10) %>%
              mutate(Ratio = RPM_chRNA.Expression.Splicing/RPM_Expression.Splicing) %>%
              summarise(n=n(), median=round(median(Ratio, na.rm=F), digits=2)) %>%
              mutate(label = str_glue("n:{n}\nmedian:{median}")),
            aes(label=label),x=-Inf, y=Inf, size=3, hjust=0, vjust=1.1) +
  scale_x_continuous(trans='log10') +
  scale_y_continuous(trans='log10') +
  coord_fixed() +
  theme(strip.text.x = element_text(size = 7), legend.position="none") +
  guides(colour = guide_legend(nrow = 2)) +
  labs(x="chRNA\nRPM across junctions", y="steady-state RNA\nRPM across junctions", color="Intron classification") +
  Rotate_x_labels

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_Final_Fig_S_ScatterReasonsForJuncClassification_UnproductiveOnlyPoints.pdf", height=5, width=5)


```



Number eQTLs per gene, hQTLs, sQTLs

```{r}
dat <- fread("../code/pi1/PairwisePi1Traits.P.all.txt.gz")

dat$PC2 %>% unique()



NumMolQTLsPerGene <- dat %>%
  filter(PC1=="Expression.Splicing" & FDR.x < 0.1) %>%
  mutate(P2 = case_when(
    PC2 %in% c("polyA.Splicing", "chRNA.Splicing", "polyA.Splicing.Subset_YRI") ~ str_extract(P2, "clu_.+$"),
    TRUE ~ P2
  )) %>%
  group_by(PC1, PC2, P2, GeneLocus) %>%
  slice(which.min(FDR.y)) %>%
  ungroup() %>%
  mutate(IsPc2SigFDR = FDR.y < 0.1) %>%
  mutate(IsPc2SigPermutationP = p_permutation.y < 0.01) %>%
  group_by(PC2, GeneLocus) %>%
  summarise(Num_PC2_QTLs_FDR = sum(IsPc2SigFDR),
            Num_PC2_QTLs_PermutationP = sum(IsPc2SigPermutationP)) %>%
  ungroup()

NumMolQTLsPerGene %>%
  filter(PC2 %in% c("H3K27AC", "polyA.Splicing")) %>%
  ggplot(aes(x=Num_PC2_QTLs_PermutationP)) +
  stat_ecdf() +
  facet_wrap(~PC2) +
  coord_cartesian(xlim=c(0,5))

NumMolQTLsPerGene %>%
  filter(PC2 %in% c("H3K27AC", "polyA.Splicing")) %>%
  ggplot(aes(x=Num_PC2_QTLs_FDR)) +
  stat_ecdf() +
  facet_wrap(~PC2) +
  coord_cartesian(xlim=c(0,5))

NumMolQTLsPerGene %>%
  ggplot(aes(x=Num_PC2_QTLs_PermutationP)) +
  stat_ecdf() +
  facet_wrap(~PC2) +
  coord_cartesian(xlim=c(0,5))
```

Ok, now let's plot that nicer for saving...

```{r}
eGenes <- dat %>%
  filter(PC1=="Expression.Splicing" & FDR.x < 0.1) %>%
  pull(P1) %>% unique()

P.Number_hQTLsForColoc <- NumMolQTLsPerGene %>%
  filter(GeneLocus %in% eGenes) %>%
  # filter(PC2 %in% c("H3K27AC", "polyA.Splicing")) %>%
  filter(PC2=="H3K27AC") %>%
  mutate(n = if_else(Num_PC2_QTLs_PermutationP>5, as.integer(5), Num_PC2_QTLs_PermutationP)) %>%
  count(PC2, n) %>%
  ggplot(aes(x=n, y=nn)) +
  geom_col() +
  scale_x_continuous(breaks=0:5, labels=c(0:4, ">4")) +
  labs(x="Number H3K27AC hQTLs w/in 100kb of gene", y="Number eGenes")
NumMolQTLsPerGene %>%
  filter(GeneLocus %in% eGenes) %>%
  # filter(PC2 %in% c("H3K27AC", "polyA.Splicing")) %>%
  filter(PC2=="H3K27AC") %>%
  mutate(n = if_else(Num_PC2_QTLs_PermutationP>5, as.integer(5), Num_PC2_QTLs_PermutationP)) %>%
  count(PC2, n)

P.Number_sQTLsForColoc <- NumMolQTLsPerGene %>%
  filter(GeneLocus %in% eGenes) %>%
  # filter(PC2 %in% c("H3K27AC", "polyA.Splicing")) %>%
  filter(PC2=="polyA.Splicing") %>%
  mutate(n = if_else(Num_PC2_QTLs_PermutationP>5, as.integer(5), Num_PC2_QTLs_PermutationP)) %>%
  count(PC2, n) %>%
  ggplot(aes(x=n, y=nn)) +
  geom_col() +
  scale_x_continuous(breaks=0:5, labels=c(0:4, ">4")) +
  labs(x="Number sQTLs within gene", y="Number eGenes")
NumMolQTLsPerGene %>%
  filter(GeneLocus %in% eGenes) %>%
  # filter(PC2 %in% c("H3K27AC", "polyA.Splicing")) %>%
  filter(PC2=="polyA.Splicing") %>%
  mutate(n = if_else(Num_PC2_QTLs_PermutationP>5, as.integer(5), Num_PC2_QTLs_PermutationP)) %>%
  count(PC2, n)

P.Number_hQTLsForColoc
P.Number_sQTLsForColoc
```

Same plots, but only considering eGenes discovered in YRI subset

```{r}
eGenes <- dat %>%
  filter(PC1=="Expression.Splicing.Subset_YRI" & FDR.x < 0.1) %>%
  pull(P1) %>% unique()

P.Number_hQTLsForColoc <- NumMolQTLsPerGene %>%
  filter(GeneLocus %in% eGenes) %>%
  # filter(PC2 %in% c("H3K27AC", "polyA.Splicing")) %>%
  filter(PC2=="H3K27AC") %>%
  mutate(n = if_else(Num_PC2_QTLs_PermutationP>5, as.integer(5), Num_PC2_QTLs_PermutationP)) %>%
  count(PC2, n) %>%
  ggplot(aes(x=n, y=nn)) +
  geom_col() +
  scale_x_continuous(breaks=0:5, labels=c(0:4, ">4")) +
  scale_y_continuous(expand=c(0,0)) +
  labs(x="Number H3K27AC hQTLs w/in 100kb of gene", y="Number eGenes")
NumMolQTLsPerGene %>%
  filter(GeneLocus %in% eGenes) %>%
  # filter(PC2 %in% c("H3K27AC", "polyA.Splicing")) %>%
  filter(PC2=="H3K27AC") %>%
  mutate(n = if_else(Num_PC2_QTLs_PermutationP>5, as.integer(5), Num_PC2_QTLs_PermutationP)) %>%
  count(PC2, n)

P.Number_sQTLsForColoc <- NumMolQTLsPerGene %>%
  filter(GeneLocus %in% eGenes) %>%
  # filter(PC2 %in% c("H3K27AC", "polyA.Splicing")) %>%
  filter(PC2=="polyA.Splicing") %>%
  mutate(n = if_else(Num_PC2_QTLs_PermutationP>5, as.integer(5), Num_PC2_QTLs_PermutationP)) %>%
  count(PC2, n) %>%
  ggplot(aes(x=n, y=nn)) +
  geom_col() +
  scale_x_continuous(breaks=0:5, labels=c(0:4, ">4")) +
  scale_y_continuous(expand=c(0,0)) +
  labs(x="Number sQTLs within gene", y="Number eGenes")
NumMolQTLsPerGene %>%
  filter(GeneLocus %in% eGenes) %>%
  # filter(PC2 %in% c("H3K27AC", "polyA.Splicing")) %>%
  filter(PC2=="polyA.Splicing") %>%
  mutate(n = if_else(Num_PC2_QTLs_PermutationP>5, as.integer(5), Num_PC2_QTLs_PermutationP)) %>%
  count(PC2, n)

P.Number_hQTLsForColoc
P.Number_sQTLsForColoc
```


```{r}
coloc.dat <- read_tsv("../code/hyprcoloc/Results/ForColoc/MolColocNonRedundantFullSplicing/tidy_results_OnlyColocalized.txt.gz") %>%
  separate(phenotype_full, into=c("PC", "P"), sep=";")

Dat.TopQTLInFamily <- dat %>%
  filter(PC1=="Expression.Splicing" & FDR.x < 0.1) %>%
  mutate(P2 = case_when(
    PC2 %in% c("polyA.Splicing", "chRNA.Splicing", "polyA.Splicing.Subset_YRI") ~ str_extract(P2, "clu_.+$"),
    TRUE ~ P2
  )) %>%
  group_by(PC1, PC2, P2, GeneLocus) %>%
  slice(which.min(FDR.y)) %>%
  ungroup() %>%
  group_by(PC1, PC2, GeneLocus) %>%
  slice(which.min(FDR.y)) %>%
  ungroup()

coloc.dat %>%
  mutate(P = case_when(
    PC %in% c("polyA.Splicing", "chRNA.Splicing", "polyA.Splicing.Subset_YRI") ~ str_extract(P, "clu_.+$"),
    TRUE ~ P
  )) %>%
  left_join(
    Dat.TopQTLInFamily %>%
      dplyr::select(Locus=GeneLocus, PC=PC2, P=P2) %>%
      mutate(TopMolQTLAroundGene = T)
  ) %>%
  replace_na(list(TopMolQTLAroundGene=F)) %>%
  group_by(Locus, snp) %>%
  filter(any(PC=="Expression.Splicing.Subset_YRI")) %>%
  ungroup() %>%
  group_by(Locus, PC) %>%
  summarise(ContainsTopMolQTL = any(TopMolQTLAroundGene)) %>%
  ungroup() %>%
  group_by(PC) %>%
  summarise(FracContainsTop = sum(ContainsTopMolQTL)/n()) %>%
  ungroup()
```


Same analysis without renaming to match all in cluster

```{r}
Dat.TopQTLInFamily2 <- dat %>%
  filter(PC1=="Expression.Splicing" & FDR.x < 0.1) %>%
  # mutate(P2 = case_when(
  #   PC2 %in% c("polyA.Splicing", "chRNA.Splicing", "polyA.Splicing.Subset_YRI") ~ str_extract(P2, "clu_.+$"),
  #   TRUE ~ P2
  # )) %>%
  # group_by(PC1, PC2, P2, GeneLocus) %>%
  # slice(which.min(FDR.y)) %>%
  # ungroup() %>%
  group_by(PC1, PC2, GeneLocus) %>%
  slice(which.min(FDR.y)) %>%
  ungroup()

dat.FractionNonPrimaryMolQTLColocs <- coloc.dat %>%
  # mutate(P = case_when(
  #   PC %in% c("polyA.Splicing", "chRNA.Splicing", "polyA.Splicing.Subset_YRI") ~ str_extract(P, "clu_.+$"),
  #   TRUE ~ P
  # )) %>%
  left_join(
    Dat.TopQTLInFamily2 %>%
      dplyr::select(Locus=GeneLocus, PC=PC2, P=P2) %>%
      mutate(TopMolQTLAroundGene = T)
  ) %>%
  replace_na(list(TopMolQTLAroundGene=F)) %>%
  group_by(Locus, snp) %>%
  filter(any(PC=="Expression.Splicing.Subset_YRI")) %>%
  ungroup() %>%
  group_by(Locus, PC) %>%
  summarise(ContainsTopMolQTL = any(TopMolQTLAroundGene)) %>%
  ungroup() %>%
  group_by(PC) %>%
  summarise(FracContainsTop = sum(ContainsTopMolQTL)/n()) %>%
  ungroup()

kable(dat.FractionNonPrimaryMolQTLColocs)


P.NonPrimaryMolQTLs_OftenColoc <- dat.FractionNonPrimaryMolQTLColocs %>%
  filter(PC %in% c("H3K27AC", "polyA.Splicing")) %>%
  mutate(Complement = 1-FracContainsTop) %>%
  gather("ContainsTop", "Fraction",-PC) %>%
  ggplot(aes(x=PC, y=Fraction, fill=ContainsTop)) +
  geom_col(position="fill") +
  scale_y_continuous(limits=c(0, 1), expand=c(0,0)) +
  scale_fill_manual(values=c("FracContainsTop"="#636363", "Complement"="#f0f0f0"), labels=c("eGene colocalizes with top molQTL", "colocalizes with non top molQTL")) +
  scale_x_discrete(labels=c("H3K27AC hQTL", "polyA RNA sQTL")) +
  labs(x=NULL, y=str_wrap("Fraction of eQTL/molQTL colocalizations", 30), fill=NULL) +
  Rotate_x_labels

P.NonPrimaryMolQTLs_OftenColoc
```


- Number of eQTLs that coloc with at least one molQTL

```{r}
eGenes <- dat %>%
  filter(PC1=="Expression.Splicing.Subset_YRI" & FDR.x < 0.1) %>%
  pull(P1) %>% unique()

coloc.dat$PC %>% unique()


coloc.dat %>%
    group_by(Locus, snp) %>%
    filter(any(PC=="Expression.Splicing.Subset_YRI")) %>%
    ungroup() %>%
    count(Locus, PC) %>%
    count(PC) %>%
  mutate(PC_relabeled = recode(PC, !!!MolQTL_Labels)) %>%
  filter(!is.na(PC_relabeled) | PC=="Any hQTL") %>%
  filter(!PC_relabeled %in% c("Splicing", "Expression; eQTLs (YRI only)")) %>%
  mutate(Assay_relabeled = recode(PC, !!!MolQTL_AssayLabels)) %>%
  mutate(FinalLabels = paste(PC_relabeled, Assay_relabeled, sep="\n")) %>%
  mutate(OrderFactor = recode(PC, !!!Factor_Order)) %>%
  mutate(FinalLabels = fct_reorder(FinalLabels, OrderFactor)) %>%
  ggplot(aes(x=FinalLabels, y=n)) +
  geom_col() +
  scale_y_continuous(
    name="Number eGenes with colocalized molQTL", 
    sec.axis = sec_axis(~ . / length(eGenes), name = "% eGenes with colocalized molQTL")
  ) +
  Rotate_x_labels +
  labs(x="Molecular QTL type, Assay")


dat.eGeneColocsWith <- bind_rows(
  coloc.dat %>%
    group_by(Locus, snp) %>%
    filter(any(PC=="Expression.Splicing.Subset_YRI")) %>%
    ungroup() %>%
    count(Locus, PC) %>%
    count(PC),
  coloc.dat %>%
    mutate(PC = case_when(
      PC %in% c("H3K4ME1", "H3K4ME3", "H3K27AC", "H3K36ME3") ~ "Any hQTL",
      TRUE ~ PC
    )) %>%
    group_by(Locus, snp) %>%
    filter(any(PC=="Expression.Splicing.Subset_YRI")) %>%
    ungroup() %>%
    count(Locus, PC) %>%
    count(PC)
) %>%
  mutate(PC_relabeled = recode(PC, !!!MolQTL_Labels)) %>%
  filter(!is.na(PC_relabeled) | PC=="Any hQTL") %>%
  filter(!PC_relabeled %in% c("Splicing", "Expression; eQTLs (YRI only)")) %>%
  mutate(Assay_relabeled = recode(PC, !!!MolQTL_AssayLabels)) %>%
  mutate(FinalLabels = paste(PC_relabeled, Assay_relabeled, sep="\n")) %>%
  mutate(OrderFactor = recode(PC, !!!Factor_Order))

P.eGeneColocsWith <- dat.eGeneColocsWith %>%
  mutate(FinalLabels = str_remove_all(FinalLabels, "\\*")) %>%
  mutate(FinalLabels = case_when(
    FinalLabels == "Any hQTL\nAny hQTL" ~ "Any hQTL*",
    TRUE ~ FinalLabels
  )) %>%
  mutate(OrderFactor = case_when(
    FinalLabels == "Any hQTL*" ~ 4.5,
    TRUE ~ OrderFactor
  )) %>%
  mutate(FinalLabels = fct_reorder(FinalLabels, OrderFactor)) %>%
  distinct(.keep_all=T) %>%
  ggplot(aes(x=FinalLabels, y=n)) +
  geom_col() +
  scale_y_continuous(
    name="Number eGenes with colocalized molQTL", 
    sec.axis = sec_axis(~ . / length(eGenes), name = "% eGenes with colocalized molQTL")
  ) +
  Rotate_x_labels +
  labs(x="Molecular QTL type, Assay")
P.eGeneColocsWith

dat.eGeneColocsWith
```


- Number of eQTLs that coloc with at least one molQTL (among non hQTL eQTLs)

```{r}
eGenes.NohQTLColoc <-
  coloc.dat %>%
  group_by(Locus, snp) %>%
  filter(any(PC=="Expression.Splicing.Subset_YRI") & !any(PC %in% c("H3K4ME3", "H3K27AC", "H3K4ME1", "H3K36ME3"))) %>%
  pull(Locus) %>% unique()

dat.eGeneColocsWith.NohQTL <-
  coloc.dat %>%
  group_by(Locus, snp) %>%
  filter(any(PC=="Expression.Splicing.Subset_YRI") & !any(PC %in% c("H3K4ME3", "H3K27AC", "H3K4ME1", "H3K36ME3"))) %>%
  ungroup() %>%
  count(Locus, PC) %>%
  count(PC) %>%
  mutate(PC_relabeled = recode(PC, !!!MolQTL_Labels)) %>%
  filter(!is.na(PC_relabeled) | PC=="Any hQTL") %>%
  filter(!PC_relabeled %in% c("Splicing", "Expression; eQTLs (YRI only)")) %>%
  mutate(Assay_relabeled = recode(PC, !!!MolQTL_AssayLabels)) %>%
  mutate(FinalLabels = paste(PC_relabeled, Assay_relabeled, sep="\n")) %>%
  mutate(OrderFactor = recode(PC, !!!Factor_Order))

P.eGeneColocsWith.nohQTL <- dat.eGeneColocsWith.NohQTL %>%
  mutate(FinalLabels = str_remove_all(FinalLabels, "\\*")) %>%
  mutate(FinalLabels = case_when(
    FinalLabels == "Any hQTL\nAny hQTL" ~ "Any hQTL*",
    TRUE ~ FinalLabels
  )) %>%
  mutate(OrderFactor = case_when(
    FinalLabels == "Any hQTL*" ~ 4.5,
    TRUE ~ OrderFactor
  )) %>%
  distinct(.keep_all=T) %>%
  mutate(FinalLabels = fct_reorder(FinalLabels, OrderFactor)) %>%
  ggplot(aes(x=FinalLabels, y=n)) +
  geom_col() +
  scale_y_continuous(
    name="Number eGenes with colocalized molQTL", 
    sec.axis = sec_axis(~ . / length(eGenes.NohQTLColoc), name = "% eGenes with colocalized molQTL\nAmong those that did not coloc with hQTL")
  ) +
  Rotate_x_labels +
  labs(x="Molecular QTL type, Assay")
P.eGeneColocsWith.nohQTL

dat.eGeneColocsWith.NohQTL


```


- Plot of number of eQTLs that coloc with an hQTL, versus those that coloc with an sQTL (and not an hQTL)

```{r}




data.frame(name="Non hQTL", n=length(eGenes.NohQTLColoc))


dat.SimplifiedNumberColocs <- bind_rows(
  dat.eGeneColocsWith %>%
    filter(PC=="Any hQTL") %>%
    mutate(name="hQTL*", order=1) %>%
    dplyr::select(name, n, order),
  # dat.eGeneColocsWith %>%
  #   filter(PC=="polyA.Splicing") %>%
  #   mutate(name="sQTL", order=1.5) %>%
  #   dplyr::select(name, n, order),
  dat.eGeneColocsWith.NohQTL %>%
    filter(PC=="APA_Total") %>%
    mutate(name="APA QTL**", order=4) %>%
    dplyr::select(name, n, order),
  dat.eGeneColocsWith.NohQTL %>%
    filter(PC=="polyA.Splicing") %>%
    mutate(name="sQTL**", order=3) %>%
    dplyr::select(name, n, order),
) %>%
  add_row(name="Any molQTL**", n=length(eGenes.NohQTLColoc), order=2) %>%
  mutate(name = fct_reorder(name, order))

P.SimplifiedNumberColocs <- ggplot(dat.SimplifiedNumberColocs, aes(x=name, y=n)) +
  geom_col(fill="#bdbdbd") +
  scale_y_continuous(expand=c(0,0), breaks=c(0, 200, 400, 600, 800), labels=c(0, "", "", "", 800)) +
  geom_text(aes(label=name), y=10, angle=90, y=-Inf, hjust=0, size=8) +
  labs(x=NULL, y="# eGenes coloc w/ molQTL") +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())
P.SimplifiedNumberColocs

kable(dat.SimplifiedNumberColocs)
```


- And finally, a plot about whether the sQTLs that do coloc with eQTL are enriched for non-productive juncs.
```{r}
IntronAnnotations <- read_tsv("../data/IntronAnnotationsFromYang.tsv.gz")

ClusterLevelQTLAnnotations <- read_delim("../code/QTLs/QTLTools/polyA.Splicing/PermutationPassForColoc.txt.gz", delim=' ') %>%
  mutate(IntronID = str_replace(phe_id, "(^.+?clu.+?):EN.+$", "\\1")) %>%
  separate(IntronID, into=c("chrom", "start", "end", "cluster"), remove=F, convert=T, sep=":") %>%
  mutate(chrom=paste0("chr", chrom)) %>%
  mutate(q = qvalue(adj_beta_pval)$qvalues) %>%
  left_join(IntronAnnotations) %>%
  group_by(cluster) %>%
  mutate(ContainsUnproductive = any(SuperAnnotation %in% c("AnnotatedJunc_UnproductiveCodingGene", "UnannotatedJunc_UnproductiveCodingGene") & q<0.1)) %>%
  ungroup()

ClusterLevelQTLAnnotations %>%
  distinct(cluster, .keep_all=T) %>%
  count(ContainsUnproductive)

dat %>%
  filter(PC1=="Expression.Splicing" & FDR.x < 0.1) %>%
  filter(PC2=="polyA.Splicing" & p_permutation.y<0.01)


Clusters.ThatColocWith_sQTL_not_hQTL <- coloc.dat %>%
  group_by(Locus, snp) %>%
  filter(any(PC=="Expression.Splicing.Subset_YRI") & !any(PC %in% c("H3K4ME3", "H3K27AC", "H3K4ME1"))) %>%
  ungroup() %>%
  filter(PC=="polyA.Splicing") %>%
  separate(P, into=c("chrom", "start", "end", "cluster"), remove=F, convert=T, sep=":") %>%
  distinct(cluster)

Clusters.AttemptColoc_butOther <- dat %>%
  filter(PC1=="Expression.Splicing" & FDR.x < 0.1) %>%
  filter(PC2=="polyA.Splicing" & p_permutation.y<0.01) %>%
  distinct(P2) %>%
  separate(P2, into=c("chrom", "start", "end", "cluster"), remove=F, convert=T, sep=":") %>%
  distinct(cluster) %>%
  filter(!cluster %in% Clusters.ThatColocWith_sQTL_not_hQTL$cluster)

dat.EnrichmentOfUnrpductiveClusters <- bind_rows(
  Clusters.AttemptColoc_butOther %>%
    mutate(Category="DoesNotColocWith_eGene"),
  Clusters.ThatColocWith_sQTL_not_hQTL %>%
    mutate(Category="DoesColocWith_eGene")
) %>%
  left_join(
    ClusterLevelQTLAnnotations %>%
      distinct(cluster, ContainsUnproductive))

test.results <- dat.EnrichmentOfUnrpductiveClusters %>%
  count(Category, ContainsUnproductive) %>%
  pivot_wider(names_from="Category", values_from ="n") %>%
  mutate(ContainsUnproductive=!ContainsUnproductive) %>%
  column_to_rownames("ContainsUnproductive") %>%
  as.matrix() %>%
  fisher.test(alternative="less")
test.results

P.EnrichmentOfUnrpductiveClusters <- dat.EnrichmentOfUnrpductiveClusters %>%
  count(Category, ContainsUnproductive) %>%
  ggplot(aes(x=Category, fill=ContainsUnproductive, y=n)) +
  geom_col() +
  scale_fill_manual(values=c("FALSE"="#377eb8", "TRUE"="#e41a1c"), labels=c("FALSE"="Productive sQTL", "TRUE"="Unproductive sQTL")) +
  scale_x_discrete(labels=c(str_wrap("sQTLs that coloc with eQTL and not hQTL", 20), str_wrap("Other sQTLs tested\nfor colocalization", 20))) +
  labs(caption=paste("One sided Fisher test P:", format.pval(test.results$p.value, 3), "\nOR:", round(1/test.results$estimate, 3)), y="Number of sQTLs", fill="sQTL type", x="") +
  Rotate_x_labels
P.EnrichmentOfUnrpductiveClusters
```

Plot about primary vs non primary sQTLs for productive vs unproductive

```{r}

dat.EnrichmentOfUnrpductiveClusters_AmongstTop <- dat %>%
  filter(PC1=="Expression.Splicing" & FDR.x < 0.1) %>%
  filter(PC2=="polyA.Splicing" & p_permutation.y<0.01) %>%
  separate(P2, into=c("chrom", "start", "end", "cluster"), remove=F, convert=T, sep=":") %>%
  group_by(cluster, GeneLocus) %>%
  summarise(cluster.minP=min(p_permutation.y)) %>%
  ungroup() %>%
  group_by(GeneLocus) %>%
  mutate(Category = cluster.minP==min(cluster.minP)) %>%
  ungroup() %>%
  left_join(
    ClusterLevelQTLAnnotations %>%
      distinct(cluster, ContainsUnproductive))

test.results <- dat.EnrichmentOfUnrpductiveClusters_AmongstTop %>%
  count(Category, ContainsUnproductive) %>%
  pivot_wider(names_from="Category", values_from ="n") %>%
  mutate(ContainsUnproductive=!ContainsUnproductive) %>%
  column_to_rownames("ContainsUnproductive") %>%
  as.matrix() %>%
  fisher.test(alternative="less")
test.results

P.EnrichmentOfUnrpductiveClusters_AmongstTop <- dat.EnrichmentOfUnrpductiveClusters_AmongstTop %>%
  count(Category, ContainsUnproductive) %>%
  ggplot(aes(x=Category, fill=ContainsUnproductive, y=n)) +
  geom_col() +
  scale_fill_manual(values=c("FALSE"="#377eb8", "TRUE"="#e41a1c"), labels=c("FALSE"="Productive sQTL", "TRUE"="Unproductive sQTL")) +
  scale_x_discrete(labels=c("TRUE"=str_wrap("Top sQTL per gene", 20), "FALSE"=str_wrap("Not top sQTL per gene", 20))) +
  labs(caption=paste("One sided Fisher test P:", format.pval(test.results$p.value, 3), "\nOR:", round(1/test.results$estimate, 3)), y="Number of sQTLs", fill="sQTL type", x="") +
  Rotate_x_labels
P.EnrichmentOfUnrpductiveClusters_AmongstTop
```

sQTLs enriched for eQTL signal, qq plot

```{r}
PC2.filter <- c("Expression.Splicing", "Expression.Splicing.Subset_YRI")
PC2.SignificanceFilter <- c("H3K27AC", "H3K4ME3", "H3K36ME3")
PC1.filter <- c("polyA.Splicing", "H3K27AC", "H3K4ME1", "H3K4ME3")
PC1.filter.Splicing <- PC1.filter[str_detect(PC1.filter, "Splicing")]
PC1.filter.NonSplicing <- PC1.filter[!str_detect(PC1.filter, "Splicing")]

dat.foreQTLQQ <- fread("../code/pi1/PairwisePi1Traits.P.all.txt.gz") %>%
  filter((PC1 %in% PC1.filter))

IntronAnnotatins <- read_tsv("../data/IntronAnnotationsFromYang.tsv.gz") %>%
  mutate(chrom = str_remove_all(chrom, "chr")) %>%
  mutate(Intron = paste(chrom, start, end, sep=":")) %>%
  filter(!str_detect(SuperAnnotation, "NoncodingGene"))

IntronAnnotatins$SuperAnnotation %>% unique()



RandomTestSNPs <- paste0("../code/QTLs/QTLTools/", PC2.filter, "/NominalPassForColoc.RandomSamplePvals.txt.gz") %>%
  setNames(PC2.filter) %>%
  lapply(fread) %>%
  bind_rows(.id="PC2") %>%
  dplyr::select(trait.x.p.in.y=V1, PC2) %>%
  mutate(SNP_group="Random test SNPs") %>%
  group_by(PC2) %>%
  sample_n(50000)


dat.foreQTLQQ.sQTLs <- dat.foreQTLQQ %>%
    filter(PC1 %in% PC1.filter.Splicing) %>%
    group_by(PC1, P1) %>%
    filter(!any((PC2 %in% PC2.SignificanceFilter) & (trait.x.p.in.y < 0.01))) %>%
    ungroup() %>%
    filter(PC2 %in% PC2.filter) %>%
    separate(P1, into=c("Intron", "Cluster"), sep=":clu", remove=F) %>%
    inner_join(
      IntronAnnotatins %>%
        dplyr::select(Intron, SuperAnnotation),
      by="Intron") %>%
    group_by(PC1, PC2, Cluster) %>%
    mutate(SNP_group = case_when(
      all(str_detect(SuperAnnotation, "Productive")) ~ "Productive sQTL cluster",
      any(str_detect(SuperAnnotation, "Unproductive")) ~ "Unproductive sQTL cluster",
      TRUE ~ "sQTL Other"
    )) %>%
    # sample_n(1) %>%
    slice(which.min(p_permutation.x)) %>%
    ungroup()

bind_rows(
  dat.foreQTLQQ.sQTLs,
  dat.foreQTLQQ %>%
    filter(PC2 %in% PC2.filter) %>%
    filter(PC1 %in% PC1.filter.NonSplicing) %>%
    mutate(SNP_group = PC1),
  RandomTestSNPs
) %>%
  group_by(SNP_group, PC2) %>%
  mutate(MyRank = rank(trait.x.p.in.y, ties.method='random')) %>%
  mutate(ExpectedP = MyRank/(max(MyRank) + 1)) %>%
  ungroup() %>%
  ggplot(aes(x=-log10(ExpectedP), y=-log10(trait.x.p.in.y), color=SNP_group)) +
  geom_abline(slope=1, intercept=0) +
  geom_point() +
  facet_wrap(~PC2, scales="free") +
  labs(y="-log10(ObservedP)", caption="sQTL clusters containing any significant unproductive sQTL intron are grouped as Unproductive sQTL\nP value from only one significant sQTL intron from cluster selected at random from cluster\nsQTLs with nominal P<0.01 for H3K36me3, H3K27Ac, or H3K4me3 removed", title="eQTL QQ plot")


dat.foreQTLQQ.sQTLs %>%
  count(SNP_group, PC2)

# pi0est function breaks sometimes. this wrapper function returns 1 when that happens
CalculatePi1_2 <- function (dat.in, ...) {
  return(tryCatch(1-pi0est(dat.in$trait.x.p.in.y, ...)$pi0, error=function(e) 1))
}

labels <- dat.foreQTLQQ.sQTLs %>%
  split(paste(.$SNP_group, .$PC2, sep=";")) %>%
  lapply(CalculatePi1_2, pi0.method="bootstrap") %>%
  unlist() %>%
  data.frame(pi1=.) %>%
  rownames_to_column("sQTLGroup_eQTLDataset") %>%
  separate(sQTLGroup_eQTLDataset, into=c("SNP_group", "PC2"), sep=";") %>%
  mutate(label=paste0("pi=", signif(pi1, digits=3)))

  
dat.foreQTLQQ.sQTLs %>%
  ggplot(aes(x=trait.x.p.in.y)) +
  geom_histogram() +
  geom_text(data = labels, aes(label=label), x=Inf, y=Inf, vjust=1, hjust=1) +
  facet_grid(SNP_group ~ PC2) +
  labs(x="P", title="eQTL P value historgram", caption="Vertical facets: eQTL dataset\nHorizontal facets: sQTL cluster type") +
  theme(strip.text = element_text(size = 8))

```

save a qqplot For lab meeting...

```{r}
eQTL.QQ <- bind_rows(
  dat.foreQTLQQ.sQTLs,
  dat.foreQTLQQ %>%
    filter(PC2 %in% PC2.filter) %>%
    filter(PC1 %in% PC1.filter.NonSplicing) %>%
    mutate(SNP_group = PC1),
  RandomTestSNPs
) %>%
  group_by(SNP_group, PC2) %>%
  mutate(MyRank = rank(trait.x.p.in.y, ties.method='random')) %>%
  mutate(ExpectedP = MyRank/(max(MyRank) + 1)) %>%
  ungroup() %>%
  filter(PC2=="Expression.Splicing") %>%
  filter(!PC1 %in% c("H3K4ME1", "H3K4ME3")) %>%
  ggplot(aes(x=-log10(ExpectedP), y=-log10(trait.x.p.in.y), color=SNP_group)) +
  geom_abline(slope=1, intercept=0) +
  geom_point() +
  scale_color_manual(values=
                       c("Random test SNPs"="#000000", "Unproductive sQTL cluster"="#e31a1c", "Productive sQTL cluster"="#1f78b4", "H3K27AC"="#6a3d9a"),
                     labels=c("Random test SNPs"="Random test SNPs", "Unproductive sQTL cluster"="Unproductive sQTL", "Productive sQTL cluster"="Productive sQTL", "H3K27AC"="H3K27AC QTL w/in 100kb")) +
  labs(y="-log10(ObservedP)", caption="sQTL clusters containing any significant unproductive sQTL intron are grouped as Unproductive sQTL\nP value from only one significant sQTL intron from cluster selected at random from cluster\nsQTLs with nominal P<0.01 for H3K36me3, H3K27Ac, or H3K4me3 removed", title="eQTL QQ plot") +
  theme(plot.subtitle=element_text(size=5))
eQTL.QQ



```

beta beta scatters

```{r}
PC1.PossibleValues <- c("polyA.Splicing", "polyA.Splicing.Subset_YRI", "chRNA.Splicing")
PC2.PossibleValues <-c("Expression.Splicing", "Expression.Splicing.Subset_YRI", "H3K4ME3", "H3K36ME3", "H3K27AC", "H3K4ME1", "MetabolicLabelled.30min", "MetabolicLabelled.60min", "chRNA.Expression.Splicing")


dat <- fread("../code/pi1/PairwisePi1Traits.P.all.txt.gz") %>%
  filter((PC1 %in% PC1.PossibleValues) & (PC2 %in% PC2.PossibleValues))

Intron.Annotations <- read_tsv("../data/IntronAnnotationsFromYang.tsv.gz") %>%
  mutate(IntronName = paste(chrom, start, end, strand, sep=":"))

dat.sQTLs.eQTLs.ForScatter <- dat %>%
  mutate(IntronName = str_replace(P1, "^(.+?:)clu_.+?([+-])$", "chr\\1\\2")) %>%
  mutate(ClusterName =  str_replace(P1, "^(.+?:).+?(clu_.+?[+-])$", "chr\\1\\2")) %>%
  left_join(Intron.Annotations)

PC1.filter = c("polyA.Splicing")
PC2.filter = c("H3K36ME3", "chRNA.Expression.Splicing" , "MetabolicLabelled.30min", "MetabolicLabelled.60min", "Expression.Splicing", "Expression.Splicing.Subset_YRI")
PC2.SignificanceFilter <- c("H3K4ME3", "H3K27AC", "H3K36ME3")

Intron.Annotations$SuperAnnotation %>% unique()

dat.sQTLs.eQTLs.ForScatter.ToPlot <- dat.sQTLs.eQTLs.ForScatter %>%
  filter(PC1 %in% PC1.filter) %>%
  group_by(P1) %>%
  filter(!any((PC2 %in% PC2.SignificanceFilter) & (trait.x.p.in.y < 0.01))) %>%
  ungroup() %>%
  filter(PC2 %in% PC2.filter) %>%
  # pull(PC2) %>% unique()
  filter(SuperAnnotation %in% c("UnannotatedJunc_UnproductiveCodingGene", "AnnotatedJunc_ProductiveCodingGene", "AnnotatedJunc_UnproductiveCodingGene", "UnannotatedJunc_ProductiveCodingGene")) %>%
  group_by(PC1, ClusterName) %>%
  filter(abs(beta.x) == max(abs(beta.x))) %>%
  ungroup() %>%
  mutate(SuperAnnotation = str_replace_all(SuperAnnotation, "_", " ")) %>%
  filter(GeneLocus == gene) %>%
  filter(FDR.x < 0.1) %>%
  group_by(P2, PC2, SuperAnnotation) %>%
  sample_n(1) %>%
  ungroup() %>%
  mutate(PC2 = factor(PC2, levels=PC2.filter)) %>%
  mutate(PC2 = recode(PC2, !!!c("chRNA.Expression.Splicing"="chRNA" , "MetabolicLabelled.30min"="30min 4sU", "MetabolicLabelled.60min"="60min 4sU", "Expression.Splicing"="polyA", "Expression.Splicing.Subset_YRI"="polyA YRI"))) %>%
  mutate(PC2 = factor(PC2, levels=c("H3K36ME3","chRNA", "30min 4sU", "60min 4sU", "polyA", "polyA YRI"))) %>%
  filter(!PC2 %in% c("polyA YRI", "H3K36ME3")) %>%
  # pull(SuperAnnotation) %>% unique()
  mutate(SuperAnnotation = recode(SuperAnnotation, !!!c("AnnotatedJunc UnproductiveCodingGene"="Unproductive", "UnannotatedJunc UnproductiveCodingGene"="Unproductive", "AnnotatedJunc ProductiveCodingGene"="Productive", "UnannotatedJunc ProductiveCodingGene"="Productive")))

dat.sQTLs.eQTLs.ForScatter.ToPlot.labels <- dat.sQTLs.eQTLs.ForScatter.ToPlot %>%
  count(SuperAnnotation, PC2) %>%
  distinct(SuperAnnotation, .keep_all=T) %>%
  dplyr::select(n, SuperAnnotation) %>%
  mutate(SuperAnnotationLabel = paste0(SuperAnnotation, " (n=", n, ")"))

library(broom)

sQTL.eQTL.beta.beta <- dat.sQTLs.eQTLs.ForScatter.ToPlot %>%
  left_join(dat.sQTLs.eQTLs.ForScatter.ToPlot.labels, by="SuperAnnotation") %>%
  nest(-SuperAnnotationLabel, -PC2) %>% 
  mutate(cor=map(data,~cor.test(.x$beta.x, .x$x.beta.in.y, method = "sp"))) %>%
  mutate(tidied = map(cor, tidy)) %>% 
  unnest(tidied, .drop = T) %>%
  unnest(data) %>%
ggplot(aes(x=beta.x, y=x.beta.in.y, color=SuperAnnotation)) +
    geom_point(alpha=0.1) +
    # geom_smooth(method = "tls", se = FALSE, color = "red", method='bootstrap') +
    # geom_smooth(method='lm', color='black') +
    geom_vline(xintercept=0, linetype='dashed') +
    geom_hline(yintercept=0, linetype='dashed') +
    scale_color_manual(values=c("Productive"="#1f78b4", "Unproductive"="#e31a1c")) +
    geom_abline(data = . %>%
                  distinct(SuperAnnotationLabel, PC2, .keep_all=T),
                aes(slope=estimate, intercept=0), size=2) +
    geom_text(
    data = . %>%
        distinct(SuperAnnotationLabel, PC2, .keep_all=T) %>%
        mutate(R=signif(estimate, 3), P=format.pval(p.value, 3)) %>%
      mutate(label = str_glue("rho:{R}\nP:{P}")),
    aes(x=-Inf, y=-Inf, label=label),
    hjust=-.1, vjust=-0.1, color='black', size=6
    ) +
    facet_grid(SuperAnnotationLabel ~ PC2, labeller = label_wrap_gen(10)) +
    theme(strip.text = element_text(size = 12), legend.position='none') +
  labs(caption = "FDR<10% sQTLs. sQTLs with nominal P<0.01 for H3K36me3, H3K27Ac, or H3K4me3 removed\nOnly top sQTL per cluster plotted", y="host gene eQTL beta", x="sQTL beta")

sQTL.eQTL.beta.beta

```

Again but classify cluster as unproductive if any are unproductive

```{r}
dat.sQTLs.eQTLs.ForScatter.ToPlot <- dat.sQTLs.eQTLs.ForScatter %>%
  filter(PC1 %in% PC1.filter) %>%
  group_by(P1) %>%
  filter(!any((PC2 %in% PC2.SignificanceFilter) & (trait.x.p.in.y < 0.01))) %>%
  ungroup() %>%
  filter(PC2 %in% PC2.filter) %>%
  # pull(PC2) %>% unique()
  filter(SuperAnnotation %in% c("UnannotatedJunc_UnproductiveCodingGene", "AnnotatedJunc_ProductiveCodingGene", "AnnotatedJunc_UnproductiveCodingGene", "UnannotatedJunc_ProductiveCodingGene")) %>%
  group_by(PC1, ClusterName) %>%
  mutate(SuperAnnotation = case_when(
    any(SuperAnnotation %in% c("UnannotatedJunc_UnproductiveCodingGene", "AnnotatedJunc_UnproductiveCodingGene")) & SuperAnnotation %in% c("UnannotatedJunc_UnproductiveCodingGene", "AnnotatedJunc_UnproductiveCodingGene") ~ "Unproductive",
    all(SuperAnnotation %in% c("UnannotatedJunc_ProductiveCodingGene", "AnnotatedJunc_ProductiveCodingGene")) & SuperAnnotation %in% c("UnannotatedJunc_ProductiveCodingGene", "AnnotatedJunc_ProductiveCodingGene") ~ "Productive",
    TRUE ~ "Productive in unproductive cluster or vice versa"
    )) %>%
  filter(abs(beta.x) == max(abs(beta.x))) %>%
  ungroup() %>%
  mutate(SuperAnnotation = str_replace_all(SuperAnnotation, "_", " ")) %>%
  filter(GeneLocus == gene) %>%
  filter(FDR.x < 0.1) %>%
  group_by(P2, PC2, SuperAnnotation) %>%
  sample_n(1) %>%
  ungroup() %>%
  mutate(PC2 = factor(PC2, levels=PC2.filter)) %>%
  mutate(PC2 = recode(PC2, !!!c("chRNA.Expression.Splicing"="chRNA" , "MetabolicLabelled.30min"="30min 4sU", "MetabolicLabelled.60min"="60min 4sU", "Expression.Splicing"="polyA", "Expression.Splicing.Subset_YRI"="polyA YRI"))) %>%
  mutate(PC2 = factor(PC2, levels=c("H3K36ME3","chRNA", "30min 4sU", "60min 4sU", "polyA", "polyA YRI"))) %>%
  filter(!PC2 %in% c("polyA YRI", "H3K36ME3")) %>%
  # pull(SuperAnnotation) %>% unique()
  mutate(SuperAnnotation = recode(SuperAnnotation, !!!c("AnnotatedJunc UnproductiveCodingGene"="Unproductive", "UnannotatedJunc UnproductiveCodingGene"="Unproductive", "AnnotatedJunc ProductiveCodingGene"="Productive", "UnannotatedJunc ProductiveCodingGene"="Productive"))) %>%
  filter(SuperAnnotation %in% c("Productive", "Unproductive"))

dat.sQTLs.eQTLs.ForScatter.ToPlot.labels <- dat.sQTLs.eQTLs.ForScatter.ToPlot %>%
  count(SuperAnnotation, PC2) %>%
  distinct(SuperAnnotation, .keep_all=T) %>%
  dplyr::select(n, SuperAnnotation) %>%
  mutate(SuperAnnotationLabel = paste0(SuperAnnotation, " (n=", n, ")"))

library(broom)

sQTL.eQTL.beta.beta <- dat.sQTLs.eQTLs.ForScatter.ToPlot %>%
  left_join(dat.sQTLs.eQTLs.ForScatter.ToPlot.labels, by="SuperAnnotation") %>%
  nest(-SuperAnnotationLabel, -PC2) %>% 
  mutate(cor=map(data,~cor.test(.x$beta.x, .x$x.beta.in.y, method = "sp"))) %>%
  mutate(tidied = map(cor, tidy)) %>% 
  unnest(tidied, .drop = T) %>%
  unnest(data) %>%
ggplot(aes(x=beta.x, y=x.beta.in.y, color=SuperAnnotation)) +
    geom_point(alpha=0.1) +
    # geom_smooth(method = "tls", se = FALSE, color = "red", method='bootstrap') +
    # geom_smooth(method='lm', color='black') +
    geom_vline(xintercept=0, linetype='dashed') +
    geom_hline(yintercept=0, linetype='dashed') +
    scale_color_manual(values=c("Productive"="#1f78b4", "Unproductive"="#e31a1c")) +
    geom_abline(data = . %>%
                  distinct(SuperAnnotationLabel, PC2, .keep_all=T),
                aes(slope=estimate, intercept=0), size=2) +
    geom_text(
    data = . %>%
        distinct(SuperAnnotationLabel, PC2, .keep_all=T) %>%
        mutate(R=signif(estimate, 3), P=format.pval(p.value, 3)) %>%
      mutate(label = str_glue("rho:{R}\nP:{P}")),
    aes(x=-Inf, y=-Inf, label=label),
    hjust=-.1, vjust=-0.1, color='black', size=6
    ) +
    facet_grid(SuperAnnotationLabel ~ PC2, labeller = label_wrap_gen(10)) +
    theme(strip.text = element_text(size = 12), legend.position='none') +
  labs(caption = "FDR<10% sQTLs. sQTLs with nominal P<0.01 for H3K36me3, H3K27Ac, or H3K4me3 removed\nOnly top sQTL per cluster plotted", y="host gene eQTL beta", x="sQTL beta")

sQTL.eQTL.beta.beta

```



hQTL eQTL beta

```{r}

PC1.filter <- c("Expression.Splicing", "MetabolicLabelled.30min", "MetabolicLabelled.60min", "H3K36ME3", "H3K27AC", "H3K4ME3", "H3K4ME1", "Expression.Splicing.Subset_YRI", "chRNA.Expression.Splicing")
dat <- fread("../code/pi1/PairwisePi1Traits.P.all.txt.gz") %>%
  filter((PC1 %in% PC1.filter) & (PC2 %in% PC1.filter))

PeaksToTSS <- Sys.glob("../code/Misc/PeaksClosestToTSS/*_assigned.tsv.gz") %>%
  setNames(str_replace(., "../code/Misc/PeaksClosestToTSS/(.+?)_assigned.tsv.gz", "\\1")) %>%
  lapply(read_tsv) %>%
  bind_rows(.id="ChromatinMark") %>%
  mutate(GenePeakPair = paste(gene, peak, sep = ";")) %>%
  distinct(ChromatinMark, peak, gene, .keep_all=T)

#purple is #6a3d9a

FDR.cutoff <- 0.1
ActivatingMarksToConsider <- c("H3K27AC", "H3K4ME3")
Recode.PCs = c("ActivatingMark_HostGenePromoter"="Promoter marks*", "H3K36ME3"="H3K36ME3", "chRNA.Expression.Splicing"="chRNA", "MetabolicLabelled.30min"="4sU 30min", "MetabolicLabelled.60min"="4sU 60min", "Expression.Splicing.Subset_YRI"="polyA YRI", "Expression.Splicing"="polyA")


dat.toPlot <- dat %>%
  filter(FDR.x < FDR.cutoff) %>%
  mutate(PC1 = case_when(
    paste(GeneLocus, P1, sep=";") %in% PeaksToTSS$GenePeakPair & PC1 %in% ActivatingMarksToConsider ~ PC1,
    P1 %in% PeaksToTSS$peak ~ "ActivatingMark_OtherGenePromoter",
    PC1 %in% c("H3K4ME1", "H3K4ME3", "H3K27AC") ~ "ActivatingMark_Enhancer",
    TRUE ~ PC1
  )) %>%
  filter(PC1 %in% ActivatingMarksToConsider) %>%
  filter(PC2 %in% c("chRNA.Expression.Splicing", "Expression.Splicing", "MetabolicLabelled.30min", "MetabolicLabelled.60min")) %>%
  mutate(PC2 = recode(PC2, !!!Recode.PCs)) %>%
  mutate(PC2 = factor(PC2, levels=c("chRNA", "4sU 30min", "4sU 60min", "polyA"))) %>%
  mutate(SuperAnnotationLabel = PC1) %>%
  nest(-SuperAnnotationLabel, -PC2) %>% 
  mutate(cor=map(data,~cor.test(.x$beta.x, .x$x.beta.in.y, method = "sp"))) %>%
  mutate(tidied = map(cor, tidy)) %>% 
  unnest(tidied, .drop = T) %>%
  unnest(data)

ggplot(dat.toPlot, aes(x=beta.x, y=x.beta.in.y)) +
  geom_point(alpha=0.1) +
    geom_vline(xintercept=0, linetype='dashed') +
    geom_hline(yintercept=0, linetype='dashed') +
    scale_color_manual(values=c("Productive"="#1f78b4", "Unproductive"="#e31a1c")) +
    geom_abline(data = . %>%
                  distinct(SuperAnnotationLabel, PC2, .keep_all=T),
                aes(slope=estimate, intercept=0), size=2) +
    geom_text(
    data = . %>%
        distinct(SuperAnnotationLabel, PC2, .keep_all=T) %>%
        mutate(R=signif(estimate, 3), P=format.pval(p.value, 3)) %>%
      mutate(label = str_glue("rho:{R}\nP:{P}")),
    aes(x=-Inf, y=Inf, label=label),
    hjust=-.1, vjust=1.1, color='black', size=6
    ) +
    facet_grid(SuperAnnotationLabel ~ PC2, labeller = label_wrap_gen(10)) +
    theme(strip.text = element_text(size = 12), legend.position='none') +
  labs(caption = "FDR<10% hQTLs at promoters.", y="eQTL beta in facet dataset", x="Promoter hQTL beta")

# ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/20230412_WIP_eQTLhQTLbetascatter.pdf", height=4, width=10)

dat.toPlot %>%
  filter(SuperAnnotationLabel == "H3K4ME3") %>%
  distinct(GeneLocus, PC2) %>%
  count(PC2)
  
eQTL.H3k27ac.beta <- dat.toPlot %>%
  filter(SuperAnnotationLabel == "H3K27AC") %>%
  mutate(SuperAnnotationLabel = "H3K27AC@TSS\n(n=1298)") %>%
ggplot(aes(x=beta.x, y=x.beta.in.y)) +
  geom_point(alpha=0.1, color="#6a3d9a") +
    geom_vline(xintercept=0, linetype='dashed') +
    geom_hline(yintercept=0, linetype='dashed') +
    geom_abline(data = . %>%
                  distinct(SuperAnnotationLabel, PC2, .keep_all=T),
                aes(slope=estimate, intercept=0), size=2) +
    geom_text(
    data = . %>%
        distinct(SuperAnnotationLabel, PC2, .keep_all=T) %>%
        mutate(R=signif(estimate, 3), P=format.pval(p.value, 3)) %>%
      mutate(label = str_glue("rho:{R}\nP:{P}")),
    aes(x=-Inf, y=Inf, label=label),
    hjust=-.1, vjust=1.1, color='black', size=6
    ) +
    facet_grid(SuperAnnotationLabel ~ PC2, labeller = label_wrap_gen(10)) +
    theme(strip.text = element_text(size = 12), legend.position='none') +
  labs(caption = "FDR<10% hQTLs at promoters.", y="eQTL beta", x="Promoter hQTL beta")

eQTL.H3k27ac.beta

dat.toPlot %>%
  filter(SuperAnnotationLabel == "H3K27AC") %>%
  distinct(GeneLocus, PC2) %>%
  count(PC2)
```


- get snps for carlos' gtex analysis:

- unproductive sQTL/eQTL SNPs.
- h3k27ac @ TSS hQTL/eQTL SNPs where lead SNP is within 10kb from promoter
- h3k27ac @ TSS hQTL/eQTL SNPs where lead SNP is > 10kb from promoter

```{r}

PC1.filter = c("polyA.Splicing")
PC2.filter = c("H3K36ME3", "chRNA.Expression.Splicing" , "MetabolicLabelled.30min", "MetabolicLabelled.60min", "Expression.Splicing", "Expression.Splicing.Subset_YRI")
PC2.SignificanceFilter <- c("H3K4ME3", "H3K27AC", "H3K36ME3")

sQTL.Nominal.hQTLs <- dat.sQTLs.eQTLs.ForScatter %>%
  filter(PC1 %in% PC1.filter) %>%
  group_by(P1) %>%
  filter(any((PC2 %in% PC2.SignificanceFilter) & (trait.x.p.in.y < 0.01))) %>%
  ungroup() %>%
  distinct(P1)


## only YRI for eQTL
# coloc.dat.fn <- "../code/hyprcoloc/Results/ForColoc/MolColocNonRedundantFullSplicing/tidy_results_OnlyColocalized.txt.gz"

## Full geuvadis for eqtl
coloc.dat.fn <- "../code/hyprcoloc/Results/ForColoc/GenewiseColoc_ChromatinAPAAndRNA/tidy_results_OnlyColocalized.txt.gz"

IntronAnnotations <- read_tsv("../data/IntronAnnotationsFromYang.tsv.gz") %>%
  unite(Intron, chrom, start, end, strand, sep=":")


dat.sQTLs.eQTLs.ForScatter.ToPlot

coloc.dat <- read_tsv(coloc.dat.fn) %>%
  separate(phenotype_full, into=c("PC", "P"), sep=";") %>%
  mutate(Intron = case_when(
    PC == "polyA.Splicing" ~ str_replace(P, "(^.+?)clu_.+?_([+-])", "chr\\1\\2"),
    TRUE ~ NA_character_
  )) %>%
  left_join(IntronAnnotations) %>%
  mutate(PC= recode(PC, "Expression.Splicing.Subset_YRI"="Expression.Splicing"))


coloc.dat$PC %>% unique()

## the sQTL/eQTL colocs, remove if top unproductive sQTL is nominal hQTL
sQTL.eQTL.colocs <- coloc.dat %>%
  group_by(snp, Locus) %>%
  filter(
    any(SuperAnnotation %in% c("AnnotatedJunc_UnproductiveCodingGene", "UnannotatedJunc_UnproductiveCodingGene")) &
    any(PC == "Expression.Splicing") 
    ) %>%
  ungroup() %>%
  filter(PC == "Expression.Splicing" | SuperAnnotation %in% c("AnnotatedJunc_UnproductiveCodingGene", "UnannotatedJunc_UnproductiveCodingGene")) %>%
  group_by(PC, snp, Locus) %>%
  filter(abs(beta) == max(abs(beta))) %>%
  ungroup() %>%
  group_by(Locus, snp) %>%
  filter(!any(P %in% sQTL.Nominal.hQTLs$P1)) %>%
  ungroup()

sQTL.eQTL.colocs %>%
  dplyr::select(snp, Locus, beta, P, PC, snp, Locus) %>%
  pivot_wider(names_from="PC", values_from=c("beta", "P")) %>%
  ggplot(aes(x=beta_polyA.Splicing, y=beta_Expression.Splicing)) +
  geom_point()
```

Ok now I also want to consider hQTLs for H3K27Ac peaks gene promoter, that are also eQTL. The top finemapped  SNP may be near the promoter (proximal hQTL/eQTL), or it may be at a distal activating element (distal hQTL/eQTL)

```{r}
sQTL.eQTL.colocs

PeaksToTSS <- Sys.glob("../code/Misc/PeaksClosestToTSS/*_assigned.tsv.gz") %>%
  setNames(str_replace(., "../code/Misc/PeaksClosestToTSS/(.+?)_assigned.tsv.gz", "\\1")) %>%
  lapply(read_tsv) %>%
  bind_rows(.id="ChromatinMark") %>%
  mutate(GenePeakPair = paste(gene, peak, sep = ";")) %>%
  distinct(ChromatinMark, peak, gene, .keep_all=T)

coloc.dat <- read_tsv(coloc.dat.fn) %>%
  separate(phenotype_full, into=c("PC", "P"), sep=";") %>%
  mutate(GenePeakPair = str_glue("{Locus};{P}")) %>%
  mutate(PC= recode(PC, "Expression.Splicing.Subset_YRI"="Expression.Splicing"))


hQTL.eQTL.colocs <- coloc.dat %>%
  filter(PC %in% c("Expression.Splicing", "H3K27AC", "H3K4ME3", "H3K4ME1")) %>%
  left_join(PeaksToTSS) %>%
  group_by(Locus, snp) %>%
  filter(any(PC=="Expression.Splicing") & any(!is.na(ChromatinMark))) %>%
  ungroup() %>%
  filter(PC=="Expression.Splicing" | peak==P)

hQTL.eQTL.colocs %>%
  distinct(Locus, snp)

hQTL.eQTL.colocs %>%
  filter(Locus %in% sQTL.eQTL.colocs$Locus) %>%
  distinct(Locus, snp)

inner_join(
  hQTL.eQTL.colocs %>%
    dplyr::select(snp, Locus, beta, P, PC) %>%
    filter(PC=="Expression.Splicing"),
  hQTL.eQTL.colocs %>%
    dplyr::select(snp, Locus, beta, P, PC, TSS_start) %>%
    filter(PC %in% c("H3K4ME3", "H3K27AC", "H3K4ME1")),
  by=c("snp", "Locus"),
  suffix=c("_Expression.Splicing", "_chromatin")
) %>%
  ggplot(aes(x=beta_chromatin, y=beta_Expression.Splicing)) +
  geom_point() +
  facet_wrap(~PC_chromatin)

```

let's also check the sQTL.eQTL colocs that are also hQTL colocs, (despite the top sQTL not being nominally significant for any hQTL, which is weird)

```{r}

sQTL.eQTL.colocs %>%
  dplyr::select(snp, Locus, beta, P, PC, snp, Locus) %>%
  pivot_wider(names_from="PC", values_from=c("beta", "P")) %>%
  mutate(Is_hQTL.Coloc = Locus %in% hQTL.eQTL.colocs$Locus) %>%
  ggplot(aes(x=beta_polyA.Splicing, y=beta_Expression.Splicing, color=Is_hQTL.Coloc)) +
  geom_point()

inner_join(
  hQTL.eQTL.colocs %>%
    dplyr::select(snp, Locus, beta, P, PC) %>%
    filter(PC=="Expression.Splicing"),
  hQTL.eQTL.colocs %>%
    dplyr::select(snp, Locus, beta, P, PC, TSS_start) %>%
    filter(PC %in% c("H3K4ME3", "H3K27AC", "H3K4ME1")),
  by=c("snp", "Locus"),
  suffix=c("_Expression.Splicing", "_chromatin")
) %>%
  mutate(Is.sQTL.Coloc = Locus %in% sQTL.eQTL.colocs$Locus) %>%
  ggplot(aes(x=beta_chromatin, y=beta_Expression.Splicing, color=Is.sQTL.Coloc)) +
  geom_point() +
  facet_wrap(~PC_chromatin)
```

Ok, as one might expect, from this I conclude that if there is a promoter hQTL, that is probably the driving mechanism over splicing.

let's filter out those hQTL.eQTL colocs from the sQTL.eQTL colocs set. Then I  will tidy up this data for Carlos.

I want the following columns in the output file:

TopSNP, TopSNPFinemapPr, eGene, eQTL_beta, molQTL_beta, molQTL_Name, molQTL_type, EffectSizeDirectionsAsExpected, Dist_from_TopSNP_to_closest_hQTL_TSS, sQTL_or_hQTL, BensClassification

From there he may choose to keep only Concordant ones, and make sure to keep only one row for each eGene, since some may be listed more than once for multiple molQTL.

```{r}
sQTL.eQTL.colocs.filtered <- sQTL.eQTL.colocs %>%
  filter(!Locus %in% hQTL.eQTL.colocs$Locus)

hQTL.eQTLs.ForCarlos <- inner_join(
  hQTL.eQTL.colocs %>%
    dplyr::select(snp, Locus, beta, P, PC, TopSNPFinemapPr, beta_se, p) %>%
    filter(PC=="Expression.Splicing"),
  hQTL.eQTL.colocs %>%
    dplyr::select(snp, Locus, beta, P, PC, TSS_start, beta_se, p) %>%
    filter(PC %in% c("H3K4ME3", "H3K27AC", "H3K4ME1")),
  by=c("snp", "Locus"),
  suffix=c("_Expression.Splicing", "_chromatin")
) %>%
  mutate(EffectSizeDirectionsAsExpected = sign(beta_Expression.Splicing)==sign(beta_chromatin)) %>%
  separate(snp, into=c("chrom", "pos", "ref", "alt"), sep=":", remove=F, convert=T) %>%
  mutate(TSS_to_SNP = abs(TSS_start - pos)) %>%
  group_by(Locus, snp) %>%
  mutate(Dist_from_TopSNP_to_closest_hQTL_TSS = min(TSS_to_SNP)) %>%
  mutate(sQTL_or_hQTL="hQTL") %>%
  mutate(BensClassification = case_when(
    Dist_from_TopSNP_to_closest_hQTL_TSS < 5000 ~ "hQTL_Proximal",
    TRUE ~ "hQTL_Distal")) %>%
  dplyr::select(TopSNP=snp, TopSNPFinemapPr, eGene=Locus, eQTL_beta=beta_Expression.Splicing, eQTL_beta_se=beta_se_Expression.Splicing, eQTL_nomP=p_Expression.Splicing, molQTL_beta=beta_chromatin, molQTL_beta_se=beta_se_chromatin, molQTL_nomP=p_chromatin, molQTL_Name=P_chromatin, molQTL_type=PC_chromatin, EffectSizeDirectionsAsExpected, Dist_from_TopSNP_to_closest_hQTL_TSS,  sQTL_or_hQTL, BensClassification)

sQTL.eQTL.colocs.ForCarlos <- sQTL.eQTL.colocs %>%
  filter(!Locus %in% hQTL.eQTL.colocs$Locus) %>%
  dplyr::select(snp, TopSNPFinemapPr, Locus, beta, P, PC, snp, Locus, beta_se, p) %>%
  pivot_wider(names_from="PC", values_from=c("beta", "P", "beta_se", "p")) %>%
  mutate(BensClassification = "sQTL", sQTL_or_hQTL="sQTL", Dist_from_TopSNP_to_closest_hQTL_TSS=NA, molQTL_type="polyA.Splicing", EffectSizeDirectionsAsExpected = !sign(beta_Expression.Splicing)==sign(beta_polyA.Splicing)) %>%
  dplyr::select(TopSNP=snp, TopSNPFinemapPr, eGene=Locus, eQTL_beta=beta_Expression.Splicing, eQTL_beta_se=beta_se_Expression.Splicing, eQTL_nomP=p_Expression.Splicing, molQTL_beta=beta_polyA.Splicing, molQTL_beta_se=beta_se_polyA.Splicing, molQTL_nomP=p_polyA.Splicing, molQTL_Name=P_polyA.Splicing, molQTL_type, EffectSizeDirectionsAsExpected, Dist_from_TopSNP_to_closest_hQTL_TSS,  sQTL_or_hQTL, BensClassification)

sQTL.vs.hQTLs.EffectOnEqtls.ForCarlos <- bind_rows(sQTL.eQTL.colocs.ForCarlos, hQTL.eQTLs.ForCarlos)

sQTL.vs.hQTLs.EffectOnEqtls.ForCarlos %>%
  distinct(eGene, .keep_all=T) %>%
  add_count(BensClassification) %>%
  mutate(Group = str_glue("{BensClassification}\n({n} eGenes)")) %>%
  ggplot(aes(x=molQTL_beta, y=eQTL_beta)) +
  geom_point() +
  facet_wrap(~Group)

sQTL.vs.hQTLs.EffectOnEqtls.ForCarlos %>%
  distinct(eGene, .keep_all=T) %>%
  add_count(BensClassification) %>%
  mutate(Group = str_glue("{BensClassification}\n({n} eGenes)")) %>%
  ggplot(aes(x=abs(eQTL_beta), color=Group)) +
  stat_ecdf() +
  labs(y="ecdf", x="abs(eQTL beta), standardized units")

sQTL.vs.hQTLs.EffectOnEqtls.ForCarlos %>%
  write_tsv("../output/eQTLs_FromSplicingVsChromatin_AcrossGTEx.ForCarlos.tsv.gz")

```


```{r, eval=F}
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_SampleSize.pdf",P.SampleSizes, width=6, height=6)

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_NumQTLs.pdf",NumQTLs, width=8, height=6)

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_PercentAnnotated.pdf",P.PercentAnnotated, width=4, height=6)
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_PercentOfUnannotated.pdf",P.PercentOfUnannotated, width=6, height=6)
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_PercentAnnotatedLegend.pdf",P.PercentJuncsLegend, width=8, height=6)


ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_Multiple_hQTLs_PerGene.pdf",P.Number_hQTLsForColoc, width=6, height=6)
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_Multiple_sQTLs_PerGene.pdf",P.Number_sQTLsForColoc, width=6, height=6)

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_NonPrimaryMolQTLsColoc.pdf",P.NonPrimaryMolQTLs_OftenColoc, width=6, height=6)

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_EnrichmentOfUnrpoductiveSqtls.pdf",P.EnrichmentOfUnrpductiveClusters, width=6, height=6)

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_EnrichmentOfUnrpoductiveSqtls_AmongstTop.pdf",P.EnrichmentOfUnrpductiveClusters_AmongstTop, width=6, height=6)


ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_eGene_MolQTL_Colocs.pdf",P.eGeneColocsWith, width=10, height=7)
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_eGene_MolQTL_NonhQTL_Colocs.pdf",P.eGeneColocsWith.nohQTL, width=10, height=7)


ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_eGene_MolQTL_Simplified.pdf",P.SimplifiedNumberColocs, width=2.7, height=2.7)

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/20230412_WIP_eQTLQQ.pdf",eQTL.QQ, height=5, width=7)

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_Final_Fig2_sQTLeQTLBetas.pdf",sQTL.eQTL.beta.beta, height=4, width=10)

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_Final_Fig2_hQTLeQTLBetas.pdf",eQTL.H3k27ac.beta, height=2.5, width=10)

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_Final_Fig_S_ScatterReasonsForJuncClassification.png",Scatter.P, height=12, width=10)



```

