---
title: "MakeFinalFigs_Fig1"
output: html_document
date: '2023-05-31'
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r setup, include=TRUE}
knitr::opts_chunk$set(echo = TRUE, warning = F, message = F)

library(tidyverse)
library(RColorBrewer)
library(data.table)
library(readxl)
library(qvalue)
library(ggrepel)
library(knitr)
library(ggbreak)



# Set theme
theme_set(
  theme_classic() +
  theme(text=element_text(size=16,  family="Helvetica")))

# I use layer a lot, to rotate long x-axis labels
Rotate_x_labels <- theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
```

Number reads per chRNA-seq dataset

```{r}

dat.reads <- read_tsv("../code/scratch/chRNA.counts.txt", col_names=c("fn", "count"))

dat.reads$count %>% sum()
  
```

Version of non-basic boxplots... similar to carlos but with jitter.

```{r}
juncs.long <- fread("../code/SplicingAnalysis/CombinedJuncTables/YRI.tsv.gz")

IntronAnnotations <- read_tsv("../data/IntronAnnotationsFromYang.tsv.gz") %>%
  mutate(stop = end-1)

juncs.long.summary <- juncs.long %>%
  dplyr::select(chrom, start, stop, strand, Dataset, IndID, Count) %>%
  inner_join(IntronAnnotations) %>%
  filter(!SuperAnnotation %in% c("AnnotatedJunc_NoncodingGene", "UnannotatedJunc_NoncodingGene")) %>%
  group_by(Dataset, IndID, SuperAnnotation) %>%
  summarise(Sum=sum(Count)) %>%
  ungroup()

P.PercentProductive <- juncs.long.summary %>%
  mutate(PlotGroup = str_detect(SuperAnnotation, "Unproductive")) %>%
  group_by(Dataset, IndID, PlotGroup) %>%
  summarise(Sum=sum(Sum)) %>%
  ungroup() %>%
  group_by(Dataset, IndID) %>%
  mutate(Percent = Sum/sum(Sum) * 100) %>%
  ungroup() %>%
  mutate(Dataset=recode(Dataset, "chRNA.Expression.Splicing"="chRNA",  "MetabolicLabelled.30min"="4sU 30min", "MetabolicLabelled.60min"="4sU 60min", "Expression.Splicing"="polyA"))  %>%
  mutate(Dataset = factor(Dataset, levels=c("chRNA", "4sU 30min", "4sU 60min","polyA"))) %>%
  ggplot(aes(x=Dataset, y=Percent, color=PlotGroup)) +
  geom_boxplot(outlier.shape = NA, position=position_identity()) +
  geom_jitter(alpha=0.5, width=0.25) +
  scale_color_manual(values=c("TRUE"="#e31a1c", "FALSE"="#1f78b4"), name="Splice junction type", labels=c("TRUE"="Unproductive", "FALSE"="Productive")) +
  scale_y_continuous(limits=c(0,100), expand=c(0,0)) +
  ggbreak::scale_y_break(c(5,95)) +
  Rotate_x_labels +
  labs(y=str_wrap("percent of splice junction reads", 20), x=NULL)

P.PercentProductive <- ggplotify::as.ggplot(print(P.PercentProductive))
P.PercentProductive

P.PercentUnproductive <- juncs.long.summary %>%
  mutate(PlotGroup = str_detect(SuperAnnotation, "Unproductive")) %>%
  group_by(Dataset, IndID, PlotGroup) %>%
  summarise(Sum=sum(Sum)) %>%
  ungroup() %>%
  group_by(Dataset, IndID) %>%
  mutate(Percent = Sum/sum(Sum) * 100) %>%
  ungroup() %>%
  filter(PlotGroup) %>%
  mutate(Dataset=recode(Dataset, "chRNA.Expression.Splicing"="chRNA",  "MetabolicLabelled.30min"="4sU 30min", "MetabolicLabelled.60min"="4sU 60min", "Expression.Splicing"="polyA"))  %>%
  mutate(Dataset = factor(Dataset, levels=c("chRNA", "4sU 30min", "4sU 60min","polyA"))) %>%
  ggplot(aes(x=Dataset, y=Percent, color=PlotGroup)) +
  geom_boxplot(outlier.shape = NA, position=position_identity()) +
  geom_jitter(alpha=0.5, width=0.25) +
  scale_color_manual(values=c("TRUE"="#e31a1c", "FALSE"="#1f78b4"), name="Splice junction type", labels=c("TRUE"="Unproductive", "FALSE"="Productive")) +
  scale_y_continuous(limits=c(0,5), expand=c(0,0)) +
  Rotate_x_labels +
  labs(y=str_wrap("Percent unproductive splice junction reads", 25), x=NULL) +
  theme(legend.position="none")
P.PercentUnproductive
```

Percent NMD

```{r}
juncs.long.summary.semigroups <- juncs.long %>%
  dplyr::select(chrom, start, stop, strand, Dataset, IndID, Count) %>%
  inner_join(IntronAnnotations) %>%
  filter(!SuperAnnotation %in% c("AnnotatedJunc_NoncodingGene", "UnannotatedJunc_NoncodingGene")) %>%
  group_by(Dataset, IndID, SemiSupergroupAnnotations, SuperAnnotation) %>%
  summarise(Sum=sum(Count)) %>%
  ungroup()

P.Percent.NMD <-
  juncs.long.summary.semigroups %>%
  mutate(PlotGroup =SemiSupergroupAnnotations=="uniquely nonsense_mediated_decay tag") %>%
  group_by(Dataset, IndID) %>%
  mutate(Percent = Sum/sum(Sum) * 100) %>%
  ungroup() %>%
  filter(PlotGroup) %>%
  mutate(Dataset=recode(Dataset, "chRNA.Expression.Splicing"="chRNA",  "MetabolicLabelled.30min"="4sU 30min", "MetabolicLabelled.60min"="4sU 60min", "Expression.Splicing"="polyA"))  %>%
  mutate(Dataset = factor(Dataset, levels=c("chRNA", "4sU 30min", "4sU 60min","polyA"))) %>%
  ggplot(aes(x=Dataset, y=Percent, color=PlotGroup)) +
  geom_boxplot(outlier.shape = NA, position=position_identity()) +
  geom_jitter(alpha=0.5, width=0.25) +
  scale_color_manual(values=c("TRUE"="#e31a1c", "FALSE"="#1f78b4"), name="Splice junction type", labels=c("TRUE"="Unproductive", "FALSE"="Productive")) +
  scale_y_continuous(limits=c(0,0.75), expand=c(0,0)) +
  Rotate_x_labels +
  labs(y=str_wrap("Percent annotated NMD splice junction reads", 25), x=NULL) +
  theme(legend.position="none")
P.Percent.NMD

```

```{r, eval=F}
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_1_PercentNMD.pdf",P.Percent.NMD, width=4, height=4)

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_1_PercentProductive.pdf",P.PercentProductive, width=6, height=4)

```

```{r}
gwas.traits <- read_tsv("../code/config/gwas_table.tsv") %>%
  dplyr::rename(GWAS.accession=gwas, gwas.trait=trait)


hyprcoloc.results <- read_tsv("../code/hyprcoloc/Results/ForGWASColoc/GWASColoc_ChromatinAPAAndRNA/results.txt.gz") %>%
  dplyr::rename(GWAS.Loci = GWASLeadSnpChrom_Pos_RefAllele_AltAllele_rsID_trait) %>%
  separate(GWAS.Loci, into=c("GWAS.LeadSNP.Chrom", "GWAS.LeadSNP.Pos", "GWAS.accession"), sep="_", remove=F) %>%
  separate_rows(ColocalizedTraits, sep = ",") %>%
  mutate(IsColocalizedWithSomething = !ColocalizedTraits == "None") %>%
  mutate(Trait = if_else(IsColocalizedWithSomething, ColocalizedTraits, DroppedTrait)) %>%
  dplyr::select(-DroppedTrait, -ColocalizedTraits) %>%
  mutate(Trait = str_replace_all(Trait, " ", "")) %>%
  mutate(GWAS.Loci = str_replace_all(GWAS.Loci, " ", "")) %>%
  mutate(Trait = if_else(Trait == GWAS.Loci, paste("GWAS",GWAS.Loci,sep = ";"),Trait)) %>%
  separate(Trait, into=c("PhenotypeClass", "Phenotype"), sep=";", remove=F) %>%
  group_by(GWAS.Loci, HyprcolocIteration) %>%
  mutate(ColocalizedClusterContainsGWASTrait = any(PhenotypeClass=="GWAS") & IsColocalizedWithSomething) %>%
  ungroup() %>%
  inner_join(gwas.traits)

hyprcoloc.results$PhenotypeClass %>% unique()

hyprcoloc.results %>%
  filter(!GWAS.accession=="IMSGC2019") %>%
  filter(`PubMed ID`=="27863252") %>%
  group_by(GWAS.Loci, HyprcolocIteration) %>%
  filter(any(PhenotypeClass=="H3K27AC") & any(PhenotypeClass=="GWAS")) %>%
  ungroup() %>%
  filter(PhenotypeClass=="H3K27AC") %>%
  distinct(Phenotype)

gwas.traits %>%
  filter(`PubMed ID`=="27863252")

hyprcoloc.results %>%
  filter(!GWAS.accession=="IMSGC2019") %>%
  filter(`PubMed ID`=="27863252") %>%
  group_by(GWAS.Loci, HyprcolocIteration) %>%
  filter(any(PhenotypeClass=="H3K27AC") & any(PhenotypeClass=="GWAS")) %>%
  ungroup() %>%
  filter(PhenotypeClass=="H3K27AC") %>%
  # filter(PhenotypeClass=="GWAS") %>%
  distinct(Phenotype)
  
```

