---
title: "MakeFinalFigs_Fig2"
output: html_document
date: '2023-05-18'
---


```{r setup, include=TRUE}
knitr::opts_chunk$set(echo = TRUE, warning = F, message = F)

library(tidyverse)
library(RColorBrewer)
library(data.table)
library(readxl)
library(qvalue)
library(ggrepel)
library(knitr)
library(ggbreak)



# Set theme
theme_set(
  theme_classic() +
  theme(text=element_text(size=16,  family="Helvetica")))

# I use layer a lot, to rotate long x-axis labels
Rotate_x_labels <- theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
```


## List

#### Number QTLs

```{r}
PhenotypeWildcardsAll <- Sys.glob("../code/QTLs/QTLTools/*/PermutationPass.FDR_Added.txt.gz") %>%
  str_replace("../code/QTLs/QTLTools/(.+?)/PermutationPass.FDR_Added.txt.gz", "\\1")

PhenotypeWildcards_ToIncluide <- c("APA_Nuclear", "APA_Total", "Expression.Splicing.Subset_YRI", "Expression.Splicing", "H3K27AC", "H3K36ME3", "H3K4ME3", "H3K4ME1", "MetabolicLabelled.30min", "MetabolicLabelled.60min", "chRNA.Expression.Splicing", "chRNA.Splicing", "polyA.Splicing", "polyA.Splicing.Subset_YRI", "MetabolicLabelled.30min.Splicing", "MetabolicLabelled.60min.Splicing")

GroupedPhenotypeWildcards_ToIncluide <- c( "chRNA.Splicing", "polyA.Splicing.Subset_YRI", "polyA.Splicing",  "MetabolicLabelled.30min.Splicing", "MetabolicLabelled.60min.Splicing")

UngroupedPhenotypeWildcards_ToIncluide <- setdiff(PhenotypeWildcards_ToIncluide, GroupedPhenotypeWildcards_ToIncluide)

Grouped.dat <- setNames(paste0("../code/QTLs/QTLTools/",GroupedPhenotypeWildcards_ToIncluide,"/GroupedPermutationPass.FDR_Added.txt.gz"), GroupedPhenotypeWildcards_ToIncluide) %>%
  lapply(fread) %>%
  bind_rows(.id="PC")

Ungrouped.dat <- setNames(paste0("../code/QTLs/QTLTools/",UngroupedPhenotypeWildcards_ToIncluide,"/PermutationPass.FDR_Added.txt.gz"), UngroupedPhenotypeWildcards_ToIncluide) %>%
  lapply(fread) %>%
  bind_rows(.id="PC")

Full.dat <- bind_rows(
  Grouped.dat, Ungrouped.dat
)

```

First plot number of individuals for each assay

```{r}
SampleSize <- read_tsv("../code/QC/SampleSize.PerPhenotype.tsv", col_names=c("fn", "dummy", "n")) %>%
  mutate(PC = str_replace(fn,"QTLs/QTLTools/(.+?)/OnlyFirstReps.sorted.qqnorm.bed.gz", "\\1")) %>%
  filter(PC %in% c("Expression.Splicing", "Expression.Splicing.Subset_YRI", "chRNA.Expression.Splicing", "MetabolicLabelled.30min", "MetabolicLabelled.60min", "H3K27AC", "H3K4ME3", "H3K4ME1", "H3K36ME3", "APA_Nuclear", "APA_Total"))

SampleSize %>%
  dplyr::select(PC, n) %>%
  kable()



SourcePublications <- setNames(c("Lappalainen et al", "Lappalainen et al", "This study", "Li et al", "Li et al", "Grubert et al", "Grubert et al", "Grubert et al", "This study", "Mittleman et al", "Mittleman et al"), SampleSize$PC)

Labels =  setNames(c("polyA RNA-seq*", "polyA RNA-seq", "chRNA-seq", "30m 4sU RNA-seq", "60m 4sU RNA-seq", "H3K4me1 ChIP-seq", "H3K27Ac ChIP-seq", "H3K4me3 ChIP-seq", "H3K36me3 Cut&Tag", "nuclear RNA 3'seq", "Total RNA 3'seq"), SampleSize$PC)

SampleSizes <- SampleSize %>%
  filter(!PC %in% "Expression.Splicing.Subset_YRI") %>%
  mutate(publication = recode(PC, !!!SourcePublications)) %>%
  mutate(Assay = factor(PC, levels=c("H3K4ME1", "H3K27AC", "H3K4ME3", "H3K36ME3", "chRNA.Expression.Splicing", "APA_Nuclear", "APA_Total", "MetabolicLabelled.30min","MetabolicLabelled.60min", "Expression.Splicing"))) %>%
  ggplot(aes(x=Assay, y=n)) +
  geom_col() +
  # geom_text(aes(label=n), color="black") +
  geom_text(aes(label=publication),angle=90, color="black", y=0, hjust=-0.05) +
  scale_x_discrete(labels=Labels) +
  scale_y_continuous(expand=c(0,0)) +
  scale_y_break(c(100,425), scales="fixed", ticklabels=c(425, 450, 475), expand=F, space=0.4) +
  Rotate_x_labels +
  labs(x="Assay", y="Sample size", caption=str_wrap("*Includes samples non-YRI donors, though some supplementary analyses only utilize the 89 YRI samples in this dataset", 30)) +
  theme(axis.text.x=element_text(size=rel(0.5)))

P.SampleSizes <- ggplotify::as.ggplot(print(SampleSizes))
P.SampleSizes
```
Now number of QTLs

```{r}
Labels.df <- read_tsv("../data/Phenotypes_recode_for_Plotting.txt")

MolQTL_Labels <- Labels.df %>% dplyr::select(PC, Shortest) %>% deframe()
MolQTL_AssayLabels <- Labels.df %>% dplyr::select(PC, Assay) %>% deframe()
Factor_Order <- Labels.df %>% dplyr::select(PC, OrderFactor) %>% deframe()

NumQTLs.dat <- Full.dat %>%
  filter(PC %in% c("APA_Total", "APA_Nuclear")) %>%
  separate(phe_id, into=c("gene", "region", "peak"), sep="_") %>%
  arrange(adj_emp_pval) %>% 
  group_by(gene) %>% slice(1) %>%
  bind_rows(
    Full.dat %>%
      filter(!PC %in% c("APA_Total", "APA_Nuclear"))
  ) %>%
  group_by(PC) %>%
  summarise(
    FDR_10 = sum(q<0.1, na.rm=T),
    FDR_05 = sum(q<0.05,  na.rm=T),
    FDR_01 = sum(q<0.01,  na.rm=T),
    TestFeats=n()) %>%
  gather("FDR", "n", -PC, -TestFeats) %>%
  mutate(PC_relabeled = recode(PC, !!!MolQTL_Labels)) %>%
  mutate(Assay_relabeled = recode(PC, !!!MolQTL_AssayLabels)) %>%
  mutate(FinalLabels = paste(PC_relabeled, Assay_relabeled, sep="\n")) %>%
  mutate(OrderFactor = recode(PC, !!!Factor_Order)) %>%
  mutate(FinalLabels = fct_reorder(FinalLabels, OrderFactor))

Num.QTLs <- ggplot(NumQTLs.dat, aes(x=FinalLabels, y=n, fill=FDR)) +
  geom_col(position="identity", width=0.8) +
  Rotate_x_labels +
  scale_y_continuous(trans="log10", breaks=10**(1:5), labels=c(10, 100, "1K", "10K", "100K")) +
  # geom_text_repel(
  #   data = . %>%
  #     gather("FDR","n",-PC),
  #     aes(label=n, y=n), angle=90, hjust=-0.05,
  #   min.segment.length = unit(0, 'lines'),
  #   color="black") +
  geom_text(
    data = . %>%
      filter(FDR== "FDR_10"),
      aes(label=str_glue("{n} / {TestFeats}")), angle=90, hjust=0,
    color="black") +
  coord_cartesian(ylim=c(100,1E5)) +
  scale_fill_manual(values=c("FDR_10"="#ffeda0", "FDR_05"="#feb24c", "FDR_01"="#f03b20"), labels=c("FDR_10"="10%", "FDR_05"="5%", "FDR_01"="1%")) +
  # scale_x_discrete(labels=c())
  labs(x="molQTL type, assay", y="Number QTLs", fill="FDR", caption=str_wrap("*Each sQTL/APA-QTL count here represents a likely independent genetic effect that may effect multiple introns/APA-sites within a local leafcutter cluster of introns/APA-sites", 40)) +
  theme(axis.text.x = element_text(size=9, hjust=1, angle=70))

Num.QTLs

NumQTLs.dat %>% kable()

NumQTLs.dat %>%
  filter(FDR=="FDR_10") %>%
  pull(n) %>% sum()

NumQTLs.dat %>%
  filter(FDR=="FDR_10") %>%
  pull(TestFeats) %>% sum()
```


### Box plots for TTC3

first do some quick data searching to figure out exactly the names of the phenotypes I want to make boxpltos for... 

```{r, eval=F}
coloc.dat <- read_tsv("../code/hyprcoloc/Results/ForColoc/MolColocNonRedundantFullSplicing/tidy_results_OnlyColocalized.txt.gz") %>%
  separate(phenotype_full, into=c("PC", "P"), sep=";")

coloc.dat %>%
  filter(str_detect(Locus, "ENSG00000075234"))

dat <- fread("../code/pi1/PairwisePi1Traits.P.all.txt.gz")

PeaksToTSS <- Sys.glob("../code/Misc/PeaksClosestToTSS/*_assigned.tsv.gz") %>%
  setNames(str_replace(., "../code/Misc/PeaksClosestToTSS/(.+?)_assigned.tsv.gz", "\\1")) %>%
  lapply(read_tsv) %>%
  bind_rows(.id="ChromatinMark") %>%
  mutate(GenePeakPair = paste(gene, peak, sep = ";")) %>%
  distinct(ChromatinMark, peak, gene, .keep_all=T)

PeaksToTSS %>%
  filter(ChromatinMark == "H3K4ME3") %>%
  filter(str_detect(gene, "ENSG00000075234"))

PeaksToTSS %>%
  filter(ChromatinMark == "H3K27AC") %>%
  filter(str_detect(gene, "ENSG00000075234"))

dat %>%
  filter(PC1 == "Expression.Splicing") %>%
  filter(str_detect(P1, "ENSG00000075234")) %>%
  # pull(PC2) %>% unique()
  filter(PC2 == "polyA.Splicing")

dat %>%
  filter(PC1 == "Expression.Splicing") %>%
  filter(str_detect(P1, "ENSG00000075234")) %>%
  filter(P2 %in% c("H3K4ME3_peak_31179", "22:46292328:46292791:clu_48515_+", "H3K27AC_peak_60679", "ENSG00000075234.17"))
  
```

So I'm looking to make boxplots for H3K4ME3_peak_31179 (hQTL), H3K27AC_peak_60679 (hQTL), ENSG00000075234.17 (polyA eQTL), and chRNA sQTL and polyA sQTL for 22:46292328:46292791:clu_48515_+

TopSNP: "22:46291323:G:A". This is the top eQTL SNP and also topSNP from hyprcoloc finemap (which uses multiple clustered traits to help jointly finemap)

Now use my helper script to get genotype/phenotype data for the top eSNP for those traits..

```{bash, eval=F}
cd ../code/scripts/GenometracksByGenotype

python ExtractPhenotypeBedByGenotype.py --Bed ../../QTLs/QTLTools/Expression.Splicing/OnlyFirstReps.sorted.qqnorm.bed.gz --VCF ../../Genotypes/1KG_GRCh38/Autosomes.vcf.gz --FeatureName ENSG00000075234.17 --SnpPos chr22:46291323 > ../../scratch/Boxplot.dat.eQTL.tsv

python ExtractPhenotypeBedByGenotype.py --Bed ../../QTLs/QTLTools/chRNA.Expression.Splicing/OnlyFirstReps.sorted.qqnorm.bed.gz --VCF ../../Genotypes/1KG_GRCh38/Autosomes.vcf.gz --FeatureName ENSG00000075234.17 --SnpPos chr22:46291323 > ../../scratch/Boxplot.dat.chRNAeQTL.tsv

python ExtractPhenotypeBedByGenotype.py --Bed ../../QTLs/QTLTools/polyA.Splicing/OnlyFirstReps.sorted.qqnorm.bed.gz --VCF ../../Genotypes/1KG_GRCh38/Autosomes.vcf.gz --FeatureName 22:46292328:46292791:clu_48515_+  --SnpPos chr22:46291323 > ../../scratch/Boxplot.dat.polyAsQTL.tsv

python ExtractPhenotypeBedByGenotype.py --Bed ../../QTLs/QTLTools/polyA.Splicing/OnlyFirstReps.sorted.qqnorm.bed.gz --VCF ../../Genotypes/1KG_GRCh38/Autosomes.vcf.gz --FeatureName 22:46292328:46292791:clu_48515_+  --SnpPos chr22:46291323 > ../../scratch/Boxplot.dat.polyAsQTL.tsv

python ExtractPhenotypeBedByGenotype.py --Bed ../../QTLs/QTLTools/H3K4ME3/OnlyFirstReps.sorted.qqnorm.bed.gz --VCF ../../Genotypes/1KG_GRCh38/Autosomes.vcf.gz --FeatureName H3K4ME3_peak_31179  --SnpPos chr22:46291323 > ../../scratch/Boxplot.dat.PromoterhQTL.tsv

python ExtractPhenotypeBedByGenotype.py --Bed ../../QTLs/QTLTools/H3K27AC/OnlyFirstReps.sorted.qqnorm.bed.gz --VCF ../../Genotypes/1KG_GRCh38/Autosomes.vcf.gz --FeatureName H3K27AC_peak_60679  --SnpPos chr22:46291323 > ../../scratch/Boxplot.dat.Promoterh3K27QTL.tsv
```

Now let's read in that data and do boxplots...

```{r}
dat.for.boxplots <- list.files("../code/scratch", pattern = "Boxplot.dat.[a-zA-Z0-9]+.tsv", full.names=T) %>%
  setNames(str_replace(., "../code/scratch/Boxplot.dat.(.+?).tsv", "\\1")) %>%
  lapply(read_tsv, col_names=c("SampleID", "P", "genotype", "Ref", "Alt", "ID"), skip=1) %>%
  bind_rows(.id="PC")

dat.for.boxplots %>%
  filter(!genotype==3) %>%
  mutate(genotype= factor(genotype, levels=c(0,1,2))) %>%
  ggplot(aes(x=genotype, y=P, group=genotype)) +
  geom_boxplot(outlier.shape = NA, aes(color=genotype)) +
  # geom_jitter(width=0.25, alpha=0.1) +
  geom_smooth(method='lm', aes(group=1), se=F, color="gray") +
  scale_x_discrete(labels=c("0"="GG", "1"="GA", "2"="AA")) +
  scale_color_manual(values=c("0"="#f03b20", "1"="#7a0177", "2"="#08519c")) +
  # scale_x_discrete() +
  facet_wrap(~PC) +
  theme(legend.position="none") +
  labs(x=NULL, color=NULL)
```

```{r, eval=F}
#colors
#000000,#525252,#d9d9d9

dat.for.boxplots$PC %>% unique()

dat.for.boxplots %>%
  filter(!genotype==3) %>%
  filter(PC=="PromoterhQTL") %>%
  mutate(genotype= factor(genotype, levels=c(0,1,2))) %>%
  ggplot(aes(x=genotype, y=P, group=genotype)) +
  geom_boxplot(outlier.shape = NA, aes(color=genotype), size=2) +
  # geom_jitter(width=0.25, alpha=0.1) +
  geom_smooth(method='lm', aes(group=1), se=F, color="gray") +
  scale_x_discrete(labels=c("0"="GG", "1"="GA", "2"="AA")) +
  scale_color_manual(values=c("0"="#000000", "1"="#525252", "2"="#d9d9d9")) +
  geom_text(data=. %>% head(1),x=-Inf, y=Inf, label=" beta=0.15\n P=0.48", hjust=0, vjust=1, size=5.5) +
  # scale_x_discrete() +
  theme(legend.position="none") +
  Rotate_x_labels +
  scale_y_continuous(limits=c(-4,4)) +
  labs(x=NULL, color=NULL, y="H3K4me3 @ promoter")
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig2_ExampleBoxplot_h3k4meQTL.pdf", height=3, width=2)

dat.for.boxplots %>%
  filter(!genotype==3) %>%
  filter(PC=="Promoterh3K27QTL") %>%
  mutate(genotype= factor(genotype, levels=c(0,1,2))) %>%
  ggplot(aes(x=genotype, y=P, group=genotype)) +
  geom_boxplot(outlier.shape = NA, aes(color=genotype), size=2) +
  # geom_jitter(width=0.25, alpha=0.1) +
  geom_smooth(method='lm', aes(group=1), se=F, color="gray") +
  scale_x_discrete(labels=c("0"="GG", "1"="GA", "2"="AA")) +
  scale_color_manual(values=c("0"="#000000", "1"="#525252", "2"="#d9d9d9")) +
  geom_text(data=. %>% head(1),x=-Inf, y=Inf, label=" beta=0.14\n P=0.10", hjust=0, vjust=1, size=5.5) +
  # scale_x_discrete() +
  theme(legend.position="none") +
  Rotate_x_labels +
  scale_y_continuous(limits=c(-4,4)) +
  labs(x=NULL, color=NULL, y="H3K27ac @ promoter")
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig2_ExampleBoxplot_h3k27acQTL.pdf", height=3, width=2)

dat.for.boxplots %>%
  filter(!genotype==3) %>%
  filter(PC=="eQTL") %>%
  mutate(genotype= factor(genotype, levels=c(0,1,2))) %>%
  ggplot(aes(x=genotype, y=P, group=genotype)) +
  geom_boxplot(outlier.shape = NA, aes(color=genotype), size=2) +
  # geom_jitter(width=0.25, alpha=0.1) +
  geom_smooth(method='lm', aes(group=1), se=F, color="gray") +
  scale_x_discrete(labels=c("0"="GG", "1"="GA", "2"="AA")) +
  scale_color_manual(values=c("0"="#000000", "1"="#525252", "2"="#d9d9d9")) +
  geom_text(data=. %>% head(1),x=-Inf, y=Inf, label=" beta=1.1\n P<5E-43", hjust=0, vjust=1, size=5.5) +
  # scale_x_discrete() +
  theme(legend.position="none") +
  Rotate_x_labels +
  scale_y_continuous(limits=c(-4,4)) +
  labs(x=NULL, color=NULL, y="TTC38 expression")
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig2_ExampleBoxplot_eQTL.pdf", height=3, width=2)

dat.for.boxplots %>%
  filter(!genotype==3) %>%
  filter(PC=="chRNAeQTL") %>%
  mutate(genotype= factor(genotype, levels=c(0,1,2))) %>%
  ggplot(aes(x=genotype, y=P, group=genotype)) +
  geom_boxplot(outlier.shape = NA, aes(color=genotype), size=2) +
  # geom_jitter(width=0.25, alpha=0.1) +
  geom_smooth(method='lm', aes(group=1), se=F, color="gray") +
  scale_x_discrete(labels=c("0"="GG", "1"="GA", "2"="AA")) +
  scale_color_manual(values=c("0"="#000000", "1"="#525252", "2"="#d9d9d9")) +
  geom_text(data=. %>% head(1),x=-Inf, y=Inf, label=" beta=0.40\n P<3E-4", hjust=0, vjust=1, size=5.5) +
  # scale_x_discrete() +
  theme(legend.position="none") +
  Rotate_x_labels +
  scale_y_continuous(limits=c(-4,4)) +
  labs(x=NULL, color=NULL, y="TTC38 expression")
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig2_ExampleBoxplot_chRNAeQTL.pdf", height=3, width=2)

dat.for.boxplots %>%
  filter(!genotype==3) %>%
  filter(PC=="polyAsQTL") %>%
  mutate(genotype= factor(genotype, levels=c(0,1,2))) %>%
  ggplot(aes(x=genotype, y=P, group=genotype)) +
  geom_boxplot(outlier.shape = NA, aes(color=genotype), size=2) +
  # geom_jitter(width=0.25, alpha=0.1) +
  geom_smooth(method='lm', aes(group=1), se=F, color="gray") +
  scale_x_discrete(labels=c("0"="GG", "1"="GA", "2"="AA")) +
  scale_color_manual(values=c("0"="#000000", "1"="#525252", "2"="#d9d9d9")) +
  geom_text(data=. %>% head(1), x=-Inf, y=Inf, label=" beta=-1.1\n P<3e-24", hjust=0, vjust=1, size=5.5) +
  # scale_x_discrete() +
  theme(legend.position="none") +
  Rotate_x_labels +
  scale_y_continuous(limits=c(-4,4)) +
  labs(x=NULL, color=NULL, y="Splicing of cassette exon")
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig2_ExampleBoxplot_polyAsQTL.pdf", height=3, width=2)

dat.for.boxplots %>%
  filter(!genotype==3) %>%
  filter(PC=="chRNAsQTL") %>%
  mutate(genotype= factor(genotype, levels=c(0,1,2))) %>%
  ggplot(aes(x=genotype, y=P, group=genotype)) +
  geom_boxplot(outlier.shape = NA, aes(color=genotype), size=2) +
  # geom_jitter(width=0.25, alpha=0.1) +
  geom_smooth(method='lm', aes(group=1), se=F, color="gray") +
  scale_x_discrete(labels=c("0"="GG", "1"="GA", "2"="AA")) +
  scale_color_manual(values=c("0"="#000000", "1"="#525252", "2"="#d9d9d9")) +
  geom_text(data=. %>% head(1),x=-Inf, y=Inf, label=" beta=-1.3\n P<4e-19", hjust=0, vjust=1, size=5.5) +
  # scale_x_discrete() +
  theme(legend.position="none") +
  Rotate_x_labels +
  scale_y_continuous(limits=c(-4,4)) +
  labs(x=NULL, color=NULL, y="Splicing of cassette exon")
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig2_ExampleBoxplot_chRNAsQTL.pdf", height=3, width=2)
```


And pygenometracks stuff for TTC38:

```{bash, eval=F}
cd project2/yangili1/bjf79/ChromatinSplicingQTLs/code
conda activate GenometracksByGenotype

python scripts/GenometracksByGenotype/AggregateBigwigsForPlotting.py --BigwigList PlotQTLs/bwList.tsv --BigwigListType KeyFile --OutputPrefix /scratch/midway2/bjf79/PlotDat --NoSashimi --VCF Genotypes/1KG_GRCh38/Autosomes.vcf.gz --SnpPos chr22:46291323  --Region chr22:46,267,416-46,294,315 --GroupSettingsFile <(cat PlotQTLs/bwList.Groups.4col.tsv | awk 'NR==1 || NR==2 || NR==3 || NR==8') --Bed12GenesToIni scripts/GenometracksByGenotype/PremadeTracks/gencode.v26.FromGTEx.genes.bed12.gz --GenotypeCustomPalette "#000000,#525252,#d9d9d9,#ffffff" -vv

pyGenomeTracks --region chr22:46,267,416-46,294,315 -out scratch/TTC38.pdf --tracks /scratch/midway2/bjf79/PlotDattracks.ini

```

or to make a condensed version, create pdf for each exon and stitch together in illustrator

```{bash, eval=F}
pyGenomeTracks --region chr22:46267350-46268754 -out scratch/TTC38.1.pdf --tracks /scratch/midway2/bjf79/PlotDattracks.ini --plotWidth 8.16279
pyGenomeTracks --region chr22:46272234-46272516 -out scratch/TTC38.2.pdf --tracks /scratch/midway2/bjf79/PlotDattracks.ini --plotWidth 1.63953
pyGenomeTracks --region chr22:46273795-46274169 -out scratch/TTC38.3.pdf --tracks /scratch/midway2/bjf79/PlotDattracks.ini --plotWidth 2.17442
pyGenomeTracks --region chr22:46275147-46275521 -out scratch/TTC38.4.pdf --tracks /scratch/midway2/bjf79/PlotDattracks.ini --plotWidth 2.17442
pyGenomeTracks --region chr22:46278485-46278761 -out scratch/TTC38.5.pdf --tracks /scratch/midway2/bjf79/PlotDattracks.ini --plotWidth 1.60465
pyGenomeTracks --region chr22:46281498-46281818 -out scratch/TTC38.6.pdf --tracks /scratch/midway2/bjf79/PlotDattracks.ini --plotWidth 1.86047
pyGenomeTracks --region chr22:46283872-46284132 -out scratch/TTC38.7.pdf --tracks /scratch/midway2/bjf79/PlotDattracks.ini --plotWidth 1.51163
pyGenomeTracks --region chr22:46285140-46285379 -out scratch/TTC38.8.pdf --tracks /scratch/midway2/bjf79/PlotDattracks.ini --plotWidth 1.38953
pyGenomeTracks --region chr22:46286972-46287254 -out scratch/TTC38.9.pdf --tracks /scratch/midway2/bjf79/PlotDattracks.ini --plotWidth 1.63953
pyGenomeTracks --region chr22:46288322-46288688 -out scratch/TTC38.10.pdf --tracks /scratch/midway2/bjf79/PlotDattracks.ini --plotWidth 2.12791
pyGenomeTracks --region chr22:46289301-46289661 -out scratch/TTC38.11.pdf --tracks /scratch/midway2/bjf79/PlotDattracks.ini --plotWidth 2.09302
pyGenomeTracks --region chr22:46289725-46289999 -out scratch/TTC38.12.pdf --tracks /scratch/midway2/bjf79/PlotDattracks.ini --plotWidth 1.59302
pyGenomeTracks --region chr22:46292102-46292428 -out scratch/TTC38.13.pdf --tracks /scratch/midway2/bjf79/PlotDattracks.ini --plotWidth 1.89535
pyGenomeTracks --region chr22:46292690-46294108 -out scratch/TTC38.14.pdf --tracks /scratch/midway2/bjf79/PlotDattracks.ini --plotWidth 8.24419
```

And also sashimi version:

```{bash, eval=F}
python scripts/GenometracksByGenotype/AggregateBigwigsForPlotting.py --BigwigList PlotQTLs/bwList.tsv --BigwigListType KeyFile --OutputPrefix /scratch/midway2/bjf79/PlotDatSashimi --VCF Genotypes/1KG_GRCh38/Autosomes.vcf.gz --SnpPos chr22:46291323  --Region chr22:46,289,843-46,292,952 --GroupSettingsFile <(cat PlotQTLs/bwList.Groups.4col.tsv | awk 'NR==1 || NR==8') --Bed12GenesToIni scripts/GenometracksByGenotype/PremadeTracks/gencode.v26.FromGTEx.genes.bed12.gz --GenotypeCustomPalette "#000000,#525252,#d9d9d9,#ffffff" --FilterJuncsByBed ../output/TTC38_RelevantJuncs.bed
```



### Classification of unannotated and unproductive juncs

Actually a supplement related to fig1

```{r}
juncs.long <- fread("../code/SplicingAnalysis/CombinedJuncTables/All.tsv.gz")

dat.KD <- fread("/project2/yangili1/cfbuenabadn/ChromatinSplicingQTLs/code/SplicingAnalysis/CombinedJuncTables/NMD_KD.tsv.gz")

juncs.long <- bind_rows(juncs.long, dat.KD)

IntronAnnotations <- read_tsv("../data/IntronAnnotationsFromYang.Updated.tsv.gz") %>%
  mutate(stop = end-1) %>%
  mutate(SuperAnnotation = recode(SuperAnnotation, "UnannotatedJunc_NoncodingJunc"="UnannotatedJunc_NoncodingGene"))

juncs.long.summary <- juncs.long %>%
  dplyr::select(chrom, start, stop, strand, Dataset, Count) %>%
  inner_join(IntronAnnotations) %>%
  group_by(Dataset, SemiSupergroupAnnotations, SuperAnnotation) %>%
  summarise(Sum=sum(Count))

juncs.long.summary$Dataset %>% unique()

P.PercentAnnotated <- juncs.long.summary %>%
  ungroup() %>%
  # filter(Dataset %in% c("Expression.Splicing", "chRNA.Expression.Splicing")) %>%
  filter(!str_detect(SuperAnnotation, "Noncoding")) %>%
  group_by(Dataset, SuperAnnotation) %>%
  summarise(TotalSum = sum(Sum)) %>%
  ungroup() %>%
  mutate(PlotGroup = if_else(SuperAnnotation=="AnnotatedJunc_ProductiveCodingGene", "Annotated productive (basic tag)", "Other")) %>%
  group_by(Dataset) %>%
  mutate(DatasetSum = sum(TotalSum)) %>%
  ungroup() %>%
  group_by(Dataset, PlotGroup) %>%
  summarise(Percent=sum(TotalSum)/DatasetSum) %>%
  ungroup() %>%
  distinct() %>%
  mutate(Dataset = factor(Dataset, levels=c("Expression.Splicing", "MetabolicLabelled.60min","MetabolicLabelled.30min","chRNA.Expression.Splicing", "HeLa.SMG6.KD", "HeLa.SMG7.KD", "HeLa.UPF1.KD", "HeLa.dKD", "HeLa.scr"))) %>%
  filter(Dataset %in% c("Expression.Splicing", "MetabolicLabelled.60min","MetabolicLabelled.30min","chRNA.Expression.Splicing")) %>%
  ggplot(aes(x=Dataset, y=Percent*100, fill=PlotGroup)) +
  geom_col() +
  scale_x_discrete(labels=c("Expression.Splicing"="steady-state\nRNA", "MetabolicLabelled.60min"="60m 4sU", "MetabolicLabelled.30min"="30m 4sU","chRNA.Expression.Splicing"="naRNA")) +
  scale_y_continuous(expand=c(0,0)) +
  scale_fill_manual(values=c("Annotated productive (basic tag)"="#1f78b4", "Other"="#969696")) + 
  scale_y_break(c(3, 97), expand=F, space=0.5) +
  labs(y="Percent of splice junction reads", fill="Intron classification") +
  theme(legend.position="none") +
  Rotate_x_labels
P.PercentAnnotated <- ggplotify::as.ggplot(print(P.PercentAnnotated))
P.PercentAnnotated

```

Another version with bars for each individual to visualize variance

```{r}
juncs.long.summary.PerInd <- juncs.long %>%
  dplyr::select(chrom, start, stop, strand, Dataset, Count, IndID) %>%
  inner_join(IntronAnnotations) %>%
  group_by(Dataset, SemiSupergroupAnnotations, SuperAnnotation, IndID) %>%
  summarise(Sum=sum(Count))


P.PercentAnnotated.PerSample.dat <- juncs.long.summary.PerInd %>%
  ungroup() %>%
  filter(!str_detect(SuperAnnotation, "Noncoding")) %>%
  group_by(Dataset, SuperAnnotation, IndID) %>%
  summarise(TotalSum = sum(Sum)) %>%
  ungroup() %>%
  mutate(PlotGroup = if_else(SuperAnnotation=="AnnotatedJunc_ProductiveCodingGene", "Annotated productive (basic tag)", "Other")) %>%
  group_by(Dataset, IndID) %>%
  mutate(DatasetSum = sum(TotalSum)) %>%
  ungroup() %>%
  group_by(Dataset, PlotGroup, IndID) %>%
  summarise(Percent=sum(TotalSum)/DatasetSum) %>%
  ungroup() %>%
  filter(PlotGroup=="Annotated productive (basic tag)") %>%
  distinct(IndID, Dataset, .keep_all=T) %>%
  mutate(Dataset = factor(Dataset, levels=c("Expression.Splicing", "MetabolicLabelled.60min","MetabolicLabelled.30min","chRNA.Expression.Splicing", "HeLa.SMG6.KD", "HeLa.SMG7.KD", "HeLa.UPF1.KD", "HeLa.dKD", "HeLa.scr"))) %>%
  arrange(Dataset, desc(Percent)) %>% 
  mutate(rn = row_number()) %>%
  mutate(Complement = 1-Percent) %>%
  dplyr::select(-PlotGroup) %>%
  gather(key="PlotGroup", value="NewPercent", Percent, Complement) %>%
  mutate(PlotGroup = relevel(factor(PlotGroup), "Percent")) %>%
  mutate(Dataset = recode(Dataset, "Expression.Splicing"="steady-state\nRNA", "MetabolicLabelled.60min"="60m 4sU", "MetabolicLabelled.30min"="30m 4sU","chRNA.Expression.Splicing"="naRNA")) %>%
  mutate(Dataset = factor(Dataset, levels=c("steady-state\nRNA", "60m 4sU","30m 4sU","naRNA", "HeLa.SMG6.KD", "HeLa.SMG7.KD", "HeLa.UPF1.KD", "HeLa.dKD", "HeLa.scr")))

P.PercentAnnotated.PerSample.dat %>%
  filter(PlotGroup == "Complement") %>%
                  group_by(Dataset) %>%
                  summarise(Median = median(NewPercent)*100)

P.PercentAnnotated.PerSample <- P.PercentAnnotated.PerSample.dat %>%
  filter(Dataset %in% c("steady-state\nRNA", "60m 4sU","30m 4sU","naRNA")) %>%
  ggplot(aes(x=rn, y=NewPercent*100, fill=PlotGroup)) +
  geom_col() +
  scale_x_discrete(labels=c("Expression.Splicing"="steady-state\nRNA", "MetabolicLabelled.60min"="60m 4sU", "MetabolicLabelled.30min"="30m 4sU","chRNA.Expression.Splicing"="chRNA")) +
  geom_hline(data = . %>%
                 filter(PlotGroup == "Complement") %>%
                  group_by(Dataset) %>%
                  summarise(Median = median(NewPercent)*100),
             aes(yintercept = Median),
             color="black", linetype='dashed', size=1.5
             ) +
  scale_y_continuous(expand=c(0,0)) +
  scale_fill_manual(values=c("Percent"="#1f78b4", "Complement"="#969696"), labels=c("Percent"="Annotated productive (basic tag)","Complement"="Other")) + 
  labs(y="Percent of splice junction reads", fill="Intron classification") +
  theme(legend.position="none") +
  facet_grid(~Dataset, scales = 'free', switch = "both") +
  scale_y_break(c(4, 96), expand=F, scales="free", space=0.1) +
  Rotate_x_labels +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  theme(
    aspect.ratio = 1,
    strip.background = element_blank(),
    strip.placement = "outside"
  )

P.PercentAnnotated.PerSample <- ggplotify::as.ggplot(print(P.PercentAnnotated.PerSample))
P.PercentAnnotated.PerSample

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_PercentAnnotatedPerSample.pdf",P.PercentAnnotated.PerSample, width=4, height=6)


```

Another plot of this style for the shRNA KD.

```{r}
P.PercentAnnotated.PerSample.NMD_KD <- P.PercentAnnotated.PerSample.dat %>%
  filter(Dataset %in% c("HeLa.dKD", "HeLa.scr")) %>%
  mutate(Dataset = recode(Dataset, "HeLa.scr"="shRNA ctl", "HeLa.dKD"="shRNA dKD")) %>%
  mutate(Dataset = relevel(factor(Dataset), "shRNA ctl")) %>%
  ggplot(aes(x=rn, y=NewPercent*100, fill=PlotGroup)) +
  geom_col() +
  scale_x_discrete(labels=c("Expression.Splicing"="steady-state\nRNA", "MetabolicLabelled.60min"="60m 4sU", "MetabolicLabelled.30min"="30m 4sU","chRNA.Expression.Splicing"="chRNA")) +
  geom_hline(data = . %>%
                 filter(PlotGroup == "Complement") %>%
                  group_by(Dataset) %>%
                  summarise(Median = median(NewPercent)*100),
             aes(yintercept = Median),
             color="black", linetype='dashed', size=1.5
             ) +
  scale_y_continuous(expand=c(0,0)) +
  scale_fill_manual(values=c("Percent"="#1f78b4", "Complement"="#969696"), labels=c("Percent"="Annotated productive (basic tag)","Complement"="Other")) + 
  scale_y_break(c(4, 96), expand=F, space=0.1) +
  labs(y="Percent of splice junction reads", fill="Intron classification") +
  theme(legend.position="none") +
  facet_grid(~Dataset, scales = 'free', switch = "both") +
  Rotate_x_labels +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  theme(
    aspect.ratio = 1,
    strip.background = element_blank(),
    strip.placement = "outside"
  )

P.PercentAnnotated.PerSample.NMD_KD <- ggplotify::as.ggplot(print(P.PercentAnnotated.PerSample.NMD_KD))
P.PercentAnnotated.PerSample.NMD_KD

# ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_PercentAnnotatedPerSample_NMDKD.pdf",P.PercentAnnotated.PerSample.NMD_KD, width=3, height=6)


P.PercentAnnotated.PerSample.NMD_KD_full <- P.PercentAnnotated.PerSample.dat %>%
  filter(str_detect(Dataset, "HeLa")) %>%
  mutate(Dataset = recode(Dataset, "HeLa.scr"="shRNA ctl", "HeLa.dKD"="shRNA dKD")) %>%
  mutate(Dataset = relevel(factor(Dataset), "shRNA ctl")) %>%
  ggplot(aes(x=rn, y=NewPercent*100, fill=PlotGroup)) +
  geom_col() +
  scale_x_discrete(labels=c("Expression.Splicing"="steady-state\nRNA", "MetabolicLabelled.60min"="60m 4sU", "MetabolicLabelled.30min"="30m 4sU","chRNA.Expression.Splicing"="chRNA")) +
  geom_hline(data = . %>%
                 filter(PlotGroup == "Complement") %>%
                  group_by(Dataset) %>%
                  summarise(Median = median(NewPercent)*100),
             aes(yintercept = Median),
             color="black", linetype='dashed', size=1.5
             ) +
  scale_y_continuous(expand=c(0,0)) +
  scale_fill_manual(values=c("Percent"="#1f78b4", "Complement"="#969696"), labels=c("Percent"="Annotated productive (basic tag)","Complement"="Other")) + 
  scale_y_break(c(4, 96), expand=F, space=0.1) +
  labs(y="Percent of splice junction reads", fill="Intron classification") +
  theme(legend.position="none") +
  facet_grid(~Dataset, scales = 'free', switch = "both") +
  Rotate_x_labels +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  theme(
    aspect.ratio = 1,
    strip.background = element_blank(),
    strip.placement = "outside"
  )

P.PercentAnnotated.PerSample.NMD_KD_full <- ggplotify::as.ggplot(print(P.PercentAnnotated.PerSample.NMD_KD_full))
P.PercentAnnotated.PerSample.NMD_KD_full

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_PercentAnnotatedPerSample_NMDKD_All.pdf",P.PercentAnnotated.PerSample.NMD_KD_full, width=4, height=6)


P.PercentAnnotated.PerSample.dat %>%
  # distinct(PlotGroup)
  filter(PlotGroup == "Complement") %>%
  group_by(Dataset) %>%
  summarise(Median = median(NewPercent)*100)


```


And lets print table of medians for custom labelling in illustrator...

```{r}
P.PercentAnnotated.PerSample.dat %>%
  ungroup() %>%
  filter(PlotGroup == "Complement") %>%
  group_by(Dataset) %>%
  summarise(Median = median(NewPercent*100, na.rm = T)) %>%
  ungroup()

```


Plot to describe What are the unannoatted junctions

```{r}

P.PercentOfUnannotated <- juncs.long.summary %>%
  ungroup() %>%
  filter(Dataset %in% c("chRNA.Expression.Splicing")) %>%
  filter(!str_detect(SuperAnnotation, "Noncoding")) %>%
  filter(SuperAnnotation!="AnnotatedJunc_ProductiveCodingGene") %>%
  mutate(PercentOfNon_AnnotatedProductive = Sum/sum(Sum)*100) %>%
  arrange(desc(SuperAnnotation), PercentOfNon_AnnotatedProductive) %>%
  mutate(n=row_number()) %>%
  mutate(SemiSupergroupAnnotations=fct_reorder(SemiSupergroupAnnotations, n)) %>%
  ggplot(aes(x=Dataset, y=PercentOfNon_AnnotatedProductive, fill=SemiSupergroupAnnotations)) +
  geom_col()
P.PercentOfUnannotated
```

One more version... with fixed scale... borrowed code from [here](https://github.com/slowkow/ggrepel/issues/161)

```{r}
position_stack_and_nudge <- function(x = 0, y = 0, vjust = 1, reverse = FALSE) {
  ggproto(NULL, PositionStackAndNudge,
    x = x,
    y = y,
    vjust = vjust,
    reverse = reverse
  )
}

#' @rdname ggplot2-ggproto
#' @format NULL
#' @usage NULL
#' @noRd
PositionStackAndNudge <- ggproto("PositionStackAndNudge", PositionStack,
  x = 0,
  y = 0,

  setup_params = function(self, data) {
    c(
        list(x = self$x, y = self$y),
        ggproto_parent(PositionStack, self)$setup_params(data)
    )
  },

  compute_layer = function(self, data, params, panel) {
    # operate on the stacked positions (updated in August 2020)
    data = ggproto_parent(PositionStack, self)$compute_layer(data, params, panel)

    x_orig <- data$x
    y_orig <- data$y
    # transform only the dimensions for which non-zero nudging is requested
    if (any(params$x != 0)) {
      if (any(params$y != 0)) {
        data <- transform_position(data, function(x) x + params$x, function(y) y + params$y)
      } else {
        data <- transform_position(data, function(x) x + params$x, NULL)
      }
    } else if (any(params$y != 0)) {
      data <- transform_position(data, function(x) x, function(y) y + params$y)
    }
    data$nudge_x <- data$x
    data$nudge_y <- data$y
    data$x <- x_orig
    data$y <- y_orig

    data
  },

  compute_panel = function(self, data, params, scales) {
      ggproto_parent(PositionStack, self)$compute_panel(data, params, scales)
  }
)
```

```{r}

AnnotationColors <- IntronAnnotations %>%
  distinct(SemiSupergroupAnnotations, SuperAnnotation) %>%
  mutate(Color = recode(SuperAnnotation, !!!c(
    "AnnotatedJunc_NoncodingGene"="#6a3d9a",
    "UnannotatedJunc_NoncodingGene"="#cab2d6",
    "AnnotatedJunc_ProductiveCodingGene"="#1f78b4",
    "UnannotatedJunc_ProductiveCodingGene"="#a6cee3",
    "AnnotatedJunc_UnproductiveCodingGene"="#e31a1c",
    "UnannotatedJunc_UnproductiveCodingGene"="#fb9a99")))

P.PercentOfUnannotated <- juncs.long.summary %>%
  ungroup() %>%
  filter(Dataset %in% c("chRNA.Expression.Splicing")) %>%
  filter(!str_detect(SuperAnnotation, "Noncoding")) %>%
  filter(SuperAnnotation!="AnnotatedJunc_ProductiveCodingGene") %>%
  mutate(PercentOfNon_AnnotatedProductive = Sum/sum(Sum)*100) %>%
  mutate(SemiSupergroupAnnotations=if_else(Sum <= 996286 & SuperAnnotation=="UnannotatedJunc_UnproductiveCodingGene", "Other predicted non-productive", SemiSupergroupAnnotations)) %>%
  group_by(Dataset, SemiSupergroupAnnotations, SuperAnnotation) %>%
  summarise(PercentOfNon_AnnotatedProductive=sum(PercentOfNon_AnnotatedProductive)) %>%
  ungroup() %>%
  arrange(desc(SuperAnnotation), PercentOfNon_AnnotatedProductive) %>%
  mutate(n=row_number()) %>%
  mutate(SemiSupergroupAnnotations=fct_reorder(SemiSupergroupAnnotations, n)) %>%
  ggplot(aes(x=Dataset, y=PercentOfNon_AnnotatedProductive, fill=SemiSupergroupAnnotations)) +
  geom_col(color='black') +
  geom_text_repel(
      aes(label = SemiSupergroupAnnotations, x=1.4),
      max.overlaps = Inf,
      min.segment.length=0,
      label.padding=1,
      position = position_stack_and_nudge(vjust = 0.5, x = 1),
      direction = 'both',
      arrow=NULL
  ) +
  scale_y_continuous(expand=c(0,0)) +
  scale_x_discrete(expand=c(0,0)) +
  scale_fill_manual(values=AnnotationColors %>% dplyr::select(SemiSupergroupAnnotations, Color) %>% deframe(), na.value="#fb9a99", labels = function(x) str_wrap(x, width = 10)) +
  theme(legend.position='none') +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  labs(y="Percent junction reads amongst other")

P.PercentOfUnannotated

P.PercentJuncsLegend.dat <- 
  AnnotationColors %>%
  distinct(SuperAnnotation, .keep_all=T) %>%
  filter(!str_detect(SuperAnnotation, "Noncoding"))

P.PercentJuncsLegend <- ggplot(P.PercentJuncsLegend.dat, aes(x=SuperAnnotation, y=1, fill=SuperAnnotation)) +
  geom_col() +
  scale_fill_manual(values=P.PercentJuncsLegend.dat %>%
                      dplyr::select(SuperAnnotation, Color) %>% deframe(), labels=c("AnnotatedJunc_UnproductiveCodingGene"="Annotated unproductive", "AnnotatedJunc_ProductiveCodingGene"="Annotated productive", "UnannotatedJunc_ProductiveCodingGene"="Unannotated unproductive", "UnannotatedJunc_UnproductiveCodingGene"="Unannotated productive")) +
  labs(fill="Intron classification") 

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_PercentOfUnannotated.pdf",P.PercentOfUnannotated, width=4, height=6)


```



Figure supplement that goes along with this... chRNA vs polyA scatter for junction RPM...


```{r}
FixBugForDuplicateAnnotationClasses <- IntronAnnotations %>%
  count(SuperAnnotation, SemiSupergroupAnnotations) %>%
  arrange(SemiSupergroupAnnotations) %>%
  add_count(SemiSupergroupAnnotations) %>%
  filter(nn==2) %>%
  distinct(SemiSupergroupAnnotations) %>%
  pull(SemiSupergroupAnnotations)
```

scatter plots, 

```{r}
#Normal
juncs.long.summary.ForReasonsPlot <- juncs.long %>%
  filter(Dataset %in% c("Expression.Splicing", "chRNA.Expression.Splicing")) %>%
  dplyr::select(chrom, start, stop, strand, Dataset, Count) %>%
  inner_join(IntronAnnotations) %>%
  group_by(chrom, start, stop, strand, Dataset, SemiSupergroupAnnotations, SuperAnnotation) %>%
  summarise(Sum=sum(Count)) %>%
  ungroup() %>%
  group_by(Dataset) %>%
  mutate(TotalSum= sum(Sum)) %>%
  mutate(RPM = Sum/TotalSum*1E6) %>%
  ungroup() %>%
  dplyr::select(-TotalSum) %>%
  pivot_wider(names_from="Dataset", values_from=c("RPM", "Sum"), values_fill = 0) %>%
  filter(!(SemiSupergroupAnnotations %in% FixBugForDuplicateAnnotationClasses & SuperAnnotation=="UnannotatedJunc_NoncodingGene"))


SupergroupLabels = c(
  "AnnotatedJunc_ProductiveCodingGene"="Annotated\nProductive",
  "UnannotatedJunc_ProductiveCodingGene"="Unannotated\nProductive",
  "AnnotatedJunc_UnproductiveCodingGene"="Annotated\nUnproductive",
  "UnannotatedJunc_UnproductiveCodingGene"="Unannotated\nUnproductive",
  "AnnotatedJunc_NoncodingGene"="Annotated\nNoncoding or hypervariable gene",
  "UnannotatedJunc_NoncodingGene"="Unannotated\nNoncoding or hypervariable gene")

Orders <- juncs.long.summary.ForReasonsPlot %>%
  distinct(SuperAnnotation, SemiSupergroupAnnotations) %>%
  mutate(SuperAnnotation = factor(SuperAnnotation, levels=names(SupergroupLabels))) %>%
  arrange(SuperAnnotation, SemiSupergroupAnnotations) %>%
  mutate(rn = row_number())

Scatter.P.dat <- juncs.long.summary.ForReasonsPlot %>%
  add_count(SemiSupergroupAnnotations) %>%
  filter(n>100) %>%
  inner_join(Orders) %>%
  mutate(SemiSupergroupAnnotations = recode(SemiSupergroupAnnotations, "basic tag"="protein_coding tag")) %>%
  mutate(SemiSupergroupAnnotations=fct_reorder(SemiSupergroupAnnotations, rn))


#shRNA
juncs.long.summary.ForReasonsPlot.shRNA <- juncs.long %>%
  filter(Dataset %in% c("HeLa.scr", "HeLa.dKD")) %>%
  dplyr::select(chrom, start, stop, strand, Dataset, Count) %>%
  inner_join(IntronAnnotations) %>%
  group_by(chrom, start, stop, strand, Dataset, SemiSupergroupAnnotations, SuperAnnotation) %>%
  summarise(Sum=sum(Count)) %>%
  ungroup() %>%
  group_by(Dataset) %>%
  mutate(TotalSum= sum(Sum)) %>%
  mutate(RPM = Sum/TotalSum*1E6) %>%
  ungroup() %>%
  dplyr::select(-TotalSum) %>%
  pivot_wider(names_from="Dataset", values_from=c("RPM", "Sum"), values_fill = 0) %>%
  filter(!(SemiSupergroupAnnotations %in% FixBugForDuplicateAnnotationClasses & SuperAnnotation=="UnannotatedJunc_NoncodingGene"))



Orders.shRNA <- juncs.long.summary.ForReasonsPlot.shRNA %>%
  distinct(SuperAnnotation, SemiSupergroupAnnotations) %>%
  mutate(SuperAnnotation = factor(SuperAnnotation, levels=names(SupergroupLabels))) %>%
  arrange(SuperAnnotation, SemiSupergroupAnnotations) %>%
  mutate(rn = row_number())

Scatter.P.dat.shRNA <- juncs.long.summary.ForReasonsPlot.shRNA %>%
  add_count(SemiSupergroupAnnotations) %>%
  filter(n>100) %>%
  inner_join(Orders.shRNA) %>%
  mutate(SemiSupergroupAnnotations = recode(SemiSupergroupAnnotations, "basic tag"="protein_coding tag")) %>%
  mutate(SemiSupergroupAnnotations=fct_reorder(SemiSupergroupAnnotations, rn))

#Filter for facets that pass threshold in both
FacetsToInclude <- intersect(
  Scatter.P.dat.shRNA$SemiSupergroupAnnotations %>% unique(),
  Scatter.P.dat$SemiSupergroupAnnotations %>% unique())

Scatter.P.dat.shRNA <- Scatter.P.dat.shRNA %>%
  filter(SemiSupergroupAnnotations %in% FacetsToInclude)
Scatter.P.dat <- Scatter.P.dat %>%
  filter(SemiSupergroupAnnotations %in% FacetsToInclude)

# Summary
Scatter.P.dat.summary <- Scatter.P.dat %>%
  filter(Sum_Expression.Splicing + Sum_chRNA.Expression.Splicing >= 10) %>%
  mutate(Ratio = RPM_chRNA.Expression.Splicing/RPM_Expression.Splicing) %>%
  group_by(SemiSupergroupAnnotations, SuperAnnotation, n) %>%
  summarise(median=round(median(Ratio, na.rm=F), digits=2)) %>%
  mutate(label = str_glue("n:{n}\nmedian:{median}"))

Scatter.P.dat.summary.shRNA <- Scatter.P.dat.shRNA %>%
  filter(Sum_HeLa.dKD + Sum_HeLa.scr >= 10) %>%
  mutate(Ratio = Sum_HeLa.dKD/Sum_HeLa.scr) %>%
  group_by(SemiSupergroupAnnotations, SuperAnnotation, n) %>%
  summarise(median=round(median(Ratio, na.rm=F), digits=2)) %>%
  mutate(label = str_glue("n:{n}\nmedian:{median}"))

Scatter.P.shRNA <- Scatter.P.dat.shRNA %>%
  ggplot(aes(x=Sum_HeLa.dKD, y=Sum_HeLa.scr)) +
  geom_point(alpha=0.1) +
  geom_abline(color='red') +
  geom_density_2d(aes(color=SuperAnnotation)) +
  geom_text(data = Scatter.P.dat.summary.shRNA, aes(label=label),x=-Inf, y=Inf, size=3, hjust=0, vjust=1.1) +
  scale_x_continuous(trans='log10') +
  scale_y_continuous(trans='log10') +
  scale_color_manual(values=AnnotationColors %>% dplyr::select(SuperAnnotation, Color) %>% deframe(), na.value="#fb9a99", labels=SupergroupLabels) +
  facet_wrap(~SemiSupergroupAnnotations, labeller = label_wrap_gen(width=14), nrow=3) +
  coord_fixed() +
  theme(strip.text.x = element_text(size = 7), legend.position="none") +
  guides(colour = guide_legend(nrow = 2)) +
  labs(title="Reason for splice junction classification", x="shRNA dKD\nRPM across junctions", y="shRNA ctrlA\nRPM across junctions", color="Intron classification") +
  Rotate_x_labels

Scatter.P <- Scatter.P.dat %>%
  ggplot(aes(x=RPM_chRNA.Expression.Splicing, y=RPM_Expression.Splicing)) +
  geom_point(alpha=0.1) +
  geom_abline(color='red') +
  geom_density_2d(aes(color=SuperAnnotation)) +
  geom_text(data = Scatter.P.dat.summary, aes(label=label),x=-Inf, y=Inf, size=3, hjust=0, vjust=1.1) +
  scale_x_continuous(trans='log10') +
  scale_y_continuous(trans='log10') +
  scale_color_manual(values=AnnotationColors %>% dplyr::select(SuperAnnotation, Color) %>% deframe(), na.value="#fb9a99", labels=SupergroupLabels) +
  facet_wrap(~SemiSupergroupAnnotations, labeller = label_wrap_gen(width=14), nrow = 3) +
  coord_fixed() +
  theme(strip.text.x = element_text(size = 7), legend.position="none") +
  guides(colour = guide_legend(nrow = 2)) +
  labs(title="Reason for splice junction classification", x="naRNA\nRPM across junctions", y="steady-state RNA\nRPM across junctions", color="Intron classification") +
  Rotate_x_labels

DummyForLegend <- Scatter.P.dat %>%
  ggplot(aes(x=RPM_chRNA.Expression.Splicing, y=RPM_Expression.Splicing)) +
  geom_density_2d(aes(color=SuperAnnotation)) +
  scale_color_manual(values=AnnotationColors %>% dplyr::select(SuperAnnotation, Color) %>% deframe(), na.value="#fb9a99", labels=SupergroupLabels) +
  theme(legend.position='bottom') +
  labs(color="Intron classification")
```

Save figures

```{r, eval=F}
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_Final_Fig_S_ScatterReasonsForJuncClassification_Final.png", Scatter.P, height=7, width=12)
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_Final_Fig_S_ScatterReasonsForJuncClassification_shRNA_Final.png", Scatter.P.shRNA, height=7, width=12)
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_Final_Fig_S_ScatterReasonsForJuncClassification_Legend.pdf", DummyForLegend, height=4, width=12)

  

```


Number eQTLs per gene, hQTLs, sQTLs

```{r}
dat <- fread("../code/pi1/PairwisePi1Traits.P.all.txt.gz")

dat$PC2 %>% unique()



NumMolQTLsPerGene <- dat %>%
  filter(PC1=="Expression.Splicing" & FDR.x < 0.1) %>%
  mutate(P2 = case_when(
    PC2 %in% c("polyA.Splicing", "chRNA.Splicing", "polyA.Splicing.Subset_YRI") ~ str_extract(P2, "clu_.+$"),
    TRUE ~ P2
  )) %>%
  group_by(PC1, PC2, P2, GeneLocus) %>%
  slice(which.min(FDR.y)) %>%
  ungroup() %>%
  mutate(IsPc2SigFDR = FDR.y < 0.1) %>%
  mutate(IsPc2SigPermutationP = p_permutation.y < 0.01) %>%
  group_by(PC2, GeneLocus) %>%
  summarise(Num_PC2_QTLs_FDR = sum(IsPc2SigFDR),
            Num_PC2_QTLs_PermutationP = sum(IsPc2SigPermutationP)) %>%
  ungroup()

NumMolQTLsPerGene %>%
  filter(PC2 %in% c("H3K27AC", "polyA.Splicing")) %>%
  ggplot(aes(x=Num_PC2_QTLs_PermutationP)) +
  stat_ecdf() +
  facet_wrap(~PC2) +
  coord_cartesian(xlim=c(0,5))

NumMolQTLsPerGene %>%
  filter(PC2 %in% c("H3K27AC", "polyA.Splicing")) %>%
  ggplot(aes(x=Num_PC2_QTLs_FDR)) +
  stat_ecdf() +
  facet_wrap(~PC2) +
  coord_cartesian(xlim=c(0,5))

NumMolQTLsPerGene %>%
  ggplot(aes(x=Num_PC2_QTLs_PermutationP)) +
  stat_ecdf() +
  facet_wrap(~PC2) +
  coord_cartesian(xlim=c(0,5))
```

Ok, now let's plot that nicer for saving...

```{r}
eGenes <- dat %>%
  filter(PC1=="Expression.Splicing" & FDR.x < 0.1) %>%
  pull(P1) %>% unique()

P.Number_hQTLsForColoc <- NumMolQTLsPerGene %>%
  filter(GeneLocus %in% eGenes) %>%
  # filter(PC2 %in% c("H3K27AC", "polyA.Splicing")) %>%
  filter(PC2=="H3K27AC") %>%
  mutate(n = if_else(Num_PC2_QTLs_PermutationP>5, as.integer(5), Num_PC2_QTLs_PermutationP)) %>%
  count(PC2, n) %>%
  ggplot(aes(x=n, y=nn)) +
  geom_col() +
  scale_x_continuous(breaks=0:5, labels=c(0:4, ">4")) +
  labs(x="Number H3K27AC hQTLs w/in 100kb of gene", y="Number eGenes")
NumMolQTLsPerGene %>%
  filter(GeneLocus %in% eGenes) %>%
  # filter(PC2 %in% c("H3K27AC", "polyA.Splicing")) %>%
  filter(PC2=="H3K27AC") %>%
  mutate(n = if_else(Num_PC2_QTLs_PermutationP>5, as.integer(5), Num_PC2_QTLs_PermutationP)) %>%
  count(PC2, n)

P.Number_sQTLsForColoc <- NumMolQTLsPerGene %>%
  filter(GeneLocus %in% eGenes) %>%
  # filter(PC2 %in% c("H3K27AC", "polyA.Splicing")) %>%
  filter(PC2=="polyA.Splicing") %>%
  mutate(n = if_else(Num_PC2_QTLs_PermutationP>5, as.integer(5), Num_PC2_QTLs_PermutationP)) %>%
  count(PC2, n) %>%
  ggplot(aes(x=n, y=nn)) +
  geom_col() +
  scale_x_continuous(breaks=0:5, labels=c(0:4, ">4")) +
  labs(x="Number sQTLs within gene", y="Number eGenes")
NumMolQTLsPerGene %>%
  filter(GeneLocus %in% eGenes) %>%
  # filter(PC2 %in% c("H3K27AC", "polyA.Splicing")) %>%
  filter(PC2=="polyA.Splicing") %>%
  mutate(n = if_else(Num_PC2_QTLs_PermutationP>5, as.integer(5), Num_PC2_QTLs_PermutationP)) %>%
  count(PC2, n)

P.Number_hQTLsForColoc
P.Number_sQTLsForColoc
```

Same plots, but only considering eGenes discovered in YRI subset

```{r}
eGenes <- dat %>%
  filter(PC1=="Expression.Splicing.Subset_YRI" & FDR.x < 0.1) %>%
  pull(P1) %>% unique()

P.Number_hQTLsForColoc <- NumMolQTLsPerGene %>%
  filter(GeneLocus %in% eGenes) %>%
  # filter(PC2 %in% c("H3K27AC", "polyA.Splicing")) %>%
  filter(PC2=="H3K27AC") %>%
  mutate(n = if_else(Num_PC2_QTLs_PermutationP>5, as.integer(5), Num_PC2_QTLs_PermutationP)) %>%
  count(PC2, n) %>%
  ggplot(aes(x=n, y=nn)) +
  geom_col() +
  scale_x_continuous(breaks=0:5, labels=c(0:4, ">4")) +
  scale_y_continuous(expand=c(0,0)) +
  labs(x="Number H3K27AC hQTLs w/in 100kb of gene", y="Number eGenes")
NumMolQTLsPerGene %>%
  filter(GeneLocus %in% eGenes) %>%
  # filter(PC2 %in% c("H3K27AC", "polyA.Splicing")) %>%
  filter(PC2=="H3K27AC") %>%
  mutate(n = if_else(Num_PC2_QTLs_PermutationP>5, as.integer(5), Num_PC2_QTLs_PermutationP)) %>%
  count(PC2, n)

P.Number_sQTLsForColoc <- NumMolQTLsPerGene %>%
  filter(GeneLocus %in% eGenes) %>%
  # filter(PC2 %in% c("H3K27AC", "polyA.Splicing")) %>%
  filter(PC2=="polyA.Splicing") %>%
  mutate(n = if_else(Num_PC2_QTLs_PermutationP>5, as.integer(5), Num_PC2_QTLs_PermutationP)) %>%
  count(PC2, n) %>%
  ggplot(aes(x=n, y=nn)) +
  geom_col() +
  scale_x_continuous(breaks=0:5, labels=c(0:4, ">4")) +
  scale_y_continuous(expand=c(0,0)) +
  labs(x="Number sQTLs within gene", y="Number eGenes")
NumMolQTLsPerGene %>%
  filter(GeneLocus %in% eGenes) %>%
  # filter(PC2 %in% c("H3K27AC", "polyA.Splicing")) %>%
  filter(PC2=="polyA.Splicing") %>%
  mutate(n = if_else(Num_PC2_QTLs_PermutationP>5, as.integer(5), Num_PC2_QTLs_PermutationP)) %>%
  count(PC2, n)

P.Number_hQTLsForColoc
P.Number_sQTLsForColoc
```


```{r}
Labels.df <- read_tsv("../data/Phenotypes_recode_for_Plotting.txt")

MolQTL_Labels <- Labels.df %>% dplyr::select(PC, Shortest) %>% deframe()
MolQTL_AssayLabels <- Labels.df %>% dplyr::select(PC, Assay) %>% deframe()
Factor_Order <- Labels.df %>% dplyr::select(PC, OrderFactor) %>% deframe()


coloc.dat <- read_tsv("../code/hyprcoloc/Results/ForColoc/MolColocNonRedundantFullSplicing/tidy_results_OnlyColocalized.txt.gz") %>%
  separate(phenotype_full, into=c("PC", "P"), sep=";")

Dat.TopQTLInFamily <- dat %>%
  filter(PC1=="Expression.Splicing" & FDR.x < 0.1) %>%
  mutate(P2 = case_when(
    PC2 %in% c("polyA.Splicing", "chRNA.Splicing", "polyA.Splicing.Subset_YRI") ~ str_extract(P2, "clu_.+$"),
    TRUE ~ P2
  )) %>%
  group_by(PC1, PC2, P2, GeneLocus) %>%
  slice(which.min(FDR.y)) %>%
  ungroup() %>%
  group_by(PC1, PC2, GeneLocus) %>%
  slice(which.min(FDR.y)) %>%
  ungroup()

coloc.dat %>%
  mutate(P = case_when(
    PC %in% c("polyA.Splicing", "chRNA.Splicing", "polyA.Splicing.Subset_YRI") ~ str_extract(P, "clu_.+$"),
    TRUE ~ P
  )) %>%
  left_join(
    Dat.TopQTLInFamily %>%
      dplyr::select(Locus=GeneLocus, PC=PC2, P=P2) %>%
      mutate(TopMolQTLAroundGene = T)
  ) %>%
  replace_na(list(TopMolQTLAroundGene=F)) %>%
  group_by(Locus, snp) %>%
  filter(any(PC=="Expression.Splicing.Subset_YRI")) %>%
  ungroup() %>%
  group_by(Locus, PC) %>%
  summarise(ContainsTopMolQTL = any(TopMolQTLAroundGene)) %>%
  ungroup() %>%
  group_by(PC) %>%
  summarise(FracContainsTop = sum(ContainsTopMolQTL)/n()) %>%
  ungroup()
```


Same analysis without renaming to match all in cluster

```{r}
Dat.TopQTLInFamily2 <- dat %>%
  filter(PC1=="Expression.Splicing" & FDR.x < 0.1) %>%
  # mutate(P2 = case_when(
  #   PC2 %in% c("polyA.Splicing", "chRNA.Splicing", "polyA.Splicing.Subset_YRI") ~ str_extract(P2, "clu_.+$"),
  #   TRUE ~ P2
  # )) %>%
  # group_by(PC1, PC2, P2, GeneLocus) %>%
  # slice(which.min(FDR.y)) %>%
  # ungroup() %>%
  group_by(PC1, PC2, GeneLocus) %>%
  slice(which.min(FDR.y)) %>%
  ungroup()

dat.FractionNonPrimaryMolQTLColocs <- coloc.dat %>%
  # mutate(P = case_when(
  #   PC %in% c("polyA.Splicing", "chRNA.Splicing", "polyA.Splicing.Subset_YRI") ~ str_extract(P, "clu_.+$"),
  #   TRUE ~ P
  # )) %>%
  left_join(
    Dat.TopQTLInFamily2 %>%
      dplyr::select(Locus=GeneLocus, PC=PC2, P=P2) %>%
      mutate(TopMolQTLAroundGene = T)
  ) %>%
  replace_na(list(TopMolQTLAroundGene=F)) %>%
  group_by(Locus, snp) %>%
  filter(any(PC=="Expression.Splicing.Subset_YRI")) %>%
  ungroup() %>%
  group_by(Locus, PC) %>%
  summarise(ContainsTopMolQTL = any(TopMolQTLAroundGene)) %>%
  ungroup() %>%
  group_by(PC) %>%
  summarise(FracContainsTop = sum(ContainsTopMolQTL)/n()) %>%
  ungroup()

kable(dat.FractionNonPrimaryMolQTLColocs)


P.NonPrimaryMolQTLs_OftenColoc <- dat.FractionNonPrimaryMolQTLColocs %>%
  filter(PC %in% c("H3K27AC", "polyA.Splicing")) %>%
  mutate(Complement = 1-FracContainsTop) %>%
  gather("ContainsTop", "Fraction",-PC) %>%
  ggplot(aes(x=PC, y=Fraction, fill=ContainsTop)) +
  geom_col(position="fill") +
  scale_y_continuous(limits=c(0, 1), expand=c(0,0)) +
  scale_fill_manual(values=c("FracContainsTop"="#636363", "Complement"="#f0f0f0"), labels=c("eGene colocalizes with top molQTL", "colocalizes with non top molQTL")) +
  scale_x_discrete(labels=c("H3K27AC hQTL", "polyA RNA sQTL")) +
  labs(x=NULL, y=str_wrap("Fraction of eQTL/molQTL colocalizations", 30), fill=NULL) +
  Rotate_x_labels

P.NonPrimaryMolQTLs_OftenColoc
```


- Number of eQTLs that coloc with at least one molQTL

```{r}
eGenes <- dat %>%
  filter(PC1=="Expression.Splicing.Subset_YRI" & FDR.x < 0.1) %>%
  pull(P1) %>% unique()

coloc.dat$PC %>% unique()


coloc.dat %>%
    group_by(Locus, snp) %>%
    filter(any(PC=="Expression.Splicing.Subset_YRI")) %>%
    ungroup() %>%
    count(Locus, PC) %>%
    count(PC) %>%
  mutate(PC_relabeled = recode(PC, !!!MolQTL_Labels)) %>%
  filter(!is.na(PC_relabeled) | PC=="Any hQTL") %>%
  filter(!PC_relabeled %in% c("Splicing", "Expression; eQTLs (YRI only)")) %>%
  mutate(Assay_relabeled = recode(PC, !!!MolQTL_AssayLabels)) %>%
  mutate(FinalLabels = paste(PC_relabeled, Assay_relabeled, sep="\n")) %>%
  mutate(OrderFactor = recode(PC, !!!Factor_Order)) %>%
  mutate(FinalLabels = fct_reorder(FinalLabels, OrderFactor)) %>%
  ggplot(aes(x=FinalLabels, y=n)) +
  geom_col() +
  scale_y_continuous(
    name="Number eGenes with colocalized molQTL", 
    sec.axis = sec_axis(~ . / length(eGenes), name = "% eGenes with colocalized molQTL")
  ) +
  Rotate_x_labels +
  labs(x="Molecular QTL type, Assay")


dat.eGeneColocsWith <- bind_rows(
  coloc.dat %>%
    group_by(Locus, snp) %>%
    filter(any(PC=="Expression.Splicing.Subset_YRI")) %>%
    ungroup() %>%
    count(Locus, PC) %>%
    count(PC),
  coloc.dat %>%
    mutate(PC = case_when(
      PC %in% c("H3K4ME1", "H3K4ME3", "H3K27AC", "H3K36ME3") ~ "Any hQTL",
      TRUE ~ PC
    )) %>%
    group_by(Locus, snp) %>%
    filter(any(PC=="Expression.Splicing.Subset_YRI")) %>%
    ungroup() %>%
    count(Locus, PC) %>%
    count(PC)
) %>%
  mutate(PC_relabeled = recode(PC, !!!MolQTL_Labels)) %>%
  filter(!is.na(PC_relabeled) | PC=="Any hQTL") %>%
  filter(!PC_relabeled %in% c("Splicing", "Expression; eQTLs (YRI only)")) %>%
  mutate(Assay_relabeled = recode(PC, !!!MolQTL_AssayLabels)) %>%
  mutate(FinalLabels = paste(PC_relabeled, Assay_relabeled, sep="\n")) %>%
  mutate(OrderFactor = recode(PC, !!!Factor_Order))

P.eGeneColocsWith <- dat.eGeneColocsWith %>%
  mutate(FinalLabels = str_remove_all(FinalLabels, "\\*")) %>%
  mutate(FinalLabels = case_when(
    FinalLabels == "Any hQTL\nAny hQTL" ~ "Any hQTL*",
    TRUE ~ FinalLabels
  )) %>%
  mutate(OrderFactor = case_when(
    FinalLabels == "Any hQTL*" ~ 4.5,
    TRUE ~ OrderFactor
  )) %>%
  mutate(FinalLabels = fct_reorder(FinalLabels, OrderFactor)) %>%
  distinct(.keep_all=T) %>%
  ggplot(aes(x=FinalLabels, y=n)) +
  geom_col() +
  scale_y_continuous(
    name="Number eGenes with colocalized molQTL", 
    sec.axis = sec_axis(~ . / length(eGenes), name = "% eGenes with colocalized molQTL")
  ) +
  Rotate_x_labels +
  labs(x="Molecular QTL type, Assay")
P.eGeneColocsWith

dat.eGeneColocsWith
```


- Number of eQTLs that coloc with at least one molQTL (among non hQTL eQTLs)

```{r}
eGenes.NohQTLColoc <-
  coloc.dat %>%
  group_by(Locus, snp) %>%
  filter(any(PC=="Expression.Splicing.Subset_YRI") & !any(PC %in% c("H3K4ME3", "H3K27AC", "H3K4ME1", "H3K36ME3"))) %>%
  pull(Locus) %>% unique()

dat.eGeneColocsWith.NohQTL <-
  coloc.dat %>%
  group_by(Locus, snp) %>%
  filter(any(PC=="Expression.Splicing.Subset_YRI") & !any(PC %in% c("H3K4ME3", "H3K27AC", "H3K4ME1", "H3K36ME3"))) %>%
  ungroup() %>%
  count(Locus, PC) %>%
  count(PC) %>%
  mutate(PC_relabeled = recode(PC, !!!MolQTL_Labels)) %>%
  filter(!is.na(PC_relabeled) | PC=="Any hQTL") %>%
  filter(!PC_relabeled %in% c("Splicing", "Expression; eQTLs (YRI only)")) %>%
  mutate(Assay_relabeled = recode(PC, !!!MolQTL_AssayLabels)) %>%
  mutate(FinalLabels = paste(PC_relabeled, Assay_relabeled, sep="\n")) %>%
  mutate(OrderFactor = recode(PC, !!!Factor_Order))

P.eGeneColocsWith.nohQTL <- dat.eGeneColocsWith.NohQTL %>%
  mutate(FinalLabels = str_remove_all(FinalLabels, "\\*")) %>%
  mutate(FinalLabels = case_when(
    FinalLabels == "Any hQTL\nAny hQTL" ~ "Any hQTL*",
    TRUE ~ FinalLabels
  )) %>%
  mutate(OrderFactor = case_when(
    FinalLabels == "Any hQTL*" ~ 4.5,
    TRUE ~ OrderFactor
  )) %>%
  distinct(.keep_all=T) %>%
  mutate(FinalLabels = fct_reorder(FinalLabels, OrderFactor)) %>%
  ggplot(aes(x=FinalLabels, y=n)) +
  geom_col() +
  scale_y_continuous(
    name="Number eGenes with colocalized molQTL", 
    sec.axis = sec_axis(~ . / length(eGenes.NohQTLColoc), name = "% eGenes with colocalized molQTL\nAmong those that did not coloc with hQTL")
  ) +
  Rotate_x_labels +
  labs(x="Molecular QTL type, Assay")
P.eGeneColocsWith.nohQTL

dat.eGeneColocsWith.NohQTL


```


- Plot of number of eQTLs that coloc with an hQTL, versus those that coloc with an sQTL (and not an hQTL)

```{r}




data.frame(name="Non hQTL", n=length(eGenes.NohQTLColoc))


dat.SimplifiedNumberColocs <- bind_rows(
  dat.eGeneColocsWith %>%
    filter(PC=="Any hQTL") %>%
    mutate(name="hQTL*", order=1) %>%
    dplyr::select(name, n, order),
  # dat.eGeneColocsWith %>%
  #   filter(PC=="polyA.Splicing") %>%
  #   mutate(name="sQTL", order=1.5) %>%
  #   dplyr::select(name, n, order),
  dat.eGeneColocsWith.NohQTL %>%
    filter(PC=="APA_Total") %>%
    mutate(name="APA QTL**", order=4) %>%
    dplyr::select(name, n, order),
  dat.eGeneColocsWith.NohQTL %>%
    filter(PC=="polyA.Splicing") %>%
    mutate(name="sQTL**", order=3) %>%
    dplyr::select(name, n, order),
) %>%
  add_row(name="Any molQTL**", n=length(eGenes.NohQTLColoc), order=2) %>%
  mutate(name = fct_reorder(name, order))

P.SimplifiedNumberColocs <- ggplot(dat.SimplifiedNumberColocs, aes(x=name, y=n)) +
  geom_col(fill="#bdbdbd") +
  scale_y_continuous(expand=c(0,0), breaks=c(0, 200, 400, 600, 800), labels=c(0, "", "", "", 800), sec.axis = sec_axis( trans=~./(831/(831+518))*100, name="Percent eQTLs with a molQTLcolocalization")) +
  geom_text(aes(label=name), y=10, angle=90, y=-Inf, hjust=0, size=8) +
  labs(x=NULL, y="# eGenes coloc w/ molQTL") +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())
P.SimplifiedNumberColocs

kable(dat.SimplifiedNumberColocs)
```


- And finally, a plot about whether the sQTLs that do coloc with eQTL are enriched for non-productive juncs.
```{r}
IntronAnnotations <- read_tsv("../data/IntronAnnotationsFromYang.tsv.gz")

ClusterLevelQTLAnnotations <- read_delim("../code/QTLs/QTLTools/polyA.Splicing/PermutationPassForColoc.txt.gz", delim=' ') %>%
  mutate(IntronID = str_replace(phe_id, "(^.+?clu.+?):EN.+$", "\\1")) %>%
  separate(IntronID, into=c("chrom", "start", "end", "cluster"), remove=F, convert=T, sep=":") %>%
  mutate(chrom=paste0("chr", chrom)) %>%
  mutate(q = qvalue(adj_beta_pval)$qvalues) %>%
  left_join(IntronAnnotations) %>%
  group_by(cluster) %>%
  mutate(ContainsUnproductive = any(SuperAnnotation %in% c("AnnotatedJunc_UnproductiveCodingGene", "UnannotatedJunc_UnproductiveCodingGene") & q<0.1)) %>%
  ungroup()

ClusterLevelQTLAnnotations %>%
  distinct(cluster, .keep_all=T) %>%
  count(ContainsUnproductive)

dat %>%
  filter(PC1=="Expression.Splicing" & FDR.x < 0.1) %>%
  filter(PC2=="polyA.Splicing" & p_permutation.y<0.01)


Clusters.ThatColocWith_sQTL_not_hQTL <- coloc.dat %>%
  group_by(Locus, snp) %>%
  filter(any(PC=="Expression.Splicing.Subset_YRI") & !any(PC %in% c("H3K4ME3", "H3K27AC", "H3K4ME1", "H3K36ME3"))) %>%
  ungroup() %>%
  filter(PC=="polyA.Splicing") %>%
  separate(P, into=c("chrom", "start", "end", "cluster"), remove=F, convert=T, sep=":") %>%
  distinct(cluster)

Clusters.AttemptColoc_butOther <- dat %>%
  filter(PC1=="Expression.Splicing" & FDR.x < 0.1) %>%
  filter(PC2=="polyA.Splicing" & p_permutation.y<0.01) %>%
  distinct(P2) %>%
  separate(P2, into=c("chrom", "start", "end", "cluster"), remove=F, convert=T, sep=":") %>%
  distinct(cluster) %>%
  filter(!cluster %in% Clusters.ThatColocWith_sQTL_not_hQTL$cluster)

dat.EnrichmentOfUnrpductiveClusters <- bind_rows(
  Clusters.AttemptColoc_butOther %>%
    mutate(Category="DoesNotColocWith_eGene"),
  Clusters.ThatColocWith_sQTL_not_hQTL %>%
    mutate(Category="DoesColocWith_eGene")
) %>%
  left_join(
    ClusterLevelQTLAnnotations %>%
      distinct(cluster, ContainsUnproductive))

test.results <- dat.EnrichmentOfUnrpductiveClusters %>%
  count(Category, ContainsUnproductive) %>%
  pivot_wider(names_from="Category", values_from ="n") %>%
  mutate(ContainsUnproductive=!ContainsUnproductive) %>%
  column_to_rownames("ContainsUnproductive") %>%
  as.matrix() %>%
  fisher.test(alternative="less")
test.results

P.EnrichmentOfUnrpductiveClusters <- dat.EnrichmentOfUnrpductiveClusters %>%
  count(Category, ContainsUnproductive) %>%
  ggplot(aes(x=Category, fill=ContainsUnproductive, y=n)) +
  geom_col() +
  scale_fill_manual(values=c("FALSE"="#377eb8", "TRUE"="#e41a1c"), labels=c("FALSE"="Productive sQTL", "TRUE"="Unproductive sQTL")) +
  scale_x_discrete(labels=c(str_wrap("sQTLs that coloc with eQTL and not hQTL", 20), str_wrap("Other sQTLs tested\nfor colocalization", 20))) +
  labs(caption=paste("One sided Fisher test P:", format.pval(test.results$p.value, 3), "\nOR:", round(1/test.results$estimate, 3)), y="Number of sQTLs", fill="sQTL type", x="") +
  Rotate_x_labels
P.EnrichmentOfUnrpductiveClusters
```

redo above simpler...

```{r}

### LEFT HERE
sQTL.type <- read_tsv("../code/SplicingAnalysis/sQTLs_p_and_u.Full.tsv.gz") %>%
    mutate(ClusterName =  str_replace(P1, "^(.+?:).+?(clu_.+?[+-])$", "chr\\1\\2"))

coloc.dat %>%
  group_by(snp, Locus) %>%
  filter(any(PC=="polyA.Splicing") & any(PC=="Expression.Splicing.Subset_YRI") & !any(PC %in% c("H3K4ME3", "H3K27AC", "H3K4ME1"))) %>%
  ungroup() %>%
  filter(PC=="polyA.Splicing") %>%
  mutate(ClusterName =  str_replace(P, "^(.+?:).+?(clu_.+?[+-])$", "chr\\1\\2")) %>%
  distinct(ClusterName, .keep_all=T) %>%
  inner_join(
    sQTL.type %>%
      distinct(ClusterName, .keep_all=T)
  ) %>%
  count(sQTL.type)

sQTL.type %>%
      distinct(ClusterName, .keep_all=T) %>%
  count(sQTL.type)

```


Plot about primary vs non primary sQTLs for productive vs unproductive

```{r}

dat.EnrichmentOfUnrpductiveClusters_AmongstTop <- dat %>%
  filter(PC1=="Expression.Splicing" & FDR.x < 0.1) %>%
  filter(PC2=="polyA.Splicing" & p_permutation.y<0.01) %>%
  separate(P2, into=c("chrom", "start", "end", "cluster"), remove=F, convert=T, sep=":") %>%
  group_by(cluster, GeneLocus) %>%
  summarise(cluster.minP=min(p_permutation.y)) %>%
  ungroup() %>%
  group_by(GeneLocus) %>%
  mutate(Category = cluster.minP==min(cluster.minP)) %>%
  ungroup() %>%
  left_join(
    ClusterLevelQTLAnnotations %>%
      distinct(cluster, ContainsUnproductive))

test.results <- dat.EnrichmentOfUnrpductiveClusters_AmongstTop %>%
  count(Category, ContainsUnproductive) %>%
  pivot_wider(names_from="Category", values_from ="n") %>%
  mutate(ContainsUnproductive=!ContainsUnproductive) %>%
  column_to_rownames("ContainsUnproductive") %>%
  as.matrix() %>%
  fisher.test(alternative="less")
test.results

P.EnrichmentOfUnrpductiveClusters_AmongstTop <- dat.EnrichmentOfUnrpductiveClusters_AmongstTop %>%
  count(Category, ContainsUnproductive) %>%
  ggplot(aes(x=Category, fill=ContainsUnproductive, y=n)) +
  geom_col() +
  scale_fill_manual(values=c("FALSE"="#377eb8", "TRUE"="#e41a1c"), labels=c("FALSE"="Productive sQTL", "TRUE"="Unproductive sQTL")) +
  scale_x_discrete(labels=c("TRUE"=str_wrap("Top sQTL per gene", 20), "FALSE"=str_wrap("Not top sQTL per gene", 20))) +
  labs(caption=paste("One sided Fisher test P:", format.pval(test.results$p.value, 3), "\nOR:", round(1/test.results$estimate, 3)), y="Number of sQTLs", fill="sQTL type", x="") +
  Rotate_x_labels
P.EnrichmentOfUnrpductiveClusters_AmongstTop
```

sQTLs enriched for eQTL signal, qq plot

```{r}
PC2.filter <- c("Expression.Splicing", "Expression.Splicing.Subset_YRI")
PC2.SignificanceFilter <- c("H3K27AC", "H3K4ME3", "H3K36ME3")
PC1.filter <- c("polyA.Splicing", "H3K27AC", "H3K4ME1", "H3K4ME3")
PC1.filter.Splicing <- PC1.filter[str_detect(PC1.filter, "Splicing")]
PC1.filter.NonSplicing <- PC1.filter[!str_detect(PC1.filter, "Splicing")]

dat.foreQTLQQ <- fread("../code/pi1/PairwisePi1Traits.P.all.txt.gz") %>%
  filter((PC1 %in% PC1.filter))

IntronAnnotatins <- read_tsv("../data/IntronAnnotationsFromYang.tsv.gz") %>%
  mutate(chrom = str_remove_all(chrom, "chr")) %>%
  mutate(Intron = paste(chrom, start, end, sep=":")) %>%
  filter(!str_detect(SuperAnnotation, "NoncodingGene"))

IntronAnnotatins$SuperAnnotation %>% unique()



RandomTestSNPs <- paste0("../code/QTLs/QTLTools/", PC2.filter, "/NominalPassForColoc.RandomSamplePvals.txt.gz") %>%
  setNames(PC2.filter) %>%
  lapply(fread) %>%
  bind_rows(.id="PC2") %>%
  dplyr::select(trait.x.p.in.y=V1, PC2) %>%
  mutate(SNP_group="Random test SNPs") %>%
  group_by(PC2) %>%
  sample_n(50000)


dat.foreQTLQQ.sQTLs <- dat.foreQTLQQ %>%
    filter(PC1 %in% PC1.filter.Splicing) %>%
    group_by(PC1, P1) %>%
    filter(!any((PC2 %in% PC2.SignificanceFilter) & (trait.x.p.in.y < 0.01))) %>%
    ungroup() %>%
    filter(PC2 %in% PC2.filter) %>%
    separate(P1, into=c("Intron", "Cluster"), sep=":clu", remove=F) %>%
    inner_join(
      IntronAnnotatins %>%
        dplyr::select(Intron, SuperAnnotation),
      by="Intron") %>%
    group_by(PC1, PC2, Cluster) %>%
    mutate(SNP_group = case_when(
      all(str_detect(SuperAnnotation, "Productive")) ~ "Productive sQTL cluster",
      any(str_detect(SuperAnnotation, "Unproductive")) ~ "Unproductive sQTL cluster",
      TRUE ~ "sQTL Other"
    )) %>%
    # sample_n(1) %>%
    slice(which.min(p_permutation.x)) %>%
    ungroup()

bind_rows(
  dat.foreQTLQQ.sQTLs,
  dat.foreQTLQQ %>%
    filter(PC2 %in% PC2.filter) %>%
    filter(PC1 %in% PC1.filter.NonSplicing) %>%
    mutate(SNP_group = PC1),
  RandomTestSNPs
) %>%
  group_by(SNP_group, PC2) %>%
  mutate(MyRank = rank(trait.x.p.in.y, ties.method='random')) %>%
  mutate(ExpectedP = MyRank/(max(MyRank) + 1)) %>%
  ungroup() %>%
  ggplot(aes(x=-log10(ExpectedP), y=-log10(trait.x.p.in.y), color=SNP_group)) +
  geom_abline(slope=1, intercept=0) +
  geom_point() +
  facet_wrap(~PC2, scales="free") +
  labs(y="-log10(ObservedP)", caption="sQTL clusters containing any significant unproductive sQTL intron are grouped as Unproductive sQTL\nP value from only one significant sQTL intron from cluster selected at random from cluster\nsQTLs with nominal P<0.01 for H3K36me3, H3K27Ac, or H3K4me3 removed", title="eQTL QQ plot")


dat.foreQTLQQ.sQTLs %>%
  count(SNP_group, PC2)

# pi0est function breaks sometimes. this wrapper function returns 1 when that happens
CalculatePi1_2 <- function (dat.in, ...) {
  return(tryCatch(1-pi0est(dat.in$trait.x.p.in.y, ...)$pi0, error=function(e) 1))
}

labels <- dat.foreQTLQQ.sQTLs %>%
  split(paste(.$SNP_group, .$PC2, sep=";")) %>%
  lapply(CalculatePi1_2, pi0.method="bootstrap") %>%
  unlist() %>%
  data.frame(pi1=.) %>%
  rownames_to_column("sQTLGroup_eQTLDataset") %>%
  separate(sQTLGroup_eQTLDataset, into=c("SNP_group", "PC2"), sep=";") %>%
  mutate(label=paste0("pi=", signif(pi1, digits=3)))

  
eQTL.P.sQTLs.hist <- dat.foreQTLQQ.sQTLs %>%
  ggplot(aes(x=trait.x.p.in.y)) +
  geom_histogram() +
  geom_text(data = labels, aes(label=label), x=Inf, y=Inf, vjust=1.5, hjust=1) +
  scale_x_continuous(breaks=c(0,0.25,0.5,0.75,1), labels=c(0,0.25, 0.5, 0.75, 1)) +
  facet_grid(SNP_group ~ PC2) +
  labs(x="P", title="eQTL P value historgram", caption="Vertical facets: eQTL dataset\nHorizontal facets: sQTL cluster type") +
  theme(strip.text = element_text(size = 8)) +
  Rotate_x_labels
eQTL.P.sQTLs.hist

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/20230906_eQTL_histogram_bysQTLType.pdf", eQTL.P.sQTLs.hist, width=4, height=4)

```

save a qqplot For lab meeting...

```{r}
eQTL.QQ <- bind_rows(
  dat.foreQTLQQ.sQTLs,
  dat.foreQTLQQ %>%
    filter(PC2 %in% PC2.filter) %>%
    filter(PC1 %in% PC1.filter.NonSplicing) %>%
    mutate(SNP_group = PC1),
  RandomTestSNPs
) %>%
  group_by(SNP_group, PC2) %>%
  mutate(MyRank = rank(trait.x.p.in.y, ties.method='random')) %>%
  mutate(ExpectedP = MyRank/(max(MyRank) + 1)) %>%
  ungroup() %>%
  filter(PC2=="Expression.Splicing") %>%
  filter(!PC1 %in% c("H3K4ME1", "H3K4ME3")) %>%
  ggplot(aes(x=-log10(ExpectedP), y=-log10(trait.x.p.in.y), color=SNP_group)) +
  geom_abline(slope=1, intercept=0) +
  geom_point() +
  scale_color_manual(values=
                       c("Random test SNPs"="#000000", "Unproductive sQTL cluster"="#e31a1c", "Productive sQTL cluster"="#1f78b4", "H3K27AC"="#6a3d9a"),
                     labels=c("Random test SNPs"="Random test SNPs", "Unproductive sQTL cluster"="Unproductive sQTL", "Productive sQTL cluster"="Productive sQTL", "H3K27AC"="H3K27AC QTL w/in 100kb")) +
  labs(y="-log10(ObservedP)", caption="sQTL clusters containing any significant unproductive sQTL intron are grouped as Unproductive sQTL\nP value from only one significant sQTL intron from cluster selected at random from cluster\nsQTLs with nominal P<0.01 for H3K36me3, H3K27Ac, or H3K4me3 removed", title="eQTL QQ plot") +
  theme(plot.subtitle=element_text(size=5))
eQTL.QQ



```

Plot about enrichment of finemap signals over genomic features...

```{r}
dat.enrichment <- read_tsv("../code/QTL_SNP_Enrichment/ResultsFromColocFinemap/GeneSet_GenewiseColoc_ChromatinAPAAndRNA_Finemap_GenewiseColoc_ChromatinAPAAndRNA/dat.tsv.gz")

dat.enrichment %>%
  distinct(Annotation)

P.SNPEnrichment <- dat.enrichment %>%
  filter(!startsWith(Annotation, "ncRNA") ) %>%
  filter(!Annotation==".") %>%
  group_by(Annotation) %>%
  summarise(Lower=quantile(Enrichment,probs=0.025, na.rm=T),
            Upper=quantile(Enrichment,probs=0.975, na.rm=T),
            Middle=quantile(Enrichment,probs=0.5, na.rm=T)) %>%
  ungroup() %>%
  mutate(IsSig = if_else(data.table::between(0, Lower, Upper), "Not significant", "Significant")) %>%
  mutate(Color = case_when(
    str_detect(Annotation, "^\\d.+Enhancer$") ~ "ChromHMM enhancer",
    str_detect(Annotation, "^\\d.+Promoter$") ~ "ChromHMM promoter",
    str_detect(Annotation, "^\\d.+Txn.*") ~ "ChromHMM transcribed region",
    str_detect(Annotation, "^Splice") ~ "Splice sites",
    # str_detect(Annotation, "miRNA_BS") ~ "miRNA binding sites",
    TRUE ~ "Other"
  )) %>%
  mutate(Annotation = if_else(str_detect(Annotation, "^\\d"), paste0("ChromHMM_",Annotation), Annotation)) %>%
  mutate(Annotation = recode(Annotation,
                             "miRNA_BS"= "miRNA binding sites",
                             "PAS_Region"="Polyadenylation site regions",
                             "SpliceAcceptor_0"="Splice acceptors (unannotated)",
                             "SpliceAcceptor_1"="Splice acceptors (annotated)",
                             "SpliceBranchpointRegion_0"="Splice branchpoint region (unannotated acceptor)",
                             "SpliceBranchpointRegion_1"="Splice branchpoint region (annotated acceptor)",
                             "SpliceDonor_0"="Splice donors (unannotated)",
                             "SpliceDonor_1"="Splice donors (annotated)"
                             )) %>%
  mutate(Annotation = fct_reorder(Annotation, Middle)) %>%
  ggplot(aes(x=Annotation)) +
  geom_col(aes(y=Middle, fill=Color)) +
  geom_errorbar(aes(ymin=Lower, ymax=Upper)) +
  scale_fill_brewer(palette = "Accent") +
  facet_wrap(~IsSig, scales="free", nrow=2) +
  theme_bw() +
  Rotate_x_labels +
  labs(y="log2 Fold Enrichment\nof PIP in genomic annotation class", fill="Type of\ngenomic annotation") +
  coord_flip()

P.SNPEnrichment

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/20230905_SNP_enrichment.pdf", P.SNPEnrichment, width=7, height=7)
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/20230905_EnrichmentOf_usQTLsInColocs.pdf", P.EnrichmentOfUnrpductiveClusters, width=5, height=6)



```


### Update for reviews...

The reviewer noticed that APA has stronger enrichment than splicing, so was left wondering what to think since in the text we note that there are more u-sQTL post-txn eQTL colocalizations than APA-QTL post-txn eQTL colocalizations... I think this is natural since there are just more splice sites than APA sites, which get's lost when we calculate the enrichment as the PIP mass over the total feature set bp space. So for the reviewer, let's make another version of this plot without the normalization by total feature space.

```{r}
featureSets <- fread("../code/QTL_SNP_Enrichment/FinemapIntersections/GenewiseColoc_ChromatinAPAAndRNA.bed.gz", col.names=c("chrom", "start", "stop", "name", "score", "featChrom", "featStart", "featStop", "Annotation", "Overlap"))

featureSets %>%
  group_by(Annotation) %>%
  summarise(featLen = sum(featStop - featStart)) %>%
  ungroup() %>%
  ggplot(aes(y=Annotation, x=featLen)) +
  geom_col() +
  scale_x_continuous(trans='log10')

dat.enrichment.forreview <- dat.enrichment %>%
  filter(!startsWith(Annotation, "ncRNA") ) %>%
  filter(!Annotation==".") %>%
  pivot_longer(names_to = "eQTLType", values_to = "PIP_mass", c("hQTL_eGenes", "Other_eGenes")) %>%
  group_by(eQTLType, BootstrapRep) %>%
  mutate(TotalPIPMass = sum(PIP_mass)) %>%
  ungroup() %>%
  group_by(Annotation, eQTLType) %>%
  summarise(Lower=quantile(PIP_mass/TotalPIPMass*100,probs=0.025, na.rm=T),
            Upper=quantile(PIP_mass/TotalPIPMass*100,probs=0.975, na.rm=T),
            Middle=quantile(PIP_mass/TotalPIPMass*100,probs=0.5, na.rm=T)) %>%
  ungroup() %>%
   mutate(Color = case_when(
    str_detect(Annotation, "^\\d.+Enhancer$") ~ "ChromHMM enhancer",
    str_detect(Annotation, "^\\d.+Promoter$") ~ "ChromHMM promoter",
    str_detect(Annotation, "^\\d.+Txn.*") ~ "ChromHMM transcribed region",
    str_detect(Annotation, "^Splice") ~ "Splice sites",
    # str_detect(Annotation, "miRNA_BS") ~ "miRNA binding sites",
    TRUE ~ "Other"
  )) %>%
  mutate(Annotation = if_else(str_detect(Annotation, "^\\d"), paste0("ChromHMM_",Annotation), Annotation)) %>%
  mutate(Annotation = recode(Annotation,
                             "miRNA_BS"= "miRNA binding sites",
                             "PAS_Region"="Polyadenylation site regions",
                             "SpliceAcceptor_0"="Splice acceptors (unannotated)",
                             "SpliceAcceptor_1"="Splice acceptors (annotated)",
                             "SpliceBranchpointRegion_0"="Splice branchpoint region (unannotated acceptor)",
                             "SpliceBranchpointRegion_1"="Splice branchpoint region (annotated acceptor)",
                             "SpliceDonor_0"="Splice donors (unannotated)",
                             "SpliceDonor_1"="Splice donors (annotated)"
                             )) %>%
  mutate(eQTLType = factor(eQTLType, levels=c("hQTL_eGenes", "Other_eGenes"))) %>%
  mutate(Annotation = fct_reorder(Annotation, Middle))

P.PIP_Mass <- ggplot(dat.enrichment.forreview, aes(x=Annotation, fill=Color, group=eQTLType)) +
  geom_col(aes(y=Middle, color=eQTLType), position=position_dodge(), width=0.8) +
  geom_errorbar(aes(ymin=Lower, ymax=Upper, fill=Color, color=eQTLType),position=position_dodge(0.9), color='black', width=.4) +
  scale_fill_brewer(palette = "Accent") +
  theme_bw() +
  Rotate_x_labels +
  labs(y="sum of PIP mass in genomic annotation class", fill="Type of\ngenomic annotation") +
  coord_flip()
P.PIP_Mass

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/20240605_PIP_Mass.pdf", P.PIP_Mass, width=7, height=7.5)

P.PIP_Mass.subset <- dat.enrichment.forreview %>%
  filter(Color == "Splice sites" | Annotation=="Polyadenylation site regions") %>%
  ggplot(aes(x=Annotation, fill=Color, group=eQTLType)) +
  geom_col(aes(y=Middle, fill=eQTLType), position=position_dodge(), width=0.8) +
  geom_errorbar(aes(ymin=Lower, ymax=Upper, fill=eQTLType),position=position_dodge(0.9), color='black', width=.4) +
  scale_fill_manual(values=c("hQTL_eGenes"="#6a3d9a", "Other_eGenes"="#969696"), labels=c("hQTL_eGenes"="transcriptionally-regulated eQTLs", "Other_eGenes"="post-transcriptional\neQTLs"), name=NULL) +
  theme_bw() +
  Rotate_x_labels +
  labs(y="percent of PIP mass in genomic annotation class", fill="Type of\ngenomic annotation") +
  coord_flip()
P.PIP_Mass.subset
```

```{r, eval=F}
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/20240605_PIP_Mass.Subset.pdf", P.PIP_Mass, width=8, height=3)
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/20240605_PIP_Mass.Subset.pdf", P.PIP_Mass.subset, width=8, height=3)
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/20240605_PIP_Mass.Subset.png", P.PIP_Mass.subset, width=8, height=3)

```


beta beta scatters

```{r}
PC1.PossibleValues <- c("polyA.Splicing", "polyA.Splicing.Subset_YRI", "chRNA.Splicing")
PC2.PossibleValues <-c("Expression.Splicing", "Expression.Splicing.Subset_YRI", "H3K4ME3", "H3K36ME3", "H3K27AC", "H3K4ME1", "MetabolicLabelled.30min", "MetabolicLabelled.60min", "chRNA.Expression.Splicing")


dat <- fread("../code/pi1/PairwisePi1Traits.P.all.txt.gz") %>%
  filter((PC1 %in% PC1.PossibleValues) & (PC2 %in% PC2.PossibleValues))

Intron.Annotations <- read_tsv("../data/IntronAnnotationsFromYang.tsv.gz") %>%
  mutate(IntronName = paste(chrom, start, end, strand, sep=":"))

dat.sQTLs.eQTLs.ForScatter <- dat %>%
  mutate(IntronName = str_replace(P1, "^(.+?:)clu_.+?([+-])$", "chr\\1\\2")) %>%
  mutate(ClusterName =  str_replace(P1, "^(.+?:).+?(clu_.+?[+-])$", "chr\\1\\2")) %>%
  left_join(Intron.Annotations)

PC1.filter = c("polyA.Splicing")
PC2.filter = c("H3K36ME3", "chRNA.Expression.Splicing" , "MetabolicLabelled.30min", "MetabolicLabelled.60min", "Expression.Splicing", "Expression.Splicing.Subset_YRI")
PC2.SignificanceFilter <- c("H3K4ME3", "H3K27AC", "H3K36ME3")

Intron.Annotations$SuperAnnotation %>% unique()

dat.sQTLs.eQTLs.ForScatter.ToPlot <- dat.sQTLs.eQTLs.ForScatter %>%
  filter(PC1 %in% PC1.filter) %>%
  group_by(P1) %>%
  filter(!any((PC2 %in% PC2.SignificanceFilter) & (trait.x.p.in.y < 0.01))) %>%
  ungroup() %>%
  filter(PC2 %in% PC2.filter) %>%
  # pull(PC2) %>% unique()
  filter(SuperAnnotation %in% c("UnannotatedJunc_UnproductiveCodingGene", "AnnotatedJunc_ProductiveCodingGene", "AnnotatedJunc_UnproductiveCodingGene", "UnannotatedJunc_ProductiveCodingGene")) %>%
  group_by(PC1, ClusterName) %>%
  filter(abs(beta.x) == max(abs(beta.x))) %>%
  ungroup() %>%
  mutate(SuperAnnotation = str_replace_all(SuperAnnotation, "_", " ")) %>%
  filter(GeneLocus == gene) %>%
  filter(FDR.x < 0.1) %>%
  group_by(P2, PC2, SuperAnnotation) %>%
  sample_n(1) %>%
  ungroup() %>%
  mutate(PC2 = factor(PC2, levels=PC2.filter)) %>%
  mutate(PC2 = recode(PC2, !!!c("chRNA.Expression.Splicing"="chRNA" , "MetabolicLabelled.30min"="30min 4sU", "MetabolicLabelled.60min"="60min 4sU", "Expression.Splicing"="polyA", "Expression.Splicing.Subset_YRI"="polyA YRI"))) %>%
  mutate(PC2 = factor(PC2, levels=c("H3K36ME3","chRNA", "30min 4sU", "60min 4sU", "polyA", "polyA YRI"))) %>%
  filter(!PC2 %in% c("polyA YRI", "H3K36ME3")) %>%
  # pull(SuperAnnotation) %>% unique()
  mutate(SuperAnnotation = recode(SuperAnnotation, !!!c("AnnotatedJunc UnproductiveCodingGene"="Unproductive", "UnannotatedJunc UnproductiveCodingGene"="Unproductive", "AnnotatedJunc ProductiveCodingGene"="Productive", "UnannotatedJunc ProductiveCodingGene"="Productive")))

dat.sQTLs.eQTLs.ForScatter.ToPlot.labels <- dat.sQTLs.eQTLs.ForScatter.ToPlot %>%
  count(SuperAnnotation, PC2) %>%
  distinct(SuperAnnotation, .keep_all=T) %>%
  dplyr::select(n, SuperAnnotation) %>%
  mutate(SuperAnnotationLabel = paste0(SuperAnnotation, " (n=", n, ")"))

library(broom)


sQTL.eQTL.beta.beta <- dat.sQTLs.eQTLs.ForScatter.ToPlot %>%
  left_join(dat.sQTLs.eQTLs.ForScatter.ToPlot.labels, by="SuperAnnotation") %>%
  nest(-SuperAnnotationLabel, -PC2) %>% 
  mutate(cor=map(data,~cor.test(.x$beta.x, .x$x.beta.in.y, method = "sp"))) %>%
  mutate(tidied = map(cor, tidy)) %>% 
  unnest(tidied, .drop = T) %>%
  unnest(data) %>%
ggplot(aes(x=beta.x, y=x.beta.in.y, color=SuperAnnotation)) +
    geom_point(alpha=0.1) +
    # geom_smooth(method = "tls", se = FALSE, color = "red", method='bootstrap') +
    # geom_smooth(method='lm', color='black') +
    geom_vline(xintercept=0, linetype='dashed') +
    geom_hline(yintercept=0, linetype='dashed') +
    scale_color_manual(values=c("Productive"="#1f78b4", "Unproductive"="#e31a1c")) +
    geom_abline(data = . %>%
                  distinct(SuperAnnotationLabel, PC2, .keep_all=T),
                aes(slope=estimate, intercept=0), size=2) +
    geom_text(
    data = . %>%
        distinct(SuperAnnotationLabel, PC2, .keep_all=T) %>%
        mutate(R=signif(estimate, 3), P=format.pval(p.value, 3)) %>%
      mutate(label = str_glue("rho:{R}\nP:{P}")),
    aes(x=-Inf, y=-Inf, label=label),
    hjust=-.1, vjust=-0.1, color='black', size=6
    ) +
    facet_grid(SuperAnnotationLabel ~ PC2, labeller = label_wrap_gen(10)) +
    theme(strip.text = element_text(size = 12), legend.position='none') +
  labs(caption = "FDR<10% sQTLs. sQTLs with nominal P<0.01 for H3K36me3, H3K27Ac, or H3K4me3 removed\nOnly top sQTL per cluster plotted", y="host gene eQTL beta", x="sQTL beta")

sQTL.eQTL.beta.beta

IntronAnnotations.regtoolsbasic <- read_tsv("../code/SplicingAnalysis/regtools_annotate_combined/basic.bed.gz")

 dat.sQTLs.eQTLs.ForScatter.ToPlot %>%
  dplyr::filter(SuperAnnotation == "Unproductive" & PC2 == "chRNA") %>%
  distinct(P1, .keep_all=T) %>%
  inner_join(IntronAnnotations.regtoolsbasic) %>%
  count(anchor) %>%
  mutate(sum= sum(n))

```

Again but classify cluster as unproductive if any are unproductive

```{r}
dat.sQTLs.eQTLs.ForScatter.ToPlot <- dat.sQTLs.eQTLs.ForScatter %>%
  filter(PC1 %in% PC1.filter) %>%
  group_by(P1) %>%
  filter(!any((PC2 %in% PC2.SignificanceFilter) & (trait.x.p.in.y < 0.01))) %>%
  ungroup() %>%
  filter(PC2 %in% PC2.filter) %>%
  # pull(PC2) %>% unique()
  filter(SuperAnnotation %in% c("UnannotatedJunc_UnproductiveCodingGene", "AnnotatedJunc_ProductiveCodingGene", "AnnotatedJunc_UnproductiveCodingGene", "UnannotatedJunc_ProductiveCodingGene")) %>%
  group_by(PC1, ClusterName) %>%
  mutate(SuperAnnotation = case_when(
    any(SuperAnnotation %in% c("UnannotatedJunc_UnproductiveCodingGene", "AnnotatedJunc_UnproductiveCodingGene")) & SuperAnnotation %in% c("UnannotatedJunc_UnproductiveCodingGene", "AnnotatedJunc_UnproductiveCodingGene") ~ "Unproductive",
    all(SuperAnnotation %in% c("UnannotatedJunc_ProductiveCodingGene", "AnnotatedJunc_ProductiveCodingGene")) & SuperAnnotation %in% c("UnannotatedJunc_ProductiveCodingGene", "AnnotatedJunc_ProductiveCodingGene") ~ "Productive",
    TRUE ~ "Productive in unproductive cluster or vice versa"
    )) %>%
  filter(abs(beta.x) == max(abs(beta.x))) %>%
  ungroup() %>%
  mutate(SuperAnnotation = str_replace_all(SuperAnnotation, "_", " ")) %>%
  filter(GeneLocus == gene) %>%
  filter(FDR.x < 0.1) %>%
  group_by(P2, PC2, SuperAnnotation) %>%
  sample_n(1) %>%
  ungroup() %>%
  mutate(PC2 = factor(PC2, levels=PC2.filter)) %>%
  mutate(PC2 = recode(PC2, !!!c("chRNA.Expression.Splicing"="chRNA" , "MetabolicLabelled.30min"="30min 4sU", "MetabolicLabelled.60min"="60min 4sU", "Expression.Splicing"="polyA", "Expression.Splicing.Subset_YRI"="polyA YRI"))) %>%
  mutate(PC2 = factor(PC2, levels=c("H3K36ME3","chRNA", "30min 4sU", "60min 4sU", "polyA", "polyA YRI"))) %>%
  filter(!PC2 %in% c("polyA YRI", "H3K36ME3")) %>%
  # pull(SuperAnnotation) %>% unique()
  mutate(SuperAnnotation = recode(SuperAnnotation, !!!c("AnnotatedJunc UnproductiveCodingGene"="Unproductive", "UnannotatedJunc UnproductiveCodingGene"="Unproductive", "AnnotatedJunc ProductiveCodingGene"="Productive", "UnannotatedJunc ProductiveCodingGene"="Productive"))) %>%
  filter(SuperAnnotation %in% c("Productive", "Unproductive"))

dat.sQTLs.eQTLs.ForScatter.ToPlot.labels <- dat.sQTLs.eQTLs.ForScatter.ToPlot %>%
  count(SuperAnnotation, PC2) %>%
  distinct(SuperAnnotation, .keep_all=T) %>%
  dplyr::select(n, SuperAnnotation) %>%
  mutate(SuperAnnotationLabel = paste0(SuperAnnotation, " (n=", n, ")"))

library(broom)

sQTL.eQTL.beta.beta <- dat.sQTLs.eQTLs.ForScatter.ToPlot %>%
  left_join(dat.sQTLs.eQTLs.ForScatter.ToPlot.labels, by="SuperAnnotation") %>%
  nest(-SuperAnnotationLabel, -PC2) %>% 
  mutate(cor=map(data,~cor.test(.x$beta.x, .x$x.beta.in.y, method = "sp"))) %>%
  mutate(tidied = map(cor, tidy)) %>% 
  unnest(tidied, .drop = T) %>%
  unnest(data) %>%
ggplot(aes(x=beta.x, y=x.beta.in.y, color=SuperAnnotation)) +
    geom_point(alpha=0.1) +
    # geom_smooth(method = "tls", se = FALSE, color = "red", method='bootstrap') +
    # geom_smooth(method='lm', color='black') +
    geom_vline(xintercept=0, linetype='dashed') +
    geom_hline(yintercept=0, linetype='dashed') +
    scale_color_manual(values=c("Productive"="#1f78b4", "Unproductive"="#e31a1c")) +
    geom_abline(data = . %>%
                  distinct(SuperAnnotationLabel, PC2, .keep_all=T),
                aes(slope=estimate, intercept=0), size=2) +
    # geom_smooth(method = deming, se = FALSE, color = "black",size=2) +
    geom_text(
    data = . %>%
        distinct(SuperAnnotationLabel, PC2, .keep_all=T) %>%
        mutate(R=signif(estimate, 3), P=format.pval(p.value, 3)) %>%
      mutate(label = str_glue("rho:{R}\nP:{P}")),
    aes(x=-Inf, y=-Inf, label=label),
    hjust=-.1, vjust=-0.1, color='black', size=6
    ) +
    facet_grid(SuperAnnotationLabel ~ PC2, labeller = label_wrap_gen(10)) +
    theme(strip.text = element_text(size = 12), legend.position='none') +
  labs(caption = "FDR<10% sQTLs. sQTLs with nominal P<0.01 for H3K36me3, H3K27Ac, or H3K4me3 removed\nOnly top sQTL per cluster plotted", y="host gene eQTL beta", x="sQTL beta")

sQTL.eQTL.beta.beta







```

### uodate for reviewer...

In lieu of two reviewer comments:

 > (paraphrasing) Why wouldn't some u-sQTLs coloc with eQTLs if you accounted for independent signals?
 
 ...one reason is power, especially when the eQTL effects are very small. True small eQTL effects can be biologically explained by small sQTL effects in terms of deltaPSI (eg, a 1%PSI to 2%PSI u-sQTL will have a weak effect on expression). Other biological reasons exist, (eg transcriptional adaptation)... but for purposes of addressing reviewers, I would like to make a plot of these scatter plots with eQTL effect in (log2FC) and PSI effect in (DeltaPSI).
 
 Furthermore, we are considering doing targeted nanopore sequencing on some genes. I'll use these plots to pick some good candidates to sequence... Looking for strong effects to validate, in well expressed genes.
 
```{r}
dat.sQTLs.eQTLs.ForScatter %>%
  inner_join(
      dat.sQTLs.eQTLs.ForScatter.ToPlot %>%
        dplyr::select(PC1, P1, GeneLocus, P2 )
  ) %>%
  dplyr::select(PC1:FDR.y) %>%
  filter(PC2 %in% c("Expression.Splicing", "chRNA.Expression.Splicing", "MetabolicLabelled.30min", "MetabolicLabelled.60min")) %>%
  bind_rows(
    dat.sQTLs.eQTLs.ForScatter.ToPlot %>%
      dplyr::select(PC1:FDR.x) %>%
      distinct() %>%
      mutate(PC2 = PC1, P2=P1)
  ) %>%
  write_tsv("../output/TraitPairsForUnnormalizedScatter.ForReviewer.tsv.gz")


```
 
 ...run python script to get betas in unnormalized units
 
```{bash, eval=F}
#run python script to get betas in unnormalized units
```

```{r}
deming.fit <- function(x, y, noise_ratio = sd(y)/sd(x)) {
  if(missing(noise_ratio) || is.null(noise_ratio)) noise_ratio <- eval(formals(sys.function(0))$noise_ratio) # this is just a complicated way to write `sd(y)/sd(x)`
  delta <-  noise_ratio^2
  x_name <- deparse(substitute(x))

  s_yy <- var(y)
  s_xx <- var(x)
  s_xy <- cov(x, y)
  beta1 <- (s_yy - delta*s_xx + sqrt((s_yy - delta*s_xx)^2 + 4*delta*s_xy^2)) / (2*s_xy)
  beta0 <- mean(y) - beta1 * mean(x) 

  res <- c(beta0 = beta0, beta1 = beta1)
  names(res) <- c("(Intercept)", x_name)
  class(res) <- "Deming"
  res
}

deming <- function(formula, data, R = 100, noise_ratio = NULL, ...){
  ret <- boot::boot(
    data = model.frame(formula, data), 
    statistic = function(data, ind) {
      data <- data[ind, ]
      args <- rlang::parse_exprs(colnames(data))
      names(args) <- c("y", "x")
      rlang::eval_tidy(rlang::expr(deming.fit(!!!args, noise_ratio = noise_ratio)), data, env = rlang::current_env())
    },
    R=R
  )
  class(ret) <- c("Deming", class(ret))
  ret  
}

predictdf.Deming <- function(model, xseq, se, level) {
  pred <- as.vector(tcrossprod(model$t0, cbind(1, xseq)))
  if(se) {
    preds <- tcrossprod(model$t, cbind(1, xseq))
    data.frame(
      x = xseq,
      y = pred,
      ymin = apply(preds, 2, function(x) quantile(x, probs = (1-level)/2)),
      ymax = apply(preds, 2, function(x) quantile(x, probs = 1-((1-level)/2)))
    )
  } else {
    return(data.frame(x = xseq, y = pred))
  }
}
```


```{r}
dat.unnormalized <- read_tsv("../output/TraitPairsBetasForUnnormalizedScatter.ForReviewer.tsv.gz")

dat.unnormalized %>%
  filter(P1==P2) %>%
  distinct(PC1, P1, .keep_all=T) %>%
  ggplot(aes(x=beta.x, y=x.beta.in.y)) +
  geom_point() +
  labs(x="Effect size,\nStandardized splicing units", y="Effect size,\nDeltaPSI")

dat.unnormalized %>%
  filter(!P1==P2) %>%
  dplyr::select(GeneLocus, singletrait_topvar.x, PC2, P2, log2FC=x.beta.in.y) %>%
  distinct() %>%
  inner_join(
    dat.sQTLs.eQTLs.ForScatter.ToPlot %>%
      dplyr::select(GeneLocus, singletrait_topvar.x, PC2, P2, NormalizedExpressionChange=x.beta.in.y) %>%
      distinct() %>%
      mutate(PC2 = recode(PC2, "polyA"="Expression.Splicing", "chRNA"="chRNA.Expression.Splicing", "30min 4sU"="MetabolicLabelled.30min", "60min 4sU"="MetabolicLabelled.60min"))
  ) %>%
  ggplot(aes(x=NormalizedExpressionChange, y=log2FC)) +
  geom_point(alpha=0.1) +
  facet_wrap(~PC2)

dat.sQTLs.eQTLs.ForScatter.ToPlot.unnormalized <- dat.unnormalized %>%
  filter(!P1==P2) %>%
  dplyr::rename(log2FC = x.beta.in.y) %>%
  inner_join(
    dat.unnormalized %>%
      filter(P1==P2) %>%
      dplyr::select(PC1, P1, GeneLocus, singletrait_topvar.x, deltaPSI=x.beta.in.y)
  ) %>%
  inner_join(
    dat.sQTLs.eQTLs.ForScatter.ToPlot %>%
      dplyr::select(P1, IntronName:SemiSupergroupAnnotations)
  ) %>%
  mutate(PC2 = recode(PC2, "Expression.Splicing"="steady-state", "chRNA.Expression.Splicing"="naRNA", "MetabolicLabelled.30min"="30min 4sU", "MetabolicLabelled.60min"="60min 4sU")) %>%
  mutate(PC2 = factor(PC2, levels=c("naRNA", "30min 4sU", "60min 4sU", "steady-state"))) %>%
  distinct(P1, P2, PC1, PC2, singletrait_topvar.x, .keep_all=T)

  
dat.sQTLs.eQTLs.ForScatter.ToPlot.labels.unnormalized <- dat.sQTLs.eQTLs.ForScatter.ToPlot.unnormalized %>%
  count(SuperAnnotation, PC2) %>%
  distinct(SuperAnnotation, .keep_all=T) %>%
  dplyr::select(n, SuperAnnotation) %>%
  mutate(SuperAnnotationLabel = paste0(SuperAnnotation, " (n=", n, ")"))

sQTL.eQTL.beta.beta.unnormalized <- dat.sQTLs.eQTLs.ForScatter.ToPlot.unnormalized %>%
  left_join(dat.sQTLs.eQTLs.ForScatter.ToPlot.labels, by="SuperAnnotation") %>%
  nest(-SuperAnnotationLabel, -PC2) %>% 
  mutate(cor=map(data,~cor.test(.x$deltaPSI, .x$log2FC, method = "sp"))) %>%
  mutate(tidied = map(cor, tidy)) %>% 
  unnest(tidied, .drop = T) %>%
  unnest(data) %>%
ggplot(aes(x=deltaPSI, y=log2FC, color=SuperAnnotation)) +
    geom_point(alpha=0.1) +
    # geom_smooth(method = deming, se = FALSE, color = "black",size=2) +
    # geom_smooth(method='lm', color='black') +
    geom_vline(xintercept=0, linetype='dashed') +
    geom_hline(yintercept=0, linetype='dashed') +
    scale_color_manual(values=c("Productive"="#1f78b4", "Unproductive"="#e31a1c")) +
    geom_abline(data = . %>%
                  distinct(SuperAnnotationLabel, PC2, .keep_all=T),
                aes(slope=estimate, intercept=0), size=2) +
    # geom_smooth(method='lm') +
    # geom_smooth(method = deming, se = FALSE, color = "black",size=2) +
    geom_text(
    data = . %>%
        distinct(SuperAnnotationLabel, PC2, .keep_all=T) %>%
        mutate(R=signif(estimate, 3), P=format.pval(p.value, 3)) %>%
      mutate(label = str_glue("rho:{R}\nP:{P}")),
    aes(x=-Inf, y=-Inf, label=label),
    hjust=-.1, vjust=-0.1, color='black', size=6
    ) +
    facet_grid(SuperAnnotationLabel ~ PC2, labeller = label_wrap_gen(10)) +
    theme(strip.text = element_text(size = 12), legend.position='none') +
  labs(caption = "FDR<10% sQTLs. sQTLs with nominal P<0.01 for H3K36me3, H3K27Ac, or H3K4me3 removed\nOnly top sQTL per cluster plotted", y="host gene eQTL beta\nlog2FC", x="sQTL beta, intronic deltaPSI") +
  coord_cartesian(xlim=c(-10,10), ylim=c(-2.5,2.5))

sQTL.eQTL.beta.beta.unnormalized

sQTL.eQTL.beta.beta.unnormalized.PlusDeming <- sQTL.eQTL.beta.beta.unnormalized +
    geom_smooth(method = deming, se = FALSE, color = "purple", method.args = list(noise_ratio=NULL))
sQTL.eQTL.beta.beta.unnormalized.PlusDeming

sQTL.eQTL.beta.beta.unnormalized.nolines <- dat.sQTLs.eQTLs.ForScatter.ToPlot.unnormalized %>%
  left_join(dat.sQTLs.eQTLs.ForScatter.ToPlot.labels, by="SuperAnnotation") %>%
  nest(-SuperAnnotationLabel, -PC2) %>% 
  mutate(cor=map(data,~cor.test(.x$deltaPSI, .x$log2FC, method = "sp"))) %>%
  mutate(tidied = map(cor, tidy)) %>% 
  unnest(tidied, .drop = T) %>%
  unnest(data) %>%
ggplot(aes(x=deltaPSI, y=log2FC, color=SuperAnnotation)) +
    geom_point(alpha=0.1) +
    # geom_smooth(method = deming, se = FALSE, color = "black",size=2) +
    # geom_smooth(method='lm', color='black') +
    geom_vline(xintercept=0, linetype='dashed') +
    geom_hline(yintercept=0, linetype='dashed') +
    scale_color_manual(values=c("Productive"="#1f78b4", "Unproductive"="#e31a1c")) +
    geom_text(
    data = . %>%
        distinct(SuperAnnotationLabel, PC2, .keep_all=T) %>%
        mutate(R=signif(estimate, 3), P=format.pval(p.value, 3)) %>%
      mutate(label = str_glue("rho:{R}\nP:{P}")),
    aes(x=-Inf, y=-Inf, label=label),
    hjust=-.1, vjust=-0.1, color='black', size=6
    ) +
    facet_grid(SuperAnnotationLabel ~ PC2, labeller = label_wrap_gen(10)) +
    theme(strip.text = element_text(size = 12), legend.position='none') +
  labs(caption = "FDR<10% sQTLs. sQTLs with nominal P<0.01 for H3K36me3, H3K27Ac, or H3K4me3 removed\nOnly top sQTL per cluster plotted", y="host gene eQTL beta\nlog2FC", x="sQTL beta, intronic deltaPSI") +
  coord_cartesian(xlim=c(-10,10), ylim=c(-1,1))
sQTL.eQTL.beta.beta.unnormalized.nolines

```
On second thought, I don't know why I plot the rho coefficient as a slope... That doesn't even make sense. Well maybe as a visual aid to highlight the change in rho across facets, and when both the x and y are on a standardized scale then maybe it looks alright, but not on this plot with unstandardized units. A better visual and more appropriate would be to plot the orthogonal regression slope... But actually I tried that above (purple line), and it's not as nice as I thought, I think partly because the orthogonal regression is still susceptible to outliers. Since this is going in the supplement or just to the reviewer, it doesn't bother me to just present this with the spearman slope as plotted above.

Ok, now let's pick some u-sQTL/eQTLs for Dylan's targeted sequencing experiment... Just pick some with a deltaPSI and log2FC effect above some threshold. Maybe I'll manually inspect the eQTL/sQTL effect on pygenometracks to confirm... Also want ones where the topSNP is in an exon that can be sequenced

```{r}
GenePos <- read_tsv("../code/QTLs/QTLTools/Expression.Splicing.Subset_YRI/OnlyFirstReps.sorted.qqnorm.bed.gz") %>%
  dplyr::select(1:4) %>%
  inner_join(
    read_tsv("../data/")
  )

# Subset exonic SNPs
dat.sQTLs.eQTLs.ForScatter.ToPlot.unnormalized %>%
  dplyr::select(chrom=singletrait_topvar_chr.x, start=singletrait_topvar_pos.x, name=singletrait_topvar.x) %>%
  mutate(end = start+1) %>%
  arrange(chrom, start, end) %>%
  distinct() %>%
  dplyr::select(chrom, start, end, name) %>%
  write_tsv("../code/scratch/PlotsToPickTargetedGenes/AllPotentialSNPs.bed")

```

```{bash, eval=F}
bedtools intersect -a scratch/PlotsToPickTargetedGenes/AllPotentialSNPs.bed -b ReferenceGenome/Annotations/GTFTools_BasicAnnotations/gencode.v34.chromasomal.exons.sorted.bed -u -sorted > scratch/PlotsToPickTargetedGenes/ExonicSNPs.bed

#Example code to look for exonic SNPs in a range that might be significant eQTL for NUDT14
tabix -h QTLs/QTLTools/Expression.Splicing/NominalPassForColoc.txt.tabix.gz  chr14:105,176,477-105,181,383 | grep -e '^#' -e 'ENSG00000183828.15' | awk -v OFS='\t' 'NR>1 {print $9, $10, $11, $8, $12}' | bedtools intersect -a - -b ReferenceGenome/Annotations/GTFTools_BasicAnnotations/gencode.v34.chromasomal.exons.sorted.bed -u

#returns 'chr14   105181225       105181243       14:105181225:GGCGGGGGCC:G       7.27844e-21', a SNP

#Example code to look for exonic SNPs in a range that might be significant eQTL for D2HGDH
tabix -h QTLs/QTLTools/Expression.Splicing/NominalPassForColoc.txt.tabix.gz  chr2:241,734,766-241,768,131 | grep -e '^#' -e 'ENSG00000180902.18' | awk -v OFS='\t' 'NR>1 {print $9, $10, $11, $8, $12}' | bedtools intersect -a - -b ReferenceGenome/Annotations/GTFTools_BasicAnnotations/gencode.v34.chromasomal.exons.sorted.bed -u

#that returns this:
```

chr2    241735134       241735134       2:241735134:C:G 0.118969
chr2    241735388       241735388       2:241735388:G:A 3.79959e-06
chr2    241751260       241751260       2:241751260:G:A 1.32053e-13
chr2    241751314       241751314       2:241751314:C:T 0.0619802
chr2    241751330       241751330       2:241751330:C:T 0.531458
chr2    241751355       241751355       2:241751355:T:C 0.000118949
chr2    241755192       241755192       2:241755192:C:T 0.163042
chr2    241755223       241755254       2:241755223:GTGTCCTTCC:G        4.23574e-07
chr2    241755234       241755234       2:241755234:T:A 0.367674
chr2    241755500       241755500       2:241755500:A:G 0.265307
chr2    241767780       241767780       2:241767780:C:A 0.0398028
chr2    241767798       241767798       2:241767798:G:A 0.576106
chr2    241768062       241768062       2:241768062:T:C 7.62866e-08

Manually checking those significant eQTLs, I see that 

2:241735388 and 2:241755223 are not a good choice (not consitutive exon),

2:241768062 and 2:241751260 ARE good choices (in constitutive exon)...



```{r}

ExonicSNPs <- read_tsv("../code/scratch/PlotsToPickTargetedGenes/ExonicSNPs.bed", col_names=c("chrom", "start", "stop", "name"))

dat.sQTLs.eQTLs.ForScatter.ToPlot.unnormalized %>%
  filter(SuperAnnotation == "Unproductive") %>%
  filter(PC2=="steady-state") %>%
  filter(abs(deltaPSI)>5) %>%
  filter(abs(log2FC) > 0.2) %>%
  filter(!sign(deltaPSI)==sign(log2FC)) %>%
  mutate(SNPPos = str_glue("{singletrait_topvar_chr.x}:{singletrait_topvar_pos.x}")) %>%
  dplyr::select(P1, SNPPos, GeneLocus, singletrait_topvar.x, symbol, singletrait_topvar_pos.x) %>%
  inner_join(GenePos, by=c("GeneLocus"="pid")) %>%
  filter(singletrait_topvar_pos.x < end & singletrait_topvar_pos.x > start) %>%
  filter(singletrait_topvar.x %in% ExonicSNPs$name) %>%
  mutate(ScriptCommand = str_glue(
    "python scripts/GenometracksByGenotype/AggregateBigwigsForPlotting.py --VCF Genotypes/1KG_GRCh38/Autosomes.vcf.gz --SnpPos {SNPPos} --GroupSettingsFile PlotQTLs/bwList.Groups.4col.chRNA_and_polyARNA.tsv --BigwigList PlotQTLs/bwList.AllPolyAAndUnstrandedchRNA.tsv --Normalization WholeGenome --Region {`#Chr`}:{start}-{end} --BigwigListType KeyFile --OutputPrefix scratch/PlotsToPickTargetedGenes/dat. --SnpName {singletrait_topvar.x}  --Bed12GenesToIni scripts/GenometracksByGenotype/PremadeTracks/gencode.v34.transcripts.bed12.gz --FilterJuncsByBed  <(tabix -h SplicingAnalysis/leafcutter/NormalizedPsiTables/PSI.Expression.Splicing.bed.gz {`#Chr`}:{start}-{end} | grep -e '^#' -e '{P1}') -v
    pyGenomeTracks --region {`#Chr`}:{start}-{end}  -out scratch/PlotsToPickTargetedGenes/Plots/{symbol}.pdf --tracks scratch/PlotsToPickTargetedGenes/dat.tracks.ini
    "
  )) %>%
  # pull(ScriptCommand)
  dplyr::select(ScriptCommand) %>%
  write_tsv("../code/scratch/PlotsToPickTargetedGenes/PlotGenes.sh", col_names = F)
```

```{bash, eval=F}
bash scratch/PlotsToPickTargetedGenes/PlotGenes.sh
```

...So after including GBP3, NAAA, NUDT14, and D2HGDH... these are the (hopefully causal or tightly linked SNPs) to look for in strains to grow:

2:241768062, D2HGDH
2:241751260, D2HGDH
14:105181225:GGCGGGGGCC:G, NUDT14
4:75920950:C:T, NAAA
1:89009452:C:T, GBP3

hQTL eQTL beta

#### Reviewer point about susie

```{r}
susie <- read_tsv("../code/FineMapping/susie_runs_Geuvadis/susie_output.tab.gz")

susie %>%
  distinct(cs_names)

susie %>%
  distinct(cs_gene, cs_names) %>%
  count(cs_gene) %>%
  count(n) %>%
  ggplot(aes(x=n, y=nn)) +
  geom_col()

susie %>%
  distinct(cs_gene, cs_names) %>%
  filter(str_detect(cs_gene, "ENSG00000180902"))
```


#### Sex balanced of dataset

```{r}
new.samples <- read_tsv("../code/config/samples.tsv")



all.1KG.samples <-read_tsv("../data/igsr_samples.tsv.gz")

new.samples %>%
  filter(Phenotype %in% c("chRNA.Expression.Splicing", "H3K36ME3")) %>%
  distinct(Phenotype, IndID, .keep_all=T) %>%
  inner_join(all.1KG.samples, by=c("IndID"="Sample name")) %>%
  count(Phenotype, Sex)
```


#### Reviewer point about how would things change if using Lindeboom rules

Remake scatter but categorizing unproductive based on Lindeboom context

```{r}
LindeboomContextAnnotations <- read_tsv("../output/20240322_ResponseToReviewerMostCommonJuncContexts.tsv.gz") %>%
  mutate(IntronName = str_replace_all(Introns, "_", ":"))

dat.sQTLs.eQTLs.ForScatter.ToPlot %>%
  inner_join(
    LindeboomContextAnnotations %>% dplyr::select(IntronName,ModeNMDFinder)
  ) %>%
  ggplot(aes(x=beta.x, y=x.beta.in.y, color=ModeNMDFinder)) +
  geom_point() +
  # geom_smooth(method='lm') +
  facet_grid(SuperAnnotation ~ PC2) +
  theme_bw()

dat.sQTLs.eQTLs.ForScatter.ToPlot %>%
  inner_join(
    LindeboomContextAnnotations %>% dplyr::select(IntronName,ModeNMDFinder)
  ) %>%
  ggplot(aes(x=beta.x, y=x.beta.in.y, color=ModeNMDFinder)) +
  geom_point() +
  # geom_smooth(method='lm') +
  facet_grid(ModeNMDFinder ~ PC2) +
  theme_bw()

# Write out spearman correlations, to possibly include to reviewer response
dat.sQTLs.eQTLs.ForScatter.ToPlot %>%
  inner_join(
    LindeboomContextAnnotations %>% dplyr::select(IntronName,ModeNMDFinder)
  ) %>%
  group_by(ModeNMDFinder, PC2) %>%
  filter(n() > 1) %>%
  summarize(COR = stats::cor.test(beta.x, x.beta.in.y, method='s')$estimate,
            pval = stats::cor.test(beta.x, x.beta.in.y, method='s')$p.value,
            n = n()
                  ) %>%
  ungroup() %>%
  dplyr::select(sQTLJunctionClassificationFromLongReadContextAndLindeboomDecisionTree=ModeNMDFinder, ExpressionEffectDataset=PC2, SpearmanCoef.SteadyStateSplicingVsExpression=COR, pval, n) %>%
  write_tsv("../code/scratch/ResponseToReviewerTable.eQTLsQTLEffectSizes.JuncsReclassified.tsv")

```

```{r}

PC1.filter <- c("Expression.Splicing", "MetabolicLabelled.30min", "MetabolicLabelled.60min", "H3K36ME3", "H3K27AC", "H3K4ME3", "H3K4ME1", "Expression.Splicing.Subset_YRI", "chRNA.Expression.Splicing")
dat <- fread("../code/pi1/PairwisePi1Traits.P.all.txt.gz") %>%
  filter((PC1 %in% PC1.filter) & (PC2 %in% PC1.filter))

PeaksToTSS <- Sys.glob("../code/Misc/PeaksClosestToTSS/*_assigned.tsv.gz") %>%
  setNames(str_replace(., "../code/Misc/PeaksClosestToTSS/(.+?)_assigned.tsv.gz", "\\1")) %>%
  lapply(read_tsv) %>%
  bind_rows(.id="ChromatinMark") %>%
  mutate(GenePeakPair = paste(gene, peak, sep = ";")) %>%
  distinct(ChromatinMark, peak, gene, .keep_all=T)

#purple is #6a3d9a

FDR.cutoff <- 0.1
ActivatingMarksToConsider <- c("H3K27AC", "H3K4ME3")
Recode.PCs = c("ActivatingMark_HostGenePromoter"="Promoter marks*", "H3K36ME3"="H3K36ME3", "chRNA.Expression.Splicing"="chRNA", "MetabolicLabelled.30min"="4sU 30min", "MetabolicLabelled.60min"="4sU 60min", "Expression.Splicing.Subset_YRI"="polyA YRI", "Expression.Splicing"="polyA")


dat.toPlot <- dat %>%
  filter(FDR.x < FDR.cutoff) %>%
  mutate(PC1 = case_when(
    paste(GeneLocus, P1, sep=";") %in% PeaksToTSS$GenePeakPair & PC1 %in% ActivatingMarksToConsider ~ PC1,
    P1 %in% PeaksToTSS$peak ~ "ActivatingMark_OtherGenePromoter",
    PC1 %in% c("H3K4ME1", "H3K4ME3", "H3K27AC") ~ "ActivatingMark_Enhancer",
    TRUE ~ PC1
  )) %>%
  filter(PC1 %in% ActivatingMarksToConsider) %>%
  filter(PC2 %in% c("chRNA.Expression.Splicing", "Expression.Splicing", "MetabolicLabelled.30min", "MetabolicLabelled.60min")) %>%
  mutate(PC2 = recode(PC2, !!!Recode.PCs)) %>%
  mutate(PC2 = factor(PC2, levels=c("chRNA", "4sU 30min", "4sU 60min", "polyA"))) %>%
  mutate(SuperAnnotationLabel = PC1) %>%
  nest(-SuperAnnotationLabel, -PC2) %>% 
  mutate(cor=map(data,~cor.test(.x$beta.x, .x$x.beta.in.y, method = "sp"))) %>%
  mutate(tidied = map(cor, tidy)) %>% 
  unnest(tidied, .drop = T) %>%
  unnest(data)

ggplot(dat.toPlot, aes(x=beta.x, y=x.beta.in.y)) +
  geom_point(alpha=0.1) +
    geom_vline(xintercept=0, linetype='dashed') +
    geom_hline(yintercept=0, linetype='dashed') +
    scale_color_manual(values=c("Productive"="#1f78b4", "Unproductive"="#e31a1c")) +
    geom_abline(data = . %>%
                  distinct(SuperAnnotationLabel, PC2, .keep_all=T),
                aes(slope=estimate, intercept=0), size=2) +
    geom_text(
    data = . %>%
        distinct(SuperAnnotationLabel, PC2, .keep_all=T) %>%
        mutate(R=signif(estimate, 3), P=format.pval(p.value, 3)) %>%
      mutate(label = str_glue("rho:{R}\nP:{P}")),
    aes(x=-Inf, y=Inf, label=label),
    hjust=-.1, vjust=1.1, color='black', size=6
    ) +
    facet_grid(SuperAnnotationLabel ~ PC2, labeller = label_wrap_gen(10)) +
    theme(strip.text = element_text(size = 12), legend.position='none') +
  labs(caption = "FDR<10% hQTLs at promoters.", y="eQTL beta in facet dataset", x="Promoter hQTL beta")

# ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/20230412_WIP_eQTLhQTLbetascatter.pdf", height=4, width=10)

dat.toPlot %>%
  filter(SuperAnnotationLabel == "H3K4ME3") %>%
  distinct(GeneLocus, PC2) %>%
  count(PC2)
  
eQTL.H3k27ac.beta <- dat.toPlot %>%
  filter(SuperAnnotationLabel == "H3K27AC") %>%
  mutate(SuperAnnotationLabel = "H3K27AC@TSS\n(n=1298)") %>%
ggplot(aes(x=beta.x, y=x.beta.in.y)) +
  geom_point(alpha=0.1, color="#6a3d9a") +
    geom_vline(xintercept=0, linetype='dashed') +
    geom_hline(yintercept=0, linetype='dashed') +
    geom_abline(data = . %>%
                  distinct(SuperAnnotationLabel, PC2, .keep_all=T),
                aes(slope=estimate, intercept=0), size=2) +
    geom_text(
    data = . %>%
        distinct(SuperAnnotationLabel, PC2, .keep_all=T) %>%
        mutate(R=signif(estimate, 3), P=format.pval(p.value, 3)) %>%
      mutate(label = str_glue("rho:{R}\nP:{P}")),
    aes(x=-Inf, y=Inf, label=label),
    hjust=-.1, vjust=1.1, color='black', size=6
    ) +
    facet_grid(SuperAnnotationLabel ~ PC2, labeller = label_wrap_gen(10)) +
    theme(strip.text = element_text(size = 12), legend.position='none') +
  labs(caption = "FDR<10% hQTLs at promoters.", y="eQTL beta", x="Promoter hQTL beta")

eQTL.H3k27ac.beta

dat.toPlot %>%
  filter(SuperAnnotationLabel == "H3K27AC") %>%
  distinct(GeneLocus, PC2) %>%
  count(PC2)
```


- get snps for carlos' gtex analysis:

- unproductive sQTL/eQTL SNPs.
- h3k27ac @ TSS hQTL/eQTL SNPs where lead SNP is within 10kb from promoter
- h3k27ac @ TSS hQTL/eQTL SNPs where lead SNP is > 10kb from promoter

```{r}

PC1.filter = c("polyA.Splicing")
PC2.filter = c("H3K36ME3", "chRNA.Expression.Splicing" , "MetabolicLabelled.30min", "MetabolicLabelled.60min", "Expression.Splicing", "Expression.Splicing.Subset_YRI")
PC2.SignificanceFilter <- c("H3K4ME3", "H3K27AC", "H3K36ME3")

sQTL.Nominal.hQTLs <- dat.sQTLs.eQTLs.ForScatter %>%
  filter(PC1 %in% PC1.filter) %>%
  group_by(P1) %>%
  filter(any((PC2 %in% PC2.SignificanceFilter) & (trait.x.p.in.y < 0.01))) %>%
  ungroup() %>%
  distinct(P1)


## only YRI for eQTL
# coloc.dat.fn <- "../code/hyprcoloc/Results/ForColoc/MolColocNonRedundantFullSplicing/tidy_results_OnlyColocalized.txt.gz"

## Full geuvadis for eqtl
coloc.dat.fn <- "../code/hyprcoloc/Results/ForColoc/GenewiseColoc_ChromatinAPAAndRNA/tidy_results_OnlyColocalized.txt.gz"

IntronAnnotations <- read_tsv("../data/IntronAnnotationsFromYang.tsv.gz") %>%
  unite(Intron, chrom, start, end, strand, sep=":")


dat.sQTLs.eQTLs.ForScatter.ToPlot

coloc.dat <- read_tsv(coloc.dat.fn) %>%
  separate(phenotype_full, into=c("PC", "P"), sep=";") %>%
  mutate(Intron = case_when(
    PC == "polyA.Splicing" ~ str_replace(P, "(^.+?)clu_.+?_([+-])", "chr\\1\\2"),
    TRUE ~ NA_character_
  )) %>%
  left_join(IntronAnnotations) %>%
  mutate(PC= recode(PC, "Expression.Splicing.Subset_YRI"="Expression.Splicing"))


coloc.dat$PC %>% unique()

## the sQTL/eQTL colocs, remove if top unproductive sQTL is nominal hQTL
sQTL.eQTL.colocs <- coloc.dat %>%
  group_by(snp, Locus) %>%
  filter(
    any(SuperAnnotation %in% c("AnnotatedJunc_UnproductiveCodingGene", "UnannotatedJunc_UnproductiveCodingGene")) &
    any(PC == "Expression.Splicing") 
    ) %>%
  ungroup() %>%
  filter(PC == "Expression.Splicing" | SuperAnnotation %in% c("AnnotatedJunc_UnproductiveCodingGene", "UnannotatedJunc_UnproductiveCodingGene")) %>%
  group_by(PC, snp, Locus) %>%
  filter(abs(beta) == max(abs(beta))) %>%
  ungroup() %>%
  group_by(Locus, snp) %>%
  filter(!any(P %in% sQTL.Nominal.hQTLs$P1)) %>%
  ungroup()

sQTL.eQTL.colocs %>%
  dplyr::select(snp, Locus, beta, P, PC, snp, Locus) %>%
  pivot_wider(names_from="PC", values_from=c("beta", "P")) %>%
  ggplot(aes(x=beta_polyA.Splicing, y=beta_Expression.Splicing)) +
  geom_point()
```

Ok now I also want to consider hQTLs for H3K27Ac peaks gene promoter, that are also eQTL. The top finemapped  SNP may be near the promoter (proximal hQTL/eQTL), or it may be at a distal activating element (distal hQTL/eQTL)

```{r}
sQTL.eQTL.colocs

PeaksToTSS <- Sys.glob("../code/Misc/PeaksClosestToTSS/*_assigned.tsv.gz") %>%
  setNames(str_replace(., "../code/Misc/PeaksClosestToTSS/(.+?)_assigned.tsv.gz", "\\1")) %>%
  lapply(read_tsv) %>%
  bind_rows(.id="ChromatinMark") %>%
  mutate(GenePeakPair = paste(gene, peak, sep = ";")) %>%
  distinct(ChromatinMark, peak, gene, .keep_all=T)

coloc.dat <- read_tsv(coloc.dat.fn) %>%
  separate(phenotype_full, into=c("PC", "P"), sep=";") %>%
  mutate(GenePeakPair = str_glue("{Locus};{P}")) %>%
  mutate(PC= recode(PC, "Expression.Splicing.Subset_YRI"="Expression.Splicing"))


hQTL.eQTL.colocs <- coloc.dat %>%
  filter(PC %in% c("Expression.Splicing", "H3K27AC", "H3K4ME3", "H3K4ME1")) %>%
  left_join(PeaksToTSS) %>%
  group_by(Locus, snp) %>%
  filter(any(PC=="Expression.Splicing") & any(!is.na(ChromatinMark))) %>%
  ungroup() %>%
  filter(PC=="Expression.Splicing" | peak==P)

hQTL.eQTL.colocs %>%
  distinct(Locus, snp)

hQTL.eQTL.colocs %>%
  filter(Locus %in% sQTL.eQTL.colocs$Locus) %>%
  distinct(Locus, snp)

inner_join(
  hQTL.eQTL.colocs %>%
    dplyr::select(snp, Locus, beta, P, PC) %>%
    filter(PC=="Expression.Splicing"),
  hQTL.eQTL.colocs %>%
    dplyr::select(snp, Locus, beta, P, PC, TSS_start) %>%
    filter(PC %in% c("H3K4ME3", "H3K27AC", "H3K4ME1")),
  by=c("snp", "Locus"),
  suffix=c("_Expression.Splicing", "_chromatin")
) %>%
  ggplot(aes(x=beta_chromatin, y=beta_Expression.Splicing)) +
  geom_point() +
  facet_wrap(~PC_chromatin)

```

let's also check the sQTL.eQTL colocs that are also hQTL colocs, (despite the top sQTL not being nominally significant for any hQTL, which is weird)

```{r}

sQTL.eQTL.colocs %>%
  dplyr::select(snp, Locus, beta, P, PC, snp, Locus) %>%
  pivot_wider(names_from="PC", values_from=c("beta", "P")) %>%
  mutate(Is_hQTL.Coloc = Locus %in% hQTL.eQTL.colocs$Locus) %>%
  ggplot(aes(x=beta_polyA.Splicing, y=beta_Expression.Splicing, color=Is_hQTL.Coloc)) +
  geom_point()

inner_join(
  hQTL.eQTL.colocs %>%
    dplyr::select(snp, Locus, beta, P, PC) %>%
    filter(PC=="Expression.Splicing"),
  hQTL.eQTL.colocs %>%
    dplyr::select(snp, Locus, beta, P, PC, TSS_start) %>%
    filter(PC %in% c("H3K4ME3", "H3K27AC", "H3K4ME1")),
  by=c("snp", "Locus"),
  suffix=c("_Expression.Splicing", "_chromatin")
) %>%
  mutate(Is.sQTL.Coloc = Locus %in% sQTL.eQTL.colocs$Locus) %>%
  ggplot(aes(x=beta_chromatin, y=beta_Expression.Splicing, color=Is.sQTL.Coloc)) +
  geom_point() +
  facet_wrap(~PC_chromatin)
```

Ok, as one might expect, from this I conclude that if there is a promoter hQTL, that is probably the driving mechanism over splicing.

let's filter out those hQTL.eQTL colocs from the sQTL.eQTL colocs set. Then I  will tidy up this data for Carlos.

I want the following columns in the output file:

TopSNP, TopSNPFinemapPr, eGene, eQTL_beta, molQTL_beta, molQTL_Name, molQTL_type, EffectSizeDirectionsAsExpected, Dist_from_TopSNP_to_closest_hQTL_TSS, sQTL_or_hQTL, BensClassification

From there he may choose to keep only Concordant ones, and make sure to keep only one row for each eGene, since some may be listed more than once for multiple molQTL.

```{r}
sQTL.eQTL.colocs.filtered <- sQTL.eQTL.colocs %>%
  filter(!Locus %in% hQTL.eQTL.colocs$Locus)

hQTL.eQTLs.ForCarlos <- inner_join(
  hQTL.eQTL.colocs %>%
    dplyr::select(snp, Locus, beta, P, PC, TopSNPFinemapPr, beta_se, p) %>%
    filter(PC=="Expression.Splicing"),
  hQTL.eQTL.colocs %>%
    dplyr::select(snp, Locus, beta, P, PC, TSS_start, beta_se, p) %>%
    filter(PC %in% c("H3K4ME3", "H3K27AC", "H3K4ME1")),
  by=c("snp", "Locus"),
  suffix=c("_Expression.Splicing", "_chromatin")
) %>%
  mutate(EffectSizeDirectionsAsExpected = sign(beta_Expression.Splicing)==sign(beta_chromatin)) %>%
  separate(snp, into=c("chrom", "pos", "ref", "alt"), sep=":", remove=F, convert=T) %>%
  mutate(TSS_to_SNP = abs(TSS_start - pos)) %>%
  group_by(Locus, snp) %>%
  mutate(Dist_from_TopSNP_to_closest_hQTL_TSS = min(TSS_to_SNP)) %>%
  mutate(sQTL_or_hQTL="hQTL") %>%
  mutate(BensClassification = case_when(
    Dist_from_TopSNP_to_closest_hQTL_TSS < 5000 ~ "hQTL_Proximal",
    TRUE ~ "hQTL_Distal")) %>%
  dplyr::select(TopSNP=snp, TopSNPFinemapPr, eGene=Locus, eQTL_beta=beta_Expression.Splicing, eQTL_beta_se=beta_se_Expression.Splicing, eQTL_nomP=p_Expression.Splicing, molQTL_beta=beta_chromatin, molQTL_beta_se=beta_se_chromatin, molQTL_nomP=p_chromatin, molQTL_Name=P_chromatin, molQTL_type=PC_chromatin, EffectSizeDirectionsAsExpected, Dist_from_TopSNP_to_closest_hQTL_TSS,  sQTL_or_hQTL, BensClassification)

sQTL.eQTL.colocs.ForCarlos <- sQTL.eQTL.colocs %>%
  filter(!Locus %in% hQTL.eQTL.colocs$Locus) %>%
  dplyr::select(snp, TopSNPFinemapPr, Locus, beta, P, PC, snp, Locus, beta_se, p) %>%
  pivot_wider(names_from="PC", values_from=c("beta", "P", "beta_se", "p")) %>%
  mutate(BensClassification = "sQTL", sQTL_or_hQTL="sQTL", Dist_from_TopSNP_to_closest_hQTL_TSS=NA, molQTL_type="polyA.Splicing", EffectSizeDirectionsAsExpected = !sign(beta_Expression.Splicing)==sign(beta_polyA.Splicing)) %>%
  dplyr::select(TopSNP=snp, TopSNPFinemapPr, eGene=Locus, eQTL_beta=beta_Expression.Splicing, eQTL_beta_se=beta_se_Expression.Splicing, eQTL_nomP=p_Expression.Splicing, molQTL_beta=beta_polyA.Splicing, molQTL_beta_se=beta_se_polyA.Splicing, molQTL_nomP=p_polyA.Splicing, molQTL_Name=P_polyA.Splicing, molQTL_type, EffectSizeDirectionsAsExpected, Dist_from_TopSNP_to_closest_hQTL_TSS,  sQTL_or_hQTL, BensClassification)

sQTL.vs.hQTLs.EffectOnEqtls.ForCarlos <- bind_rows(sQTL.eQTL.colocs.ForCarlos, hQTL.eQTLs.ForCarlos)

sQTL.vs.hQTLs.EffectOnEqtls.ForCarlos %>%
  distinct(eGene, .keep_all=T) %>%
  add_count(BensClassification) %>%
  mutate(Group = str_glue("{BensClassification}\n({n} eGenes)")) %>%
  ggplot(aes(x=molQTL_beta, y=eQTL_beta)) +
  geom_point() +
  facet_wrap(~Group)

sQTL.vs.hQTLs.EffectOnEqtls.ForCarlos %>%
  distinct(eGene, .keep_all=T) %>%
  add_count(BensClassification) %>%
  mutate(Group = str_glue("{BensClassification}\n({n} eGenes)")) %>%
  ggplot(aes(x=abs(eQTL_beta), color=Group)) +
  stat_ecdf() +
  labs(y="ecdf", x="abs(eQTL beta), standardized units")

sQTL.vs.hQTLs.EffectOnEqtls.ForCarlos %>%
  write_tsv("../output/eQTLs_FromSplicingVsChromatin_AcrossGTEx.ForCarlos.tsv.gz")

```


Ok, Carlos ran QTLtools with log2CPM as units for expression matrix, so beta can be interpreted (across all tissues) as the log2FC per dose of alt allele.

I've added that to "MakeFinalFigs_Fig3.Rmd"

### stuff to plot Fig3A (Illustrative coloc examples)

I'm going to filter for coloc examples like LTBP4, and plot them all, and then manually pick the one I visually like... The criteria I am filtering for are where there is an H3K27AC QTL that colocs with the eQTL, and a seperate cluster with at least two sQTLs that are all productive, so that the sQTL is a truly independent signal from the eQTL.

```{r}
coloc.dat.ForIllustrativePlot <- read_tsv("../code/hyprcoloc/Results/ForColoc/GenewiseColoc_ChromatinAPAAndRNA/tidy_results_OnlyColocalized.txt.gz") %>%
  separate(phenotype_full, into=c("PC", "P"), sep=";", remove=F) 

IntronAnnotations <- read_tsv("../data/IntronAnnotationsFromYang.Updated.tsv.gz") %>%
  filter(!str_detect(SuperAnnotation, "Noncoding")) %>%
  mutate(ProductiveOrUnproductive = if_else(str_detect(SuperAnnotation, "Unproductive"), "Unproductive", "Productive")) %>%
  unite(Int, chrom, start, end, strand, sep=":" )

dat.CandidatesToPlot <- coloc.dat.ForIllustrativePlot %>%
  group_by(Locus) %>%
  mutate(NumColocClusters = n_distinct(snp)) %>%
  filter(NumColocClusters > 1) %>%
  ungroup() %>%
  arrange(Locus, snp) %>%
  group_by(Locus, snp) %>%
  add_count(PC) %>%
  mutate(
    ClusterType = case_when(
      (any(PC=="H3K27AC") & any(PC=="Expression.Splicing")) ~ "hQTL+eQTL",
      any(PC=="polyA.Splicing" & n>1) & (all(!PC=="Expression.Splicing")) ~ "sQTL",
       TRUE ~ "Other"
    )) %>%
  ungroup() %>%
  group_by(Locus) %>%
  filter(any(ClusterType=="hQTL+eQTL") & any(ClusterType=="sQTL")) %>%
  ungroup() %>%
  filter(ClusterType %in% c("hQTL+eQTL", "sQTL")) %>%
  filter(PC %in% c("Expression.Splicing", "polyA.Splicing", "H3K27AC")) %>%
  mutate(Int = case_when(
    PC == "polyA.Splicing" ~ str_replace(P, "^(.+?):clu_.+?_([+-])$", "chr\\1:\\2"),
    TRUE ~ NA_character_
  )) %>%
  left_join(
    IntronAnnotations
  ) %>%
  group_by(Locus, snp) %>%
  filter(all(ProductiveOrUnproductive=="Productive") | (ClusterType=="hQTL+eQTL" & !PC=="polyA.Splicing") ) %>%
  ungroup() %>%
  arrange(Locus, snp, PC, p) %>%
  group_by(Locus) %>%
  filter(any(ClusterType=="hQTL+eQTL") & any(ClusterType=="sQTL")) %>%
  ungroup() %>%
  # distinct(Locus)
  group_by(Locus, snp, PC) %>%
  mutate(rn = row_number()) %>%
  filter((rn==1 & PC=="H3K27AC") | 
           (rn==1 & PC=="Expression.Splicing") |
           (rn <=2 & PC=="polyA.Splicing") ) %>%
  ungroup() %>%
  dplyr::select(-n, -rn) %>%
  group_by(Locus, snp) %>%
  add_count(PC) %>%
  filter(! (PC=="polyA.Splicing" & n==1)) %>%
  ungroup() %>%
  group_by(Locus, PC) %>%
  mutate(minP = min(p)) %>%
  ungroup() %>%
  group_by(Locus, snp) %>%
  filter(any(PC=="polyA.Splicing" & p==minP) | (!PC=="polyA.Splicing")) %>%
  ungroup() %>%
  add_count(Locus)

dat.CandidatesToPlot
```

For each of these, I need data for the boxplot, and for the coloc scatter plot

```{r, eval=F}
fn <- "../code/scratch/DatForColocExampleBoxplots.tsv.gz"
#Check its existence
if (file.exists(fn)) {
  #Delete file if it exists
  file.remove(fn)
}


dat.CandidatesToPlot %>%
  mutate(SnpPos = str_replace(snp, "(^.+?:.+?):.+?:.+?$", "chr\\1")) %>%
  mutate(cmd = str_glue("python ExtractPhenotypeBedByGenotype.py --Bed ../../QTLs/QTLTools/{PC}/OnlyFirstReps.sorted.qqnorm.bed.gz --VCF ../../Genotypes/1KG_GRCh38/Autosomes.vcf.gz --FeatureName {P} --SnpPos {SnpPos} --SnpName {snp} --Long --ReportBedAsColumn --NoHeader -v | gzip - >> ../../scratch/DatForColocExampleBoxplots.tsv.gz")) %>%
    dplyr::select(cmd) %>%
  write_tsv("../code/scratch/DatForColocExampleBoxplots.Commands.sh", col_names = F)
```

Boxplots

```{r}
dir.create("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/ColocExamples/")

Coloc.exampleDat.Boxplots <- read_tsv("../code/scratch/DatForColocExampleBoxplots.tsv.gz", col_names=c("IndID", "genotype", "ref", "alt", "snp", "P", "value", "fn")) %>%
  separate(P, into=c("chrom", "start", "stop", "P", "grp_name", "strand"), sep=";", convert=T) %>%
  mutate(PC = str_replace(fn, "../../QTLs/QTLTools/(.+?)/OnlyFirstReps.sorted.qqnorm.bed.gz", "\\1"))

dat.CandidatesToPlot.BoxPlotDatSplit <- dat.CandidatesToPlot %>%
  group_by(Locus, snp, PC) %>%
  filter(row_number()==1) %>%
  ungroup() %>%
  inner_join(Coloc.exampleDat.Boxplots) %>%
  split(.$Locus)

dat.CandidatesToPlot$PC %>% unique()

library(ggnewscale)

for (i in seq_along(dat.CandidatesToPlot.BoxPlotDatSplit)){
  print(i)
  if (i<=100){
    dat.ToBoxplot <- 
    dat.CandidatesToPlot.BoxPlotDatSplit[[i]] %>%
      filter(!genotype==3) %>%
      mutate(FacetGroup = recode(PC, "polyA.Splicing"="sQTL", "Expression.Splicing"="eQTL", "H3K27AC"="hQTL")) %>%
      mutate(FacetGroup = factor(FacetGroup, levels=c("eQTL", "hQTL", "sQTL"))) %>%
      mutate(genotype= factor(genotype, levels=c(0,1,2)))

    Coloc.Box.P <- ggplot(dat.ToBoxplot, aes(x=genotype, y=value, group=genotype)) +
      geom_boxplot(outlier.shape = NA, aes(color=genotype), size=1) +
      scale_color_manual(values=c("0"="#000000","1"="#525252","2"="#d9d9d9")) +
      new_scale_color() +
      geom_rect(xmin=-Inf, xmax=Inf, ymin=-Inf, ymax=Inf, alpha=0, size=2, aes(color=FacetGroup)) +
      scale_color_manual(values=c("eQTL"="#ff7f00", "hQTL"="#6a3d9a", "sQTL"="#1f78b4")) +
      geom_smooth(method='lm', aes(group=1), se=F, color="gray") +
      facet_wrap(~FacetGroup, nrow=1) +
      theme(legend.position="none") +
      labs(x="Genotype at top molQTL SNP", color=NULL, y="Normalized Phenotype")
    ggsave(paste0("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/ColocExamples/", Locus, ".box.pdf"), Coloc.Box.P, height=3, width=5)

    
    Locus <- dat.CandidatesToPlot.BoxPlotDatSplit[[i]]$Locus %>% unique()
    Contains_molQTLs <- dat.ToBoxplot %>% distinct(PC, P, FacetGroup, Locus) %>% 
      filter(!FacetGroup=="eQTL") %>% nrow() >= 1
    Contains_eQTL <- dat.ToBoxplot %>% distinct(PC, P, FacetGroup, Locus) %>%
      filter(FacetGroup=="eQTL") %>% nrow() == 1

    if (Contains_molQTLs & Contains_eQTL){
          SummaryStats <- fread(paste0("../code/hyprcoloc/LociWiseSummaryStatsInput/ForColoc/", Locus, ".txt.gz")) %>%
      mutate(PC=str_replace(source_file, "QTLs/QTLTools/(.+?)/NominalPassForColocChunks/.+$", "\\1")) %>%
      dplyr::select(Locus = gwas_locus, P=phenotype, PC, p, snp  ) %>%
      inner_join(
        dat.ToBoxplot %>% distinct(PC, P, FacetGroup, Locus)
      )
    
      Coloc.Scatter.P <- SummaryStats %>%
        dplyr::select(-P, -PC) %>%
        pivot_wider(names_from = "FacetGroup", values_from="p") %>%
        drop_na() %>%
        gather(key="molQTL", value="molQTL_p", -Locus, -snp, -eQTL) %>%
        ggplot(aes(x=-log10(molQTL_p), y=-log10(eQTL))) +
        geom_point(alpha=0.5, color='gray', shape=19) +
        geom_rect(xmin=-Inf, xmax=-Inf, ymin=-Inf, ymax=Inf, color = '#ff7f00', size=2) +
        geom_rect(data = . %>%
                     distinct(molQTL, .keep_all=T), ymin = -Inf, ymax=-Inf, xmin=-Inf, xmax=Inf, size=3, aes(color=molQTL)) +
        facet_wrap(~molQTL, scales="free") +
        scale_color_manual(values=c("hQTL"="#6a3d9a", "sQTL"="#1f78b4")) +
        theme(axis.line=element_blank(), legend.position='none') +
        theme(
          strip.background = element_blank(),
          strip.text.x = element_blank()
        )
    ggsave(paste0("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/ColocExamples/", Locus, ".scatter.pdf"), Coloc.Scatter.P, height=3, width=5)
    }
  }
}


```




```{r, eval=F}
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_SampleSize.pdf",P.SampleSizes, width=6, height=6)

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_NumQTLs.pdf",NumQTLs, width=8, height=6)

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_PercentAnnotated.pdf",P.PercentAnnotated, width=4, height=6)
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_PercentOfUnannotated.pdf",P.PercentOfUnannotated, width=3, height=6)
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_PercentAnnotatedLegend.pdf",P.PercentJuncsLegend, width=8, height=6)

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_PercentAnnotatedPerSample.pdf",P.PercentAnnotated.PerSample, width=4, height=6)




ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_Multiple_hQTLs_PerGene.pdf",P.Number_hQTLsForColoc, width=6, height=6)
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_Multiple_sQTLs_PerGene.pdf",P.Number_sQTLsForColoc, width=6, height=6)

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_NonPrimaryMolQTLsColoc.pdf",P.NonPrimaryMolQTLs_OftenColoc, width=6, height=6)

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_EnrichmentOfUnrpoductiveSqtls.pdf",P.EnrichmentOfUnrpductiveClusters, width=6, height=6)

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_EnrichmentOfUnrpoductiveSqtls_AmongstTop.pdf",P.EnrichmentOfUnrpductiveClusters_AmongstTop, width=6, height=6)


ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_eGene_MolQTL_Colocs.pdf",P.eGeneColocsWith, width=10, height=7)
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_eGene_MolQTL_NonhQTL_Colocs.pdf",P.eGeneColocsWith.nohQTL, width=10, height=7)


ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_FinalFig_S_eGene_MolQTL_Simplified.pdf",P.SimplifiedNumberColocs, width=2.7, height=2.7)

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/20230412_WIP_eQTLQQ.pdf",eQTL.QQ, height=5, width=7)

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_Final_Fig2_sQTLeQTLBetas.pdf",sQTL.eQTL.beta.beta, height=4, width=10)

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/20240506_eQTL_sQTL_UnnormalizedUnitsScatter.pdf",sQTL.eQTL.beta.beta.unnormalized, height=4, width=10)
ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/20240506_eQTL_sQTL_UnnormalizedUnitsScatterOrthogonalRegression.pdf",sQTL.eQTL.beta.beta.unnormalized.PlusDeming, height=4, width=10)

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/20240506_eQTL_sQTL_UnnormalizedUnitsScatterNoLines.pdf",sQTL.eQTL.beta.beta.unnormalized.nolines, height=4, width=10)

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_Final_Fig2_hQTLeQTLBetas.pdf",eQTL.H3k27ac.beta, height=2.5, width=10)


ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_Final_Fig_S_ScatterReasonsForJuncClassification.pdf",Scatter.P, height=12, width=10)

ggsave("/project2/yangili1/carlos_and_ben_shared/rough_figs/OriginalSubplots/2023_Final_Fig_S_ScatterReasonsForJuncClassification.png",Scatter.P, height=12, width=10)

```

